// @tempi/core-elements/dist/components/form/v4/constants.es.js
import { t } from '../../../i18n.es.js';

const DEFAULT_IMAGE_CONFIG = {
  width: 100,
  radius: 8,
  backgroundSize: "cover"
};
const DEFAULT_POPUP_SUCCESS = (popupName) => ({
  a7V6wzgv00: {
    id: "a7V6wzgv00",
    parentId: "ROOT",
    tag: "popupv1",
    resolvedName: "PopupEditorV1",
    style: {
      width: 400,
      height: 200,
      backgroundColor: "#fff",
      padding: {
        left: 16,
        right: 16,
        bottom: 16,
        top: 16
      },
      backgroundSize: "cover",
      borderRadius: 12
    },
    isActive: false,
    isCanvas: true,
    customAttributes: {
      popupv1: {
        maskClosable: true,
        maskStyle: {
          backgroundColor: "#0000004a"
        },
        position: "default",
        name: popupName
      }
    },
    children: ["BFbkQmE9WI", "RlaZkRZUBx", "TdeG0UMb4y"]
  },
  BFbkQmE9WI: {
    id: "BFbkQmE9WI",
    parentId: "a7V6wzgv00",
    tag: "heading",
    resolvedName: "HeadingEditor",
    style: {
      backgroundSize: "cover"
    },
    isActive: false,
    isCanvas: false,
    customAttributes: {
      heading: {
        text: `<h2 style="text-align: center">${t("Th\xF4ng b\xE1o")}</h2>`
      }
    },
    children: []
  },
  RlaZkRZUBx: {
    id: "RlaZkRZUBx",
    parentId: "a7V6wzgv00",
    tag: "paragraph",
    resolvedName: "ParagraphEditor",
    style: {
      backgroundSize: "cover",
      margin: {
        top: 16
      }
    },
    isActive: false,
    isCanvas: false,
    customAttributes: {
      paragraph: {
        text: '<p><span style="font-size: 14px">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut elit tellus</span></p>'
      }
    },
    children: []
  },
  TdeG0UMb4y: {
    id: "TdeG0UMb4y",
    parentId: "a7V6wzgv00",
    tag: "buttonv2",
    resolvedName: "ButtonEditorV2",
    style: {
      justifyContent: "center",
      margin: {
        top: 16
      }
    },
    isActive: false,
    isCanvas: false,
    customAttributes: {
      buttonv2: {
        text: `<p>${t("X\xE1c nh\u1EADn")}</p>`,
        buttonStyle: {
          backgroundColor: "rgba(37, 141, 255, 1)",
          color: "rgba(255, 255, 255, 1)",
          height: 32,
          width: 120,
          borderRadius: 999,
          backgroundSize: "cover"
        },
        iconConfig: {
          position: "left",
          spacing: 8,
          show: true
        }
      }
    },
    children: []
  }
});
const DEFAULT_FORM = {
  formStyle: {
    flexDirection: "column",
    gap: 8
  },
  inputStyle: {
    borderRadius: 4
  },
  showLabel: true,
  submits: [],
  hiddenFields: [],
  response: {
    custom: []
  },
  inputs: [
    {
      type: "text",
      label: t("H\u1ECD t\xEAn"),
      required: true,
      name: "name",
      commonStyle: { height: 32, width: "100%" }
    },
    {
      type: "text",
      label: t("N\u1ED9i dung"),
      name: "content",
      commonStyle: { height: 32, width: "100%" }
    }
  ]
};

export { DEFAULT_FORM, DEFAULT_IMAGE_CONFIG, DEFAULT_POPUP_SUCCESS };
