// @tempi/core-elements/dist/components/form/v4/selector.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var coreEditor = require('@tempi/core-editor');
var React = require('react');
var constants = require('./constants.js');
var quickSetting = require('./quick-setting.js');
var setting = require('./setting.js');
var styleSetting = require('./style-setting.js');
var ui = require('./ui.js');
var propKey = require('../../../constants/propKey.js');
var selector = require('../../section/selector.js');
var selector$1 = require('../../root/selector.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const FormV4Editor = coreEditor.withSelector(
  (props) => {
    return /* @__PURE__ */ jsxRuntime.jsx(ui.FormV4, __spreadProps(__spreadValues({
      style: { pointerEvents: "none" }
    }, props), {
      isEditingMode: true
    }));
  },
  {
    displayName: "Form",
    tag: ui.FormV4.displayName.toLowerCase(),
    customAttributes: setting.FormSetting,
    commonAttributes: styleSetting.FormStyleSetting,
    advanceAttributes: () => React__default["default"].createElement(coreEditor.AdvanceAttributes, {
      propKey: propKey.PROP_KEY.FormStyle
    }),
    quickSetting: quickSetting.FormQuickSetting,
    props: constants.DEFAULT_FORM,
    rules: {
      canMoveIn: () => false,
      canDrop: (targetNode) => {
        const ignoreTargets = [
          coreEditor.getEditorName({ RowEditor: selector.RowEditor }),
          coreEditor.getEditorName({ RootEditor: selector$1.RootEditor })
        ];
        return !ignoreTargets.includes(targetNode.data.name);
      }
    }
  },
  {
    style: {
      cursor: "pointer"
    }
  }
);

exports.FormV4Editor = FormV4Editor;
