// @tempi/core-elements/dist/components/form/v4/selector.es.js
import { jsx } from 'react/jsx-runtime';
import { withSelector, AdvanceAttributes, getEditorName } from '@tempi/core-editor';
import React__default from 'react';
import { DEFAULT_FORM } from './constants.es.js';
import { FormQuickSetting } from './quick-setting.es.js';
import { FormSetting } from './setting.es.js';
import { FormStyleSetting } from './style-setting.es.js';
import { FormV4 } from './ui.es.js';
import { PROP_KEY } from '../../../constants/propKey.es.js';
import { RowEditor } from '../../section/selector.es.js';
import { RootEditor } from '../../root/selector.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const FormV4Editor = withSelector(
  (props) => {
    return /* @__PURE__ */ jsx(FormV4, __spreadProps(__spreadValues({
      style: { pointerEvents: "none" }
    }, props), {
      isEditingMode: true
    }));
  },
  {
    displayName: "Form",
    tag: FormV4.displayName.toLowerCase(),
    customAttributes: FormSetting,
    commonAttributes: FormStyleSetting,
    advanceAttributes: () => React__default.createElement(AdvanceAttributes, {
      propKey: PROP_KEY.FormStyle
    }),
    quickSetting: FormQuickSetting,
    props: DEFAULT_FORM,
    rules: {
      canMoveIn: () => false,
      canDrop: (targetNode) => {
        const ignoreTargets = [
          getEditorName({ RowEditor }),
          getEditorName({ RootEditor })
        ];
        return !ignoreTargets.includes(targetNode.data.name);
      }
    }
  },
  {
    style: {
      cursor: "pointer"
    }
  }
);

export { FormV4Editor };
