// @tempi/core-elements/dist/components/form/v4/setting.es.js
import { jsx, Fragment, jsxs } from 'react/jsx-runtime';
import { t } from '../../../i18n.es.js';
import { useProp, useBuilder, TrackingConfig } from '@tempi/core-editor';
import { Collapse } from 'antd';
import { DEFAULT_LEAD_FORM } from '../../lead-form/constants.es.js';
import { LeadStorageDataConfig } from '../../lead-form/LeadStorageDataConfig.es.js';
import { useStorageData } from '../../../configs/storage-v2/hook/useStorageData.es.js';
import { FormFieldSettingCustom } from './configs/FormFieldSettingCustom.es.js';
import { FormFieldSetting } from './configs/FormFieldSetting.es.js';
import { ResponseConfig } from './configs/ResponseConfig.es.js';
import { CaptchaConfig } from './configs/CaptchaConfig.es.js';
import { LayoutSetting } from './configs/LayoutSetting.es.js';
import { PROP_KEY } from '../../../constants/propKey.es.js';
import { StorageDataConfig } from '../../../configs/storage-v2/StorageDataConfig.es.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
const { Panel } = Collapse;
const FormSetting = ({
  formType = "default"
}) => {
  const [hiddenFields] = useProp(PROP_KEY.HiddenFields);
  const [submits] = useProp(PROP_KEY.Submits);
  const isOldVersion = (hiddenFields == null ? void 0 : hiddenFields.length) || (submits == null ? void 0 : submits.length);
  const { isPrebuildEditor } = useBuilder();
  const {
    handleFetchFormAccount,
    loading,
    configStorage,
    handleSaveDataConfig,
    ggSheetsAccounts,
    allowConfigResponseByCode
  } = useStorageData();
  const renderSettingOldVersion = () => {
    return /* @__PURE__ */ jsx(Fragment, {
      children: t("Vui l\xF2ng k\xE9o th\u1EA3 Form \u0111\u1EC3 c\u1EADp nh\u1EADt t\xEDnh n\u0103ng m\u1EDBi")
    });
  };
  const renderStorageData = (formType2) => {
    if (isPrebuildEditor) {
      return /* @__PURE__ */ jsx(Fragment, {});
    }
    if (formType2 === "leadForm") {
      return /* @__PURE__ */ jsx(Panel, {
        header: t("C\u1EA5u h\xECnh l\u01B0u d\u1EEF li\u1EC7u"),
        children: /* @__PURE__ */ jsx(LeadStorageDataConfig, __spreadValues({}, {
          handleFetchFormAccount,
          loading,
          configStorage,
          handleSaveDataConfig,
          ggSheetsAccounts
        }))
      }, "DATA");
    }
    return /* @__PURE__ */ jsx(Panel, {
      header: t("C\u1EA5u h\xECnh l\u01B0u d\u1EEF li\u1EC7u"),
      children: /* @__PURE__ */ jsx(StorageDataConfig, __spreadValues({}, {
        handleFetchFormAccount,
        loading,
        configStorage,
        handleSaveDataConfig,
        ggSheetsAccounts
      }))
    }, "DATA");
  };
  const renderSetting = (formType2) => {
    return /* @__PURE__ */ jsxs(Fragment, {
      children: [
        renderStorageData(formType2),
        /* @__PURE__ */ jsx(Panel, {
          header: t("C\u1EA5u h\xECnh danh s\xE1ch tr\u01B0\u1EDDng"),
          children: formType2 === "leadForm" ? /* @__PURE__ */ jsx(FormFieldSettingCustom, {
            formFieldOptions: DEFAULT_LEAD_FORM.inputs
          }) : /* @__PURE__ */ jsx(FormFieldSetting, {})
        }, "FORM_FIELDS"),
        /* @__PURE__ */ jsx(Panel, {
          header: t("C\u1EA5u h\xECnh ph\u1EA3n h\u1ED3i"),
          children: /* @__PURE__ */ jsx(ResponseConfig, __spreadValues({}, { allowConfigResponseByCode }))
        }, "RESPONSE"),
        /* @__PURE__ */ jsx(Panel, {
          header: t("C\u1EA5u h\xECnh l\u01B0u Captcha"),
          children: /* @__PURE__ */ jsx(CaptchaConfig, {})
        }, "CAPTCHA"),
        /* @__PURE__ */ jsx(Panel, {
          header: t("C\u1EA5u h\xECnh l\u01B0u Tracking"),
          children: /* @__PURE__ */ jsx(TrackingConfig, {
            config: "form",
            layout: "vertical"
          })
        }, "TRACKING"),
        formType2 === "subscriptionForm" && /* @__PURE__ */ jsx(Panel, {
          header: t("C\u1EA5u h\xECnh b\u1ED1 c\u1EE5c"),
          children: /* @__PURE__ */ jsx(LayoutSetting, {})
        }, "LAYOUT")
      ]
    });
  };
  return /* @__PURE__ */ jsx(Collapse, {
    expandIconPosition: "end",
    bordered: false,
    defaultActiveKey: isPrebuildEditor ? ["DATA"] : ["FORM_FIELDS"],
    children: isOldVersion ? renderSettingOldVersion() : renderSetting(formType)
  });
};

export { FormSetting };
