// @tempi/core-elements/dist/components/form/v4/setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var antd = require('antd');
var constants = require('../../lead-form/constants.js');
var LeadStorageDataConfig = require('../../lead-form/LeadStorageDataConfig.js');
var useStorageData = require('../../../configs/storage-v2/hook/useStorageData.js');
var FormFieldSettingCustom = require('./configs/FormFieldSettingCustom.js');
var FormFieldSetting = require('./configs/FormFieldSetting.js');
var ResponseConfig = require('./configs/ResponseConfig.js');
var CaptchaConfig = require('./configs/CaptchaConfig.js');
var LayoutSetting = require('./configs/LayoutSetting.js');
var propKey = require('../../../constants/propKey.js');
var StorageDataConfig = require('../../../configs/storage-v2/StorageDataConfig.js');

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
const { Panel } = antd.Collapse;
const FormSetting = ({
  formType = "default"
}) => {
  const [hiddenFields] = coreEditor.useProp(propKey.PROP_KEY.HiddenFields);
  const [submits] = coreEditor.useProp(propKey.PROP_KEY.Submits);
  const isOldVersion = (hiddenFields == null ? void 0 : hiddenFields.length) || (submits == null ? void 0 : submits.length);
  const { isPrebuildEditor } = coreEditor.useBuilder();
  const {
    handleFetchFormAccount,
    loading,
    configStorage,
    handleSaveDataConfig,
    ggSheetsAccounts,
    allowConfigResponseByCode
  } = useStorageData.useStorageData();
  const renderSettingOldVersion = () => {
    return /* @__PURE__ */ jsxRuntime.jsx(jsxRuntime.Fragment, {
      children: i18n.t("Vui l\xF2ng k\xE9o th\u1EA3 Form \u0111\u1EC3 c\u1EADp nh\u1EADt t\xEDnh n\u0103ng m\u1EDBi")
    });
  };
  const renderStorageData = (formType2) => {
    if (isPrebuildEditor) {
      return /* @__PURE__ */ jsxRuntime.jsx(jsxRuntime.Fragment, {});
    }
    if (formType2 === "leadForm") {
      return /* @__PURE__ */ jsxRuntime.jsx(Panel, {
        header: i18n.t("C\u1EA5u h\xECnh l\u01B0u d\u1EEF li\u1EC7u"),
        children: /* @__PURE__ */ jsxRuntime.jsx(LeadStorageDataConfig.LeadStorageDataConfig, __spreadValues({}, {
          handleFetchFormAccount,
          loading,
          configStorage,
          handleSaveDataConfig,
          ggSheetsAccounts
        }))
      }, "DATA");
    }
    return /* @__PURE__ */ jsxRuntime.jsx(Panel, {
      header: i18n.t("C\u1EA5u h\xECnh l\u01B0u d\u1EEF li\u1EC7u"),
      children: /* @__PURE__ */ jsxRuntime.jsx(StorageDataConfig.StorageDataConfig, __spreadValues({}, {
        handleFetchFormAccount,
        loading,
        configStorage,
        handleSaveDataConfig,
        ggSheetsAccounts
      }))
    }, "DATA");
  };
  const renderSetting = (formType2) => {
    return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
      children: [
        renderStorageData(formType2),
        /* @__PURE__ */ jsxRuntime.jsx(Panel, {
          header: i18n.t("C\u1EA5u h\xECnh danh s\xE1ch tr\u01B0\u1EDDng"),
          children: formType2 === "leadForm" ? /* @__PURE__ */ jsxRuntime.jsx(FormFieldSettingCustom.FormFieldSettingCustom, {
            formFieldOptions: constants.DEFAULT_LEAD_FORM.inputs
          }) : /* @__PURE__ */ jsxRuntime.jsx(FormFieldSetting.FormFieldSetting, {})
        }, "FORM_FIELDS"),
        /* @__PURE__ */ jsxRuntime.jsx(Panel, {
          header: i18n.t("C\u1EA5u h\xECnh ph\u1EA3n h\u1ED3i"),
          children: /* @__PURE__ */ jsxRuntime.jsx(ResponseConfig.ResponseConfig, __spreadValues({}, { allowConfigResponseByCode }))
        }, "RESPONSE"),
        /* @__PURE__ */ jsxRuntime.jsx(Panel, {
          header: i18n.t("C\u1EA5u h\xECnh l\u01B0u Captcha"),
          children: /* @__PURE__ */ jsxRuntime.jsx(CaptchaConfig.CaptchaConfig, {})
        }, "CAPTCHA"),
        /* @__PURE__ */ jsxRuntime.jsx(Panel, {
          header: i18n.t("C\u1EA5u h\xECnh l\u01B0u Tracking"),
          children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.TrackingConfig, {
            config: "form",
            layout: "vertical"
          })
        }, "TRACKING"),
        formType2 === "subscriptionForm" && /* @__PURE__ */ jsxRuntime.jsx(Panel, {
          header: i18n.t("C\u1EA5u h\xECnh b\u1ED1 c\u1EE5c"),
          children: /* @__PURE__ */ jsxRuntime.jsx(LayoutSetting.LayoutSetting, {})
        }, "LAYOUT")
      ]
    });
  };
  return /* @__PURE__ */ jsxRuntime.jsx(antd.Collapse, {
    expandIconPosition: "end",
    bordered: false,
    defaultActiveKey: isPrebuildEditor ? ["DATA"] : ["FORM_FIELDS"],
    children: isOldVersion ? renderSettingOldVersion() : renderSetting(formType)
  });
};

exports.FormSetting = FormSetting;
