// @tempi/core-elements/dist/components/form/v4/style-setting.es.js
import { jsx, Fragment, jsxs } from 'react/jsx-runtime';
import { t } from '../../../i18n.es.js';
import { useProp, ColorPickerConfig, BackgroundImageConfig, SelectConfig, AlignmentConfig, InputNumberConfig } from '@tempi/core-editor';
import { UnitEnum } from '@tempi/core-renderer';
import { Collapse } from 'antd';
import { useEffect } from 'react';
import { PROP_KEY } from '../../../constants/propKey.es.js';
import { TextConfig } from '../../../configs/TextConfig/TextConfig.es.js';

const { Panel } = Collapse;
const FormStyleSetting = ({
  isSubscriptionForm = false
}) => {
  const [, setAlignItems] = useProp(
    PROP_KEY.joinKey(PROP_KEY.FormStyle, PROP_KEY.AlignItems)
  );
  const [flexDirection] = useProp(
    PROP_KEY.joinKey(PROP_KEY.FormStyle, PROP_KEY.FlexDirection)
  );
  const [, setJustifyContent] = useProp(
    PROP_KEY.joinKey(PROP_KEY.FormStyle, PROP_KEY.JustifyContent)
  );
  const flexDirectionOptions = [
    {
      label: t("common.horizontal"),
      value: "row"
    },
    {
      label: t("common.vertical"),
      value: "column"
    }
  ];
  useEffect(() => {
    if (flexDirection === "column") {
      setJustifyContent(void 0);
    }
    if (flexDirection === "row") {
      !isSubscriptionForm && setAlignItems(void 0);
    }
  }, [flexDirection]);
  return /* @__PURE__ */ jsx(Fragment, {
    children: /* @__PURE__ */ jsxs(Collapse, {
      expandIconPosition: "end",
      bordered: false,
      children: [
        /* @__PURE__ */ jsx(Collapse.Panel, {
          header: t("M\xE0u n\u1EC1n"),
          children: /* @__PURE__ */ jsx(ColorPickerConfig, {
            propKey: PROP_KEY.joinKey(
              PROP_KEY.FormStyle,
              PROP_KEY.BackgroundColor
            ),
            formItemProps: {
              label: t("M\xE0u n\u1EC1n")
            },
            gradientPicker: true
          })
        }, "BACKGROUND_COLOR"),
        /* @__PURE__ */ jsx(Collapse.Panel, {
          header: t("\u1EA2nh n\u1EC1n"),
          children: /* @__PURE__ */ jsx(BackgroundImageConfig, {
            propKey: PROP_KEY.FormStyle
          })
        }, "BACKGROUND_IMAGE"),
        /* @__PURE__ */ jsxs(Panel, {
          header: "Form",
          children: [
            /* @__PURE__ */ jsx(SelectConfig, {
              formItemProps: {
                label: t("Ki\u1EC3u hi\u1EC3n th\u1ECB")
              },
              options: flexDirectionOptions,
              propKey: PROP_KEY.joinKey(
                PROP_KEY.FormStyle,
                PROP_KEY.FlexDirection
              )
            }),
            flexDirection === "column" && /* @__PURE__ */ jsx(AlignmentConfig, {
              propKey: PROP_KEY.joinKey(
                PROP_KEY.FormStyle,
                PROP_KEY.AlignItems
              ),
              buttonGroupProps: {
                listAlignment: ["flex-start", "center", "flex-end"]
              },
              formItemProps: {
                label: t("C\u0103n ch\u1EC9nh")
              }
            }),
            flexDirection === "row" && /* @__PURE__ */ jsx(AlignmentConfig, {
              propKey: PROP_KEY.joinKey(
                PROP_KEY.FormStyle,
                PROP_KEY.JustifyContent
              ),
              buttonGroupProps: {
                listAlignment: ["flex-start", "center", "flex-end"]
              },
              formItemProps: {
                label: t("C\u0103n ch\u1EC9nh")
              }
            }),
            /* @__PURE__ */ jsx(InputNumberConfig, {
              formItemProps: {
                label: t("Kho\u1EA3ng c\xE1ch h\xE0ng")
              },
              unitProps: { defaultValue: UnitEnum.pixel },
              propKey: PROP_KEY.joinKey(PROP_KEY.FormStyle, PROP_KEY.RowGap)
            }),
            /* @__PURE__ */ jsx(ColorPickerConfig, {
              propKey: PROP_KEY.joinKey(PROP_KEY.LabelStyle, PROP_KEY.Color),
              formItemProps: {
                label: t("M\xE0u nh\xE3n")
              }
            }),
            /* @__PURE__ */ jsx(TextConfig, {
              propKey: PROP_KEY.LabelStyle,
              title: t("Ch\u1EEF nh\xE3n"),
              config: { fontSize: true }
            })
          ]
        }, "FORM"),
        /* @__PURE__ */ jsxs(Panel, {
          header: "Field",
          children: [
            /* @__PURE__ */ jsx(ColorPickerConfig, {
              propKey: PROP_KEY.joinKey(PROP_KEY.InputStyle, PROP_KEY.Color),
              formItemProps: {
                label: t("M\xE0u ch\u1EEF input")
              }
            }),
            /* @__PURE__ */ jsx(ColorPickerConfig, {
              propKey: PROP_KEY.joinKey(PROP_KEY.OptionStyle, PROP_KEY.Color),
              formItemProps: {
                label: t("M\xE0u l\u1EF1a ch\u1ECDn")
              }
            }),
            /* @__PURE__ */ jsx(TextConfig, {
              propKey: PROP_KEY.InputStyle,
              title: t("Ch\u1EEF input"),
              config: { fontSize: true }
            }),
            /* @__PURE__ */ jsx(ColorPickerConfig, {
              propKey: PROP_KEY.joinKey(
                PROP_KEY.InputStyle,
                PROP_KEY.BorderColor
              ),
              formItemProps: {
                label: t("M\xE0u vi\u1EC1n")
              }
            }),
            /* @__PURE__ */ jsx(InputNumberConfig, {
              formItemProps: {
                label: t("K\xEDch th\u01B0\u1EDBc vi\u1EC1n")
              },
              unitProps: { defaultValue: UnitEnum.pixel },
              propKey: PROP_KEY.joinKey(
                PROP_KEY.InputStyle,
                PROP_KEY.BorderWidth
              )
            }),
            /* @__PURE__ */ jsx(InputNumberConfig, {
              formItemProps: {
                label: t("Bo g\xF3c")
              },
              unitProps: { defaultValue: UnitEnum.pixel },
              propKey: PROP_KEY.joinKey(
                PROP_KEY.InputStyle,
                PROP_KEY.BorderRadius
              )
            }),
            /* @__PURE__ */ jsx(ColorPickerConfig, {
              propKey: PROP_KEY.joinKey(PROP_KEY.ErrorStyle, PROP_KEY.Color),
              formItemProps: {
                label: t("M\xE0u th\xF4ng b\xE1o l\u1ED7i")
              }
            })
          ]
        }, "FIELD")
      ]
    })
  });
};

export { FormStyleSetting };
