// @tempi/core-elements/dist/components/product-list/selector.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var coreEditor = require('@tempi/core-editor');
var constants = require('./constants.js');
var quickSetting = require('./quick-setting.js');
var setting = require('./setting.js');
var styleSetting = require('./style-setting.js');
var ui = require('./ui.js');
var selector = require('../section/selector.js');
var selector$1 = require('../root/selector.js');

const TempiProductListEditor = coreEditor.withSelector(ui.TempiProductList, {
  displayName: "Danh s\xE1ch s\u1EA3n ph\u1EA9m",
  tag: ui.TempiProductList.displayName.toLowerCase(),
  advanceAttributes: coreEditor.AdvanceAttributes,
  customAttributes: setting.Setting,
  commonAttributes: styleSetting.StyleSetting,
  quickSetting: quickSetting.QuickSetting,
  rules: {
    canDrop: (targetNode) => {
      const ignoreTargets = [
        coreEditor.getEditorName({ RowEditor: selector.RowEditor }),
        coreEditor.getEditorName({ RootEditor: selector$1.RootEditor })
      ];
      return !ignoreTargets.includes(targetNode.data.name);
    }
  },
  props: constants.DEFAULT_PRODUCT_LIST
});

exports.TempiProductListEditor = TempiProductListEditor;
