// @tempi/core-elements/dist/components/product-list/selector.es.js
import { withSelector, AdvanceAttributes, getEditorName } from '@tempi/core-editor';
import { DEFAULT_PRODUCT_LIST } from './constants.es.js';
import { QuickSetting } from './quick-setting.es.js';
import { Setting } from './setting.es.js';
import { StyleSetting } from './style-setting.es.js';
import { TempiProductList } from './ui.es.js';
import { RowEditor } from '../section/selector.es.js';
import { RootEditor } from '../root/selector.es.js';

const TempiProductListEditor = withSelector(TempiProductList, {
  displayName: "Danh s\xE1ch s\u1EA3n ph\u1EA9m",
  tag: TempiProductList.displayName.toLowerCase(),
  advanceAttributes: AdvanceAttributes,
  customAttributes: Setting,
  commonAttributes: StyleSetting,
  quickSetting: QuickSetting,
  rules: {
    canDrop: (targetNode) => {
      const ignoreTargets = [
        getEditorName({ RowEditor }),
        getEditorName({ RootEditor })
      ];
      return !ignoreTargets.includes(targetNode.data.name);
    }
  },
  props: DEFAULT_PRODUCT_LIST
});

export { TempiProductListEditor };
