// @tempi/core-elements/dist/components/product-list/style-setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var propKey = require('../../constants/propKey.js');

const StyleSetting = () => {
  const [showButton] = coreEditor.useProp(propKey.PROP_KEY.ShowButton);
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
        formItemProps: { label: i18n.t("M\xE0u t\xEAn s\u1EA3n ph\u1EA9m") },
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.NameStyle, propKey.PROP_KEY.Color)
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.RangeSliderConfig, {
        formItemProps: {
          label: i18n.t("C\u1EE1 ch\u1EEF t\xEAn s\u1EA3n ph\u1EA9m")
        },
        sliderProps: {
          min: 12,
          max: 20,
          defaultValue: 16
        },
        unitProps: {
          defaultValue: "px"
        },
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.NameStyle, propKey.PROP_KEY.FontSize)
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
        formItemProps: { label: i18n.t("M\xE0u gi\xE1 s\u1EA3n ph\u1EA9m") },
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.PriceStyle, propKey.PROP_KEY.Color)
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.RangeSliderConfig, {
        formItemProps: {
          label: i18n.t("C\u1EE1 ch\u1EEF gi\xE1 s\u1EA3n ph\u1EA9m")
        },
        sliderProps: {
          min: 10,
          max: 20,
          defaultValue: 14
        },
        unitProps: {
          defaultValue: "px"
        },
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.PriceStyle, propKey.PROP_KEY.FontSize)
      }),
      showButton && /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
            formItemProps: { label: i18n.t("M\xE0u t\xEAn n\xFAt b\u1EA5m") },
            propKey: propKey.PROP_KEY.joinKey(
              propKey.PROP_KEY.OrderButton,
              propKey.PROP_KEY.ButtonStyle,
              propKey.PROP_KEY.Color
            )
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
            formItemProps: { label: i18n.t("M\xE0u n\u1EC1n n\xFAt b\u1EA5m") },
            propKey: propKey.PROP_KEY.joinKey(
              propKey.PROP_KEY.OrderButton,
              propKey.PROP_KEY.ButtonStyle,
              propKey.PROP_KEY.BackgroundColor
            )
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
            formItemProps: { label: i18n.t("M\xE0u khi di chu\u1ED9t v\xE0o n\xFAt b\u1EA5m") },
            propKey: propKey.PROP_KEY.joinKey(
              propKey.PROP_KEY.OrderButton,
              propKey.PROP_KEY.ButtonStyle,
              propKey.PROP_KEY.HoverBackgroundColor
            )
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.RangeSliderConfig, {
            formItemProps: {
              label: i18n.t("C\u1EE1 ch\u1EEF t\xEAn n\xFAt b\u1EA5m")
            },
            sliderProps: {
              min: 10,
              max: 20
            },
            unitProps: {
              defaultValue: "px"
            },
            propKey: propKey.PROP_KEY.joinKey(
              propKey.PROP_KEY.OrderButton,
              propKey.PROP_KEY.ButtonStyle,
              propKey.PROP_KEY.FontSize
            )
          })
        ]
      })
    ]
  });
};

exports.StyleSetting = StyleSetting;
