// @tempi/core-elements/dist/components/product-list/style-setting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { useProp, ColorPickerConfig, RangeSliderConfig } from '@tempi/core-editor';
import { PROP_KEY } from '../../constants/propKey.es.js';

const StyleSetting = () => {
  const [showButton] = useProp(PROP_KEY.ShowButton);
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(ColorPickerConfig, {
        formItemProps: { label: t("M\xE0u t\xEAn s\u1EA3n ph\u1EA9m") },
        propKey: PROP_KEY.joinKey(PROP_KEY.NameStyle, PROP_KEY.Color)
      }),
      /* @__PURE__ */ jsx(RangeSliderConfig, {
        formItemProps: {
          label: t("C\u1EE1 ch\u1EEF t\xEAn s\u1EA3n ph\u1EA9m")
        },
        sliderProps: {
          min: 12,
          max: 20,
          defaultValue: 16
        },
        unitProps: {
          defaultValue: "px"
        },
        propKey: PROP_KEY.joinKey(PROP_KEY.NameStyle, PROP_KEY.FontSize)
      }),
      /* @__PURE__ */ jsx(ColorPickerConfig, {
        formItemProps: { label: t("M\xE0u gi\xE1 s\u1EA3n ph\u1EA9m") },
        propKey: PROP_KEY.joinKey(PROP_KEY.PriceStyle, PROP_KEY.Color)
      }),
      /* @__PURE__ */ jsx(RangeSliderConfig, {
        formItemProps: {
          label: t("C\u1EE1 ch\u1EEF gi\xE1 s\u1EA3n ph\u1EA9m")
        },
        sliderProps: {
          min: 10,
          max: 20,
          defaultValue: 14
        },
        unitProps: {
          defaultValue: "px"
        },
        propKey: PROP_KEY.joinKey(PROP_KEY.PriceStyle, PROP_KEY.FontSize)
      }),
      showButton && /* @__PURE__ */ jsxs(Fragment, {
        children: [
          /* @__PURE__ */ jsx(ColorPickerConfig, {
            formItemProps: { label: t("M\xE0u t\xEAn n\xFAt b\u1EA5m") },
            propKey: PROP_KEY.joinKey(
              PROP_KEY.OrderButton,
              PROP_KEY.ButtonStyle,
              PROP_KEY.Color
            )
          }),
          /* @__PURE__ */ jsx(ColorPickerConfig, {
            formItemProps: { label: t("M\xE0u n\u1EC1n n\xFAt b\u1EA5m") },
            propKey: PROP_KEY.joinKey(
              PROP_KEY.OrderButton,
              PROP_KEY.ButtonStyle,
              PROP_KEY.BackgroundColor
            )
          }),
          /* @__PURE__ */ jsx(ColorPickerConfig, {
            formItemProps: { label: t("M\xE0u khi di chu\u1ED9t v\xE0o n\xFAt b\u1EA5m") },
            propKey: PROP_KEY.joinKey(
              PROP_KEY.OrderButton,
              PROP_KEY.ButtonStyle,
              PROP_KEY.HoverBackgroundColor
            )
          }),
          /* @__PURE__ */ jsx(RangeSliderConfig, {
            formItemProps: {
              label: t("C\u1EE1 ch\u1EEF t\xEAn n\xFAt b\u1EA5m")
            },
            sliderProps: {
              min: 10,
              max: 20
            },
            unitProps: {
              defaultValue: "px"
            },
            propKey: PROP_KEY.joinKey(
              PROP_KEY.OrderButton,
              PROP_KEY.ButtonStyle,
              PROP_KEY.FontSize
            )
          })
        ]
      })
    ]
  });
};

export { StyleSetting };
