// @tempi/core-elements/dist/components/product-list/ui.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { defaultCarouselOptions, defaultGridOptions } from './constants.es.js';
import { useCheckout } from './hooks.es.js';
import { CheckoutPopup } from './components/CheckoutPopup/CheckoutPopup.es.js';
import { CheckoutSuccessPopup } from './components/CheckoutSuccessPopup/CheckoutSuccessPopup.es.js';
import { ProductCarousel } from './components/ProductCarousel/ProductCarousel.es.js';
import { ProductGrid } from './components/ProductGrid/ProductGrid.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const TempiProductList = (_a) => {
  var _b = _a, {
    isEditingMode,
    configStorage
  } = _b, restProps = __objRest(_b, [
    "isEditingMode",
    "configStorage"
  ]);
  const {
    selectedProduct,
    setSelectedProduct,
    loading,
    doneCheckout,
    setDoneCheckout,
    handleFinishCheckout
  } = useCheckout(configStorage);
  const handleClickProduct = (restProps == null ? void 0 : restProps.eventType) === "showPopup" ? setSelectedProduct : void 0;
  const renderContent = () => {
    if ((restProps == null ? void 0 : restProps.layout) === "carousel")
      return /* @__PURE__ */ jsx(ProductCarousel, __spreadProps(__spreadValues({
        isEditingMode
      }, restProps), {
        carouselOptions: __spreadValues(__spreadValues({}, defaultCarouselOptions), restProps == null ? void 0 : restProps.carouselOptions),
        onClick: handleClickProduct
      }));
    return /* @__PURE__ */ jsx(ProductGrid, __spreadProps(__spreadValues({
      isEditingMode
    }, restProps), {
      gridOptions: __spreadValues(__spreadValues({}, defaultGridOptions), restProps == null ? void 0 : restProps.gridOptions),
      onClick: handleClickProduct
    }));
  };
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      renderContent(),
      /* @__PURE__ */ jsx(CheckoutPopup, {
        open: !!selectedProduct,
        setOpen: () => setSelectedProduct(void 0),
        loading,
        onFinish: handleFinishCheckout
      }),
      /* @__PURE__ */ jsx(CheckoutSuccessPopup, {
        open: doneCheckout,
        setOpen: () => setDoneCheckout(false)
      })
    ]
  });
};
TempiProductList.displayName = "TempiProductList";

export { TempiProductList };
