// @tempi/core-elements/dist/components/popup/v1/quick-setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var coreRenderer = require('@tempi/core-renderer');
var styled = require('@emotion/styled');
var BackgroundQuickSetting = require('../../../quick-configs/BackgroundQuickSetting/BackgroundQuickSetting.js');
var DuplicatePopup = require('../../../quick-configs/DuplicatePopup/DuplicatePopup.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const PopupQuickSetting = ({ setShow }) => {
  const { actions } = coreEditor.useEditor();
  const { parentId, id } = coreEditor.useNode((node) => {
    return {
      parentId: node.data.parent
    };
  });
  const {
    isEditingPopup,
    relatedNodeId,
    setRelatedNodeId,
    setShowBubbleSetting
  } = coreEditor.useBuilder();
  const handleClosePopupConfig = () => {
    actions.setCustom(coreRenderer.ROOT, (custom) => custom.isEditingPopup = void 0);
    setShow(false);
    if (relatedNodeId == null ? void 0 : relatedNodeId.componentId) {
      if (relatedNodeId.componentId === coreRenderer.BUBBLE) {
        setTimeout(() => {
          setShowBubbleSetting(true);
          setRelatedNodeId((prev) => __spreadProps(__spreadValues({}, prev), {
            popupId: id,
            componentId: void 0
          }));
        }, 50);
      } else {
        actions.selectNode(relatedNodeId == null ? void 0 : relatedNodeId.componentId);
        setRelatedNodeId((prev) => __spreadProps(__spreadValues({}, prev), {
          popupId: id,
          componentId: void 0
        }));
      }
    } else {
      actions.selectNode(parentId);
    }
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(Wrapper, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(BackgroundQuickSetting.BackgroundQuickSetting, {}),
          /* @__PURE__ */ jsxRuntime.jsx(DuplicatePopup.DuplicatePopup, {}),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.OpenSettingDialog, {}),
          !isEditingPopup && /* @__PURE__ */ jsxRuntime.jsx(coreEditor.DeleteNode, {})
        ]
      }),
      !isEditingPopup && setShow && /* @__PURE__ */ jsxRuntime.jsx(coreEditor.QuickSettingButton, {
        onClick: handleClosePopupConfig,
        children: i18n.t("\u0110\xF3ng")
      })
    ]
  });
};
const Wrapper = styled__default["default"].div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding-right: 5px;
  background: #fff;
  border-radius: 999px;
  position: absolute;
  top: -50px;
  right: 0;
`;

exports.PopupQuickSetting = PopupQuickSetting;
