// @tempi/core-elements/dist/components/popup/v1/selector.es.js
import { jsxs } from 'react/jsx-runtime';
import { useNode, useProp, crafting, AdvanceAttributes, getEditorName } from '@tempi/core-editor';
import React__default, { useEffect } from 'react';
import { DEFAULT_POPUP_V1 } from './constants.es.js';
import { usePopup } from './hooks.es.js';
import { PopupQuickSetting } from './quick-setting.es.js';
import { PopupSetting } from './setting.es.js';
import { PopupV1 } from './ui.es.js';
import { RootEditor } from '../../root/selector.es.js';
import { ColEditor } from '../../col/selector.es.js';
import { RowEditor } from '../../section/selector.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const PopupEditorV1 = (_a) => {
  var _b = _a, {
    children,
    quickSetting
  } = _b, rest = __objRest(_b, [
    "children",
    "quickSetting"
  ]);
  const {
    connectors: { connect },
    node
  } = useNode((node2) => ({
    node: node2
  }));
  const { show, setShow } = usePopup();
  const [name, setName] = useProp("name");
  const [id, setId] = useProp("id");
  useEffect(() => {
    if (!id)
      setId(node.id);
    if (!name)
      setName(node.id);
  }, []);
  return /* @__PURE__ */ jsxs(PopupV1, __spreadProps(__spreadValues({
    show,
    isEditorMode: true,
    ref: (ref) => connect(ref)
  }, rest), {
    children: [
      quickSetting || React__default.createElement(PopupQuickSetting, { setShow }),
      children
    ]
  }));
};
PopupEditorV1.craft = crafting({
  displayName: "Popup",
  tag: PopupV1.displayName.toLowerCase(),
  customAttributes: PopupSetting,
  advanceAttributes: AdvanceAttributes,
  props: DEFAULT_POPUP_V1,
  rules: {
    canDrag: () => false,
    canMoveIn: (incomingNodes) => {
      const ignoreSources = [
        getEditorName({ RootEditor }),
        getEditorName({ ColEditor }),
        getEditorName({ RowEditor })
      ];
      return incomingNodes.every(
        (incomingNode) => !ignoreSources.includes(incomingNode.data.name)
      );
    }
  }
});

export { PopupEditorV1 };
