// @tempi/core-elements/dist/components/map/selector.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { withSelector, AdvanceAttributes, getEditorName } from '@tempi/core-editor';
import { DEFAULT_MAP } from './constants.es.js';
import { MapQuickSetting } from './quick-setting.es.js';
import { MapSetting } from './setting.es.js';
import { Map } from './ui.es.js';
import { RowEditor } from '../section/selector.es.js';
import { RootEditor } from '../root/selector.es.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
const Inner = (props) => {
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx("div", {
        style: { position: "absolute", inset: 0 }
      }),
      /* @__PURE__ */ jsx(Map, __spreadValues({
        style: { pointerEvents: "none" }
      }, props))
    ]
  });
};
const MapEditor = withSelector(
  Inner,
  {
    displayName: "B\u1EA3n \u0111\u1ED3",
    tag: Map.displayName.toLowerCase(),
    advanceAttributes: AdvanceAttributes,
    quickSetting: MapQuickSetting,
    customAttributes: MapSetting,
    props: DEFAULT_MAP,
    rules: {
      canDrop: (targetNode) => {
        const ignoreTargets = [
          getEditorName({ RowEditor }),
          getEditorName({ RootEditor })
        ];
        return !ignoreTargets.includes(targetNode.data.name);
      }
    }
  },
  {
    style: {
      position: "relative"
    }
  }
);

export { MapEditor };
