// @tempi/core-elements/dist/components/map/setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var ContentWrapper = require('../../core/ContentWrapper/ContentWrapper.js');
var propKey = require('../../constants/propKey.js');

const MapSetting = () => {
  const zoomOptions = [
    { label: "map_config.zoom.world", value: 2 },
    { label: "map_config.zoom.country", value: 6 },
    { label: "map_config.zoom.area", value: 9 },
    { label: "map_config.zoom.province", value: 12 },
    { label: "map_config.zoom.district", value: 14 },
    { label: "map_config.zoom.road", value: 17 },
    { label: "map_config.zoom.home", value: 20 }
  ];
  return /* @__PURE__ */ jsxRuntime.jsxs(ContentWrapper.ContentWrapper, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputConfig, {
        propKey: propKey.PROP_KEY.Address,
        formItemProps: {
          label: i18n.t("\u0110\u1ECBa ch\u1EC9")
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
        options: i18n.getOptionsI18n(i18n.t, zoomOptions, "label"),
        propKey: propKey.PROP_KEY.Zoom,
        formItemProps: {
          label: i18n.t("Ph\xF3ng to"),
          width: 150
        }
      })
    ]
  });
};

exports.MapSetting = MapSetting;
