// @tempi/core-elements/dist/components/map/setting.es.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { t, getOptionsI18n } from '../../i18n.es.js';
import { InputConfig, SelectConfig } from '@tempi/core-editor';
import { ContentWrapper } from '../../core/ContentWrapper/ContentWrapper.es.js';
import { PROP_KEY } from '../../constants/propKey.es.js';

const MapSetting = () => {
  const zoomOptions = [
    { label: "map_config.zoom.world", value: 2 },
    { label: "map_config.zoom.country", value: 6 },
    { label: "map_config.zoom.area", value: 9 },
    { label: "map_config.zoom.province", value: 12 },
    { label: "map_config.zoom.district", value: 14 },
    { label: "map_config.zoom.road", value: 17 },
    { label: "map_config.zoom.home", value: 20 }
  ];
  return /* @__PURE__ */ jsxs(ContentWrapper, {
    children: [
      /* @__PURE__ */ jsx(InputConfig, {
        propKey: PROP_KEY.Address,
        formItemProps: {
          label: t("\u0110\u1ECBa ch\u1EC9")
        }
      }),
      /* @__PURE__ */ jsx(SelectConfig, {
        options: getOptionsI18n(t, zoomOptions, "label"),
        propKey: PROP_KEY.Zoom,
        formItemProps: {
          label: t("Ph\xF3ng to"),
          width: 150
        }
      })
    ]
  });
};

export { MapSetting };
