// @tempi/core-editor/dist/configs/EventConfig/EventConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var React = require('react');
var antd = require('antd');
var coreRenderer = require('@tempi/core-renderer');
var reactFrameComponent = require('react-frame-component');
var SelectPopupConfig = require('./SelectPopupConfig.js');
var TrackingConfig = require('../TrackingConfig/TrackingConfig.js');
var TekoTrackingConfig = require('../TekoTrackingConfig/TekoTrackingConfig.js');
var SwitchConfig = require('../SwitchConfig/SwitchConfig.js');
var BuilderContext = require('../../contexts/BuilderContext.js');
var propKey = require('../../constants/propKey.js');
var usePropSeparateDevice = require('../../hooks/usePropSeparateDevice.js');
var useEditor = require('../../hooks/useEditor.js');
var SelectConfig = require('../SelectConfig/SelectConfig.js');
var InputConfig = require('../InputConfig/InputConfig.js');
var InputNumberConfig = require('../InputNumberConfig/InputNumberConfig.js');

const EventConfig = ({
  propKey: propKey$1,
  visible,
  setVisible,
  separateDevice,
  listOptions
}) => {
  const { popupListSelection, pageListInWebsiteSelection, relatedNodeId } = BuilderContext.useBuilder();
  const [warning, setWarning] = React.useState("");
  const eventTypePropKey = propKey.PROP_KEY.joinKey(
    propKey$1,
    propKey.PROP_KEY.Event,
    propKey.PROP_KEY.EventType
  );
  const linkPropKey = propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.Event, propKey.PROP_KEY.Link);
  const emailPropKey = propKey.PROP_KEY.joinKey(
    propKey$1,
    propKey.PROP_KEY.Event,
    propKey.PROP_KEY.Email
  );
  const phonePropKey = propKey.PROP_KEY.joinKey(
    propKey$1,
    propKey.PROP_KEY.Event,
    propKey.PROP_KEY.Phone
  );
  const inPagePropKey = propKey.PROP_KEY.joinKey(
    propKey$1,
    propKey.PROP_KEY.Event,
    propKey.PROP_KEY.InPage
  );
  const togglePopupPropKey = propKey.PROP_KEY.joinKey(
    propKey$1,
    propKey.PROP_KEY.Event,
    propKey.PROP_KEY.TogglePopup
  );
  const togglePopupListPropKey = propKey.PROP_KEY.joinKey(
    propKey$1,
    propKey.PROP_KEY.Event,
    propKey.PROP_KEY.TogglePopupList
  );
  const openPageInWebsite = propKey.PROP_KEY.joinKey(
    propKey$1,
    propKey.PROP_KEY.Event,
    propKey.PROP_KEY.OpenPageOfWebsite
  );
  const [eventType, setEventType] = usePropSeparateDevice.usePropSeparateDevice(
    eventTypePropKey,
    separateDevice
  );
  const [phone] = usePropSeparateDevice.usePropSeparateDevice(phonePropKey, separateDevice);
  const [email] = usePropSeparateDevice.usePropSeparateDevice(emailPropKey, separateDevice);
  const { enableTekoTracking } = BuilderContext.useBuilder();
  const { nodes } = useEditor.useEditor((state) => {
    return {
      nodes: state.nodes
    };
  });
  const { document: frameDocument } = reactFrameComponent.useFrame();
  const elementList = React.useMemo(() => {
    return [
      ...new Set(Object.keys(nodes).map((key) => nodes[key].data.props.id))
    ].filter((option) => option).map((id) => ({ label: id, value: id }));
  }, [eventType]);
  const getActionTypeOption = () => {
    const actionTypeOptions = [
      {
        label: i18n.t("M\u1EDF li\xEAn k\u1EBFt"),
        value: "link"
      },
      {
        label: i18n.t("Cu\u1ED9n trong trang"),
        value: "inPage"
      },
      {
        label: "Email",
        value: "email"
      },
      {
        label: i18n.t("S\u1ED1 \u0111i\u1EC7n tho\u1EA1i"),
        value: "phone"
      },
      {
        label: i18n.t("M\u1EDF popup"),
        value: "openPopup"
      },
      {
        label: i18n.t("\u0110\xF3ng popup"),
        value: "closePopupList"
      }
    ];
    if (pageListInWebsiteSelection == null ? void 0 : pageListInWebsiteSelection.length) {
      actionTypeOptions.unshift({
        label: i18n.t("M\u1EDF trang trong website"),
        value: "openPageOfWebsite"
      });
    }
    if (listOptions)
      return actionTypeOptions.filter(
        ({ value }) => listOptions.includes(value)
      );
    return actionTypeOptions;
  };
  const validatePhone = (phone2) => {
    if (!phone2)
      return;
    if (!coreRenderer.PATTERN_VALIDATION.phone.test(phone2)) {
      setWarning(i18n.t("S\u1ED1 \u0111i\u1EC7n tho\u1EA1i ch\u01B0a \u0111\xFAng \u0111\u1ECBnh d\u1EA1ng"));
    } else
      setWarning("");
  };
  const validateEmail = (email2) => {
    if (!email2)
      return;
    if (!coreRenderer.PATTERN_VALIDATION.email.test(email2)) {
      setWarning(i18n.t("Nh\u1EADp email \u0111\u1ECBnh d\u1EA1ng abc@gmai.com"));
    } else
      setWarning("");
  };
  const onClose = () => {
    setVisible(false);
  };
  React.useEffect(() => {
    if (eventType === "phone")
      validatePhone(phone);
    if (eventType === "email")
      validateEmail(email);
  }, [eventType, phone, email]);
  React.useEffect(() => {
    if (relatedNodeId == null ? void 0 : relatedNodeId.popupId) {
      setVisible(true);
    }
  }, []);
  const renderValues = (eventType2) => {
    switch (eventType2) {
      case "link":
        return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
          children: [
            /* @__PURE__ */ jsxRuntime.jsx(InputConfig.InputConfig, {
              separateDevice,
              formItemProps: {
                label: i18n.t("Nh\u1EADp URL"),
                style: {
                  marginBottom: "0.5rem"
                }
              },
              inputProps: {
                placeholder: i18n.t("Nh\u1EADp URL"),
                style: {
                  width: "100%"
                }
              },
              propKey: linkPropKey + ".href"
            }),
            /* @__PURE__ */ jsxRuntime.jsx(SwitchConfig.SwitchConfig, {
              separateDevice,
              formItemProps: {
                label: i18n.t("M\u1EDF trong th\u1EBB m\u1EDBi"),
                layout: "horizontal",
                style: {
                  display: "flex",
                  justifyContent: "space-between",
                  alignItems: "center",
                  marginBottom: "0.5rem"
                }
              },
              propKey: linkPropKey + ".openNewTab"
            }),
            /* @__PURE__ */ jsxRuntime.jsx(SwitchConfig.SwitchConfig, {
              separateDevice,
              formItemProps: {
                label: i18n.t("Li\xEAn k\u1EBFt nofollow"),
                layout: "horizontal",
                style: {
                  display: "flex",
                  justifyContent: "space-between",
                  alignItems: "center",
                  marginBottom: "0.5rem"
                }
              },
              propKey: linkPropKey + ".nofollow"
            })
          ]
        });
      case "inPage":
        return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
          children: [
            /* @__PURE__ */ jsxRuntime.jsx(SelectConfig.SelectConfig, {
              separateDevice,
              formItemProps: {
                label: i18n.t("Ph\u1EA7n t\u1EED"),
                style: {
                  marginBottom: "0.5rem"
                }
              },
              selectProps: {
                placeholder: i18n.t("M\u1EB7c \u0111\u1ECBnh"),
                style: { width: "100%" },
                popupMatchSelectWidth: false
              },
              options: elementList,
              propKey: inPagePropKey
            }),
            /* @__PURE__ */ jsxRuntime.jsx(InputNumberConfig.InputNumberConfig, {
              separateDevice,
              propKey: propKey.PROP_KEY.joinKey(
                propKey.PROP_KEY.Event,
                propKey.PROP_KEY.SectionMarginTop
              ),
              formItemProps: {
                label: "Section margin-top",
                style: {
                  marginBottom: "0.5rem"
                }
              },
              unitProps: { defaultValue: coreRenderer.UnitEnum.pixel },
              inputNumberProps: {
                style: { width: "100%" }
              }
            })
          ]
        });
      case "email":
        return /* @__PURE__ */ jsxRuntime.jsx(InputConfig.InputConfig, {
          separateDevice,
          formItemProps: {
            label: "Email",
            warning,
            layout: "vertical",
            style: {
              marginBottom: "0.5rem"
            }
          },
          inputProps: {
            style: {
              width: "100%"
            },
            placeholder: i18n.t("Nh\u1EADp email")
          },
          propKey: emailPropKey
        });
      case "phone":
        return /* @__PURE__ */ jsxRuntime.jsx(InputConfig.InputConfig, {
          separateDevice,
          formItemProps: {
            label: i18n.t("S\u1ED1 \u0111i\u1EC7n tho\u1EA1i"),
            warning,
            layout: "vertical",
            style: {
              marginBottom: "0.5rem"
            }
          },
          inputProps: {
            placeholder: i18n.t("Nh\u1EADp s\u1ED1 \u0111i\u1EC7n tho\u1EA1i c\xF3 10 ch\u1EEF s\u1ED1"),
            style: { width: "100%" }
          },
          propKey: phonePropKey
        });
      case "openPopup":
        return /* @__PURE__ */ jsxRuntime.jsx(SelectPopupConfig.SelectPopupConfig, {
          separateDevice,
          label: i18n.t("M\u1EDF popup"),
          options: popupListSelection,
          propKey: togglePopupPropKey
        });
      case "closePopupList":
        return /* @__PURE__ */ jsxRuntime.jsx(SelectPopupConfig.SelectPopupConfig, {
          separateDevice,
          label: i18n.t("\u0110\xF3ng popup"),
          options: popupListSelection,
          propKey: togglePopupListPropKey
        });
      case "openPageOfWebsite":
        return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
          children: [
            /* @__PURE__ */ jsxRuntime.jsx(SelectConfig.SelectConfig, {
              separateDevice,
              formItemProps: {
                style: {
                  marginBottom: "0.5rem"
                }
              },
              selectProps: {
                placeholder: i18n.t("M\u1EB7c \u0111\u1ECBnh"),
                style: { width: "100%" },
                popupMatchSelectWidth: false
              },
              options: pageListInWebsiteSelection,
              propKey: propKey.PROP_KEY.joinKey(openPageInWebsite, propKey.PROP_KEY.Href)
            }),
            /* @__PURE__ */ jsxRuntime.jsx(SwitchConfig.SwitchConfig, {
              separateDevice,
              formItemProps: {
                label: i18n.t("M\u1EDF trong th\u1EBB m\u1EDBi"),
                layout: "horizontal",
                style: {
                  display: "flex",
                  justifyContent: "space-between",
                  alignItems: "center",
                  marginBottom: "0.5rem"
                }
              },
              propKey: propKey.PROP_KEY.joinKey(openPageInWebsite, propKey.PROP_KEY.OpenNewTab)
            })
          ]
        });
    }
  };
  return /* @__PURE__ */ jsxRuntime.jsx(antd.Modal, {
    closable: false,
    zIndex: 1080,
    open: visible,
    onCancel: onClose,
    title: i18n.t("C\xE0i \u0111\u1EB7t s\u1EF1 ki\u1EC7n nh\u1EA5p chu\u1ED9t"),
    destroyOnClose: true,
    width: 450,
    footer: [
      /* @__PURE__ */ jsxRuntime.jsx(antd.Button, {
        type: "primary",
        shape: "round",
        onClick: onClose,
        children: i18n.t("\u0110\xF3ng")
      })
    ],
    getContainer: () => (frameDocument == null ? void 0 : frameDocument.body) || (document == null ? void 0 : document.body),
    children: /* @__PURE__ */ jsxRuntime.jsx(antd.Tabs, {
      items: [
        {
          label: i18n.t("S\u1EF1 ki\u1EC7n"),
          key: "1",
          children: /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
            children: [
              /* @__PURE__ */ jsxRuntime.jsx(SelectConfig.SelectConfig, {
                options: getActionTypeOption(),
                formItemProps: {
                  label: i18n.t("Lo\u1EA1i s\u1EF1 ki\u1EC7n"),
                  style: { marginBottom: "0.5rem" }
                },
                selectProps: {
                  placeholder: i18n.t("Ch\u1ECDn s\u1EF1 ki\u1EC7n"),
                  style: { width: "100%" },
                  onChange: (val) => {
                    setEventType(val);
                    setWarning("");
                  },
                  value: eventType
                }
              }),
              renderValues(eventType)
            ]
          })
        },
        {
          label: i18n.t("Chuy\u1EC3n \u0111\u1ED5i (Tracking)"),
          key: "2",
          children: /* @__PURE__ */ jsxRuntime.jsx(TrackingConfig.TrackingConfig, {
            separateDevice,
            config: "event",
            propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.Event)
          })
        },
        enableTekoTracking && {
          label: "Teko tracking",
          key: "3",
          children: /* @__PURE__ */ jsxRuntime.jsx(TekoTrackingConfig.TekoTrackingConfig, {
            propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.Event)
          })
        }
      ]
    })
  });
};

exports.EventConfig = EventConfig;
