// @tempi/core-editor/dist/components/ContextMenu/ContextMenu.es.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import styled from '@emotion/styled';
import { useNode } from '@tempi/dnd-core';
import { Dropdown } from 'antd';
import { useEditor } from '../../hooks/useEditor.es.js';
import { useBuilder } from '../../contexts/BuilderContext.es.js';

const ContextMenu = (props) => {
  const { children } = props;
  const {
    actions,
    duplicateNode,
    getSiblings
  } = useEditor();
  const { id, node, parentId } = useNode((node2) => ({
    node: node2,
    parentId: node2.data.parent
  }));
  const { setShowSettingDialog, frame } = useBuilder();
  const onDelete = () => {
    actions.delete(id);
  };
  const onDuplicate = () => {
    const index = getSiblings(id).findIndex((el) => el === id);
    duplicateNode(node, parentId, index + 1);
  };
  const onOpenSetting = () => {
    actions.selectNode(id);
    setShowSettingDialog(true);
  };
  const items = [
    {
      label: /* @__PURE__ */ jsxs(MenuLabel, {
        children: [
          /* @__PURE__ */ jsx("span", {
            children: t("C\xE0i \u0111\u1EB7t")
          }),
          /* @__PURE__ */ jsx("span", {
            children: "\u2318 E"
          })
        ]
      }),
      key: "1",
      onClick: onOpenSetting
    },
    {
      label: /* @__PURE__ */ jsxs(MenuLabel, {
        children: [
          /* @__PURE__ */ jsx("span", {
            children: t("Nh\xE2n b\u1EA3n")
          }),
          /* @__PURE__ */ jsx("span", {
            children: "\u2318 D"
          })
        ]
      }),
      key: "4",
      onClick: onDuplicate
    },
    {
      label: /* @__PURE__ */ jsxs(MenuLabel, {
        children: [
          /* @__PURE__ */ jsx("span", {
            children: t("Xo\xE1")
          }),
          /* @__PURE__ */ jsx("span", {
            children: "\u2318 \u232B"
          })
        ]
      }),
      key: "5",
      onClick: onDelete
    }
  ];
  return /* @__PURE__ */ jsx(Dropdown, {
    getPopupContainer: () => frame == null ? void 0 : frame.body,
    menu: { items, style: { width: 160 } },
    trigger: ["contextMenu"],
    children: /* @__PURE__ */ jsx("div", {
      children
    })
  });
};
const MenuLabel = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export { ContextMenu };
