// @tempi/core-elements/dist/components/header/selector/Logo/LogoEditor.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var coreEditor = require('@tempi/core-editor');
var React = require('react');
var quickSetting = require('../../../image/quick-setting.js');
var setting = require('../../../image/setting.js');
var constants = require('../../constants.js');
var Logo = require('../../ui/Logo/Logo.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

const LogoEditor = coreEditor.withSelectorDirectElement(Logo.Logo, {
  displayName: Logo.Logo.displayName,
  tag: Logo.Logo.displayName.toLowerCase(),
  advanceAttributes: coreEditor.AdvanceAttributes,
  quickSetting: quickSetting.ImageQuickSetting,
  customAttributes: () => React__default["default"].createElement(setting.ImageSettings, {
    showZoomPreview: false,
    showAlignment: false
  }),
  props: constants.DEFAULT_HEADER.logo,
  rules: {
    canDrag: () => false
  }
});

exports.LogoEditor = LogoEditor;
