// @tempi/core-renderer/dist/constants/fonts.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const pbFontFamily = [
  "Alfa Slab One",
  "Anton",
  "Arima Madurai",
  "Arimo",
  "Athiti",
  "Bahianita",
  "Baloo Bhaina",
  "Barlow",
  "Barlow Condensed",
  "Bungee",
  "Bungee Inline",
  "Bungee Outline",
  "Bungee Shade",
  "Cabin",
  "Chonburi",
  "Cormorant Upright",
  "Crimson Pro",
  "Dancing Script",
  "Dosis",
  "EB Garamond",
  "Exo",
  "Epilogue",
  "Fira Sans",
  "Hepta Slab",
  "Inconsolata",
  "Inter",
  "Itim",
  "Josefin Sans",
  "Judson",
  "Jura",
  "Kanit",
  "K2D",
  "Lexend",
  "Literata",
  "Lobster",
  "Lora",
  "Manrope",
  "Maven Pro",
  "Merriweather",
  "Montserrat",
  "Mulish",
  "Niramit",
  "Noto Serif",
  "Open Sans",
  "Oswald",
  "Pattaya",
  "Paytone One",
  "Philosopher",
  "Play",
  "Playfair Display",
  "Poppins",
  "Prata",
  "Quicksand",
  "Raleway",
  "Roboto",
  "Roboto Slab",
  "Rokkit",
  "Source Sans Pro",
  "Space Grotesk",
  "Sriracha",
  "Taviraj",
  "Thasadith",
  "Tinos",
  "Trirong",
  "VT323",
  "Work Sans",
  "Yeseva One"
];
const pbFontWeight = {
  light: 300,
  regular: 400,
  medium: 500,
  bold: 700
};
const fontWeightQueryParams = Object.values(pbFontWeight).join(";");
const fontQueryParams = pbFontFamily.map((font) => {
  return `family=${encodeURIComponent(font)}:wght@${fontWeightQueryParams}`;
}).join("&");

exports.fontQueryParams = fontQueryParams;
exports.pbFontFamily = pbFontFamily;
exports.pbFontWeight = pbFontWeight;
