// @tempi/core-renderer/dist/constants/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var React = require('react');
var fonts = require('./fonts.js');
var reactHelmet = require('react-helmet');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const landingLinks = [
  {
    rel: "preconnect",
    href: "https://fonts.googleapis.com"
  },
  {
    rel: "preconnect",
    href: "https://fonts.gstatic.com",
    crossOrigin: "anonymous"
  },
  {
    rel: "preLoad stylesheet",
    href: `https://fonts.googleapis.com/css2?${fonts.fontQueryParams}&display=swap`,
    as: "style",
    type: "text/css"
  },
  {
    rel: "preLoad stylesheet",
    href: "https://cdnjs.cloudflare.com/ajax/libs/animate.css/4.1.1/animate.min.css",
    as: "style",
    type: "text/css"
  }
];
const LandingLinks = () => /* @__PURE__ */ jsxRuntime.jsx(reactHelmet.Helmet, {
  children: landingLinks.map((link, index) => /* @__PURE__ */ React.createElement("link", __spreadProps(__spreadValues({}, link), {
    key: index
  })))
});
const publicBffUrl = {
  local: "http://localhost:8081",
  dev: "https://public-builder-bff.dev.tekoapis.net",
  stag: "https://public-builder-bff.stag.tekoapis.net",
  production: "https://public-bff.tempi.vn",
  "test-1": "https://public-builder-bff.test-1.tekoapis.net",
  internal: "http://public-bff-api.page-management"
};

exports.fontQueryParams = fonts.fontQueryParams;
exports.pbFontFamily = fonts.pbFontFamily;
exports.pbFontWeight = fonts.pbFontWeight;
exports.LandingLinks = LandingLinks;
exports.landingLinks = landingLinks;
exports.publicBffUrl = publicBffUrl;
