// @tempi/core-elements/dist/components/youtube/quick-setting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { YoutubeOutlined } from '@ant-design/icons';
import { SubQuickSetting, QuickSettingButton, OpenSettingDialog, DuplicateNode, DeleteNode, InputConfig, SwitchConfig } from '@tempi/core-editor';
import { ResizeQuickSetting } from '../../quick-configs/ResizeQuickSetting/ResizeQuickSetting.es.js';

const YoutubeSetting = () => {
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(InputConfig, {
        propKey: "link",
        formItemProps: {
          label: t("Li\xEAn k\u1EBFt Youtube")
        },
        inputProps: {
          placeholder: t("Nh\u1EADp URL")
        }
      }),
      /* @__PURE__ */ jsx(SwitchConfig, {
        propKey: "autoplay",
        formItemProps: {
          label: t("T\u1EF1 \u0111\u1ED9ng ph\xE1t tr\xEAn m\xE1y t\xEDnh")
        }
      })
    ]
  });
};
const YoutubeQuickSetting = () => {
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(SubQuickSetting, {
        content: YoutubeSetting,
        children: /* @__PURE__ */ jsx(QuickSettingButton, {
          children: /* @__PURE__ */ jsx(YoutubeOutlined, {})
        })
      }),
      /* @__PURE__ */ jsx(ResizeQuickSetting, {
        configs: { allowConfigHeight: false }
      }),
      /* @__PURE__ */ jsx(OpenSettingDialog, {}),
      /* @__PURE__ */ jsx(DuplicateNode, {}),
      /* @__PURE__ */ jsx(DeleteNode, {})
    ]
  });
};

export { YoutubeQuickSetting };
