// @tempi/core-elements/dist/components/divider/quick-setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var reactIcons = require('@fluentui/react-icons');
var coreEditor = require('@tempi/core-editor');
var DividerLineSetting = require('./configs/DividerLineSetting.js');
var DividerTypeSettings = require('./configs/DividerTypeSettings.js');
var AlignmentQuickSetting = require('../../quick-configs/AlignmentQuickSetting/AlignmentQuickSetting.js');
var propKey = require('../../constants/propKey.js');

const DividerQuickSetting = () => {
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SubQuickSetting, {
        content: /* @__PURE__ */ jsxRuntime.jsx(DividerLineSetting.DividerLineSetting, {}),
        children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.QuickSettingButton, {
          tooltip: i18n.t("T\xF9y ch\u1EC9nh \u0111\u01B0\u1EDDng ph\xE2n chia"),
          children: /* @__PURE__ */ jsxRuntime.jsx(reactIcons.SplitHorizontalRegular, {
            fontSize: 16
          })
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SubQuickSetting, {
        content: /* @__PURE__ */ jsxRuntime.jsx(DividerTypeSettings.DividerTypeSetting, {}),
        children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.QuickSettingButton, {
          tooltip: i18n.t("Th\xEAm th\xE0nh ph\u1EA7n"),
          children: /* @__PURE__ */ jsxRuntime.jsx(reactIcons.TextExpandRegular, {
            fontSize: 16
          })
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx(AlignmentQuickSetting.AlignmentQuickSetting, {
        propKey: propKey.PROP_KEY.Align,
        listAlignment: ["left", "center", "right"]
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.DuplicateNode, {}),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.DeleteNode, {}),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.OpenSettingDialog, {})
    ]
  });
};

exports.DividerQuickSetting = DividerQuickSetting;
