// @tempi/core-elements/dist/components/divider/quick-setting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { SplitHorizontalRegular, TextExpandRegular } from '@fluentui/react-icons';
import { SubQuickSetting, QuickSettingButton, DuplicateNode, DeleteNode, OpenSettingDialog } from '@tempi/core-editor';
import { DividerLineSetting } from './configs/DividerLineSetting.es.js';
import { DividerTypeSetting } from './configs/DividerTypeSettings.es.js';
import { AlignmentQuickSetting } from '../../quick-configs/AlignmentQuickSetting/AlignmentQuickSetting.es.js';
import { PROP_KEY } from '../../constants/propKey.es.js';

const DividerQuickSetting = () => {
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(SubQuickSetting, {
        content: /* @__PURE__ */ jsx(DividerLineSetting, {}),
        children: /* @__PURE__ */ jsx(QuickSettingButton, {
          tooltip: t("T\xF9y ch\u1EC9nh \u0111\u01B0\u1EDDng ph\xE2n chia"),
          children: /* @__PURE__ */ jsx(SplitHorizontalRegular, {
            fontSize: 16
          })
        })
      }),
      /* @__PURE__ */ jsx(SubQuickSetting, {
        content: /* @__PURE__ */ jsx(DividerTypeSetting, {}),
        children: /* @__PURE__ */ jsx(QuickSettingButton, {
          tooltip: t("Th\xEAm th\xE0nh ph\u1EA7n"),
          children: /* @__PURE__ */ jsx(TextExpandRegular, {
            fontSize: 16
          })
        })
      }),
      /* @__PURE__ */ jsx(AlignmentQuickSetting, {
        propKey: PROP_KEY.Align,
        listAlignment: ["left", "center", "right"]
      }),
      /* @__PURE__ */ jsx(DuplicateNode, {}),
      /* @__PURE__ */ jsx(DeleteNode, {}),
      /* @__PURE__ */ jsx(OpenSettingDialog, {})
    ]
  });
};

export { DividerQuickSetting };
