// @tempi/core-elements/dist/components/divider/selector.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var coreEditor = require('@tempi/core-editor');
var constants = require('./constants.js');
var quickSetting = require('./quick-setting.js');
var ui = require('./ui.js');
var TiptapEditorSimple = require('../../configs/TipTapEditor/TiptapEditorSimple.js');
var selector = require('../section/selector.js');
var selector$1 = require('../root/selector.js');

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
const DividerEditor = coreEditor.withSelector(
  (props) => {
    return /* @__PURE__ */ jsxRuntime.jsx(ui.PbDivider, __spreadValues({
      customText: /* @__PURE__ */ jsxRuntime.jsx(TiptapEditorSimple.TiptapEditorSimple, {
        propKey: "textConfig.text"
      })
    }, props));
  },
  {
    displayName: "\u0110\u01B0\u1EDDng ph\xE2n chia",
    tag: ui.PbDivider.displayName.toLowerCase(),
    quickSetting: quickSetting.DividerQuickSetting,
    advanceAttributes: coreEditor.AdvanceAttributes,
    props: constants.DEFAULT_DIVIDER,
    rules: {
      canDrop: (targetNode) => {
        const ignoreTargets = [
          coreEditor.getEditorName({ RowEditor: selector.RowEditor }),
          coreEditor.getEditorName({ RootEditor: selector$1.RootEditor })
        ];
        return !ignoreTargets.includes(targetNode.data.name);
      }
    }
  },
  {
    style: {
      cursor: "pointer"
    }
  }
);

exports.DividerEditor = DividerEditor;
