// @tempi/core-elements/dist/components/sample-component/selector.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var coreEditor = require('@tempi/core-editor');
var constants = require('./constants.js');
var quickSetting = require('./quick-setting.js');
var setting = require('./setting.js');
var ui = require('./ui.js');
var selector = require('../section/selector.js');
var selector$1 = require('../root/selector.js');

var _a, _b;
const SampleComponentEditor = coreEditor.withSelector(ui.SampleComponent, {
  displayName: "Tempi Sample",
  tag: ((_b = (_a = ui.SampleComponent) == null ? void 0 : _a.displayName) == null ? void 0 : _b.toLowerCase()) || "Tempi Sample",
  quickSetting: quickSetting.QuickSetting,
  customAttributes: setting.SampleComponentSetting,
  advanceAttributes: coreEditor.AdvanceAttributes,
  props: constants.DEFAULT_SAMPLE_COMPONENT,
  rules: {
    canDrop: (targetNode) => {
      const ignoreTargets = [
        coreEditor.getEditorName({ RowEditor: selector.RowEditor }),
        coreEditor.getEditorName({ RootEditor: selector$1.RootEditor })
      ];
      return !ignoreTargets.includes(targetNode.data.name);
    }
  }
});

exports.SampleComponentEditor = SampleComponentEditor;
exports["default"] = SampleComponentEditor;
