// @tempi/core-renderer/dist/utils/getComponentProps.es.js
import clsx from 'clsx';
import { validateIsVisibleComponent } from './date-time.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const getComponentProps = (compData, configs, ssrData) => {
  var _a, _b;
  const {
    customAttributes,
    style,
    elementId,
    id,
    startTime,
    endTime,
    ruleRepeat,
    date,
    duration,
    tag,
    children
  } = compData || {};
  const props = __spreadValues(__spreadProps(__spreadValues({}, customAttributes[tag] || {}), {
    commonStyle: style || {},
    id: elementId,
    "data-id": elementId,
    "data-uniq-id": id,
    dataUniqId: id
  }), configs);
  const clsName = clsx({
    "hidden-xs": (_a = props.responsive) == null ? void 0 : _a.hiddenXs,
    "hidden-md": (_b = props.responsive) == null ? void 0 : _b.hiddenMd
  });
  const isVisible = validateIsVisibleComponent({
    startTime,
    endTime,
    ruleRepeat,
    date,
    duration
  });
  return {
    key: id,
    props,
    className: clsName,
    ssrData,
    isVisible,
    nodes: children || []
  };
};

export { getComponentProps };
