// @tempi/core-editor/dist/quick-configs/DeleteNode/DeleteNode.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var reactIcons = require('@fluentui/react-icons');
var dndCore = require('@tempi/dnd-core');
var useEditor = require('../../hooks/useEditor.js');
var QuickSettingButton = require('../../components/QuickSettingButton/QuickSettingButton.js');

const DeleteNode = ({ customDelete }) => {
  const { actions, query } = useEditor.useEditor();
  const { id, isDeletable } = dndCore.useNode((node) => ({
    isDeletable: query.node(node.id).isDeletable(),
    name: node.data.name,
    parent: node.data.parent
  }));
  const onDelete = (e) => {
    e.stopPropagation();
    if (typeof customDelete === "function")
      customDelete(id);
    else
      actions.delete(id);
  };
  if (!isDeletable)
    return null;
  return /* @__PURE__ */ jsxRuntime.jsx(QuickSettingButton.QuickSettingButton, {
    onClick: onDelete,
    children: /* @__PURE__ */ jsxRuntime.jsx(reactIcons.Delete16Regular, {})
  });
};

exports.DeleteNode = DeleteNode;
