// @tempi/core-elements/dist/components/header/constants.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var i18n = require('../../i18n.js');
var index = require('../../types/index.js');
var type = require('./type.js');

const DEFAULT_HEADER = {
  header: {
    alignment: "right",
    widthType: index.WidthTypeEnum.full,
    toggle: {
      icon: {
        color: "rgba(0, 0, 0, 1)",
        size: 24
      }
    },
    commonStyle: {
      backgroundColor: "rgba(255, 255, 255, 1)",
      minHeight: 80,
      padding: {
        left: 16,
        right: 16
      },
      position: "relative"
    }
  },
  logo: {
    alignment: "left",
    commonStyle: {
      height: 48
    },
    src: "https://lh3.googleusercontent.com/trFHKrybkS_WgcUruWcZFLbrZ6fDHCIw6_xMc-7Gaxr_FeZfZN0JAqeYwcAxA1jGDUs609yRC0WHX3FLqJQEAuG927ik9OBD"
  },
  link: {
    type: type.MenuItemType.link,
    itemData: {
      text: "<h5>Lorems</h5>"
    }
  },
  button: {
    type: type.MenuItemType.button,
    itemData: {
      text: "<p>Lorems</p>",
      buttonStyle: {
        backgroundColor: "#7367f0",
        color: "rgba(255, 255, 255, 1)",
        height: 32,
        minWidth: 120,
        borderRadius: 999
      },
      iconConfig: {
        position: "left",
        spacing: 8,
        show: true,
        icon: {
          size: 24
        }
      }
    }
  },
  centerMenuItems: [
    i18n.t("Gi\u1EDBi thi\u1EC7u"),
    i18n.t("Ch\xFAng t\xF4i"),
    i18n.t("Khuy\u1EBFn m\xE3i"),
    i18n.t("S\u1EA3n ph\u1EA9m"),
    i18n.t("Ph\u1EA3n h\u1ED3i")
  ].map((text) => ({
    type: type.MenuItemType.link,
    itemData: {
      text: `<h5>${text}</h5>`
    }
  })),
  rightMenuItems: [
    {
      type: type.MenuItemType.button,
      itemData: {
        text: `<p>${i18n.t("\u0110\u0103ng k\xFD")}</p>`,
        buttonStyle: {
          backgroundColor: "#7367f0",
          color: "rgba(255, 255, 255, 1)",
          height: 32,
          minWidth: 120,
          borderRadius: 999
        },
        iconConfig: {
          position: "left",
          spacing: 8,
          show: true,
          icon: {
            size: 24
          }
        }
      }
    }
  ]
};

exports.DEFAULT_HEADER = DEFAULT_HEADER;
