// @tempi/core-elements/dist/next/renderer.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var dynamic = require('next/dynamic');
var ui = require('../components/root/ui.js');
var ui$1 = require('../components/icon/ui.js');
var ui$2 = require('../components/section/ui.js');
var ui$3 = require('../components/container/ui.js');
var ui$4 = require('../components/row/ui.js');
var ui$5 = require('../components/col/ui.js');
var ui$6 = require('../components/button/v2/ui.js');
var ui$7 = require('../components/paragraph/ui.js');
var ui$8 = require('../components/heading/ui.js');
var ui$9 = require('../components/spacer/ui.js');
var ui$a = require('../components/countdown/ui.js');
var ui$b = require('../components/image/ui.js');
var ui$c = require('../components/gallery/ui.js');
var ui$d = require('../components/html-code/ui.js');
var ui$e = require('../components/number-counter/ui.js');
var ui$f = require('../components/slider/v1/ui.js');
var Header = require('../components/header/ui/Header/Header.js');
var Menu = require('../components/header/ui/Menu/Menu.js');
var MenuItem = require('../components/header/ui/MenuItem/MenuItem.js');
var Logo = require('../components/header/ui/Logo/Logo.js');
var MenuCol = require('../components/header/ui/MenuCol/MenuCol.js');
var ui$g = require('../components/website-header/ui.js');
var ui$h = require('../components/video/ui.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var dynamic__default = /*#__PURE__*/_interopDefaultLegacy(dynamic);

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
const Accordion = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/accordion/ui/index.js'); }).then((mod) => mod.Accordion)
);
Accordion.displayName = "Accordion";
const AccordionItem = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/accordion/ui/index.js'); }).then((mod) => mod.AccordionItem)
);
AccordionItem.displayName = "AccordionItem";
const AccordionContent = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/accordion/ui/index.js'); }).then((mod) => mod.AccordionContent)
);
AccordionContent.displayName = "AccordionContent";
const ContentSlider = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/content-slider/ui/index.js'); }).then((mod) => mod.ContentSlider)
);
ContentSlider.displayName = "ContentSlider";
const ContentSliderItem = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/content-slider/ui/index.js'); }).then(
    (mod) => mod.ContentSliderItem
  )
);
ContentSliderItem.displayName = "ContentSliderItem";
const Divider = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/divider/ui.js'); }).then((mod) => mod.PbDivider)
);
Divider.displayName = "Divider";
const FormV4 = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/form/v4/ui.js'); }).then((mod) => mod.FormV4)
);
FormV4.displayName = "FormV4";
const ListItemContentV2 = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/list/v2/ui/index.js'); }).then((mod) => mod.ListItemContentV2)
);
ListItemContentV2.displayName = "ListItemContentV2";
const ListItemIconV2 = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/list/v2/ui/index.js'); }).then((mod) => mod.ListItemIconV2)
);
ListItemIconV2.displayName = "ListItemIconV2";
const ListItemV2 = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/list/v2/ui/index.js'); }).then((mod) => mod.ListItemV2)
);
ListItemV2.displayName = "ListItemV2";
const LuckyDrawV2 = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/lucky-draw/v2/ui.js'); }).then((mod) => mod.LuckyDrawV2)
);
LuckyDrawV2.displayName = "LuckyDrawV2";
const Map = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/map/ui.js'); }).then((mod) => mod.Map)
);
Map.displayName = "Map";
const PageFlip = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/page-flip/ui.js'); }).then((mod) => mod.PageFlip)
);
PageFlip.displayName = "PageFlip";
const PopupV1 = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/popup/v1/ui.js'); }).then((mod) => mod.PopupV1)
);
PopupV1.displayName = "PopupV1";
const SocialIcon = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/social-icon/ui.js'); }).then((mod) => mod.SocialIcon)
);
SocialIcon.displayName = "SocialIcon";
const StoreList = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/store-list/ui.js'); }).then((mod) => mod.StoreList)
);
StoreList.displayName = "StoreList";
const Tab = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/tab/ui/index.js'); }).then((mod) => mod.Tab)
);
Tab.displayName = "Tab";
const TabItem = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/tab/ui/index.js'); }).then((mod) => mod.TabItem)
);
TabItem.displayName = "TabItem";
const Table = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/table/ui.js'); }).then((mod) => mod.Table)
);
Table.displayName = "Table";
const TestimonialV2 = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/testimonial/v2/ui/index.js'); }).then((mod) => mod.TestimonialV2)
);
TestimonialV2.displayName = "TestimonialV2";
const TestimonialV2Item = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/testimonial/v2/ui/index.js'); }).then(
    (mod) => mod.TestimonialV2Item
  )
);
TestimonialV2Item.displayName = "TestimonialV2Item";
const TestimonialCarousel = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/testimonial-carousel/ui.js'); }).then(
    (mod) => mod.TestimonialCarousel
  )
);
TestimonialCarousel.displayName = "TestimonialCarousel";
const Youtube = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/youtube/ui.js'); }).then((mod) => mod.Youtube)
);
Youtube.displayName = "Youtube";
const WebsiteFooter = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/websitefooter/ui.js'); }).then((mod) => mod.WebsiteFooter)
);
WebsiteFooter.displayName = "WebsiteFooter";
const ProductList = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/teko-product-list/ui/index.js'); }).then((mod) => mod.ProductList)
);
ProductList.displayName = "TekoProductListV1";
const ProductListContent = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/teko-product-list/ui/index.js'); }).then(
    (mod) => mod.ProductListContent
  )
);
ProductListContent.displayName = "TekoProductListContentV1";
const GlobalBlock = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/global-block/ui.js'); }).then((mod) => mod.GlobalBlock)
);
GlobalBlock.displayName = "GlobalBlock";
const ContactBubble = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/contact-bubble/ui.js'); }).then((mod) => mod.ContactBubble)
);
ContactBubble.displayName = "ContactBubble";
const SubscriptionFormV2 = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/form/v4/ui.js'); }).then((mod) => {
    const FormV42 = mod.FormV4;
    const SubscriptionFormV22 = (props) => {
      return /* @__PURE__ */ jsxRuntime.jsx(FormV42, __spreadValues({}, props));
    };
    return { default: SubscriptionFormV22 };
  })
);
SubscriptionFormV2.displayName = "SubscriptionFormV2";
const PopupRoot = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/popuproot/ui.js'); }).then((mod) => mod.PopupRoot)
);
PopupRoot.displayName = "PopupRoot";
const Page404 = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/404/ui.js'); }).then((mod) => mod.Page404)
);
Page404.displayName = "404";
const TempiProductList = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/product-list/ui.js'); }).then((mod) => mod.TempiProductList)
);
TempiProductList.displayName = "TempiProductList";
const QRCode = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/qr-code/ui.js'); }).then((mod) => mod.QRCode)
);
QRCode.displayName = "QRCode";
const ArticleList = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/article-list/ui.js'); }).then((mod) => mod.ArticleList)
);
ArticleList.displayName = "ArticleList";
const ArticleHeader = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/article-header/ui.js'); }).then((mod) => mod.ArticleHeader)
);
ArticleHeader.displayName = "ArticleHeader";
const SampleComponent = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/sample-component/ui.js'); }).then(
    (mod) => mod.SampleComponent
  )
);
SampleComponent.displayName = "SampleComponent";
const ExamScore = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/exam-score/ui.js'); }).then((mod) => {
    return mod.ExamScore;
  })
);
ExamScore.displayName = "ExamScore";
const LeadForm = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/lead-form/ui.js'); }).then((mod) => {
    return mod.LeadForm;
  })
);
LeadForm.displayName = "LeadForm";
const InlineBanner = dynamic__default["default"](
  () => Promise.resolve().then(function () { return require('../components/inline-banner/ui.js'); }).then((mod) => {
    return mod.InlineBanner;
  })
);
InlineBanner.displayName = "InlineBanner";
const components = {
  Root: ui.Root,
  PbIcon: ui$1.PbIcon,
  Section: ui$2.PbSection,
  Container: ui$3.PbContainer,
  Row: ui$4.PbRow,
  Col: ui$5.PbCol,
  ButtonV2: ui$6.ButtonV2,
  Paragraph: ui$7.Paragraph,
  Heading: ui$8.Heading,
  Divider,
  Spacer: ui$9.Spacer,
  Youtube,
  Countdown: ui$a.Countdown,
  Map,
  Image: ui$b.Image,
  Gallery: ui$c.Gallery,
  HtmlCode: ui$d.HtmlCode,
  PageFlip,
  NumberCounter: ui$e.NumberCounter,
  SocialIcon,
  StoreList,
  Slider: ui$f.Slider,
  Table,
  Accordion,
  AccordionContent,
  AccordionItem,
  ListItemV2,
  ListItemContentV2,
  ListItemIconV2,
  FormV4,
  SubscriptionFormV2,
  PopupV1,
  Header: Header.Header,
  Menu: Menu.Menu,
  MenuItem: MenuItem.MenuItem,
  Logo: Logo.Logo,
  MenuCol: MenuCol.MenuCol,
  LuckyDrawV2,
  Tab,
  TabItem,
  TestimonialCarousel,
  TestimonialV2,
  TestimonialV2Item,
  ContentSlider,
  ContentSliderItem,
  ProductList,
  ProductListContent,
  WebsiteFooter,
  GlobalBlock,
  WebsiteHeader: ui$g.WebsiteHeader,
  PopupRoot,
  ContactBubble,
  Page404,
  TempiProductList,
  QRCode,
  ExamScore,
  Video: ui$h.Video,
  ArticleList,
  ArticleHeader,
  SampleComponent,
  LeadForm,
  InlineBanner
};

exports.components = components;
