// @tempi/core-elements/dist/configs/ShadowConfig/ShadowConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var styled = require('@emotion/styled');
var coreEditor = require('@tempi/core-editor');
var propKey = require('../../constants/propKey.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

const ShadowConfig = ({
  propKey: propKey$1,
  shadowProps = ["x", "y", "blur", "spread"],
  separateDevice
}) => {
  const props = shadowProps.map((property) => ({
    label: property,
    key: propKey.PROP_KEY.joinKey(propKey$1, property)
  }));
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
        formItemProps: { label: i18n.t("M\xE0u b\xF3ng") },
        propKey: `${propKey$1}.color`
      }),
      /* @__PURE__ */ jsxRuntime.jsx(InputGroup, {
        children: props.map(({ label, key }) => /* @__PURE__ */ jsxRuntime.jsxs(FormItem, {
          children: [
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
              propKey: key,
              formItemProps: { style: { marginBottom: 0 } },
              separateDevice
            }),
            /* @__PURE__ */ jsxRuntime.jsx(Label, {
              children: label
            })
          ]
        }, key))
      })
    ]
  });
};
const InputGroup = styled__default["default"].div`
  display: flex;
  max-width: 100%;
  height: 32px;
  margin-bottom: 28px;
  box-shadow: 0 2px 5px var(--box-shadow-color);
`;
const FormItem = styled__default["default"].div`
  flex: 1;
  min-width: 0;
`;
const Label = styled__default["default"].div`
  text-transform: capitalize;
  font-weight: 500;
  font-size: 12px;
  line-height: 16px;
  text-align: center;
  padding: 4px 4px 0;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
`;

exports.ShadowConfig = ShadowConfig;
