// @tempi/core-elements/dist/components/contact-bubble/components/BubbleItem.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var styled = require('@emotion/styled');
var React = require('react');
var type = require('../type.js');
var coreRenderer = require('@tempi/core-renderer');
var track = require('../../../utils/track.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const EventActionWrapper = ({ children, item }) => {
  const ref = React.useRef();
  const getPropsLink = () => {
    const href = item == null ? void 0 : item.href;
    if (!href)
      return;
    const phoneNumber = href.replace(/-|\.|\s/g, "");
    if (coreRenderer.PATTERN_VALIDATION.phone.test(phoneNumber))
      return {
        href: `tel:${phoneNumber}`,
        target: "_self"
      };
    return {
      href,
      target: "_blank"
    };
  };
  const openPopup = () => {
    var _a, _b, _c, _d;
    (_d = (_c = (_b = (_a = ref.current) == null ? void 0 : _a.ownerDocument) == null ? void 0 : _b.querySelector) == null ? void 0 : _c.call(
      _b,
      `[data-uniq-id="${item == null ? void 0 : item.popup}"]`
    )) == null ? void 0 : _d.click();
  };
  switch (item.eventType) {
    case type.EventType.POPUP:
      return /* @__PURE__ */ jsxRuntime.jsx("div", {
        ref,
        onClick: openPopup,
        style: { cursor: "pointer" },
        children
      });
    default:
      return /* @__PURE__ */ jsxRuntime.jsx("a", __spreadProps(__spreadValues({}, getPropsLink()), {
        rel: "noreferrer",
        children
      }));
  }
};
const BubbleItem = ({ item, position, isEditorMode }) => {
  var _a, _b, _c, _d;
  if (!item)
    return null;
  const tekoTracking = (_a = item == null ? void 0 : item.event) == null ? void 0 : _a.tekoTrackingEvent;
  return /* @__PURE__ */ jsxRuntime.jsxs(Wrapper, __spreadProps(__spreadValues({
    position
  }, !isEditorMode && track.getAttrTracking(tekoTracking)), {
    children: [
      (item == null ? void 0 : item.icon) && /* @__PURE__ */ jsxRuntime.jsx(EventActionWrapper, {
        item,
        children: /* @__PURE__ */ jsxRuntime.jsx("img", __spreadProps(__spreadValues({
          src: (_b = item.icon) == null ? void 0 : _b.src,
          alt: item.title,
          height: item.size
        }, ((_c = item.icon) == null ? void 0 : _c.src) && {
          srcSet: coreRenderer.buildSrcsetDPR((_d = item.icon) == null ? void 0 : _d.src, item.size)
        }), {
          style: {
            aspectRatio: "auto 1/1"
          }
        }))
      }),
      /* @__PURE__ */ jsxRuntime.jsx(Label, {
        children: item == null ? void 0 : item.title
      })
    ]
  }));
};
const getFlexDirection = (position) => {
  switch (position) {
    case type.ContactBubblePosition.BOTTOM_LEFT:
    case type.ContactBubblePosition.CENTER_LEFT:
      return "row";
    case type.ContactBubblePosition.BOTTOM_RIGHT:
    case type.ContactBubblePosition.CENTER_RIGHT:
      return "row-reverse";
    default:
      return "row";
  }
};
const Wrapper = styled__default["default"].div(
  ({ position }) => `
a:hover ~ p, div:hover ~ p {
  display: block;
}

display: flex;
align-items: center;
flex-direction: row-reverse;
gap: 0.8em;
flex-direction: ${getFlexDirection(position)};
`
);
const Label = styled__default["default"].p`
  background: rgba(0, 0, 0, 0.85);
  color: #fff;
  padding: 5px;
  border-radius: 5px;
  position: relative;
  display: none;
  padding: 6px 10px;
`;

exports.BubbleItem = BubbleItem;
