// @tempi/core-elements/dist/components/countdown/constants.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var i18n = require('../../i18n.js');
var coreRenderer = require('@tempi/core-renderer');

const getStartOfDay = (date) => {
  return new Date(date.setHours(0, 0, 0, 0));
};
const DEFAULT_COUNTDOWN = {
  countdownType: "due",
  due: getStartOfDay(new Date()).toISOString(),
  showDays: true,
  showHours: true,
  showMinutes: true,
  showSeconds: true,
  showLabel: true,
  labelDays: i18n.t("Ng\xE0y"),
  labelHours: i18n.t("Gi\u1EDD"),
  labelMinutes: i18n.t("Ph\xFAt"),
  labelSeconds: i18n.t("Gi\xE2y"),
  countdownStyle: {
    elementGap: 20,
    labelGap: 0,
    align: coreRenderer.Alignment.left,
    layout: "vertical",
    labelPosition: "baseline",
    digitStyle: {
      fontSize: 40
    },
    labelStyle: {
      textAlign: "center"
    },
    timeUnitStyle: {
      padding: {
        top: 10,
        bottom: 10,
        left: 10,
        right: 10
      }
    }
  }
};

exports.DEFAULT_COUNTDOWN = DEFAULT_COUNTDOWN;
exports.getStartOfDay = getStartOfDay;
