// @tempi/core-editor/dist/utils/craft.es.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const defaultCommonStyles = {
  commonStyle: {
    margin: {
      top: void 0,
      bottom: void 0,
      left: void 0,
      right: void 0
    },
    padding: {
      top: void 0,
      bottom: void 0,
      left: void 0,
      right: void 0
    },
    backgroundColor: void 0,
    backgroundRepeat: void 0,
    backgroundPosition: void 0,
    backgroundSize: void 0,
    width: void 0,
    maxWidth: void 0,
    minWidth: void 0,
    height: void 0,
    maxHeight: void 0,
    minHeight: void 0
  }
};
const defaultResponsiveVisiblity = {
  responsive: {
    hiddenXs: false,
    hiddenMd: false
  }
};
const crafting = (_a) => {
  var _b = _a, {
    props = {},
    related = {},
    customAttributes,
    commonAttributes,
    advanceAttributes,
    effectAttributes,
    rules,
    custom = {},
    tag,
    commonStyle = {},
    quickSetting
  } = _b, rest = __objRest(_b, [
    "props",
    "related",
    "customAttributes",
    "commonAttributes",
    "advanceAttributes",
    "effectAttributes",
    "rules",
    "custom",
    "tag",
    "commonStyle",
    "quickSetting"
  ]);
  const relatedData = __spreadValues({}, related);
  if (customAttributes) {
    relatedData["customAttributes"] = customAttributes;
  }
  if (commonAttributes) {
    relatedData["commonAttributes"] = commonAttributes;
  }
  if (advanceAttributes) {
    relatedData["advanceAttributes"] = advanceAttributes;
  }
  if (quickSetting) {
    relatedData["quickSetting"] = quickSetting;
  }
  if (effectAttributes) {
    relatedData["effectAttributes"] = effectAttributes;
  }
  return __spreadValues({
    props: __spreadValues(__spreadValues(__spreadProps(__spreadValues({}, defaultCommonStyles), {
      commonStyle
    }), defaultResponsiveVisiblity), props),
    related: relatedData,
    rules,
    custom: __spreadValues(__spreadValues({}, custom), { tag })
  }, rest);
};
const getEditorName = (objEditor) => {
  var _a;
  return (_a = Object.keys(objEditor)) == null ? void 0 : _a[0];
};

export { crafting, getEditorName };
