// @tempi/core-editor/dist/constants/propKey.es.js
const PROP_KEY = {
  Action: "action",
  ActionButton: "actionButton",
  ActiveName: "activeName",
  ActiveTabStyle: "activeTabStyle",
  Address: "address",
  Align: "align",
  AlignItems: "alignItems",
  Alignment: "alignment",
  AlignThumbnail: "alignThumbnail",
  AllowSetupThumbnail: "allowSetupThumbnail",
  Alt: "alt",
  AltStyle: "altStyle",
  AnimationDelay: "animation.delay",
  AnimationDuration: "animation.duration",
  AnimationName: "animation.name",
  AnimationRepeatNumber: "animation.repeatNumber",
  AnimationRepeatType: "animation.repeatType",
  ApiResponsePopup: "responsePopup",
  AppliedDomains: "appliedDomains",
  ArrowStyle: "arrowStyle",
  AutoPassParams: "autoPassParams",
  Autoplay: "autoplay",
  AutoSwipeTimeMilliseconds: "autoSwipeTimeMilliseconds",
  Background: "background",
  BackgroundColor: "backgroundColor",
  BackgroundColorHover: "hover.backgroundColor",
  BackgroundImage: "backgroundImage",
  BackgroundImagePercent: "bgImagePercent",
  BackgroundImageWidth: "bgImageWidth",
  BackgroundMarqueeDirection: "backgroundMarqueeDirection",
  BackgroundMarqueeSpeed: "backgroundMarqueeSpeed",
  BackgroundOverlay: "backgroundOverlay",
  BackgroundOverlayType: "backgroundOverlayType",
  BackgroundPosition: "backgroundPosition",
  BackgroundRepeat: "backgroundRepeat",
  BackgroundSize: "backgroundSize",
  BackgroundVideo: "backgroundVideo",
  BarStyle: "barStyle",
  BlockType: "blockType",
  Border: "border",
  BorderBottomLeftRadius: "borderBottomLeftRadius",
  BorderBottomLeftRadiusUnit: "borderBottomLeftRadiusUnit",
  BorderBottomRightRadius: "borderBottomRightRadius",
  BorderBottomRightRadiusUnit: "borderBottomRightRadiusUnit",
  BorderColor: "borderColor",
  BorderColorHover: "hover.borderColor",
  BorderRadius: "borderRadius",
  BorderRadiusUnit: "borderRadiusUnit",
  BorderStyle: "borderStyle",
  BorderTopLeftRadius: "borderTopLeftRadius",
  BorderTopLeftRadiusUnit: "borderTopLeftRadiusUnit",
  BorderTopRightRadius: "borderTopRightRadius",
  BorderTopRightRadiusUnit: "borderTopRightRadiusUnit",
  BorderWidth: "borderWidth",
  Bottom: "bottom",
  BottomBanners: "bottomBanners",
  BubbleStyle: "bubbleStyle",
  Button: "button",
  ButtonStyle: "buttonStyle",
  CanvasConfig: "canvasConfig",
  Captcha: "captcha",
  CarouselSettings: "carouselSettings",
  CategoryMenuItems: "categoryMenuItems",
  CircleStyle: "circleStyle",
  CityId: "cityId",
  CloseIcon: "closeIcon",
  Code: "code",
  CollectionSlug: "collectionSlug",
  Color: "color",
  ColorHover: "hover.color",
  ColumnCount: "columnCount",
  ColumnGap: "columnGap",
  CommonStyle: "commonStyle",
  Config: "config",
  ConfigStorage: "configStorage",
  Content: "content",
  ContentAlign: "contentAlign",
  ContentMaxWidthKey: "row.commonStyle.maxWidth",
  ContentName: "contentName",
  ContentStyle: "contentStyle",
  ContentType: "contentType",
  ContentWidthType: "contentWidthType",
  ConversionGGAdsId: "conversionGGAdsId",
  ConversionGGAdsLabel: "conversionGGAdsLabel",
  ConversionValue: "conversionValue",
  CornerOption: "cornerOption",
  CornerType: "cornerType",
  CountdownStyle: "countdownStyle",
  Crop: "crop",
  Custom: "custom",
  Data: "data",
  Date: "date",
  DefaultOverlay: "defaultOverlay",
  Delay: "delay",
  Description: "description",
  Desktop: "desktop",
  Device: "device",
  DigitStyle: "digitStyle",
  DisplayCloseButton: "displayCloseButton",
  DisplayOnClickElement: "displayOnClickElement",
  DisplayOnExitPage: "displayOnExitPage",
  DisplayOnScrollTo: "displayOnScrollTo",
  DisplayOnVisitPage: "displayOnVisitPage",
  DistrictId: "districtId",
  Divider: "divider",
  DividerStyle: "dividerStyle",
  Duration: "duration",
  ElementGap: "elementGap",
  Email: "email",
  EndTime: "endTime",
  ErrorStyle: "errorStyle",
  Event: "event",
  EventType: "eventType",
  Failed: "failed",
  FlexDirection: "flexDirection",
  FlexWrap: "flexWrap",
  FontFamily: "fontFamily",
  FontSize: "fontSize",
  FontStyle: "fontStyle",
  FontVariant: "fontVariant",
  FontWeight: "fontWeight",
  ForegroundColor: "foregroundColor",
  FormAccountId: "formAccountId",
  FormConfigId: "formConfigId",
  FormStyle: "formStyle",
  Gap: "gap",
  GoogleImageParams: "googleImageParams",
  GradientColor: "gradientColor",
  GridCols: "gridCols",
  GridRows: "gridRows",
  Gutter: "gutter",
  Headers: "headers",
  HeaderStyle: "headerStyle",
  HeadingEditor: "headingEditor",
  Height: "height",
  HeightUnit: "heightUnit",
  HiddenFields: "hiddenFields",
  HiddenMd: "responsive.hiddenMd",
  HiddenXs: "responsive.hiddenXs",
  HideAfterClick: "hideAfterClick",
  HoverBackgroundColor: "hover.backgroundColor",
  HoverBorderColor: "hover.borderColor",
  HoverColor: "hover.color",
  HoverEffect: "hoverEffect",
  HoverTransparent: "hover.transparent",
  HoverZoom: "hover.zoom",
  Href: "href",
  HtmlTag: "htmlTag",
  Icon: "icon",
  IconConfig: "iconConfig",
  IconStyle: "iconStyle",
  Id: "id",
  Image: "image",
  ImageAlt: "imageAlt",
  ImageConfig: "imageConfig",
  Images: "images",
  ImageStyle: "imageStyle",
  ImageUrl: "imageUrl",
  InPage: "inPage",
  Inputs: "inputs",
  InputStyle: "inputStyle",
  Interval: "interval",
  IsMainPopup: "isMainPopup",
  ItemData: "itemData",
  ItemKey: "itemKey",
  Items: "items",
  ItemSettings: "itemSettings",
  ItemsPerRow: "itemsPerRow",
  ItemStyle: "itemStyle",
  JustifyContent: "justifyContent",
  Key: "key",
  Label: "label",
  LabelGap: "labelGap",
  LabelPosition: "labelPosition",
  LabelStyle: "labelStyle",
  Layout: "layout",
  Left: "left",
  LetterSpacing: "letterSpacing",
  Limit: "limit",
  LineHeight: "lineHeight",
  Link: "link",
  Logo: "logo",
  LogoConfig: "logoConfig",
  LogoDesktop: "logoDesktop",
  LogoMobile: "logoMobile",
  Loop: "loop",
  MainBanner: "mainBanner",
  MainColor: "mainColor",
  MainConnection: "mainConnection",
  Map: "map",
  Margin: "margin",
  MaskClosable: "maskClosable",
  MaskStyle: "maskStyle",
  MaxHeight: "maxHeight",
  MaxWidth: "maxWidth",
  Md: "md.flex",
  Menu: "menu",
  MenuConfig: "menuConfig",
  MenuItemConfig: "menuItemConfig",
  MenuItemId: "menuItemId",
  MenuZone: "menuZone",
  Method: "method",
  MinHeight: "minHeight",
  MinHeightUnit: "minHeightUnit",
  MinWidth: "minWidth",
  Mobile: "mobile",
  Multiple: "multiple",
  MyColors: "colors",
  Name: "name",
  NameStyle: "nameStyle",
  NavigationColor: "navigationColor",
  NavigationStyle: "navigationStyle",
  Number: "number",
  NumberItemPerRow: "numberItemPerRow",
  NumberPosition: "numberPosition",
  NumberStyle: "numberStyle",
  NumberType: "numberType",
  ObjectFit: "objectFit",
  OpenNewTab: "openNewTab",
  OpenPageOfWebsite: "openPageOfWebsite",
  Options: "options",
  OptionStyle: "optionStyle",
  OrderBy: "orderBy",
  Origin: "origin",
  OutOfMove: "outOfMove",
  OutOfPrize: "outOfPrize",
  OutOfTurn: "outOfTurn",
  Padding: "padding",
  PageMaxWidth: "pageMaxWidth",
  PageWidthType: "pageWidthType",
  PaginationColor: "paginationColor",
  PaginationPosition: "paginationPosition",
  PaginationStyle: "paginationStyle",
  ParagraphEditor: "paragraphEditor",
  PauseOnHover: "pauseOnHover",
  PauseOnMouseEnter: "pauseOnMouseEnter",
  Payload: "payload",
  Percent: "percent",
  Percentage: "percentage",
  PerpendicularPrize: "perpendicularPrize",
  Phone: "phone",
  Placeholder: "placeholder",
  Popup: "popup",
  PopupButton: "popupButton",
  Position: "position",
  PositionType: "positionType",
  PrefixNumber: "prefixNumber",
  PriceColor: "priceColor",
  PrimaryColor: "primaryColor",
  Prize: "prize",
  PrizeAngle: "prizeAngle",
  Radius: "radius",
  RedirectUrl: "redirectUrl",
  RegionName: "regionName",
  Request: "request",
  RequestKey: "requestKey",
  Required: "required",
  Response: "response",
  ResponseType: "responseType",
  ResponsiveSettings: "responsiveSettings",
  Rewind: "rewind",
  Right: "right",
  RightBanners: "rightBanners",
  RightItems: "rightItems",
  Rotate: "rotate",
  Roulette: "roulette",
  Row: "row",
  RowGap: "rowGap",
  RuleRepeat: "ruleRepeat",
  ScriptCustom: "scriptCustom",
  SectionMarginTop: "sectionMarginTop",
  Shadow: "shadow",
  SheetResponsePopup: "sheetResponsePopup",
  Show: "show",
  ShowAlt: "showAlt",
  ShowCloseBtn: "showCloseBtn",
  ShowCloseIcon: "showCloseIcon",
  ShowLabel: "showLabel",
  ShowMegaMenu: "showMegaMenu",
  ShowThumbnail: "showThumbnail",
  ShowTurn: "showTurn",
  SiteKey: "siteKey",
  Size: "size",
  Skin: "skin",
  SlidePerColumn: "slidesPerColumn",
  SlidePerView: "slidesPerView",
  SlidePerViewMobile: "slidesPerViewMobile",
  Slug: "slug",
  SortBy: "sortBy",
  Spacing: "spacing",
  SpacingWithMenu: "spacingWithMenu",
  Speed: "speed",
  SpinTrigger: "spinTrigger",
  SpinTurn: "spinTurn",
  Src: "src",
  StartTime: "startTime",
  StickyTop: "stickyTop",
  Stores: "stores",
  StrokeSize: "strokeSize",
  Style: "style",
  SubmitButton: "submitButton",
  Submits: "submits",
  SubmitTime: "submitTime",
  Success: "success",
  SuffixNumber: "suffixNumber",
  SyncDeviceStyle: "syncDeviceStyle",
  Tabs: "tabs",
  TagId: "tagId",
  Target: "target",
  TekoTrackingEvent: "tekoTrackingEvent",
  TerminalCode: "terminalCode",
  Text: "text",
  TextAfter: "textAfter",
  TextAlign: "textAlign",
  TextBefore: "textBefore",
  TextConfig: "textConfig",
  TextDecoration: "textDecoration",
  TextShadow: "textShadow",
  TextStyle: "textStyle",
  TextTransform: "textTransform",
  TimeUnitStyle: "timeUnitStyle",
  TimeVisibility: "timeVisibility",
  Title: "title",
  TitleStyle: "titleStyle",
  Toggle: "toggle",
  TogglePopup: "togglePopup",
  TogglePopupList: "togglePopupList",
  Top: "top",
  TopBanner: "topBanner",
  TopBarMenu: "topBarMenu",
  Total: "total",
  TrackingEventName: "trackingEventName",
  TriggerUpdate: "triggerUpdate",
  Type: "type",
  Unit: "unit",
  UploadSrc: "uploadSrc",
  Url: "url",
  Valid: "valid",
  Value: "value",
  Version: "version",
  VerticalAlignment: "verticalAlignment",
  ViewMoreStyle: "viewMoreStyle",
  ViewMoreUrl: "viewMoreUrl",
  Weight: "weight",
  Width: "width",
  WidthType: "widthType",
  WidthUnit: "widthUnit",
  WorkHours: "workHours",
  WrapperStyle: "wrapperStyle",
  Xs: "xs.flex",
  ZIndex: "zIndex",
  Zoom: "zoom",
  ZoomPreview: "zoomPreview",
  joinKey: (...args) => args.filter(Boolean).join(".")
};

export { PROP_KEY };
