// @tempi/core-elements/dist/configs/Swiper/SwiperSettingV2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var icons = require('@ant-design/icons');
var iconsDev = require('@tempi/icons-dev');
var coreEditor = require('@tempi/core-editor');
var antd = require('antd');
var ButtonGroupSwiper = require('./ButtonGroupSwiper.js');
var propKey = require('../../constants/propKey.js');

const rowNumberConfig = [
  {
    icon: /* @__PURE__ */ jsxRuntime.jsx(iconsDev.AntdCustomIcon, {
      name: "PbSliderOneRowFilled",
      style: { fontSize: 30 }
    }),
    value: 1
  },
  {
    icon: /* @__PURE__ */ jsxRuntime.jsx(iconsDev.AntdCustomIcon, {
      name: "PbSliderTwoRowFilled",
      style: { fontSize: 30 }
    }),
    value: 2
  },
  {
    icon: /* @__PURE__ */ jsxRuntime.jsx(iconsDev.AntdCustomIcon, {
      name: "PbSliderThreeRowFilled",
      style: { fontSize: 30 }
    }),
    value: 3
  }
];
const arrowStyleConfig = [
  {
    icon: /* @__PURE__ */ jsxRuntime.jsx(iconsDev.AntdCustomIcon, {
      name: "PbNavigationCurvedFilled",
      style: { fontSize: 20 }
    }),
    value: 1
  },
  {
    icon: /* @__PURE__ */ jsxRuntime.jsx(iconsDev.AntdCustomIcon, {
      name: "PbNavigationRoundFilled",
      style: { fontSize: 20 }
    }),
    value: 2
  },
  { icon: /* @__PURE__ */ jsxRuntime.jsx(icons.LeftOutlined, {
    style: { fontSize: 20 }
  }), value: 3 }
];
const paginationStyleConfig = [
  {
    icon: /* @__PURE__ */ jsxRuntime.jsx(iconsDev.AntdCustomIcon, {
      name: "PbPaginationDotsFilled",
      style: { fontSize: 48 }
    }),
    value: 1
  },
  {
    icon: /* @__PURE__ */ jsxRuntime.jsx(iconsDev.AntdCustomIcon, {
      name: "PbPaginationDotLineFilled",
      style: { fontSize: 36 }
    }),
    value: 2
  },
  {
    icon: /* @__PURE__ */ jsxRuntime.jsx(iconsDev.AntdCustomIcon, {
      name: "PbPaginationDotBigFilled",
      style: { fontSize: 48 }
    }),
    value: 3
  },
  {
    icon: /* @__PURE__ */ jsxRuntime.jsx(iconsDev.AntdCustomIcon, {
      name: "PbPaginationLineFilled",
      style: { fontSize: 36 }
    }),
    value: 4
  }
];
const SwiperSettingV2 = ({
  propKey: propKey$1
}) => {
  const [slidePerRow, setSlidePerRow] = coreEditor.useProp(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.GridRows)
  );
  const [arrowStyle, setArrowStyle] = coreEditor.useProp(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.ArrowStyle)
  );
  const [paginationStyle, setPaginationStyle] = coreEditor.useProp(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.PaginationStyle)
  );
  const [autoplay] = coreEditor.useProp(propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.Autoplay));
  const [interval, setInterval] = coreEditor.useProp(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.Interval)
  );
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
        label: i18n.t("S\u1ED1 h\xE0ng"),
        children: /* @__PURE__ */ jsxRuntime.jsx(ButtonGroupSwiper.ButtonGroupSwiper, {
          configValue: slidePerRow,
          setConfigValue: setSlidePerRow,
          configuration: rowNumberConfig
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
        propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.GridCols),
        inputNumberProps: { min: 1 },
        formItemProps: {
          label: i18n.t("S\u1ED1 th\xE0nh ph\u1EA7n tr\xEAn 1 h\xE0ng")
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
        label: i18n.t("Ki\u1EC3u m\u0169i t\xEAn"),
        layout: "vertical",
        children: /* @__PURE__ */ jsxRuntime.jsx(ButtonGroupSwiper.ButtonGroupSwiper, {
          configValue: arrowStyle,
          setConfigValue: setArrowStyle,
          configuration: arrowStyleConfig
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
        label: i18n.t("Ki\u1EC3u ph\xE2n trang"),
        layout: "vertical",
        children: /* @__PURE__ */ jsxRuntime.jsx(ButtonGroupSwiper.ButtonGroupSwiper, {
          configValue: paginationStyle,
          setConfigValue: setPaginationStyle,
          configuration: paginationStyleConfig
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
        propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.PaginationPosition),
        options: [
          { label: i18n.t("Ph\xEDa trong"), value: "inside" },
          { label: i18n.t("Ph\xEDa ngo\xE0i"), value: "outside" }
        ],
        formItemProps: {
          label: i18n.t("V\u1ECB tr\xED ph\xE2n trang")
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
        propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.PaginationColor),
        formItemProps: {
          label: i18n.t("M\xE0u ph\xE2n trang")
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
        propKey: "crop",
        formItemProps: {
          label: i18n.t("C\u1EAFt \u1EA3nh")
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
        propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.Rewind),
        formItemProps: {
          label: i18n.t("L\u1EB7p l\u1EA1i")
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
        propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.Autoplay),
        formItemProps: {
          label: i18n.t("T\u1EF1 \u0111\u1ED9ng tr\u01B0\u1EE3t")
        }
      }),
      autoplay && /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
            propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.PauseOnHover),
            formItemProps: {
              label: i18n.t("D\u1EEBng chu\u1ED9t khi di chu\u1ED9t")
            }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
            label: i18n.t("T\u1ED1c \u0111\u1ED9 (gi\xE2y)"),
            children: /* @__PURE__ */ jsxRuntime.jsx(antd.InputNumber, {
              style: { width: 120 },
              value: interval / 1e3,
              onChange: (val) => val && setInterval(Number(val) * 1e3)
            })
          })
        ]
      })
    ]
  });
};

exports.SwiperSettingV2 = SwiperSettingV2;
