// @tempi/core-editor/dist/components/ColorPickerModal/ColorPickerModalV2.es.js
import { jsxs, jsx, Fragment } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { Col, Modal, Button, Row } from 'antd';
import get from 'lodash.get';
import set from 'lodash.set';
import { useState, useEffect } from 'react';
import { GradientType, ColorType, ROOT, parseColor, stringifyColor, getRgbaFromHex } from '@tempi/core-renderer';
import { HexAlphaColorPicker, HexColorInput } from 'react-colorful';
import styled from '@emotion/styled';
import { useFrame } from 'react-frame-component';
import { sampleColors } from './constants.es.js';
import GradientBar from './GradientBar.es.js';
import { getHexColor, rgbaStringToObject, rgbaToString } from '../../utils/color.es.js';
import { SelectConfig } from '../../configs/SelectConfig/SelectConfig.es.js';
import { InputNumberConfig } from '../../configs/InputNumberConfig/InputNumberConfig.es.js';
import { useEditor } from '../../hooks/useEditor.es.js';
import { PROP_KEY } from '../../constants/propKey.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const gradientTypeOptions = [
  { label: "Linear", value: GradientType.linear },
  { label: "Radial", value: GradientType.radial }
];
const getRGBAColor = (color) => {
  var _a;
  if (!color)
    return null;
  let rgbaColor = color;
  if (/^#.+$/.test(color)) {
    rgbaColor = getRgbaFromHex(color);
  }
  const rgbaObject = (_a = rgbaStringToObject(rgbaColor)) == null ? void 0 : _a.rgba;
  return rgbaToString(rgbaObject);
};
const colorTypeOptions = [
  { label: "Solid", value: ColorType.solid },
  { label: "Gradient", value: ColorType.gradient }
];
const ColorPickerModalV2 = ({
  color = stringifyColor({
    colorType: ColorType.solid,
    colorValue: "rgba(255,255,255,1)"
  }),
  visible,
  setVisible,
  onChange,
  hasGradientColor
}) => {
  const {
    actions: { history },
    myColors
  } = useEditor((state) => {
    var _a;
    const rootNode = state.nodes[ROOT];
    return {
      myColors: get((_a = rootNode == null ? void 0 : rootNode.data) == null ? void 0 : _a.props, PROP_KEY.MyColors, [])
    };
  });
  const { document: frameDocument } = useFrame();
  const [colorType, setColorType] = useState();
  const [selectedGradientItem, setSelectedGradientItem] = useState();
  const [gradientColor, setGradientColor] = useState();
  const [hexInput, setHexInput] = useState();
  const getCurrentColorValue = () => {
    if (colorType === ColorType.gradient)
      return gradientColor;
    if (!hexInput)
      return "";
    return getRgbaFromHex(hexInput).replaceAll(" ", "");
  };
  const currentColorValue = getCurrentColorValue();
  useEffect(() => {
    var _a, _b, _c;
    if (visible) {
      const colorObjInit = parseColor(color);
      const initColorType = (colorObjInit == null ? void 0 : colorObjInit.colorType) || ColorType.solid;
      const initColorValue = (colorObjInit == null ? void 0 : colorObjInit.colorValue) || "rgba(255,255,255,1)";
      const initSelectedGradientItem = 0;
      const isGradientColor = initColorType === ColorType.gradient;
      const initGradientColor = isGradientColor ? initColorValue : void 0;
      setColorType(initColorType);
      setGradientColor(initGradientColor);
      setSelectedGradientItem(initSelectedGradientItem);
      const initCurrentColor = ((_c = isGradientColor ? (_b = (_a = initColorValue == null ? void 0 : initColorValue.colors) == null ? void 0 : _a[initSelectedGradientItem]) == null ? void 0 : _b.color : initColorValue) == null ? void 0 : _c.toString()) || "";
      setHexInput(getHexColor(initCurrentColor));
    }
  }, [visible]);
  useEffect(() => {
    var _a, _b;
    if (colorType === ColorType.gradient && selectedGradientItem !== void 0) {
      setHexInput(
        getHexColor((_b = (_a = gradientColor == null ? void 0 : gradientColor.colors) == null ? void 0 : _a[selectedGradientItem]) == null ? void 0 : _b.color)
      );
    }
  }, [selectedGradientItem]);
  const onChangeGradientType = (value) => {
    if (value === GradientType.linear) {
      onChangeGradient("radius", 90);
    }
    onChangeGradient("gradientType", value);
  };
  const onChangeGradientRadius = (e) => {
    const radius = !e.target.value ? 0 : parseInt(e.target.value);
    onChangeGradient("radius", radius);
  };
  const onChangeColorType = (colorType2) => {
    var _a, _b;
    if (colorType2 === ColorType.solid) {
      setGradientColor(void 0);
      setHexInput(getHexColor("rgba(255,255,255,1)"));
    } else {
      const defaultGradientColor = {
        colors: [
          { color: "rgb(254,214,227)", stop: 0 },
          { color: "rgba(168,237,234)", stop: 100 }
        ],
        radius: 90,
        gradientType: GradientType.linear
      };
      setGradientColor(defaultGradientColor);
      setHexInput(getHexColor((_b = (_a = defaultGradientColor == null ? void 0 : defaultGradientColor.colors) == null ? void 0 : _a[0]) == null ? void 0 : _b.color));
    }
    setColorType(colorType2);
    setSelectedGradientItem(0);
  };
  const onChangeColor = (hexColor) => {
    setHexInput(hexColor);
    if (colorType === ColorType.gradient && gradientColor && hexColor) {
      const clone = JSON.parse(JSON.stringify(gradientColor.colors));
      clone[selectedGradientItem].color = getRgbaFromHex(hexColor).replaceAll(
        " ",
        ""
      );
      onChangeGradient("colors", clone);
    }
  };
  const renderSquarePicker = (label, colors) => {
    if (!(colors == null ? void 0 : colors.length))
      return null;
    return /* @__PURE__ */ jsxs(Fragment, {
      children: [
        /* @__PURE__ */ jsx(Label, {
          children: label
        }),
        /* @__PURE__ */ jsx(CircleColorPicker, {
          children: colors.map((color2, index) => {
            var _a;
            const rgbaColor = getRGBAColor(color2);
            const hexColor = (_a = rgbaStringToObject(rgbaColor)) == null ? void 0 : _a.hex;
            return /* @__PURE__ */ jsx(ColorSquare, {
              title: hexColor,
              style: { backgroundColor: color2 },
              onClick: () => {
                onChangeColor(hexColor);
              }
            }, index);
          })
        })
      ]
    });
  };
  const onClose = () => setVisible(false);
  const onOk = () => {
    const selectedColor = {
      colorType,
      colorValue: currentColorValue
    };
    onChange(stringifyColor(selectedColor));
    if (colorType === ColorType.solid) {
      let newMyColors = [...myColors];
      const isExistInMyColors = myColors.some(
        (color2) => getRGBAColor(color2) === currentColorValue
      );
      if (!isExistInMyColors) {
        if (myColors.length === 10) {
          newMyColors = [...myColors.slice(1), currentColorValue];
        } else
          newMyColors.push(currentColorValue);
        history.ignore().setProp(ROOT, (prop) => {
          set(prop, PROP_KEY.MyColors, newMyColors);
        });
      }
    }
    onClose();
  };
  const onChangeGradient = (name, value) => {
    setGradientColor((prev) => __spreadProps(__spreadValues({}, prev), {
      [name]: Array.isArray(value) ? [...value] : value
    }));
  };
  return /* @__PURE__ */ jsxs(StyledModal, {
    open: visible,
    title: t("Ch\u1ECDn m\xE0u"),
    onCancel: onClose,
    width: 300,
    bodyStyle: { padding: "8px 8px" },
    zIndex: 9999,
    destroyOnClose: true,
    getContainer: () => (frameDocument == null ? void 0 : frameDocument.body) || (document == null ? void 0 : document.body),
    footer: /* @__PURE__ */ jsxs(Footer, {
      children: [
        /* @__PURE__ */ jsx(ResetButton, {
          type: "link",
          onClick: () => {
            onChange(void 0);
            onClose();
          },
          children: t("\u0110\u1EB7t l\u1EA1i")
        }),
        /* @__PURE__ */ jsxs("div", {
          style: { display: "flex" },
          children: [
            /* @__PURE__ */ jsx(StyledButton, {
              onClick: onClose,
              children: t("common.cancel")
            }),
            /* @__PURE__ */ jsx(StyledButton, {
              type: "primary",
              onClick: onOk,
              children: "OK"
            })
          ]
        })
      ]
    }),
    children: [
      /* @__PURE__ */ jsxs(Header, {
        children: [
          hasGradientColor && /* @__PURE__ */ jsx(SelectConfig, {
            selectProps: {
              allowClear: false,
              value: colorType,
              options: colorTypeOptions,
              onChange: onChangeColorType,
              getPopupContainer: (trigger) => trigger
            }
          }),
          /* @__PURE__ */ jsx(TransparentBackground, {
            children: /* @__PURE__ */ jsx(ColorDiv, {
              colorConfig: { colorType, colorValue: currentColorValue }
            })
          })
        ]
      }),
      /* @__PURE__ */ jsx(StyledColorPicker, {
        color: hexInput,
        onChange: onChangeColor
      }),
      colorType === ColorType.gradient && /* @__PURE__ */ jsx(GradientBar, {
        gradientColor,
        selectedGradientItem,
        setSelectedGradientItem,
        onChange: onChangeGradient
      }),
      /* @__PURE__ */ jsxs(StyledRow, {
        gutter: [8, 8],
        className: "mt-base",
        children: [
          /* @__PURE__ */ jsx(Col, {
            span: 11,
            children: /* @__PURE__ */ jsx(StyledHexColorInput, {
              alpha: true,
              prefixed: true,
              color: hexInput,
              onChange: onChangeColor
            })
          }),
          colorType === ColorType.gradient && /* @__PURE__ */ jsxs(Fragment, {
            children: [
              /* @__PURE__ */ jsx(Col, {
                span: 8,
                children: /* @__PURE__ */ jsx(SelectConfig, {
                  selectProps: {
                    allowClear: false,
                    value: gradientColor == null ? void 0 : gradientColor.gradientType,
                    options: gradientTypeOptions,
                    onChange: onChangeGradientType,
                    getPopupContainer: (trigger) => trigger
                  }
                })
              }),
              /* @__PURE__ */ jsxs(Col, {
                span: 5,
                children: [
                  /* @__PURE__ */ jsx(StyledNumberInput, {
                    formItemProps: {
                      onChange: onChangeGradientRadius,
                      value: gradientColor == null ? void 0 : gradientColor.radius
                    },
                    inputNumberProps: {
                      disabled: (gradientColor == null ? void 0 : gradientColor.gradientType) === GradientType.radial,
                      min: 0,
                      max: 360,
                      value: gradientColor == null ? void 0 : gradientColor.radius
                    }
                  }),
                  (gradientColor == null ? void 0 : gradientColor.gradientType) === GradientType.linear && /* @__PURE__ */ jsx(Degree, {
                    children: "\xB0"
                  })
                ]
              })
            ]
          })
        ]
      }),
      colorType === ColorType.solid && renderSquarePicker(t("M\xE0u c\u1EE7a t\xF4i"), myColors),
      renderSquarePicker(t("M\xE0u m\u1EABu"), sampleColors)
    ]
  });
};
const Label = styled.div`
  font-size: 12px;
  font-weight: 500;
  margin: 8px 0 4px;
`;
const StyledModal = styled(Modal)`
  &.ant-modal {
    top: 80px;
  }

  .ant-modal {
    &-header,
    &-footer {
      padding: 8px;
      border-radius: 8px;
    }

    &-title {
      font-size: 14px;
    }

    &-close-x {
      width: 32px;
      height: 32px;
      line-height: 32px;
      font-size: 14px;
    }
    &-content {
      border-radius: 8px;
      padding: 8px;
    }
  }
`;
const StyledColorPicker = styled(HexAlphaColorPicker)`
  width: 100% !important;
  margin-bottom: 16px;

  .react-colorful {
    &__pointer {
      width: 16px;
      height: 16px;
    }

    &__alpha,
    &__hue {
      height: 12px;
      margin-top: 12px;
      border-radius: 8px;
    }
  }
`;
const StyledHexColorInput = styled(HexColorInput)`
  width: 100%;
  padding: 4px;
  text-align: center;
  border: 1px solid var(--border-color);
  border-radius: 4px;
  &:focus {
    outline: none;
  }
`;
const CircleColorPicker = styled.div`
  display: flex;
  flex-wrap: wrap;
  row-gap: 8px;
  column-gap: 8px;
`;
const ColorSquare = styled.div`
  width: 25px;
  height: 25px;
  border-radius: 6px;
  transform: scale(1);
  transition: all 100ms ease 0s;
  cursor: pointer;

  &:hover {
    transform: scale(1.2);
    border: 2px solid #d9d9d9;
  }
`;
const Footer = styled.div`
  display: flex;
  justify-content: space-between;
`;
const ResetButton = styled(Button)`
  color: var(--text-color);
  padding-left: 0;
  padding-right: 0;
`;
const Header = styled.div`
  height: 32px;
  display: flex;
  justify-content: space-between;
  margin-bottom: 8px;
`;
const TransparentBackground = styled.div`
  border-radius: 4px;
  background-image: url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill-opacity=".05"><path d="M8 0h8v8H8zM0 8h8v8H0z"/></svg>');
`;
const ColorDiv = styled.div(
  ({ colorConfig }) => `
    width: 32px;
    height: 32px;
    border-radius: 4px;
    margin-left: auto;
    background: ${stringifyColor(colorConfig)};
    border: 1px solid rgba(0, 0, 0, 0.05);
`
);
const StyledNumberInput = styled(InputNumberConfig)`
  .ant-input-number-handler-wrap {
    display: none;
  }
`;
const Degree = styled.div`
  font-size: 24px;
  position: absolute;
  top: -18px;
  right: -8px;
`;
const StyledRow = styled(Row)`
  margin-top: 16px;
`;
const StyledButton = styled(Button)`
  border-radius: 6px;
`;

export { ColorPickerModalV2 };
