// @tempi/core-editor/dist/components/ColorPickerModal/RgbaField.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var styled = require('@emotion/styled');
var EditableInput = require('./EditableInput.js');
var style = require('./style.js');
var color = require('../../utils/color.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const RgbaField = ({
  color: color$1,
  onChange,
  className
}) => {
  var _a;
  const rgbaObject = (_a = color.rgbaStringToObject(color$1)) == null ? void 0 : _a.rgba;
  const renderRGBAInputs = () => {
    return ["r", "g", "b", "a"].map((key) => /* @__PURE__ */ jsxRuntime.jsxs("div", {
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(EditableInput.EditableInput, {
          value: (rgbaObject == null ? void 0 : rgbaObject[key]) || null,
          onChange: (value) => {
            onChange(
              color.rgbaToString(__spreadProps(__spreadValues({}, rgbaObject), {
                [key]: value
              }))
            );
          },
          min: 0,
          max: key === "a" ? 1 : 255,
          precision: key === "a" ? 2 : void 0
        }),
        /* @__PURE__ */ jsxRuntime.jsx(StyledLabel, {
          children: key
        })
      ]
    }, key));
  };
  return /* @__PURE__ */ jsxRuntime.jsx("div", {
    className,
    style: { display: "flex", columnGap: 4 },
    children: renderRGBAInputs()
  });
};
const StyledLabel = styled__default["default"].label(style.labelCss);

exports.RgbaField = RgbaField;
