// @tempi/core-elements/dist/quick-configs/ColorPickerQuickSetting/ColorPickerQuickSetting.es.js
import { jsx, jsxs } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { Color16Regular } from '@fluentui/react-icons';
import { SubQuickSetting, QuickSettingButton, ColorPickerConfig, BorderConfig } from '@tempi/core-editor';
import { Tabs } from 'antd';
import { QuickSettingTabWrapper } from '../../core/QuickSettingTabWrapper/QuickSettingTabWrapper.es.js';
import { ShadowConfig } from '../../configs/ShadowConfig/ShadowConfig.es.js';
import { PROP_KEY } from '../../constants/propKey.es.js';

const ColorPickerQuickSetting = ({
  propKey = PROP_KEY.CommonStyle,
  track,
  gradientPicker = false
}) => {
  const content = () => {
    return /* @__PURE__ */ jsxs(QuickSettingTabWrapper, {
      children: [
        /* @__PURE__ */ jsx(Tabs.TabPane, {
          tab: t("N\u1EC1n"),
          children: /* @__PURE__ */ jsx(ColorPickerConfig, {
            propKey: PROP_KEY.joinKey(propKey, PROP_KEY.BackgroundColor),
            formItemProps: {
              label: t("M\xE0u n\u1EC1n")
            },
            gradientPicker
          })
        }, "1"),
        /* @__PURE__ */ jsx(Tabs.TabPane, {
          tab: t("Vi\u1EC1n"),
          children: /* @__PURE__ */ jsx(BorderConfig, {
            propKey
          })
        }, "2"),
        /* @__PURE__ */ jsx(Tabs.TabPane, {
          tab: t("B\xF3ng"),
          children: /* @__PURE__ */ jsx(ShadowConfig, {
            propKey: PROP_KEY.joinKey(propKey, PROP_KEY.Shadow)
          })
        }, "3")
      ]
    });
  };
  return /* @__PURE__ */ jsx(SubQuickSetting, {
    content,
    children: /* @__PURE__ */ jsx(QuickSettingButton, {
      onClick: () => track == null ? void 0 : track(),
      tooltip: t("M\xE0u n\u1EC1n"),
      children: /* @__PURE__ */ jsx(Color16Regular, {})
    })
  });
};

export { ColorPickerQuickSetting };
