// @tempi/core-elements/dist/quick-configs/ColorPickerQuickSetting/ColorPickerQuickSetting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var reactIcons = require('@fluentui/react-icons');
var coreEditor = require('@tempi/core-editor');
var antd = require('antd');
var QuickSettingTabWrapper = require('../../core/QuickSettingTabWrapper/QuickSettingTabWrapper.js');
var ShadowConfig = require('../../configs/ShadowConfig/ShadowConfig.js');
var propKey = require('../../constants/propKey.js');

const ColorPickerQuickSetting = ({
  propKey: propKey$1 = propKey.PROP_KEY.CommonStyle,
  track,
  gradientPicker = false
}) => {
  const content = () => {
    return /* @__PURE__ */ jsxRuntime.jsxs(QuickSettingTabWrapper.QuickSettingTabWrapper, {
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(antd.Tabs.TabPane, {
          tab: i18n.t("N\u1EC1n"),
          children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
            propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.BackgroundColor),
            formItemProps: {
              label: i18n.t("M\xE0u n\u1EC1n")
            },
            gradientPicker
          })
        }, "1"),
        /* @__PURE__ */ jsxRuntime.jsx(antd.Tabs.TabPane, {
          tab: i18n.t("Vi\u1EC1n"),
          children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.BorderConfig, {
            propKey: propKey$1
          })
        }, "2"),
        /* @__PURE__ */ jsxRuntime.jsx(antd.Tabs.TabPane, {
          tab: i18n.t("B\xF3ng"),
          children: /* @__PURE__ */ jsxRuntime.jsx(ShadowConfig.ShadowConfig, {
            propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.Shadow)
          })
        }, "3")
      ]
    });
  };
  return /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SubQuickSetting, {
    content,
    children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.QuickSettingButton, {
      onClick: () => track == null ? void 0 : track(),
      tooltip: i18n.t("M\xE0u n\u1EC1n"),
      children: /* @__PURE__ */ jsxRuntime.jsx(reactIcons.Color16Regular, {})
    })
  });
};

exports.ColorPickerQuickSetting = ColorPickerQuickSetting;
