// @tempi/core-elements/dist/configs/TextConfig/TextConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var coreRenderer = require('@tempi/core-renderer');
var antd = require('antd');
var get = require('lodash.get');
var fonts = require('../../constants/fonts.js');
var propKey = require('../../constants/propKey.js');
var ShadowConfig = require('../ShadowConfig/ShadowConfig.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var get__default = /*#__PURE__*/_interopDefaultLegacy(get);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const { Panel } = antd.Collapse;
const TextConfig = ({
  propKey: propKey$1 = "",
  config,
  title
}) => {
  var _a, _b;
  const { rootNode } = coreEditor.useEditor((state) => ({
    rootNode: state.nodes[coreRenderer.ROOT]
  }));
  const { realDevice } = coreEditor.useBuilder();
  const fontFamilyKey = propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.FontFamily);
  const [fontFamily, setFontFamily] = coreEditor.useProp(fontFamilyKey);
  return /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
    layout: "vertical",
    label: !title ? i18n.t("Ki\u1EC3u ch\u1EEF") : title,
    children: /* @__PURE__ */ jsxRuntime.jsxs(antd.Collapse, {
      defaultActiveKey: [],
      children: [
        /* @__PURE__ */ jsxRuntime.jsxs(Panel, {
          header: i18n.t("Ph\xF4ng ch\u1EEF"),
          children: [
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
              label: i18n.t("Ph\xF4ng ch\u1EEF"),
              children: /* @__PURE__ */ jsxRuntime.jsx(antd.Select, __spreadProps(__spreadValues({}, {
                virtual: false,
                style: {
                  width: 140
                },
                showSearch: true,
                placeholder: i18n.t("Ph\xF4ng ch\u1EEF"),
                value: fontFamily || get__default["default"](
                  (_b = (_a = rootNode == null ? void 0 : rootNode.data) == null ? void 0 : _a.props) == null ? void 0 : _b[realDevice],
                  propKey.PROP_KEY.joinKey(propKey.PROP_KEY.CommonStyle, propKey.PROP_KEY.FontFamily)
                ),
                onChange: setFontFamily,
                getPopupContainer: (trigger) => trigger,
                allowClear: true
              }), {
                children: fonts.fontFamilyOptions.map((el) => /* @__PURE__ */ jsxRuntime.jsx(antd.Select.Option, {
                  value: el.value,
                  children: /* @__PURE__ */ jsxRuntime.jsx("span", {
                    style: { fontFamily: el.value },
                    children: el.label
                  })
                }, el.value))
              }))
            }),
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
              propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.FontWeight),
              options: fonts.fontWeightOptions,
              formItemProps: {
                label: i18n.t("\u0110\u1ED9 d\xE0y ph\xF4ng ch\u1EEF")
              },
              selectProps: {
                allowClear: true,
                placeholder: i18n.t("M\u1EB7c \u0111\u1ECBnh")
              }
            }),
            (config == null ? void 0 : config.fontSize) && /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
              propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.FontSize),
              formItemProps: __spreadValues({
                label: i18n.t("C\u1EE1 ch\u1EEF")
              }, config.isCountdownTextSetting && {
                tooltip: i18n.t(
                  "\u1EDE mobile, c\u1EE1 ch\u1EEF >30px \u0111\u01B0\u1EE3c t\u1EF1 \u0111\u1ED9ng gi\u1EA3m xu\u1ED1ng 30px \u0111\u1EC3 \u0111\u1EA3m b\u1EA3o hi\u1EC3n th\u1ECB"
                )
              }),
              unitProps: { defaultValue: coreRenderer.UnitEnum.pixel }
            }),
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
              propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.LineHeight),
              formItemProps: {
                label: i18n.t("Kho\u1EA3ng c\xE1ch d\xF2ng")
              },
              unitProps: { defaultValue: coreRenderer.UnitEnum.pixel }
            }),
            (config == null ? void 0 : config.textAlign) && /* @__PURE__ */ jsxRuntime.jsx(coreEditor.AlignmentConfig, {
              propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.TextAlign),
              buttonGroupProps: {
                listAlignment: ["left", "center", "right", "justify"]
              },
              formItemProps: {
                label: i18n.t("C\u0103n ch\u1EC9nh ch\u1EEF")
              }
            })
          ]
        }, "FONT"),
        /* @__PURE__ */ jsxRuntime.jsxs(Panel, {
          header: i18n.t("Ki\u1EC3u & Kho\u1EA3ng c\xE1ch"),
          children: [
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
              propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.LetterSpacing),
              formItemProps: {
                label: i18n.t("Kho\u1EA3ng c\xE1ch")
              },
              unitProps: { defaultValue: coreRenderer.UnitEnum.pixel }
            }),
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
              propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.TextTransform),
              options: fonts.textTransformOptions,
              formItemProps: {
                label: i18n.t("Thay \u0111\u1ED5i")
              },
              selectProps: {
                allowClear: true,
                placeholder: i18n.t("M\u1EB7c \u0111\u1ECBnh")
              }
            }),
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
              propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.TextDecoration),
              options: fonts.textDecorationOptions,
              formItemProps: { label: i18n.t("Trang tr\xED") },
              selectProps: {
                allowClear: true,
                placeholder: i18n.t("M\u1EB7c \u0111\u1ECBnh")
              }
            }),
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
              propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.FontStyle),
              options: fonts.fontStyleOptions,
              formItemProps: { label: i18n.t("Ki\u1EC3u") },
              selectProps: {
                allowClear: true,
                placeholder: i18n.t("M\u1EB7c \u0111\u1ECBnh")
              }
            }),
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
              propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.FontVariant),
              options: fonts.fontVariantOptions,
              formItemProps: { label: i18n.t("Bi\u1EBFn th\u1EC3") },
              selectProps: {
                allowClear: true,
                placeholder: i18n.t("M\u1EB7c \u0111\u1ECBnh")
              }
            })
          ]
        }, "STYLE"),
        (config == null ? void 0 : config.textShadow) && /* @__PURE__ */ jsxRuntime.jsx(Panel, {
          header: i18n.t("\u0110\u1ED5 b\xF3ng ch\u1EEF"),
          children: /* @__PURE__ */ jsxRuntime.jsx(ShadowConfig.ShadowConfig, {
            propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.TextShadow),
            shadowProps: ["x", "y", "blur"]
          })
        }, "TextShadow")
      ]
    })
  });
};

exports.TextConfig = TextConfig;
