// @tempi/core-elements/dist/configs/TextConfig/TextConfigV2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var coreRenderer = require('@tempi/core-renderer');
var antd = require('antd');
var get = require('lodash.get');
var TextInputNumberConfig = require('./TextInputNumberConfig.js');
var fonts = require('../../constants/fonts.js');
var propKey = require('../../constants/propKey.js');
var getEditorAttributeValue = require('../../utils/getEditorAttributeValue.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var get__default = /*#__PURE__*/_interopDefaultLegacy(get);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const TextConfigV2 = ({
  config = {},
  editor
}) => {
  var _a, _b, _c, _d;
  const { heading = false, responsiveFontSize = false } = config;
  const { realDevice } = coreEditor.useBuilder();
  const { rootNode } = coreEditor.useEditor((state) => ({
    rootNode: state.nodes[coreRenderer.ROOT]
  }));
  const fontFamilyKey = propKey.PROP_KEY.joinKey(
    propKey.PROP_KEY.CommonStyle,
    propKey.PROP_KEY.FontFamily
  );
  const headingAttributes = editor == null ? void 0 : editor.getAttributes("heading");
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      heading && /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
        formItemProps: { label: i18n.t("Ti\xEAu \u0111\u1EC1"), placeholder: i18n.t("M\u1EB7c \u0111\u1ECBnh") },
        selectProps: {
          value: headingAttributes == null ? void 0 : headingAttributes.level,
          onChange: (value) => {
            var _a2;
            (_a2 = editor == null ? void 0 : editor.commands) == null ? void 0 : _a2.executeCmd(
              ({ commands }) => commands.setHeading(__spreadProps(__spreadValues({}, headingAttributes), {
                level: value
              })),
              true
            );
          }
        },
        options: [...Array(6).keys()].map((_, index) => ({
          label: i18n.t("Ti\xEAu \u0111\u1EC1 {{var0}}", { var0: index + 1 }),
          value: index + 1
        }))
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
        formItemProps: {
          label: i18n.t("Ph\xF4ng ch\u1EEF")
        },
        selectProps: {
          showSearch: true,
          dropdownMatchSelectWidth: false,
          placeholder: i18n.t("Ph\xF4ng ch\u1EEF"),
          value: getEditorAttributeValue.getEditorAttributeValue(editor, propKey.PROP_KEY.FontFamily) || get__default["default"]((_b = (_a = rootNode == null ? void 0 : rootNode.data) == null ? void 0 : _a.props) == null ? void 0 : _b[realDevice], fontFamilyKey),
          onChange: (value) => {
            var _a2;
            (_a2 = editor == null ? void 0 : editor.commands) == null ? void 0 : _a2.executeCmd(
              ({ commands }) => commands.setFontFamily(value),
              true
            );
          },
          children: (_c = fonts.fontFamilyOptions) == null ? void 0 : _c.map((el) => /* @__PURE__ */ jsxRuntime.jsx(antd.Select.Option, {
            value: el.value,
            children: /* @__PURE__ */ jsxRuntime.jsx("span", {
              style: { fontFamily: el.value },
              children: el.label
            })
          }, el.value))
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
        options: fonts.fontWeightOptions,
        formItemProps: {
          label: i18n.t("\u0110\u1ED9 d\xE0y ph\xF4ng ch\u1EEF")
        },
        selectProps: {
          placeholder: i18n.t("M\u1EB7c \u0111\u1ECBnh"),
          value: (_d = getEditorAttributeValue.getEditorAttributeValue(
            editor,
            propKey.PROP_KEY.FontWeight
          )) == null ? void 0 : _d.toString(),
          onChange: (value) => {
            var _a2;
            (_a2 = editor == null ? void 0 : editor.commands) == null ? void 0 : _a2.executeCmd(
              ({ commands }) => commands.setFontWeight(value),
              true
            );
          }
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(TextInputNumberConfig.TextInputNumberConfig, {
        initValue: coreEditor.removeUnit(
          getEditorAttributeValue.getEditorAttributeValue(
            editor,
            propKey.PROP_KEY.FontSize,
            responsiveFontSize ? realDevice : void 0
          )
        ),
        formItemProps: { label: i18n.t("C\u1EE1 ch\u1EEF") },
        inputNumberProps: {
          placeholder: i18n.t("M\u1EB7c \u0111\u1ECBnh")
        },
        unitProps: { defaultValue: coreRenderer.UnitEnum.pixel },
        updateValue: (value) => {
          var _a2;
          (_a2 = editor == null ? void 0 : editor.commands) == null ? void 0 : _a2.executeCmd(
            ({ commands }) => commands.setFontSize(
              coreRenderer.unitParse(value),
              responsiveFontSize ? realDevice : void 0
            ),
            true
          );
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(TextInputNumberConfig.TextInputNumberConfig, {
        initValue: coreEditor.removeUnit(
          getEditorAttributeValue.getEditorAttributeValue(editor, propKey.PROP_KEY.LineHeight)
        ),
        formItemProps: {
          label: i18n.t("Kho\u1EA3ng c\xE1ch d\xF2ng")
        },
        inputNumberProps: {
          placeholder: i18n.t("M\u1EB7c \u0111\u1ECBnh")
        },
        unitProps: { defaultValue: coreRenderer.UnitEnum.pixel },
        updateValue: (value) => {
          var _a2;
          (_a2 = editor == null ? void 0 : editor.commands) == null ? void 0 : _a2.executeCmd(
            ({ commands }) => commands.setLineHeight(coreRenderer.unitParse(value)),
            true
          );
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(TextInputNumberConfig.TextInputNumberConfig, {
        initValue: coreEditor.removeUnit(
          coreEditor.removeUnit(getEditorAttributeValue.getEditorAttributeValue(editor, propKey.PROP_KEY.LetterSpacing))
        ),
        formItemProps: {
          label: i18n.t("Kho\u1EA3ng c\xE1ch ch\u1EEF")
        },
        inputNumberProps: {
          placeholder: i18n.t("M\u1EB7c \u0111\u1ECBnh")
        },
        unitProps: { defaultValue: coreRenderer.UnitEnum.pixel },
        updateValue: (value) => {
          var _a2;
          (_a2 = editor == null ? void 0 : editor.commands) == null ? void 0 : _a2.executeCmd(
            ({ commands }) => commands.setLetterSpacing(coreRenderer.unitParse(value)),
            true
          );
        }
      })
    ]
  });
};

exports.TextConfigV2 = TextConfigV2;
