// @tempi/core-elements/dist/quick-configs/table/TableSetting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var icons = require('@ant-design/icons');
var styled = require('@emotion/styled');
var coreEditor = require('@tempi/core-editor');
var antd = require('antd');
var utils = require('./utils.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

const TableSetting = ({ editor }) => {
  const content = () => {
    return /* @__PURE__ */ jsxRuntime.jsxs(ContentWrapper, {
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
          label: i18n.t("Ti\xEAu \u0111\u1EC1 theo h\xE0ng"),
          children: /* @__PURE__ */ jsxRuntime.jsx(antd.Switch, {
            checked: utils.isToggledHeaderRow(editor),
            onChange: () => editor.chain().focus().toggleHeaderRow().run()
          })
        }),
        /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
          label: i18n.t("Ti\xEAu \u0111\u1EC1 theo c\u1ED9t"),
          children: /* @__PURE__ */ jsxRuntime.jsx(antd.Switch, {
            checked: utils.isToggledHeaderColumn(editor),
            onChange: () => editor.chain().focus().toggleHeaderColumn().run()
          })
        })
      ]
    });
  };
  return /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SubQuickSetting, {
    content,
    trigger: "click",
    children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.QuickSettingButton, {
      tooltip: i18n.t("T\xF9y ch\u1EC9nh b\u1EA3ng"),
      children: /* @__PURE__ */ jsxRuntime.jsx(icons.TableOutlined, {
        style: { fontSize: 16 }
      })
    })
  });
};
const ContentWrapper = styled__default["default"].div`
  padding-top: 8px;
`;

exports.TableSetting = TableSetting;
