// @tempi/core-elements/dist/quick-configs/table/TableSetting.es.js
import { jsx, jsxs } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { TableOutlined } from '@ant-design/icons';
import styled from '@emotion/styled';
import { SubQuickSetting, QuickSettingButton, FormItem } from '@tempi/core-editor';
import { Switch } from 'antd';
import { isToggledHeaderRow, isToggledHeaderColumn } from './utils.es.js';

const TableSetting = ({ editor }) => {
  const content = () => {
    return /* @__PURE__ */ jsxs(ContentWrapper, {
      children: [
        /* @__PURE__ */ jsx(FormItem, {
          label: t("Ti\xEAu \u0111\u1EC1 theo h\xE0ng"),
          children: /* @__PURE__ */ jsx(Switch, {
            checked: isToggledHeaderRow(editor),
            onChange: () => editor.chain().focus().toggleHeaderRow().run()
          })
        }),
        /* @__PURE__ */ jsx(FormItem, {
          label: t("Ti\xEAu \u0111\u1EC1 theo c\u1ED9t"),
          children: /* @__PURE__ */ jsx(Switch, {
            checked: isToggledHeaderColumn(editor),
            onChange: () => editor.chain().focus().toggleHeaderColumn().run()
          })
        })
      ]
    });
  };
  return /* @__PURE__ */ jsx(SubQuickSetting, {
    content,
    trigger: "click",
    children: /* @__PURE__ */ jsx(QuickSettingButton, {
      tooltip: t("T\xF9y ch\u1EC9nh b\u1EA3ng"),
      children: /* @__PURE__ */ jsx(TableOutlined, {
        style: { fontSize: 16 }
      })
    })
  });
};
const ContentWrapper = styled.div`
  padding-top: 8px;
`;

export { TableSetting };
