// @tempi/core-elements/dist/quick-configs/table/TableColorQuickSetting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var styled = require('@emotion/styled');
var reactIcons = require('@fluentui/react-icons');
var coreEditor = require('@tempi/core-editor');
var propKey = require('../../constants/propKey.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

const TableColorQuickSetting = () => {
  const content = () => {
    return /* @__PURE__ */ jsxRuntime.jsxs(ContentWrapper, {
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
          propKey: propKey.PROP_KEY.HeaderBackgroundColor,
          placeholder: i18n.t("Ch\u1ECDn m\xE0u"),
          formItemProps: {
            label: i18n.t("M\xE0u n\u1EC1n ti\xEAu \u0111\u1EC1")
          }
        }),
        /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
          propKey: propKey.PROP_KEY.BackgroundColor,
          placeholder: i18n.t("Ch\u1ECDn m\xE0u"),
          formItemProps: {
            label: i18n.t("M\xE0u n\u1EC1n n\u1ED9i dung")
          }
        }),
        /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
          propKey: propKey.PROP_KEY.Color,
          placeholder: i18n.t("Ch\u1ECDn m\xE0u"),
          formItemProps: {
            label: i18n.t("M\xE0u ch\u1EEF")
          }
        }),
        /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
          propKey: propKey.PROP_KEY.BorderColor,
          placeholder: i18n.t("Ch\u1ECDn m\xE0u"),
          formItemProps: {
            label: i18n.t("M\xE0u vi\u1EC1n")
          }
        })
      ]
    });
  };
  return /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SubQuickSetting, {
    content,
    trigger: "click",
    children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.QuickSettingButton, {
      tooltip: i18n.t("T\xF9y ch\u1EC9nh m\xE0u"),
      children: /* @__PURE__ */ jsxRuntime.jsx(reactIcons.Color16Regular, {})
    })
  });
};
const ContentWrapper = styled__default["default"].div`
  padding-top: 8px;
`;

exports.TableColorQuickSetting = TableColorQuickSetting;
