// @tempi/core-elements/dist/quick-configs/table/TableColorQuickSetting.es.js
import { jsx, jsxs } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import styled from '@emotion/styled';
import { Color16Regular } from '@fluentui/react-icons';
import { SubQuickSetting, QuickSettingButton, ColorPickerConfig } from '@tempi/core-editor';
import { PROP_KEY } from '../../constants/propKey.es.js';

const TableColorQuickSetting = () => {
  const content = () => {
    return /* @__PURE__ */ jsxs(ContentWrapper, {
      children: [
        /* @__PURE__ */ jsx(ColorPickerConfig, {
          propKey: PROP_KEY.HeaderBackgroundColor,
          placeholder: t("Ch\u1ECDn m\xE0u"),
          formItemProps: {
            label: t("M\xE0u n\u1EC1n ti\xEAu \u0111\u1EC1")
          }
        }),
        /* @__PURE__ */ jsx(ColorPickerConfig, {
          propKey: PROP_KEY.BackgroundColor,
          placeholder: t("Ch\u1ECDn m\xE0u"),
          formItemProps: {
            label: t("M\xE0u n\u1EC1n n\u1ED9i dung")
          }
        }),
        /* @__PURE__ */ jsx(ColorPickerConfig, {
          propKey: PROP_KEY.Color,
          placeholder: t("Ch\u1ECDn m\xE0u"),
          formItemProps: {
            label: t("M\xE0u ch\u1EEF")
          }
        }),
        /* @__PURE__ */ jsx(ColorPickerConfig, {
          propKey: PROP_KEY.BorderColor,
          placeholder: t("Ch\u1ECDn m\xE0u"),
          formItemProps: {
            label: t("M\xE0u vi\u1EC1n")
          }
        })
      ]
    });
  };
  return /* @__PURE__ */ jsx(SubQuickSetting, {
    content,
    trigger: "click",
    children: /* @__PURE__ */ jsx(QuickSettingButton, {
      tooltip: t("T\xF9y ch\u1EC9nh m\xE0u"),
      children: /* @__PURE__ */ jsx(Color16Regular, {})
    })
  });
};
const ContentWrapper = styled.div`
  padding-top: 8px;
`;

export { TableColorQuickSetting };
