// @tempi/core-elements/dist/components/image/selector.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var coreEditor = require('@tempi/core-editor');
var coreRenderer = require('@tempi/core-renderer');
var constants = require('./constants.js');
var quickSetting = require('./quick-setting.js');
var setting = require('./setting.js');
var ui = require('./ui.js');
var selector = require('../section/selector.js');
var selector$1 = require('../root/selector.js');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const { defaultImageSvg: defaultImage } = coreRenderer.mappingAssets;
const ImageEditor = (props) => {
  const _a = props, { src, alignment = "center" } = _a, rest = __objRest(_a, ["src", "alignment"]);
  const {
    connectors: { connect, drag }
  } = coreEditor.useNode();
  return /* @__PURE__ */ jsxRuntime.jsx(ui.ImageWrapper, {
    alignment,
    ref: (ref) => connect(drag(ref)),
    children: /* @__PURE__ */ jsxRuntime.jsx(ui.Image, __spreadProps(__spreadValues({
      src: src || defaultImage
    }, rest), {
      event: null,
      isEditingMode: true
    }))
  });
};
ImageEditor.craft = coreEditor.crafting({
  displayName: "common.image",
  tag: ui.Image.displayName.toLowerCase(),
  quickSetting: quickSetting.ImageQuickSetting,
  advanceAttributes: coreEditor.AdvanceAttributes,
  customAttributes: setting.ImageSettings,
  effectAttributes: coreEditor.EffectAttributes,
  props: constants.DEFAULT_IMAGE,
  rules: {
    canDrop: (targetNode) => {
      const ignoreTargets = [
        coreEditor.getEditorName({ RowEditor: selector.RowEditor }),
        coreEditor.getEditorName({ RootEditor: selector$1.RootEditor })
      ];
      return !ignoreTargets.includes(targetNode.data.name);
    }
  }
});

exports.ImageEditor = ImageEditor;
