// @tempi/core-elements/dist/components/image/setting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { useProp, ImageConfig, InputConfig, AlignmentConfig, CornerConfig, SizeAttribute, SwitchConfig } from '@tempi/core-editor';
import { PROP_KEY } from '../../constants/propKey.es.js';

const ImageSettings = ({
  srcPropKey = PROP_KEY.Src,
  showAlignment = true,
  showZoomPreview = true
}) => {
  const [imageSrc] = useProp(srcPropKey);
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(ImageConfig, {
        propKey: PROP_KEY.Src,
        widthPropKey: PROP_KEY.joinKey(
          PROP_KEY.GoogleImageParams,
          PROP_KEY.Width
        ),
        layout: "vertical",
        allowCropImage: false
      }),
      imageSrc && /* @__PURE__ */ jsxs(Fragment, {
        children: [
          /* @__PURE__ */ jsx(InputConfig, {
            propKey: PROP_KEY.Alt,
            formItemProps: {
              label: t("V\u0103n ba\u0309n thay th\xEA\u0301"),
              layout: "vertical"
            },
            inputProps: {
              placeholder: t("Nh\xE2\u0323p ch\u01B0\u0303")
            }
          }),
          showAlignment && /* @__PURE__ */ jsx(AlignmentConfig, {
            formItemProps: {
              label: t("C\u0103n ch\u1EC9nh")
            },
            propKey: PROP_KEY.Alignment
          }),
          /* @__PURE__ */ jsx(CornerConfig, {
            propKey: PROP_KEY.CommonStyle
          }),
          /* @__PURE__ */ jsx(SizeAttribute, {}),
          showZoomPreview && /* @__PURE__ */ jsx(SwitchConfig, {
            formItemProps: {
              label: t("Ph\xF3ng to khi click \u1EA3nh"),
              description: t("Kh\xF4ng \xE1p d\u1EE5ng \u1EDF ch\u1EBF \u0111\u1ED9 ch\u1EC9nh s\u1EEDa")
            },
            propKey: PROP_KEY.ZoomPreview
          })
        ]
      })
    ]
  });
};

export { ImageSettings };
