// @tempi/core-elements/dist/components/testimonial/v2/selector/TestimonialV2Editor/TestimonialV2Editor.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var coreEditor = require('@tempi/core-editor');
var TestimonialV2QuickSetting = require('./TestimonialV2QuickSetting.js');
var TestimonialV2Setting = require('./TestimonialV2Setting.js');
var TestimonialV2 = require('../../ui/TestimonialV2/TestimonialV2.js');
var selector = require('../../../../section/selector.js');
var selector$1 = require('../../../../root/selector.js');

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
const TestimonialV2Editor = coreEditor.withSelector(
  (props) => {
    return /* @__PURE__ */ jsxRuntime.jsx(TestimonialV2.TestimonialV2, __spreadValues({
      allowTouchMove: false
    }, props));
  },
  {
    displayName: "C\u1EA3m nh\u1EADn kh\xE1ch h\xE0ng",
    tag: TestimonialV2.TestimonialV2.displayName.toLowerCase(),
    advanceAttributes: coreEditor.AdvanceAttributes,
    quickSetting: TestimonialV2QuickSetting.TestimonialV2QuickSetting,
    customAttributes: TestimonialV2Setting.TestimonialV2Setting,
    rules: {
      canDrop: (targetNode) => {
        const ignoreTargets = [
          coreEditor.getEditorName({ RowEditor: selector.RowEditor }),
          coreEditor.getEditorName({ RootEditor: selector$1.RootEditor })
        ];
        return !ignoreTargets.includes(targetNode.data.name);
      }
    }
  }
);

exports.TestimonialV2Editor = TestimonialV2Editor;
