// @tempi/core-elements/dist/core/Indicator/AddBlockButton.es.js
import { jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import styled from '@emotion/styled';
import { AddRegular } from '@fluentui/react-icons';
import { useEditor, useNode, useBuilder } from '@tempi/core-editor';
import { Button } from 'antd';

const AddBlockButton = () => {
  const {
    query: { node },
    hoveredNodes,
    getParent
  } = useEditor((state) => ({
    hoveredNodes: state.events.hovered
  }));
  const { id, isHover, allDescendants } = useNode(({ events, id: id2 }) => ({
    isHover: events.hovered,
    allDescendants: node(id2).descendants(true)
  }));
  const { setShowModalSelectBlock, setInsertIndex, device } = useBuilder();
  const hoveredChilds = allDescendants.some((el) => hoveredNodes.has(el));
  const showBottomIndicator = hoveredChilds || isHover;
  return showBottomIndicator && /* @__PURE__ */ jsx(StyledAddBlockButton, {
    type: "primary",
    shape: "round",
    size: device === "desktop" ? "middle" : "small",
    onClick: () => {
      const currentIndex = getParent(id).data.nodes.findIndex(
        (childId) => childId === id
      );
      setInsertIndex(currentIndex + 1);
      setShowModalSelectBlock(true);
    },
    children: device === "desktop" ? t("common.add_block") : /* @__PURE__ */ jsx(AddRegular, {
      fontSize: 14
    })
  });
};
const StyledAddBlockButton = styled(Button)`
  padding: 0 2em;
  position: absolute !important;
  bottom: 0px;
  left: 50%;
  transform: translate(-50%, 50%);
  font-size: 0.9em;
  z-index: 1000;
`;

export { AddBlockButton };
