// @tempi/core-editor/dist/configs/SelectConfig/SelectConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var antd = require('antd');
var styled = require('@emotion/styled');
var React = require('react');
var FormItem = require('../../components/FormItem/FormItem.js');
var usePropSeparateDevice = require('../../hooks/usePropSeparateDevice.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const SelectConfig = (props) => {
  const _a = props, { formItemProps, selectProps } = _a, rest = __objRest(_a, ["formItemProps", "selectProps"]);
  const { layout = "horizontal" } = formItemProps || {};
  return /* @__PURE__ */ jsxRuntime.jsx(FormItem.FormItem, __spreadProps(__spreadValues({
    layout
  }, formItemProps), {
    children: /* @__PURE__ */ jsxRuntime.jsx(Select, __spreadValues(__spreadValues({
      allowClear: true,
      style: { width: layout === "vertical" ? "100%" : 140 }
    }, selectProps), rest))
  }));
};
const Select = (_a) => {
  var _b = _a, {
    propKey,
    options,
    separateDevice,
    usePropOptions,
    requireExistedValue,
    value
  } = _b, rest = __objRest(_b, [
    "propKey",
    "options",
    "separateDevice",
    "usePropOptions",
    "requireExistedValue",
    "value"
  ]);
  const [prop, setProp] = usePropSeparateDevice.usePropSeparateDevice(
    propKey,
    separateDevice,
    usePropOptions
  );
  const isSelectRef = React.useRef(false);
  React.useEffect(() => {
    isSelectRef.current = false;
  }, [JSON.stringify(options)]);
  const getValue = () => {
    let val = value != null ? value : prop;
    if (requireExistedValue && !isSelectRef.current && (options == null ? void 0 : options.findIndex(({ value: value2 }) => value2 === val)) === -1) {
      val = null;
    }
    return val;
  };
  return /* @__PURE__ */ jsxRuntime.jsx(StyledAntSelect, __spreadProps(__spreadValues({
    style: {
      width: 140
    },
    dropdownStyle: { zIndex: 1080 },
    value: getValue(),
    onChange: setProp,
    options,
    placeholder: i18n.t("Ch\u1ECDn gi\xE1 tr\u1ECB")
  }, rest), {
    onSelect: () => isSelectRef.current = true
  }));
};
const StyledAntSelect = styled__default["default"](antd.Select)`
  .ant-select-selector {
    border-radius: 4px !important;
  }
`;

exports.Select = Select;
exports.SelectConfig = SelectConfig;
