// @tempi/core-renderer/dist/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var builder = require('./builder.js');
var index = require('./constants/index.js');
var render = require('./render.js');
var generate = require('./utils/colors/generate.js');
var Image = require('./components/Image/Image.js');
var ImageOptimize = require('./components/ImageOptimize/ImageOptimize.js');
var Link = require('./components/Link/Link.js');
var LazyComponent = require('./components/LazyComponent/LazyComponent.js');
var BackgroundVideo = require('./components/BackgroundVideo/BackgroundVideo.js');
var NodeElements = require('./components/NodeElement/NodeElements.js');
var reset = require('./constants/reset.js');
var fonts = require('./constants/fonts.js');
var regex = require('./constants/regex.js');
var root = require('./constants/root.js');
var mappingAssets = require('./constants/mapping-assets.js');
var common = require('./constants/common.js');
var rcCascader = require('./constants/rc-cascader.js');
var animation = require('./constants/animation.js');
var StoreContext = require('./contexts/StoreContext.js');
var BreakpointWidthContext = require('./contexts/BreakpointWidthContext.js');
var PageDataContext = require('./contexts/PageDataContext.js');
var useInitValue = require('./contexts/useInitValue.js');
var PreviewGroupContext = require('./contexts/PreviewGroupContext.js');
var useInjectLinksToHTML = require('./hooks/useInjectLinksToHTML.js');
var useTimer = require('./hooks/useTimer.js');
var useTimeVisibility = require('./hooks/useTimeVisibility.js');
var useIntersectionObserver = require('./hooks/useIntersectionObserver.js');
var useAnimation = require('./hooks/useAnimation.js');
var useInjectDynamicLinks = require('./hooks/useInjectDynamicLinks.js');
var useEndpointHandler = require('./hooks/useEndpointHandler.js');
var useLocationChange = require('./hooks/useLocationChange.js');
var useEndpointHandlerV2 = require('./hooks/useEndpointHandlerV2.js');
var useGetDynamicValue = require('./hooks/useGetDynamicValue.js');
var style = require('./types/style.js');
var common$1 = require('./types/common.js');
var index$1 = require('./utils/api/index.js');
var error = require('./utils/api/error.js');
var buildSizes = require('./utils/buildSizes.js');
var buildSrcset = require('./utils/buildSrcset.js');
var buildSrcsetDPR = require('./utils/buildSrcsetDPR.js');
var util = require('./utils/colors/util.js');
var computeStringHandleBarsExpression = require('./utils/computeStringHandleBarsExpression.js');
var dateTime = require('./utils/date-time.js');
var getBreakpointWidthFromStyle = require('./utils/getBreakpointWidthFromStyle.js');
var getOffset = require('./utils/getOffset.js');
var getSrcSetImageWidth = require('./utils/getSrcSetImageWidth.js');
var image = require('./utils/image.js');
var isAbsoluteWidth = require('./utils/isAbsoluteWidth.js');
var isGCSUrl = require('./utils/isGCSUrl.js');
var render$1 = require('./utils/render.js');
var transform = require('./utils/transform.js');
var youtubeParser = require('./utils/youtubeParser.js');
var getComponentProps = require('./utils/getComponentProps.js');
var track = require('./utils/track.js');
var RenderPage = require('./renderPage/RenderPage.js');



exports.builder = builder.builder;
exports.LandingLinks = index.LandingLinks;
exports.landingLinks = index.landingLinks;
exports.publicBffUrl = index.publicBffUrl;
exports.BuilderComponent = render.BuilderComponent;
exports.generate = generate;
exports.ImageBase = Image.ImageBase;
exports.ImageOptimize = ImageOptimize.ImageOptimize;
exports.Link = Link.Link;
exports.LazyComponent = LazyComponent.LazyComponent;
exports.BackgroundVideo = BackgroundVideo.BackgroundVideo;
exports.NodeElement = NodeElements.NodeElement;
exports.reset = reset.reset;
exports.fontQueryParams = fonts.fontQueryParams;
exports.pbFontFamily = fonts.pbFontFamily;
exports.pbFontWeight = fonts.pbFontWeight;
exports.PATTERN_VALIDATION = regex.PATTERN_VALIDATION;
exports.ARTICLE_HEADER = root.ARTICLE_HEADER;
exports.BUBBLE = root.BUBBLE;
exports.COMPONENT_TAG = root.COMPONENT_TAG;
exports.FOOTER = root.FOOTER;
exports.HEADER = root.HEADER;
exports.MAX_DPR = root.MAX_DPR;
exports.MINESTONE_M_WIDTH = root.MINESTONE_M_WIDTH;
exports.ROOT = root.ROOT;
exports.ROOT_SELECTOR_NAME = root.ROOT_SELECTOR_NAME;
exports.TEMPI_MAIN_IFRAME = root.TEMPI_MAIN_IFRAME;
exports.mappingAssets = mappingAssets.mappingAssets;
exports.BG_LAZY_CLASS_NAME = common.BG_LAZY_CLASS_NAME;
exports.BG_VISIBLE_LAZY_CLASS_NAME = common.BG_VISIBLE_LAZY_CLASS_NAME;
exports.rcCascaderCss = rcCascader.rcCascaderCss;
exports.animationCss = animation.animationCss;
exports.StoreProvider = StoreContext.StoreProvider;
exports.useStore = StoreContext.useStore;
exports.BreakpointWidthProvider = BreakpointWidthContext.BreakpointWidthProvider;
exports.useBreakpointWidth = BreakpointWidthContext.useBreakpointWidth;
exports.PageDataProvider = PageDataContext.PageDataProvider;
exports.usePageData = PageDataContext.usePageData;
exports.useInitValue = useInitValue.useInitValue;
exports.PreviewGroupProvider = PreviewGroupContext.PreviewGroupProvider;
exports.usePreviewGroup = PreviewGroupContext.usePreviewGroup;
exports.useInjectLinksToHTML = useInjectLinksToHTML.useInjectLinksToHTML;
exports.useTimer = useTimer.useTimer;
exports.useTimeVisibility = useTimeVisibility.useTimeVisibility;
exports.useIntersectionObserver = useIntersectionObserver.useIntersectionObserver;
exports.useAnimation = useAnimation.useAnimation;
exports.useInjectDynamicLinks = useInjectDynamicLinks.useInjectDynamicLinks;
exports.useEndpointHandler = useEndpointHandler.useEndpointHandler;
exports.useLocationChange = useLocationChange.useLocationChange;
exports.useEndpointHandlerV2 = useEndpointHandlerV2.useEndpointHandlerV2;
exports.useGetDynamicValue = useGetDynamicValue.useGetDynamicValue;
exports.Alignment = style.Alignment;
exports.ColorType = style.ColorType;
exports.GradientType = style.GradientType;
exports.OverlayType = style.OverlayType;
exports.UnitEnum = style.UnitEnum;
exports.VerticalAlignment = style.VerticalAlignment;
exports.RepeatTypeEnum = common$1.RepeatTypeEnum;
exports.RuleRepeatEnum = common$1.RuleRepeatEnum;
exports.UnitRepeatEnum = common$1.UnitRepeatEnum;
exports.Api = index$1.Api;
exports.ApiError = error.ApiError;
exports.buildSizes = buildSizes.buildSizes;
exports.buildSrcset = buildSrcset.buildSrcset;
exports.buildSrcsetDPR = buildSrcsetDPR.buildSrcsetDPR;
exports.getRgbaFromHex = util.getRgbaFromHex;
exports.computeStringHandleBarsExpression = computeStringHandleBarsExpression.computeStringHandleBarsExpression;
exports.getAllHandleBarsVariables = computeStringHandleBarsExpression.getAllHandleBarsVariables;
exports.isHandleBarsVariable = computeStringHandleBarsExpression.isHandleBarsVariable;
exports.calculateTimeDelta = dateTime.calculateTimeDelta;
exports.parseRuleRepeat = dateTime.parseRuleRepeat;
exports.validateIsVisibleComponent = dateTime.validateIsVisibleComponent;
exports.getBreakpointWidthFromStyle = getBreakpointWidthFromStyle.getBreakpointWidthFromStyle;
exports.getOffset = getOffset.getOffset;
exports.getSrcSetImageWidth = getSrcSetImageWidth.getSrcSetImageWidth;
exports.constructGoogleImageParams = image.constructGoogleImageParams;
exports.formatImageUrlQuality = image.formatImageUrlQuality;
exports.getImageUrlFromGoogleUserContent = image.getImageUrlFromGoogleUserContent;
exports.preProcesssorGoogleImageParams = image.preProcesssorGoogleImageParams;
exports.isAbsoluteWidth = isAbsoluteWidth.isAbsoluteWidth;
exports.isGCSUrl = isGCSUrl.isGCSUrl;
exports.buildComponentDict = render$1.buildComponentDict;
exports.getPropsByDevice = render$1.getPropsByDevice;
exports.getTargetOfTrackingFromEvent = render$1.getTargetOfTrackingFromEvent;
exports.getTrackingData = render$1.getTrackingData;
exports.debounce = transform.debounce;
exports.escapeString = transform.escapeString;
exports.formatCurrencyVND = transform.formatCurrencyVND;
exports.generateRandomId = transform.generateRandomId;
exports.getAnimation = transform.getAnimation;
exports.getBackgroundImageMarquee = transform.getBackgroundImageMarquee;
exports.getBackgroundOverlay = transform.getBackgroundOverlay;
exports.getBoxShadow = transform.getBoxShadow;
exports.getCommonCss = transform.getCommonCss;
exports.getScrollbarWidth = transform.getScrollbarWidth;
exports.getStyle = transform.getStyle;
exports.parseColor = transform.parseColor;
exports.pixelParse = transform.pixelParse;
exports.stringifyColor = transform.stringifyColor;
exports.unescapeString = transform.unescapeString;
exports.unitParse = transform.unitParse;
exports.youtubeParser = youtubeParser.youtubeParser;
exports.getComponentProps = getComponentProps.getComponentProps;
exports.getAttrTracking = track.getAttrTracking;
exports.RenderPage = RenderPage.RenderPage;
