// @tempi/core-elements/dist/components/inline-banner/selector.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var coreEditor = require('@tempi/core-editor');
var constants = require('./constants.js');
var quickSetting = require('./quick-setting.js');
var ui = require('./ui.js');
var setting = require('./setting.js');
var React = require('react');
var RenderFloatingQuickSetting = require('../../core/RenderFloatingQuickSetting/RenderFloatingQuickSetting.js');
var selector = require('../root/selector.js');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const InlineBannerEditor = (_a) => {
  var _b = _a, {
    children,
    quickSetting: quickSetting$1
  } = _b, rest = __objRest(_b, [
    "children",
    "quickSetting"
  ]);
  const {
    connectors: { connect, drag },
    node
  } = coreEditor.useNode((node2) => ({
    node: node2
  }));
  const [name, setName] = coreEditor.useProp("name");
  const [id, setId] = coreEditor.useProp("id");
  React.useEffect(() => {
    if (!id)
      setId(node.id);
    if (!name)
      setName(node.id);
  }, []);
  return /* @__PURE__ */ jsxRuntime.jsx("div", {
    ref: (ref) => connect(drag(ref)),
    children: /* @__PURE__ */ jsxRuntime.jsx(ui.InlineBanner, __spreadProps(__spreadValues({
      isEditorMode: true
    }, rest), {
      children: /* @__PURE__ */ jsxRuntime.jsx(RenderFloatingQuickSetting.RenderFloatingQuickSetting, {
        quickSetting: quickSetting$1 || /* @__PURE__ */ jsxRuntime.jsx(quickSetting.InlineBannerQuickSetting, {}),
        children
      })
    }))
  });
};
InlineBannerEditor.craft = coreEditor.crafting({
  displayName: "Inline banner",
  tag: ui.InlineBanner.displayName.toLowerCase(),
  isCanvas: true,
  advanceAttributes: coreEditor.AdvanceAttributes,
  customAttributes: setting.InlineBannerSetting,
  props: constants.DEFAULT_INLINE_BANNER,
  rules: {
    canDrop: () => false,
    canDrag: () => false,
    canMoveIn: (incomingNodes) => {
      const ignoreSources = [coreEditor.getEditorName({ RootEditor: selector.RootEditor })];
      return incomingNodes.every(
        (incomingNode) => !ignoreSources.includes(incomingNode.data.name)
      );
    }
  }
});

exports.InlineBannerEditor = InlineBannerEditor;
