// @tempi/core-elements/dist/components/content-slider/selector/ContentSliderHeadingEditor/ContentSliderHeadingEditor.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var coreEditor = require('@tempi/core-editor');
var React = require('react');
var editorQuickSetting = require('../../../heading/editor-quick-setting.js');
var quickSetting = require('../../../heading/quick-setting.js');
var constant = require('../../constant.js');
var propKey = require('../../../../constants/propKey.js');
var ui = require('../../../heading/ui.js');
var TipTapEditorV2 = require('../../../../configs/TipTapEditor/TipTapEditorV2.js');

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const ContentSliderHeadingEditor = coreEditor.withSelector(
  (_a) => {
    var _b = _a, { id } = _b, props = __objRest(_b, ["id"]);
    const { parentId } = coreEditor.useNode((node) => ({
      parentId: node.data.parent
    }));
    const {
      enabled,
      actions: { history }
    } = coreEditor.useEditor((state) => ({
      enabled: state.options.enabled
    }));
    const [text] = coreEditor.useProp(propKey.PROP_KEY.Text);
    React.useEffect(() => {
      history.throttle().setProp(parentId, (props2) => {
        const htmlRegex = /(<([^>]+)>)/gi;
        props2.headingText = text.replace(htmlRegex, "");
      });
    }, [text, history, parentId]);
    if (!enabled)
      return /* @__PURE__ */ jsxRuntime.jsx(ui.Heading, __spreadValues({
        id
      }, props));
    return /* @__PURE__ */ jsxRuntime.jsx(TipTapEditorV2.TipTapEditorV2, {
      propKey: propKey.PROP_KEY.Text,
      quickSetting: quickSetting.HeadingQuickSetting,
      editorQuickSetting: editorQuickSetting.EditorQuickSetting,
      children: /* @__PURE__ */ jsxRuntime.jsx(ui.Heading, {
        commonStyle: props == null ? void 0 : props.commonStyle
      })
    });
  },
  {
    displayName: "Ti\xEAu \u0111\u1EC1",
    tag: ui.Heading.displayName.toLowerCase(),
    advanceAttributes: coreEditor.AdvanceAttributes,
    props: constant.DEFAULT_CONTENT_SLIDER.heading,
    rules: {
      canDrag: () => false
    }
  }
);

exports.ContentSliderHeadingEditor = ContentSliderHeadingEditor;
