// @tempi/core-elements/dist/components/product-list/configs/StorageDataSetting.es.js
import { jsx } from 'react/jsx-runtime';
import { t } from '../../../i18n.es.js';
import { Form, Modal, Spin } from 'antd';
import { useEffect } from 'react';
import { useFrame } from 'react-frame-component';
import { DATA_KEYS } from '../constants.es.js';
import { useStorageData } from '../../../configs/storage/hook/useStorageData.es.js';
import { ConfigStorageData } from '../../../configs/storage/ConfigStorageData.es.js';

const StorageDataSetting = ({
  open,
  setOpen
}) => {
  const { document: frameDocument } = useFrame();
  const [form] = Form.useForm();
  const formConfigs = Form.useWatch("configs", form);
  const {
    getFormConfigs,
    ggSheetsAccounts,
    handleFetchFormAccount,
    handleSubmitStorageData,
    initStorageData,
    loadingForm,
    configStorageId
  } = useStorageData({ form });
  useEffect(() => {
    if (open) {
      getFormConfigs(configStorageId, initStorageData);
      handleFetchFormAccount();
    }
  }, [open]);
  const handlerSaveModal = async (value) => {
    await handleSubmitStorageData({ configs: value.configs });
    setOpen(false);
  };
  return /* @__PURE__ */ jsx(Modal, {
    title: t("L\u01B0u d\u1EEF li\u1EC7u"),
    zIndex: 2e3,
    onCancel: () => setOpen(false),
    open,
    okText: t("L\u01B0u"),
    destroyOnClose: true,
    cancelText: t("common.cancel"),
    bodyStyle: { paddingTop: 0 },
    getContainer: () => (frameDocument == null ? void 0 : frameDocument.body) || (document == null ? void 0 : document.body),
    onOk: () => form.submit(),
    cancelButtonProps: {
      shape: "round"
    },
    okButtonProps: {
      loading: loadingForm,
      shape: "round",
      disabled: formConfigs == null ? void 0 : formConfigs.some(
        (item) => (item == null ? void 0 : item.statusGoogleSheetUrl) === "validating"
      )
    },
    children: /* @__PURE__ */ jsx(Form, {
      form,
      onFinish: handlerSaveModal,
      layout: "vertical",
      scrollToFirstError: true,
      children: /* @__PURE__ */ jsx(Spin, {
        spinning: loadingForm,
        children: /* @__PURE__ */ jsx(ConfigStorageData, {
          form,
          ggSheetsAccounts,
          handleFetchFormAccount,
          suggestionAttributes: Object.values(DATA_KEYS)
        })
      })
    })
  });
};

export { StorageDataSetting };
