// @tempi/core-elements/dist/components/header/ui/Menu/MobileMenu.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import styled from '@emotion/styled';
import { Icon } from '@tempi/icons';
import { useContext, useState } from 'react';
import { HeaderContext } from '../../context.es.js';
import { getPropsIcon } from '../../../../utils/getPropsIcon.es.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
const MobileMenu = ({ children }) => {
  const { toggle } = useContext(HeaderContext);
  const [visibleMenu, setVisibleMenu] = useState(false);
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(ToggleButton, {
        className: "toggle-button",
        onClick: () => setVisibleMenu(!visibleMenu),
        children: visibleMenu ? /* @__PURE__ */ jsx(Icon, __spreadValues({
          name: "CloseOutlined"
        }, getPropsIcon(toggle == null ? void 0 : toggle.icon))) : /* @__PURE__ */ jsx(Icon, __spreadValues({
          name: "MenuOutlined"
        }, getPropsIcon(toggle == null ? void 0 : toggle.icon)))
      }),
      /* @__PURE__ */ jsx(DropdownMenu, {
        className: `dropdown-menu ${visibleMenu ? "show" : ""}`,
        children
      })
    ]
  });
};
const ToggleButton = styled.div`
  margin-left: auto;
  cursor: pointer;
  display: flex;
  align-items: center;
  padding: 0 8px;
`;
const DropdownMenu = styled.div`
  position: absolute;
  top: 80%;
  left: 0;
  right: 0;
  z-index: 100;

  display: flex;
  flex-direction: column;
  align-content: center;
  row-gap: 8px;

  height: auto;
  padding: 16px;
  overflow-y: auto;

  opacity: 0;
  visibility: hidden;
  transition: all 0.2s ease-in-out;

  background: inherit;
  box-shadow: 0 10px 16px rgba(10, 10, 10, 0.1);

  &.show {
    top: 100%;
    opacity: 1;
    visibility: visible;
  }
`;

export { MobileMenu };
