// @tempi/core-editor/dist/configs/ImageConfig/ImageUploadButton.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var React = require('react');
var ImageUpload = require('./ImageUpload.js');
var styled = require('@emotion/styled');
var antd = require('antd');
var icons = require('@ant-design/icons');
var default_image = require('../../assets/images/default_image.js');
var uuid = require('uuid');
var file = require('../../utils/file.js');
var usePropSeparateDevice = require('../../hooks/usePropSeparateDevice.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const ImageUploadButton = (_a) => {
  var _b = _a, {
    multiple = false,
    imagePropKey,
    widthPropKey,
    listType = "picture-card",
    placeholder = /* @__PURE__ */ jsxRuntime.jsx(jsxRuntime.Fragment, {
      children: i18n.t("T\u1EA3i \u1EA3nh l\xEAn (T\u1ED1i \u0111a 20MB)")
    }),
    deleteImage,
    separateDevice,
    layout = "horizontal",
    defaultNewImage,
    showButtonUpload,
    stateImageUpload
  } = _b, rest = __objRest(_b, [
    "multiple",
    "imagePropKey",
    "widthPropKey",
    "listType",
    "placeholder",
    "deleteImage",
    "separateDevice",
    "layout",
    "defaultNewImage",
    "showButtonUpload",
    "stateImageUpload"
  ]);
  const [image, setImage] = usePropSeparateDevice.usePropSeparateDevice(imagePropKey, separateDevice);
  const [, setWidth] = usePropSeparateDevice.usePropSeparateDevice(widthPropKey, separateDevice);
  const [uploading, setUploading] = React.useState(false);
  const { token: themeToken } = antd.theme.useToken();
  const mergeImage = stateImageUpload ? stateImageUpload.image : image;
  const mergeSetImage = stateImageUpload ? stateImageUpload.setImage : setImage;
  const mergeSetWidth = stateImageUpload ? stateImageUpload.setWidth : setWidth;
  const uploadAction = async (data, file$1) => {
    if (!data)
      return;
    const imageUrl = data.imageUrl;
    if (imageUrl && imagePropKey) {
      if (widthPropKey) {
        const img = await file.getImageObject(file$1);
        mergeSetWidth(img.width);
      }
      mergeSetImage(imageUrl);
    }
    setUploading(data.uploading);
  };
  const uploadImageListAction = async (data, file$1) => {
    if (!data)
      return;
    const imageUrl = data.imageUrl;
    if (imageUrl && imagePropKey) {
      const newObject = __spreadValues({
        src: imageUrl,
        googleImageParams: {
          width: 0
        }
      }, defaultNewImage);
      if (newObject.id) {
        newObject.id = uuid.v4();
      }
      const img = await file.getImageObject(file$1);
      newObject.googleImageParams.width = img.width;
      const newImage = [...image];
      newImage.push(newObject);
      mergeSetImage(newImage);
    }
    setUploading(data.uploading);
  };
  const renderContent = () => {
    if (multiple) {
      return /* @__PURE__ */ jsxRuntime.jsx(antd.Button, {
        block: true,
        type: "primary",
        icon: /* @__PURE__ */ jsxRuntime.jsx(icons.UploadOutlined, {}),
        loading: uploading,
        children: placeholder
      });
    }
    if (mergeImage && mergeImage != default_image && !uploading) {
      return /* @__PURE__ */ jsxRuntime.jsx(PreviewImage, {
        src: mergeImage,
        alt: "uploaded img"
      });
    }
    return /* @__PURE__ */ jsxRuntime.jsx("div", {
      style: { width: "100%" },
      children: uploading ? /* @__PURE__ */ jsxRuntime.jsx(icons.LoadingOutlined, {}) : /* @__PURE__ */ jsxRuntime.jsx(jsxRuntime.Fragment, {
        children: showButtonUpload ? /* @__PURE__ */ jsxRuntime.jsx(antd.Button, {
          icon: /* @__PURE__ */ jsxRuntime.jsx(icons.UploadOutlined, {}),
          loading: uploading,
          children: placeholder
        }) : /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
          children: [
            /* @__PURE__ */ jsxRuntime.jsx(icons.UploadOutlined, {}),
            /* @__PURE__ */ jsxRuntime.jsx("div", {
              style: { marginTop: 4 },
              children: placeholder
            })
          ]
        })
      })
    });
  };
  return /* @__PURE__ */ jsxRuntime.jsxs("div", {
    style: { position: "relative" },
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(ImageUpload.ImageUpload, __spreadProps(__spreadValues({
        multiple,
        showUploadList: false,
        disabled: uploading,
        listType,
        onUpdateImageUrl: multiple ? uploadImageListAction : uploadAction,
        layout
      }, rest), {
        children: renderContent()
      })),
      !multiple && !!mergeImage && /* @__PURE__ */ jsxRuntime.jsx(icons.CloseCircleFilled, {
        onClick: () => {
          if (deleteImage)
            deleteImage();
          else {
            mergeSetImage(void 0);
          }
        },
        style: {
          cursor: "pointer",
          color: themeToken.colorText,
          position: "absolute",
          top: -6,
          right: -6,
          fontSize: 16,
          borderRadius: 8
        }
      })
    ]
  });
};
const PreviewImage = styled__default["default"].img`
  width: 100%;
  height: 100%;
  object-fit: contain;
`;

exports.ImageUploadButton = ImageUploadButton;
