// @tempi/core-elements/dist/components/paragraph/editor-quick-setting.es.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import styled from '@emotion/styled';
import { TextLineSpacingRegular } from '@fluentui/react-icons';
import { useBuilder, useEditor, PROP_KEY, InputSelectConfig, removeUnit, OpenSettingDialog, SelectConfig } from '@tempi/core-editor';
import { ROOT, unitParse } from '@tempi/core-renderer';
import { Select } from 'antd';
import get from 'lodash.get';
import { NODE_TYPES } from '../../configs/TipTapEditor/constants.es.js';
import { MAX_FONT_SIZE } from './constants.es.js';
import { TextColorQuickSettingButton } from '../../quick-configs/typography/TextColorQuickSettingButton.es.js';
import { fontFamilyOptions, fontSizeOptions, fontWeightOptions } from '../../constants/fonts.es.js';
import { TextInputNumberConfig } from '../../configs/TextConfig/TextInputNumberConfig.es.js';
import { TextListQuickSetting } from '../../quick-configs/typography/TextListQuickSetting.es.js';
import { getEditorAttributeValue } from '../../utils/getEditorAttributeValue.es.js';
import { TextAlignQuickSetting } from '../../quick-configs/typography/TextAlignQuickSetting.es.js';
import { BoldQuickSetting } from '../../quick-configs/typography/BoldQuickSetting.es.js';
import { ItalicQuickSetting } from '../../quick-configs/typography/ItalicQuickSetting.es.js';
import { UnderlineQuickSetting } from '../../quick-configs/typography/UnderlineQuickSetting.es.js';
import { StrikeThroughQuickSetting } from '../../quick-configs/typography/StrikeThroughQuickSetting.es.js';
import { TextLinkQuickSetting } from '../../quick-configs/typography/TextLinkQuickSetting.es.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const headingTypePrefix = "h";
const headingLevel = [1, 2, 3, 4, 5, 6];
const paragraphType = "paragraph";
const typeOptions = [
  ...headingLevel.map((level) => `${headingTypePrefix}${level}`),
  paragraphType
];
const EditorQuickSetting = ({ editor }) => {
  var _a, _b, _c, _d, _e;
  const { realDevice } = useBuilder();
  const { rootNode } = useEditor((state) => ({
    rootNode: state.nodes[ROOT]
  }));
  const fontFamilyKey = PROP_KEY.joinKey(
    PROP_KEY.CommonStyle,
    PROP_KEY.FontFamily
  );
  const fontWeightKey = PROP_KEY.joinKey(
    PROP_KEY.CommonStyle,
    PROP_KEY.FontWeight
  );
  const getTypeValue = () => {
    const headingLevel2 = editor == null ? void 0 : editor.getAttributes(NODE_TYPES.heading).level;
    if (headingLevel2)
      return `${headingTypePrefix}${headingLevel2}`;
    return paragraphType;
  };
  const onChangeType = (value) => {
    var _b2, _c2;
    const _a2 = editor.getAttributes(
      editor.isActive(NODE_TYPES.heading) ? NODE_TYPES.heading : NODE_TYPES.paragraph
    ), restAttributes = __objRest(_a2, ["fontSize", "level"]);
    if (value === paragraphType) {
      (_b2 = editor == null ? void 0 : editor.commands) == null ? void 0 : _b2.executeCmd(
        ({ chain }) => chain().setParagraph().updateAttributes(NODE_TYPES.paragraph, restAttributes).run()
      );
    } else {
      (_c2 = editor == null ? void 0 : editor.commands) == null ? void 0 : _c2.executeCmd(
        ({ chain }) => chain().toggleHeading({
          level: parseInt(value.replace(headingTypePrefix, ""))
        }).updateAttributes(NODE_TYPES.heading, restAttributes).setFontSize(null, realDevice).run()
      );
    }
  };
  const onChangeFontSize = (value) => {
    var _a2, _b2;
    if (!value) {
      if (editor.isActive(NODE_TYPES.heading))
        (_a2 = editor == null ? void 0 : editor.commands) == null ? void 0 : _a2.executeCmd(
          ({ commands }) => commands.setFontSize(null, realDevice)
        );
      return;
    }
    let fontSize = typeof value === "number" ? value : parseInt(value);
    if (fontSize < 0)
      return;
    fontSize = fontSize >= MAX_FONT_SIZE ? MAX_FONT_SIZE : fontSize;
    (_b2 = editor == null ? void 0 : editor.commands) == null ? void 0 : _b2.executeCmd(
      ({ commands }) => commands.setFontSize(fontSize + "px", realDevice)
    );
  };
  return /* @__PURE__ */ jsxs("div", {
    children: [
      /* @__PURE__ */ jsxs(Row, {
        children: [
          /* @__PURE__ */ jsx(TextColorQuickSettingButton, {
            editor,
            styleKey: "color",
            onChange: (color) => {
              var _a2;
              (_a2 = editor == null ? void 0 : editor.commands) == null ? void 0 : _a2.executeCmd(
                ({ commands }) => commands.setColor(color)
              );
            }
          }),
          /* @__PURE__ */ jsx(TextSelectConfig, {
            formItemProps: {
              style: { marginBottom: 0 }
            },
            selectProps: {
              style: { width: 110 },
              dropdownMatchSelectWidth: false,
              bordered: false,
              allowClear: false,
              value: getTypeValue(),
              onChange: onChangeType,
              children: typeOptions == null ? void 0 : typeOptions.map((val) => {
                const isHeading = val !== paragraphType;
                const label = isHeading ? t("Ti\xEAu \u0111\u1EC1 {{var0}}", {
                  var0: val.replace(headingTypePrefix, "")
                }) : t("\u0110o\u1EA1n v\u0103n");
                return /* @__PURE__ */ jsx(Select.Option, {
                  value: val,
                  children: /* @__PURE__ */ jsx("span", {
                    style: __spreadValues({}, isHeading && { fontWeight: 500 }),
                    children: label
                  })
                }, val);
              })
            }
          }),
          /* @__PURE__ */ jsx(TextSelectConfig, {
            formItemProps: {
              style: { marginBottom: 0 }
            },
            selectProps: {
              showSearch: true,
              dropdownMatchSelectWidth: false,
              bordered: false,
              placeholder: t("Ph\xF4ng ch\u1EEF"),
              value: getEditorAttributeValue(editor, PROP_KEY.FontFamily) || get((_b = (_a = rootNode == null ? void 0 : rootNode.data) == null ? void 0 : _a.props) == null ? void 0 : _b[realDevice], fontFamilyKey),
              onChange: (value) => {
                var _a2;
                (_a2 = editor == null ? void 0 : editor.commands) == null ? void 0 : _a2.executeCmd(
                  ({ commands }) => commands.setFontFamily(value)
                );
              },
              children: (_c = fontFamilyOptions) == null ? void 0 : _c.map((el) => /* @__PURE__ */ jsx(Select.Option, {
                value: el.value,
                children: /* @__PURE__ */ jsx("span", {
                  style: { fontFamily: el.value },
                  children: el.label
                })
              }, el.value))
            }
          }),
          /* @__PURE__ */ jsx(InputSelectConfig, {
            options: fontSizeOptions,
            width: 85,
            formItemProps: {
              style: { marginBottom: 0 }
            },
            value: removeUnit(
              getEditorAttributeValue(editor, PROP_KEY.FontSize, realDevice)
            ),
            onChange: onChangeFontSize,
            inputProps: {
              placeholder: t("C\u1EE1 ch\u1EEF"),
              type: "number",
              min: 0,
              max: MAX_FONT_SIZE
            }
          }),
          /* @__PURE__ */ jsx(TextSelectConfig, {
            formItemProps: {
              style: { marginBottom: 0 }
            },
            selectProps: {
              style: { width: 96 },
              dropdownMatchSelectWidth: false,
              bordered: false,
              placeholder: t("\u0110\u1ED9 d\xE0y"),
              value: getEditorAttributeValue(editor, PROP_KEY.FontWeight) || get((_e = (_d = rootNode == null ? void 0 : rootNode.data) == null ? void 0 : _d.props) == null ? void 0 : _e[realDevice], fontWeightKey),
              onChange: (value) => {
                var _a2;
                (_a2 = editor == null ? void 0 : editor.commands) == null ? void 0 : _a2.executeCmd(
                  ({ commands }) => commands.setFontWeight(value)
                );
              },
              options: fontWeightOptions
            }
          }),
          /* @__PURE__ */ jsx(TextInputNumberConfig, {
            formItemProps: {
              style: { marginBottom: 0 }
            },
            initValue: removeUnit(
              getEditorAttributeValue(editor, PROP_KEY.LineHeight)
            ),
            inputNumberProps: {
              style: {
                width: 80
              },
              bordered: false,
              controls: false,
              prefix: /* @__PURE__ */ jsx(TextLineSpacingRegular, {
                fontSize: 16
              }),
              placeholder: "Auto"
            },
            updateValue: (val) => {
              var _a2;
              (_a2 = editor == null ? void 0 : editor.commands) == null ? void 0 : _a2.executeCmd(
                ({ commands }) => commands.setLineHeight(unitParse(val))
              );
            }
          })
        ]
      }),
      /* @__PURE__ */ jsxs(Row, {
        children: [
          /* @__PURE__ */ jsx(TextAlignQuickSetting, {
            editor
          }),
          /* @__PURE__ */ jsx(TextListQuickSetting, {
            editor
          }),
          /* @__PURE__ */ jsx(BoldQuickSetting, {
            editor
          }),
          /* @__PURE__ */ jsx(ItalicQuickSetting, {
            editor
          }),
          /* @__PURE__ */ jsx(UnderlineQuickSetting, {
            editor
          }),
          /* @__PURE__ */ jsx(StrikeThroughQuickSetting, {
            editor
          }),
          /* @__PURE__ */ jsx(TextLinkQuickSetting, {
            editor
          }),
          /* @__PURE__ */ jsx(OpenSettingDialog, {})
        ]
      })
    ]
  });
};
const Row = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;
const TextSelectConfig = styled(SelectConfig)`
  padding-left: 12px;
  padding-right: 12px;

  .ant-select-selector {
    padding: 0 !important;
  }

  .ant-select-selection-search {
    inset-inline-start: 0 !important;
  }
`;

export { EditorQuickSetting };
