// @tempi/core-elements/dist/components/article-list/selector.es.js
import { withSelector, AdvanceAttributes, getEditorName } from '@tempi/core-editor';
import { DEFAULT_ARTICLE_LIST } from './constants.es.js';
import { QuickSetting } from './quick-setting.es.js';
import { ArticleList } from './ui.es.js';
import { ArticleListStyleSetting } from './style-setting.es.js';
import { ArticleListSetting } from './setting/ArticleListSetting.es.js';
import { RowEditor } from '../section/selector.es.js';
import { RootEditor } from '../root/selector.es.js';

const ArticleListEditor = withSelector(ArticleList, {
  displayName: "Danh s\xE1ch b\xE0i vi\u1EBFt",
  tag: ArticleList.displayName.toLowerCase(),
  quickSetting: QuickSetting,
  customAttributes: ArticleListSetting,
  props: DEFAULT_ARTICLE_LIST,
  advanceAttributes: AdvanceAttributes,
  commonAttributes: ArticleListStyleSetting,
  rules: {
    canDrop: (targetNode) => {
      const ignoreTargets = [
        getEditorName({ RowEditor }),
        getEditorName({ RootEditor })
      ];
      return !ignoreTargets.includes(targetNode.data.name);
    }
  }
});

export { ArticleListEditor };
