// @tempi/core-elements/dist/components/article-list/selector.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var coreEditor = require('@tempi/core-editor');
var constants = require('./constants.js');
var quickSetting = require('./quick-setting.js');
var ui = require('./ui.js');
var styleSetting = require('./style-setting.js');
var ArticleListSetting = require('./setting/ArticleListSetting.js');
var selector = require('../section/selector.js');
var selector$1 = require('../root/selector.js');

const ArticleListEditor = coreEditor.withSelector(ui.ArticleList, {
  displayName: "Danh s\xE1ch b\xE0i vi\u1EBFt",
  tag: ui.ArticleList.displayName.toLowerCase(),
  quickSetting: quickSetting.QuickSetting,
  customAttributes: ArticleListSetting.ArticleListSetting,
  props: constants.DEFAULT_ARTICLE_LIST,
  advanceAttributes: coreEditor.AdvanceAttributes,
  commonAttributes: styleSetting.ArticleListStyleSetting,
  rules: {
    canDrop: (targetNode) => {
      const ignoreTargets = [
        coreEditor.getEditorName({ RowEditor: selector.RowEditor }),
        coreEditor.getEditorName({ RootEditor: selector$1.RootEditor })
      ];
      return !ignoreTargets.includes(targetNode.data.name);
    }
  }
});

exports.ArticleListEditor = ArticleListEditor;
