// @tempi/core-elements/dist/components/article-list/style-setting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { RangeSliderConfig, ColorPickerConfig, SelectConfig } from '@tempi/core-editor';
import { PROP_KEY } from '../../constants/propKey.es.js';

const ArticleListStyleSetting = () => {
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(RangeSliderConfig, {
        propKey: PROP_KEY.TitleFontSize,
        formItemProps: {
          label: t("C\u1EE1 ch\u1EEF ti\xEAu \u0111\u1EC1")
        },
        sliderProps: {
          min: 12,
          max: 72
        },
        unitProps: {
          defaultValue: "px"
        }
      }),
      /* @__PURE__ */ jsx(ColorPickerConfig, {
        propKey: PROP_KEY.joinKey(PROP_KEY.CommonStyle, PROP_KEY.Color),
        formItemProps: {
          label: t("M\xE0u ch\u1EEF")
        }
      }),
      /* @__PURE__ */ jsx(ColorPickerConfig, {
        propKey: PROP_KEY.joinKey(
          PROP_KEY.CommonStyle,
          PROP_KEY.BackgroundColor
        ),
        formItemProps: {
          label: t("M\xE0u n\u1EC1n")
        },
        gradientPicker: true
      }),
      /* @__PURE__ */ jsx(SelectConfig, {
        propKey: PROP_KEY.CategoryType,
        formItemProps: {
          label: t("Ki\u1EC3u hi\u1EC3n th\u1ECB danh m\u1EE5c")
        },
        selectProps: {
          options: [
            {
              label: t("N\xFAt b\u1EA5m"),
              value: "button"
            },
            {
              label: t("Ch\u1EEF"),
              value: "text"
            }
          ]
        }
      }),
      /* @__PURE__ */ jsx(ColorPickerConfig, {
        propKey: PROP_KEY.CategoryColor,
        formItemProps: {
          label: t("M\xE0u danh m\u1EE5c")
        }
      })
    ]
  });
};

export { ArticleListStyleSetting };
