// @tempi/core-elements/dist/components/article-list/style-setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var propKey = require('../../constants/propKey.js');

const ArticleListStyleSetting = () => {
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.RangeSliderConfig, {
        propKey: propKey.PROP_KEY.TitleFontSize,
        formItemProps: {
          label: i18n.t("C\u1EE1 ch\u1EEF ti\xEAu \u0111\u1EC1")
        },
        sliderProps: {
          min: 12,
          max: 72
        },
        unitProps: {
          defaultValue: "px"
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.CommonStyle, propKey.PROP_KEY.Color),
        formItemProps: {
          label: i18n.t("M\xE0u ch\u1EEF")
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
        propKey: propKey.PROP_KEY.joinKey(
          propKey.PROP_KEY.CommonStyle,
          propKey.PROP_KEY.BackgroundColor
        ),
        formItemProps: {
          label: i18n.t("M\xE0u n\u1EC1n")
        },
        gradientPicker: true
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
        propKey: propKey.PROP_KEY.CategoryType,
        formItemProps: {
          label: i18n.t("Ki\u1EC3u hi\u1EC3n th\u1ECB danh m\u1EE5c")
        },
        selectProps: {
          options: [
            {
              label: i18n.t("N\xFAt b\u1EA5m"),
              value: "button"
            },
            {
              label: i18n.t("Ch\u1EEF"),
              value: "text"
            }
          ]
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
        propKey: propKey.PROP_KEY.CategoryColor,
        formItemProps: {
          label: i18n.t("M\xE0u danh m\u1EE5c")
        }
      })
    ]
  });
};

exports.ArticleListStyleSetting = ArticleListStyleSetting;
