// @tempi/core-elements/dist/quick-configs/QuickLayout/QuickLayout.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var reactIcons = require('@fluentui/react-icons');
var coreEditor = require('@tempi/core-editor');
var antd = require('antd');
var React = require('react');
var iconsDev = require('@tempi/icons-dev');
var layoutUtils = require('./layoutUtils.js');
var set = require('lodash.set');
var selector = require('../../components/col/selector.js');
var propKey = require('../../constants/propKey.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var set__default = /*#__PURE__*/_interopDefaultLegacy(set);

const QuickLayout = () => {
  const [open, setOpen] = React.useState(false);
  const [form] = antd.Form.useForm();
  const { actions, query } = coreEditor.useEditor();
  const { id, columnIds } = coreEditor.useNode((node) => ({
    columnIds: node.data.nodes
  }));
  const getCurrentLayoutString = () => {
    const currentLayout = columnIds.map((colId) => {
      var _a, _b, _c;
      const colProps = (_b = (_a = query.node(colId).get()) == null ? void 0 : _a.data) == null ? void 0 : _b.props;
      return (_c = colProps == null ? void 0 : colProps.md) == null ? void 0 : _c.flex;
    });
    if (currentLayout.length === 0) {
      return "";
    }
    if (currentLayout.length === 1) {
      return `${currentLayout[0]}`;
    }
    return currentLayout.join(" + ");
  };
  const updateColumnWidth = (colId, percentage) => {
    actions.history.throttle().setProp(colId, (props) => {
      set__default["default"](props, propKey.PROP_KEY.Md, `${coreEditor.floorPercentage(percentage)}%`);
    });
  };
  const addColumnWithSpecificWidthPercentage = (widthPercentage) => {
    const element = /* @__PURE__ */ jsxRuntime.jsx(coreEditor.Element, {
      canvas: true,
      is: selector.ColEditor,
      md: { flex: `${coreEditor.floorPercentage(widthPercentage)}%` },
      xs: { flex: "100%" }
    });
    const newColNodeTree = query.parseReactElement(element).toNodeTree();
    actions.addNodeTree(newColNodeTree, id);
  };
  const onFinish = ({ layout }) => {
    const newRowLayout = layoutUtils.dissectStringIntoPercentages(layout);
    if (newRowLayout.length !== columnIds.length) {
      if (newRowLayout.length < columnIds.length) {
        newRowLayout.forEach((percentage, index) => {
          updateColumnWidth(columnIds[index], percentage);
        });
        const columnIdsToDelete = columnIds.slice(newRowLayout.length);
        columnIdsToDelete.forEach((colId) => {
          actions.history.throttle().delete(colId);
        });
      }
      newRowLayout.forEach((percentage, index) => {
        updateColumnWidth(columnIds[index], percentage);
      });
      const newColWidthPercentage = newRowLayout.slice(columnIds.length);
      const numberOfNewColumns = newRowLayout.length - columnIds.length;
      for (let i = 0; i < numberOfNewColumns; i++) {
        addColumnWithSpecificWidthPercentage(newColWidthPercentage[i]);
      }
    } else {
      newRowLayout.forEach((percentage, index) => {
        updateColumnWidth(columnIds[index], percentage);
      });
    }
    const sumPercentage = newRowLayout.reduce((a, b) => a + b, 0);
    if (sumPercentage < 100) {
      const blankPercentage = 100 - sumPercentage;
      if (blankPercentage >= 8) {
        addColumnWithSpecificWidthPercentage(blankPercentage);
      }
    }
    setOpen(false);
  };
  const onSelectLayout = (layout) => {
    form.setFieldsValue({
      layout: layoutUtils.getLayoutStringByColumnNumber(layout)
    });
  };
  React.useEffect(() => {
    form.setFieldsValue({
      layout: getCurrentLayoutString()
    });
  }, [columnIds, form]);
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.QuickSettingButton, {
        tooltip: i18n.t("Ch\u1EC9nh s\u1EEDa b\u1ED1 c\u1EE5c h\xE0ng"),
        onClick: () => setOpen(true),
        children: /* @__PURE__ */ jsxRuntime.jsx(reactIcons.TextColumnThreeRegular, {
          style: { fontSize: 16 }
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx(antd.Modal, {
        open,
        title: i18n.t("C\xE0i \u0111\u1EB7t nhanh b\u1ED1 c\u1EE5c h\xE0ng"),
        onOk: form.submit,
        okText: i18n.t("C\u1EADp nh\u1EADt"),
        cancelText: i18n.t("\u0110\xF3ng"),
        onCancel: () => {
          setOpen(false);
          form.setFieldsValue({
            layout: getCurrentLayoutString()
          });
        },
        children: /* @__PURE__ */ jsxRuntime.jsxs(antd.Form, {
          onFinish,
          layout: "vertical",
          form,
          children: [
            /* @__PURE__ */ jsxRuntime.jsx("div", {
              style: { margin: "1em 0 0 0" },
              children: i18n.t("Ch\u1ECDn nhanh b\u1ED1 c\u1EE5c")
            }),
            /* @__PURE__ */ jsxRuntime.jsx("div", {
              children: layoutUtils.layoutIcons.map((icon, index) => /* @__PURE__ */ jsxRuntime.jsx("span", {
                style: {
                  cursor: "pointer",
                  fontSize: 36,
                  marginRight: "0.2em"
                },
                title: i18n.t("{{var0}} c\u1ED9t", { var0: index + 1 }),
                onClick: () => onSelectLayout(index + 1),
                children: /* @__PURE__ */ jsxRuntime.jsx(iconsDev.AntdCustomIcon, {
                  name: icon
                })
              }, index))
            }),
            /* @__PURE__ */ jsxRuntime.jsx(antd.Form.Item, {
              label: i18n.t("S\u1EEDa b\u1ED1 c\u1EE5c th\u1EE7 c\xF4ng"),
              name: "layout",
              rules: [
                { required: true, message: i18n.t("Vui l\xF2ng nh\u1EADp ho\u1EB7c ch\u1ECDn b\u1ED1 c\u1EE5c") },
                { validator: layoutUtils.validateLayoutSyntax }
              ],
              children: /* @__PURE__ */ jsxRuntime.jsx(antd.Input, {})
            }),
            /* @__PURE__ */ jsxRuntime.jsx(antd.Typography.Text, {
              type: "secondary",
              children: i18n.t(
                "Thay \u0111\u1ED5i b\u1ED1 c\u1EE5c h\xE0ng th\u1EE7 c\xF4ng b\u1EB1ng c\xE1ch nh\u1EADp chi\u1EC1u r\u1ED9ng c\u1EE7a t\u1EEBng c\u1ED9t theo d\u1EA1ng %"
              )
            })
          ]
        })
      })
    ]
  });
};

exports.QuickLayout = QuickLayout;
