// @tempi/core-editor/dist/configs/ColorPickerConfig/ColorPickerConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var React = require('react');
var ColorPickerV2 = require('../../components/ColorPickerV2/ColorPickerV2.js');
var usePropSeparateDevice = require('../../hooks/usePropSeparateDevice.js');
var propKey = require('../../constants/propKey.js');
var FormItem = require('../../components/FormItem/FormItem.js');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const ColorPickerConfig = ({
  propKey: propKey$1,
  placeholder,
  formItemProps,
  colorPickerProps = {},
  gradientPicker = false,
  separateDevice
}) => {
  var _a, _b;
  const [value, handleChange] = usePropSeparateDevice.usePropSeparateDevice(propKey$1, separateDevice);
  const newPropKey = (_a = propKey$1 == null ? void 0 : propKey$1.replace) == null ? void 0 : _a.call(
    propKey$1,
    propKey.PROP_KEY.BackgroundColor,
    propKey.PROP_KEY.Background
  );
  const [valueV2, handleChangeGradientColor] = usePropSeparateDevice.usePropSeparateDevice(
    newPropKey,
    separateDevice
  );
  const [show, setShow] = React.useState(false);
  const isChangeBackgroundColorToBackground = ((_b = propKey$1 == null ? void 0 : propKey$1.includes) == null ? void 0 : _b.call(propKey$1, propKey.PROP_KEY.BackgroundColor)) && (!!valueV2 || gradientPicker);
  React.useEffect(() => {
    if (isChangeBackgroundColorToBackground && value) {
      handleChange(void 0);
      if (!valueV2)
        handleChangeGradientColor(value);
    }
  }, []);
  return /* @__PURE__ */ jsxRuntime.jsx(FormItem.FormItem, __spreadProps(__spreadValues({
    label: i18n.t("Ch\u1ECDn m\xE0u")
  }, formItemProps), {
    children: /* @__PURE__ */ jsxRuntime.jsx(ColorPickerV2.ColorPickerV2, __spreadValues({
      color: isChangeBackgroundColorToBackground ? valueV2 : value,
      onChangeColor: isChangeBackgroundColorToBackground ? handleChangeGradientColor : handleChange,
      showModal: show,
      setShowModal: setShow,
      placeholder: placeholder || i18n.t("Ch\u1ECDn m\xE0u"),
      hasGradientColor: gradientPicker
    }, colorPickerProps))
  }));
};

exports.ColorPickerConfig = ColorPickerConfig;
