// @tempi/core-elements/dist/configs/TipTapEditor/utils/custom-kit.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var core = require('@tiptap/core');
var Table = require('@tiptap/extension-table');
var TableCell = require('@tiptap/extension-table-cell');
var TableHeader = require('@tiptap/extension-table-header');
var TableRow = require('@tiptap/extension-table-row');
var TextAlign = require('@tiptap/extension-text-align');
var TextStyle = require('@tiptap/extension-text-style');
var StarterKit = require('@tiptap/starter-kit');
var constants = require('../constants.js');
var backgroundColor = require('./background-color.js');
var bold = require('./bold.js');
var bulletList = require('./bullet-list.js');
var color = require('./color.js');
var customCommands = require('./custom-commands.js');
var fontFamily = require('./font-family.js');
var fontSize = require('./font-size.js');
var fontWeight = require('./font-weight.js');
var letterSpacing = require('./letter-spacing.js');
var lineHeight = require('./line-height.js');
var link = require('./link.js');
var listItem = require('./list-item.js');
var orderedList = require('./ordered-list.js');
var textTransform = require('./text-transform.js');
var underline = require('./underline.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var Table__default = /*#__PURE__*/_interopDefaultLegacy(Table);
var TableCell__default = /*#__PURE__*/_interopDefaultLegacy(TableCell);
var TableHeader__default = /*#__PURE__*/_interopDefaultLegacy(TableHeader);
var TableRow__default = /*#__PURE__*/_interopDefaultLegacy(TableRow);
var TextAlign__default = /*#__PURE__*/_interopDefaultLegacy(TextAlign);
var TextStyle__default = /*#__PURE__*/_interopDefaultLegacy(TextStyle);
var StarterKit__default = /*#__PURE__*/_interopDefaultLegacy(StarterKit);

const CustomKit = core.Extension.create({
  name: "customKit",
  addExtensions() {
    const extensions = [
      customCommands.CustomCommands,
      fontFamily.FontFamily,
      fontSize.FontSize,
      fontWeight.FontWeight,
      letterSpacing.LetterSpacing,
      lineHeight.LineHeight,
      TextStyle__default["default"]
    ];
    const starterKitOptions = {};
    const {
      backgroundColor: backgroundColor$1,
      bold: bold$1,
      color: color$1,
      link: link$1,
      list,
      table,
      textAlign,
      textTransform: textTransform$1,
      underline: underline$1
    } = this.options;
    if (backgroundColor$1 !== false) {
      extensions.push(backgroundColor.BackgroundColor);
    }
    if (bold$1 !== false) {
      extensions.push(bold.Bold);
      starterKitOptions.bold = false;
    }
    if (color$1 !== false) {
      extensions.push(color.Color);
    }
    if (link$1 !== false) {
      extensions.push(link.Link);
    }
    if (list !== false) {
      extensions.push(bulletList.BulletList, orderedList.OrderedList, listItem.ListItem);
      starterKitOptions.bulletList = false;
      starterKitOptions.orderedList = false;
      starterKitOptions.listItem = false;
    }
    if (table !== false) {
      extensions.push(Table__default["default"], TableHeader__default["default"], TableRow__default["default"], TableCell__default["default"]);
    }
    if (textAlign !== false) {
      extensions.push(
        TextAlign__default["default"].configure({
          types: [constants.NODE_TYPES.heading, constants.NODE_TYPES.paragraph],
          defaultAlignment: "left"
        })
      );
    }
    if (textTransform$1 !== false) {
      extensions.push(textTransform.TextTransform);
    }
    if (underline$1 !== false) {
      extensions.push(underline.Underline);
    }
    extensions.push(StarterKit__default["default"].configure(starterKitOptions));
    return extensions;
  }
});

exports.CustomKit = CustomKit;
