// @tempi/core-elements/dist/configs/TipTapEditor/utils/font-weight.es.js
import { Extension } from '@tiptap/core';
import { NODE_TYPES } from '../constants.es.js';
import { updateTextAttribute } from '../../../utils/updateTextAttribute.es.js';

const FontWeight = Extension.create({
  name: "fontWeight",
  addOptions() {
    return {
      types: [
        NODE_TYPES.orderedList,
        NODE_TYPES.bulletList,
        NODE_TYPES.listItem,
        NODE_TYPES.heading,
        NODE_TYPES.paragraph,
        NODE_TYPES.textStyle
      ]
    };
  },
  addGlobalAttributes() {
    return [
      {
        types: this.options.types,
        attributes: {
          fontWeight: {
            default: null,
            parseHTML: (element) => parseInt(element.style.fontWeight) || null,
            renderHTML: (attributes) => {
              if (attributes.fontWeight !== null) {
                return {
                  style: `font-weight: ${attributes.fontWeight}`
                };
              }
            }
          }
        }
      }
    ];
  },
  addCommands() {
    return {
      setFontWeight: (fontWeight) => (commandProps) => {
        return updateTextAttribute(commandProps, this.options.types, {
          fontWeight
        });
      }
    };
  }
});

export { FontWeight };
