// @tempi/core-elements/dist/configs/TipTapEditor/utils/font-weight.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var core = require('@tiptap/core');
var constants = require('../constants.js');
var updateTextAttribute = require('../../../utils/updateTextAttribute.js');

const FontWeight = core.Extension.create({
  name: "fontWeight",
  addOptions() {
    return {
      types: [
        constants.NODE_TYPES.orderedList,
        constants.NODE_TYPES.bulletList,
        constants.NODE_TYPES.listItem,
        constants.NODE_TYPES.heading,
        constants.NODE_TYPES.paragraph,
        constants.NODE_TYPES.textStyle
      ]
    };
  },
  addGlobalAttributes() {
    return [
      {
        types: this.options.types,
        attributes: {
          fontWeight: {
            default: null,
            parseHTML: (element) => parseInt(element.style.fontWeight) || null,
            renderHTML: (attributes) => {
              if (attributes.fontWeight !== null) {
                return {
                  style: `font-weight: ${attributes.fontWeight}`
                };
              }
            }
          }
        }
      }
    ];
  },
  addCommands() {
    return {
      setFontWeight: (fontWeight) => (commandProps) => {
        return updateTextAttribute.updateTextAttribute(commandProps, this.options.types, {
          fontWeight
        });
      }
    };
  }
});

exports.FontWeight = FontWeight;
