// @tempi/core-elements/dist/configs/TipTapEditor/utils/font-size.es.js
import { Extension } from '@tiptap/core';
import { NODE_TYPES } from '../constants.es.js';
import { getVarFromStyle, renderResponsiveAttr } from './common.es.js';
import { TEXT_STYLE_KEY } from '../../../constants/fonts.es.js';
import { updateTextAttribute } from '../../../utils/updateTextAttribute.es.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
const FontSize = Extension.create({
  name: "fontSize",
  addOptions() {
    return {
      types: [
        NODE_TYPES.orderedList,
        NODE_TYPES.bulletList,
        NODE_TYPES.listItem,
        NODE_TYPES.heading,
        NODE_TYPES.paragraph,
        NODE_TYPES.textStyle
      ]
    };
  },
  addGlobalAttributes() {
    return [
      {
        types: this.options.types,
        attributes: {
          fontSize: {
            default: null,
            parseHTML: (element) => element.style.fontSize || null,
            renderHTML: (attributes) => {
              if (attributes.fontSize !== null) {
                return {
                  style: `font-size: ${attributes.fontSize}`
                };
              }
            }
          },
          fontSizeDesktop: {
            default: null,
            parseHTML: (element) => {
              const fontSize = getVarFromStyle(element, TEXT_STYLE_KEY.fontSize, "desktop") || element.style.fontSize;
              return fontSize || null;
            },
            renderHTML: (attributes) => {
              return renderResponsiveAttr(
                attributes,
                TEXT_STYLE_KEY.fontSize,
                "desktop"
              );
            }
          },
          fontSizeMobile: {
            default: null,
            parseHTML: (element) => {
              const fontSize = getVarFromStyle(element, TEXT_STYLE_KEY.fontSize, "mobile") || element.style.fontSize;
              return fontSize || null;
            },
            renderHTML: (attributes) => {
              return renderResponsiveAttr(
                attributes,
                TEXT_STYLE_KEY.fontSize,
                "mobile"
              );
            }
          }
        }
      }
    ];
  },
  addCommands() {
    return {
      setFontSize: (fontSize, device) => (commandProps) => {
        let attributes = {
          fontSize
        };
        if (device) {
          attributes = __spreadValues({
            fontSize: null
          }, device === "mobile" ? { fontSizeMobile: fontSize } : { fontSizeDesktop: fontSize });
        }
        return updateTextAttribute(
          commandProps,
          this.options.types,
          attributes
        );
      }
    };
  }
});

export { FontSize };
