// @tempi/core-elements/dist/configs/TipTapEditor/utils/letter-spacing.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var core = require('@tiptap/core');
var constants = require('../constants.js');
var updateTextAttribute = require('../../../utils/updateTextAttribute.js');

const LetterSpacing = core.Extension.create({
  name: "letterSpacing",
  addOptions() {
    return {
      types: [
        constants.NODE_TYPES.orderedList,
        constants.NODE_TYPES.bulletList,
        constants.NODE_TYPES.listItem,
        constants.NODE_TYPES.heading,
        constants.NODE_TYPES.paragraph,
        constants.NODE_TYPES.textStyle
      ]
    };
  },
  addGlobalAttributes() {
    return [
      {
        types: this.options.types,
        attributes: {
          letterSpacing: {
            default: null,
            parseHTML: (element) => element.style.letterSpacing || null,
            renderHTML: (attributes) => {
              if (attributes.letterSpacing !== null) {
                return {
                  style: `letter-spacing: ${attributes.letterSpacing}`
                };
              }
            }
          }
        }
      }
    ];
  },
  addCommands() {
    return {
      setLetterSpacing: (letterSpacing) => (commandProps) => {
        return updateTextAttribute.updateTextAttribute(commandProps, this.options.types, {
          letterSpacing
        });
      }
    };
  }
});

exports.LetterSpacing = LetterSpacing;
