// @tempi/core-elements/dist/configs/TipTapEditor/utils/letter-spacing.es.js
import { Extension } from '@tiptap/core';
import { NODE_TYPES } from '../constants.es.js';
import { updateTextAttribute } from '../../../utils/updateTextAttribute.es.js';

const LetterSpacing = Extension.create({
  name: "letterSpacing",
  addOptions() {
    return {
      types: [
        NODE_TYPES.orderedList,
        NODE_TYPES.bulletList,
        NODE_TYPES.listItem,
        NODE_TYPES.heading,
        NODE_TYPES.paragraph,
        NODE_TYPES.textStyle
      ]
    };
  },
  addGlobalAttributes() {
    return [
      {
        types: this.options.types,
        attributes: {
          letterSpacing: {
            default: null,
            parseHTML: (element) => element.style.letterSpacing || null,
            renderHTML: (attributes) => {
              if (attributes.letterSpacing !== null) {
                return {
                  style: `letter-spacing: ${attributes.letterSpacing}`
                };
              }
            }
          }
        }
      }
    ];
  },
  addCommands() {
    return {
      setLetterSpacing: (letterSpacing) => (commandProps) => {
        return updateTextAttribute(commandProps, this.options.types, {
          letterSpacing
        });
      }
    };
  }
});

export { LetterSpacing };
