// @tempi/core-elements/dist/configs/TipTapEditor/utils/line-height.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var core = require('@tiptap/core');
var constants = require('../constants.js');
var updateTextAttribute = require('../../../utils/updateTextAttribute.js');

const LineHeight = core.Extension.create({
  name: "lineHeight",
  addOptions() {
    return {
      types: [
        constants.NODE_TYPES.orderedList,
        constants.NODE_TYPES.bulletList,
        constants.NODE_TYPES.listItem,
        constants.NODE_TYPES.heading,
        constants.NODE_TYPES.paragraph,
        constants.NODE_TYPES.textStyle
      ]
    };
  },
  addGlobalAttributes() {
    return [
      {
        types: this.options.types,
        attributes: {
          lineHeight: {
            default: null,
            parseHTML: (element) => element.style.lineHeight || null,
            renderHTML: (attributes) => {
              if (attributes.lineHeight !== null) {
                return {
                  style: `line-height: ${attributes.lineHeight}`
                };
              }
            }
          }
        }
      }
    ];
  },
  addCommands() {
    return {
      setLineHeight: (lineHeight) => (commandProps) => {
        return updateTextAttribute.updateTextAttribute(commandProps, this.options.types, {
          lineHeight
        });
      }
    };
  }
});

exports.LineHeight = LineHeight;
