// @tempi/core-elements/dist/configs/TipTapEditor/utils/line-height.es.js
import { Extension } from '@tiptap/core';
import { NODE_TYPES } from '../constants.es.js';
import { updateTextAttribute } from '../../../utils/updateTextAttribute.es.js';

const LineHeight = Extension.create({
  name: "lineHeight",
  addOptions() {
    return {
      types: [
        NODE_TYPES.orderedList,
        NODE_TYPES.bulletList,
        NODE_TYPES.listItem,
        NODE_TYPES.heading,
        NODE_TYPES.paragraph,
        NODE_TYPES.textStyle
      ]
    };
  },
  addGlobalAttributes() {
    return [
      {
        types: this.options.types,
        attributes: {
          lineHeight: {
            default: null,
            parseHTML: (element) => element.style.lineHeight || null,
            renderHTML: (attributes) => {
              if (attributes.lineHeight !== null) {
                return {
                  style: `line-height: ${attributes.lineHeight}`
                };
              }
            }
          }
        }
      }
    ];
  },
  addCommands() {
    return {
      setLineHeight: (lineHeight) => (commandProps) => {
        return updateTextAttribute(commandProps, this.options.types, {
          lineHeight
        });
      }
    };
  }
});

export { LineHeight };
