// @tempi/core-elements/dist/configs/TipTapEditor/utils/text-transform.es.js
import { Extension } from '@tiptap/core';
import { NODE_TYPES } from '../constants.es.js';
import { updateTextAttribute } from '../../../utils/updateTextAttribute.es.js';

const TextTransform = Extension.create({
  name: "textTransform",
  addOptions() {
    return {
      types: [
        NODE_TYPES.orderedList,
        NODE_TYPES.bulletList,
        NODE_TYPES.listItem,
        NODE_TYPES.heading,
        NODE_TYPES.paragraph,
        NODE_TYPES.textStyle
      ]
    };
  },
  addGlobalAttributes() {
    return [
      {
        types: this.options.types,
        attributes: {
          textTransform: {
            default: null,
            parseHTML: (element) => element.style.textTransform || null,
            renderHTML: (attributes) => {
              if (attributes.textTransform !== null) {
                return {
                  style: `text-transform: ${attributes.textTransform}`
                };
              }
            }
          }
        }
      }
    ];
  },
  addCommands() {
    return {
      setTextTransform: (textTransform) => (commandProps) => {
        return updateTextAttribute(commandProps, this.options.types, {
          textTransform
        });
      }
    };
  }
});

export { TextTransform };
