// @tempi/core-elements/dist/components/button-submit/v3/quick-setting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../../i18n.es.js';
import { StarOutlined } from '@ant-design/icons';
import { AntdCustomIcon } from '@tempi/icons-dev';
import { SubQuickSetting, QuickSettingButton, OpenSettingDialog } from '@tempi/core-editor';
import { PROP_KEY } from '../../../constants/propKey.es.js';
import { TypographyQuickSetting } from '../../../quick-configs/typography/TypographyQuickSetting.es.js';
import { ButtonConfig } from '../../button/v2/configs/ButtonConfig.es.js';
import { IconButtonConfig } from '../../button/v2/configs/IconButtonConfig.es.js';
import { AlignmentQuickSetting } from '../../../quick-configs/AlignmentQuickSetting/AlignmentQuickSetting.es.js';
import { ButtonColorConfig } from '../../button/v2/configs/ButtonColorConfig.es.js';

const ButtonSummitFormQuickSetting = ({
  editor
}) => {
  const alignPropKey = PROP_KEY.joinKey(
    PROP_KEY.CommonStyle,
    PROP_KEY.JustifyContent
  );
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(TypographyQuickSetting, {
        editor
      }),
      /* @__PURE__ */ jsx(SubQuickSetting, {
        content: /* @__PURE__ */ jsx(ButtonConfig, {}),
        children: /* @__PURE__ */ jsx(QuickSettingButton, {
          tooltip: t("T\xF9y ch\u1EC9nh n\xFAt b\u1EA5m"),
          children: /* @__PURE__ */ jsx(AntdCustomIcon, {
            name: "ButtonPointerOutlined",
            style: {
              fontSize: 16
            }
          })
        })
      }),
      /* @__PURE__ */ jsx(SubQuickSetting, {
        content: /* @__PURE__ */ jsx(IconButtonConfig, {}),
        children: /* @__PURE__ */ jsx(QuickSettingButton, {
          tooltip: t("T\xF9y ch\u1EC9nh bi\u1EC3u t\u01B0\u1EE3ng"),
          children: /* @__PURE__ */ jsx(StarOutlined, {})
        })
      }),
      /* @__PURE__ */ jsx(AlignmentQuickSetting, {
        propKey: alignPropKey,
        listAlignment: ["flex-start", "center", "flex-end"]
      }),
      /* @__PURE__ */ jsx(ButtonColorConfig, {}),
      /* @__PURE__ */ jsx(OpenSettingDialog, {})
    ]
  });
};

export { ButtonSummitFormQuickSetting };
