// @tempi/core-elements/dist/components/button-submit/v3/selector.es.js
import { jsx } from 'react/jsx-runtime';
import { useEffect } from 'react';
import { useNode, useBuilder, useProp, RenderQuickSetting, crafting, AdvanceAttributes } from '@tempi/core-editor';
import { DEFAULT_BUTTON_SUBMIT_V3 } from './constants.es.js';
import { ButtonSummitFormQuickSetting } from './quick-setting.es.js';
import { useCustomTiptap } from '../../../configs/TipTapEditor/hooks.es.js';
import { ButtonWrapper, ButtonV2 } from '../../button/v2/ui.es.js';
import { PROP_KEY } from '../../../constants/propKey.es.js';
import { CustomKit } from '../../../configs/TipTapEditor/utils/custom-kit.es.js';
import { EditorContent } from '../../../configs/TipTapEditor/components/EditorContent.es.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const ButtonSubmitV3Editor = (props) => {
  const _a = props, { commonStyle } = _a, rest = __objRest(_a, ["commonStyle"]);
  const {
    connectors: { connect, drag },
    isActive
  } = useNode((node) => ({
    isActive: node.events.selected,
    parent: node.data.parent
  }));
  const { track } = useBuilder();
  const [width, setWidth] = useProp("buttonStyle.width");
  const [height, setHeight] = useProp("buttonStyle.height");
  useEffect(() => {
    if (!width)
      setWidth(160);
    if (!height)
      setHeight(40);
  }, []);
  useEffect(() => {
    if (isActive) {
      track == null ? void 0 : track({
        regionName: "ButtonSubmitEditorSetting",
        contentName: "quickSettingBtn"
      });
    }
  }, [isActive]);
  const { editor, setIsEditable } = useCustomTiptap(
    {
      editorOptions: {
        extensions: [
          CustomKit.configure({
            backgroundColor: false,
            bold: false,
            color: false,
            link: false,
            list: false,
            table: false,
            textTransform: false,
            underline: false
          })
        ]
      }
    },
    PROP_KEY.Text
  );
  return /* @__PURE__ */ jsx(RenderQuickSetting, {
    open: isActive,
    content: /* @__PURE__ */ jsx(ButtonSummitFormQuickSetting, {
      editor
    }),
    children: /* @__PURE__ */ jsx(ButtonWrapper, {
      ref: (ref) => connect(drag(ref)),
      commonStyle,
      style: { pointerEvents: "initial" },
      onClick: (e) => {
        e.preventDefault();
      },
      children: /* @__PURE__ */ jsx(ButtonV2, __spreadValues({
        commonStyle,
        customText: /* @__PURE__ */ jsx(EditorContent, {
          editor,
          setIsEditable
        }),
        isEditingMode: true
      }, rest))
    })
  });
};
ButtonSubmitV3Editor.craft = crafting({
  displayName: "N\xFAt b\u1EA5m",
  tag: ButtonV2.displayName.toLowerCase(),
  advanceAttributes: AdvanceAttributes,
  props: DEFAULT_BUTTON_SUBMIT_V3,
  rules: {
    canDrag: () => false
  }
});

export { ButtonSubmitV3Editor };
