// @tempi/core-elements/dist/components/contact-bubble/settings/BubbleItemSetting.es.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { t } from '../../../i18n.es.js';
import styled from '@emotion/styled';
import { useEditor, getEditorName, usePropSeparateDevice, ImageConfig, InputConfig, RangeSliderConfig, SelectConfig, SelectPopupConfig } from '@tempi/core-editor';
import { ROOT } from '@tempi/core-renderer';
import { useState } from 'react';
import { EventType } from '../type.es.js';
import { theme } from 'antd';
import { TrackingSettingModal } from './TrackingSettingModal.es.js';
import { PopupEditorV1 } from '../../popup/v1/selector.es.js';
import { PROP_KEY } from '../../../constants/propKey.es.js';

const eventOptions = [
  { value: EventType.HREF, label: t("\u0110\u01B0\u1EDDng d\u1EABn URL/S\u1ED1 \u0111i\u1EC7n tho\u1EA1i") },
  { value: EventType.POPUP, label: t("M\u1EDF popup") }
];
const BubbleItemSetting = ({
  index,
  device
}) => {
  var _a;
  const { token } = theme.useToken();
  const [showTrackingSetting, setShowTrackingSetting] = useState(false);
  const { nodes } = useEditor((state) => {
    return { nodes: state.nodes };
  });
  const popupListSelection = [
    ...new Set(
      (_a = nodes[ROOT]) == null ? void 0 : _a.data.nodes.filter(
        (key) => {
          var _a2, _b, _c;
          return ((_b = (_a2 = nodes == null ? void 0 : nodes[key]) == null ? void 0 : _a2.data) == null ? void 0 : _b.name) === getEditorName({ PopupEditorV1 }) && !((_c = nodes[key]) == null ? void 0 : _c.data.props.isMainPopup);
        }
      )
    )
  ].map((id) => ({
    label: nodes[id].data.props.name,
    value: id
  }));
  const bubbleItemPropKey = PROP_KEY.joinKey(
    device === "desktop" ? PROP_KEY.Desktop : PROP_KEY.Mobile,
    PROP_KEY.Items,
    index.toString()
  );
  const [eventType, setEventType] = usePropSeparateDevice(
    PROP_KEY.joinKey(bubbleItemPropKey, PROP_KEY.EventType)
  );
  const renderEventConfig = (eventType2) => {
    switch (eventType2) {
      case EventType.POPUP:
        return /* @__PURE__ */ jsx(SelectPopupConfig, {
          label: t("M\u1EDF popup"),
          placeholder: t("Ch\u1ECDn popup"),
          options: popupListSelection,
          propKey: PROP_KEY.joinKey(bubbleItemPropKey, PROP_KEY.Popup),
          itemIndex: index,
          device
        });
      default:
        return /* @__PURE__ */ jsx(InputConfig, {
          propKey: PROP_KEY.joinKey(bubbleItemPropKey, PROP_KEY.Href),
          formItemProps: {
            label: t("\u0110\u01B0\u1EDDng d\u1EABn URL/S\u1ED1 \u0111i\u1EC7n tho\u1EA1i"),
            layout: "vertical"
          },
          inputProps: {
            placeholder: t("Nh\u1EADp \u0111\u01B0\u1EDDng d\u1EABn URL ho\u1EB7c s\u1ED1 \u0111i\u1EC7n tho\u1EA1i")
          }
        });
    }
  };
  return /* @__PURE__ */ jsxs(Wrapper, {
    children: [
      /* @__PURE__ */ jsx(ImageConfig, {
        propKey: PROP_KEY.joinKey(
          bubbleItemPropKey,
          PROP_KEY.Icon,
          PROP_KEY.Src
        )
      }),
      /* @__PURE__ */ jsx(InputConfig, {
        propKey: PROP_KEY.joinKey(bubbleItemPropKey, PROP_KEY.Title),
        formItemProps: {
          label: t("contact_bubble_item_title"),
          layout: "vertical"
        },
        inputProps: {
          placeholder: t("Nh\u1EADp n\u1ED9i dung")
        }
      }),
      /* @__PURE__ */ jsx(RangeSliderConfig, {
        formItemProps: { label: t("Chi\u1EC1u cao n\xFAt") },
        propKey: PROP_KEY.joinKey(bubbleItemPropKey, PROP_KEY.Size),
        sliderProps: {
          defaultValue: 50,
          min: 10,
          max: 100
        },
        unitProps: {
          defaultValue: "px"
        }
      }),
      /* @__PURE__ */ jsx(SelectConfig, {
        options: eventOptions,
        formItemProps: {
          label: t("click_event"),
          style: { marginBottom: "0.5rem" }
        },
        selectProps: {
          allowClear: false,
          placeholder: t("Ch\u1ECDn s\u1EF1 ki\u1EC7n"),
          style: { width: "100%" },
          onChange: (val) => {
            setEventType(val);
          },
          value: eventType || EventType.HREF
        }
      }),
      renderEventConfig(eventType),
      /* @__PURE__ */ jsx("div", {
        style: { color: token.colorPrimary, cursor: "pointer" },
        onClick: () => setShowTrackingSetting(true),
        children: t("conversion_tracking_setting")
      }),
      /* @__PURE__ */ jsx(TrackingSettingModal, {
        bubbleItemPropKey,
        isOpen: showTrackingSetting,
        setShowTrackingSetting
      })
    ]
  });
};
const Wrapper = styled.div`
  margin-top: 1em;
`;

export { BubbleItemSetting };
