// @tempi/core-elements/dist/components/contact-bubble/settings/TrackingSettingModal.es.js
import { jsx } from 'react/jsx-runtime';
import { Modal, Button, Tabs } from 'antd';
import { useFrame } from 'react-frame-component';
import { t } from '../../../i18n.es.js';
import { useBuilder, TrackingConfig, TekoTrackingConfig } from '@tempi/core-editor';
import { PROP_KEY } from '../../../constants/propKey.es.js';

const TrackingSettingModal = ({
  separateDevice,
  bubbleItemPropKey,
  isOpen,
  setShowTrackingSetting
}) => {
  const { enableTekoTracking } = useBuilder();
  const { document: frameDocument } = useFrame();
  const onClose = () => {
    setShowTrackingSetting(false);
  };
  const getTabItems = () => {
    const tabItems = [
      {
        label: t("Chuy\u1EC3n \u0111\u1ED5i (Tracking)"),
        key: "1",
        children: /* @__PURE__ */ jsx(TrackingConfig, {
          separateDevice,
          config: "event",
          propKey: PROP_KEY.joinKey(bubbleItemPropKey, PROP_KEY.Event)
        })
      }
    ];
    if (enableTekoTracking)
      tabItems.push({
        label: "Teko tracking",
        key: "3",
        children: /* @__PURE__ */ jsx(TekoTrackingConfig, {
          propKey: PROP_KEY.joinKey(bubbleItemPropKey, PROP_KEY.Event)
        })
      });
    return tabItems;
  };
  return /* @__PURE__ */ jsx(Modal, {
    open: isOpen,
    zIndex: 1080,
    title: t("conversion_tracking_setting"),
    destroyOnClose: true,
    footer: [
      /* @__PURE__ */ jsx(Button, {
        type: "primary",
        shape: "round",
        onClick: onClose,
        children: t("\u0110\xF3ng")
      })
    ],
    getContainer: () => (frameDocument == null ? void 0 : frameDocument.body) || (document == null ? void 0 : document.body),
    children: /* @__PURE__ */ jsx(Tabs, {
      items: getTabItems()
    })
  });
};

export { TrackingSettingModal };
