// @tempi/core-elements/dist/components/contact-bubble/settings/TrackingSettingModal.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var antd = require('antd');
var reactFrameComponent = require('react-frame-component');
var i18n = require('../../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var propKey = require('../../../constants/propKey.js');

const TrackingSettingModal = ({
  separateDevice,
  bubbleItemPropKey,
  isOpen,
  setShowTrackingSetting
}) => {
  const { enableTekoTracking } = coreEditor.useBuilder();
  const { document: frameDocument } = reactFrameComponent.useFrame();
  const onClose = () => {
    setShowTrackingSetting(false);
  };
  const getTabItems = () => {
    const tabItems = [
      {
        label: i18n.t("Chuy\u1EC3n \u0111\u1ED5i (Tracking)"),
        key: "1",
        children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.TrackingConfig, {
          separateDevice,
          config: "event",
          propKey: propKey.PROP_KEY.joinKey(bubbleItemPropKey, propKey.PROP_KEY.Event)
        })
      }
    ];
    if (enableTekoTracking)
      tabItems.push({
        label: "Teko tracking",
        key: "3",
        children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.TekoTrackingConfig, {
          propKey: propKey.PROP_KEY.joinKey(bubbleItemPropKey, propKey.PROP_KEY.Event)
        })
      });
    return tabItems;
  };
  return /* @__PURE__ */ jsxRuntime.jsx(antd.Modal, {
    open: isOpen,
    zIndex: 1080,
    title: i18n.t("conversion_tracking_setting"),
    destroyOnClose: true,
    footer: [
      /* @__PURE__ */ jsxRuntime.jsx(antd.Button, {
        type: "primary",
        shape: "round",
        onClick: onClose,
        children: i18n.t("\u0110\xF3ng")
      })
    ],
    getContainer: () => (frameDocument == null ? void 0 : frameDocument.body) || (document == null ? void 0 : document.body),
    children: /* @__PURE__ */ jsxRuntime.jsx(antd.Tabs, {
      items: getTabItems()
    })
  });
};

exports.TrackingSettingModal = TrackingSettingModal;
