// @tempi/core-elements/dist/components/website-header/constants.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var i18n = require('../../i18n.js');
var coreRenderer = require('@tempi/core-renderer');

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
const MOBILE_WIDTH = 767;
const DEFAULT_WEBSITE_HEADER_DESKTOP = {
  stickyTop: true,
  menuConfig: {
    commonStyle: {
      columnGap: 32
    },
    align: coreRenderer.Alignment.right
  },
  logoConfig: {
    show: true,
    src: "https://lh3.googleusercontent.com/bhg6FoxZPA5S6q50_Gmu6YgMt2GcKxpjsbnaRvvHUwSV9JSnAj83TpbJEHIJjgKnbNoZmM7Mnx0pBlUau5CG2yIOG5v7rGc",
    align: coreRenderer.Alignment.left,
    commonStyle: {
      width: 100
    },
    spacingWithMenu: 32
  },
  menuItemConfig: {
    commonStyle: {
      color: "#000000",
      hover: {
        color: "#008cff"
      },
      fontSize: 14
    }
  },
  languageConfig: {
    align: coreRenderer.Alignment.right,
    flagStyle: "round",
    layout: "short",
    showFlag: true,
    flagSize: 24
  },
  commonStyle: {
    backgroundColor: "#ffffff",
    padding: {
      top: 30,
      right: 16,
      bottom: 30,
      left: 16
    }
  }
};
const DEFAULT_WEBSITE_HEADER_MOBILE = {
  commonStyle: {
    padding: {
      top: 24,
      right: 16,
      bottom: 24,
      left: 16
    }
  }
};
const DEFAULT_WEBSITE_HEADER = {
  desktop: __spreadValues({}, DEFAULT_WEBSITE_HEADER_DESKTOP),
  mobile: __spreadValues({}, DEFAULT_WEBSITE_HEADER_MOBILE)
};
const DEFAULT_BUTTONS = [
  {
    text: "",
    buttonStyle: {
      backgroundColor: "#7367F0",
      color: "#ffffff",
      height: 40,
      width: 100,
      borderRadius: 16,
      borderStyle: "solid",
      borderWidth: 0
    }
  }
];
[
  {
    id: "5eBqBWoa",
    name: i18n.t("Trang ch\u1EE7"),
    target: "SELF",
    isHidden: false,
    linkType: "PAGE",
    page: {
      id: 455,
      isHomepage: true,
      slug: null
    }
  },
  {
    id: "iCScxX63",
    name: "Blog",
    target: "BLANK",
    linkType: "PAGE",
    page: {
      id: 456,
      isHomepage: false,
      slug: "add-page-test"
    }
  },
  {
    id: "LvTwfsZx",
    name: "About",
    target: "BLANK",
    isHidden: true,
    linkType: "PAGE",
    page: {
      id: 458,
      slug: "test-122312",
      isHomepage: false
    }
  },
  {
    id: "7DWki3lv",
    name: "Services",
    target: "SELF",
    isHidden: false,
    linkType: "PAGE",
    page: {
      id: 419,
      slug: "test-add-page-3",
      isHomepage: false
    }
  },
  {
    id: "LebWjo9l",
    name: "Child1",
    linkType: "PAGE",
    parentId: "7DWki3lv",
    page: {
      id: 459,
      slug: "test-12123",
      isHomepage: false
    }
  },
  {
    id: "moar1sju",
    name: "Contact",
    target: "SELF",
    linkType: "PAGE",
    page: {
      id: 460,
      slug: "test-123123123",
      isHomepage: false
    }
  },
  {
    id: "aVtmfMh_",
    href: "https://vnshop.vn/thung-48-bich-sua-tuoi-tiet-trung-khong-duong-dalatmilk-220ml--s210201486?sku=210201486",
    name: "Child2",
    target: "BLANK",
    linkType: "LINK",
    nofollow: true,
    parentId: "moar1sju"
  },
  {
    id: "IOm6L3zm",
    href: "https://vnshop.vn/thung-48-bich-sua-tuoi-tiet-trung-khong-duong-dalatmilk-220ml--s210201486?sku=210201486",
    name: "Child3",
    target: "BLANK",
    linkType: "LINK",
    nofollow: true,
    parentId: "moar1sju"
  },
  {
    id: "Nuklf8V1",
    name: "Gallery",
    target: "SELF",
    linkType: "PAGE",
    page: {
      id: 454,
      slug: "test-qiwueqw-queqwe",
      isHomepage: false
    }
  },
  {
    id: "iR183_D0",
    href: "https://vnshop.vn/thung-48-bich-sua-tuoi-tiet-trung-khong-duong-dalatmilk-220ml--s210201486?sku=210201486",
    name: "Child4",
    target: "BLANK",
    isHidden: true,
    linkType: "LINK",
    nofollow: true,
    parentId: "Nuklf8V1"
  },
  {
    id: "1hr5sQFD",
    href: "https://vnshop.vn/thung-48-bich-sua-tuoi-tiet-trung-khong-duong-dalatmilk-220ml--s210201486?sku=210201486",
    name: "Child5",
    target: "BLANK",
    linkType: "LINK",
    nofollow: true,
    parentId: "Nuklf8V1"
  },
  {
    id: "vx6PjNLA",
    name: "Lorem Ipsum",
    target: "SELF",
    isHidden: false,
    linkType: "PAGE",
    page: {
      id: 448,
      slug: "test-add-2",
      isHomepage: false
    }
  }
];
const LIGHT_ACTIVE_ITEM_BACKGROUND_COLOR = "#f3f4f6";
const DARK_ACTIVE_ITEM_BACKGROUND_COLOR = "#4b5563";

exports.DARK_ACTIVE_ITEM_BACKGROUND_COLOR = DARK_ACTIVE_ITEM_BACKGROUND_COLOR;
exports.DEFAULT_BUTTONS = DEFAULT_BUTTONS;
exports.DEFAULT_WEBSITE_HEADER = DEFAULT_WEBSITE_HEADER;
exports.DEFAULT_WEBSITE_HEADER_DESKTOP = DEFAULT_WEBSITE_HEADER_DESKTOP;
exports.DEFAULT_WEBSITE_HEADER_MOBILE = DEFAULT_WEBSITE_HEADER_MOBILE;
exports.LIGHT_ACTIVE_ITEM_BACKGROUND_COLOR = LIGHT_ACTIVE_ITEM_BACKGROUND_COLOR;
exports.MOBILE_WIDTH = MOBILE_WIDTH;
