// @tempi/core-elements/dist/components/website-header/ContentSetting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { usePropSeparateDevice, SwitchConfig, AlignmentConfig, RangeSliderConfig, ImageConfig } from '@tempi/core-editor';
import { UnitEnum } from '@tempi/core-renderer';
import { Collapse } from 'antd';
import { useEffect } from 'react';
import { ButtonSetting } from './ButtonSetting.es.js';
import { LanguageSetting } from './LanguageSetting.es.js';
import { SearchSetting } from './SearchSetting.es.js';
import { PROP_KEY } from '../../constants/propKey.es.js';

const { Panel } = Collapse;
const HeaderV2ContentSetting = () => {
  const [showLogo] = usePropSeparateDevice(
    PROP_KEY.joinKey(PROP_KEY.LogoConfig, PROP_KEY.Show),
    true
  );
  const [paddingTop] = usePropSeparateDevice(
    PROP_KEY.joinKey(PROP_KEY.CommonStyle, PROP_KEY.Padding, PROP_KEY.Top),
    true
  );
  const [, setPaddingBottom] = usePropSeparateDevice(
    PROP_KEY.joinKey(PROP_KEY.CommonStyle, PROP_KEY.Padding, PROP_KEY.Bottom),
    true
  );
  useEffect(() => {
    setPaddingBottom(paddingTop);
  }, [paddingTop]);
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(SwitchConfig, {
        formItemProps: {
          label: t("G\u1EAFn \u0111\u1EA7u trang khi cu\u1ED9n xu\u1ED1ng")
        },
        propKey: PROP_KEY.joinKey(PROP_KEY.StickyTop),
        separateDevice: true
      }),
      /* @__PURE__ */ jsxs(Collapse, {
        expandIconPosition: "end",
        bordered: false,
        children: [
          /* @__PURE__ */ jsxs(Panel, {
            header: "Menu",
            children: [
              /* @__PURE__ */ jsx(AlignmentConfig, {
                formItemProps: {
                  label: t("V\u1ECB tr\xED menu")
                },
                propKey: PROP_KEY.joinKey(PROP_KEY.MenuConfig, PROP_KEY.Align),
                separateDevice: true
              }),
              /* @__PURE__ */ jsx(RangeSliderConfig, {
                formItemProps: {
                  label: t("Kho\u1EA3ng c\xE1ch gi\u1EEFa c\xE1c menu")
                },
                propKey: PROP_KEY.joinKey(
                  PROP_KEY.MenuConfig,
                  PROP_KEY.CommonStyle,
                  PROP_KEY.ColumnGap
                ),
                separateDevice: true,
                sliderProps: {
                  min: 0,
                  max: 64
                },
                inputNumberProps: {
                  min: 0,
                  max: 64,
                  style: {
                    width: 100
                  }
                },
                unitProps: { defaultValue: UnitEnum.pixel }
              }),
              /* @__PURE__ */ jsx(RangeSliderConfig, {
                formItemProps: {
                  label: t("Kho\u1EA3ng c\xE1ch tr\xEAn / d\u01B0\u1EDBi menu")
                },
                propKey: PROP_KEY.joinKey(
                  PROP_KEY.CommonStyle,
                  PROP_KEY.Padding,
                  PROP_KEY.Top
                ),
                inputNumberProps: {
                  min: 0,
                  max: 128,
                  style: {
                    width: 100
                  }
                },
                sliderProps: {
                  min: 0,
                  max: 128
                },
                unitProps: { defaultValue: UnitEnum.pixel },
                separateDevice: true
              })
            ]
          }, "Menu"),
          /* @__PURE__ */ jsxs(Panel, {
            header: "Logo",
            children: [
              /* @__PURE__ */ jsx(SwitchConfig, {
                formItemProps: {
                  label: t("Hi\u1EC3n th\u1ECB Logo")
                },
                propKey: PROP_KEY.joinKey(PROP_KEY.LogoConfig, PROP_KEY.Show),
                separateDevice: true
              }),
              showLogo && /* @__PURE__ */ jsxs(Fragment, {
                children: [
                  /* @__PURE__ */ jsx(ImageConfig, {
                    propKey: PROP_KEY.joinKey(PROP_KEY.LogoConfig, PROP_KEY.Src),
                    widthPropKey: PROP_KEY.joinKey(
                      PROP_KEY.LogoConfig,
                      PROP_KEY.GoogleImageParams,
                      PROP_KEY.Width
                    ),
                    altPropKey: PROP_KEY.joinKey(PROP_KEY.LogoConfig, PROP_KEY.Alt),
                    separateDevice: true
                  }),
                  /* @__PURE__ */ jsx(RangeSliderConfig, {
                    sliderProps: {
                      min: 30,
                      max: 350
                    },
                    formItemProps: {
                      label: t("Chi\u1EC1u r\u1ED9ng logo")
                    },
                    propKey: PROP_KEY.joinKey(
                      PROP_KEY.LogoConfig,
                      PROP_KEY.CommonStyle,
                      PROP_KEY.Width
                    ),
                    inputNumberProps: {
                      min: 30,
                      max: 350,
                      style: {
                        width: 100
                      }
                    },
                    unitProps: { defaultValue: UnitEnum.pixel },
                    separateDevice: true
                  }),
                  /* @__PURE__ */ jsx(RangeSliderConfig, {
                    formItemProps: {
                      label: t("Kho\u1EA3ng c\xE1ch logo v\u1EDBi menu")
                    },
                    propKey: PROP_KEY.joinKey(
                      PROP_KEY.LogoConfig,
                      PROP_KEY.SpacingWithMenu
                    ),
                    sliderProps: {
                      min: 0,
                      max: 64
                    },
                    inputNumberProps: {
                      min: 0,
                      max: 64,
                      style: {
                        width: 100
                      }
                    },
                    unitProps: { defaultValue: UnitEnum.pixel },
                    separateDevice: true
                  }),
                  /* @__PURE__ */ jsx(AlignmentConfig, {
                    formItemProps: {
                      label: t("V\u1ECB tr\xED logo")
                    },
                    propKey: PROP_KEY.joinKey(PROP_KEY.LogoConfig, PROP_KEY.Align),
                    separateDevice: true
                  })
                ]
              })
            ]
          }, "Logo"),
          /* @__PURE__ */ jsx(Panel, {
            header: t("N\xFAt k\xEAu g\u1ECDi h\xE0nh \u0111\u1ED9ng"),
            children: /* @__PURE__ */ jsx(ButtonSetting, {})
          }, "ActionButton"),
          /* @__PURE__ */ jsx(Panel, {
            header: t("N\xFAt chuy\u1EC3n \u0111\u1ED5i ng\xF4n ng\u1EEF"),
            children: /* @__PURE__ */ jsx(LanguageSetting, {})
          }, "LanguageSetting")
        ]
      }),
      /* @__PURE__ */ jsx(SearchSetting, {})
    ]
  });
};

export { HeaderV2ContentSetting };
