// @tempi/core-elements/dist/components/website-header/LanguageSetting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { usePropSeparateDevice, AlignmentConfig, SelectConfig, SwitchConfig, RangeSliderConfig } from '@tempi/core-editor';
import { PROP_KEY } from '../../constants/propKey.es.js';

const LanguageSetting = () => {
  const [languageLayout] = usePropSeparateDevice(
    PROP_KEY.joinKey(PROP_KEY.LanguageConfig, PROP_KEY.Layout),
    true
  );
  const languageLayoutOptions = [
    { value: "full", label: t("T\xEAn \u0111\u1EA7y \u0111\u1EE7") },
    { value: "short", label: t("T\xEAn vi\u1EBFt t\u1EAFt") },
    { value: "flag", label: t("Ch\u1EC9 hi\u1EC3n th\u1ECB c\u1EDD") }
  ];
  const flagStyleOptions = [
    { value: "round", label: t("H\xECnh tr\xF2n") },
    { value: "rectangle", label: t("H\xECnh ch\u1EEF nh\u1EADt") }
  ];
  const isShowSwitchFlag = languageLayout !== "flag";
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(AlignmentConfig, {
        formItemProps: {
          label: t("V\u1ECB tr\xED n\xFAt")
        },
        propKey: PROP_KEY.joinKey(PROP_KEY.LanguageConfig, PROP_KEY.Align),
        separateDevice: true,
        buttonGroupProps: {
          listAlignment: ["left", "right"]
        }
      }),
      /* @__PURE__ */ jsx(SelectConfig, {
        formItemProps: {
          label: t("Ki\u1EC3u hi\u1EC3n th\u1ECB ng\xF4n ng\u1EEF")
        },
        propKey: PROP_KEY.joinKey(PROP_KEY.LanguageConfig, PROP_KEY.Layout),
        selectProps: { options: languageLayoutOptions, allowClear: false },
        separateDevice: true
      }),
      isShowSwitchFlag && /* @__PURE__ */ jsx(SwitchConfig, {
        formItemProps: {
          label: t("Hi\u1EC3n th\u1ECB c\u1EDD")
        },
        propKey: PROP_KEY.joinKey(PROP_KEY.LanguageConfig, PROP_KEY.ShowFlag),
        separateDevice: true
      }),
      /* @__PURE__ */ jsx(SelectConfig, {
        formItemProps: {
          label: t("Ki\u1EC3u hi\u1EC3n th\u1ECB c\u1EDD")
        },
        propKey: PROP_KEY.joinKey(PROP_KEY.LanguageConfig, PROP_KEY.FlagStyle),
        selectProps: { options: flagStyleOptions, allowClear: false },
        separateDevice: true
      }),
      /* @__PURE__ */ jsx(RangeSliderConfig, {
        formItemProps: {
          label: t("K\xEDch th\u01B0\u1EDBc c\u1EDD")
        },
        sliderProps: {
          min: 12,
          max: 50
        },
        propKey: PROP_KEY.joinKey(PROP_KEY.LanguageConfig, PROP_KEY.FlagSize),
        unitProps: {
          defaultValue: "px"
        },
        separateDevice: true
      })
    ]
  });
};

export { LanguageSetting };
