// @tempi/core-elements/dist/components/website-header/LanguageSetting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var propKey = require('../../constants/propKey.js');

const LanguageSetting = () => {
  const [languageLayout] = coreEditor.usePropSeparateDevice(
    propKey.PROP_KEY.joinKey(propKey.PROP_KEY.LanguageConfig, propKey.PROP_KEY.Layout),
    true
  );
  const languageLayoutOptions = [
    { value: "full", label: i18n.t("T\xEAn \u0111\u1EA7y \u0111\u1EE7") },
    { value: "short", label: i18n.t("T\xEAn vi\u1EBFt t\u1EAFt") },
    { value: "flag", label: i18n.t("Ch\u1EC9 hi\u1EC3n th\u1ECB c\u1EDD") }
  ];
  const flagStyleOptions = [
    { value: "round", label: i18n.t("H\xECnh tr\xF2n") },
    { value: "rectangle", label: i18n.t("H\xECnh ch\u1EEF nh\u1EADt") }
  ];
  const isShowSwitchFlag = languageLayout !== "flag";
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.AlignmentConfig, {
        formItemProps: {
          label: i18n.t("V\u1ECB tr\xED n\xFAt")
        },
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.LanguageConfig, propKey.PROP_KEY.Align),
        separateDevice: true,
        buttonGroupProps: {
          listAlignment: ["left", "right"]
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
        formItemProps: {
          label: i18n.t("Ki\u1EC3u hi\u1EC3n th\u1ECB ng\xF4n ng\u1EEF")
        },
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.LanguageConfig, propKey.PROP_KEY.Layout),
        selectProps: { options: languageLayoutOptions, allowClear: false },
        separateDevice: true
      }),
      isShowSwitchFlag && /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
        formItemProps: {
          label: i18n.t("Hi\u1EC3n th\u1ECB c\u1EDD")
        },
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.LanguageConfig, propKey.PROP_KEY.ShowFlag),
        separateDevice: true
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
        formItemProps: {
          label: i18n.t("Ki\u1EC3u hi\u1EC3n th\u1ECB c\u1EDD")
        },
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.LanguageConfig, propKey.PROP_KEY.FlagStyle),
        selectProps: { options: flagStyleOptions, allowClear: false },
        separateDevice: true
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.RangeSliderConfig, {
        formItemProps: {
          label: i18n.t("K\xEDch th\u01B0\u1EDBc c\u1EDD")
        },
        sliderProps: {
          min: 12,
          max: 50
        },
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.LanguageConfig, propKey.PROP_KEY.FlagSize),
        unitProps: {
          defaultValue: "px"
        },
        separateDevice: true
      })
    ]
  });
};

exports.LanguageSetting = LanguageSetting;
