// @tempi/core-elements/dist/components/website-header/selector.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var ContentSetting = require('./ContentSetting.js');
var StyleSetting = require('./StyleSetting.js');
var QuickSetting = require('./QuickSetting.js');
var coreEditor = require('@tempi/core-editor');
var ui = require('./ui.js');
var RenderFloatingQuickSetting = require('../../core/RenderFloatingQuickSetting/RenderFloatingQuickSetting.js');
var AddBlockButton = require('../../core/Indicator/AddBlockButton.js');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const WebsiteHeaderEditor = (props) => {
  const {
    connectors: { connect }
  } = coreEditor.useNode();
  const { menu } = coreEditor.useBuilder();
  return /* @__PURE__ */ jsxRuntime.jsx("div", {
    ref: connect,
    children: /* @__PURE__ */ jsxRuntime.jsxs(RenderFloatingQuickSetting.RenderFloatingQuickSetting, {
      quickSetting: /* @__PURE__ */ jsxRuntime.jsx(QuickSetting.HeaderV2QuickSetting, {}),
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(ui.WebsiteHeader, __spreadProps(__spreadValues({}, props), {
          menu,
          isEditingMode: true
        })),
        /* @__PURE__ */ jsxRuntime.jsx(AddBlockButton.AddBlockButton, {})
      ]
    })
  });
};
WebsiteHeaderEditor.craft = coreEditor.crafting({
  displayName: ui.WebsiteHeader.displayName,
  tag: ui.WebsiteHeader.displayName.toLowerCase(),
  customAttributes: ContentSetting.HeaderV2ContentSetting,
  commonAttributes: StyleSetting.HeaderV2StyleSetting,
  rules: {
    canDrag: () => false
  }
});

exports.WebsiteHeaderEditor = WebsiteHeaderEditor;
