// @tempi/core-elements/dist/components/website-header/StyleSetting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { ColorPickerConfig, RangeSliderConfig, TextConfig, ShadowConfig } from '@tempi/core-editor';
import { PROP_KEY } from '../../constants/propKey.es.js';

const HeaderV2StyleSetting = () => {
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(ColorPickerConfig, {
        formItemProps: {
          label: t("M\xE0u n\u1EC1n")
        },
        propKey: PROP_KEY.joinKey(
          PROP_KEY.CommonStyle,
          PROP_KEY.BackgroundColor
        ),
        separateDevice: true,
        gradientPicker: true
      }),
      /* @__PURE__ */ jsx(ColorPickerConfig, {
        formItemProps: {
          label: t("M\xE0u menu")
        },
        propKey: PROP_KEY.joinKey(
          PROP_KEY.MenuItemConfig,
          PROP_KEY.CommonStyle,
          PROP_KEY.Color
        ),
        separateDevice: true
      }),
      /* @__PURE__ */ jsx(ColorPickerConfig, {
        formItemProps: {
          label: t("M\xE0u hover menu")
        },
        propKey: PROP_KEY.joinKey(
          PROP_KEY.MenuItemConfig,
          PROP_KEY.CommonStyle,
          PROP_KEY.HoverColor
        ),
        separateDevice: true
      }),
      /* @__PURE__ */ jsx(RangeSliderConfig, {
        formItemProps: { label: t("C\u1EE1 ch\u1EEF menu") },
        sliderProps: {
          step: 1,
          min: 12,
          max: 72,
          defaultValue: 14
        },
        propKey: PROP_KEY.joinKey(
          PROP_KEY.MenuItemConfig,
          PROP_KEY.CommonStyle,
          PROP_KEY.FontSize
        ),
        separateDevice: true,
        showInput: true,
        unitProps: {
          defaultValue: "px"
        }
      }),
      /* @__PURE__ */ jsx(TextConfig, {
        formItemProps: { label: t("Ki\u1EC3u ch\u1EEF menu") },
        propKey: PROP_KEY.joinKey(
          PROP_KEY.MenuItemConfig,
          PROP_KEY.CommonStyle
        )
      }),
      /* @__PURE__ */ jsx(ShadowConfig, {})
    ]
  });
};

export { HeaderV2StyleSetting };
