// @tempi/core-elements/dist/components/col/quick-setting.es.js
import { jsx, Fragment, jsxs } from 'react/jsx-runtime';
import { DragOutlined } from '@ant-design/icons';
import styled from '@emotion/styled';
import { ChevronDownRegular, Copy16Regular } from '@fluentui/react-icons';
import { useEditor, useNode, useBuilder, RenderQuickSetting, EventQuickSetting, OpenSettingDialog, DeleteNode, QuickSettingButton } from '@tempi/core-editor';
import { useState } from 'react';
import { MoveLeft } from '../../quick-configs/MoveLeft/MoveLeft.es.js';
import { MoveRight } from '../../quick-configs/MoveRight/MoveRight.es.js';

const ColQuickSetting = () => {
  const { actions, query } = useEditor();
  const {
    connectors: { drag },
    id,
    parentId
  } = useNode((node) => ({
    parentId: node.data.parent
  }));
  const { track } = useBuilder();
  const [enabledDrag, setEnableDrag] = useState(false);
  const onDeleteColumn = () => {
    actions.delete(id);
    const numOfColAfterDelete = query.node(parentId).childNodes().length;
    if (numOfColAfterDelete === 0)
      actions.history.throttle().delete(parentId);
  };
  return /* @__PURE__ */ jsx(Fragment, {
    children: /* @__PURE__ */ jsx(IndicatorDiv, {
      children: /* @__PURE__ */ jsx(RenderQuickSetting, {
        trigger: "click",
        content: /* @__PURE__ */ jsxs(Fragment, {
          children: [
            /* @__PURE__ */ jsx(MoveLeft, {}),
            /* @__PURE__ */ jsx(MoveRight, {}),
            /* @__PURE__ */ jsx(EventQuickSetting, {}),
            /* @__PURE__ */ jsx(OpenSettingDialog, {}),
            /* @__PURE__ */ jsx(CustomColDuplicate, {}),
            /* @__PURE__ */ jsx(DeleteNode, {
              customDelete: onDeleteColumn
            })
          ]
        }),
        onOpenChange: (visible) => {
          if (visible) {
            track == null ? void 0 : track({
              regionName: "ColQuickSetting",
              contentName: "quickSettingBtn"
            });
          }
        },
        destroyTooltipOnHide: false,
        children: /* @__PURE__ */ jsx(ColumnQuickSettingBtn, {
          ref: drag,
          onClick: () => actions.selectNode(id),
          onMouseEnter: () => setEnableDrag(true),
          onMouseLeave: () => setEnableDrag(false),
          children: enabledDrag ? /* @__PURE__ */ jsx(DragOutlined, {
            style: {
              color: "#fff",
              pointerEvents: "none",
              cursor: "move"
            }
          }) : /* @__PURE__ */ jsx(ChevronDownRegular, {
            fontSize: 12,
            style: {
              color: "#fff",
              pointerEvents: "none",
              cursor: "unset"
            }
          })
        })
      })
    })
  });
};
const IndicatorDiv = styled.div`
  font-size: 12px;
  line-height: 12px;
  position: absolute !important;
  display: flex;
  align-items: center;
  justify-content: left;
  right: -10px;
  top: -10px;
  z-index: 999;
`;
const ColumnQuickSettingBtn = styled.div`
  width: 18px;
  height: 18px;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #ffa500 !important;
  cursor: pointer;
  border-radius: 999px;
  transition-duration: 0.2s;
  &:hover {
    width: 24px;
    height: 24px;
  }
`;
const CustomColDuplicate = () => {
  const { duplicateNode, getSiblings } = useEditor();
  const { id, node, parentId } = useNode((node2) => ({
    node: node2,
    parentId: node2.data.parent
  }));
  return /* @__PURE__ */ jsx(QuickSettingButton, {
    onClick: () => {
      const siblings = getSiblings(id);
      if (siblings.length >= 12)
        return;
      const index = siblings.findIndex((el) => el === id);
      duplicateNode(node, parentId, index + 1);
    },
    children: /* @__PURE__ */ jsx(Copy16Regular, {})
  });
};

export { ColQuickSetting, CustomColDuplicate };
