// @tempi/core-elements/dist/components/col/style-setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var antd = require('antd');
var propKey = require('../../constants/propKey.js');

const ColStyle = ({
  propKey: propKey$1 = propKey.PROP_KEY.CommonStyle
}) => {
  return /* @__PURE__ */ jsxRuntime.jsx(jsxRuntime.Fragment, {
    children: /* @__PURE__ */ jsxRuntime.jsxs(antd.Collapse, {
      expandIconPosition: "end",
      bordered: false,
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(antd.Collapse.Panel, {
          header: i18n.t("M\xE0u n\u1EC1n"),
          children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
            propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.BackgroundColor),
            formItemProps: {
              label: i18n.t("M\xE0u n\u1EC1n")
            },
            gradientPicker: true
          })
        }, "background-color"),
        /* @__PURE__ */ jsxRuntime.jsx(antd.Collapse.Panel, {
          header: i18n.t("\u1EA2nh n\u1EC1n"),
          children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.BackgroundImageConfig, {
            propKey: propKey$1,
            allowConfingMarqueeBgImage: true
          })
        }, "background-image"),
        /* @__PURE__ */ jsxRuntime.jsx(antd.Collapse.Panel, {
          header: i18n.t("Video n\u1EC1n"),
          children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.BackgroundVideoConfig, {
            propKey: propKey$1
          })
        }, "background-video"),
        /* @__PURE__ */ jsxRuntime.jsxs(antd.Collapse.Panel, {
          header: i18n.t("Vi\u1EC1n v\xE0 bo g\xF3c"),
          children: [
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.BorderConfig, {
              propKey: propKey$1
            }),
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ShadowConfig, {})
          ]
        }, "border-and-radius")
      ]
    })
  });
};

exports.ColStyle = ColStyle;
