// @tempi/core-elements/dist/components/col/style-setting.es.js
import { jsx, Fragment, jsxs } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { ColorPickerConfig, BackgroundImageConfig, BackgroundVideoConfig, BorderConfig, ShadowConfig } from '@tempi/core-editor';
import { Collapse } from 'antd';
import { PROP_KEY } from '../../constants/propKey.es.js';

const ColStyle = ({
  propKey = PROP_KEY.CommonStyle
}) => {
  return /* @__PURE__ */ jsx(Fragment, {
    children: /* @__PURE__ */ jsxs(Collapse, {
      expandIconPosition: "end",
      bordered: false,
      children: [
        /* @__PURE__ */ jsx(Collapse.Panel, {
          header: t("M\xE0u n\u1EC1n"),
          children: /* @__PURE__ */ jsx(ColorPickerConfig, {
            propKey: PROP_KEY.joinKey(propKey, PROP_KEY.BackgroundColor),
            formItemProps: {
              label: t("M\xE0u n\u1EC1n")
            },
            gradientPicker: true
          })
        }, "background-color"),
        /* @__PURE__ */ jsx(Collapse.Panel, {
          header: t("\u1EA2nh n\u1EC1n"),
          children: /* @__PURE__ */ jsx(BackgroundImageConfig, {
            propKey,
            allowConfingMarqueeBgImage: true
          })
        }, "background-image"),
        /* @__PURE__ */ jsx(Collapse.Panel, {
          header: t("Video n\u1EC1n"),
          children: /* @__PURE__ */ jsx(BackgroundVideoConfig, {
            propKey
          })
        }, "background-video"),
        /* @__PURE__ */ jsxs(Collapse.Panel, {
          header: t("Vi\u1EC1n v\xE0 bo g\xF3c"),
          children: [
            /* @__PURE__ */ jsx(BorderConfig, {
              propKey
            }),
            /* @__PURE__ */ jsx(ShadowConfig, {})
          ]
        }, "border-and-radius")
      ]
    })
  });
};

export { ColStyle };
