// @tempi/core-elements/dist/components/col/ui.es.js
import { jsx, jsxs } from 'react/jsx-runtime';
import styled from '@emotion/styled';
import { getStyle, useAnimation, useIntersectionObserver, BreakpointWidthProvider, BackgroundVideo, BG_LAZY_CLASS_NAME, BG_VISIBLE_LAZY_CLASS_NAME } from '@tempi/core-renderer';
import clsx from 'clsx';
import React__default, { useRef } from 'react';
import { parseFlexPercent } from './utils.es.js';
import { Col } from '../../core/Grid/Col.es.js';
import { MAPPING_VERTICAL_ALIGNMENT_FLEX } from '../../constants/grid.es.js';
import { Link } from '../../core/Link/index.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const ColContent = styled.div(
  ({
    commonStyle,
    verticalAlignment
  }) => `
  ${getStyle(commonStyle, { lazyBgImage: true })}
  display: flex;
  flex-direction: column;
  justify-content: ${MAPPING_VERTICAL_ALIGNMENT_FLEX[verticalAlignment] || MAPPING_VERTICAL_ALIGNMENT_FLEX.top}
`
);
const PbCol = React__default.forwardRef(
  (props, ref) => {
    var _b, _c, _d;
    const _a = props, { commonStyle, verticalAlignment, children, event } = _a, rest = __objRest(_a, ["commonStyle", "verticalAlignment", "children", "event"]);
    const colRef = useRef();
    const { style: animationStyle, animationClassName } = useAnimation(
      commonStyle == null ? void 0 : commonStyle.animation,
      colRef
    );
    const isIntersected = useIntersectionObserver(colRef, {
      freezeOnceVisible: true
    });
    return /* @__PURE__ */ jsx(BreakpointWidthProvider, {
      sWidth: {
        value: parseFlexPercent((_b = rest == null ? void 0 : rest.xs) == null ? void 0 : _b.flex),
        unit: "%"
      },
      mWidth: {
        value: parseFlexPercent((_c = rest == null ? void 0 : rest.md) == null ? void 0 : _c.flex),
        unit: "%"
      },
      children: /* @__PURE__ */ jsxs(Col, __spreadProps(__spreadValues({
        ref
      }, rest), {
        children: [
          (commonStyle == null ? void 0 : commonStyle.backgroundVideo) && /* @__PURE__ */ jsx(BackgroundVideo, __spreadProps(__spreadValues({}, commonStyle.backgroundVideo), {
            background: (_d = commonStyle.background) == null ? void 0 : _d.toString()
          })),
          /* @__PURE__ */ jsx(Link, __spreadProps(__spreadValues({}, event), {
            children: /* @__PURE__ */ jsx(ColContent, {
              ref: colRef,
              commonStyle,
              verticalAlignment,
              style: animationStyle,
              className: clsx(
                {
                  [BG_LAZY_CLASS_NAME]: true,
                  [BG_VISIBLE_LAZY_CLASS_NAME]: isIntersected
                },
                animationClassName
              ),
              children
            })
          }))
        ]
      }))
    });
  }
);
PbCol.displayName = Col.displayName;

export { PbCol };
