// @tempi/core-editor/dist/components/ContextMenu/ContextMenu.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var styled = require('@emotion/styled');
var dndCore = require('@tempi/dnd-core');
var antd = require('antd');
var useEditor = require('../../hooks/useEditor.js');
var BuilderContext = require('../../contexts/BuilderContext.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

const ContextMenu = (props) => {
  const { children } = props;
  const {
    actions,
    duplicateNode,
    getSiblings
  } = useEditor.useEditor();
  const { id, node, parentId } = dndCore.useNode((node2) => ({
    node: node2,
    parentId: node2.data.parent
  }));
  const { setShowSettingDialog, frame } = BuilderContext.useBuilder();
  const onDelete = () => {
    actions.delete(id);
  };
  const onDuplicate = () => {
    const index = getSiblings(id).findIndex((el) => el === id);
    duplicateNode(node, parentId, index + 1);
  };
  const onOpenSetting = () => {
    actions.selectNode(id);
    setShowSettingDialog(true);
  };
  const items = [
    {
      label: /* @__PURE__ */ jsxRuntime.jsxs(MenuLabel, {
        children: [
          /* @__PURE__ */ jsxRuntime.jsx("span", {
            children: i18n.t("C\xE0i \u0111\u1EB7t")
          }),
          /* @__PURE__ */ jsxRuntime.jsx("span", {
            children: "\u2318 E"
          })
        ]
      }),
      key: "1",
      onClick: onOpenSetting
    },
    {
      label: /* @__PURE__ */ jsxRuntime.jsxs(MenuLabel, {
        children: [
          /* @__PURE__ */ jsxRuntime.jsx("span", {
            children: i18n.t("Nh\xE2n b\u1EA3n")
          }),
          /* @__PURE__ */ jsxRuntime.jsx("span", {
            children: "\u2318 D"
          })
        ]
      }),
      key: "4",
      onClick: onDuplicate
    },
    {
      label: /* @__PURE__ */ jsxRuntime.jsxs(MenuLabel, {
        children: [
          /* @__PURE__ */ jsxRuntime.jsx("span", {
            children: i18n.t("Xo\xE1")
          }),
          /* @__PURE__ */ jsxRuntime.jsx("span", {
            children: "\u2318 \u232B"
          })
        ]
      }),
      key: "5",
      onClick: onDelete
    }
  ];
  return /* @__PURE__ */ jsxRuntime.jsx(antd.Dropdown, {
    getPopupContainer: () => frame == null ? void 0 : frame.body,
    menu: { items, style: { width: 160 } },
    trigger: ["contextMenu"],
    children: /* @__PURE__ */ jsxRuntime.jsx("div", {
      children
    })
  });
};
const MenuLabel = styled__default["default"].div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

exports.ContextMenu = ContextMenu;
