// @tempi/core-elements/dist/components/teko-product-list/selector/ProductListEditor/ProductListEditor.es.js
import { withSelector, AdvanceAttributes, getEditorName } from '@tempi/core-editor';
import { QuickSetting } from './QuickSetting.es.js';
import { ProductList } from '../../ui/ProductList/ProductList.es.js';
import { HeadingEditor } from '../../../heading/selector.es.js';
import { ButtonEditorV2 } from '../../../button/v2/selector.es.js';
import { RowEditor } from '../../../section/selector.es.js';
import { RootEditor } from '../../../root/selector.es.js';

const ProductListEditor = withSelector(ProductList, {
  displayName: "Danh s\xE1ch s\u1EA3n ph\u1EA9m",
  tag: ProductList.displayName.toLowerCase(),
  advanceAttributes: AdvanceAttributes,
  quickSetting: QuickSetting,
  rules: {
    canMoveIn: (incomingNodes) => {
      const allowSources = [
        getEditorName({ HeadingEditor }),
        getEditorName({ ButtonEditorV2 })
      ];
      if (incomingNodes.length === 1) {
        return allowSources.includes(incomingNodes[0].data.name);
      }
      return false;
    },
    canDrop: (targetNode) => {
      const ignoreTargets = [
        getEditorName({ RowEditor }),
        getEditorName({ RootEditor })
      ];
      return !ignoreTargets.includes(targetNode.data.name);
    }
  }
});

export { ProductListEditor };
