// @tempi/core-elements/dist/components/teko-product-list/selector/ProductListEditor/ProductListEditor.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var coreEditor = require('@tempi/core-editor');
var QuickSetting = require('./QuickSetting.js');
var ProductList = require('../../ui/ProductList/ProductList.js');
var selector = require('../../../heading/selector.js');
var selector$1 = require('../../../button/v2/selector.js');
var selector$2 = require('../../../section/selector.js');
var selector$3 = require('../../../root/selector.js');

const ProductListEditor = coreEditor.withSelector(ProductList.ProductList, {
  displayName: "Danh s\xE1ch s\u1EA3n ph\u1EA9m",
  tag: ProductList.ProductList.displayName.toLowerCase(),
  advanceAttributes: coreEditor.AdvanceAttributes,
  quickSetting: QuickSetting.QuickSetting,
  rules: {
    canMoveIn: (incomingNodes) => {
      const allowSources = [
        coreEditor.getEditorName({ HeadingEditor: selector.HeadingEditor }),
        coreEditor.getEditorName({ ButtonEditorV2: selector$1.ButtonEditorV2 })
      ];
      if (incomingNodes.length === 1) {
        return allowSources.includes(incomingNodes[0].data.name);
      }
      return false;
    },
    canDrop: (targetNode) => {
      const ignoreTargets = [
        coreEditor.getEditorName({ RowEditor: selector$2.RowEditor }),
        coreEditor.getEditorName({ RootEditor: selector$3.RootEditor })
      ];
      return !ignoreTargets.includes(targetNode.data.name);
    }
  }
});

exports.ProductListEditor = ProductListEditor;
