// @tempi/core-elements/dist/components/teko-product-list/selector/ProductListContentEditor/ProductListContentEditor.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var coreEditor = require('@tempi/core-editor');
var constants = require('../../constants.js');
var QuickSetting = require('./QuickSetting.js');
var Setting = require('./Setting.js');
var ProductListContent = require('../../ui/ProductListContent/ProductListContent.js');

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
const ProductListContentEditor = coreEditor.withSelector(
  (props) => {
    const { editorTerminal, realDevice: device } = coreEditor.useBuilder();
    const isMobile = device === "mobile";
    return /* @__PURE__ */ jsxRuntime.jsx(ProductListContent.ProductListContent, __spreadValues({
      isMobile,
      editorTerminal
    }, props));
  },
  {
    displayName: "Danh s\xE1ch s\u1EA3n ph\u1EA9m",
    rules: {
      canDrag: () => false
    },
    tag: ProductListContent.ProductListContent.displayName.toLowerCase(),
    advanceAttributes: coreEditor.AdvanceAttributes,
    customAttributes: Setting.Setting,
    quickSetting: QuickSetting.QuickSetting,
    props: constants.DEFAULT_PRODUCT_LIST
  }
);

exports.ProductListContentEditor = ProductListContentEditor;
