// @tempi/core-elements/dist/components/teko-product-list/selector/ProductListContentEditor/ProductListContentEditor.es.js
import { jsx } from 'react/jsx-runtime';
import { withSelector, useBuilder, AdvanceAttributes } from '@tempi/core-editor';
import { DEFAULT_PRODUCT_LIST } from '../../constants.es.js';
import { QuickSetting } from './QuickSetting.es.js';
import { Setting } from './Setting.es.js';
import { ProductListContent } from '../../ui/ProductListContent/ProductListContent.es.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
const ProductListContentEditor = withSelector(
  (props) => {
    const { editorTerminal, realDevice: device } = useBuilder();
    const isMobile = device === "mobile";
    return /* @__PURE__ */ jsx(ProductListContent, __spreadValues({
      isMobile,
      editorTerminal
    }, props));
  },
  {
    displayName: "Danh s\xE1ch s\u1EA3n ph\u1EA9m",
    rules: {
      canDrag: () => false
    },
    tag: ProductListContent.displayName.toLowerCase(),
    advanceAttributes: AdvanceAttributes,
    customAttributes: Setting,
    quickSetting: QuickSetting,
    props: DEFAULT_PRODUCT_LIST
  }
);

export { ProductListContentEditor };
