// @tempi/core-elements/dist/components/teko-product-list/selector/ProductListContentEditor/QuickSetting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../../i18n.js');
var icons = require('@ant-design/icons');
var coreEditor = require('@tempi/core-editor');
var antd = require('antd');
var React = require('react');
var SelectParent = require('../../../../quick-configs/SelectParent/SelectParent.js');
var BackgroundImageQuickSetting = require('../../../../quick-configs/BackgroundImageQuickSetting/BackgroundImageQuickSetting.js');

const QuickSetting = () => {
  const { editorTerminal, setEditorTerminal, useTerminalId } = coreEditor.useBuilder();
  const [open, setOpen] = React.useState(false);
  const [text, setText] = React.useState("");
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(SelectParent.SelectParent, {}),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.QuickSettingButton, {
        tooltip: i18n.t("Xem tr\u01B0\u01A1\u0301c danh sa\u0301ch sa\u0309n ph\xE2\u0309m hi\xEA\u0309n thi\u0323 tr\xEAn \u0111i\xEA\u0309m ba\u0301n"),
        children: /* @__PURE__ */ jsxRuntime.jsx(icons.EyeOutlined, {
          onClick: () => {
            setOpen(true);
            setText(editorTerminal ? `${editorTerminal}` : "");
          }
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx(BackgroundImageQuickSetting.BackgroundImageQuickSetting, {}),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.OpenSettingDialog, {}),
      /* @__PURE__ */ jsxRuntime.jsx(antd.Modal, {
        title: i18n.t("Xem tr\u01B0\u1EDBc danh s\xE1ch s\u1EA3n ph\u1EA9m"),
        open,
        onCancel: () => setOpen(false),
        destroyOnClose: true,
        onOk: () => {
          setOpen(false);
          setEditorTerminal(text);
        },
        children: useTerminalId ? /* @__PURE__ */ jsxRuntime.jsx(antd.InputNumber, {
          style: { width: "100%" },
          placeholder: i18n.t("Nh\u1EADp ID \u0111i\u1EC3m b\xE1n"),
          value: text,
          onChange: (val) => setText(val)
        }) : /* @__PURE__ */ jsxRuntime.jsx(antd.Input, {
          placeholder: i18n.t("Nh\u1EADp m\xE3 \u0111i\u1EC3m b\xE1n"),
          value: text,
          onChange: (e) => setText(e.target.value)
        })
      })
    ]
  });
};

exports.QuickSetting = QuickSetting;
