// @tempi/core-elements/dist/components/teko-product-list/selector/ProductListContentEditor/QuickSetting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../../../i18n.es.js';
import { EyeOutlined } from '@ant-design/icons';
import { useBuilder, QuickSettingButton, OpenSettingDialog } from '@tempi/core-editor';
import { Modal, InputNumber, Input } from 'antd';
import { useState } from 'react';
import { SelectParent } from '../../../../quick-configs/SelectParent/SelectParent.es.js';
import { BackgroundImageQuickSetting } from '../../../../quick-configs/BackgroundImageQuickSetting/BackgroundImageQuickSetting.es.js';

const QuickSetting = () => {
  const { editorTerminal, setEditorTerminal, useTerminalId } = useBuilder();
  const [open, setOpen] = useState(false);
  const [text, setText] = useState("");
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(SelectParent, {}),
      /* @__PURE__ */ jsx(QuickSettingButton, {
        tooltip: t("Xem tr\u01B0\u01A1\u0301c danh sa\u0301ch sa\u0309n ph\xE2\u0309m hi\xEA\u0309n thi\u0323 tr\xEAn \u0111i\xEA\u0309m ba\u0301n"),
        children: /* @__PURE__ */ jsx(EyeOutlined, {
          onClick: () => {
            setOpen(true);
            setText(editorTerminal ? `${editorTerminal}` : "");
          }
        })
      }),
      /* @__PURE__ */ jsx(BackgroundImageQuickSetting, {}),
      /* @__PURE__ */ jsx(OpenSettingDialog, {}),
      /* @__PURE__ */ jsx(Modal, {
        title: t("Xem tr\u01B0\u1EDBc danh s\xE1ch s\u1EA3n ph\u1EA9m"),
        open,
        onCancel: () => setOpen(false),
        destroyOnClose: true,
        onOk: () => {
          setOpen(false);
          setEditorTerminal(text);
        },
        children: useTerminalId ? /* @__PURE__ */ jsx(InputNumber, {
          style: { width: "100%" },
          placeholder: t("Nh\u1EADp ID \u0111i\u1EC3m b\xE1n"),
          value: text,
          onChange: (val) => setText(val)
        }) : /* @__PURE__ */ jsx(Input, {
          placeholder: t("Nh\u1EADp m\xE3 \u0111i\u1EC3m b\xE1n"),
          value: text,
          onChange: (e) => setText(e.target.value)
        })
      })
    ]
  });
};

export { QuickSetting };
