// @tempi/core-elements/dist/components/teko-product-list/selector/ProductListContentEditor/Setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../../i18n.js');
var icons = require('@ant-design/icons');
var coreEditor = require('@tempi/core-editor');
var antd = require('antd');
var React = require('react');
var CTAButtonSetting = require('./CTAButtonSetting.js');
var propKey = require('../../../../constants/propKey.js');
var ContentWrapper = require('../../../../core/ContentWrapper/ContentWrapper.js');
var SwiperSettingV2 = require('../../../../configs/Swiper/SwiperSettingV2.js');

const { Panel } = antd.Collapse;
const Setting = () => {
  var _a;
  const [layout] = coreEditor.useProp(propKey.PROP_KEY.Type);
  const [actionButton, setActionButton] = coreEditor.useProp(
    propKey.PROP_KEY.joinKey(propKey.PROP_KEY.ActionButton)
  );
  const [menuItemId, setMenuItemId] = coreEditor.useProp(
    propKey.PROP_KEY.joinKey(propKey.PROP_KEY.MenuItemId)
  );
  const [input, setInput] = React.useState("");
  const [tab, setTab] = React.useState(0);
  React.useEffect(() => {
    setInput(menuItemId);
  }, [menuItemId]);
  React.useEffect(() => {
    if (layout === "carousel" && actionButton === "loadMore") {
      setActionButton(void 0);
    }
  }, [layout]);
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsxs(ContentWrapper.ContentWrapper, {
        children: [
          /* @__PURE__ */ jsxRuntime.jsxs(antd.Radio.Group, {
            style: { display: "flex", marginBottom: 12 },
            value: tab,
            onChange: (e) => setTab(e.target.value),
            children: [
              /* @__PURE__ */ jsxRuntime.jsx(antd.Radio.Button, {
                style: { flex: 1 },
                value: 0,
                children: i18n.t("C\u1EA5u h\xECnh")
              }),
              /* @__PURE__ */ jsxRuntime.jsx(antd.Radio.Button, {
                style: { flex: 1 },
                value: 1,
                children: "CTA"
              })
            ]
          }),
          tab === 0 && /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
            children: [
              menuItemId && /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
                label: "Menu item ID",
                layout: "vertical",
                children: /* @__PURE__ */ jsxRuntime.jsx(antd.Input.Search, {
                  value: input,
                  onChange: (e) => setInput(e.target.value),
                  onSearch: () => setMenuItemId(input),
                  enterButton: /* @__PURE__ */ jsxRuntime.jsx(antd.Button, {
                    disabled: (input || "") === (((_a = menuItemId == null ? void 0 : menuItemId.toString) == null ? void 0 : _a.call(menuItemId)) || ""),
                    type: "primary",
                    icon: /* @__PURE__ */ jsxRuntime.jsx(icons.CheckOutlined, {}),
                    children: i18n.t("\xC1p d\u1EE5ng")
                  })
                })
              }),
              /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputConfig, {
                formItemProps: {
                  layout: "vertical",
                  label: i18n.t("Slug trang danh s\xE1ch s\u1EA3n ph\u1EA9m")
                },
                propKey: propKey.PROP_KEY.Slug,
                inputProps: {
                  style: { width: "100%" },
                  placeholder: "/c/khuyen-mai"
                }
              }),
              /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
                propKey: propKey.PROP_KEY.Limit,
                formItemProps: {
                  label: i18n.t("S\u1ED1 l\u01B0\u1EE3ng s\u1EA3n ph\u1EA9m")
                },
                inputNumberProps: {
                  max: 999
                }
              }),
              /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
                propKey: propKey.PROP_KEY.Type,
                options: [
                  {
                    label: "Grid",
                    value: "grid"
                  },
                  {
                    label: "Carousel",
                    value: "carousel"
                  }
                ],
                formItemProps: {
                  label: i18n.t("B\u1ED1 c\u1EE5c")
                }
              })
            ]
          }),
          tab === 1 && /* @__PURE__ */ jsxRuntime.jsx(CTAButtonSetting.CTAButtonSetting, {})
        ]
      }),
      tab === 0 && /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(antd.Collapse, {
            defaultActiveKey: [],
            expandIconPosition: "end",
            bordered: false,
            children: /* @__PURE__ */ jsxRuntime.jsx(Panel, {
              header: i18n.t("C\xE0i \u0111\u1EB7t c\xE1c ph\u1EA7n t\u1EED"),
              children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
                formItemProps: {
                  label: i18n.t("M\xE0u hi\u1EC3n th\u1ECB gi\xE1")
                },
                propKey: propKey.PROP_KEY.joinKey(
                  propKey.PROP_KEY.ItemSettings,
                  propKey.PROP_KEY.PriceColor
                )
              })
            }, "ITEM")
          }),
          layout === "grid" && /* @__PURE__ */ jsxRuntime.jsxs(antd.Collapse, {
            defaultActiveKey: [],
            expandIconPosition: "end",
            bordered: false,
            children: [
              /* @__PURE__ */ jsxRuntime.jsxs(Panel, {
                header: i18n.t("Hi\u1EC3n th\u1ECB tr\xEAn m\xE1y t\xEDnh"),
                children: [
                  /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
                    propKey: propKey.PROP_KEY.joinKey(
                      propKey.PROP_KEY.ResponsiveSettings,
                      propKey.PROP_KEY.Desktop,
                      propKey.PROP_KEY.ItemsPerRow
                    ),
                    formItemProps: {
                      label: i18n.t("S\u1ED1 ph\u1EA7n t\u1EED 1 h\xE0ng")
                    },
                    inputNumberProps: {
                      max: 10
                    }
                  }),
                  /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
                    propKey: propKey.PROP_KEY.joinKey(
                      propKey.PROP_KEY.ResponsiveSettings,
                      propKey.PROP_KEY.Desktop,
                      propKey.PROP_KEY.Gap
                    ),
                    formItemProps: {
                      label: i18n.t("Kho\u1EA3ng c\xE1ch (px)")
                    }
                  })
                ]
              }, "DESKTOP"),
              /* @__PURE__ */ jsxRuntime.jsxs(Panel, {
                header: i18n.t("Hi\u1EC3n th\u1ECB tr\xEAn \u0111i\u1EC7n tho\u1EA1i"),
                children: [
                  /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
                    propKey: propKey.PROP_KEY.joinKey(
                      propKey.PROP_KEY.ResponsiveSettings,
                      propKey.PROP_KEY.Mobile,
                      propKey.PROP_KEY.ItemsPerRow
                    ),
                    formItemProps: {
                      label: i18n.t("S\u1ED1 ph\u1EA7n t\u1EED 1 h\xE0ng")
                    },
                    inputNumberProps: {
                      max: 10
                    }
                  }),
                  /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
                    propKey: propKey.PROP_KEY.joinKey(
                      propKey.PROP_KEY.ResponsiveSettings,
                      propKey.PROP_KEY.Mobile,
                      propKey.PROP_KEY.Gap
                    ),
                    formItemProps: {
                      label: i18n.t("Kho\u1EA3ng c\xE1ch (px)")
                    }
                  })
                ]
              }, "MOBILE")
            ]
          }),
          layout === "carousel" && /* @__PURE__ */ jsxRuntime.jsxs(antd.Collapse, {
            defaultActiveKey: [],
            expandIconPosition: "end",
            children: [
              /* @__PURE__ */ jsxRuntime.jsx(Panel, {
                header: i18n.t("Hi\u1EC3n th\u1ECB tr\xEAn m\xE1y t\xEDnh"),
                children: /* @__PURE__ */ jsxRuntime.jsx(SwiperSettingV2.SwiperSettingV2, {
                  propKey: propKey.PROP_KEY.joinKey(
                    propKey.PROP_KEY.CarouselSettings,
                    propKey.PROP_KEY.Desktop
                  )
                })
              }, "DESKTOP"),
              /* @__PURE__ */ jsxRuntime.jsx(Panel, {
                header: i18n.t("Hi\u1EC3n th\u1ECB tr\xEAn \u0111i\u1EC7n tho\u1EA1i"),
                children: /* @__PURE__ */ jsxRuntime.jsx(SwiperSettingV2.SwiperSettingV2, {
                  propKey: propKey.PROP_KEY.joinKey(
                    propKey.PROP_KEY.CarouselSettings,
                    propKey.PROP_KEY.Mobile
                  )
                })
              }, "MOBILE")
            ]
          })
        ]
      })
    ]
  });
};

exports.Setting = Setting;
