// @tempi/core-editor/dist/configs/ColorPickerConfig/ColorPickerConfig.es.js
import { jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { useState, useEffect } from 'react';
import { ColorPickerV2 } from '../../components/ColorPickerV2/ColorPickerV2.es.js';
import { usePropSeparateDevice } from '../../hooks/usePropSeparateDevice.es.js';
import { PROP_KEY } from '../../constants/propKey.es.js';
import { FormItem } from '../../components/FormItem/FormItem.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const ColorPickerConfig = ({
  propKey,
  placeholder,
  formItemProps,
  colorPickerProps = {},
  gradientPicker = false,
  separateDevice
}) => {
  var _a, _b;
  const [value, handleChange] = usePropSeparateDevice(propKey, separateDevice);
  const newPropKey = (_a = propKey == null ? void 0 : propKey.replace) == null ? void 0 : _a.call(
    propKey,
    PROP_KEY.BackgroundColor,
    PROP_KEY.Background
  );
  const [valueV2, handleChangeGradientColor] = usePropSeparateDevice(
    newPropKey,
    separateDevice
  );
  const [show, setShow] = useState(false);
  const isChangeBackgroundColorToBackground = ((_b = propKey == null ? void 0 : propKey.includes) == null ? void 0 : _b.call(propKey, PROP_KEY.BackgroundColor)) && (!!valueV2 || gradientPicker);
  useEffect(() => {
    if (isChangeBackgroundColorToBackground && value) {
      handleChange(void 0);
      if (!valueV2)
        handleChangeGradientColor(value);
    }
  }, []);
  return /* @__PURE__ */ jsx(FormItem, __spreadProps(__spreadValues({
    label: t("Ch\u1ECDn m\xE0u")
  }, formItemProps), {
    children: /* @__PURE__ */ jsx(ColorPickerV2, __spreadValues({
      color: isChangeBackgroundColorToBackground ? valueV2 : value,
      onChangeColor: isChangeBackgroundColorToBackground ? handleChangeGradientColor : handleChange,
      showModal: show,
      setShowModal: setShow,
      placeholder: placeholder || t("Ch\u1ECDn m\xE0u"),
      hasGradientColor: gradientPicker
    }, colorPickerProps))
  }));
};

export { ColorPickerConfig };
