// @tempi/core-elements/dist/components/testimonial/v2/constants.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var i18n = require('../../../i18n.js');
var constants = require('../../paragraph/constants.js');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const DEFAULT_TESTIMONIAL_V2 = {
  slider: {
    slidesPerView: 2,
    slidesPerColumn: 1,
    navigation: true,
    navigationStyle: 1,
    pagination: true,
    paginationStyle: 2,
    paginationColor: "rgba(37, 141, 255, 1)",
    paginationPosition: "outside",
    autoplay: false,
    loop: false,
    spaceBetween: 12,
    speed: 3e3,
    pauseOnMouseEnter: true,
    virtual: false
  },
  sliderItem: {
    commonStyle: {
      padding: {
        top: 20,
        right: 20,
        bottom: 20,
        left: 20
      }
    }
  },
  image: {
    src: "https://images.unsplash.com/photo-1543270122-f7a11ad44f3a?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=687&q=80",
    commonStyle: {
      height: 56,
      width: 56,
      border: { borderRadius: 28 },
      objectFit: "cover",
      margin: { top: 10, bottom: 15 }
    }
  },
  content: {
    text: `<p style="text-align:center;"><em><span style="font-size:15px">${i18n.t(
      "T\xF4i r\u1EA5t h\xE0i l\xF2ng v\u1EC1 s\u1EA3n ph\u1EA9m d\u1ECBch v\u1EE5 c\u1EE7a c\xF4ng ty."
    )}<br>${i18n.t(
      "T\xF4i s\u1EBD gi\u1EDBi thi\u1EC7u b\u1EA1n b\xE8 v\xE0 ti\u1EBFp t\u1EE5c \u1EE7ng h\u1ED9 trong th\u1EDDi gian t\u1EDBi."
    )}
    </span></em></p>`,
    commonStyle: {
      margin: { bottom: 10 }
    }
  },
  name: {
    text: `<p style="text-align: center; margin-bottom:unset"><span style="font-size: 18px; text-align: center;"><strong>${i18n.t(
      "Nguy\u1EC5n V\u0103n A"
    )}</strong></span></p>`,
    commonStyle: {}
  },
  title: {
    text: `<p style="text-align: center;"><span style="color: #258DFF;">${i18n.t(
      "Nh\xE2n vi\xEAn b\xE1n h\xE0ng"
    )}</span></p>`,
    commonStyle: {
      margin: { bottom: 10 }
    }
  },
  commonStyle: {
    padding: {
      left: 40,
      right: 40
    }
  },
  paragraph: __spreadProps(__spreadValues({}, constants.DEFAULT_PARAGRAPH), {
    commonStyle: {
      textAlign: "center",
      margin: {
        bottom: 16
      }
    }
  })
};

exports.DEFAULT_TESTIMONIAL_V2 = DEFAULT_TESTIMONIAL_V2;
