// @tempi/core-editor/dist/components/ColorPickerModal/ColorPickerModal.es.js
import { jsxs, jsx, Fragment } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { Button, Modal } from 'antd';
import get from 'lodash.get';
import set from 'lodash.set';
import { useState, useEffect } from 'react';
import { ROOT, getRgbaFromHex } from '@tempi/core-renderer';
import { RgbaStringColorPicker, HexColorInput } from 'react-colorful';
import { RgbaField } from './RgbaField.es.js';
import { inputCss, labelCss } from './style.es.js';
import styled from '@emotion/styled';
import { useFrame } from 'react-frame-component';
import { sampleColors } from './constants.es.js';
import { useEditor } from '../../hooks/useEditor.es.js';
import { PROP_KEY } from '../../constants/propKey.es.js';
import { rgbaStringToObject, rgbaToString } from '../../utils/color.es.js';

const getRGBAColor = (color) => {
  var _a;
  if (!color)
    return null;
  let rgbaColor = color;
  if (/^#.+$/.test(color)) {
    rgbaColor = getRgbaFromHex(color);
  }
  const rgbaObject = (_a = rgbaStringToObject(rgbaColor)) == null ? void 0 : _a.rgba;
  return rgbaToString(rgbaObject);
};
const ColorPickerModal = ({
  color = "rgba(0, 0, 0, 1)",
  visible,
  setVisible,
  onChange
}) => {
  var _a;
  const {
    actions: { history },
    myColors
  } = useEditor((state) => {
    var _a2;
    const rootNode = state.nodes[ROOT];
    return {
      myColors: get((_a2 = rootNode.data) == null ? void 0 : _a2.props, PROP_KEY.MyColors, [])
    };
  });
  const { document: frameDocument } = useFrame();
  const [selectedRgbaColor, setSelectedRgbaColor] = useState();
  const currentColor = selectedRgbaColor || "";
  useEffect(() => {
    if (visible) {
      const rgbaString = getRGBAColor(color);
      setSelectedRgbaColor(rgbaString);
    }
  }, [visible]);
  const renderCirclePicker = (label, colors) => {
    if (!(colors == null ? void 0 : colors.length))
      return null;
    return /* @__PURE__ */ jsxs(Fragment, {
      children: [
        /* @__PURE__ */ jsx(Label, {
          children: label
        }),
        /* @__PURE__ */ jsx(CircleColorPicker, {
          children: colors.map((color2, index) => {
            var _a2;
            const rgbaColor = getRGBAColor(color2);
            const hexColor = (_a2 = rgbaStringToObject(rgbaColor)) == null ? void 0 : _a2.hex;
            return /* @__PURE__ */ jsx(ColorCircle, {
              title: hexColor,
              style: { backgroundColor: color2 },
              className: rgbaColor === selectedRgbaColor ? "active" : "",
              onClick: () => {
                setSelectedRgbaColor(rgbaColor);
              }
            }, index);
          })
        })
      ]
    });
  };
  const onClose = () => setVisible(false);
  const onOk = () => {
    var _a2;
    const hexColor = (_a2 = rgbaStringToObject(selectedRgbaColor)) == null ? void 0 : _a2.hex;
    onChange(hexColor);
    let newMyColors = [...myColors];
    let isExistInMyColors = false;
    for (const color2 of myColors) {
      const rgba = getRGBAColor(color2);
      if (rgba === selectedRgbaColor) {
        isExistInMyColors = true;
        break;
      }
    }
    if (!isExistInMyColors) {
      if (myColors.length === 10) {
        newMyColors = [...myColors.slice(1), hexColor];
      } else
        newMyColors.push(hexColor);
      history.ignore().setProp(ROOT, (prop) => {
        set(prop, PROP_KEY.MyColors, newMyColors);
      });
    }
    onClose();
  };
  return /* @__PURE__ */ jsxs(StyledModal, {
    open: visible,
    title: t("Ch\u1ECDn m\xE0u"),
    onCancel: onClose,
    width: 300,
    bodyStyle: { padding: "16px 8px" },
    zIndex: 9999,
    destroyOnClose: true,
    getContainer: () => (frameDocument == null ? void 0 : frameDocument.body) || (document == null ? void 0 : document.body),
    footer: /* @__PURE__ */ jsxs(Footer, {
      children: [
        /* @__PURE__ */ jsx(ResetButton, {
          type: "link",
          onClick: () => {
            onChange();
            onClose();
          },
          children: t("\u0110\u1EB7t l\u1EA1i")
        }),
        /* @__PURE__ */ jsxs("div", {
          style: { display: "flex" },
          children: [
            /* @__PURE__ */ jsx(Button, {
              onClick: onClose,
              children: t("common.cancel")
            }),
            /* @__PURE__ */ jsx(Button, {
              type: "primary",
              onClick: onOk,
              children: "OK"
            })
          ]
        })
      ]
    }),
    children: [
      /* @__PURE__ */ jsx(StyledColorPicker, {
        color: currentColor,
        onChange: setSelectedRgbaColor
      }),
      /* @__PURE__ */ jsxs("div", {
        style: { display: "flex", columnGap: 16 },
        children: [
          /* @__PURE__ */ jsxs("div", {
            style: { flex: 1 },
            children: [
              /* @__PURE__ */ jsx(StyledHexColorInput, {
                alpha: true,
                prefixed: true,
                color: (_a = rgbaStringToObject(selectedRgbaColor)) == null ? void 0 : _a.hex,
                onChange: (color2) => {
                  setSelectedRgbaColor(getRgbaFromHex(color2));
                }
              }),
              /* @__PURE__ */ jsx(InputLabel, {
                children: "hex"
              })
            ]
          }),
          /* @__PURE__ */ jsx(StyledRgbaField, {
            color: currentColor,
            onChange: setSelectedRgbaColor
          })
        ]
      }),
      renderCirclePicker(t("M\xE0u c\u1EE7a t\xF4i"), myColors),
      renderCirclePicker(t("M\xE0u m\u1EABu"), sampleColors)
    ]
  });
};
const Label = styled.div`
  font-size: 12px;
  font-weight: 500;
  margin: 8px 0 4px;
`;
const StyledModal = styled(Modal)`
  &.ant-modal {
    top: 80px;
  }

  .ant-modal {
    &-header,
    &-footer {
      padding: 8px;
    }

    &-title {
      font-size: 14px;
    }

    &-close-x {
      width: 32px;
      height: 32px;
      line-height: 32px;
      font-size: 14px;
    }
  }
`;
const StyledColorPicker = styled(RgbaStringColorPicker)`
  width: 100% !important;
  margin-bottom: 16px;

  .react-colorful {
    &__pointer {
      width: 16px;
      height: 16px;
    }

    &__alpha,
    &__hue {
      height: 12px;
      margin-top: 12px;
      border-radius: 8px;
    }
  }
`;
const StyledHexColorInput = styled(HexColorInput)(inputCss);
const InputLabel = styled.label(labelCss);
const CircleColorPicker = styled.div`
  display: flex;
  flex-wrap: wrap;
  row-gap: 8px;
  column-gap: 8px;
`;
const ColorCircle = styled.div`
  width: 20px;
  height: 20px;
  border-radius: 50%;
  transform: scale(1);
  transition: all 100ms ease 0s;
  cursor: pointer;
  border: 1px solid #d9d9d9;

  &.active {
    border: 2px solid #000;
    transform: scale(1.1);
  }

  &:hover {
    transform: scale(1.2);
  }
`;
const Footer = styled.div`
  display: flex;
  justify-content: space-between;
`;
const ResetButton = styled(Button)`
  padding-left: 0;
  padding-right: 0;
`;
const StyledRgbaField = styled(RgbaField)`
  flex: 2;
`;

export { ColorPickerModal };
