// @tempi/core-elements/dist/components/lead-form/constants.es.js
import { t } from '../../i18n.es.js';
import { PROP_KEY } from '../../constants/propKey.es.js';

const CDPBff = {
  dev: "https://cdp-bff.dev.tekoapis.net",
  stag: "https://cdp-bff.stag.tekoapis.net",
  production: "https://cdp-bff.tekoapis.com"
};
const DEFAULT_LEAD_FORM = {
  formStyle: {
    flexDirection: "column",
    gap: 8
  },
  inputStyle: {
    borderRadius: 4
  },
  showLabel: true,
  hiddenFields: [],
  response: {
    custom: []
  },
  submits: [],
  inputs: [
    {
      type: "text",
      label: t("H\u1ECD v\xE0 t\xEAn"),
      name: "fullname",
      commonStyle: { height: 32, width: "100%" },
      disabledFormFieldSetting: [PROP_KEY.Name, PROP_KEY.Type]
    },
    {
      type: "tel",
      label: t("S\u1ED1 \u0111i\u1EC7n tho\u1EA1i"),
      name: "phone",
      commonStyle: { height: 32, width: "100%" },
      required: true,
      disabledFormFieldSetting: [
        PROP_KEY.Name,
        PROP_KEY.Type,
        PROP_KEY.Required
      ]
    },
    {
      type: "email",
      label: "Email",
      name: "email",
      commonStyle: { height: 32, width: "100%" },
      required: true,
      disabledFormFieldSetting: [
        PROP_KEY.Name,
        PROP_KEY.Type,
        PROP_KEY.Required
      ]
    },
    {
      type: "text",
      label: t("M\xE3 kh\xE1ch h\xE0ng"),
      name: "externalId",
      commonStyle: { height: 32, width: "100%" },
      required: true,
      disabledFormFieldSetting: [
        PROP_KEY.Name,
        PROP_KEY.Type,
        PROP_KEY.Required
      ]
    },
    {
      type: "radio-group",
      label: t("Gi\u1EDBi t\xEDnh"),
      name: "gender",
      commonStyle: { flexDirection: "column", width: "100%" },
      disabledFormFieldSetting: [PROP_KEY.Name, PROP_KEY.Type],
      options: [
        {
          value: "M",
          label: "Nam"
        },
        {
          value: "F",
          label: t("N\u1EEF")
        },
        {
          value: "U",
          label: t("Kh\xE1c")
        }
      ]
    },
    {
      type: "date",
      label: t("Ng\xE0y sinh"),
      name: "birthday",
      commonStyle: { height: 32, width: "100%" },
      disabledFormFieldSetting: [PROP_KEY.Name, PROP_KEY.Type]
    },
    {
      type: "location",
      label: t("\u0110\u1ECBa ch\u1EC9"),
      name: "address",
      commonStyle: { height: 32, width: "100%" },
      disabledFormFieldSetting: [PROP_KEY.Name, PROP_KEY.Type]
    },
    {
      type: "text",
      label: t("S\u1ED1 nh\xE0, \u0111\u01B0\u1EDDng ph\u1ED1"),
      name: "addressStreet",
      commonStyle: { height: 32, width: "100%" },
      disabledFormFieldSetting: [PROP_KEY.Name, PROP_KEY.Type]
    },
    {
      type: "checkbox-group",
      label: t("S\u1EDF th\xEDch"),
      name: "interest",
      commonStyle: { flexDirection: "column", width: "100%" },
      disabledFormFieldSetting: [PROP_KEY.Name, PROP_KEY.Type],
      options: [
        {
          value: "option1",
          label: "Option 1"
        },
        {
          value: "option2",
          label: "Option 2"
        }
      ]
    },
    {
      type: "text",
      label: "Tag 1",
      name: "tag1",
      commonStyle: { height: 32, width: "100%" },
      disabledFormFieldSetting: [PROP_KEY.Name],
      typeSelectable: ["text", "checkbox-group", "radio-group"]
    },
    {
      type: "text",
      label: "Tag 2",
      name: "tag2",
      commonStyle: { height: 32, width: "100%" },
      disabledFormFieldSetting: [PROP_KEY.Name],
      typeSelectable: ["text", "checkbox-group", "radio-group"]
    },
    {
      type: "text",
      label: "Tag 3",
      name: "tag3",
      commonStyle: { height: 32, width: "100%" },
      disabledFormFieldSetting: [PROP_KEY.Name],
      typeSelectable: ["text", "checkbox-group", "radio-group"]
    }
  ]
};

export { CDPBff, DEFAULT_LEAD_FORM };
