// @tempi/core-editor/dist/components/ColorPickerV2/ColorPickerV2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var styled = require('@emotion/styled');
var icons = require('@ant-design/icons');
var coreRenderer = require('@tempi/core-renderer');
var ColorPickerModalV2 = require('../ColorPickerModal/ColorPickerModalV2.js');
var color = require('../../utils/color.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

const ColorPickerV2 = ({
  color: color$1,
  showModal,
  setShowModal,
  onChangeColor,
  placeholder,
  hasGradientColor
}) => {
  const colorObj = coreRenderer.parseColor(color$1);
  const hexColor = (colorObj == null ? void 0 : colorObj.colorType) === coreRenderer.ColorType.solid ? color.getHexColor(color$1) : void 0;
  const togglePickerModal = () => {
    setShowModal(!showModal);
  };
  const onDeleteColor = (e) => {
    onChangeColor(void 0);
    e.stopPropagation();
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(Wrapper, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(ColorDivWrapper, {
        onClick: togglePickerModal,
        children: color$1 ? /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
          children: [
            /* @__PURE__ */ jsxRuntime.jsxs(ColorDisplayWrapper, {
              children: [
                /* @__PURE__ */ jsxRuntime.jsx(ColorDiv, {
                  color: color$1
                }),
                /* @__PURE__ */ jsxRuntime.jsx(ColorTextDiv, {
                  children: hexColor || ""
                })
              ]
            }),
            /* @__PURE__ */ jsxRuntime.jsx(DeleteColorWrapper, {
              onClick: onDeleteColor,
              children: /* @__PURE__ */ jsxRuntime.jsx(icons.CloseOutlined, {})
            })
          ]
        }) : /* @__PURE__ */ jsxRuntime.jsx(Placeholder, {
          children: placeholder || i18n.t("Ch\u1ECDn m\xE0u")
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx(ColorPickerModalV2.ColorPickerModalV2, {
        visible: showModal,
        setVisible: setShowModal,
        color: color$1,
        onChange: onChangeColor,
        hasGradientColor
      })
    ]
  });
};
const Wrapper = styled__default["default"].div`
  position: relative;
`;
const ColorDiv = styled__default["default"].div(
  ({ color }) => `
    width: 24px;
    height: 24px;
    border-radius: 4px;
    margin-left: 4px;
    background: ${color};
    margin-right: 8px;
`
);
const ColorDivWrapper = styled__default["default"].div`
  position: relative;
  width: 140px;
  height: 32px;
  display: flex;
  align-items: center;
  border: 1px solid ${(props) => props.theme.colorBorder};
  border-radius: 4px;
  &:hover {
    cursor: pointer;
  }
`;
const ColorTextDiv = styled__default["default"].div`
  font-size: 13px;
  line-height: 20px;
`;
const Placeholder = styled__default["default"].div`
  font-size: 13px;
  line-height: 18px;
  color: var(--placeholder-color);
  padding: 12px;
`;
const ColorDisplayWrapper = styled__default["default"].div`
  display: flex;
  align-items: center;
  width: 100%;
`;
const DeleteColorWrapper = styled__default["default"].div`
  margin: auto 4px;
  right: 0;
  position: absolute;
`;

exports.ColorPickerV2 = ColorPickerV2;
