// @tempi/core-editor/dist/components/ColorPickerV2/ColorPickerV2.es.js
import { jsxs, jsx, Fragment } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import styled from '@emotion/styled';
import { CloseOutlined } from '@ant-design/icons';
import { parseColor, ColorType } from '@tempi/core-renderer';
import { ColorPickerModalV2 } from '../ColorPickerModal/ColorPickerModalV2.es.js';
import { getHexColor } from '../../utils/color.es.js';

const ColorPickerV2 = ({
  color,
  showModal,
  setShowModal,
  onChangeColor,
  placeholder,
  hasGradientColor
}) => {
  const colorObj = parseColor(color);
  const hexColor = (colorObj == null ? void 0 : colorObj.colorType) === ColorType.solid ? getHexColor(color) : void 0;
  const togglePickerModal = () => {
    setShowModal(!showModal);
  };
  const onDeleteColor = (e) => {
    onChangeColor(void 0);
    e.stopPropagation();
  };
  return /* @__PURE__ */ jsxs(Wrapper, {
    children: [
      /* @__PURE__ */ jsx(ColorDivWrapper, {
        onClick: togglePickerModal,
        children: color ? /* @__PURE__ */ jsxs(Fragment, {
          children: [
            /* @__PURE__ */ jsxs(ColorDisplayWrapper, {
              children: [
                /* @__PURE__ */ jsx(ColorDiv, {
                  color
                }),
                /* @__PURE__ */ jsx(ColorTextDiv, {
                  children: hexColor || ""
                })
              ]
            }),
            /* @__PURE__ */ jsx(DeleteColorWrapper, {
              onClick: onDeleteColor,
              children: /* @__PURE__ */ jsx(CloseOutlined, {})
            })
          ]
        }) : /* @__PURE__ */ jsx(Placeholder, {
          children: placeholder || t("Ch\u1ECDn m\xE0u")
        })
      }),
      /* @__PURE__ */ jsx(ColorPickerModalV2, {
        visible: showModal,
        setVisible: setShowModal,
        color,
        onChange: onChangeColor,
        hasGradientColor
      })
    ]
  });
};
const Wrapper = styled.div`
  position: relative;
`;
const ColorDiv = styled.div(
  ({ color }) => `
    width: 24px;
    height: 24px;
    border-radius: 4px;
    margin-left: 4px;
    background: ${color};
    margin-right: 8px;
`
);
const ColorDivWrapper = styled.div`
  position: relative;
  width: 140px;
  height: 32px;
  display: flex;
  align-items: center;
  border: 1px solid ${(props) => props.theme.colorBorder};
  border-radius: 4px;
  &:hover {
    cursor: pointer;
  }
`;
const ColorTextDiv = styled.div`
  font-size: 13px;
  line-height: 20px;
`;
const Placeholder = styled.div`
  font-size: 13px;
  line-height: 18px;
  color: var(--placeholder-color);
  padding: 12px;
`;
const ColorDisplayWrapper = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
`;
const DeleteColorWrapper = styled.div`
  margin: auto 4px;
  right: 0;
  position: absolute;
`;

export { ColorPickerV2 };
