// @tempi/core-elements/dist/components/global-block/header-v2/ui/GlobalHeader.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var coreRenderer = require('@tempi/core-renderer');
var clsx = require('clsx');
var React = require('react');
var hooks = require('../../../header/hooks.js');
var utils = require('./utils.js');
var type = require('../../../header/type.js');
var Header = require('./components/Header.js');
var Content = require('./components/Content.js');
var Logo = require('./components/Logo.js');
var ToggleButton = require('./components/ToggleButton.js');
var NavigationBar = require('./components/NavigationBar.js');
var Menu = require('./components/Menu.js');
var Item = require('./components/Item.js');
var DropdownItem = require('./components/DropdownItem.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var clsx__default = /*#__PURE__*/_interopDefaultLegacy(clsx);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const GlobalHeader = (props) => {
  const [mobileMenuOpen, setMobileMenuOpen] = React.useState(false);
  const {
    commonStyle,
    logoConfig,
    menuConfig,
    menu = [],
    isEditingMode,
    stickyTop
  } = props || {};
  const grid = utils.calculateGrid(props);
  const menuOpenClassName = clsx__default["default"]({ open: mobileMenuOpen });
  const tree = utils.flattenToTreeMax2Level(
    (menu || []).filter((el) => !el.isHidden)
  );
  const { headerRef, headerHeight } = hooks.useHeaderPosition(
    stickyTop ? type.PositionHeader.STICKY : type.PositionHeader.NORMAL
  );
  React.useEffect(() => {
    const handleClickOutside = (event) => {
      var _a, _b;
      if (!((_b = (_a = headerRef == null ? void 0 : headerRef.current) == null ? void 0 : _a.contains) == null ? void 0 : _b.call(_a, event.target))) {
        setMobileMenuOpen(false);
      }
    };
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, [headerRef]);
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(Header.Header, {
        commonStyle,
        ref: headerRef,
        grid,
        headerHeight,
        children: /* @__PURE__ */ jsxRuntime.jsxs(Content.Content, {
          grid,
          commonStyle,
          children: [
            (logoConfig == null ? void 0 : logoConfig.show) && /* @__PURE__ */ jsxRuntime.jsx(Logo.Logo, __spreadProps(__spreadValues({}, !isEditingMode ? { href: "/" } : {}), {
              grid,
              children: /* @__PURE__ */ jsxRuntime.jsx(Logo.LogoImg, {
                alt: logoConfig == null ? void 0 : logoConfig.alt,
                src: coreRenderer.getImageUrlFromGoogleUserContent(
                  logoConfig == null ? void 0 : logoConfig.src,
                  logoConfig == null ? void 0 : logoConfig.googleImageParams
                ),
                grid,
                loading: "lazy"
              })
            })),
            /* @__PURE__ */ jsxRuntime.jsx(ToggleButton.ToggleButton, {
              className: menuOpenClassName,
              grid,
              onClick: () => setMobileMenuOpen(!mobileMenuOpen)
            }),
            /* @__PURE__ */ jsxRuntime.jsx(NavigationBar.Navbar, {
              grid,
              children: /* @__PURE__ */ jsxRuntime.jsx(Menu.Menu, {
                className: menuOpenClassName,
                grid,
                children: tree.map((item) => {
                  var _a;
                  return /* @__PURE__ */ jsxRuntime.jsx(Item.Item, {
                    children: ((_a = item == null ? void 0 : item.children) == null ? void 0 : _a.length) > 0 ? /* @__PURE__ */ jsxRuntime.jsx(DropdownItem.DropdownItem, {
                      item,
                      grid,
                      alignMenu: menuConfig == null ? void 0 : menuConfig.align,
                      isEditingMode,
                      backgroundColor: commonStyle.backgroundColor || "white"
                    }) : /* @__PURE__ */ jsxRuntime.jsx(Item.ItemLabel, {
                      children: /* @__PURE__ */ jsxRuntime.jsx(Item.ItemContentWrapper, {
                        alignMenu: menuConfig == null ? void 0 : menuConfig.align,
                        children: /* @__PURE__ */ jsxRuntime.jsxs(Item.ItemContent, __spreadProps(__spreadValues(__spreadProps(__spreadValues({
                          alignMenu: menuConfig == null ? void 0 : menuConfig.align
                        }, !isEditingMode ? {
                          href: utils.getUrlOfMenuItem(item)
                        } : {}), {
                          grid,
                          target: utils.getTargetUrl(item.target)
                        }), (item == null ? void 0 : item.linkType) === "LINK" && item.nofollow ? { rel: "nofollow" } : {}), {
                          children: [
                            (item == null ? void 0 : item.iconUrl) && /* @__PURE__ */ jsxRuntime.jsx(Item.ItemIcon, {
                              src: item.iconUrl,
                              loading: "lazy"
                            }),
                            item == null ? void 0 : item.name
                          ]
                        }))
                      })
                    })
                  }, item.id);
                })
              })
            })
          ]
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx("div", {
        className: "placeholder"
      })
    ]
  });
};

exports.GlobalHeader = GlobalHeader;
