// @tempi/core-elements/dist/renderer.es.js
import { jsx } from 'react/jsx-runtime';
import { Root } from './components/root/ui.es.js';
import { PbDivider } from './components/divider/ui.es.js';
import { Spacer } from './components/spacer/ui.es.js';
import { Youtube } from './components/youtube/ui.es.js';
import { Countdown } from './components/countdown/ui.es.js';
import { Map } from './components/map/ui.es.js';
import { Gallery } from './components/gallery/ui.es.js';
import { HtmlCode } from './components/html-code/ui.es.js';
import { PageFlip } from './components/page-flip/ui.es.js';
import { NumberCounter } from './components/number-counter/ui.es.js';
import { SocialIcon } from './components/social-icon/ui.es.js';
import { StoreList } from './components/store-list/ui.es.js';
import { Slider } from './components/slider/v1/ui.es.js';
import { Table } from './components/table/ui.es.js';
import { LuckyDrawV2 } from './components/lucky-draw/v2/ui.es.js';
import { TestimonialCarousel } from './components/testimonial-carousel/ui.es.js';
import { WebsiteFooter } from './components/websitefooter/ui.es.js';
import { GlobalBlock } from './components/global-block/ui.es.js';
import { WebsiteHeader } from './components/website-header/ui.es.js';
import { PopupRoot } from './components/popuproot/ui.es.js';
import { ContactBubble } from './components/contact-bubble/ui.es.js';
import { Page404 } from './components/404/ui.es.js';
import { TempiProductList } from './components/product-list/ui.es.js';
import { QRCode } from './components/qr-code/ui.es.js';
import { ExamScore } from './components/exam-score/ui.es.js';
import { Video } from './components/video/ui.es.js';
import { ArticleList } from './components/article-list/ui.es.js';
import { ArticleHeader } from './components/article-header/ui.es.js';
import { SampleComponent } from './components/sample-component/ui.es.js';
import { LeadForm } from './components/lead-form/ui.es.js';
import { InlineBanner } from './components/inline-banner/ui.es.js';
import { PbIcon } from './components/icon/ui.es.js';
import { PbSection } from './components/section/ui.es.js';
import { PbContainer } from './components/container/ui.es.js';
import { PbRow } from './components/row/ui.es.js';
import { PbCol } from './components/col/ui.es.js';
import { ButtonV2 } from './components/button/v2/ui.es.js';
import { Paragraph } from './components/paragraph/ui.es.js';
import { Heading } from './components/heading/ui.es.js';
import { Image } from './components/image/ui.es.js';
import { Accordion } from './components/accordion/ui/Accordion.es.js';
import { AccordionContent } from './components/accordion/ui/AccordionContent.es.js';
import { AccordionItem } from './components/accordion/ui/AccordionItem.es.js';
import { ListItemV2 } from './components/list/v2/ui/ListItemV2.es.js';
import { ListItemContentV2 } from './components/list/v2/ui/ListItemContentV2.es.js';
import { ListItemIconV2 } from './components/list/v2/ui/ListItemIconV2.es.js';
import { FormV4 } from './components/form/v4/ui.es.js';
import { PopupV1 } from './components/popup/v1/ui.es.js';
import { Header } from './components/header/ui/Header/Header.es.js';
import { Menu } from './components/header/ui/Menu/Menu.es.js';
import { MenuItem } from './components/header/ui/MenuItem/MenuItem.es.js';
import { Logo } from './components/header/ui/Logo/Logo.es.js';
import { MenuCol } from './components/header/ui/MenuCol/MenuCol.es.js';
import { Tab } from './components/tab/ui/Tab/Tab.es.js';
import { TabItem } from './components/tab/ui/TabItem/TabItem.es.js';
import { TestimonialV2 } from './components/testimonial/v2/ui/TestimonialV2/TestimonialV2.es.js';
import { TestimonialV2Item } from './components/testimonial/v2/ui/TestimonialItemV2/TestimonialItemV2.es.js';
import { ContentSlider } from './components/content-slider/ui/ContentSlider/ContentSlider.es.js';
import { ContentSliderItem } from './components/content-slider/ui/ContentSliderItem/ContentSliderItem.es.js';
import { ProductList } from './components/teko-product-list/ui/ProductList/ProductList.es.js';
import { ProductListContent } from './components/teko-product-list/ui/ProductListContent/ProductListContent.es.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
const SubscriptionFormV2 = (props) => {
  return /* @__PURE__ */ jsx(FormV4, __spreadValues({}, props));
};
SubscriptionFormV2.displayName = "SubscriptionFormV2";
const components = {
  Root,
  PbIcon,
  Section: PbSection,
  Container: PbContainer,
  Row: PbRow,
  Col: PbCol,
  ButtonV2,
  Paragraph,
  Heading,
  Divider: PbDivider,
  Spacer,
  Youtube,
  Countdown,
  Map,
  Image,
  Gallery,
  HtmlCode,
  PageFlip,
  NumberCounter,
  SocialIcon,
  StoreList,
  Slider,
  Table,
  Accordion,
  AccordionContent,
  AccordionItem,
  ListItemV2,
  ListItemContentV2,
  ListItemIconV2,
  FormV4,
  SubscriptionFormV2,
  PopupV1,
  Header,
  Menu,
  MenuItem,
  Logo,
  MenuCol,
  LuckyDrawV2,
  Tab,
  TabItem,
  TestimonialCarousel,
  TestimonialV2,
  TestimonialV2Item,
  ContentSlider,
  ContentSliderItem,
  ProductList,
  ProductListContent,
  WebsiteFooter,
  GlobalBlock,
  WebsiteHeader,
  PopupRoot,
  ContactBubble,
  Page404,
  TempiProductList,
  QRCode,
  ExamScore,
  Video,
  ArticleList,
  ArticleHeader,
  SampleComponent,
  LeadForm,
  InlineBanner
};

export { components };
