// @tempi/core-elements/dist/selectors.es.js
import { IconEditor } from './components/icon/selector.es.js';
export { IconEditor } from './components/icon/selector.es.js';
import { ButtonEditor } from './components/button/v1/selector.es.js';
export { ButtonEditor } from './components/button/v1/selector.es.js';
import { DividerEditor } from './components/divider/selector.es.js';
export { DividerEditor } from './components/divider/selector.es.js';
import { SpacerEditor } from './components/spacer/selector.es.js';
export { SpacerEditor } from './components/spacer/selector.es.js';
import { YoutubeEditor } from './components/youtube/selector.es.js';
export { YoutubeEditor } from './components/youtube/selector.es.js';
import { CountdownEditor } from './components/countdown/selector.es.js';
export { CountdownEditor } from './components/countdown/selector.es.js';
import { MapEditor } from './components/map/selector.es.js';
export { MapEditor } from './components/map/selector.es.js';
import { GalleryEditor } from './components/gallery/selector.es.js';
export { GalleryEditor } from './components/gallery/selector.es.js';
import { HtmlCodeEditor } from './components/html-code/selector.es.js';
export { HtmlCodeEditor } from './components/html-code/selector.es.js';
import { PageFlipEditor } from './components/page-flip/selector.es.js';
export { PageFlipEditor } from './components/page-flip/selector.es.js';
import { NumberCounterEditor } from './components/number-counter/selector.es.js';
export { NumberCounterEditor } from './components/number-counter/selector.es.js';
import { SocialIconEditor } from './components/social-icon/selector.es.js';
export { SocialIconEditor } from './components/social-icon/selector.es.js';
import { StoreListEditor } from './components/store-list/selector.es.js';
export { StoreListEditor } from './components/store-list/selector.es.js';
import { SliderEditor } from './components/slider/v1/selector.es.js';
export { SliderEditor } from './components/slider/v1/selector.es.js';
import { SliderEditorV2 } from './components/slider/v2/selector.es.js';
export { SliderEditorV2 } from './components/slider/v2/selector.es.js';
import { TableEditor } from './components/table/selector.es.js';
export { TableEditor } from './components/table/selector.es.js';
import { ListEditor } from './components/list/v1/selector/ListEditor.es.js';
export { ListEditor } from './components/list/v1/selector/ListEditor.es.js';
import { ListItemEditor } from './components/list/v1/selector/ListItemEditor.es.js';
export { ListItemEditor } from './components/list/v1/selector/ListItemEditor.es.js';
import { ListItemContentEditor } from './components/list/v1/selector/ListItemContentEditor.es.js';
export { ListItemContentEditor } from './components/list/v1/selector/ListItemContentEditor.es.js';
import { ListItemIconEditor } from './components/list/v1/selector/ListItemIconEditor.es.js';
export { ListItemIconEditor } from './components/list/v1/selector/ListItemIconEditor.es.js';
import { ListItemEditorV2 } from './components/list/v2/selector/ListItemV2/ListItemEditorV2.es.js';
export { ListItemEditorV2 } from './components/list/v2/selector/ListItemV2/ListItemEditorV2.es.js';
import { ListItemIconEditorV2 } from './components/list/v2/selector/ListItemIconV2/ListItemIconEditorV2.es.js';
export { ListItemIconEditorV2 } from './components/list/v2/selector/ListItemIconV2/ListItemIconEditorV2.es.js';
import { ListItemContentEditorV2 } from './components/list/v2/selector/ListItemContentV2/ListItemContentEditorV2.es.js';
export { ListItemContentEditorV2 } from './components/list/v2/selector/ListItemContentV2/ListItemContentEditorV2.es.js';
import { FormV4Editor } from './components/form/v4/selector.es.js';
export { FormV4Editor } from './components/form/v4/selector.es.js';
import { ButtonSubmitV3Editor } from './components/button-submit/v3/selector.es.js';
export { ButtonSubmitV3Editor } from './components/button-submit/v3/selector.es.js';
import { FormEditor } from './components/form/v1/selector.es.js';
export { FormEditor } from './components/form/v1/selector.es.js';
import { FormV2Editor } from './components/form/v2/selector.es.js';
export { FormV2Editor } from './components/form/v2/selector.es.js';
import { FormV3Editor } from './components/form/v3/selector.es.js';
export { FormV3Editor } from './components/form/v3/selector.es.js';
import { ButtonSubmitEditor } from './components/button-submit/v1/selector.es.js';
export { ButtonSubmitEditor } from './components/button-submit/v1/selector.es.js';
import { SubscriptionForm } from './components/subscription-form/v1/selector.es.js';
export { SubscriptionForm } from './components/subscription-form/v1/selector.es.js';
import { SubscriptionFormV2 } from './components/subscription-form/v2/selector.es.js';
export { SubscriptionFormV2 } from './components/subscription-form/v2/selector.es.js';
import { HeaderEditor } from './components/header/selector/Header/HeaderEditor.es.js';
export { HeaderEditor } from './components/header/selector/Header/HeaderEditor.es.js';
import { MenuColEditor } from './components/header/selector/MenuCol/MenuColEditor.es.js';
export { MenuColEditor } from './components/header/selector/MenuCol/MenuColEditor.es.js';
import { LogoEditor } from './components/header/selector/Logo/LogoEditor.es.js';
export { LogoEditor } from './components/header/selector/Logo/LogoEditor.es.js';
import { MenuEditor } from './components/header/selector/Menu/MenuEditor.es.js';
export { MenuEditor } from './components/header/selector/Menu/MenuEditor.es.js';
import { LuckyDrawEditor } from './components/lucky-draw/v1/selector.es.js';
export { LuckyDrawEditor } from './components/lucky-draw/v1/selector.es.js';
import { LuckyDrawV2Editor } from './components/lucky-draw/v2/selector.es.js';
export { LuckyDrawV2Editor } from './components/lucky-draw/v2/selector.es.js';
import { TabEditor } from './components/tab/selector/TabEditor/TabEditor.es.js';
export { TabEditor } from './components/tab/selector/TabEditor/TabEditor.es.js';
import { TestimonialCarouselEditor } from './components/testimonial-carousel/selector.es.js';
export { TestimonialCarouselEditor } from './components/testimonial-carousel/selector.es.js';
import { TestimonialV2Editor } from './components/testimonial/v2/selector/TestimonialV2Editor/TestimonialV2Editor.es.js';
export { TestimonialV2Editor } from './components/testimonial/v2/selector/TestimonialV2Editor/TestimonialV2Editor.es.js';
import { ContentSliderEditor } from './components/content-slider/selector/ContentSliderEditor/ContentSliderEditor.es.js';
export { ContentSliderEditor } from './components/content-slider/selector/ContentSliderEditor/ContentSliderEditor.es.js';
import { ContentSliderItemEditor } from './components/content-slider/selector/ContentSliderItemEditor/ContentSliderItemEditor.es.js';
export { ContentSliderItemEditor } from './components/content-slider/selector/ContentSliderItemEditor/ContentSliderItemEditor.es.js';
import { ContentSliderHeadingEditor } from './components/content-slider/selector/ContentSliderHeadingEditor/ContentSliderHeadingEditor.es.js';
export { ContentSliderHeadingEditor } from './components/content-slider/selector/ContentSliderHeadingEditor/ContentSliderHeadingEditor.es.js';
import { ProductListEditor } from './components/teko-product-list/selector/ProductListEditor/ProductListEditor.es.js';
import { ProductListContentEditor } from './components/teko-product-list/selector/ProductListContentEditor/ProductListContentEditor.es.js';
import { WebsiteHeaderEditor } from './components/website-header/selector.es.js';
import { ContactBubbleEditor } from './components/contact-bubble/selector.es.js';
import { TempiProductListEditor } from './components/product-list/selector.es.js';
import { QRCodeEditor } from './components/qr-code/selector.es.js';
import { ExamScoreEditor } from './components/exam-score/selector.es.js';
export { ExamScoreEditor } from './components/exam-score/selector.es.js';
import { VideoEditor } from './components/video/selector.es.js';
import { ArticleListEditor } from './components/article-list/selector.es.js';
export { ArticleListEditor } from './components/article-list/selector.es.js';
import { ArticleHeaderEditor } from './components/article-header/selector.es.js';
export { ArticleHeaderEditor } from './components/article-header/selector.es.js';
import { SampleComponentEditor } from './components/sample-component/selector.es.js';
export { SampleComponentEditor } from './components/sample-component/selector.es.js';
import { LeadFormEditor } from './components/lead-form/selector.es.js';
import { InlineBannerEditor } from './components/inline-banner/selector.es.js';
import { RootEditor } from './components/root/selector.es.js';
export { RootEditor } from './components/root/selector.es.js';
import { ColEditor } from './components/col/selector.es.js';
export { ColEditor } from './components/col/selector.es.js';
import { RowEditor } from './components/section/selector.es.js';
export { RowEditor } from './components/section/selector.es.js';
import { ButtonEditorV2 } from './components/button/v2/selector.es.js';
export { ButtonEditorV2 } from './components/button/v2/selector.es.js';
import { ParagraphEditor } from './components/paragraph/selector.es.js';
export { ParagraphEditor } from './components/paragraph/selector.es.js';
import { HeadingEditor } from './components/heading/selector.es.js';
export { HeadingEditor } from './components/heading/selector.es.js';
import { ImageEditor } from './components/image/selector.es.js';
export { ImageEditor } from './components/image/selector.es.js';
import { AccordionEditor } from './components/accordion/selector/Accordion/AccordionEditor.es.js';
export { AccordionEditor } from './components/accordion/selector/Accordion/AccordionEditor.es.js';
import { AccordionContentEditor } from './components/accordion/selector/AccordionContent/AccordionContentEditor.es.js';
export { AccordionContentEditor } from './components/accordion/selector/AccordionContent/AccordionContentEditor.es.js';
import { AccordionItemEditor } from './components/accordion/selector/AccordionItem/AccordionItemEditor.es.js';
export { AccordionItemEditor } from './components/accordion/selector/AccordionItem/AccordionItemEditor.es.js';
import { PopupEditorV1 } from './components/popup/v1/selector.es.js';
export { PopupEditorV1 } from './components/popup/v1/selector.es.js';
import { MenuItemEditor } from './components/header/selector/MenuItem/MenuItemEditor.es.js';
export { MenuItemEditor } from './components/header/selector/MenuItem/MenuItemEditor.es.js';
import { TabItemEditor } from './components/tab/selector/TabItemEditor/TabItemEditor.es.js';
export { TabItemEditor } from './components/tab/selector/TabItemEditor/TabItemEditor.es.js';
import { TestimonialItemV2Editor } from './components/testimonial/v2/selector/TestimonialItemV2Editor/TestimonialItemV2Editor.es.js';
export { TestimonialItemV2Editor } from './components/testimonial/v2/selector/TestimonialItemV2Editor/TestimonialItemV2Editor.es.js';
import { GlobalBlockEditor } from './components/global-block/selector.es.js';
export { GlobalBlockEditor } from './components/global-block/selector.es.js';
import { WebsiteFooterEditor } from './components/websitefooter/selector.es.js';

const selectors = {
  RootEditor,
  IconEditor,
  ColEditor,
  RowEditor,
  ButtonEditor,
  ButtonEditorV2,
  ParagraphEditor,
  HeadingEditor,
  DividerEditor,
  SpacerEditor,
  YoutubeEditor,
  CountdownEditor,
  MapEditor,
  ImageEditor,
  GalleryEditor,
  HtmlCodeEditor,
  PageFlipEditor,
  NumberCounterEditor,
  SocialIconEditor,
  StoreListEditor,
  SliderEditor,
  SliderEditorV2,
  TableEditor,
  AccordionEditor,
  AccordionContentEditor,
  AccordionItemEditor,
  ListEditor,
  ListItemEditor,
  ListItemContentEditor,
  ListItemIconEditor,
  ListItemEditorV2,
  ListItemIconEditorV2,
  ListItemContentEditorV2,
  FormV4Editor,
  ButtonSubmitV3Editor,
  FormEditor,
  FormV2Editor,
  FormV3Editor,
  ButtonSubmitEditor,
  SubscriptionForm,
  SubscriptionFormV2,
  PopupEditorV1,
  HeaderEditor,
  MenuColEditor,
  LogoEditor,
  MenuItemEditor,
  MenuEditor,
  LuckyDrawEditor,
  LuckyDrawV2Editor,
  TabEditor,
  TabItemEditor,
  TestimonialCarouselEditor,
  TestimonialItemV2Editor,
  TestimonialV2Editor,
  ContentSliderEditor,
  ContentSliderItemEditor,
  ContentSliderHeadingEditor,
  GlobalBlockEditor,
  ProductListEditor,
  ProductListContentEditor,
  WebsiteHeaderEditor,
  WebsiteFooterEditor,
  ContactBubbleEditor,
  TempiProductListEditor,
  QRCodeEditor,
  ExamScoreEditor,
  VideoEditor,
  ArticleListEditor,
  ArticleHeaderEditor,
  SampleComponentEditor,
  LeadFormEditor,
  InlineBannerEditor
};

export { selectors };
