// @tempi/core-elements/dist/selectors.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var selector$f = require('./components/icon/selector.js');
var selector = require('./components/button/v1/selector.js');
var selector$6 = require('./components/divider/selector.js');
var selector$t = require('./components/spacer/selector.js');
var selector$z = require('./components/youtube/selector.js');
var selector$5 = require('./components/countdown/selector.js');
var selector$j = require('./components/map/selector.js');
var selector$b = require('./components/gallery/selector.js');
var selector$e = require('./components/html-code/selector.js');
var selector$l = require('./components/page-flip/selector.js');
var selector$k = require('./components/number-counter/selector.js');
var selector$s = require('./components/social-icon/selector.js');
var selector$u = require('./components/store-list/selector.js');
var selector$q = require('./components/slider/v1/selector.js');
var selector$r = require('./components/slider/v2/selector.js');
var selector$x = require('./components/table/selector.js');
var ListEditor = require('./components/list/v1/selector/ListEditor.js');
var ListItemEditor = require('./components/list/v1/selector/ListItemEditor.js');
var ListItemContentEditor = require('./components/list/v1/selector/ListItemContentEditor.js');
var ListItemIconEditor = require('./components/list/v1/selector/ListItemIconEditor.js');
var ListItemEditorV2 = require('./components/list/v2/selector/ListItemV2/ListItemEditorV2.js');
var ListItemIconEditorV2 = require('./components/list/v2/selector/ListItemIconV2/ListItemIconEditorV2.js');
var ListItemContentEditorV2 = require('./components/list/v2/selector/ListItemContentV2/ListItemContentEditorV2.js');
var selector$a = require('./components/form/v4/selector.js');
var selector$3 = require('./components/button-submit/v3/selector.js');
var selector$7 = require('./components/form/v1/selector.js');
var selector$8 = require('./components/form/v2/selector.js');
var selector$9 = require('./components/form/v3/selector.js');
var selector$2 = require('./components/button-submit/v1/selector.js');
var selector$v = require('./components/subscription-form/v1/selector.js');
var selector$w = require('./components/subscription-form/v2/selector.js');
var HeaderEditor = require('./components/header/selector/Header/HeaderEditor.js');
var MenuColEditor = require('./components/header/selector/MenuCol/MenuColEditor.js');
var LogoEditor = require('./components/header/selector/Logo/LogoEditor.js');
var MenuEditor = require('./components/header/selector/Menu/MenuEditor.js');
var selector$h = require('./components/lucky-draw/v1/selector.js');
var selector$i = require('./components/lucky-draw/v2/selector.js');
var TabEditor = require('./components/tab/selector/TabEditor/TabEditor.js');
var selector$y = require('./components/testimonial-carousel/selector.js');
var TestimonialV2Editor = require('./components/testimonial/v2/selector/TestimonialV2Editor/TestimonialV2Editor.js');
var ContentSliderEditor = require('./components/content-slider/selector/ContentSliderEditor/ContentSliderEditor.js');
var ContentSliderItemEditor = require('./components/content-slider/selector/ContentSliderItemEditor/ContentSliderItemEditor.js');
var ContentSliderHeadingEditor = require('./components/content-slider/selector/ContentSliderHeadingEditor/ContentSliderHeadingEditor.js');
var ProductListEditor = require('./components/teko-product-list/selector/ProductListEditor/ProductListEditor.js');
var ProductListContentEditor = require('./components/teko-product-list/selector/ProductListContentEditor/ProductListContentEditor.js');
var selector$E = require('./components/website-header/selector.js');
var selector$G = require('./components/contact-bubble/selector.js');
var selector$H = require('./components/product-list/selector.js');
var selector$I = require('./components/qr-code/selector.js');
var selector$A = require('./components/exam-score/selector.js');
var selector$J = require('./components/video/selector.js');
var selector$B = require('./components/article-list/selector.js');
var selector$C = require('./components/article-header/selector.js');
var selector$D = require('./components/sample-component/selector.js');
var selector$K = require('./components/lead-form/selector.js');
var selector$L = require('./components/inline-banner/selector.js');
var selector$o = require('./components/root/selector.js');
var selector$4 = require('./components/col/selector.js');
var selector$p = require('./components/section/selector.js');
var selector$1 = require('./components/button/v2/selector.js');
var selector$m = require('./components/paragraph/selector.js');
var selector$d = require('./components/heading/selector.js');
var selector$g = require('./components/image/selector.js');
var AccordionEditor = require('./components/accordion/selector/Accordion/AccordionEditor.js');
var AccordionContentEditor = require('./components/accordion/selector/AccordionContent/AccordionContentEditor.js');
var AccordionItemEditor = require('./components/accordion/selector/AccordionItem/AccordionItemEditor.js');
var selector$n = require('./components/popup/v1/selector.js');
var MenuItemEditor = require('./components/header/selector/MenuItem/MenuItemEditor.js');
var TabItemEditor = require('./components/tab/selector/TabItemEditor/TabItemEditor.js');
var TestimonialItemV2Editor = require('./components/testimonial/v2/selector/TestimonialItemV2Editor/TestimonialItemV2Editor.js');
var selector$c = require('./components/global-block/selector.js');
var selector$F = require('./components/websitefooter/selector.js');

const selectors = {
  RootEditor: selector$o.RootEditor,
  IconEditor: selector$f.IconEditor,
  ColEditor: selector$4.ColEditor,
  RowEditor: selector$p.RowEditor,
  ButtonEditor: selector.ButtonEditor,
  ButtonEditorV2: selector$1.ButtonEditorV2,
  ParagraphEditor: selector$m.ParagraphEditor,
  HeadingEditor: selector$d.HeadingEditor,
  DividerEditor: selector$6.DividerEditor,
  SpacerEditor: selector$t.SpacerEditor,
  YoutubeEditor: selector$z.YoutubeEditor,
  CountdownEditor: selector$5.CountdownEditor,
  MapEditor: selector$j.MapEditor,
  ImageEditor: selector$g.ImageEditor,
  GalleryEditor: selector$b.GalleryEditor,
  HtmlCodeEditor: selector$e.HtmlCodeEditor,
  PageFlipEditor: selector$l.PageFlipEditor,
  NumberCounterEditor: selector$k.NumberCounterEditor,
  SocialIconEditor: selector$s.SocialIconEditor,
  StoreListEditor: selector$u.StoreListEditor,
  SliderEditor: selector$q.SliderEditor,
  SliderEditorV2: selector$r.SliderEditorV2,
  TableEditor: selector$x.TableEditor,
  AccordionEditor: AccordionEditor.AccordionEditor,
  AccordionContentEditor: AccordionContentEditor.AccordionContentEditor,
  AccordionItemEditor: AccordionItemEditor.AccordionItemEditor,
  ListEditor: ListEditor.ListEditor,
  ListItemEditor: ListItemEditor.ListItemEditor,
  ListItemContentEditor: ListItemContentEditor.ListItemContentEditor,
  ListItemIconEditor: ListItemIconEditor.ListItemIconEditor,
  ListItemEditorV2: ListItemEditorV2.ListItemEditorV2,
  ListItemIconEditorV2: ListItemIconEditorV2.ListItemIconEditorV2,
  ListItemContentEditorV2: ListItemContentEditorV2.ListItemContentEditorV2,
  FormV4Editor: selector$a.FormV4Editor,
  ButtonSubmitV3Editor: selector$3.ButtonSubmitV3Editor,
  FormEditor: selector$7.FormEditor,
  FormV2Editor: selector$8.FormV2Editor,
  FormV3Editor: selector$9.FormV3Editor,
  ButtonSubmitEditor: selector$2.ButtonSubmitEditor,
  SubscriptionForm: selector$v.SubscriptionForm,
  SubscriptionFormV2: selector$w.SubscriptionFormV2,
  PopupEditorV1: selector$n.PopupEditorV1,
  HeaderEditor: HeaderEditor.HeaderEditor,
  MenuColEditor: MenuColEditor.MenuColEditor,
  LogoEditor: LogoEditor.LogoEditor,
  MenuItemEditor: MenuItemEditor.MenuItemEditor,
  MenuEditor: MenuEditor.MenuEditor,
  LuckyDrawEditor: selector$h.LuckyDrawEditor,
  LuckyDrawV2Editor: selector$i.LuckyDrawV2Editor,
  TabEditor: TabEditor.TabEditor,
  TabItemEditor: TabItemEditor.TabItemEditor,
  TestimonialCarouselEditor: selector$y.TestimonialCarouselEditor,
  TestimonialItemV2Editor: TestimonialItemV2Editor.TestimonialItemV2Editor,
  TestimonialV2Editor: TestimonialV2Editor.TestimonialV2Editor,
  ContentSliderEditor: ContentSliderEditor.ContentSliderEditor,
  ContentSliderItemEditor: ContentSliderItemEditor.ContentSliderItemEditor,
  ContentSliderHeadingEditor: ContentSliderHeadingEditor.ContentSliderHeadingEditor,
  GlobalBlockEditor: selector$c.GlobalBlockEditor,
  ProductListEditor: ProductListEditor.ProductListEditor,
  ProductListContentEditor: ProductListContentEditor.ProductListContentEditor,
  WebsiteHeaderEditor: selector$E.WebsiteHeaderEditor,
  WebsiteFooterEditor: selector$F.WebsiteFooterEditor,
  ContactBubbleEditor: selector$G.ContactBubbleEditor,
  TempiProductListEditor: selector$H.TempiProductListEditor,
  QRCodeEditor: selector$I.QRCodeEditor,
  ExamScoreEditor: selector$A.ExamScoreEditor,
  VideoEditor: selector$J.VideoEditor,
  ArticleListEditor: selector$B.ArticleListEditor,
  ArticleHeaderEditor: selector$C.ArticleHeaderEditor,
  SampleComponentEditor: selector$D.SampleComponentEditor,
  LeadFormEditor: selector$K.LeadFormEditor,
  InlineBannerEditor: selector$L.InlineBannerEditor
};

exports.IconEditor = selector$f.IconEditor;
exports.ButtonEditor = selector.ButtonEditor;
exports.DividerEditor = selector$6.DividerEditor;
exports.SpacerEditor = selector$t.SpacerEditor;
exports.YoutubeEditor = selector$z.YoutubeEditor;
exports.CountdownEditor = selector$5.CountdownEditor;
exports.MapEditor = selector$j.MapEditor;
exports.GalleryEditor = selector$b.GalleryEditor;
exports.HtmlCodeEditor = selector$e.HtmlCodeEditor;
exports.PageFlipEditor = selector$l.PageFlipEditor;
exports.NumberCounterEditor = selector$k.NumberCounterEditor;
exports.SocialIconEditor = selector$s.SocialIconEditor;
exports.StoreListEditor = selector$u.StoreListEditor;
exports.SliderEditor = selector$q.SliderEditor;
exports.SliderEditorV2 = selector$r.SliderEditorV2;
exports.TableEditor = selector$x.TableEditor;
exports.ListEditor = ListEditor.ListEditor;
exports.ListItemEditor = ListItemEditor.ListItemEditor;
exports.ListItemContentEditor = ListItemContentEditor.ListItemContentEditor;
exports.ListItemIconEditor = ListItemIconEditor.ListItemIconEditor;
exports.ListItemEditorV2 = ListItemEditorV2.ListItemEditorV2;
exports.ListItemIconEditorV2 = ListItemIconEditorV2.ListItemIconEditorV2;
exports.ListItemContentEditorV2 = ListItemContentEditorV2.ListItemContentEditorV2;
exports.FormV4Editor = selector$a.FormV4Editor;
exports.ButtonSubmitV3Editor = selector$3.ButtonSubmitV3Editor;
exports.FormEditor = selector$7.FormEditor;
exports.FormV2Editor = selector$8.FormV2Editor;
exports.FormV3Editor = selector$9.FormV3Editor;
exports.ButtonSubmitEditor = selector$2.ButtonSubmitEditor;
exports.SubscriptionForm = selector$v.SubscriptionForm;
exports.SubscriptionFormV2 = selector$w.SubscriptionFormV2;
exports.HeaderEditor = HeaderEditor.HeaderEditor;
exports.MenuColEditor = MenuColEditor.MenuColEditor;
exports.LogoEditor = LogoEditor.LogoEditor;
exports.MenuEditor = MenuEditor.MenuEditor;
exports.LuckyDrawEditor = selector$h.LuckyDrawEditor;
exports.LuckyDrawV2Editor = selector$i.LuckyDrawV2Editor;
exports.TabEditor = TabEditor.TabEditor;
exports.TestimonialCarouselEditor = selector$y.TestimonialCarouselEditor;
exports.TestimonialV2Editor = TestimonialV2Editor.TestimonialV2Editor;
exports.ContentSliderEditor = ContentSliderEditor.ContentSliderEditor;
exports.ContentSliderItemEditor = ContentSliderItemEditor.ContentSliderItemEditor;
exports.ContentSliderHeadingEditor = ContentSliderHeadingEditor.ContentSliderHeadingEditor;
exports.ExamScoreEditor = selector$A.ExamScoreEditor;
exports.ArticleListEditor = selector$B.ArticleListEditor;
exports.ArticleHeaderEditor = selector$C.ArticleHeaderEditor;
exports.SampleComponentEditor = selector$D.SampleComponentEditor;
exports.RootEditor = selector$o.RootEditor;
exports.ColEditor = selector$4.ColEditor;
exports.RowEditor = selector$p.RowEditor;
exports.ButtonEditorV2 = selector$1.ButtonEditorV2;
exports.ParagraphEditor = selector$m.ParagraphEditor;
exports.HeadingEditor = selector$d.HeadingEditor;
exports.ImageEditor = selector$g.ImageEditor;
exports.AccordionEditor = AccordionEditor.AccordionEditor;
exports.AccordionContentEditor = AccordionContentEditor.AccordionContentEditor;
exports.AccordionItemEditor = AccordionItemEditor.AccordionItemEditor;
exports.PopupEditorV1 = selector$n.PopupEditorV1;
exports.MenuItemEditor = MenuItemEditor.MenuItemEditor;
exports.TabItemEditor = TabItemEditor.TabItemEditor;
exports.TestimonialItemV2Editor = TestimonialItemV2Editor.TestimonialItemV2Editor;
exports.GlobalBlockEditor = selector$c.GlobalBlockEditor;
exports.selectors = selectors;
