// @tempi/core-elements/dist/components/product-list/components/ProductGrid/ProductGrid.stories.js
import { jsx as _jsx } from "react/jsx-runtime";
import { ProductGrid } from './ProductGrid';
const defaultProductGrid = {
    alignment: 'left',
    gridOptions: {
        columns: 5,
        rows: 1,
    },
    layout: 'grid',
    items: [
        {
            name: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry',
            price: 500000,
            image: {
                src: 'https://vcdn1-dulich.vnecdn.net/2021/07/16/8-1626444967.jpg?w=1200&h=0&q=100&dpr=1&fit=crop&s=GfgGn4dNuKZexy1BGkAUNA',
            },
            link: 'https://phongvu.vn/msi-gaming-gf63-thin-11sc-665vn-i5-11400h--s221101658',
        },
        {
            name: 'Product 1',
            price: 20000,
            discount: 30,
            image: {
                src: 'https://lh3.googleusercontent.com/fKUsNfKwAQggLUyMJ2TpHJ6IT3YI2suMdruCmrVlpbgPy4DErIw60__FBElMmKgtkCAYWFi4wAgnEaEwQeQ',
            },
            link: 'https://phongvu.vn/msi-gaming-gf63-thin-11sc-665vn-i5-11400h--s221101658',
        },
        {
            name: 'Product 1',
            price: 20000,
            discount: 30,
            image: {
                src: 'https://lh3.googleusercontent.com/HDchawn0E7MWVK60_1CVQafezWgK5sqhYFlkCEOkKksTIAehk6BYW2P55qxtwy8CO3WBVI558YrEs8cGoqAKKhNuARtZ168',
            },
        },
        {
            name: 'Product 1',
            price: 20000,
            discount: 30,
            image: {
                src: 'https://lh3.googleusercontent.com/XY6Olaf3hkIL7V42TzBDTuYCLT4VQ6CWDIAGuRmLSWU_JZTywcdHaP5_BVTlo10y9RUh6JOdvDRE27wJrIO36HovMdNAO7C81Q',
            },
        },
        {
            name: 'Product 1',
            price: 20000,
            discount: 30,
            image: {
                src: 'https://lh3.googleusercontent.com/fKUsNfKwAQggLUyMJ2TpHJ6IT3YI2suMdruCmrVlpbgPy4DErIw60__FBElMmKgtkCAYWFi4wAgnEaEwQeQ',
            },
        },
        {
            name: 'Product 1',
            price: 20000,
            discount: 30,
            image: {
                src: 'https://lh3.googleusercontent.com/fKUsNfKwAQggLUyMJ2TpHJ6IT3YI2suMdruCmrVlpbgPy4DErIw60__FBElMmKgtkCAYWFi4wAgnEaEwQeQ',
            },
        },
    ],
};
export default {
    title: 'CoreElements/ProductGrid',
};
const Template = (args) => (_jsx("div", { children: _jsx(ProductGrid, Object.assign({}, args)) }));
export const Default = Template.bind({});
Default.args = defaultProductGrid;
