// @tempi/core-elements/dist/components/websitefooter/selector.es.js
import { jsx } from 'react/jsx-runtime';
import { useNode, crafting, getEditorName } from '@tempi/core-editor';
import { WebsiteFooter } from './ui.es.js';
import { DEFAULT_WEBSITE_FOOTER } from './constants.es.js';
import { WebsiteFooterSetting } from './quick-setting.es.js';
import { RenderFooterSetting } from '../../core/RenderFooterSetting/RenderFooterSetting.es.js';
import { RowEditor } from '../section/selector.es.js';

const WebsiteFooterEditor = ({ children }) => {
  const {
    connectors: { connect }
  } = useNode();
  return /* @__PURE__ */ jsx("div", {
    ref: connect,
    style: {
      width: "100%",
      position: "relative"
    },
    children: /* @__PURE__ */ jsx(RenderFooterSetting, {
      footerSetting: /* @__PURE__ */ jsx(WebsiteFooterSetting, {}),
      children: /* @__PURE__ */ jsx(WebsiteFooter, {
        children
      })
    })
  });
};
WebsiteFooterEditor.craft = crafting({
  displayName: "Website Footer",
  tag: WebsiteFooter.displayName.toLowerCase(),
  rules: {
    canDrag: () => false,
    canMoveIn: (incomingNodes) => {
      return incomingNodes == null ? void 0 : incomingNodes.every(
        (incomingNode) => incomingNode.data.name === getEditorName({ RowEditor })
      );
    }
  },
  props: DEFAULT_WEBSITE_FOOTER
});

export { WebsiteFooterEditor };
