// @tempi/core-elements/dist/components/websitefooter/selector.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var coreEditor = require('@tempi/core-editor');
var ui = require('./ui.js');
var constants = require('./constants.js');
var quickSetting = require('./quick-setting.js');
var RenderFooterSetting = require('../../core/RenderFooterSetting/RenderFooterSetting.js');
var selector = require('../section/selector.js');

const WebsiteFooterEditor = ({ children }) => {
  const {
    connectors: { connect }
  } = coreEditor.useNode();
  return /* @__PURE__ */ jsxRuntime.jsx("div", {
    ref: connect,
    style: {
      width: "100%",
      position: "relative"
    },
    children: /* @__PURE__ */ jsxRuntime.jsx(RenderFooterSetting.RenderFooterSetting, {
      footerSetting: /* @__PURE__ */ jsxRuntime.jsx(quickSetting.WebsiteFooterSetting, {}),
      children: /* @__PURE__ */ jsxRuntime.jsx(ui.WebsiteFooter, {
        children
      })
    })
  });
};
WebsiteFooterEditor.craft = coreEditor.crafting({
  displayName: "Website Footer",
  tag: ui.WebsiteFooter.displayName.toLowerCase(),
  rules: {
    canDrag: () => false,
    canMoveIn: (incomingNodes) => {
      return incomingNodes == null ? void 0 : incomingNodes.every(
        (incomingNode) => incomingNode.data.name === coreEditor.getEditorName({ RowEditor: selector.RowEditor })
      );
    }
  },
  props: constants.DEFAULT_WEBSITE_FOOTER
});

exports.WebsiteFooterEditor = WebsiteFooterEditor;
