// @tempi/core-elements/dist/components/teko-product-list/ui/NoProductsFound/NoProductsFound.es.js
import { jsxs, jsx } from 'react/jsx-runtime';
import styled from '@emotion/styled';
import { mappingAssets } from '@tempi/core-renderer';
import { Image } from '../../../image/ui.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const { noProductFoundImage } = mappingAssets;
const NoProductsFound = (_a) => {
  var _b = _a, {
    text
  } = _b, props = __objRest(_b, [
    "text"
  ]);
  return /* @__PURE__ */ jsxs(Wrapper, __spreadProps(__spreadValues({}, props), {
    children: [
      /* @__PURE__ */ jsx(Image, {
        width: 200,
        src: noProductFoundImage
      }),
      !!text && /* @__PURE__ */ jsx(Text, {
        children: text
      })
    ]
  }));
};
const Wrapper = styled.div`
  text-align: center;
  width: 100%;
  padding: 0.5rem 0;
`;
const Text = styled.div`
  margin-top: 0.5rem;
  font-size: 0.9rem;
  color: #82869e;
`;

export { NoProductsFound };
