// @tempi/core-editor/dist/components/SubQuickSetting/SubQuickSetting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var antd = require('antd');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const SubQuickSetting = (_a) => {
  var _b = _a, {
    children,
    overlayInnerStyle
  } = _b, rest = __objRest(_b, [
    "children",
    "overlayInnerStyle"
  ]);
  return /* @__PURE__ */ jsxRuntime.jsx(antd.Popover, __spreadProps(__spreadValues({
    trigger: "click",
    placement: "bottom",
    overlayInnerStyle: __spreadValues({
      padding: 12,
      borderRadius: 8,
      width: 310
    }, overlayInnerStyle),
    zIndex: 1065,
    getPopupContainer: (trigger) => trigger
  }, rest), {
    children
  }));
};

exports.SubQuickSetting = SubQuickSetting;
