// @tempi/core-elements/dist/components/store-list/configs/StoreDataConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../i18n.js');
var sortable = require('@dnd-kit/sortable');
var utilities = require('@dnd-kit/utilities');
var styled = require('@emotion/styled');
var reactIcons = require('@fluentui/react-icons');
var coreEditor = require('@tempi/core-editor');
var antd = require('antd');
var hooks = require('../hooks.js');
var propKey = require('../../../constants/propKey.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const StoreDataConfig = ({ index, store, activeField, setActiveField, onDuplicate, onRemove }) => {
  const { setNodeRef, attributes, listeners, transform } = sortable.useSortable({
    id: store.dndId
  });
  const [province, setProvince] = coreEditor.useProp(
    propKey.PROP_KEY.joinKey(propKey.PROP_KEY.Stores, index.toString(), propKey.PROP_KEY.CityId)
  );
  const [district, setDistrict] = coreEditor.useProp(
    propKey.PROP_KEY.joinKey(propKey.PROP_KEY.Stores, index.toString(), propKey.PROP_KEY.DistrictId)
  );
  const { cities, districts } = hooks.useGetCitiesAndDistricts();
  const getOptions = () => {
    const options = [];
    cities.forEach((city) => {
      const localDistrict = districts.filter(
        (d) => d.idProvince === city.idProvince
      );
      options.push({
        label: city.name,
        value: city.idProvince,
        children: localDistrict.map((local) => ({
          label: local.name,
          value: local.idDistrict
        }))
      });
    });
    return options;
  };
  const filter = (inputValue, path) => path.some(
    (option) => coreEditor.nomalizeText(option.label).toLowerCase().indexOf(coreEditor.nomalizeText(inputValue.toLowerCase())) > -1
  );
  const onChangeLocation = (value) => {
    const [firstLevel, secondLevel] = value || [];
    setProvince(firstLevel);
    setDistrict(secondLevel);
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(StyledTagField, {
    ref: setNodeRef,
    style: { transform: utilities.CSS.Transform.toString(transform) },
    children: [
      /* @__PURE__ */ jsxRuntime.jsxs(StyledTagHeader, {
        onClick: () => {
          if (activeField === index)
            setActiveField(void 0);
          else
            setActiveField(index);
        },
        children: [
          /* @__PURE__ */ jsxRuntime.jsxs("div", {
            className: "flex",
            children: [
              /* @__PURE__ */ jsxRuntime.jsx(StyledAction, __spreadProps(__spreadValues(__spreadValues({}, listeners), attributes), {
                style: { cursor: "move", marginRight: 8 },
                children: /* @__PURE__ */ jsxRuntime.jsx(reactIcons.ReOrderDotsVerticalRegular, {
                  style: {
                    fontSize: 20
                  }
                })
              })),
              /* @__PURE__ */ jsxRuntime.jsx("div", {
                children: store.name
              })
            ]
          }),
          /* @__PURE__ */ jsxRuntime.jsxs("div", {
            className: "flex",
            children: [
              /* @__PURE__ */ jsxRuntime.jsx(StyledAction, {
                style: { marginRight: 8 },
                onClick: (e) => {
                  e.stopPropagation();
                  onDuplicate();
                },
                children: /* @__PURE__ */ jsxRuntime.jsx(reactIcons.CopyRegular, {
                  style: {
                    fontSize: 16
                  }
                })
              }),
              /* @__PURE__ */ jsxRuntime.jsx(StyledAction, {
                onClick: (e) => {
                  e.stopPropagation();
                  onRemove();
                },
                children: /* @__PURE__ */ jsxRuntime.jsx(reactIcons.DeleteRegular, {
                  style: {
                    fontSize: 16
                  }
                })
              })
            ]
          })
        ]
      }),
      activeField === index && /* @__PURE__ */ jsxRuntime.jsxs(StyledTagContent, {
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
            label: i18n.t("V\u1ECB tr\xED"),
            children: /* @__PURE__ */ jsxRuntime.jsx(antd.Cascader, {
              options: getOptions(),
              onChange: onChangeLocation,
              placeholder: i18n.t("Ch\u1ECDn \u0111\u1ECBa \u0111i\u1EC3m"),
              showSearch: { filter },
              getPopupContainer: (trigger) => trigger,
              style: { width: 140 },
              value: [province, district],
              allowClear: false
            })
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputConfig, {
            formItemProps: {
              label: i18n.t("T\xEAn")
            },
            propKey: propKey.PROP_KEY.joinKey(
              propKey.PROP_KEY.Stores,
              index.toString(),
              propKey.PROP_KEY.Name
            )
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputConfig, {
            formItemProps: {
              label: i18n.t("\u0110\u1ECBa ch\u1EC9")
            },
            propKey: propKey.PROP_KEY.joinKey(
              propKey.PROP_KEY.Stores,
              index.toString(),
              propKey.PROP_KEY.Address
            )
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputConfig, {
            formItemProps: {
              label: i18n.t("S\u1ED1 \u0111i\u1EC7n tho\u1EA1i")
            },
            propKey: propKey.PROP_KEY.joinKey(
              propKey.PROP_KEY.Stores,
              index.toString(),
              propKey.PROP_KEY.Phone
            )
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputConfig, {
            formItemProps: {
              label: i18n.t("Gi\u1EDD l\xE0m vi\u1EC7c")
            },
            propKey: propKey.PROP_KEY.joinKey(
              propKey.PROP_KEY.Stores,
              index.toString(),
              propKey.PROP_KEY.WorkHours
            )
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputConfig, {
            formItemProps: {
              label: i18n.t("B\u1EA3n \u0111\u1ED3")
            },
            propKey: propKey.PROP_KEY.joinKey(
              propKey.PROP_KEY.Stores,
              index.toString(),
              propKey.PROP_KEY.Map
            )
          })
        ]
      })
    ]
  }, index);
};
const StyledTagField = styled__default["default"].div`
  justify-content: space-between;
  padding: 0.5rem;
  border-radius: 4px;
  border: 1px solid var(--border-color);
  margin-bottom: 8px;
`;
const StyledTagHeader = styled__default["default"].div`
  display: flex;
  flex-grow: 1;
  align-items: center;
  justify-content: space-between;
  cursor: pointer;
`;
const StyledTagContent = styled__default["default"].div`
  padding: 0.5rem;
`;
const StyledAction = styled__default["default"].div`
  background-color: transparent;
  cursor: pointer;
`;

exports.StoreDataConfig = StoreDataConfig;
