// @tempi/core-editor/dist/configs/CodeEditorConfig/CodeEditorConfig.es.js
import { jsx } from 'react/jsx-runtime';
import React, { useEffect } from 'react';
import { useProp } from '../../hooks/useProp.es.js';
import { FormItem } from '../../components/FormItem/FormItem.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const AceEditor = React.lazy(() => import('react-ace'));
const CodeEditorConfig = ({
  propKey,
  formItemProps
}) => {
  const [prop, setProp] = useProp(propKey);
  useEffect(() => {
    const importDepsAce = async () => {
      await import('ace-builds/src-noconflict/mode-java');
      await import('ace-builds/src-noconflict/theme-github');
    };
    importDepsAce();
  }, []);
  return /* @__PURE__ */ jsx(React.Suspense, {
    fallback: /* @__PURE__ */ jsx("div", {}),
    children: /* @__PURE__ */ jsx(FormItem, __spreadProps(__spreadValues({}, formItemProps), {
      children: /* @__PURE__ */ jsx(AceEditor, {
        value: prop,
        onChange: (value) => setProp(value),
        mode: "html",
        fontSize: 14,
        height: "300px",
        width: "100%",
        setOptions: {
          showLineNumbers: true,
          tabSize: 2
        },
        highlightActiveLine: true
      })
    }))
  });
};

export { CodeEditorConfig };
