// @tempi/core-elements/dist/components/spacer/quick-setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var icons = require('@ant-design/icons');
var coreEditor = require('@tempi/core-editor');
var coreRenderer = require('@tempi/core-renderer');

const SpacerSetting = () => {
  return /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
    propKey: "height",
    formItemProps: {
      label: i18n.t("Kho\u1EA3ng c\xE1ch")
    },
    unitProps: { defaultValue: coreRenderer.UnitEnum.pixel }
  });
};
const SpacerQuickSetting = () => {
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SubQuickSetting, {
        content: SpacerSetting,
        children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.QuickSettingButton, {
          children: /* @__PURE__ */ jsxRuntime.jsx(icons.ColumnHeightOutlined, {})
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.OpenSettingDialog, {}),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.DuplicateNode, {}),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.DeleteNode, {})
    ]
  });
};

exports.SpacerQuickSetting = SpacerQuickSetting;
