// @tempi/core-elements/dist/components/spacer/quick-setting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { ColumnHeightOutlined } from '@ant-design/icons';
import { SubQuickSetting, QuickSettingButton, OpenSettingDialog, DuplicateNode, DeleteNode, InputNumberConfig } from '@tempi/core-editor';
import { UnitEnum } from '@tempi/core-renderer';

const SpacerSetting = () => {
  return /* @__PURE__ */ jsx(InputNumberConfig, {
    propKey: "height",
    formItemProps: {
      label: t("Kho\u1EA3ng c\xE1ch")
    },
    unitProps: { defaultValue: UnitEnum.pixel }
  });
};
const SpacerQuickSetting = () => {
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(SubQuickSetting, {
        content: SpacerSetting,
        children: /* @__PURE__ */ jsx(QuickSettingButton, {
          children: /* @__PURE__ */ jsx(ColumnHeightOutlined, {})
        })
      }),
      /* @__PURE__ */ jsx(OpenSettingDialog, {}),
      /* @__PURE__ */ jsx(DuplicateNode, {}),
      /* @__PURE__ */ jsx(DeleteNode, {})
    ]
  });
};

export { SpacerQuickSetting };
