// @tempi/core-editor/dist/configs/CornerConfig/CornerConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var styled = require('@emotion/styled');
var reactIcons = require('@fluentui/react-icons');
var coreRenderer = require('@tempi/core-renderer');
var antd = require('antd');
var usePropSeparateDevice = require('../../hooks/usePropSeparateDevice.js');
var propKey = require('../../constants/propKey.js');
var FormItem = require('../../components/FormItem/FormItem.js');
var RangeSliderConfig = require('../RangeSliderConfig/RangeSliderConfig.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

const cornerTypes = ["common", "detail"];
const MAX_BORDER_RADIUS = 100;
const CornerConfig = ({
  propKey: propKey$1,
  separateDevice
}) => {
  const [cornerType, setCornerType] = usePropSeparateDevice.usePropSeparateDevice(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.CornerType),
    separateDevice
  );
  const [topLeftRadius, setTopLeftRadius] = usePropSeparateDevice.usePropSeparateDevice(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.BorderTopLeftRadius),
    separateDevice
  );
  const [topRightRadius, setTopRightRadius] = usePropSeparateDevice.usePropSeparateDevice(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.BorderTopRightRadius),
    separateDevice
  );
  const [bottomLeftRadius, setBottomLeftRadius] = usePropSeparateDevice.usePropSeparateDevice(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.BorderBottomLeftRadius),
    separateDevice
  );
  const [bottomRightRadius, setBottomRightRadius] = usePropSeparateDevice.usePropSeparateDevice(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.BorderBottomRightRadius),
    separateDevice
  );
  const [borderRadius] = usePropSeparateDevice.usePropSeparateDevice(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.BorderRadius),
    separateDevice
  );
  const [, setBorderRadiusUnit] = usePropSeparateDevice.usePropSeparateDevice(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.BorderRadiusUnit),
    separateDevice
  );
  const [, setTopRightRadiusUnit] = usePropSeparateDevice.usePropSeparateDevice(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.BorderTopRightRadiusUnit),
    separateDevice
  );
  const [, setTopLeftRadiusUnit] = usePropSeparateDevice.usePropSeparateDevice(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.BorderTopLeftRadiusUnit),
    separateDevice
  );
  const [, setBottomLeftRadiusUnit] = usePropSeparateDevice.usePropSeparateDevice(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.BorderBottomLeftRadiusUnit),
    separateDevice
  );
  const [, setBottomRightRadiusUnit] = usePropSeparateDevice.usePropSeparateDevice(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.BorderBottomRightRadiusUnit),
    separateDevice
  );
  const clickOption = (value) => {
    if (value === "detail" && !topLeftRadius && !topRightRadius && !bottomLeftRadius && !bottomRightRadius) {
      setTopLeftRadius(borderRadius);
      setTopRightRadius(borderRadius);
      setBottomLeftRadius(borderRadius);
      setBottomRightRadius(borderRadius);
    }
    setCornerType(value);
  };
  const renderIcon = (value) => {
    switch (value) {
      case "common":
        return /* @__PURE__ */ jsxRuntime.jsx(reactIcons.BorderOutsideRegular, {
          fontSize: 16
        });
      case "detail":
        return /* @__PURE__ */ jsxRuntime.jsx(reactIcons.BorderTopBottomRegular, {
          fontSize: 16
        });
    }
  };
  const onChangeUnit = (unit, setUnit) => {
    setUnit(unit);
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(FormItem.FormItem, {
        label: i18n.t("Bo g\xF3c"),
        layout: "horizontal",
        children: /* @__PURE__ */ jsxRuntime.jsx(antd.Radio.Group, {
          value: cornerType,
          optionType: "button",
          buttonStyle: "solid",
          defaultValue: cornerType || "common",
          children: cornerTypes.map((value, index) => /* @__PURE__ */ jsxRuntime.jsx(antd.Radio, {
            onClick: () => clickOption(value),
            value,
            children: renderIcon(value)
          }, index))
        })
      }),
      cornerType === "detail" ? /* @__PURE__ */ jsxRuntime.jsxs(WrapperDetailCornerConfig, {
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(RangeSliderConfig.RangeSliderConfig, {
            propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.BorderTopLeftRadius),
            formItemProps: {
              label: i18n.t("Tr\xE1i tr\xEAn"),
              layout: "horizontal"
            },
            sliderProps: { min: 0, max: MAX_BORDER_RADIUS, defaultValue: 0 },
            unitProps: {
              defaultValue: coreRenderer.UnitEnum.percent,
              propKey: propKey.PROP_KEY.joinKey(
                propKey$1,
                propKey.PROP_KEY.BorderTopLeftRadiusUnit
              ),
              options: [coreRenderer.UnitEnum.percent, coreRenderer.UnitEnum.pixel],
              onChange: (value) => onChangeUnit(value, setTopLeftRadiusUnit)
            }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(RangeSliderConfig.RangeSliderConfig, {
            propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.BorderTopRightRadius),
            formItemProps: {
              label: i18n.t("Ph\u1EA3i tr\xEAn"),
              layout: "horizontal"
            },
            sliderProps: { min: 0, max: MAX_BORDER_RADIUS, defaultValue: 0 },
            unitProps: {
              defaultValue: coreRenderer.UnitEnum.percent,
              propKey: propKey.PROP_KEY.joinKey(
                propKey$1,
                propKey.PROP_KEY.BorderTopRightRadiusUnit
              ),
              options: [coreRenderer.UnitEnum.percent, coreRenderer.UnitEnum.pixel],
              onChange: (value) => onChangeUnit(value, setTopRightRadiusUnit)
            }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(RangeSliderConfig.RangeSliderConfig, {
            propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.BorderBottomLeftRadius),
            formItemProps: {
              label: i18n.t("Tr\xE1i d\u01B0\u1EDBi"),
              layout: "horizontal"
            },
            sliderProps: { min: 0, max: MAX_BORDER_RADIUS, defaultValue: 0 },
            unitProps: {
              defaultValue: coreRenderer.UnitEnum.percent,
              propKey: propKey.PROP_KEY.joinKey(
                propKey$1,
                propKey.PROP_KEY.BorderBottomLeftRadiusUnit
              ),
              options: [coreRenderer.UnitEnum.percent, coreRenderer.UnitEnum.pixel],
              onChange: (value) => onChangeUnit(value, setBottomLeftRadiusUnit)
            }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(RangeSliderConfig.RangeSliderConfig, {
            propKey: propKey.PROP_KEY.joinKey(
              propKey$1,
              propKey.PROP_KEY.BorderBottomRightRadius
            ),
            formItemProps: {
              label: i18n.t("Ph\u1EA3i d\u01B0\u1EDBi"),
              layout: "horizontal"
            },
            sliderProps: { min: 0, max: MAX_BORDER_RADIUS, defaultValue: 0 },
            unitProps: {
              defaultValue: coreRenderer.UnitEnum.percent,
              propKey: propKey.PROP_KEY.joinKey(
                propKey$1,
                propKey.PROP_KEY.BorderBottomRightRadiusUnit
              ),
              options: [coreRenderer.UnitEnum.percent, coreRenderer.UnitEnum.pixel],
              onChange: (value) => onChangeUnit(value, setBottomRightRadiusUnit)
            }
          })
        ]
      }) : /* @__PURE__ */ jsxRuntime.jsx(RangeSliderConfig.RangeSliderConfig, {
        propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.BorderRadius),
        sliderProps: { min: 0, max: MAX_BORDER_RADIUS, defaultValue: 0 },
        unitProps: {
          defaultValue: coreRenderer.UnitEnum.percent,
          propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.BorderRadiusUnit),
          options: [coreRenderer.UnitEnum.percent, coreRenderer.UnitEnum.pixel],
          onChange: (value) => onChangeUnit(value, setBorderRadiusUnit)
        }
      })
    ]
  });
};
const WrapperDetailCornerConfig = styled__default["default"].div`
  padding: 0px 8px 8px 8px;
  .ant-row {
    width: 83% !important;
  }
`;

exports.CornerConfig = CornerConfig;
