// @tempi/core-editor/dist/configs/TimingConfig/SelectSegment.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var styled = require('@emotion/styled');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

const optionDayInWeek = [
  {
    label: "CN",
    value: 0
  },
  {
    label: "T2",
    value: 1
  },
  {
    label: "T3",
    value: 2
  },
  {
    label: "T4",
    value: 3
  },
  {
    label: "T5",
    value: 4
  },
  {
    label: "T6",
    value: 5
  },
  {
    label: "T7",
    value: 6
  }
];
const SelectSegment = ({
  arrSelected = [1],
  onchange
}) => {
  return /* @__PURE__ */ jsxRuntime.jsx(StyledDiv, {
    children: optionDayInWeek.map((item) => /* @__PURE__ */ jsxRuntime.jsx(StyledItem, {
      className: arrSelected.includes(item.value) ? "active" : "",
      onClick: () => {
        const isSelected = !arrSelected.includes(item.value);
        if (isSelected || !isSelected && (arrSelected == null ? void 0 : arrSelected.length) > 1) {
          onchange(item.value);
        }
      },
      children: item.label
    }, item.value))
  });
};
const StyledDiv = styled__default["default"].div`
  display: flex;
  justify-content: space-between;
  width: 100%;
  border: 1px solid #d8d6de;
  border-radius: 4px;
  max-width: 320px;
`;
const StyledItem = styled__default["default"].div`
  &.active,
  &:hover {
    color: #7367f0;
  }
  padding: 4px 8px;
  cursor: pointer;
`;

exports.SelectSegment = SelectSegment;
