// @tempi/core-elements/dist/components/page-flip/constants.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var coreRenderer = require('@tempi/core-renderer');

const DEFAULT_PAGE_FLIP = {
  commonStyle: {
    justifyContent: "center"
  },
  width: 595,
  height: 842,
  minWidth: 315,
  minHeight: 400,
  maxWidth: 2e3,
  maxHeight: 2500,
  size: "stretch",
  drawShadow: true,
  flippingTime: 1e3,
  usePortrait: true,
  startZIndex: 0,
  autoSize: true,
  showCover: false,
  mobileScrollSupport: true,
  maxShadowOpacity: 0.5,
  align: coreRenderer.Alignment.center,
  images: [
    {
      src: "https://lh3.googleusercontent.com/fme57ZJ5E8LDAuiBOdfSCl6Ox2J-dm_tkSJFpPH84KJH-FiFHmmJuu7tLYWCx8aIbPHYUtXjmMqkkrD9S3N3spmmZGFhyAc",
      googleImageParams: {
        width: 0
      }
    },
    {
      src: "https://lh3.googleusercontent.com/4CyrPB24nPsyzkRiOvhkHjoXnp85CZk0PGV24jHy3Cg6V3b0Xpwa7sUq-FrseKv1V5ANBEF2_u6kQZhjacrrs8TNTCTRznc",
      googleImageParams: {
        width: 0
      }
    },
    {
      src: "https://lh3.googleusercontent.com/eubbkpQN-TWjh6Jghnb28geuiiHBtUjKWfPfSYjp2Y5nAIPrJ84U00yUTiLFAquesy15gkRg1sgZchdpdTMM0tYxpqlzLbw",
      googleImageParams: {
        width: 0
      }
    },
    {
      src: "https://lh3.googleusercontent.com/1DMzHybfiLsRu3VAkUUiSTwX75KFOwgcImCqrXdk4JrIJnI7BRkqPv3Ou8YUWk1p_8nef39GZ9oyBvBB16cFXfuKFh8Yhmo=rw-w1754",
      googleImageParams: {
        width: 0
      }
    }
  ]
};

exports.DEFAULT_PAGE_FLIP = DEFAULT_PAGE_FLIP;
