// @tempi/core-elements/dist/components/page-flip/quick-setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var coreEditor = require('@tempi/core-editor');
var AlignmentQuickSetting = require('../../quick-configs/AlignmentQuickSetting/AlignmentQuickSetting.js');
var propKey = require('../../constants/propKey.js');

const PageFlipQuickSetting = () => {
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ImageListQuickSetting, {
        propKey: "images",
        actions: ["editInfo"],
        imageHeight: 500
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.OpenSettingDialog, {}),
      /* @__PURE__ */ jsxRuntime.jsx(AlignmentQuickSetting.AlignmentQuickSetting, {
        propKey: propKey.PROP_KEY.Align,
        listAlignment: ["left", "center", "right"]
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.DuplicateNode, {}),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.DeleteNode, {})
    ]
  });
};

exports.PageFlipQuickSetting = PageFlipQuickSetting;
