// @tempi/core-elements/dist/components/page-flip/selector.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var coreEditor = require('@tempi/core-editor');
var constants = require('./constants.js');
var quickSetting = require('./quick-setting.js');
var setting = require('./setting.js');
var ui = require('./ui.js');
var selector = require('../section/selector.js');
var selector$1 = require('../root/selector.js');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const Inner = (props) => {
  return /* @__PURE__ */ jsxRuntime.jsx(ui.PageFlip, __spreadProps(__spreadValues({
    style: { pointerEvents: "none" }
  }, props), {
    mobileScrollSupport: true
  }));
};
const PageFlipEditor = coreEditor.withSelector(
  Inner,
  {
    displayName: "Trang s\xE1ch",
    tag: ui.PageFlip.displayName.toLowerCase(),
    quickSetting: quickSetting.PageFlipQuickSetting,
    customAttributes: setting.PageFlipSetting,
    advanceAttributes: coreEditor.AdvanceAttributes,
    props: constants.DEFAULT_PAGE_FLIP,
    rules: {
      canDrop: (targetNode) => {
        const ignoreTargets = [
          coreEditor.getEditorName({ RowEditor: selector.RowEditor }),
          coreEditor.getEditorName({ RootEditor: selector$1.RootEditor })
        ];
        return !ignoreTargets.includes(targetNode.data.name);
      }
    }
  },
  {
    style: { cursor: "pointer" }
  }
);

exports.PageFlipEditor = PageFlipEditor;
