// @tempi/core-elements/dist/components/page-flip/selector.es.js
import { jsx } from 'react/jsx-runtime';
import { withSelector, AdvanceAttributes, getEditorName } from '@tempi/core-editor';
import { DEFAULT_PAGE_FLIP } from './constants.es.js';
import { PageFlipQuickSetting } from './quick-setting.es.js';
import { PageFlipSetting } from './setting.es.js';
import { PageFlip } from './ui.es.js';
import { RowEditor } from '../section/selector.es.js';
import { RootEditor } from '../root/selector.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const Inner = (props) => {
  return /* @__PURE__ */ jsx(PageFlip, __spreadProps(__spreadValues({
    style: { pointerEvents: "none" }
  }, props), {
    mobileScrollSupport: true
  }));
};
const PageFlipEditor = withSelector(
  Inner,
  {
    displayName: "Trang s\xE1ch",
    tag: PageFlip.displayName.toLowerCase(),
    quickSetting: PageFlipQuickSetting,
    customAttributes: PageFlipSetting,
    advanceAttributes: AdvanceAttributes,
    props: DEFAULT_PAGE_FLIP,
    rules: {
      canDrop: (targetNode) => {
        const ignoreTargets = [
          getEditorName({ RowEditor }),
          getEditorName({ RootEditor })
        ];
        return !ignoreTargets.includes(targetNode.data.name);
      }
    }
  },
  {
    style: { cursor: "pointer" }
  }
);

export { PageFlipEditor };
