// @tempi/core-elements/dist/components/page-flip/setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var coreRenderer = require('@tempi/core-renderer');
var React = require('react');
var propKey = require('../../constants/propKey.js');

const PageFlipSetting = () => {
  const [checkedDrawShadow, setCheckedDrawShadow] = React.useState(false);
  const handleChangeDrawShadow = (checked) => {
    setCheckedDrawShadow(checked);
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.AlignmentConfig, {
        propKey: propKey.PROP_KEY.Align,
        buttonGroupProps: {
          listAlignment: ["left", "center", "right"]
        },
        formItemProps: {
          label: i18n.t("C\u0103n ch\u1EC9nh")
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
        formItemProps: {
          label: i18n.t("Chi\u1EC1u r\u1ED9ng")
        },
        unitProps: { defaultValue: coreRenderer.UnitEnum.pixel },
        propKey: propKey.PROP_KEY.Width
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
        formItemProps: {
          label: i18n.t("Chi\u1EC1u cao")
        },
        unitProps: { defaultValue: coreRenderer.UnitEnum.pixel },
        propKey: propKey.PROP_KEY.Height
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchToShow, {
        show: checkedDrawShadow,
        setShow: handleChangeDrawShadow,
        formItemProps: {
          label: i18n.t("\u0110\u1ED5 b\xF3ng")
        },
        children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
          formItemProps: {
            label: i18n.t("\u0110\u1ED9 m\u1EDD b\xF3ng l\u1EDBn nh\u1EA5t")
          },
          propKey: "MaxShadowOpacity",
          unitProps: { defaultValue: coreRenderer.UnitEnum.pixel }
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
        propKey: "flippingTime",
        formItemProps: {
          label: i18n.t("T\u1ED1c \u0111\u1ED9 l\u1EADt (ms)")
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
        propKey: "showCover",
        formItemProps: {
          label: i18n.t("Hi\u1EC3n th\u1ECB b\xECa")
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
        propKey: "mobileScrollSupport",
        formItemProps: {
          label: i18n.t("H\u1ED7 tr\u1EE3 scroll tr\xEAn thi\u1EBFt b\u1ECB di d\u1ED9ng")
        }
      })
    ]
  });
};

exports.PageFlipSetting = PageFlipSetting;
