// @tempi/core-elements/dist/components/page-flip/setting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { AlignmentConfig, InputNumberConfig, SwitchToShow, SwitchConfig } from '@tempi/core-editor';
import { UnitEnum } from '@tempi/core-renderer';
import { useState } from 'react';
import { PROP_KEY } from '../../constants/propKey.es.js';

const PageFlipSetting = () => {
  const [checkedDrawShadow, setCheckedDrawShadow] = useState(false);
  const handleChangeDrawShadow = (checked) => {
    setCheckedDrawShadow(checked);
  };
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(AlignmentConfig, {
        propKey: PROP_KEY.Align,
        buttonGroupProps: {
          listAlignment: ["left", "center", "right"]
        },
        formItemProps: {
          label: t("C\u0103n ch\u1EC9nh")
        }
      }),
      /* @__PURE__ */ jsx(InputNumberConfig, {
        formItemProps: {
          label: t("Chi\u1EC1u r\u1ED9ng")
        },
        unitProps: { defaultValue: UnitEnum.pixel },
        propKey: PROP_KEY.Width
      }),
      /* @__PURE__ */ jsx(InputNumberConfig, {
        formItemProps: {
          label: t("Chi\u1EC1u cao")
        },
        unitProps: { defaultValue: UnitEnum.pixel },
        propKey: PROP_KEY.Height
      }),
      /* @__PURE__ */ jsx(SwitchToShow, {
        show: checkedDrawShadow,
        setShow: handleChangeDrawShadow,
        formItemProps: {
          label: t("\u0110\u1ED5 b\xF3ng")
        },
        children: /* @__PURE__ */ jsx(InputNumberConfig, {
          formItemProps: {
            label: t("\u0110\u1ED9 m\u1EDD b\xF3ng l\u1EDBn nh\u1EA5t")
          },
          propKey: "MaxShadowOpacity",
          unitProps: { defaultValue: UnitEnum.pixel }
        })
      }),
      /* @__PURE__ */ jsx(InputNumberConfig, {
        propKey: "flippingTime",
        formItemProps: {
          label: t("T\u1ED1c \u0111\u1ED9 l\u1EADt (ms)")
        }
      }),
      /* @__PURE__ */ jsx(SwitchConfig, {
        propKey: "showCover",
        formItemProps: {
          label: t("Hi\u1EC3n th\u1ECB b\xECa")
        }
      }),
      /* @__PURE__ */ jsx(SwitchConfig, {
        propKey: "mobileScrollSupport",
        formItemProps: {
          label: t("H\u1ED7 tr\u1EE3 scroll tr\xEAn thi\u1EBFt b\u1ECB di d\u1ED9ng")
        }
      })
    ]
  });
};

export { PageFlipSetting };
