// @tempi/core-elements/dist/components/gallery/constants.es.js
const images = [
  {
    src: "https://lh3.googleusercontent.com/S5XWZViVkkH_toNSE6PQwaBR-Gx5vMuoT5AM8YYcMLMgBgD-SXBjzeEXPGgKRQxn7bjjCITOrLdgwpbhGX5NRnogwdl9tq8"
  },
  {
    src: "https://lh3.googleusercontent.com/1zB-8ZR8g5FdDgF_v4LdlFH0dI8sQFRhmYGIbM0CZ9zfL1U6w4FegeiJl9bwi5aUuA4Qh1BnuO35cL3xud4f2vOIjQa4fUG6"
  },
  {
    src: "https://lh3.googleusercontent.com/1fBqf-h09_8ykO6ZHbwhHyvszBnwxOgTOHwwZvQYmHNL-bwg-IAV0kf5xEp_2Z8kNiWEBYvKfhoRMWzmBeo83vgBm7FkIOLU"
  },
  {
    src: "https://lh3.googleusercontent.com/BZ7vM54nVgQA_N3G4mm6kn6Zxs4bWHG0oXKL5ZADOgy_eec0r2zBoSIZbxHXgil9R9qW36DGa8tkh94kwqUv0L_siFD9_R2X"
  },
  {
    src: "https://lh3.googleusercontent.com/eXIcx4DFKl_itUisfZJNdazzMmmb--hLjWn5E-rupuGC74-5LcLSgcZaufyPExx4DQTF4OA-2ZHJutRkZUx_Bjyfu_IKF1HCMA"
  },
  {
    src: "https://lh3.googleusercontent.com/zbcFIq-E8d9Jq6mSU1FHw9nFRmccL-3tWtCsNE218iwLRSMMmrxwK15juu9FOhczJ3aZ0OlZPEY3NUunpgZtwEB_2v3CnHs"
  },
  {
    src: "https://lh3.googleusercontent.com/HNGuPzB4HL2bSuyZBBZ1OE7fb5luDa541EbcTWuVa_zBXomqgtzFNGfyS4Drt4Lqk9lNT3PHG0qEdbO-51j5afy5wo2xBcE"
  },
  {
    src: "https://lh3.googleusercontent.com/5J-0315733Zt-sVBldmAkaICrp7E0-djemAfBz7csyaa0K1rwu4cm31X9y6Es5G0_HD0I2Z73KFJTMypp8bMws1P-uWXfFJX"
  }
];
const DEFAULT_GALLERY = {
  images,
  desktop: {
    columnCount: 4,
    gap: 20,
    layout: "masonry",
    zoomPreview: true
  }
};

export { DEFAULT_GALLERY };
