// @tempi/core-elements/dist/components/gallery/selector.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var coreEditor = require('@tempi/core-editor');
var constants = require('./constants.js');
var quickSetting = require('./quick-setting.js');
var setting = require('./setting.js');
var ui = require('./ui.js');
var selector = require('../section/selector.js');
var selector$1 = require('../root/selector.js');

const GalleryEditor = coreEditor.withSelector(
  ui.Gallery,
  {
    displayName: "Th\u01B0 vi\u1EC7n \u1EA3nh",
    tag: ui.Gallery.displayName.toLowerCase(),
    advanceAttributes: coreEditor.AdvanceAttributes,
    quickSetting: quickSetting.GalleryQuickSetting,
    customAttributes: setting.GallerySetting,
    props: constants.DEFAULT_GALLERY,
    rules: {
      canDrop: (targetNode) => {
        const ignoreTargets = [
          coreEditor.getEditorName({ RowEditor: selector.RowEditor }),
          coreEditor.getEditorName({ RootEditor: selector$1.RootEditor })
        ];
        return !ignoreTargets.includes(targetNode.data.name);
      }
    }
  },
  {
    style: { cursor: "pointer" }
  }
);

exports.GalleryEditor = GalleryEditor;
