// @tempi/core-elements/dist/components/gallery/selector.es.js
import { withSelector, AdvanceAttributes, getEditorName } from '@tempi/core-editor';
import { DEFAULT_GALLERY } from './constants.es.js';
import { GalleryQuickSetting } from './quick-setting.es.js';
import { GallerySetting } from './setting.es.js';
import { Gallery } from './ui.es.js';
import { RowEditor } from '../section/selector.es.js';
import { RootEditor } from '../root/selector.es.js';

const GalleryEditor = withSelector(
  Gallery,
  {
    displayName: "Th\u01B0 vi\u1EC7n \u1EA3nh",
    tag: Gallery.displayName.toLowerCase(),
    advanceAttributes: AdvanceAttributes,
    quickSetting: GalleryQuickSetting,
    customAttributes: GallerySetting,
    props: DEFAULT_GALLERY,
    rules: {
      canDrop: (targetNode) => {
        const ignoreTargets = [
          getEditorName({ RowEditor }),
          getEditorName({ RootEditor })
        ];
        return !ignoreTargets.includes(targetNode.data.name);
      }
    }
  },
  {
    style: { cursor: "pointer" }
  }
);

export { GalleryEditor };
