// @tempi/core-elements/dist/components/lucky-draw/v2/core/Wheel/PercentageWarning.es.js
import { jsx, jsxs } from 'react/jsx-runtime';
import { t } from '../../../../../i18n.es.js';
import styled from '@emotion/styled';
import { Icon } from '@tempi/icons';

const PercentageWarning = ({ totalPercentage = 100 }) => {
  if (totalPercentage === 100)
    return null;
  return /* @__PURE__ */ jsx(Wrapper, {
    children: /* @__PURE__ */ jsxs(Content, {
      children: [
        /* @__PURE__ */ jsx(Icon, {
          name: "InforOutlined",
          style: { fontSize: 40 }
        }),
        /* @__PURE__ */ jsx("div", {
          children: t("T\u1ED5ng x\xE1c xu\u1EA5t r\u01A1i qu\xE0 ph\u1EA3i b\u1EB1ng 100%.")
        }),
        /* @__PURE__ */ jsxs("div", {
          children: [
            t("T\u1ED5ng x\xE1c xu\u1EA5t hi\u1EC7n t\u1EA1i:"),
            totalPercentage,
            "%"
          ]
        })
      ]
    })
  });
};
const Wrapper = styled.div`
  position: absolute;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: grid;
  place-items: center;
`;
const Content = styled.div`
  color: #e6983a;
  width: 304px;
  background-color: #fffbf5;
  text-align: center;
  border-radius: 8px;
  padding: 8px 12px;
`;

export { PercentageWarning };
