// @tempi/core-elements/dist/components/lucky-draw/v2/core/Wheel/SpinBorder.es.js
import { jsx } from 'react/jsx-runtime';
import styled from '@emotion/styled';
import { getImageUrlFromGoogleUserContent } from '@tempi/core-renderer';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const SpinBorder = ({ image }) => {
  const { src, googleImageParams } = image || {};
  if (!src)
    return null;
  return /* @__PURE__ */ jsx(StyledBorderImage, __spreadProps(__spreadValues({
    alt: ""
  }, image), {
    src: getImageUrlFromGoogleUserContent(src, googleImageParams),
    loading: "lazy"
  }));
};
const StyledBorderImage = styled.img`
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
`;

export { SpinBorder };
