// @tempi/core-elements/dist/components/button/v2/configs/ButtonColorConfig.es.js
import { jsx, jsxs } from 'react/jsx-runtime';
import { t } from '../../../../i18n.es.js';
import { Color16Regular } from '@fluentui/react-icons';
import { SubQuickSetting, QuickSettingButton, ColorPickerConfig, BorderConfig } from '@tempi/core-editor';
import { Tabs } from 'antd';
import { PROP_KEY } from '../../../../constants/propKey.es.js';
import { QuickSettingTabWrapper } from '../../../../core/QuickSettingTabWrapper/QuickSettingTabWrapper.es.js';
import { ShadowConfig } from '../../../../configs/ShadowConfig/ShadowConfig.es.js';

const ButtonColorConfig = ({
  propKey,
  separateDevice
} = {}) => {
  const buttonStyleKey = PROP_KEY.joinKey(propKey, PROP_KEY.ButtonStyle);
  const content = () => /* @__PURE__ */ jsxs(QuickSettingTabWrapper, {
    children: [
      /* @__PURE__ */ jsxs(Tabs.TabPane, {
        tab: t("N\u1EC1n"),
        children: [
          /* @__PURE__ */ jsx(ColorPickerConfig, {
            propKey: PROP_KEY.joinKey(buttonStyleKey, PROP_KEY.BackgroundColor),
            formItemProps: {
              label: t("M\xE0u n\u1EC1n")
            },
            gradientPicker: true,
            separateDevice
          }),
          /* @__PURE__ */ jsx(ColorPickerConfig, {
            formItemProps: {
              label: t("M\xE0u khi di chu\u1ED9t")
            },
            propKey: PROP_KEY.joinKey(
              buttonStyleKey,
              PROP_KEY.HoverBackgroundColor
            ),
            gradientPicker: true,
            separateDevice
          })
        ]
      }, "1"),
      /* @__PURE__ */ jsxs(Tabs.TabPane, {
        tab: t("Ch\u1EEF"),
        children: [
          /* @__PURE__ */ jsx(ColorPickerConfig, {
            formItemProps: {
              label: t("M\xE0u ch\u1EEF")
            },
            propKey: PROP_KEY.joinKey(buttonStyleKey, PROP_KEY.Color),
            separateDevice
          }),
          /* @__PURE__ */ jsx(ColorPickerConfig, {
            formItemProps: {
              label: t("M\xE0u khi di chu\u1ED9t")
            },
            propKey: PROP_KEY.joinKey(buttonStyleKey, PROP_KEY.HoverColor),
            separateDevice
          })
        ]
      }, "2"),
      /* @__PURE__ */ jsx(Tabs.TabPane, {
        tab: t("Vi\u1EC1n"),
        children: /* @__PURE__ */ jsx(BorderConfig, {
          separateDevice,
          propKey: buttonStyleKey
        })
      }, "3"),
      /* @__PURE__ */ jsx(Tabs.TabPane, {
        tab: t("B\xF3ng"),
        children: /* @__PURE__ */ jsx(ShadowConfig, {
          propKey: PROP_KEY.joinKey(buttonStyleKey, PROP_KEY.Shadow),
          separateDevice
        })
      }, "4")
    ]
  });
  return /* @__PURE__ */ jsx(SubQuickSetting, {
    content,
    children: /* @__PURE__ */ jsx(QuickSettingButton, {
      tooltip: t("M\xE0u kh\u1ED1i"),
      children: /* @__PURE__ */ jsx(Color16Regular, {})
    })
  });
};

export { ButtonColorConfig };
