// @tempi/core-elements/dist/components/button/v2/configs/ButtonConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../../i18n.js');
var constants = require('../constants.js');
var styled = require('@emotion/styled');
var coreRenderer = require('@tempi/core-renderer');
var coreEditor = require('@tempi/core-editor');
var ui = require('../ui.js');
var propKey = require('../../../../constants/propKey.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const {
  primary: primaryStyle,
  primaryRound: primaryRoundStyle,
  outline: outlineStyle,
  outlineRound: outlineRoundStyle,
  link: linkStyle
} = constants.DEFAULT_BUTTON_STYLE;
const ButtonConfig = ({
  propKey: propKey$1,
  separateDevice
} = {}) => {
  const [buttonStyle, setButtonStyle] = coreEditor.usePropSeparateDevice(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.ButtonStyle),
    separateDevice
  );
  const widthPropKey = propKey.PROP_KEY.joinKey(
    propKey$1,
    propKey.PROP_KEY.ButtonStyle,
    propKey.PROP_KEY.Width
  );
  const heightPropKey = propKey.PROP_KEY.joinKey(
    propKey$1,
    propKey.PROP_KEY.ButtonStyle,
    propKey.PROP_KEY.Height
  );
  const chooseButttonStyle = (style) => {
    setButtonStyle(__spreadValues(__spreadValues({}, buttonStyle), style));
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
        label: i18n.t("Ki\u1EC3u"),
        layout: "vertical",
        children: /* @__PURE__ */ jsxRuntime.jsxs(ButtonStyleWrapper, {
          children: [
            /* @__PURE__ */ jsxRuntime.jsx(ui.ButtonV2, __spreadProps(__spreadValues({
              onClick: () => chooseButttonStyle(__spreadProps(__spreadValues({}, primaryStyle), {
                background: (buttonStyle == null ? void 0 : buttonStyle.background) || (buttonStyle == null ? void 0 : buttonStyle.backgroundColor) || primaryStyle.background
              }))
            }, constants.DEFAULT_BUTTON.primary), {
              commonStyle: {
                justifyContent: "center"
              },
              text: "Primary"
            })),
            /* @__PURE__ */ jsxRuntime.jsx(ui.ButtonV2, __spreadProps(__spreadValues({
              onClick: () => chooseButttonStyle(__spreadProps(__spreadValues({}, primaryRoundStyle), {
                background: (buttonStyle == null ? void 0 : buttonStyle.background) || (buttonStyle == null ? void 0 : buttonStyle.backgroundColor) || primaryStyle.background
              }))
            }, constants.DEFAULT_BUTTON.primaryRound), {
              commonStyle: {
                justifyContent: "center"
              },
              text: "Primary Round"
            })),
            /* @__PURE__ */ jsxRuntime.jsx(ui.ButtonV2, __spreadProps(__spreadValues({
              onClick: () => chooseButttonStyle(__spreadProps(__spreadValues({}, outlineStyle), {
                borderWidth: (buttonStyle == null ? void 0 : buttonStyle.borderWidth) || outlineStyle.borderWidth,
                borderColor: (buttonStyle == null ? void 0 : buttonStyle.borderColor) || outlineStyle.borderColor,
                borderStyle: (buttonStyle == null ? void 0 : buttonStyle.borderStyle) || outlineStyle.borderStyle
              }))
            }, constants.DEFAULT_BUTTON.outline), {
              commonStyle: {
                justifyContent: "center"
              },
              text: "Outline"
            })),
            /* @__PURE__ */ jsxRuntime.jsx(ui.ButtonV2, __spreadProps(__spreadValues({
              onClick: () => chooseButttonStyle(__spreadProps(__spreadValues({}, outlineRoundStyle), {
                borderWidth: (buttonStyle == null ? void 0 : buttonStyle.borderWidth) || outlineRoundStyle.borderWidth,
                borderColor: (buttonStyle == null ? void 0 : buttonStyle.borderColor) || outlineRoundStyle.borderColor,
                borderStyle: (buttonStyle == null ? void 0 : buttonStyle.borderStyle) || outlineRoundStyle.borderStyle
              }))
            }, constants.DEFAULT_BUTTON.outlineRound), {
              commonStyle: {
                justifyContent: "center"
              },
              text: "Outline Round"
            })),
            /* @__PURE__ */ jsxRuntime.jsx(ui.ButtonV2, __spreadProps(__spreadValues({
              onClick: () => chooseButttonStyle(linkStyle)
            }, constants.DEFAULT_BUTTON.link), {
              commonStyle: {
                justifyContent: "center"
              },
              text: "Link"
            }))
          ]
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
        formItemProps: {
          label: i18n.t("Chi\u1EC1u r\u1ED9ng")
        },
        unitProps: { defaultValue: coreRenderer.UnitEnum.pixel },
        propKey: widthPropKey,
        separateDevice
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
        formItemProps: {
          label: i18n.t("Chi\u1EC1u cao")
        },
        unitProps: { defaultValue: coreRenderer.UnitEnum.pixel },
        propKey: heightPropKey,
        separateDevice
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
        formItemProps: {
          label: i18n.t("Bo g\xF3c")
        },
        unitProps: { defaultValue: coreRenderer.UnitEnum.pixel },
        propKey: propKey.PROP_KEY.joinKey(
          propKey$1,
          propKey.PROP_KEY.ButtonStyle,
          propKey.PROP_KEY.BorderRadius
        ),
        separateDevice
      })
    ]
  });
};
const ButtonStyleWrapper = styled__default["default"].div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 8px;
`;

exports.ButtonConfig = ButtonConfig;
