// @tempi/core-elements/dist/components/global-block/header-v2/style-setting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../../i18n.es.js';
import { ColorPickerConfig } from '@tempi/core-editor';
import { PROP_KEY } from '../../../constants/propKey.es.js';

const HeaderV2StyleSetting = () => {
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(ColorPickerConfig, {
        formItemProps: {
          label: t("M\xE0u n\u1EC1n")
        },
        propKey: PROP_KEY.joinKey(
          PROP_KEY.CommonStyle,
          PROP_KEY.BackgroundColor
        ),
        separateDevice: true,
        gradientPicker: true
      }),
      /* @__PURE__ */ jsx(ColorPickerConfig, {
        formItemProps: {
          label: t("M\xE0u menu")
        },
        propKey: PROP_KEY.joinKey(
          PROP_KEY.MenuItemConfig,
          PROP_KEY.CommonStyle,
          PROP_KEY.Color
        ),
        separateDevice: true
      }),
      /* @__PURE__ */ jsx(ColorPickerConfig, {
        formItemProps: {
          label: t("M\xE0u hover menu")
        },
        propKey: PROP_KEY.joinKey(
          PROP_KEY.MenuItemConfig,
          PROP_KEY.CommonStyle,
          PROP_KEY.HoverColor
        ),
        separateDevice: true
      })
    ]
  });
};

export { HeaderV2StyleSetting };
