// @tempi/core-elements/dist/components/global-block/header-v2/style-setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var propKey = require('../../../constants/propKey.js');

const HeaderV2StyleSetting = () => {
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
        formItemProps: {
          label: i18n.t("M\xE0u n\u1EC1n")
        },
        propKey: propKey.PROP_KEY.joinKey(
          propKey.PROP_KEY.CommonStyle,
          propKey.PROP_KEY.BackgroundColor
        ),
        separateDevice: true,
        gradientPicker: true
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
        formItemProps: {
          label: i18n.t("M\xE0u menu")
        },
        propKey: propKey.PROP_KEY.joinKey(
          propKey.PROP_KEY.MenuItemConfig,
          propKey.PROP_KEY.CommonStyle,
          propKey.PROP_KEY.Color
        ),
        separateDevice: true
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
        formItemProps: {
          label: i18n.t("M\xE0u hover menu")
        },
        propKey: propKey.PROP_KEY.joinKey(
          propKey.PROP_KEY.MenuItemConfig,
          propKey.PROP_KEY.CommonStyle,
          propKey.PROP_KEY.HoverColor
        ),
        separateDevice: true
      })
    ]
  });
};

exports.HeaderV2StyleSetting = HeaderV2StyleSetting;
