// @tempi/core-editor/dist/configs/TekoTrackingConfig/TekoTrackingConfig.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { PROP_KEY } from '../../constants/propKey.es.js';
import { InputConfig } from '../InputConfig/InputConfig.es.js';

const TekoTrackingConfig = ({
  propKey = PROP_KEY.Event,
  separateDevice
}) => {
  const regionName = PROP_KEY.joinKey(
    propKey,
    PROP_KEY.TekoTrackingEvent,
    PROP_KEY.RegionName
  );
  const contentName = PROP_KEY.joinKey(
    propKey,
    PROP_KEY.TekoTrackingEvent,
    PROP_KEY.ContentName
  );
  const target = PROP_KEY.joinKey(
    propKey,
    PROP_KEY.TekoTrackingEvent,
    PROP_KEY.Target
  );
  const payload = PROP_KEY.joinKey(
    propKey,
    PROP_KEY.TekoTrackingEvent,
    PROP_KEY.Payload
  );
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(InputConfig, {
        separateDevice,
        propKey: regionName,
        formItemProps: {
          label: "Region name",
          style: {
            display: "flex",
            justifyContent: "space-between",
            alignItems: "center",
            marginBottom: "0.5rem"
          }
        },
        inputProps: {
          placeholder: "Region name",
          style: { width: "180px" }
        }
      }),
      /* @__PURE__ */ jsx(InputConfig, {
        separateDevice,
        propKey: contentName,
        formItemProps: {
          label: "Content name",
          style: {
            display: "flex",
            justifyContent: "space-between",
            alignItems: "center",
            marginBottom: "0.5rem"
          }
        },
        inputProps: {
          placeholder: "Content name",
          style: { width: "180px" }
        }
      }),
      /* @__PURE__ */ jsx(InputConfig, {
        separateDevice,
        propKey: target,
        formItemProps: {
          label: "Target",
          style: {
            display: "flex",
            justifyContent: "space-between",
            alignItems: "center",
            marginBottom: "0.5rem"
          }
        },
        inputProps: {
          placeholder: "Target",
          style: { width: "180px" }
        }
      }),
      /* @__PURE__ */ jsx(InputConfig, {
        separateDevice,
        propKey: payload,
        formItemProps: {
          label: "Payload",
          style: {
            display: "flex",
            justifyContent: "space-between",
            alignItems: "center",
            marginBottom: "0.5rem"
          }
        },
        inputProps: {
          placeholder: "Payload",
          style: { width: "180px" }
        }
      })
    ]
  });
};

export { TekoTrackingConfig };
