// @tempi/core-elements/dist/components/popup/v1/configs/ConfigPopup.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../../../i18n.es.js';
import { useEditor, useProp, InputConfig, SwitchConfig, ColorPickerConfig, InputNumberConfig, SelectConfig, ResponsiveAttribute } from '@tempi/core-editor';
import { UnitEnum } from '@tempi/core-renderer';
import { useMemo } from 'react';
import { PROP_KEY } from '../../../../constants/propKey.es.js';

const ConfigPopup = () => {
  const { nodes } = useEditor((state) => {
    return { nodes: state.nodes };
  });
  const [position] = useProp(PROP_KEY.Position);
  const [showCloseIcon] = useProp(PROP_KEY.ShowCloseIcon);
  const [displayOnVisitPage] = useProp("displayOnVisitPage");
  const topKey = PROP_KEY.joinKey(PROP_KEY.CommonStyle, "top");
  const leftKey = PROP_KEY.joinKey(PROP_KEY.CommonStyle, "left");
  const rightKey = PROP_KEY.joinKey(PROP_KEY.CommonStyle, "right");
  const bottomKey = PROP_KEY.joinKey(PROP_KEY.CommonStyle, "bottom");
  const positionOptions = [
    { label: t("M\u1EB7c \u0111\u1ECBnh"), value: "default" },
    { label: t("Tr\xEAn c\xF9ng tr\xE1i"), value: "topLeft" },
    { label: t("Tr\xEAn c\xF9ng gi\u1EEFa"), value: "topCenter" },
    { label: t("Tr\xEAn c\xF9ng ph\u1EA3i"), value: "topRight" },
    { label: t("Gi\u1EEFa tr\xE1i"), value: "centerLeft" },
    { label: t("Gi\u1EEFa ph\u1EA3i"), value: "centerRight" },
    { label: t("D\u01B0\u1EDBi d\xF9ng tr\xE1i"), value: "bottomLeft" },
    { label: t("D\u01B0\u1EDBi c\xF9ng gi\u1EEFa"), value: "bottomCenter" },
    { label: t("D\u01B0\u1EDBi c\xF9ng ph\u1EA3i"), value: "bottomRight" }
  ];
  const elementList = useMemo(() => {
    return [
      ...new Set(Object.keys(nodes).map((key) => nodes[key].data.props.id))
    ].filter((option) => option).map((id) => ({ label: id, value: id }));
  }, []);
  const renderDistance = (position2) => {
    switch (position2) {
      case "topLeft":
        return /* @__PURE__ */ jsxs(Fragment, {
          children: [
            /* @__PURE__ */ jsx(InputNumberConfig, {
              formItemProps: {
                label: t("Kho\u1EA3ng c\xE1ch tr\xEAn")
              },
              propKey: topKey
            }),
            /* @__PURE__ */ jsx(InputNumberConfig, {
              formItemProps: {
                label: t("Kho\u1EA3ng c\xE1ch tr\xE1i")
              },
              propKey: leftKey
            })
          ]
        });
      case "topCenter":
        return /* @__PURE__ */ jsx(InputNumberConfig, {
          formItemProps: {
            label: t("Kho\u1EA3ng c\xE1ch tr\xEAn")
          },
          propKey: topKey
        });
      case "topRight":
        return /* @__PURE__ */ jsxs(Fragment, {
          children: [
            /* @__PURE__ */ jsx(InputNumberConfig, {
              formItemProps: {
                label: t("Kho\u1EA3ng c\xE1ch tr\xEAn")
              },
              propKey: topKey
            }),
            /* @__PURE__ */ jsx(InputNumberConfig, {
              formItemProps: {
                label: t("Kho\u1EA3ng c\xE1ch ph\u1EA3i")
              },
              propKey: rightKey
            })
          ]
        });
      case "centerLeft":
        return /* @__PURE__ */ jsx(InputNumberConfig, {
          formItemProps: {
            label: t("Kho\u1EA3ng c\xE1ch tr\xE1i")
          },
          propKey: leftKey
        });
      case "centerRight":
        return /* @__PURE__ */ jsx(InputNumberConfig, {
          formItemProps: {
            label: t("Kho\u1EA3ng c\xE1ch ph\u1EA3i")
          },
          propKey: rightKey
        });
      case "bottomLeft":
        return /* @__PURE__ */ jsxs(Fragment, {
          children: [
            /* @__PURE__ */ jsx(InputNumberConfig, {
              formItemProps: {
                label: t("Kho\u1EA3ng c\xE1ch d\u01B0\u1EDBi")
              },
              propKey: bottomKey
            }),
            /* @__PURE__ */ jsx(InputNumberConfig, {
              formItemProps: {
                label: t("Kho\u1EA3ng c\xE1ch tr\xE1i")
              },
              propKey: leftKey
            })
          ]
        });
      case "bottomCenter":
        return /* @__PURE__ */ jsx(InputNumberConfig, {
          formItemProps: {
            label: t("Kho\u1EA3ng c\xE1ch d\u01B0\u1EDBi")
          },
          propKey: bottomKey
        });
      case "bottomRight":
        return /* @__PURE__ */ jsxs(Fragment, {
          children: [
            /* @__PURE__ */ jsx(InputNumberConfig, {
              formItemProps: {
                label: t("Kho\u1EA3ng c\xE1ch d\u01B0\u1EDBi")
              },
              propKey: bottomKey
            }),
            /* @__PURE__ */ jsx(InputNumberConfig, {
              formItemProps: {
                label: t("Kho\u1EA3ng c\xE1ch ph\u1EA3i")
              },
              propKey: rightKey
            })
          ]
        });
    }
  };
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(InputConfig, {
        formItemProps: {
          label: t("T\xEAn")
        },
        propKey: "name"
      }),
      /* @__PURE__ */ jsx(SwitchConfig, {
        formItemProps: {
          label: t("Hi\u1EC3n th\u1ECB icon \u0111\xF3ng popup")
        },
        propKey: PROP_KEY.ShowCloseIcon
      }),
      showCloseIcon && /* @__PURE__ */ jsxs(Fragment, {
        children: [
          /* @__PURE__ */ jsx(ColorPickerConfig, {
            formItemProps: { label: t("M\xE0u s\u1EAFc icon") },
            propKey: PROP_KEY.joinKey(PROP_KEY.CloseIcon, PROP_KEY.Color)
          }),
          /* @__PURE__ */ jsx(InputNumberConfig, {
            formItemProps: {
              label: t("K\xEDch th\u01B0\u1EDBc icon")
            },
            propKey: PROP_KEY.joinKey(PROP_KEY.CloseIcon, PROP_KEY.Size),
            unitProps: { defaultValue: UnitEnum.pixel }
          })
        ]
      }),
      /* @__PURE__ */ jsx(SwitchConfig, {
        formItemProps: {
          label: t("Hi\u1EC7n khi v\xE0o trang")
        },
        propKey: "displayOnVisitPage"
      }),
      displayOnVisitPage && /* @__PURE__ */ jsx(InputNumberConfig, {
        formItemProps: {
          label: t("\u0110\u1ED9 tr\u1EC5 (s)")
        },
        propKey: "delay"
      }),
      /* @__PURE__ */ jsx(SwitchConfig, {
        formItemProps: {
          label: t("Hi\u1EC7n khi tho\xE1t trang")
        },
        propKey: "displayOnExitPage"
      }),
      /* @__PURE__ */ jsx(SelectConfig, {
        formItemProps: {
          label: t("Hi\u1EC7n khi cu\u1ED9n \u0111\u1EBFn")
        },
        options: elementList,
        propKey: "displayOnScrollTo"
      }),
      /* @__PURE__ */ jsx(SwitchConfig, {
        formItemProps: {
          label: t(
            "Hi\u1EC3n th\u1ECB m\u1ED9t l\u1EA7n tr\xEAn m\u1ED9t tab browser (ch\u1EC9 \xE1p d\u1EE5ng v\u1EDBi popup hi\u1EC7n khi v\xE0o/tho\xE1t trang)"
          )
        },
        propKey: "displayOncePerSession"
      }),
      /* @__PURE__ */ jsx(InputNumberConfig, {
        formItemProps: {
          label: t("Chi\u1EC1u r\u1ED9ng")
        },
        propKey: PROP_KEY.joinKey(PROP_KEY.CommonStyle, PROP_KEY.Width),
        unitProps: { defaultValue: UnitEnum.pixel }
      }),
      /* @__PURE__ */ jsx(InputNumberConfig, {
        formItemProps: {
          label: t("Chi\u1EC1u cao")
        },
        propKey: PROP_KEY.joinKey(PROP_KEY.CommonStyle, PROP_KEY.Height),
        unitProps: { defaultValue: UnitEnum.pixel }
      }),
      /* @__PURE__ */ jsx(ColorPickerConfig, {
        propKey: "maskStyle.backgroundColor",
        formItemProps: {
          label: t("M\xE0u Backdrop")
        }
      }),
      /* @__PURE__ */ jsx(SelectConfig, {
        options: positionOptions,
        propKey: PROP_KEY.Position,
        formItemProps: { label: t("V\u1ECB tr\xED") }
      }),
      renderDistance(position),
      /* @__PURE__ */ jsx(InputNumberConfig, {
        formItemProps: {
          label: "Z-Index"
        },
        propKey: PROP_KEY.ZIndex
      }),
      /* @__PURE__ */ jsx(ResponsiveAttribute, {})
    ]
  });
};

export { ConfigPopup };
