// @tempi/core-elements/dist/components/popup/v1/configs/ConfigPopup.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var coreRenderer = require('@tempi/core-renderer');
var React = require('react');
var propKey = require('../../../../constants/propKey.js');

const ConfigPopup = () => {
  const { nodes } = coreEditor.useEditor((state) => {
    return { nodes: state.nodes };
  });
  const [position] = coreEditor.useProp(propKey.PROP_KEY.Position);
  const [showCloseIcon] = coreEditor.useProp(propKey.PROP_KEY.ShowCloseIcon);
  const [displayOnVisitPage] = coreEditor.useProp("displayOnVisitPage");
  const topKey = propKey.PROP_KEY.joinKey(propKey.PROP_KEY.CommonStyle, "top");
  const leftKey = propKey.PROP_KEY.joinKey(propKey.PROP_KEY.CommonStyle, "left");
  const rightKey = propKey.PROP_KEY.joinKey(propKey.PROP_KEY.CommonStyle, "right");
  const bottomKey = propKey.PROP_KEY.joinKey(propKey.PROP_KEY.CommonStyle, "bottom");
  const positionOptions = [
    { label: i18n.t("M\u1EB7c \u0111\u1ECBnh"), value: "default" },
    { label: i18n.t("Tr\xEAn c\xF9ng tr\xE1i"), value: "topLeft" },
    { label: i18n.t("Tr\xEAn c\xF9ng gi\u1EEFa"), value: "topCenter" },
    { label: i18n.t("Tr\xEAn c\xF9ng ph\u1EA3i"), value: "topRight" },
    { label: i18n.t("Gi\u1EEFa tr\xE1i"), value: "centerLeft" },
    { label: i18n.t("Gi\u1EEFa ph\u1EA3i"), value: "centerRight" },
    { label: i18n.t("D\u01B0\u1EDBi d\xF9ng tr\xE1i"), value: "bottomLeft" },
    { label: i18n.t("D\u01B0\u1EDBi c\xF9ng gi\u1EEFa"), value: "bottomCenter" },
    { label: i18n.t("D\u01B0\u1EDBi c\xF9ng ph\u1EA3i"), value: "bottomRight" }
  ];
  const elementList = React.useMemo(() => {
    return [
      ...new Set(Object.keys(nodes).map((key) => nodes[key].data.props.id))
    ].filter((option) => option).map((id) => ({ label: id, value: id }));
  }, []);
  const renderDistance = (position2) => {
    switch (position2) {
      case "topLeft":
        return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
          children: [
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
              formItemProps: {
                label: i18n.t("Kho\u1EA3ng c\xE1ch tr\xEAn")
              },
              propKey: topKey
            }),
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
              formItemProps: {
                label: i18n.t("Kho\u1EA3ng c\xE1ch tr\xE1i")
              },
              propKey: leftKey
            })
          ]
        });
      case "topCenter":
        return /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
          formItemProps: {
            label: i18n.t("Kho\u1EA3ng c\xE1ch tr\xEAn")
          },
          propKey: topKey
        });
      case "topRight":
        return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
          children: [
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
              formItemProps: {
                label: i18n.t("Kho\u1EA3ng c\xE1ch tr\xEAn")
              },
              propKey: topKey
            }),
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
              formItemProps: {
                label: i18n.t("Kho\u1EA3ng c\xE1ch ph\u1EA3i")
              },
              propKey: rightKey
            })
          ]
        });
      case "centerLeft":
        return /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
          formItemProps: {
            label: i18n.t("Kho\u1EA3ng c\xE1ch tr\xE1i")
          },
          propKey: leftKey
        });
      case "centerRight":
        return /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
          formItemProps: {
            label: i18n.t("Kho\u1EA3ng c\xE1ch ph\u1EA3i")
          },
          propKey: rightKey
        });
      case "bottomLeft":
        return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
          children: [
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
              formItemProps: {
                label: i18n.t("Kho\u1EA3ng c\xE1ch d\u01B0\u1EDBi")
              },
              propKey: bottomKey
            }),
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
              formItemProps: {
                label: i18n.t("Kho\u1EA3ng c\xE1ch tr\xE1i")
              },
              propKey: leftKey
            })
          ]
        });
      case "bottomCenter":
        return /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
          formItemProps: {
            label: i18n.t("Kho\u1EA3ng c\xE1ch d\u01B0\u1EDBi")
          },
          propKey: bottomKey
        });
      case "bottomRight":
        return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
          children: [
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
              formItemProps: {
                label: i18n.t("Kho\u1EA3ng c\xE1ch d\u01B0\u1EDBi")
              },
              propKey: bottomKey
            }),
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
              formItemProps: {
                label: i18n.t("Kho\u1EA3ng c\xE1ch ph\u1EA3i")
              },
              propKey: rightKey
            })
          ]
        });
    }
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputConfig, {
        formItemProps: {
          label: i18n.t("T\xEAn")
        },
        propKey: "name"
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
        formItemProps: {
          label: i18n.t("Hi\u1EC3n th\u1ECB icon \u0111\xF3ng popup")
        },
        propKey: propKey.PROP_KEY.ShowCloseIcon
      }),
      showCloseIcon && /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
            formItemProps: { label: i18n.t("M\xE0u s\u1EAFc icon") },
            propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.CloseIcon, propKey.PROP_KEY.Color)
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
            formItemProps: {
              label: i18n.t("K\xEDch th\u01B0\u1EDBc icon")
            },
            propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.CloseIcon, propKey.PROP_KEY.Size),
            unitProps: { defaultValue: coreRenderer.UnitEnum.pixel }
          })
        ]
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
        formItemProps: {
          label: i18n.t("Hi\u1EC7n khi v\xE0o trang")
        },
        propKey: "displayOnVisitPage"
      }),
      displayOnVisitPage && /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
        formItemProps: {
          label: i18n.t("\u0110\u1ED9 tr\u1EC5 (s)")
        },
        propKey: "delay"
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
        formItemProps: {
          label: i18n.t("Hi\u1EC7n khi tho\xE1t trang")
        },
        propKey: "displayOnExitPage"
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
        formItemProps: {
          label: i18n.t("Hi\u1EC7n khi cu\u1ED9n \u0111\u1EBFn")
        },
        options: elementList,
        propKey: "displayOnScrollTo"
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
        formItemProps: {
          label: i18n.t(
            "Hi\u1EC3n th\u1ECB m\u1ED9t l\u1EA7n tr\xEAn m\u1ED9t tab browser (ch\u1EC9 \xE1p d\u1EE5ng v\u1EDBi popup hi\u1EC7n khi v\xE0o/tho\xE1t trang)"
          )
        },
        propKey: "displayOncePerSession"
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
        formItemProps: {
          label: i18n.t("Chi\u1EC1u r\u1ED9ng")
        },
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.CommonStyle, propKey.PROP_KEY.Width),
        unitProps: { defaultValue: coreRenderer.UnitEnum.pixel }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
        formItemProps: {
          label: i18n.t("Chi\u1EC1u cao")
        },
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.CommonStyle, propKey.PROP_KEY.Height),
        unitProps: { defaultValue: coreRenderer.UnitEnum.pixel }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
        propKey: "maskStyle.backgroundColor",
        formItemProps: {
          label: i18n.t("M\xE0u Backdrop")
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
        options: positionOptions,
        propKey: propKey.PROP_KEY.Position,
        formItemProps: { label: i18n.t("V\u1ECB tr\xED") }
      }),
      renderDistance(position),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
        formItemProps: {
          label: "Z-Index"
        },
        propKey: propKey.PROP_KEY.ZIndex
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ResponsiveAttribute, {})
    ]
  });
};

exports.ConfigPopup = ConfigPopup;
