// @tempi/core-editor/dist/configs/CustomRadioConfig/CustomRadioConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var styled = require('@emotion/styled');
var antd = require('antd');
var usePropSeparateDevice = require('../../hooks/usePropSeparateDevice.js');
var FormItem = require('../../components/FormItem/FormItem.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

const CustomRadioConfig = ({
  propKey,
  seperateDevice,
  formItemProps,
  options
}) => {
  const [value, setValue] = usePropSeparateDevice.usePropSeparateDevice(propKey, seperateDevice);
  return /* @__PURE__ */ jsxRuntime.jsx(FormItem.FormItem, {
    layout: "vertical",
    label: formItemProps == null ? void 0 : formItemProps.label,
    children: /* @__PURE__ */ jsxRuntime.jsx(antd.Row, {
      gutter: 16,
      children: options.map((option, index) => /* @__PURE__ */ jsxRuntime.jsxs(antd.Col, {
        span: Math.round(24 / options.length),
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(LayoutSelectButton, {
            block: true,
            type: value === option.value ? "primary" : "default",
            onClick: () => setValue(option.value),
            children: option.children
          }),
          !!(option == null ? void 0 : option.label) && /* @__PURE__ */ jsxRuntime.jsx("div", {
            style: { textAlign: "center" },
            children: option == null ? void 0 : option.label
          })
        ]
      }, index))
    })
  });
};
const LayoutSelectButton = styled__default["default"](antd.Button)(
  ({ type }) => `
    height: unset;
    margin-bottom: 0.5em;
    padding-top: 1em;
    padding-bottom: 1em;
  
    ${type === "default" ? "color: #82868B;" : ""}
  `
);

exports.CustomRadioConfig = CustomRadioConfig;
