// @tempi/core-elements/dist/quick-configs/MoveLeft/MoveLeft.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var reactIcons = require('@fluentui/react-icons');
var coreEditor = require('@tempi/core-editor');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const MoveLeft = ({ attrTracking = {} }) => {
  const { actions, getSiblings } = coreEditor.useEditor();
  const { id, parentId } = coreEditor.useNode((node) => ({
    parentId: node.data.parent
  }));
  const index = getSiblings(id).indexOf(id);
  const onMoveLeft = () => {
    if (index > 0) {
      actions.move(id, parentId, index - 1);
      actions.selectNode();
    }
  };
  return /* @__PURE__ */ jsxRuntime.jsx(coreEditor.QuickSettingButton, __spreadProps(__spreadValues({
    tooltip: i18n.t("D\u1ECBch ph\u1EA7n t\u1EED sang tr\xE1i"),
    onClick: onMoveLeft
  }, attrTracking), {
    children: /* @__PURE__ */ jsxRuntime.jsx(reactIcons.ArrowLeftRegular, {
      fontSize: 16
    })
  }));
};

exports.MoveLeft = MoveLeft;
