// @tempi/core-elements/dist/components/product-list/components/ProductCard/ProductCard.stories.js
import { jsx as _jsx } from "react/jsx-runtime";
import { ProductCard } from './ProductCard';
const defaultProducts = [
    {
        product: {
            name: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry',
            price: 500000,
            image: {
                src: 'https://vcdn1-dulich.vnecdn.net/2021/07/16/8-1626444967.jpg?w=1200&h=0&q=100&dpr=1&fit=crop&s=GfgGn4dNuKZexy1BGkAUNA',
            },
            link: 'https://phongvu.vn/msi-gaming-gf63-thin-11sc-665vn-i5-11400h--s221101658',
        },
        alignment: 'left',
    },
    {
        product: {
            name: 'Product 1',
            price: 20000,
            discount: 30,
            image: {
                src: 'https://lh3.googleusercontent.com/fKUsNfKwAQggLUyMJ2TpHJ6IT3YI2suMdruCmrVlpbgPy4DErIw60__FBElMmKgtkCAYWFi4wAgnEaEwQeQ',
            },
            link: 'https://phongvu.vn/msi-gaming-gf63-thin-11sc-665vn-i5-11400h--s221101658',
        },
        alignment: 'left',
    },
    {
        product: {
            name: 'Product 1',
            price: 20000,
            discount: 30,
            image: {
                src: 'https://lh3.googleusercontent.com/HDchawn0E7MWVK60_1CVQafezWgK5sqhYFlkCEOkKksTIAehk6BYW2P55qxtwy8CO3WBVI558YrEs8cGoqAKKhNuARtZ168',
            },
        },
        alignment: 'left',
    },
    {
        product: {
            name: 'Product 1',
            price: 20000,
            discount: 30,
            image: {
                src: 'https://lh3.googleusercontent.com/XY6Olaf3hkIL7V42TzBDTuYCLT4VQ6CWDIAGuRmLSWU_JZTywcdHaP5_BVTlo10y9RUh6JOdvDRE27wJrIO36HovMdNAO7C81Q',
            },
        },
        alignment: 'left',
    },
    {
        product: {
            name: 'Product 1',
            price: 20000,
            discount: 30,
            image: {
                src: 'https://lh3.googleusercontent.com/fKUsNfKwAQggLUyMJ2TpHJ6IT3YI2suMdruCmrVlpbgPy4DErIw60__FBElMmKgtkCAYWFi4wAgnEaEwQeQ',
            },
        },
        alignment: 'left',
    },
    {
        product: {
            name: 'Product 1',
            price: 20000,
            discount: 30,
            image: {
                src: 'https://lh3.googleusercontent.com/fKUsNfKwAQggLUyMJ2TpHJ6IT3YI2suMdruCmrVlpbgPy4DErIw60__FBElMmKgtkCAYWFi4wAgnEaEwQeQ',
            },
        },
        alignment: 'left',
    },
];
export default {
    title: 'CoreElements/ProductCard',
};
const Template = () => {
    return (_jsx("div", Object.assign({ style: {
            width: 768,
            height: 500,
            display: 'grid',
            alignSelf: 'stretch',
            gap: 16,
            gridTemplateColumns: 'repeat(4, 1fr)',
            padding: 4,
        } }, { children: defaultProducts.map((el, index) => (_jsx("div", { children: _jsx(ProductCard, Object.assign({}, el)) }, index))) })));
};
export const Default = Template.bind({});
