// @tempi/core-elements/dist/quick-configs/typography/TextColorQuickSettingButton.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var styled = require('@emotion/styled');
var coreEditor = require('@tempi/core-editor');
var React = require('react');
var getEditorAttributeValue = require('../../utils/getEditorAttributeValue.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

const TextColorQuickSettingButton = ({ editor, styleKey, onChange }) => {
  const [visible, setVisible] = React.useState(false);
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.QuickSettingButton, {
        tooltip: i18n.t("M\xE0u ch\u1EEF"),
        onClick: () => {
          setVisible(true);
        },
        children: /* @__PURE__ */ jsxRuntime.jsx(ColorPicker, {
          color: getEditorAttributeValue.getEditorAttributeValue(editor, styleKey)
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerModalV2, {
        visible,
        setVisible,
        color: getEditorAttributeValue.getEditorAttributeValue(editor, styleKey),
        onChange
      })
    ]
  });
};
const ColorPicker = styled__default["default"].div(
  ({ color }) => `
  width: 16px;
  height: 16px;
  border-radius: 4px;
  border: 1px solid #7A7A7A;

  ${color ? `background-color: ${color};` : ""}
`
);

exports.TextColorQuickSettingButton = TextColorQuickSettingButton;
