// @tempi/core-elements/dist/quick-configs/typography/TextColorQuickSettingButton.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import styled from '@emotion/styled';
import { QuickSettingButton, ColorPickerModalV2 } from '@tempi/core-editor';
import { useState } from 'react';
import { getEditorAttributeValue } from '../../utils/getEditorAttributeValue.es.js';

const TextColorQuickSettingButton = ({ editor, styleKey, onChange }) => {
  const [visible, setVisible] = useState(false);
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(QuickSettingButton, {
        tooltip: t("M\xE0u ch\u1EEF"),
        onClick: () => {
          setVisible(true);
        },
        children: /* @__PURE__ */ jsx(ColorPicker, {
          color: getEditorAttributeValue(editor, styleKey)
        })
      }),
      /* @__PURE__ */ jsx(ColorPickerModalV2, {
        visible,
        setVisible,
        color: getEditorAttributeValue(editor, styleKey),
        onChange
      })
    ]
  });
};
const ColorPicker = styled.div(
  ({ color }) => `
  width: 16px;
  height: 16px;
  border-radius: 4px;
  border: 1px solid #7A7A7A;

  ${color ? `background-color: ${color};` : ""}
`
);

export { TextColorQuickSettingButton };
