// @tempi/core-elements/dist/quick-configs/typography/TextLinkQuickSetting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { useState, useMemo, useEffect } from 'react';
import { useBuilder, QuickSettingButton, SelectConfig, TekoTrackingConfig, PROP_KEY, SwitchConfig } from '@tempi/core-editor';
import { Link16Regular } from '@fluentui/react-icons';
import { TextInputConfig } from '../../configs/TextConfig/TextInputConfig.es.js';
import { Modal, Button, Tabs } from 'antd';
import { useFrame } from 'react-frame-component';

const TextLinkQuickSetting = ({
  editor
}) => {
  const [visible, setVisible] = useState(false);
  const [eventType, setEventType] = useState();
  const { enableTekoTracking } = useBuilder();
  const { menu } = useBuilder();
  const { document: frameDocument } = useFrame();
  const link = (editor == null ? void 0 : editor.getAttributes("link")) || {};
  const websiteMenuItemIds = useMemo(() => {
    var _a;
    return (_a = menu == null ? void 0 : menu.map) == null ? void 0 : _a.call(menu, (item) => ({
      label: item.name,
      value: item.id
    }));
  }, [JSON.stringify(menu)]);
  useEffect(() => {
    if (!visible)
      return;
    if (link == null ? void 0 : link.pageId) {
      setEventType("openPageOfWebsite");
      return;
    }
    setEventType("link");
  }, [visible]);
  const getActionTypeOption = () => {
    const actionTypeOptions = [
      {
        label: t("\u0110\u1ECBa ch\u1EC9 web"),
        value: "link"
      }
    ];
    if (websiteMenuItemIds == null ? void 0 : websiteMenuItemIds.length) {
      actionTypeOptions.unshift({
        label: t("M\u1EDF trang trong website c\u1EE7a b\u1EA1n"),
        value: "openPageOfWebsite"
      });
    }
    return actionTypeOptions;
  };
  const handleSetLink = (link2) => {
    var _a, _b;
    (_b = (_a = editor == null ? void 0 : editor.commands) == null ? void 0 : _a.executeCmd) == null ? void 0 : _b.call(
      _a,
      ({ commands }) => commands.setCustomLink(link2)
    );
  };
  const handleUnsetlink = () => {
    var _a, _b;
    (_b = (_a = editor == null ? void 0 : editor.commands) == null ? void 0 : _a.executeCmd) == null ? void 0 : _b.call(_a, ({ commands }) => commands.unsetLink());
  };
  const onChangeEventType = (event) => {
    setEventType(event);
    if (!event) {
      handleUnsetlink();
    }
  };
  const onClose = () => {
    setVisible(false);
  };
  const renderSettings = () => {
    var _a, _b;
    const renderLinkValueConfig = () => {
      switch (eventType) {
        case "link":
          return /* @__PURE__ */ jsx(TextInputConfig, {
            initValue: link == null ? void 0 : link.href,
            formItemProps: { label: t("\u0110\u01B0\u1EDDng d\u1EABn"), layout: "vertical" },
            inputProps: {
              placeholder: t("Nh\u1EADp \u0111\u01B0\u1EDDng d\u1EABn")
            },
            updateValue: (val) => {
              const href = val == null ? void 0 : val.trim();
              if (href) {
                handleSetLink({
                  href,
                  pageId: void 0
                });
                return;
              }
              handleUnsetlink();
            }
          });
        case "openPageOfWebsite":
          return /* @__PURE__ */ jsx(SelectConfig, {
            requireExistedValue: true,
            formItemProps: {
              style: {
                marginBottom: "0.5rem"
              }
            },
            selectProps: {
              placeholder: t("Ch\u1ECDn trang"),
              style: { width: "100%" },
              value: link == null ? void 0 : link.pageId,
              onChange: (pageId) => {
                if (pageId) {
                  handleSetLink({
                    href: void 0,
                    pageId
                  });
                  return;
                }
                handleUnsetlink();
              }
            },
            options: websiteMenuItemIds
          });
        default:
          return null;
      }
    };
    const linkValueConfig = renderLinkValueConfig();
    if (!linkValueConfig)
      return null;
    return /* @__PURE__ */ jsxs(Fragment, {
      children: [
        linkValueConfig,
        /* @__PURE__ */ jsx(SwitchConfig, {
          formItemProps: { label: t("M\u1EDF trong tab m\u1EDBi") },
          switchProps: {
            checked: link.target === "_blank",
            onChange: (checked) => {
              handleSetLink({
                target: checked ? "_blank" : "_self"
              });
            }
          }
        }),
        /* @__PURE__ */ jsx(SwitchConfig, {
          formItemProps: {
            label: t("Li\xEAn k\u1EBFt nofollow"),
            description: t("C\xE1c c\xF4ng c\u1EE5 t\xECm ki\u1EBFm s\u1EBD b\u1ECF qua li\xEAn k\u1EBFt nofollow")
          },
          switchProps: {
            checked: (_b = (_a = link.rel) == null ? void 0 : _a.includes) == null ? void 0 : _b.call(_a, "nofollow"),
            onChange: (checked) => {
              handleSetLink({
                rel: checked ? "nofollow" : null
              });
            }
          }
        })
      ]
    });
  };
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(QuickSettingButton, {
        tooltip: t("G\u0103\u0301n link"),
        onClick: () => setVisible(true),
        children: /* @__PURE__ */ jsx(Link16Regular, {})
      }),
      visible && /* @__PURE__ */ jsx(Modal, {
        closable: false,
        zIndex: 1080,
        open: visible,
        onCancel: onClose,
        title: t("C\xE0i \u0111\u1EB7t s\u1EF1 ki\u1EC7n nh\u1EA5p chu\u1ED9t"),
        destroyOnClose: true,
        width: 450,
        footer: [
          /* @__PURE__ */ jsx(Button, {
            type: "primary",
            shape: "round",
            onClick: onClose,
            children: t("\u0110\xF3ng")
          })
        ],
        getContainer: () => (frameDocument == null ? void 0 : frameDocument.body) || (document == null ? void 0 : document.body),
        children: /* @__PURE__ */ jsx(Tabs, {
          items: [
            {
              label: t("S\u1EF1 ki\u1EC7n"),
              key: "1",
              children: /* @__PURE__ */ jsxs(Fragment, {
                children: [
                  /* @__PURE__ */ jsx(SelectConfig, {
                    options: getActionTypeOption(),
                    formItemProps: {
                      label: t("Lo\u1EA1i s\u1EF1 ki\u1EC7n"),
                      style: { marginBottom: "0.5rem" }
                    },
                    selectProps: {
                      placeholder: t("Ch\u1ECDn s\u1EF1 ki\u1EC7n"),
                      style: { width: "100%" },
                      onChange: onChangeEventType,
                      value: eventType
                    }
                  }),
                  renderSettings()
                ]
              })
            },
            enableTekoTracking && {
              label: "Teko tracking",
              key: "2",
              children: /* @__PURE__ */ jsx(TekoTrackingConfig, {
                propKey: PROP_KEY.Event
              })
            }
          ]
        })
      })
    ]
  });
};

export { TextLinkQuickSetting };
