// @tempi/core-elements/dist/quick-configs/typography/TextColorQuickSetting.es.js
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { BorderOutsideRegular } from '@fluentui/react-icons';
import { SubQuickSetting, QuickSettingButton, ColorPickerConfig } from '@tempi/core-editor';
import { getEditorAttributeValue } from '../../utils/getEditorAttributeValue.es.js';

const TextColorQuickSetting = ({ editor }) => {
  const content = () => {
    return /* @__PURE__ */ jsxs(Fragment, {
      children: [
        /* @__PURE__ */ jsx(ColorPickerConfig, {
          formItemProps: {
            label: t("M\xE0u ch\u1EEF")
          },
          colorPickerProps: {
            color: getEditorAttributeValue(editor, "color"),
            onChangeColor: (color) => {
              var _a, _b;
              (_b = (_a = editor == null ? void 0 : editor.commands) == null ? void 0 : _a.executeCmd) == null ? void 0 : _b.call(
                _a,
                ({ commands }) => commands.setColor(color)
              );
            }
          }
        }),
        /* @__PURE__ */ jsx(ColorPickerConfig, {
          formItemProps: {
            label: t("M\xE0u n\u1EC1n ch\u1EEF")
          },
          colorPickerProps: {
            color: getEditorAttributeValue(editor, "backgroundColor"),
            onChangeColor: (backgroundColor) => {
              var _a, _b;
              (_b = (_a = editor == null ? void 0 : editor.commands) == null ? void 0 : _a.executeCmd) == null ? void 0 : _b.call(
                _a,
                ({ commands }) => commands.setBackgroundColor(backgroundColor)
              );
            }
          }
        })
      ]
    });
  };
  return /* @__PURE__ */ jsx(SubQuickSetting, {
    content,
    children: /* @__PURE__ */ jsx(QuickSettingButton, {
      tooltip: t("M\xE0u ch\u1EEF"),
      children: /* @__PURE__ */ jsx(BorderOutsideRegular, {
        fontSize: 16
      })
    })
  });
};

export { TextColorQuickSetting };
