// @tempi/core-elements/dist/components/website-header/components/Item.es.js
import styled from '@emotion/styled';
import { Alignment } from '@tempi/core-renderer';
import { MOBILE_WIDTH } from '../constants.es.js';

const Item = styled.li`
  user-select: none;
`;
const ItemLabel = styled.label`
  display: block;
`;
const ItemContentWrapper = styled.div(
  ({ alignMenu, hoverColor, isActive }) => `
  display: flex;
  align-items: center;
  text-align: left;
  @media screen and (max-width: ${MOBILE_WIDTH}px) {
    grid-template-columns: 
      ${alignMenu === Alignment.left ? "auto 1fr auto" : ""}
      ${alignMenu === Alignment.center ? "1fr auto 1fr" : ""}
      ${alignMenu === Alignment.right ? "1fr auto" : ""}
    ;
    display: grid;
  }

  ${isActive ? `
    &{
      color: ${hoverColor};
      
      svg {
        color: ${hoverColor} !important;
        }
    }` : ""}

  ${hoverColor ? `
    &:hover {
      color: ${hoverColor};
      
      svg {
        color: ${hoverColor} !important;
        }
    }` : ""}
`
);
const ItemContent = styled.a(({ grid, alignMenu, isActive }) => {
  const {
    menuItemSpacing,
    menuColor,
    menuHoverColor,
    mobileNavigationTextAlign,
    menuFontSize,
    menuFontStyle,
    menuFontWeight,
    menuTextDecoration
  } = grid || {};
  return `
  position: relative;
  display: inline-flex;
  align-self: center;
  margin: 0 0 0 ${menuItemSpacing}px;
  text-decoration: ${menuTextDecoration ? `${menuTextDecoration} !important` : "none"};
  white-space: nowrap;
  cursor: pointer;
  font-size: ${typeof menuFontSize === "number" ? `${menuFontSize}px` : menuFontSize};
  ${menuFontWeight ? `font-weight: ${menuFontWeight};` : ""}
  ${menuFontStyle ? `font-style: ${menuFontStyle};` : ""}

  ${isActive ? `
      color: ${menuHoverColor};` : `color: ${menuColor};`}

  &:hover {
    text-decoration: underline;
    text-underline-offset: 7px;
    color: ${menuHoverColor};
  }

  @media screen and (max-width: ${MOBILE_WIDTH}px){
    margin: 0;
    text-align: ${mobileNavigationTextAlign};

    grid-column: 
    ${alignMenu === Alignment.left ? "1/2" : ""}
    ${alignMenu === Alignment.center ? "2/3" : ""}
    ${alignMenu === Alignment.right ? "3/4" : ""}
  }
`;
});
const ItemIcon = styled.img`
  width: 20px;
  height: 20px;
  margin-right: 0.4em;
`;

export { Item, ItemContent, ItemContentWrapper, ItemIcon, ItemLabel };
