// @tempi/core-elements/dist/components/website-header/components/Menu.es.js
import styled from '@emotion/styled';
import { MOBILE_WIDTH } from '../constants.es.js';

const Menu = styled.ul(({ grid }) => {
  const {
    navigationTextAlign,
    mobileMenuItemSpacing,
    mobileNavigationJustifySelf,
    paddingRight,
    mobileNavigationTextAlign,
    mobileBackground = "#ffffff"
  } = grid;
  return `
    padding-left: unset;
    display: inline-flex;
    flex-flow: row wrap;
    row-gap: 16px;
    align-items: center;
    height: 100%;
    text-align: ${navigationTextAlign};
    list-style: none;
  
    @media screen and (max-width: ${MOBILE_WIDTH}px) {
      position: absolute;
      top: 100%;
      right: 0;
      left: 0;
      z-index: 1;
      display: grid;
      grid-gap: ${mobileMenuItemSpacing || 16}px 0;
      justify-self: ${mobileNavigationJustifySelf};
      width: auto;
      height: 0;
      max-height: 450px;
      padding: ${mobileMenuItemSpacing || 16}px ${mobileMenuItemSpacing || 16}px ${mobileMenuItemSpacing || 16}px ${paddingRight}px;
      margin: 0;
      overflow-y: auto;
      text-align: ${mobileNavigationTextAlign};
      pointer-events: none;
      background-size: cover;
      box-shadow: 0 6px 14px -14px #0000001a;
      opacity: 0;
      transform: translateY(-1em);
      transition-timing-function: cubic-bezier(.45,0,.1,1);
      transition-duration: .2s;
      transition-property: opacity,transform;
      background: ${mobileBackground};
  
      &.open {
        height: auto;
        pointer-events: auto;
        opacity: 1;
        transform: translate(0);
        transition-timing-function: cubic-bezier(.45,0,.1,1);
        transition-duration: .2s;
        transition-property: opacity,transform;
      }
    };
  `;
});

export { Menu };
