// @tempi/core-editor/dist/configs/CodeEditorConfig/CodeEditorConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var React = require('react');
var useProp = require('../../hooks/useProp.js');
var FormItem = require('../../components/FormItem/FormItem.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

function _interopNamespace(e) {
  if (e && e.__esModule) return e;
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () { return e[k]; }
        });
      }
    });
  }
  n["default"] = e;
  return Object.freeze(n);
}

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const AceEditor = React__default["default"].lazy(() => Promise.resolve().then(function () { return /*#__PURE__*/_interopNamespace(require('react-ace')); }));
const CodeEditorConfig = ({
  propKey,
  formItemProps
}) => {
  const [prop, setProp] = useProp.useProp(propKey);
  React.useEffect(() => {
    const importDepsAce = async () => {
      await Promise.resolve().then(function () { return /*#__PURE__*/_interopNamespace(require('ace-builds/src-noconflict/mode-java')); });
      await Promise.resolve().then(function () { return /*#__PURE__*/_interopNamespace(require('ace-builds/src-noconflict/theme-github')); });
    };
    importDepsAce();
  }, []);
  return /* @__PURE__ */ jsxRuntime.jsx(React__default["default"].Suspense, {
    fallback: /* @__PURE__ */ jsxRuntime.jsx("div", {}),
    children: /* @__PURE__ */ jsxRuntime.jsx(FormItem.FormItem, __spreadProps(__spreadValues({}, formItemProps), {
      children: /* @__PURE__ */ jsxRuntime.jsx(AceEditor, {
        value: prop,
        onChange: (value) => setProp(value),
        mode: "html",
        fontSize: 14,
        height: "300px",
        width: "100%",
        setOptions: {
          showLineNumbers: true,
          tabSize: 2
        },
        highlightActiveLine: true
      })
    }))
  });
};

exports.CodeEditorConfig = CodeEditorConfig;
