// @tempi/core-editor/dist/components/PrevNext/PrevNext.es.js
import { jsxs, jsx } from 'react/jsx-runtime';
import styled from '@emotion/styled';
import { ArrowHookUpLeft20Regular, ArrowHookUpRight20Regular } from '@fluentui/react-icons';
import { Space, Button } from 'antd';
import { useBuilder } from '../../contexts/BuilderContext.es.js';
import { useEditor } from '../../hooks/useEditor.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const PrevNext = ({ disabled }) => {
  const { getAttrTracking, isEditingText } = useBuilder();
  const {
    canUndo,
    canRedo,
    actions: { history }
  } = useEditor((_, query) => {
    return {
      canUndo: !isEditingText && query.history.canUndo() && !disabled,
      canRedo: !isEditingText && query.history.canRedo() && !disabled
    };
  });
  const onUndo = () => {
    canUndo && history.undo();
  };
  const onRedo = () => {
    canUndo && history.redo();
  };
  return /* @__PURE__ */ jsxs(Space, {
    children: [
      /* @__PURE__ */ jsx(IconButton, __spreadProps(__spreadValues({}, getAttrTracking({
        regionName: "pageEditor",
        contentName: "undoBtn"
      })), {
        onClick: onUndo,
        type: "text",
        disabled: !canUndo,
        icon: /* @__PURE__ */ jsx(ArrowHookUpLeft20Regular, {
          color: canUndo ? "var(--text-color)" : "var(--text-secondary-color)"
        })
      })),
      /* @__PURE__ */ jsx(IconButton, __spreadProps(__spreadValues({}, getAttrTracking({
        regionName: "pageEditor",
        contentName: "redoBtn"
      })), {
        onClick: onRedo,
        type: "text",
        disabled: !canRedo,
        icon: /* @__PURE__ */ jsx(ArrowHookUpRight20Regular, {
          color: canRedo ? "var(--text-color)" : "var(--text-secondary-color)"
        })
      }))
    ]
  });
};
const IconButton = styled(Button)`
  display: flex;
  align-items: center;
  width: fit-content;
  padding: 0;
  background: transparent !important;

  &:not([disabled]):hover {
    path,
    stroke {
      fill: var(--text-color);
    }
  }
`;

export { PrevNext };
