// @tempi/core-elements/dist/components/countdown/configs/CustomizeCountDownDialog .js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../i18n.js');
var icons = require('@ant-design/icons');
var coreEditor = require('@tempi/core-editor');

const CustomizeCountDownDialog = () => {
  const content = /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.DatePickerConfig, {
        propKey: "due",
        dateTimePickerProps: {
          showTime: { format: "HH:mm" },
          format: "DD/MM/YYYY HH:mm"
        },
        formItemProps: {
          label: i18n.t("Th\u1EDDi gian k\u1EBFt th\xFAc")
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
        formItemProps: {
          label: i18n.t("Hi\u1EC7n s\u1ED1 ng\xE0y")
        },
        propKey: "showDays"
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
        formItemProps: {
          label: i18n.t("Hi\u1EC7n s\u1ED1 gi\u1EDD")
        },
        propKey: "showHours"
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
        formItemProps: {
          label: i18n.t("Hi\u1EC7n s\u1ED1 ph\xFAt")
        },
        propKey: "showMinutes"
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
        formItemProps: {
          label: i18n.t("Hi\u1EC7n s\u1ED1 gi\xE2y")
        },
        propKey: "showSeconds"
      })
    ]
  });
  return /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SubQuickSetting, {
    content,
    trigger: "click",
    placement: "bottom",
    children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.QuickSettingButton, {
      tooltip: i18n.t("T\xF9y ch\u1EC9nh \u0111\u1ED3ng h\u1ED3 \u0111\u1EBFm ng\u01B0\u1EE3c"),
      children: /* @__PURE__ */ jsxRuntime.jsx(icons.ClockCircleOutlined, {
        style: {
          fontSize: "16px"
        }
      })
    })
  });
};

exports.CustomizeCountDownDialog = CustomizeCountDownDialog;
