// @tempi/core-elements/dist/components/countdown/configs/CustomizeCountDownDialog .es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../../i18n.es.js';
import { ClockCircleOutlined } from '@ant-design/icons';
import { DatePickerConfig, SwitchConfig, SubQuickSetting, QuickSettingButton } from '@tempi/core-editor';

const CustomizeCountDownDialog = () => {
  const content = /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(DatePickerConfig, {
        propKey: "due",
        dateTimePickerProps: {
          showTime: { format: "HH:mm" },
          format: "DD/MM/YYYY HH:mm"
        },
        formItemProps: {
          label: t("Th\u1EDDi gian k\u1EBFt th\xFAc")
        }
      }),
      /* @__PURE__ */ jsx(SwitchConfig, {
        formItemProps: {
          label: t("Hi\u1EC7n s\u1ED1 ng\xE0y")
        },
        propKey: "showDays"
      }),
      /* @__PURE__ */ jsx(SwitchConfig, {
        formItemProps: {
          label: t("Hi\u1EC7n s\u1ED1 gi\u1EDD")
        },
        propKey: "showHours"
      }),
      /* @__PURE__ */ jsx(SwitchConfig, {
        formItemProps: {
          label: t("Hi\u1EC7n s\u1ED1 ph\xFAt")
        },
        propKey: "showMinutes"
      }),
      /* @__PURE__ */ jsx(SwitchConfig, {
        formItemProps: {
          label: t("Hi\u1EC7n s\u1ED1 gi\xE2y")
        },
        propKey: "showSeconds"
      })
    ]
  });
  return /* @__PURE__ */ jsx(SubQuickSetting, {
    content,
    trigger: "click",
    placement: "bottom",
    children: /* @__PURE__ */ jsx(QuickSettingButton, {
      tooltip: t("T\xF9y ch\u1EC9nh \u0111\u1ED3ng h\u1ED3 \u0111\u1EBFm ng\u01B0\u1EE3c"),
      children: /* @__PURE__ */ jsx(ClockCircleOutlined, {
        style: {
          fontSize: "16px"
        }
      })
    })
  });
};

export { CustomizeCountDownDialog };
