// @tempi/core-elements/dist/components/countdown/configs/DisplayDistanceDialog.es.js
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
import { t } from '../../../i18n.es.js';
import { useProp, SubQuickSetting, QuickSettingButton, SwitchConfig, InputNumberConfig, SelectConfig } from '@tempi/core-editor';
import { RowTripleRegular } from '@fluentui/react-icons';
import { PROP_KEY } from '../../../constants/propKey.es.js';

const DisplayDistanceDialog = () => {
  const [showLabel] = useProp(PROP_KEY.ShowLabel);
  const [countdownLayout] = useProp(
    PROP_KEY.joinKey(PROP_KEY.CountdownStyle, PROP_KEY.Layout)
  );
  const countdownDisplayOptions = [
    { value: "vertical", label: t("common.vertical") },
    { value: "horizontal", label: t("common.horizontal") }
  ];
  const labelPositionOptions = [
    { value: "baseline", label: t("Ngang h\xE0ng") },
    { value: "center", label: t("Trung t\xE2m") }
  ];
  const content = () => {
    return /* @__PURE__ */ jsxs(Fragment, {
      children: [
        /* @__PURE__ */ jsx(SwitchConfig, {
          formItemProps: {
            label: t("Hi\u1EC3n th\u1ECB nh\xE3n")
          },
          propKey: PROP_KEY.ShowLabel
        }),
        /* @__PURE__ */ jsx(InputNumberConfig, {
          formItemProps: {
            label: t("Kho\u1EA3ng c\xE1ch gi\u1EEFa c\xE1c ph\u1EA7n t\u1EED")
          },
          propKey: PROP_KEY.joinKey(
            PROP_KEY.CountdownStyle,
            PROP_KEY.ElementGap
          )
        }),
        showLabel && /* @__PURE__ */ jsx(InputNumberConfig, {
          formItemProps: {
            label: t("Kho\u1EA3ng c\xE1ch trong nh\xE3n")
          },
          propKey: PROP_KEY.joinKey(
            PROP_KEY.CountdownStyle,
            PROP_KEY.LabelGap
          )
        }),
        /* @__PURE__ */ jsx(SelectConfig, {
          formItemProps: {
            label: t("Ki\u1EC3u hi\u1EC3n th\u1ECB")
          },
          propKey: PROP_KEY.joinKey(PROP_KEY.CountdownStyle, PROP_KEY.Layout),
          options: countdownDisplayOptions
        }),
        countdownLayout === "horizontal" && /* @__PURE__ */ jsx(SelectConfig, {
          formItemProps: {
            label: t("V\u1ECB tr\xED nh\xE3n")
          },
          propKey: PROP_KEY.joinKey(
            PROP_KEY.CountdownStyle,
            PROP_KEY.LabelPosition
          ),
          options: labelPositionOptions
        })
      ]
    });
  };
  return /* @__PURE__ */ jsx(SubQuickSetting, {
    content,
    trigger: "click",
    placement: "bottom",
    children: /* @__PURE__ */ jsx(QuickSettingButton, {
      tooltip: t("Hi\u1EC3n th\u1ECB v\xE0 kho\u1EA3ng c\xE1ch"),
      children: /* @__PURE__ */ jsx(RowTripleRegular, {})
    })
  });
};

export { DisplayDistanceDialog };
