// @tempi/core-elements/dist/components/countdown/configs/TypographyCountdownQuickSetting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../i18n.js');
var reactIcons = require('@fluentui/react-icons');
var coreEditor = require('@tempi/core-editor');
var antd = require('antd');
var TextConfig = require('../../../configs/TextConfig/TextConfig.js');
var propKey = require('../../../constants/propKey.js');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const { Panel } = antd.Collapse;
const digitPropKey = propKey.PROP_KEY.joinKey(
  propKey.PROP_KEY.CountdownStyle,
  propKey.PROP_KEY.DigitStyle
);
const labelPropKey = propKey.PROP_KEY.joinKey(
  propKey.PROP_KEY.CountdownStyle,
  propKey.PROP_KEY.LabelStyle
);
const content = (props) => /* @__PURE__ */ jsxRuntime.jsxs(antd.Collapse, __spreadProps(__spreadValues({}, props), {
  children: [
    /* @__PURE__ */ jsxRuntime.jsxs(Panel, {
      header: i18n.t("Ch\u1EEF s\u1ED1"),
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
          propKey: propKey.PROP_KEY.joinKey(digitPropKey, propKey.PROP_KEY.Color)
        }),
        /* @__PURE__ */ jsxRuntime.jsx(TextConfig.TextConfig, {
          propKey: propKey.PROP_KEY.joinKey(digitPropKey),
          config: {
            fontSize: true,
            textAlign: true,
            isCountdownTextSetting: true
          }
        })
      ]
    }, "DIGIT"),
    /* @__PURE__ */ jsxRuntime.jsxs(Panel, {
      header: i18n.t("Nh\xE3n"),
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
          propKey: propKey.PROP_KEY.joinKey(labelPropKey, propKey.PROP_KEY.Color)
        }),
        /* @__PURE__ */ jsxRuntime.jsx(TextConfig.TextConfig, {
          propKey: propKey.PROP_KEY.joinKey(labelPropKey),
          config: {
            fontSize: true,
            textAlign: true,
            isCountdownTextSetting: true
          }
        })
      ]
    }, "LABEL")
  ]
}));
const TypographyCountdownQuickSetting = (props) => {
  return /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SubQuickSetting, {
    content: () => content(props),
    children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.QuickSettingButton, {
      children: /* @__PURE__ */ jsxRuntime.jsx(reactIcons.TextTRegular, {})
    })
  });
};

exports.TypographyCountdownQuickSetting = TypographyCountdownQuickSetting;
