// @tempi/core-elements/dist/components/article-list/setting/ArticleListSetting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../../i18n.es.js';
import styled from '@emotion/styled';
import { ArrowSyncRegular } from '@fluentui/react-icons';
import { useProp, useBuilder, SelectConfig, SwitchConfig, ColorPickerConfig, FormItem, RangeSliderConfig } from '@tempi/core-editor';
import { useStore } from '@tempi/core-renderer';
import { theme, Tooltip, Typography } from 'antd';
import { useState, useEffect } from 'react';
import { LoadMoreType } from '../type.es.js';
import { PROP_KEY } from '../../../constants/propKey.es.js';

const ArticleListSetting = () => {
  const {
    token: { colorPrimary }
  } = theme.useToken();
  const [layout] = useProp(PROP_KEY.Layout);
  const { getWebsiteCategories, setOpenArticleSidebar } = useBuilder();
  const { globalData } = useStore();
  const [categories, setCategories] = useState([]);
  const [loadMoreType] = useProp(PROP_KEY.LoadMoreType);
  const [triggerUpdate, setTriggerUpdate] = useProp(PROP_KEY.TriggerUpdate);
  const { websiteId } = globalData || {};
  useEffect(() => {
    if (!websiteId || typeof getWebsiteCategories !== "function")
      return;
    getWebsiteCategories(websiteId).then((res) => {
      setCategories(res);
    });
  }, [websiteId]);
  const renderGridConfig = () => /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(RangeSliderConfig, {
        propKey: PROP_KEY.joinKey(PROP_KEY.GridConfig, PROP_KEY.Row),
        formItemProps: {
          label: t("S\u1ED1 h\xE0ng")
        },
        sliderProps: {
          min: 1,
          max: 10
        }
      }),
      /* @__PURE__ */ jsx(RangeSliderConfig, {
        propKey: PROP_KEY.joinKey(PROP_KEY.GridConfig, PROP_KEY.Column),
        formItemProps: {
          label: t("S\u1ED1 l\u01B0\u1EE3ng b\xE0i vi\u1EBFt tr\xEAn m\u1ED9t h\xE0ng")
        },
        sliderProps: {
          min: 1,
          max: 6
        }
      }),
      /* @__PURE__ */ jsx(RangeSliderConfig, {
        propKey: PROP_KEY.joinKey(PROP_KEY.GridConfig, PROP_KEY.Gutter),
        formItemProps: {
          label: t("Kho\u1EA3ng c\xE1ch gi\u1EEFa c\xE1c b\xE0i vi\u1EBFt")
        },
        sliderProps: {
          min: 4,
          max: 40
        },
        unitProps: {
          defaultValue: "px"
        }
      })
    ]
  });
  const renderListConfig = () => /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(RangeSliderConfig, {
        propKey: PROP_KEY.joinKey(PROP_KEY.ListConfig, PROP_KEY.Row),
        formItemProps: {
          label: t("S\u1ED1 l\u01B0\u1EE3ng b\xE0i vi\u1EBFt hi\u1EC3n th\u1ECB")
        },
        sliderProps: {
          min: 1,
          max: 50
        }
      }),
      /* @__PURE__ */ jsx(RangeSliderConfig, {
        propKey: PROP_KEY.joinKey(PROP_KEY.ListConfig, PROP_KEY.Gutter),
        formItemProps: {
          label: t("Kho\u1EA3ng c\xE1ch gi\u1EEFa c\xE1c b\xE0i vi\u1EBFt")
        },
        sliderProps: {
          min: 4,
          max: 40
        },
        unitProps: {
          defaultValue: "px"
        }
      })
    ]
  });
  const refreshArticleList = () => {
    setTriggerUpdate(!triggerUpdate);
  };
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsxs("div", {
        style: {
          display: "flex",
          justifyContent: "space-between",
          alignItems: "center",
          color: colorPrimary
        },
        children: [
          /* @__PURE__ */ jsx(StyledLink, {
            onClick: () => setOpenArticleSidebar == null ? void 0 : setOpenArticleSidebar(true),
            children: t("Qu\u1EA3n l\xFD b\xE0i vi\u1EBFt")
          }),
          /* @__PURE__ */ jsx(Tooltip, {
            zIndex: 1080,
            placement: "top",
            color: "black",
            title: t("L\xE0m m\u1EDBi danh s\xE1ch"),
            children: /* @__PURE__ */ jsx(StyledLink, {
              onClick: () => refreshArticleList(),
              children: /* @__PURE__ */ jsx(ArrowSyncRegular, {
                fontSize: 20
              })
            })
          })
        ]
      }),
      /* @__PURE__ */ jsx(SelectConfig, {
        formItemProps: {
          label: t("Ki\u1EC3u hi\u1EC3n th\u1ECB")
        },
        propKey: PROP_KEY.Layout,
        options: [
          { label: t("L\u01B0\u1EDBi"), value: "grid" },
          { label: t("Danh s\xE1ch"), value: "list" }
        ]
      }),
      /* @__PURE__ */ jsx(SelectConfig, {
        formItemProps: {
          label: t("Danh m\u1EE5c"),
          layout: "vertical"
        },
        propKey: PROP_KEY.CategoryIds,
        options: categories == null ? void 0 : categories.map((category) => ({
          label: category.name,
          value: category.id
        })),
        selectProps: {
          mode: "multiple",
          filterOption: (input, option) => {
            var _a, _b;
            return ((_b = (_a = option == null ? void 0 : option.label) == null ? void 0 : _a.toString().toLowerCase()) != null ? _b : "").includes(
              input.toLowerCase()
            );
          }
        }
      }),
      /* @__PURE__ */ jsx(Typography.Text, {
        type: "secondary",
        style: { marginBottom: "1em", display: "block" },
        children: t("Hi\u1EC3n th\u1ECB t\u1EA5t c\u1EA3 b\xE0i vi\u1EBFt n\u1EBFu kh\xF4ng ch\u1ECDn danh m\u1EE5c n\xE0o")
      }),
      layout === "grid" ? renderGridConfig() : renderListConfig(),
      /* @__PURE__ */ jsx(SwitchConfig, {
        propKey: PROP_KEY.IsOpenNewTab,
        formItemProps: {
          label: t("M\u1EDF tab m\u1EDBi khi b\u1EA5m v\xE0o m\u1ED7i b\xE0i vi\u1EBFt")
        },
        switchProps: {
          defaultChecked: true
        }
      }),
      /* @__PURE__ */ jsx(SelectConfig, {
        propKey: PROP_KEY.LoadMoreType,
        options: [
          {
            label: t('Hi\u1EC3n th\u1ECB n\xFAt "Xem th\xEAm"'),
            value: LoadMoreType.BUTTON
          },
          {
            label: t("Ph\xE2n trang"),
            value: LoadMoreType.PAGINATION
          },
          {
            label: t("Kh\xF4ng hi\u1EC3n th\u1ECB th\xEAm b\xE0i vi\u1EBFt"),
            value: LoadMoreType.NONE
          }
        ],
        formItemProps: {
          label: t("Ki\u1EC3u hi\u1EC3n th\u1ECB th\xEAm b\xE0i vi\u1EBFt"),
          layout: "vertical"
        },
        selectProps: {
          style: {
            width: "100%"
          },
          defaultValue: LoadMoreType.BUTTON,
          allowClear: false
        }
      }),
      loadMoreType === LoadMoreType.PAGINATION && /* @__PURE__ */ jsx(ColorPickerConfig, {
        formItemProps: {
          label: t("M\xE0u ph\xE2n trang")
        },
        propKey: PROP_KEY.PaginationColor
      }),
      /* @__PURE__ */ jsx(FormItem, {
        label: t("Hi\u1EC3n th\u1ECB"),
        layout: "vertical",
        children: /* @__PURE__ */ jsxs(DisplayConfigContainer, {
          children: [
            /* @__PURE__ */ jsx(SwitchConfig, {
              propKey: PROP_KEY.joinKey(
                PROP_KEY.DisplayConfig,
                PROP_KEY.ShowImage
              ),
              formItemProps: {
                label: t("\u1EA2nh b\xE0i vi\u1EBFt")
              },
              switchProps: {
                defaultChecked: true
              }
            }),
            /* @__PURE__ */ jsx(SwitchConfig, {
              propKey: PROP_KEY.joinKey(
                PROP_KEY.DisplayConfig,
                PROP_KEY.ShowTitle
              ),
              formItemProps: {
                label: t("Ti\xEAu \u0111\u1EC1")
              },
              switchProps: {
                defaultChecked: true
              }
            }),
            /* @__PURE__ */ jsx(SwitchConfig, {
              propKey: PROP_KEY.joinKey(
                PROP_KEY.DisplayConfig,
                PROP_KEY.ShowDesc
              ),
              formItemProps: {
                label: t("M\xF4 t\u1EA3")
              },
              switchProps: {
                defaultChecked: true
              }
            }),
            /* @__PURE__ */ jsx(SwitchConfig, {
              propKey: PROP_KEY.joinKey(
                PROP_KEY.DisplayConfig,
                PROP_KEY.ShowPublishDate
              ),
              formItemProps: {
                label: t("Ng\xE0y xu\u1EA5t b\u1EA3n")
              },
              switchProps: {
                defaultChecked: true
              }
            }),
            /* @__PURE__ */ jsx(SwitchConfig, {
              propKey: PROP_KEY.joinKey(
                PROP_KEY.DisplayConfig,
                PROP_KEY.ShowAuthor
              ),
              formItemProps: {
                label: t("T\xE1c gi\u1EA3")
              },
              switchProps: {
                defaultChecked: true
              }
            }),
            /* @__PURE__ */ jsx(SwitchConfig, {
              propKey: PROP_KEY.joinKey(
                PROP_KEY.DisplayConfig,
                PROP_KEY.ShowCategories
              ),
              formItemProps: {
                label: t("Danh m\u1EE5c")
              },
              switchProps: {
                defaultChecked: true
              }
            })
          ]
        })
      })
    ]
  });
};
const DisplayConfigContainer = styled.div`
  margin: 8px 16px;
  h5 {
    font-weight: 400 !important;
  }
`;
const StyledLink = styled.div`
  cursor: pointer;
  padding: 8px 0;
`;

export { ArticleListSetting };
