// @tempi/core-elements/dist/components/article-list/setting/ArticleListSetting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../i18n.js');
var styled = require('@emotion/styled');
var reactIcons = require('@fluentui/react-icons');
var coreEditor = require('@tempi/core-editor');
var coreRenderer = require('@tempi/core-renderer');
var antd = require('antd');
var React = require('react');
var type = require('../type.js');
var propKey = require('../../../constants/propKey.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

const ArticleListSetting = () => {
  const {
    token: { colorPrimary }
  } = antd.theme.useToken();
  const [layout] = coreEditor.useProp(propKey.PROP_KEY.Layout);
  const { getWebsiteCategories, setOpenArticleSidebar } = coreEditor.useBuilder();
  const { globalData } = coreRenderer.useStore();
  const [categories, setCategories] = React.useState([]);
  const [loadMoreType] = coreEditor.useProp(propKey.PROP_KEY.LoadMoreType);
  const [triggerUpdate, setTriggerUpdate] = coreEditor.useProp(propKey.PROP_KEY.TriggerUpdate);
  const { websiteId } = globalData || {};
  React.useEffect(() => {
    if (!websiteId || typeof getWebsiteCategories !== "function")
      return;
    getWebsiteCategories(websiteId).then((res) => {
      setCategories(res);
    });
  }, [websiteId]);
  const renderGridConfig = () => /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.RangeSliderConfig, {
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.GridConfig, propKey.PROP_KEY.Row),
        formItemProps: {
          label: i18n.t("S\u1ED1 h\xE0ng")
        },
        sliderProps: {
          min: 1,
          max: 10
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.RangeSliderConfig, {
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.GridConfig, propKey.PROP_KEY.Column),
        formItemProps: {
          label: i18n.t("S\u1ED1 l\u01B0\u1EE3ng b\xE0i vi\u1EBFt tr\xEAn m\u1ED9t h\xE0ng")
        },
        sliderProps: {
          min: 1,
          max: 6
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.RangeSliderConfig, {
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.GridConfig, propKey.PROP_KEY.Gutter),
        formItemProps: {
          label: i18n.t("Kho\u1EA3ng c\xE1ch gi\u1EEFa c\xE1c b\xE0i vi\u1EBFt")
        },
        sliderProps: {
          min: 4,
          max: 40
        },
        unitProps: {
          defaultValue: "px"
        }
      })
    ]
  });
  const renderListConfig = () => /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.RangeSliderConfig, {
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.ListConfig, propKey.PROP_KEY.Row),
        formItemProps: {
          label: i18n.t("S\u1ED1 l\u01B0\u1EE3ng b\xE0i vi\u1EBFt hi\u1EC3n th\u1ECB")
        },
        sliderProps: {
          min: 1,
          max: 50
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.RangeSliderConfig, {
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.ListConfig, propKey.PROP_KEY.Gutter),
        formItemProps: {
          label: i18n.t("Kho\u1EA3ng c\xE1ch gi\u1EEFa c\xE1c b\xE0i vi\u1EBFt")
        },
        sliderProps: {
          min: 4,
          max: 40
        },
        unitProps: {
          defaultValue: "px"
        }
      })
    ]
  });
  const refreshArticleList = () => {
    setTriggerUpdate(!triggerUpdate);
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsxs("div", {
        style: {
          display: "flex",
          justifyContent: "space-between",
          alignItems: "center",
          color: colorPrimary
        },
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(StyledLink, {
            onClick: () => setOpenArticleSidebar == null ? void 0 : setOpenArticleSidebar(true),
            children: i18n.t("Qu\u1EA3n l\xFD b\xE0i vi\u1EBFt")
          }),
          /* @__PURE__ */ jsxRuntime.jsx(antd.Tooltip, {
            zIndex: 1080,
            placement: "top",
            color: "black",
            title: i18n.t("L\xE0m m\u1EDBi danh s\xE1ch"),
            children: /* @__PURE__ */ jsxRuntime.jsx(StyledLink, {
              onClick: () => refreshArticleList(),
              children: /* @__PURE__ */ jsxRuntime.jsx(reactIcons.ArrowSyncRegular, {
                fontSize: 20
              })
            })
          })
        ]
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
        formItemProps: {
          label: i18n.t("Ki\u1EC3u hi\u1EC3n th\u1ECB")
        },
        propKey: propKey.PROP_KEY.Layout,
        options: [
          { label: i18n.t("L\u01B0\u1EDBi"), value: "grid" },
          { label: i18n.t("Danh s\xE1ch"), value: "list" }
        ]
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
        formItemProps: {
          label: i18n.t("Danh m\u1EE5c"),
          layout: "vertical"
        },
        propKey: propKey.PROP_KEY.CategoryIds,
        options: categories == null ? void 0 : categories.map((category) => ({
          label: category.name,
          value: category.id
        })),
        selectProps: {
          mode: "multiple",
          filterOption: (input, option) => {
            var _a, _b;
            return ((_b = (_a = option == null ? void 0 : option.label) == null ? void 0 : _a.toString().toLowerCase()) != null ? _b : "").includes(
              input.toLowerCase()
            );
          }
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(antd.Typography.Text, {
        type: "secondary",
        style: { marginBottom: "1em", display: "block" },
        children: i18n.t("Hi\u1EC3n th\u1ECB t\u1EA5t c\u1EA3 b\xE0i vi\u1EBFt n\u1EBFu kh\xF4ng ch\u1ECDn danh m\u1EE5c n\xE0o")
      }),
      layout === "grid" ? renderGridConfig() : renderListConfig(),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
        propKey: propKey.PROP_KEY.IsOpenNewTab,
        formItemProps: {
          label: i18n.t("M\u1EDF tab m\u1EDBi khi b\u1EA5m v\xE0o m\u1ED7i b\xE0i vi\u1EBFt")
        },
        switchProps: {
          defaultChecked: true
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
        propKey: propKey.PROP_KEY.LoadMoreType,
        options: [
          {
            label: i18n.t('Hi\u1EC3n th\u1ECB n\xFAt "Xem th\xEAm"'),
            value: type.LoadMoreType.BUTTON
          },
          {
            label: i18n.t("Ph\xE2n trang"),
            value: type.LoadMoreType.PAGINATION
          },
          {
            label: i18n.t("Kh\xF4ng hi\u1EC3n th\u1ECB th\xEAm b\xE0i vi\u1EBFt"),
            value: type.LoadMoreType.NONE
          }
        ],
        formItemProps: {
          label: i18n.t("Ki\u1EC3u hi\u1EC3n th\u1ECB th\xEAm b\xE0i vi\u1EBFt"),
          layout: "vertical"
        },
        selectProps: {
          style: {
            width: "100%"
          },
          defaultValue: type.LoadMoreType.BUTTON,
          allowClear: false
        }
      }),
      loadMoreType === type.LoadMoreType.PAGINATION && /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
        formItemProps: {
          label: i18n.t("M\xE0u ph\xE2n trang")
        },
        propKey: propKey.PROP_KEY.PaginationColor
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
        label: i18n.t("Hi\u1EC3n th\u1ECB"),
        layout: "vertical",
        children: /* @__PURE__ */ jsxRuntime.jsxs(DisplayConfigContainer, {
          children: [
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
              propKey: propKey.PROP_KEY.joinKey(
                propKey.PROP_KEY.DisplayConfig,
                propKey.PROP_KEY.ShowImage
              ),
              formItemProps: {
                label: i18n.t("\u1EA2nh b\xE0i vi\u1EBFt")
              },
              switchProps: {
                defaultChecked: true
              }
            }),
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
              propKey: propKey.PROP_KEY.joinKey(
                propKey.PROP_KEY.DisplayConfig,
                propKey.PROP_KEY.ShowTitle
              ),
              formItemProps: {
                label: i18n.t("Ti\xEAu \u0111\u1EC1")
              },
              switchProps: {
                defaultChecked: true
              }
            }),
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
              propKey: propKey.PROP_KEY.joinKey(
                propKey.PROP_KEY.DisplayConfig,
                propKey.PROP_KEY.ShowDesc
              ),
              formItemProps: {
                label: i18n.t("M\xF4 t\u1EA3")
              },
              switchProps: {
                defaultChecked: true
              }
            }),
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
              propKey: propKey.PROP_KEY.joinKey(
                propKey.PROP_KEY.DisplayConfig,
                propKey.PROP_KEY.ShowPublishDate
              ),
              formItemProps: {
                label: i18n.t("Ng\xE0y xu\u1EA5t b\u1EA3n")
              },
              switchProps: {
                defaultChecked: true
              }
            }),
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
              propKey: propKey.PROP_KEY.joinKey(
                propKey.PROP_KEY.DisplayConfig,
                propKey.PROP_KEY.ShowAuthor
              ),
              formItemProps: {
                label: i18n.t("T\xE1c gi\u1EA3")
              },
              switchProps: {
                defaultChecked: true
              }
            }),
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
              propKey: propKey.PROP_KEY.joinKey(
                propKey.PROP_KEY.DisplayConfig,
                propKey.PROP_KEY.ShowCategories
              ),
              formItemProps: {
                label: i18n.t("Danh m\u1EE5c")
              },
              switchProps: {
                defaultChecked: true
              }
            })
          ]
        })
      })
    ]
  });
};
const DisplayConfigContainer = styled__default["default"].div`
  margin: 8px 16px;
  h5 {
    font-weight: 400 !important;
  }
`;
const StyledLink = styled__default["default"].div`
  cursor: pointer;
  padding: 8px 0;
`;

exports.ArticleListSetting = ArticleListSetting;
