// @tempi/core-renderer/dist/utils/colors/core/mix-css-color/index.js
'use strict';

var index$1 = require('../parse-css-color/index.js');
var index = require('../pure-color/index.js');

function parseColor(color) {
  const res = index$1(color);
  if (res === null)
    return null;
  if (res.type === "hsl")
    res.values = index.hsl2rgb(res.values);
  return res;
}
function mix(color1, color2, percentage = 50) {
  const c1 = parseColor(color1);
  const c2 = parseColor(color2);
  if (!c1 || !c2)
    return null;
  const p = Math.min(Math.max(0, percentage), 100) / 100;
  const w = p * 2 - 1;
  const a = c1.alpha - c2.alpha;
  const w1 = ((w * a === -1 ? w : (w + a) / (1 + w * a)) + 1) / 2;
  const w2 = 1 - w1;
  const [r, g, b] = c1.values.map(
    (c, i) => Math.round(c1.values[i] * w1 + c2.values[i] * w2)
  );
  const alpha = parseFloat((c1.alpha * p + c2.alpha * (1 - p)).toFixed(8));
  return {
    hex: index.rgb2hex([r, g, b]),
    hexa: index.rgb2hex([r, g, b, alpha]),
    rgba: [r, g, b, alpha],
    hsla: [...index.rgb2hsl([r, g, b]).map(Math.round), alpha]
  };
}

module.exports = mix;
