// @tempi/core-elements/dist/configs/storage-v2/components/DataConfigRow.es.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { t } from '../../../i18n.es.js';
import { Open16Regular, Edit16Regular, Delete16Regular } from '@fluentui/react-icons';
import { List, Avatar, Typography, Popconfirm } from 'antd';
import clsx from 'clsx';
import { useMemo } from 'react';
import { FormAccountType } from '../../storage/type.es.js';

const DataConfigRow = ({
  config,
  onEdit,
  onDelete,
  editable = true,
  deletable = true
}) => {
  const isLink = config.type === FormAccountType.GOOGLE_SHEET;
  const displayItem = useMemo(() => {
    switch (config.type) {
      case FormAccountType.EMAIL:
        return {
          title: "Gmail",
          description: config.email,
          avatar: "https://lh3.googleusercontent.com/s7AGeYlPg8QEw7DavsCAXDE03wi7x-wCPPIQd_8WS3BD6trUjPOTkVgTkO87uh5KY5N6061lS2BiuLUxhBxOpxnR3LIGKjsh=w100"
        };
      case FormAccountType.GOOGLE_SHEET:
        return {
          title: "Google Sheet",
          description: config.spreadSheetUrl,
          avatar: "https://lh3.googleusercontent.com/_ly8MwaWDoV9vIWjmmaSHu4FDF-WERnUPO2YlMyr-p033ndrRF4dGkv8XNpQ2FcWUdnGxqE65SWGXAOcosCcpN1UOtYQBfKBOg=w100"
        };
      case FormAccountType.API:
        return {
          title: "API",
          description: config.endpoint,
          avatar: "https://lh3.googleusercontent.com/1Qq-5HxqJa9fPpnxLwBiJKpYHOIeMLaebukSxfQ-gFatc91lzkqribVvjLTuYxevxg6y1Eer9aaF8LY5_7VGLvcUO-yyUo6s=w100"
        };
      default:
        return {};
    }
  }, [config]);
  const onClick = () => {
    if (!isLink)
      return;
    window == null ? void 0 : window.open(config.spreadSheetUrl, "_blank");
  };
  return /* @__PURE__ */ jsxs(List.Item, {
    className: "px-0",
    children: [
      /* @__PURE__ */ jsx(List.Item.Meta, {
        avatar: /* @__PURE__ */ jsx(Avatar, {
          src: displayItem.avatar
        }),
        title: /* @__PURE__ */ jsx("div", {
          children: displayItem.title
        }),
        description: /* @__PURE__ */ jsxs("div", {
          className: clsx(
            "d-flex align-items-center",
            isLink && "cursor-pointer"
          ),
          onClick,
          children: [
            /* @__PURE__ */ jsx(Typography.Text, {
              type: "secondary",
              style: { width: "85%" },
              ellipsis: {
                tooltip: {
                  title: displayItem.description,
                  zIndex: 1081
                }
              },
              children: displayItem.description
            }),
            isLink && /* @__PURE__ */ jsx(Open16Regular, {
              className: "ml-quarter"
            })
          ]
        })
      }),
      /* @__PURE__ */ jsxs("div", {
        children: [
          editable && /* @__PURE__ */ jsx(Edit16Regular, {
            style: { color: "#00000073" },
            className: "mx-half cursor-pointer",
            onClick: onEdit
          }),
          deletable && /* @__PURE__ */ jsx(Popconfirm, {
            title: t("Xo\xE1 k\u1EBFt n\u1ED1i"),
            description: t("B\u1EA1n c\xF3 ch\u1EAFc ch\u1EAFn mu\u1ED1n xo\xE1 k\u1EBFt n\u1ED1i n\xE0y?"),
            onConfirm: onDelete,
            okText: t("Xo\xE1"),
            cancelText: t("common.cancel"),
            zIndex: 1082,
            placement: "topRight",
            children: /* @__PURE__ */ jsx(Delete16Regular, {
              className: "cursor-pointer",
              style: { color: "#00000073" }
            })
          })
        ]
      })
    ]
  });
};

export { DataConfigRow };
