// @tempi/core-elements/dist/configs/storage-v2/components/DataConfigRow.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../i18n.js');
var reactIcons = require('@fluentui/react-icons');
var antd = require('antd');
var clsx = require('clsx');
var React = require('react');
var type = require('../../storage/type.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var clsx__default = /*#__PURE__*/_interopDefaultLegacy(clsx);

const DataConfigRow = ({
  config,
  onEdit,
  onDelete,
  editable = true,
  deletable = true
}) => {
  const isLink = config.type === type.FormAccountType.GOOGLE_SHEET;
  const displayItem = React.useMemo(() => {
    switch (config.type) {
      case type.FormAccountType.EMAIL:
        return {
          title: "Gmail",
          description: config.email,
          avatar: "https://lh3.googleusercontent.com/s7AGeYlPg8QEw7DavsCAXDE03wi7x-wCPPIQd_8WS3BD6trUjPOTkVgTkO87uh5KY5N6061lS2BiuLUxhBxOpxnR3LIGKjsh=w100"
        };
      case type.FormAccountType.GOOGLE_SHEET:
        return {
          title: "Google Sheet",
          description: config.spreadSheetUrl,
          avatar: "https://lh3.googleusercontent.com/_ly8MwaWDoV9vIWjmmaSHu4FDF-WERnUPO2YlMyr-p033ndrRF4dGkv8XNpQ2FcWUdnGxqE65SWGXAOcosCcpN1UOtYQBfKBOg=w100"
        };
      case type.FormAccountType.API:
        return {
          title: "API",
          description: config.endpoint,
          avatar: "https://lh3.googleusercontent.com/1Qq-5HxqJa9fPpnxLwBiJKpYHOIeMLaebukSxfQ-gFatc91lzkqribVvjLTuYxevxg6y1Eer9aaF8LY5_7VGLvcUO-yyUo6s=w100"
        };
      default:
        return {};
    }
  }, [config]);
  const onClick = () => {
    if (!isLink)
      return;
    window == null ? void 0 : window.open(config.spreadSheetUrl, "_blank");
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(antd.List.Item, {
    className: "px-0",
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(antd.List.Item.Meta, {
        avatar: /* @__PURE__ */ jsxRuntime.jsx(antd.Avatar, {
          src: displayItem.avatar
        }),
        title: /* @__PURE__ */ jsxRuntime.jsx("div", {
          children: displayItem.title
        }),
        description: /* @__PURE__ */ jsxRuntime.jsxs("div", {
          className: clsx__default["default"](
            "d-flex align-items-center",
            isLink && "cursor-pointer"
          ),
          onClick,
          children: [
            /* @__PURE__ */ jsxRuntime.jsx(antd.Typography.Text, {
              type: "secondary",
              style: { width: "85%" },
              ellipsis: {
                tooltip: {
                  title: displayItem.description,
                  zIndex: 1081
                }
              },
              children: displayItem.description
            }),
            isLink && /* @__PURE__ */ jsxRuntime.jsx(reactIcons.Open16Regular, {
              className: "ml-quarter"
            })
          ]
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsxs("div", {
        children: [
          editable && /* @__PURE__ */ jsxRuntime.jsx(reactIcons.Edit16Regular, {
            style: { color: "#00000073" },
            className: "mx-half cursor-pointer",
            onClick: onEdit
          }),
          deletable && /* @__PURE__ */ jsxRuntime.jsx(antd.Popconfirm, {
            title: i18n.t("Xo\xE1 k\u1EBFt n\u1ED1i"),
            description: i18n.t("B\u1EA1n c\xF3 ch\u1EAFc ch\u1EAFn mu\u1ED1n xo\xE1 k\u1EBFt n\u1ED1i n\xE0y?"),
            onConfirm: onDelete,
            okText: i18n.t("Xo\xE1"),
            cancelText: i18n.t("common.cancel"),
            zIndex: 1082,
            placement: "topRight",
            children: /* @__PURE__ */ jsxRuntime.jsx(reactIcons.Delete16Regular, {
              className: "cursor-pointer",
              style: { color: "#00000073" }
            })
          })
        ]
      })
    ]
  });
};

exports.DataConfigRow = DataConfigRow;
