// @tempi/core-elements/dist/components/lucky-draw/v2/setting.es.js
import { jsx, jsxs } from 'react/jsx-runtime';
import { t } from '../../../i18n.es.js';
import { useProp } from '@tempi/core-editor';
import { Collapse } from 'antd';
import { LuckyDrawProvider } from './context/LuckyDrawProvider.es.js';
import { PrizeListConfig } from './configs/PrizeListConfig.es.js';
import { ResponseConfig } from './configs/ResponseConfig.es.js';
import { ConditionConfig } from './configs/ConditionConfig.es.js';
import { PROP_KEY } from '../../../constants/propKey.es.js';

const { Panel } = Collapse;
const LuckyDrawSetting = () => {
  const [id] = useProp(PROP_KEY.Id);
  return /* @__PURE__ */ jsx(LuckyDrawProvider, {
    id,
    children: /* @__PURE__ */ jsxs(Collapse, {
      defaultActiveKey: ["PrizeConfig", "ResponseConfig", "ConditionConfig"],
      expandIconPosition: "end",
      bordered: false,
      children: [
        /* @__PURE__ */ jsx(Panel, {
          header: t("C\u1EA5u h\xECnh gi\u1EA3i th\u01B0\u1EDFng"),
          children: /* @__PURE__ */ jsx(PrizeListConfig, {})
        }, "PrizeConfig"),
        /* @__PURE__ */ jsx(Panel, {
          header: t("Ph\u1EA3n h\u1ED3i"),
          children: /* @__PURE__ */ jsx(ResponseConfig, {})
        }, "ResponseConfig"),
        /* @__PURE__ */ jsx(Panel, {
          header: t("\u0110i\u1EC1u ki\u1EC7n tham gia & tr\xFAng gi\u1EA3i"),
          children: /* @__PURE__ */ jsx(ConditionConfig, {})
        }, "ConditionConfig")
      ]
    })
  });
};

export { LuckyDrawSetting };
