// @tempi/core-elements/dist/components/lucky-draw/v2/setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var antd = require('antd');
var LuckyDrawProvider = require('./context/LuckyDrawProvider.js');
var PrizeListConfig = require('./configs/PrizeListConfig.js');
var ResponseConfig = require('./configs/ResponseConfig.js');
var ConditionConfig = require('./configs/ConditionConfig.js');
var propKey = require('../../../constants/propKey.js');

const { Panel } = antd.Collapse;
const LuckyDrawSetting = () => {
  const [id] = coreEditor.useProp(propKey.PROP_KEY.Id);
  return /* @__PURE__ */ jsxRuntime.jsx(LuckyDrawProvider.LuckyDrawProvider, {
    id,
    children: /* @__PURE__ */ jsxRuntime.jsxs(antd.Collapse, {
      defaultActiveKey: ["PrizeConfig", "ResponseConfig", "ConditionConfig"],
      expandIconPosition: "end",
      bordered: false,
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(Panel, {
          header: i18n.t("C\u1EA5u h\xECnh gi\u1EA3i th\u01B0\u1EDFng"),
          children: /* @__PURE__ */ jsxRuntime.jsx(PrizeListConfig.PrizeListConfig, {})
        }, "PrizeConfig"),
        /* @__PURE__ */ jsxRuntime.jsx(Panel, {
          header: i18n.t("Ph\u1EA3n h\u1ED3i"),
          children: /* @__PURE__ */ jsxRuntime.jsx(ResponseConfig.ResponseConfig, {})
        }, "ResponseConfig"),
        /* @__PURE__ */ jsxRuntime.jsx(Panel, {
          header: i18n.t("\u0110i\u1EC1u ki\u1EC7n tham gia & tr\xFAng gi\u1EA3i"),
          children: /* @__PURE__ */ jsxRuntime.jsx(ConditionConfig.ConditionConfig, {})
        }, "ConditionConfig")
      ]
    })
  });
};

exports.LuckyDrawSetting = LuckyDrawSetting;
