// @tempi/core-elements/dist/components/article-list/components/ArticleCard.es.js
import { jsxs, jsx } from 'react/jsx-runtime';
import styled from '@emotion/styled';
import dayjs from 'dayjs';
import { useStore } from '@tempi/core-renderer';
import { Category } from './Category.es.js';
import { DEFAULT_ARTICLE_THUMBNAIL, DEFAULT_TITLE_FONT_SIZE } from '../constants.es.js';

const ArticleCard = ({
  article,
  layout = "list",
  displayConfig = {},
  isOpenNewTab = true,
  titleFontSize = DEFAULT_TITLE_FONT_SIZE,
  categoryColor = "#7367f0",
  categoryType = "button"
}) => {
  const { device } = useStore();
  const isMobile = device === "mobile";
  if (!article)
    return null;
  const {
    name,
    description,
    extraInfo,
    articleInfo,
    canonical,
    categories = []
  } = article || {};
  const {
    showAuthor = true,
    showTitle = true,
    showDesc = true,
    showImage = true,
    showPublishDate = true,
    showCategories = true
  } = displayConfig;
  return /* @__PURE__ */ jsxs(Wrapper, {
    layout: isMobile ? "grid" : layout,
    href: canonical,
    target: isOpenNewTab ? "_blank" : "_self",
    rel: "noreferrer",
    children: [
      showImage && /* @__PURE__ */ jsx(Container, {
        children: /* @__PURE__ */ jsx(Thumbnail, {
          layout,
          children: /* @__PURE__ */ jsx(ThumbnailContent, {
            children: /* @__PURE__ */ jsx("img", {
              src: (extraInfo == null ? void 0 : extraInfo.metaImage) || DEFAULT_ARTICLE_THUMBNAIL,
              width: "100%",
              height: "100%",
              style: {
                objectFit: "cover"
              },
              alt: name
            })
          })
        })
      }),
      /* @__PURE__ */ jsxs(Container, {
        children: [
          showCategories && (categories == null ? void 0 : categories.length) > 0 && /* @__PURE__ */ jsx(CategoryWrapper, {
            children: categories.map((category) => /* @__PURE__ */ jsx(Category, {
              category,
              color: categoryColor,
              type: categoryType
            }, category.id))
          }),
          showTitle && /* @__PURE__ */ jsx(Title, {
            style: { fontSize: titleFontSize },
            children: name
          }),
          showDesc && /* @__PURE__ */ jsx(Description, {
            children: description
          }),
          showAuthor && /* @__PURE__ */ jsx(AuthorName, {
            children: articleInfo == null ? void 0 : articleInfo.authorName
          }),
          showPublishDate && /* @__PURE__ */ jsx(PublishedAt, {
            children: dayjs.unix(articleInfo == null ? void 0 : articleInfo.publishedAt).format("DD/MM/YYYY")
          })
        ]
      })
    ]
  });
};
const Wrapper = styled.a(
  ({ layout }) => layout === "grid" ? `
    display: flex;
    flex-direction: column;
    gap: 16px;
    ` : `
    display: grid;
    grid-template-columns: 1fr 2fr;
    grid-column-gap: 16px;
    grid-row-gap: 16px;
    `
);
const Title = styled.div`
  font-weight: bold;
  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  line-clamp: 2;
  -webkit-box-orient: vertical;
  margin-bottom: 1em;
  line-height: 1.5;
`;
const Description = styled.div`
  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  line-clamp: 3;
  -webkit-box-orient: vertical;
  margin-bottom: 0.5em;
  line-height: 1.5;
`;
const Thumbnail = styled.div(
  ({ layout }) => `
position: relative;
padding-bottom: 67%;
overflow: hidden;
border-radius: 8px;
`
);
const AuthorName = styled.div`
  margin-bottom: 0.5em;
`;
const PublishedAt = styled.div`
  margin-bottom: 0.5em;
`;
const ThumbnailContent = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
`;
const CategoryWrapper = styled.div`
  display: flex;
  gap: 8px;
  flex-wrap: wrap;
  font-weight: 500;
  margin-bottom: 0.5em;
`;
const Container = styled.div`
  width: 100%;
`;

export { ArticleCard };
