// @tempi/core-elements/dist/components/article-list/components/LoadMoreButton.es.js
import { jsx, Fragment } from 'react/jsx-runtime';
import { t } from '../../../i18n.es.js';
import styled from '@emotion/styled';
import { ButtonV2 } from '../../button/v2/ui.es.js';

const LoadMoreButton = ({ loading, isEmpty, hasMore, loadMore }) => {
  if (!loading && !isEmpty) {
    return /* @__PURE__ */ jsx(Fragment, {
      children: hasMore ? /* @__PURE__ */ jsx(ButtonV2, {
        text: t("Xem th\xEAm"),
        onClick: loadMore,
        buttonStyle: {
          background: "white",
          hover: {
            color: "#777777"
          },
          border: {
            borderStyle: "solid",
            borderWidth: 1,
            borderColor: "#d9d9d9",
            borderRadius: 4
          },
          padding: { bottom: 8, top: 8, left: 16, right: 16 },
          margin: { top: 16, bottom: 16 }
        },
        commonStyle: { justifyContent: "center" }
      }) : /* @__PURE__ */ jsx(ArticleListFooter, {
        children: t("Kh\xF4ng c\xF2n b\xE0i vi\u1EBFt n\xE0o")
      })
    });
  } else
    return null;
};
const ArticleListFooter = styled.div`
  text-align: center;
  margin: 2em 0;
  color: gray;
`;

export { LoadMoreButton };
