// @tempi/core-elements/dist/components/accordion/selector/AccordionContent/AccordionContentEditor.es.js
import { jsx } from 'react/jsx-runtime';
import { AddCircleFilled } from '@fluentui/react-icons';
import { useBuilder, useNode, useEditor, crafting, getEditorName } from '@tempi/core-editor';
import { theme } from 'antd';
import { useEffect } from 'react';
import { DEFAULT_ACCORDION } from '../../constants.es.js';
import { AccordionContent } from '../../ui/AccordionContent.es.js';
import { Empty } from '../../../../core/Empty/Empty.es.js';
import { AccordionItemEditor } from '../AccordionItem/AccordionItemEditor.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const { useToken } = theme;
const AccordionContentEditor = (_a) => {
  var _b = _a, {
    children
  } = _b, rest = __objRest(_b, [
    "children"
  ]);
  const { setOpenElementMenu } = useBuilder();
  const { token: themeToken } = useToken();
  const {
    id,
    isActive,
    connectors: { connect }
  } = useNode((node) => ({
    isActive: node.events.selected
  }));
  const {
    actions: { selectNode },
    ancestors
  } = useEditor((_, query) => ({
    ancestors: query.node(id).ancestors()
  }));
  useEffect(() => {
    if (isActive) {
      selectNode(ancestors[ancestors.length - 1]);
    }
  }, [isActive]);
  const defaultPlaceholder = /* @__PURE__ */ jsx(AddCircleFilled, {
    style: {
      fontSize: 24,
      color: themeToken.colorPrimary,
      cursor: "pointer"
    },
    onClick: (e) => {
      e.stopPropagation();
      setOpenElementMenu(true);
    }
  });
  return /* @__PURE__ */ jsx(AccordionContent, __spreadProps(__spreadValues({
    ref: connect
  }, rest), {
    children: children || /* @__PURE__ */ jsx(Empty, {
      children: defaultPlaceholder
    })
  }));
};
AccordionContentEditor.craft = crafting({
  displayName: "N\u1ED9i dung thu g\u1ECDn (content)",
  tag: AccordionContent.displayName.toLowerCase(),
  props: DEFAULT_ACCORDION.accordionContent,
  rules: {
    canMoveIn: (incomingNodes) => {
      const ignoreSources = [getEditorName({ AccordionItemEditor })];
      return incomingNodes.every(
        (node) => !ignoreSources.includes(node.data.name)
      );
    }
  }
});

export { AccordionContentEditor };
