// @tempi/core-elements/dist/components/content-slider/constant.es.js
import { DEFAULT_BUTTON } from '../button/v2/constants.es.js';
import { DEFAULT_HEADING } from '../heading/constants.es.js';
import { DEFAULT_PARAGRAPH } from '../paragraph/constants.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const DEFAULT_CONTENT_SLIDER = {
  slider: {
    paginationColor: "rgba(37, 141, 255, 1)",
    paginationPosition: "inside",
    arrows: true,
    arrowStyle: 0,
    slidesPerView: 1
  },
  sliderItem: {
    commonStyle: {
      padding: {
        top: 50,
        right: 50,
        bottom: 50,
        left: 50
      }
    }
  },
  heading: __spreadProps(__spreadValues({}, DEFAULT_HEADING), {
    commonStyle: {
      margin: {
        bottom: 16
      },
      textAlign: "center"
    }
  }),
  paragraph: __spreadProps(__spreadValues({}, DEFAULT_PARAGRAPH), {
    commonStyle: {
      textAlign: "center",
      margin: {
        bottom: 16
      }
    }
  }),
  button: __spreadValues({
    commonStyle: {
      justifyContent: "center"
    }
  }, DEFAULT_BUTTON.primary)
};

export { DEFAULT_CONTENT_SLIDER };
