// @tempi/core-editor/dist/components/UploadImageButton/UploadImageButton.es.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { Button, notification } from 'antd';
import { useState, useRef } from 'react';
import { v4 } from 'uuid';
import { useBuilder } from '../../contexts/BuilderContext.es.js';

const MAX_IMAGE_UPLOAD = 5;
const demoUploadImage = (file, _type) => {
  return URL.createObjectURL(file);
};
const UploadImageButton = ({
  acceptedFiles = [
    "image/jpg",
    "image/jpeg",
    "image/png",
    "image/gif",
    "image/webp"
  ],
  handlerAfterUpload
}) => {
  const [uploading, setUploading] = useState(false);
  const fileInputRef = useRef();
  const { uploadFileHandler = demoUploadImage } = useBuilder();
  const [uid, setUid] = useState(v4());
  const onClick = () => {
    if (!fileInputRef) {
      return;
    }
    const parent = fileInputRef.current.parentNode;
    parent.focus();
    parent.querySelector("button").blur();
    fileInputRef.current.click();
  };
  const onChange = (e) => {
    try {
      const { files } = e.target;
      const fileList = [...files];
      if ((files == null ? void 0 : files.length) > MAX_IMAGE_UPLOAD) {
        notification.error({
          message: t(
            "Vui l\xF2ng ch\u1EC9 t\u1EA3i l\xEAn t\u1ED1i \u0111a {{MAX_IMAGE_UPLOAD}} \u1EA3nh 1 l\u1EA7n!",
            { MAX_IMAGE_UPLOAD }
          )
        });
        return;
      }
      const errorAcceptedFile = fileList.some(
        (file) => !acceptedFiles.includes(file.type)
      );
      if (errorAcceptedFile) {
        notification.error({
          message: t("Ch\u1EC9 upload c\xE1c \u0111\u1ECBnh d\u1EA1ng file: {{acceptedFiles}}", {
            acceptedFiles
          })
        });
        return;
      }
      uploadFiles(fileList);
    } finally {
      reset();
    }
  };
  const uploadFiles = (files) => {
    setUploading(true);
    const postFiles = files.map((file) => {
      return uploadFileHandler(file, "image");
    });
    Promise.all(postFiles).then((fileList) => {
      handlerAfterUpload == null ? void 0 : handlerAfterUpload(fileList);
    }).finally(() => setUploading(false));
  };
  const reset = () => {
    setUid(v4());
  };
  return /* @__PURE__ */ jsxs("div", {
    onKeyDown: onClick,
    onClick,
    children: [
      /* @__PURE__ */ jsx("input", {
        ref: fileInputRef,
        disabled: uploading,
        id: "image-file",
        type: "file",
        hidden: true,
        multiple: true,
        onClick: (e) => e.stopPropagation(),
        onChange: (e) => onChange(e)
      }, uid),
      /* @__PURE__ */ jsx(Button, {
        loading: uploading,
        shape: "round",
        children: t("T\u1EA3i l\xEAn \u1EA3nh s\u1EA3n ph\u1EA9m (t\u1ED1i \u0111a 20MB)")
      })
    ]
  });
};

export { UploadImageButton };
