// @tempi/core-elements/dist/components/list/v2/selector/ListItemV2/ListItemConfig.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../../../../i18n.es.js';
import { useProp, InputNumberConfig, SelectConfig } from '@tempi/core-editor';
import { UnitEnum } from '@tempi/core-renderer';
import { PROP_KEY } from '../../../../../constants/propKey.es.js';

const ListItemConfig = () => {
  const [verticalAlignIcon, setVerticalAlignIcon] = useProp(
    PROP_KEY.joinKey(PROP_KEY.IconStyle, PROP_KEY.AlignItems)
  );
  const [, setVerticalAlignListItem] = useProp(
    PROP_KEY.joinKey(PROP_KEY.CommonStyle, PROP_KEY.AlignItems)
  );
  const verticalAlignmentOptions = [
    { label: t("Ph\xEDa tr\xEAn"), value: "flex-start" },
    { label: t("\u1EDE gi\u1EEFa"), value: "center" },
    { label: t("Ph\xEDa d\u01B0\u1EDBi"), value: "flex-end" }
  ];
  const onChangeVerticalAlignment = (value) => {
    setVerticalAlignIcon(value);
    setVerticalAlignListItem(value);
  };
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(InputNumberConfig, {
        formItemProps: {
          label: t("Kho\u1EA3ng c\xE1ch")
        },
        unitProps: { defaultValue: UnitEnum.pixel },
        propKey: PROP_KEY.joinKey(PROP_KEY.IconConfig, PROP_KEY.Spacing)
      }),
      /* @__PURE__ */ jsx(SelectConfig, {
        options: verticalAlignmentOptions,
        formItemProps: {
          label: t("C\u0103n ch\u1EC9nh theo chi\u1EC1u d\u1ECDc")
        },
        selectProps: {
          placeholder: t("M\u1EB7c \u0111\u1ECBnh"),
          onChange: onChangeVerticalAlignment,
          value: verticalAlignIcon
        }
      })
    ]
  });
};

export { ListItemConfig };
