// @tempi/core-elements/dist/components/list/v2/selector/ListItemV2/ListItemQuickSetting.es.js
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
import styled from '@emotion/styled';
import { AppsListFilled, ChevronDownRegular } from '@fluentui/react-icons';
import { useEditor, useNode, useBuilder, RenderQuickSetting, SubQuickSetting, QuickSettingButton, OpenSettingDialog, DuplicateNode, DeleteNode } from '@tempi/core-editor';
import { ListItemConfig } from './ListItemConfig.es.js';
import { AlignmentQuickSetting } from '../../../../../quick-configs/AlignmentQuickSetting/AlignmentQuickSetting.es.js';
import { PROP_KEY } from '../../../../../constants/propKey.es.js';
import { ColorPickerQuickSetting } from '../../../../../quick-configs/ColorPickerQuickSetting/ColorPickerQuickSetting.es.js';

const ListItemQuickSetting = () => {
  const { actions } = useEditor();
  const { id } = useNode();
  const { track } = useBuilder();
  return /* @__PURE__ */ jsx(IndicatorDiv, {
    children: /* @__PURE__ */ jsx(RenderQuickSetting, {
      trigger: "click",
      onOpenChange: (visible) => {
        if (visible) {
          track == null ? void 0 : track({
            regionName: "ListItemIconSetting",
            contentName: "quickSettingBtn"
          });
        }
      },
      content: /* @__PURE__ */ jsxs(Fragment, {
        children: [
          /* @__PURE__ */ jsx(SubQuickSetting, {
            content: ListItemConfig,
            children: /* @__PURE__ */ jsx(QuickSettingButton, {
              children: /* @__PURE__ */ jsx(AppsListFilled, {
                fontSize: 16
              })
            })
          }),
          /* @__PURE__ */ jsx(AlignmentQuickSetting, {
            propKey: PROP_KEY.joinKey(
              PROP_KEY.CommonStyle,
              PROP_KEY.JustifyContent
            ),
            listAlignment: ["flex-start", "center", "flex-end"]
          }),
          /* @__PURE__ */ jsx(ColorPickerQuickSetting, {}),
          /* @__PURE__ */ jsx(OpenSettingDialog, {}),
          /* @__PURE__ */ jsx(DuplicateNode, {}),
          /* @__PURE__ */ jsx(DeleteNode, {})
        ]
      }),
      children: /* @__PURE__ */ jsx(ColumnQuickSettingBtn, {
        onClick: () => actions.selectNode(id),
        children: /* @__PURE__ */ jsx(ChevronDownRegular, {
          fontSize: 12,
          style: { pointerEvents: "none" }
        })
      })
    })
  });
};
const IndicatorDiv = styled.div`
  font-size: 12px;
  line-height: 12px;
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: left;
  right: -8px;
  top: -8px;
  z-index: 999;
`;
const ColumnQuickSettingBtn = styled.div`
  width: 18px;
  height: 18px;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #7d7d7d;
  cursor: pointer;
  border-radius: 999px;
  transition-duration: 0.2s;
`;

export { ListItemQuickSetting };
