// @tempi/core-elements/dist/components/list/v2/selector/ListItemV2/ListItemConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var coreRenderer = require('@tempi/core-renderer');
var propKey = require('../../../../../constants/propKey.js');

const ListItemConfig = () => {
  const [verticalAlignIcon, setVerticalAlignIcon] = coreEditor.useProp(
    propKey.PROP_KEY.joinKey(propKey.PROP_KEY.IconStyle, propKey.PROP_KEY.AlignItems)
  );
  const [, setVerticalAlignListItem] = coreEditor.useProp(
    propKey.PROP_KEY.joinKey(propKey.PROP_KEY.CommonStyle, propKey.PROP_KEY.AlignItems)
  );
  const verticalAlignmentOptions = [
    { label: i18n.t("Ph\xEDa tr\xEAn"), value: "flex-start" },
    { label: i18n.t("\u1EDE gi\u1EEFa"), value: "center" },
    { label: i18n.t("Ph\xEDa d\u01B0\u1EDBi"), value: "flex-end" }
  ];
  const onChangeVerticalAlignment = (value) => {
    setVerticalAlignIcon(value);
    setVerticalAlignListItem(value);
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
        formItemProps: {
          label: i18n.t("Kho\u1EA3ng c\xE1ch")
        },
        unitProps: { defaultValue: coreRenderer.UnitEnum.pixel },
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.IconConfig, propKey.PROP_KEY.Spacing)
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
        options: verticalAlignmentOptions,
        formItemProps: {
          label: i18n.t("C\u0103n ch\u1EC9nh theo chi\u1EC1u d\u1ECDc")
        },
        selectProps: {
          placeholder: i18n.t("M\u1EB7c \u0111\u1ECBnh"),
          onChange: onChangeVerticalAlignment,
          value: verticalAlignIcon
        }
      })
    ]
  });
};

exports.ListItemConfig = ListItemConfig;
