// @tempi/core-elements/dist/components/list/v2/selector/ListItemV2/ListItemEditorV2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var styled = require('@emotion/styled');
var coreEditor = require('@tempi/core-editor');
var constants = require('../../constants.js');
var ListItemQuickSetting = require('./ListItemQuickSetting.js');
var ListItemV2 = require('../../ui/ListItemV2.js');
var selector = require('../../../../section/selector.js');
var selector$1 = require('../../../../root/selector.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const ListItemEditorV2 = (props) => {
  const { query, hoveredNodes } = coreEditor.useEditor((state) => ({
    enabled: state.options.enabled,
    hoveredNodes: state.events.hovered
  }));
  const {
    connectors: { connect, drag },
    isActive,
    isHover,
    allDescendants
  } = coreEditor.useNode((node) => {
    return {
      isActive: node.events.selected,
      isHover: node.events.hovered,
      allDescendants: query.node(node.id).descendants(true)
    };
  });
  const { children } = props;
  const showQuickSetting = allDescendants.some((el) => hoveredNodes.has(el)) || isActive || isHover;
  return /* @__PURE__ */ jsxRuntime.jsxs(WrapperListItem, {
    ref: (ref) => connect(drag(ref)),
    style: {
      position: "relative"
    },
    isShowBorder: showQuickSetting && !isActive && !isHover,
    children: [
      showQuickSetting && /* @__PURE__ */ jsxRuntime.jsx(ListItemQuickSetting.ListItemQuickSetting, {}),
      /* @__PURE__ */ jsxRuntime.jsx(ListItemV2.ListItemV2, __spreadProps(__spreadValues({}, props), {
        children
      }))
    ]
  });
};
const WrapperListItem = styled__default["default"].div(
  ({ isShowBorder }) => `
   ${isShowBorder ? `&::after {
          content: '';
          border: 1px dashed #4d4d4d;
          position: absolute;
          left: 0;
          top: 0;
          width: 100%;
          height: 100%;
          pointer-events: none;
          display: block;
        }` : ""}
`
);
ListItemEditorV2.craft = coreEditor.crafting({
  displayName: "Danh s\xE1ch bi\u1EC3u t\u01B0\u1EE3ng",
  tag: ListItemV2.ListItemV2.displayName.toLowerCase(),
  advanceAttributes: coreEditor.AdvanceAttributes,
  props: constants.DEFAULT_LIST.listItem,
  rules: {
    canMoveIn: () => false,
    canDrop: (targetNode) => {
      const ignoreTargets = [
        coreEditor.getEditorName({ RowEditor: selector.RowEditor }),
        coreEditor.getEditorName({ RootEditor: selector$1.RootEditor })
      ];
      return !ignoreTargets.includes(targetNode.data.name);
    }
  }
});

exports.ListItemEditorV2 = ListItemEditorV2;
