// @tempi/core-elements/dist/components/row/quick-setting.es.js
import { jsxs, jsx, Fragment } from 'react/jsx-runtime';
import styled from '@emotion/styled';
import { SettingsRegular, ChevronDownRegular } from '@fluentui/react-icons';
import { useEditor, useNode, useBuilder, RenderQuickSetting, QuickSettingButton, DuplicateNode, DeleteNode } from '@tempi/core-editor';

const RowQuickSetting = () => {
  const { actions } = useEditor();
  const { id } = useNode();
  const { setShowSettingDialog, track } = useBuilder();
  return /* @__PURE__ */ jsxs(IndicatorDiv, {
    children: [
      /* @__PURE__ */ jsx(RenderQuickSetting, {
        trigger: "click",
        onOpenChange: (visible) => {
          if (visible) {
            track == null ? void 0 : track({
              regionName: "RowSetting",
              contentName: "quickSettingBtn"
            });
          }
        },
        content: /* @__PURE__ */ jsxs(Fragment, {
          children: [
            /* @__PURE__ */ jsx(QuickSettingButton, {
              onClick: () => {
                actions.selectNode(id);
                setShowSettingDialog(true);
                track == null ? void 0 : track({
                  regionName: "commonSetting",
                  contentName: "advancedSettingBtn"
                });
              },
              children: /* @__PURE__ */ jsx(SettingsRegular, {
                style: { fontSize: 16 }
              })
            }),
            /* @__PURE__ */ jsx(DuplicateNode, {}),
            /* @__PURE__ */ jsx(DeleteNode, {})
          ]
        }),
        children: /* @__PURE__ */ jsx(RowQuickSettingBtn, {
          onClick: () => actions.selectNode(id),
          children: /* @__PURE__ */ jsx(ChevronDownRegular, {
            fontSize: 12,
            style: {
              color: "#fff",
              pointerEvents: "none"
            }
          })
        })
      }),
      /* @__PURE__ */ jsx(ActiveBorder, {
        className: "active-border"
      })
    ]
  });
};
const IndicatorDiv = styled.div`
  position: absolute !important;
  top: -10px;
  left: -10px;
  width: 100%;
  height: 100%;
`;
const RowQuickSettingBtn = styled.div`
  position: relative;
  z-index: 1000;
  width: 18px;
  height: 18px;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #879294 !important;
  cursor: pointer;
  border-radius: 50%;

  &:hover {
    & ~ .active-border {
      border-color: #879294;
    }
  }
`;
const ActiveBorder = styled.div`
  position: absolute;
  top: 9px;
  left: 9px;
  z-index: -1;
  width: 100%;
  height: 100%;
  border: 1px dashed transparent;
  transition: 0.2s linear;
`;

export { RowQuickSetting };
