// @tempi/core-elements/dist/components/row/ui.es.js
import { jsxs, jsx } from 'react/jsx-runtime';
import styled from '@emotion/styled';
import { getStyle, useIntersectionObserver, useAnimation, BackgroundVideo, BG_LAZY_CLASS_NAME, BG_VISIBLE_LAZY_CLASS_NAME } from '@tempi/core-renderer';
import clsx from 'clsx';
import React__default, { useRef } from 'react';
import { Row } from '../../core/Grid/Row.es.js';
import { MAPPING_VERTICAL_ALIGNMENT_FLEX } from '../../constants/grid.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const StyledPbRow = styled(Row)(
  ({
    verticalAlignment,
    commonStyle
  }) => `
  ${getStyle(commonStyle, { lazyBgImage: true })}
  align-items: ${MAPPING_VERTICAL_ALIGNMENT_FLEX[verticalAlignment] || MAPPING_VERTICAL_ALIGNMENT_FLEX.top};
`
);
const PbRow = React__default.forwardRef(
  (props, passRef) => {
    var _b;
    const _a = props, {
      commonStyle,
      children,
      verticalAlignment,
      gutter,
      className
    } = _a, rest = __objRest(_a, [
      "commonStyle",
      "children",
      "verticalAlignment",
      "gutter",
      "className"
    ]);
    const rowRef = useRef(null);
    const ref = passRef || rowRef;
    const isIntersected = useIntersectionObserver(ref, {
      freezeOnceVisible: true
    });
    const { style: animationStyle, animationClassName } = useAnimation(
      commonStyle == null ? void 0 : commonStyle.animation,
      rowRef
    );
    return /* @__PURE__ */ jsxs(StyledPbRow, __spreadProps(__spreadValues({
      verticalAlignment,
      ref: (el) => {
        if (typeof passRef === "function") {
          passRef(el);
        } else if (passRef) {
          passRef.current = el;
        }
        rowRef.current = el;
      },
      commonStyle,
      style: animationStyle,
      gutter: gutter || [16, 16],
      className: clsx(
        className,
        {
          [BG_LAZY_CLASS_NAME]: true,
          [BG_VISIBLE_LAZY_CLASS_NAME]: isIntersected
        },
        animationClassName
      )
    }, rest), {
      children: [
        (commonStyle == null ? void 0 : commonStyle.backgroundVideo) && /* @__PURE__ */ jsx(BackgroundVideo, __spreadProps(__spreadValues({}, commonStyle.backgroundVideo), {
          background: (_b = commonStyle.background) == null ? void 0 : _b.toString()
        })),
        children
      ]
    }));
  }
);
PbRow.displayName = Row.displayName;

export { PbRow };
