// @tempi/core-editor/dist/configs/EventConfig/SelectPopupConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var antd = require('antd');
var styled = require('@emotion/styled');
var dndCore = require('@tempi/dnd-core');
var React = require('react');
var usePropSeparateDevice = require('../../hooks/usePropSeparateDevice.js');
var BuilderContext = require('../../contexts/BuilderContext.js');
var SelectConfig = require('../SelectConfig/SelectConfig.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const SelectPopupConfig = (props) => {
  const _a = props, { label, placeholder, itemIndex, device } = _a, rest = __objRest(_a, ["label", "placeholder", "itemIndex", "device"]);
  const [_, setProp] = usePropSeparateDevice.usePropSeparateDevice(
    rest.propKey,
    rest.separateDevice,
    rest.usePropOptions
  );
  const { setShowModalSelectPopup, relatedNodeId, setRelatedNodeId } = BuilderContext.useBuilder();
  const { id } = dndCore.useNode();
  const handleCreatePopup = () => {
    setShowModalSelectPopup(true);
    setRelatedNodeId({ componentId: id, itemIndex, device });
  };
  React.useEffect(() => {
    if (relatedNodeId == null ? void 0 : relatedNodeId.popupId) {
      setProp(relatedNodeId.popupId);
      setRelatedNodeId(void 0);
    }
  }, [relatedNodeId == null ? void 0 : relatedNodeId.popupId]);
  return /* @__PURE__ */ jsxRuntime.jsx(SelectConfig.SelectConfig, __spreadValues({
    selectProps: {
      placeholder: placeholder || i18n.t("M\u1EB7c \u0111\u1ECBnh"),
      style: { width: "100%" },
      popupMatchSelectWidth: false,
      showSearch: true,
      optionFilterProp: "label",
      notFoundContent: rest.options.length ? /* @__PURE__ */ jsxRuntime.jsx(antd.Empty, {
        style: { margin: 8 },
        image: antd.Empty.PRESENTED_IMAGE_SIMPLE
      }) : /* @__PURE__ */ jsxRuntime.jsx(jsxRuntime.Fragment, {}),
      dropdownRender: (option) => /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
        children: [
          option,
          /* @__PURE__ */ jsxRuntime.jsxs(CreatePopup, {
            onClick: handleCreatePopup,
            children: [
              "+",
              ` `,
              i18n.t("Th\xEAm m\u1EDBi popup")
            ]
          })
        ]
      })
    },
    formItemProps: {
      label,
      style: {
        marginBottom: "0.5rem"
      }
    }
  }, rest));
};
const CreatePopup = styled__default["default"].div`
  color: #7367f0;
  padding: 5px;
  width: 100%;
  cursor: pointer;
`;

exports.SelectPopupConfig = SelectPopupConfig;
