// @tempi/core-elements/dist/core/Spin/Spin.es.js
import { jsx, Fragment, jsxs } from 'react/jsx-runtime';
import styled from '@emotion/styled';
import { StyledSpinningInner } from './SpinningInner.es.js';

const Spin = ({
  spinning = true,
  children,
  style
}) => {
  if (!spinning)
    return /* @__PURE__ */ jsx(Fragment, {
      children
    });
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(NestedSpinning, {
        children: /* @__PURE__ */ jsx(StyledSpinning, {
          children: /* @__PURE__ */ jsx(StyledSpinningInner, {})
        })
      }),
      /* @__PURE__ */ jsx(StyledContainer, {
        style,
        children
      })
    ]
  });
};
const NestedSpinning = styled.div`
  position: relative;
`;
const StyledSpinning = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  z-index: 4;
  display: block;
  width: 100%;
  height: 100%;
  max-height: 400px;
  top: 40px;
  left: calc(50% - 40px);
`;
const StyledContainer = styled.div`
  clear: both;
  opacity: 0.5;
  user-select: none;
  pointer-events: none;
`;

export { Spin };
