// @tempi/core-elements/dist/components/accordion/selector/AccordionItem/AccordionItemEditor.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var coreEditor = require('@tempi/core-editor');
var React = require('react');
var constants = require('../../constants.js');
var AccordionItem = require('../../ui/AccordionItem.js');
var AccordionEditor = require('../Accordion/AccordionEditor.js');

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
const AccordionItemEditor = (props) => {
  const {
    connectors: { connect, drag },
    isActive,
    parent,
    id
  } = coreEditor.useNode((node) => ({
    isActive: node.events.selected,
    parent: node.data.parent
  }));
  const {
    actions: { selectNode }
  } = coreEditor.useEditor(({ events }) => ({ selectedNodeIds: events.selected }));
  React.useEffect(() => {
    if (isActive) {
      selectNode(parent);
    }
  }, [isActive]);
  return /* @__PURE__ */ jsxRuntime.jsx(AccordionItem.AccordionItem, __spreadValues({
    dataUniqId: id,
    ref: (ref) => connect(drag(ref))
  }, props));
};
AccordionItemEditor.craft = coreEditor.crafting({
  displayName: "N\u1ED9i dung thu g\u1ECDn (item)",
  tag: AccordionItem.AccordionItem.displayName.toLowerCase(),
  props: constants.DEFAULT_ACCORDION.accordionItem,
  rules: {
    canMoveIn: () => false,
    canDrop: (targetNode) => {
      const allowTargets = [coreEditor.getEditorName({ AccordionEditor: AccordionEditor.AccordionEditor })];
      return allowTargets.includes(targetNode.data.name);
    }
  }
});

exports.AccordionItemEditor = AccordionItemEditor;
