// @tempi/core-elements/dist/components/form/v4/configs/CaptchaConfig.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../../../i18n.es.js';
import { usePropSeparateDevice, SelectConfig, InputConfig, InputNumberConfig, usePropOptionsUnitPercent } from '@tempi/core-editor';
import { UnitEnum } from '@tempi/core-renderer';
import { PROP_KEY } from '../../../../constants/propKey.es.js';

const CaptchaConfig = () => {
  const versionPropKey = PROP_KEY.joinKey(PROP_KEY.Captcha, PROP_KEY.Version);
  const [version] = usePropSeparateDevice(versionPropKey);
  const captchaVersions = [
    {
      label: t("H\u1ED9p ki\u1EC3m 'T\xF4i kh\xF4ng ph\u1EA3i ng\u01B0\u1EDDi m\xE1y'"),
      value: "checkboxv2"
    },
    {
      label: t("Huy hi\u1EC7u Captcha trong n\u1EC1n (V2)"),
      value: "invisiblev2"
    },
    {
      label: t("Huy hi\u1EC7u Captcha trong n\u1EC1n (V3)"),
      value: "invisiblev3"
    }
  ];
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(SelectConfig, {
        propKey: versionPropKey,
        selectProps: {
          allowClear: true,
          options: captchaVersions,
          placeholder: t("Ch\u1ECDn gi\xE1 tr\u1ECB")
        },
        formItemProps: { layout: "vertical", label: t("Lo\u1EA1i Captcha") }
      }),
      /* @__PURE__ */ jsx(InputConfig, {
        formItemProps: { layout: "vertical", label: t("Kho\xE1 trang") },
        propKey: PROP_KEY.joinKey(PROP_KEY.Captcha, PROP_KEY.SiteKey)
      }),
      /* @__PURE__ */ jsx(InputConfig, {
        formItemProps: { layout: "vertical", label: t("T\xEAn l\u1EA5y d\u1EEF li\u1EC7u") },
        propKey: PROP_KEY.joinKey(PROP_KEY.Captcha, PROP_KEY.RequestKey)
      }),
      version === "checkboxv2" && /* @__PURE__ */ jsx(InputNumberConfig, {
        propKey: PROP_KEY.joinKey(PROP_KEY.Captcha, PROP_KEY.Width),
        formItemProps: { label: t("Chi\u1EC1u r\u1ED9ng c\u1ED9t"), layout: "vertical" },
        unitProps: { defaultValue: UnitEnum.percent },
        usePropOptions: usePropOptionsUnitPercent,
        style: { width: "100%" }
      }),
      version === "invisiblev3" && /* @__PURE__ */ jsx(InputConfig, {
        formItemProps: { label: t("H\xE0nh \u0111\u1ED9ng"), layout: "vertical" },
        propKey: PROP_KEY.joinKey(PROP_KEY.Captcha, PROP_KEY.Action)
      })
    ]
  });
};

export { CaptchaConfig };
