// @tempi/core-elements/dist/components/form/v4/configs/CaptchaConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var coreRenderer = require('@tempi/core-renderer');
var propKey = require('../../../../constants/propKey.js');

const CaptchaConfig = () => {
  const versionPropKey = propKey.PROP_KEY.joinKey(propKey.PROP_KEY.Captcha, propKey.PROP_KEY.Version);
  const [version] = coreEditor.usePropSeparateDevice(versionPropKey);
  const captchaVersions = [
    {
      label: i18n.t("H\u1ED9p ki\u1EC3m 'T\xF4i kh\xF4ng ph\u1EA3i ng\u01B0\u1EDDi m\xE1y'"),
      value: "checkboxv2"
    },
    {
      label: i18n.t("Huy hi\u1EC7u Captcha trong n\u1EC1n (V2)"),
      value: "invisiblev2"
    },
    {
      label: i18n.t("Huy hi\u1EC7u Captcha trong n\u1EC1n (V3)"),
      value: "invisiblev3"
    }
  ];
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
        propKey: versionPropKey,
        selectProps: {
          allowClear: true,
          options: captchaVersions,
          placeholder: i18n.t("Ch\u1ECDn gi\xE1 tr\u1ECB")
        },
        formItemProps: { layout: "vertical", label: i18n.t("Lo\u1EA1i Captcha") }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputConfig, {
        formItemProps: { layout: "vertical", label: i18n.t("Kho\xE1 trang") },
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.Captcha, propKey.PROP_KEY.SiteKey)
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputConfig, {
        formItemProps: { layout: "vertical", label: i18n.t("T\xEAn l\u1EA5y d\u1EEF li\u1EC7u") },
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.Captcha, propKey.PROP_KEY.RequestKey)
      }),
      version === "checkboxv2" && /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.Captcha, propKey.PROP_KEY.Width),
        formItemProps: { label: i18n.t("Chi\u1EC1u r\u1ED9ng c\u1ED9t"), layout: "vertical" },
        unitProps: { defaultValue: coreRenderer.UnitEnum.percent },
        usePropOptions: coreEditor.usePropOptionsUnitPercent,
        style: { width: "100%" }
      }),
      version === "invisiblev3" && /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputConfig, {
        formItemProps: { label: i18n.t("H\xE0nh \u0111\u1ED9ng"), layout: "vertical" },
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.Captcha, propKey.PROP_KEY.Action)
      })
    ]
  });
};

exports.CaptchaConfig = CaptchaConfig;
