// @tempi/core-elements/dist/components/form/v4/configs/FormFieldSetting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../../../i18n.es.js';
import { DndContext } from '@dnd-kit/core';
import { restrictToVerticalAxis, restrictToWindowEdges, restrictToFirstScrollableAncestor } from '@dnd-kit/modifiers';
import { SortableContext, verticalListSortingStrategy, useSortable, arrayMove } from '@dnd-kit/sortable';
import { CSS } from '@dnd-kit/utilities';
import styled from '@emotion/styled';
import { CopyRegular, DeleteRegular, ReOrderDotsVerticalRegular } from '@fluentui/react-icons';
import { useProp, SwitchConfig, SelectConfig, InputConfig, InputNumberConfig, TextareaToOptionConfig, usePropOptionsUnitPercent } from '@tempi/core-editor';
import { UnitEnum } from '@tempi/core-renderer';
import { Button, Typography } from 'antd';
import { useState, useEffect } from 'react';
import { DEFAULT_IMAGE_CONFIG } from '../constants.es.js';
import { OptionImageSetting } from './OptionImageSetting.es.js';
import { PROP_KEY } from '../../../../constants/propKey.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const FieldConfig = ({
  index,
  input,
  onDuplicate,
  onRemove,
  activeField,
  setActiveField,
  labelStyle
}) => {
  var _a;
  const { setNodeRef, attributes, listeners, transform } = useSortable({
    id: input.name
  });
  const defaultInputTypeOptions = [
    { value: "text", label: t("form_config.field.text") },
    { value: "tel", label: t("form_config.field.tel") },
    { value: "email", label: t("form_config.field.mail") },
    { value: "number", label: t("form_config.field.number") },
    { value: "textarea", label: t("form_config.field.paragraph") },
    { value: "select-box", label: t("form_config.field.select_box") },
    { value: "radio-group", label: t("form_config.field.radio_group") },
    {
      value: "radio-group-image",
      label: t("form_config.field.radio_image_group")
    },
    { value: "checkbox-group", label: t("form_config.field.checkbox_group") },
    {
      value: "checkbox-group-image",
      label: t("form_config.field.radio_image_group")
    },
    { value: "date", label: t("form_config.field.date") },
    { value: "time", label: t("form_config.field.time") },
    { value: "datetime-local", label: t("form_config.field.datetime-local") },
    { value: "file", label: t("form_config.field.file") },
    {
      value: "location",
      label: t("form_config.field.location")
    },
    { value: "address", label: t("form_config.field.address") }
  ];
  const inputTypeOptions = ((_a = input == null ? void 0 : input.typeSelectable) == null ? void 0 : _a.length) ? defaultInputTypeOptions.filter(
    (type2) => input.typeSelectable.includes(type2.value)
  ) : defaultInputTypeOptions;
  const flexDirectionOptions = [
    {
      label: t("common.horizontal"),
      value: "row"
    },
    {
      label: t("common.vertical"),
      value: "column"
    }
  ];
  const [type, setType] = useProp(
    PROP_KEY.joinKey(PROP_KEY.Inputs, index.toString(), PROP_KEY.Type)
  );
  const [multiple] = useProp(
    PROP_KEY.joinKey(PROP_KEY.Inputs, index.toString(), PROP_KEY.Multiple)
  );
  const [options, setOptions] = useProp(
    PROP_KEY.joinKey(PROP_KEY.Inputs, index.toString(), PROP_KEY.Options)
  );
  const [gap, setGap] = useProp(
    PROP_KEY.joinKey(
      PROP_KEY.Inputs,
      index.toString(),
      PROP_KEY.CommonStyle,
      PROP_KEY.Gap
    )
  );
  const [imageConfig, setImageConfig] = useProp(
    PROP_KEY.joinKey(PROP_KEY.Inputs, index.toString(), PROP_KEY.ImageConfig)
  );
  const [, setFlexDirectionWrap] = useProp(
    PROP_KEY.joinKey(
      PROP_KEY.Inputs,
      index.toString(),
      PROP_KEY.CommonStyle,
      PROP_KEY.FlexDirection
    )
  );
  const [, setHeightWrap] = useProp(
    PROP_KEY.joinKey(
      PROP_KEY.Inputs,
      index.toString(),
      PROP_KEY.CommonStyle,
      PROP_KEY.Height
    )
  );
  const [placeholder, setPlaceholder] = useProp(
    PROP_KEY.joinKey(PROP_KEY.Inputs, index.toString(), PROP_KEY.Placeholder)
  );
  useEffect(() => {
    if (["checkbox-group", "radio-group"].includes(type)) {
      setFlexDirectionWrap("column");
      setHeightWrap(void 0);
    }
    if (type === "select-box" && multiple) {
      setHeightWrap(void 0);
    }
    if (["checkbox-group-image", "radio-group-image"].includes(type)) {
      if (!(options == null ? void 0 : options.length)) {
        setOptions([
          {
            label: "",
            value: t("L\u1EF1a ch\u1ECDn 1")
          },
          {
            label: "",
            value: t("L\u1EF1a ch\u1ECDn 2")
          }
        ]);
      }
      if (!imageConfig) {
        setImageConfig(DEFAULT_IMAGE_CONFIG);
      }
      setFlexDirectionWrap("row");
      setHeightWrap(void 0);
      if (!gap) {
        setGap(20);
      }
    }
  }, [type, multiple]);
  const generateDefaultPlaceHolder = (inputType) => {
    switch (inputType) {
      case "date":
        setPlaceholder("mm/dd/yyyy");
        break;
      case "time":
        setPlaceholder("--:-- --");
        break;
      case "datetime-local":
        setPlaceholder("mm/dd/yyyy --:-- --");
        break;
      default:
        setPlaceholder("");
        break;
    }
  };
  const isDisableSetting = (propKey) => {
    return ((input == null ? void 0 : input.disabledFormFieldSetting) || []).includes(propKey);
  };
  useEffect(() => {
    if (!placeholder) {
      generateDefaultPlaceHolder(type);
    }
  }, []);
  return /* @__PURE__ */ jsx(Fragment, {
    children: /* @__PURE__ */ jsxs(StyledTagField, {
      ref: setNodeRef,
      style: { transform: CSS.Transform.toString(transform) },
      children: [
        /* @__PURE__ */ jsxs(StyledTagHeader, {
          onClick: () => {
            if (activeField === index)
              setActiveField(void 0);
            else
              setActiveField(index);
          },
          children: [
            /* @__PURE__ */ jsxs("div", {
              className: "flex",
              children: [
                /* @__PURE__ */ jsx(StyledAction, __spreadProps(__spreadValues(__spreadValues({}, listeners), attributes), {
                  style: { cursor: "move", marginRight: 8 },
                  children: /* @__PURE__ */ jsx(ReOrderDotsVerticalRegular, {
                    style: {
                      fontSize: 20
                    }
                  })
                })),
                /* @__PURE__ */ jsx("div", {
                  style: labelStyle,
                  children: input.name || input.label
                })
              ]
            }),
            /* @__PURE__ */ jsxs("div", {
              className: "flex",
              children: [
                onDuplicate && /* @__PURE__ */ jsx(StyledAction, {
                  style: { marginRight: 8 },
                  onClick: (e) => {
                    e.stopPropagation();
                    onDuplicate();
                  },
                  children: /* @__PURE__ */ jsx(CopyRegular, {
                    style: {
                      fontSize: 16
                    }
                  })
                }),
                /* @__PURE__ */ jsx(StyledAction, {
                  onClick: (e) => {
                    e.stopPropagation();
                    onRemove();
                  },
                  children: /* @__PURE__ */ jsx(DeleteRegular, {
                    style: {
                      fontSize: 16
                    }
                  })
                })
              ]
            })
          ]
        }),
        activeField === index && /* @__PURE__ */ jsxs(StyledTagContent, {
          children: [
            /* @__PURE__ */ jsx(SelectConfig, {
              formItemProps: {
                label: t("Lo\u1EA1i"),
                layout: "vertical"
              },
              selectProps: {
                style: {
                  width: "100%"
                },
                value: type,
                onChange: (value) => {
                  setType(value);
                  generateDefaultPlaceHolder(value);
                },
                disabled: isDisableSetting(PROP_KEY.Type)
              },
              options: inputTypeOptions
            }),
            /* @__PURE__ */ jsx(InputConfig, {
              showTooltip: true,
              tooltipProps: {
                zIndex: 2001,
                title: t("data_name_tooltip")
              },
              formItemProps: {
                label: t("T\xEAn l\u1EA5y d\u1EEF li\u1EC7u")
              },
              propKey: PROP_KEY.joinKey(
                PROP_KEY.Inputs,
                index.toString(),
                PROP_KEY.Name
              ),
              inputProps: {
                disabled: isDisableSetting(PROP_KEY.Name)
              }
            }),
            /* @__PURE__ */ jsx(InputConfig, {
              formItemProps: {
                label: t("Nh\xE3n")
              },
              propKey: PROP_KEY.joinKey(
                PROP_KEY.Inputs,
                index.toString(),
                PROP_KEY.Label
              )
            }),
            /* @__PURE__ */ jsx(InputConfig, {
              formItemProps: {
                label: t("Ch\u1EEF g\u1EE3i nh\u1EAFc")
              },
              propKey: PROP_KEY.joinKey(
                PROP_KEY.Inputs,
                index.toString(),
                PROP_KEY.Placeholder
              )
            }),
            /* @__PURE__ */ jsx(SwitchConfig, {
              formItemProps: {
                label: t("B\u1EAFt bu\u1ED9c nh\u1EADp")
              },
              switchProps: {
                disabled: isDisableSetting(PROP_KEY.Required)
              },
              propKey: PROP_KEY.joinKey(
                PROP_KEY.Inputs,
                index.toString(),
                PROP_KEY.Required
              )
            }),
            type === "select-box" && /* @__PURE__ */ jsx(SwitchConfig, {
              formItemProps: {
                label: t("Ch\u1ECDn nhi\u1EC1u")
              },
              propKey: PROP_KEY.joinKey(
                PROP_KEY.Inputs,
                index.toString(),
                PROP_KEY.Multiple
              )
            }),
            [
              "radio-group",
              "checkbox-group",
              "radio-group-image",
              "checkbox-group-image"
            ].includes(type) && /* @__PURE__ */ jsxs(Fragment, {
              children: [
                /* @__PURE__ */ jsx(SelectConfig, {
                  formItemProps: {
                    label: t("Ki\u1EC3u hi\u1EC3n th\u1ECB")
                  },
                  options: flexDirectionOptions,
                  propKey: PROP_KEY.joinKey(
                    PROP_KEY.Inputs,
                    index.toString(),
                    PROP_KEY.CommonStyle,
                    PROP_KEY.FlexDirection
                  )
                }),
                /* @__PURE__ */ jsx(InputNumberConfig, {
                  formItemProps: {
                    label: t("Kho\u1EA3ng c\xE1ch")
                  },
                  unitProps: { defaultValue: UnitEnum.pixel },
                  propKey: PROP_KEY.joinKey(
                    PROP_KEY.Inputs,
                    index.toString(),
                    PROP_KEY.CommonStyle,
                    PROP_KEY.Gap
                  )
                })
              ]
            }),
            ["radio-group", "checkbox-group", "select-box"].includes(type) && /* @__PURE__ */ jsx(TextareaToOptionConfig, {
              textareaProps: {
                rows: 4
              },
              formItemProps: {
                label: t("Danh s\xE1ch ch\u1ECDn"),
                tip: /* @__PURE__ */ jsx(Typography.Text, {
                  type: "secondary",
                  style: { marginTop: 4 },
                  children: t(
                    'Nh\u1EADp m\u1ED7i option tr\xEAn 1 d\xF2ng \u0111\u01A1n. \u0110\u1EC3 ph\xE2n bi\u1EC7t gi\u1EEFa label v\xE0 value, s\u1EED d\u1EE5ng k\xFD t\u1EF1 "|"'
                  )
                }),
                layout: "vertical"
              },
              propKey: PROP_KEY.joinKey(
                PROP_KEY.Inputs,
                index.toString(),
                PROP_KEY.Options
              )
            }),
            ["radio-group-image", "checkbox-group-image"].includes(type) && /* @__PURE__ */ jsxs(Fragment, {
              children: [
                /* @__PURE__ */ jsx(OptionImageSetting, {
                  index
                }),
                /* @__PURE__ */ jsx(InputNumberConfig, {
                  formItemProps: {
                    label: t("Chi\u1EC1u r\u1ED9ng h\xECnh \u1EA3nh")
                  },
                  unitProps: { defaultValue: UnitEnum.pixel },
                  propKey: PROP_KEY.joinKey(
                    PROP_KEY.Inputs,
                    index.toString(),
                    PROP_KEY.ImageConfig,
                    PROP_KEY.Width
                  )
                }),
                /* @__PURE__ */ jsx(InputNumberConfig, {
                  formItemProps: {
                    label: t("Bo vi\u1EC1n h\xECnh \u1EA3nh")
                  },
                  unitProps: { defaultValue: UnitEnum.pixel },
                  propKey: PROP_KEY.joinKey(
                    PROP_KEY.Inputs,
                    index.toString(),
                    PROP_KEY.ImageConfig,
                    PROP_KEY.Radius
                  )
                }),
                /* @__PURE__ */ jsx(SelectConfig, {
                  propKey: PROP_KEY.joinKey(
                    PROP_KEY.Inputs,
                    index.toString(),
                    PROP_KEY.ImageConfig,
                    PROP_KEY.BackgroundSize
                  ),
                  options: [
                    { label: t("T\u1EF1 \u0111\u1ED9ng"), value: "auto" },
                    { label: t("L\xE0m \u0111\u1EA7y"), value: "cover" },
                    { label: t("V\u1EEBa m\xE0n h\xECnh"), value: "contain" }
                  ],
                  formItemProps: {
                    label: t("T\u1EC9 l\u1EC7")
                  },
                  selectProps: {
                    defaultValue: "cover"
                  }
                })
              ]
            }),
            /* @__PURE__ */ jsx(InputNumberConfig, {
              formItemProps: {
                label: t("Chi\u1EC1u r\u1ED9ng")
              },
              unitProps: { defaultValue: UnitEnum.percent },
              usePropOptions: usePropOptionsUnitPercent,
              propKey: PROP_KEY.joinKey(
                PROP_KEY.Inputs,
                index.toString(),
                PROP_KEY.CommonStyle,
                PROP_KEY.Width
              )
            }),
            /* @__PURE__ */ jsx(InputNumberConfig, {
              formItemProps: {
                label: t("Chi\u1EC1u cao")
              },
              unitProps: { defaultValue: UnitEnum.pixel },
              propKey: PROP_KEY.joinKey(
                PROP_KEY.Inputs,
                index.toString(),
                PROP_KEY.CommonStyle,
                PROP_KEY.Height
              )
            })
          ]
        })
      ]
    }, index)
  });
};
const FormFieldSetting = () => {
  const [inputs, setInputs] = useProp(PROP_KEY.Inputs);
  const [activeField, setActiveField] = useState();
  const addField = () => {
    setInputs([
      ...inputs,
      {
        type: "text",
        name: "form_item" + (inputs.length + 1),
        label: "form_item" + (inputs.length + 1),
        commonStyle: { height: 32, width: "100%" }
      }
    ]);
  };
  const removeField = (index) => {
    const cloneInputs = [...inputs];
    cloneInputs.splice(index, 1);
    setInputs(cloneInputs);
  };
  const duplicateField = (index) => {
    const newInput = __spreadProps(__spreadValues({}, inputs[index]), {
      name: inputs[index].name + new Date().getTime().toString()
    });
    const cloneInputs = [...inputs];
    cloneInputs.splice(index + 1, 0, newInput);
    setInputs(cloneInputs);
  };
  const handleDragEnd = ({ active, over }) => {
    if (active.id !== over.id) {
      const _inputs = [...inputs];
      const oldIndex = _inputs.findIndex((el) => el.name === active.id);
      const newIndex = _inputs.findIndex((el) => el.name === over.id);
      setInputs(arrayMove(_inputs, oldIndex, newIndex));
    }
  };
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(DndContext, {
        modifiers: [
          restrictToVerticalAxis,
          restrictToWindowEdges,
          restrictToFirstScrollableAncestor
        ],
        onDragEnd: handleDragEnd,
        children: /* @__PURE__ */ jsx(SortableContext, {
          items: inputs.map((el) => el.name),
          strategy: verticalListSortingStrategy,
          children: inputs.map((el, index) => /* @__PURE__ */ jsx(FieldConfig, {
            activeField,
            setActiveField,
            input: el,
            index,
            onDuplicate: () => duplicateField(index),
            onRemove: () => removeField(index)
          }, index))
        })
      }),
      /* @__PURE__ */ jsx(Button, {
        style: {
          marginTop: 8,
          marginBottom: 8
        },
        type: "primary",
        block: true,
        onClick: addField,
        shape: "round",
        children: t("Th\xEAm tr\u01B0\u1EDDng")
      }),
      /* @__PURE__ */ jsx(SwitchConfig, {
        propKey: PROP_KEY.ShowLabel,
        formItemProps: { label: t("Hi\u1EC3n th\u1ECB nh\xE3n") }
      })
    ]
  });
};
const StyledTagField = styled.div`
  justify-content: space-between;
  padding: 0.5rem;
  border-radius: 4px;
  border: 1px solid var(--border-color);
  margin-bottom: 8px;
`;
const StyledTagHeader = styled.div`
  display: flex;
  flex-grow: 1;
  align-items: center;
  justify-content: space-between;
  cursor: pointer;
`;
const StyledTagContent = styled.div`
  padding: 0.5rem;
`;
const StyledAction = styled.div`
  background-color: transparent;
  cursor: pointer;
`;

export { FieldConfig, FormFieldSetting };
