// @tempi/core-elements/dist/components/form/v4/configs/FormFieldSettingCustom.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../../i18n.js');
var React = require('react');
var coreEditor = require('@tempi/core-editor');
var core = require('@dnd-kit/core');
var modifiers = require('@dnd-kit/modifiers');
var sortable = require('@dnd-kit/sortable');
var FormFieldSetting = require('./FormFieldSetting.js');
var antd = require('antd');
var propKey = require('../../../../constants/propKey.js');

const FormFieldSettingCustom = ({
  formFieldOptions,
  requiredInputs = ["phone", "email", "externalId"],
  noticeRequiredFieldText = i18n.t("C\u1EA7n c\xF3 \xEDt nh\u1EA5t 1 trong c\xE1c tr\u01B0\u1EDDng {{var0}}", {
    var0: requiredInputs.join("/")
  })
}) => {
  const [inputs, setInputs] = coreEditor.useProp(propKey.PROP_KEY.Inputs);
  const [activeField, setActiveField] = React.useState();
  const [selectedAdditionalOption, setSelectedAdditionalOption] = React.useState();
  const additionalOptions = formFieldOptions.filter(
    (option) => !inputs.find((input) => input.name === option.name)
  );
  const addField = () => {
    setInputs([...inputs, selectedAdditionalOption]);
    setSelectedAdditionalOption(void 0);
  };
  const removeField = (index) => {
    const cloneInputs = [...inputs];
    const removedInput = cloneInputs.splice(index, 1);
    if (requiredInputs.includes(removedInput[0].name)) {
      const canRemoveRequiredField = cloneInputs.some(
        (item) => requiredInputs.includes(item.name)
      );
      if (!canRemoveRequiredField) {
        antd.Modal.info({
          title: i18n.t("Th\xF4ng b\xE1o"),
          content: noticeRequiredFieldText,
          centered: true,
          width: 500,
          zIndex: 9999
        });
        return;
      }
    }
    setInputs(cloneInputs);
  };
  const onChange = (value, option) => {
    setSelectedAdditionalOption(option);
  };
  const handleDragEnd = ({ active, over }) => {
    if (active.id !== over.id) {
      const _inputs = [...inputs];
      const oldIndex = _inputs.findIndex((el) => el.name === active.id);
      const newIndex = _inputs.findIndex((el) => el.name === over.id);
      setInputs(sortable.arrayMove(_inputs, oldIndex, newIndex));
    }
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      requiredInputs.length && /* @__PURE__ */ jsxRuntime.jsx("div", {
        style: { marginBottom: 8 },
        children: /* @__PURE__ */ jsxRuntime.jsxs(antd.Typography.Text, {
          type: "secondary",
          children: [
            i18n.t("*L\u01B0u \xFD:"),
            noticeRequiredFieldText
          ]
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx(core.DndContext, {
        modifiers: [
          modifiers.restrictToVerticalAxis,
          modifiers.restrictToWindowEdges,
          modifiers.restrictToFirstScrollableAncestor
        ],
        onDragEnd: handleDragEnd,
        children: /* @__PURE__ */ jsxRuntime.jsx(sortable.SortableContext, {
          items: inputs.map((el) => el.name),
          strategy: sortable.verticalListSortingStrategy,
          children: inputs.map((el, index) => /* @__PURE__ */ jsxRuntime.jsx(FormFieldSetting.FieldConfig, {
            activeField,
            setActiveField,
            input: el,
            index,
            onRemove: () => removeField(index),
            labelStyle: requiredInputs.includes(el.name) ? { color: "#e10202" } : void 0
          }, index))
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsxs("div", {
        children: [
          /* @__PURE__ */ jsxRuntime.jsx("h4", {
            children: i18n.t("Th\xEAm tr\u01B0\u1EDDng")
          }),
          /* @__PURE__ */ jsxRuntime.jsx(antd.Select, {
            options: additionalOptions,
            style: { width: "100%" },
            dropdownStyle: { zIndex: 9999 },
            placeholder: i18n.t("Ch\u1ECDn tr\u01B0\u1EDDng c\u1EA7n th\xEAm"),
            onChange,
            allowClear: true,
            value: selectedAdditionalOption,
            disabled: !additionalOptions.length,
            fieldNames: { label: "name", value: "name" }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(antd.Button, {
            style: {
              marginTop: 8,
              marginBottom: 8
            },
            type: "primary",
            block: true,
            onClick: addField,
            shape: "round",
            disabled: !selectedAdditionalOption,
            children: i18n.t("Th\xEAm tr\u01B0\u1EDDng")
          })
        ]
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
        propKey: propKey.PROP_KEY.ShowLabel,
        formItemProps: { label: i18n.t("Hi\u1EC3n th\u1ECB nh\xE3n") }
      })
    ]
  });
};

exports.FormFieldSettingCustom = FormFieldSettingCustom;
