// @tempi/core-elements/dist/components/form/v4/configs/OptionImageSetting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../../i18n.js');
var icons = require('@ant-design/icons');
var core = require('@dnd-kit/core');
var modifiers = require('@dnd-kit/modifiers');
var sortable = require('@dnd-kit/sortable');
var utilities = require('@dnd-kit/utilities');
var styled = require('@emotion/styled');
var coreEditor = require('@tempi/core-editor');
var React = require('react');
var propKey = require('../../../../constants/propKey.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const OptionImageSetting = ({
  index: inputIndex
}) => {
  const [options, setOptions] = coreEditor.useProp(
    propKey.PROP_KEY.joinKey(propKey.PROP_KEY.Inputs, inputIndex.toString(), propKey.PROP_KEY.Options)
  );
  const [activeOption, setActiveOption] = React.useState();
  const addOption = () => {
    const opts = options || [];
    const defaultText = i18n.t("L\u1EF1a ch\u1ECDn {{var0}}", { var0: opts.length + 1 });
    setOptions([
      ...opts,
      {
        label: "",
        value: defaultText
      }
    ]);
  };
  const removeOption = (index) => {
    const cloneOptions = [...options || []];
    cloneOptions.splice(index, 1);
    setOptions(cloneOptions);
  };
  const handleDragEnd = ({ active, over }) => {
    if (active.id !== (over == null ? void 0 : over.id)) {
      const _options = [...options || []];
      const oldIndex = _options.findIndex((el) => el.value === active.id);
      const newIndex = _options.findIndex((el) => el.value === (over == null ? void 0 : over.id));
      setOptions(sortable.arrayMove(_options, oldIndex, newIndex));
    }
  };
  return Array.isArray(options) ? /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
    label: i18n.t("Danh s\xE1ch l\u1EF1a ch\u1ECDn"),
    layout: "vertical",
    children: /* @__PURE__ */ jsxRuntime.jsxs("div", {
      style: { marginTop: 8, marginBottom: 16 },
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(core.DndContext, {
          modifiers: [
            modifiers.restrictToVerticalAxis,
            modifiers.restrictToWindowEdges,
            modifiers.restrictToFirstScrollableAncestor
          ],
          onDragEnd: handleDragEnd,
          children: /* @__PURE__ */ jsxRuntime.jsx(sortable.SortableContext, {
            items: options.map((el) => el.value),
            strategy: sortable.verticalListSortingStrategy,
            children: options.map((el, idx) => /* @__PURE__ */ jsxRuntime.jsx(OptionImageConfig, {
              inputIndex,
              activeOption: activeOption || 0,
              setActiveOption,
              option: el,
              index: idx,
              onRemove: () => removeOption(idx)
            }, idx))
          })
        }),
        /* @__PURE__ */ jsxRuntime.jsxs(StyledTextLink, {
          style: {
            display: "flex",
            justifyContent: "center",
            alignItems: "center",
            marginBottom: 8
          },
          onClick: addOption,
          children: [
            /* @__PURE__ */ jsxRuntime.jsx(icons.PlusOutlined, {
              style: { marginRight: 4 }
            }),
            i18n.t("Th\xEAm l\u1EF1a ch\u1ECDn")
          ]
        })
      ]
    })
  }) : null;
};
const OptionImageConfig = ({
  inputIndex,
  index,
  option,
  onRemove,
  activeOption,
  setActiveOption
}) => {
  const { setNodeRef, attributes, listeners, transform } = sortable.useSortable({
    id: option.value
  });
  return /* @__PURE__ */ jsxRuntime.jsx(jsxRuntime.Fragment, {
    children: /* @__PURE__ */ jsxRuntime.jsxs(StyledTagField, {
      ref: setNodeRef,
      style: { transform: utilities.CSS.Transform.toString(transform) },
      children: [
        /* @__PURE__ */ jsxRuntime.jsxs(StyledTagHeader, {
          onClick: () => {
            if (activeOption === index)
              setActiveOption(void 0);
            else
              setActiveOption(index);
          },
          children: [
            /* @__PURE__ */ jsxRuntime.jsxs("div", {
              className: "flex",
              children: [
                /* @__PURE__ */ jsxRuntime.jsx(StyledAction, __spreadProps(__spreadValues(__spreadValues({}, listeners), attributes), {
                  style: { cursor: "move", marginRight: 8 },
                  children: /* @__PURE__ */ jsxRuntime.jsx(icons.HolderOutlined, {
                    style: {
                      fontSize: 20
                    }
                  })
                })),
                /* @__PURE__ */ jsxRuntime.jsx("div", {
                  children: option.value || option.label
                })
              ]
            }),
            /* @__PURE__ */ jsxRuntime.jsx("div", {
              className: "flex",
              children: /* @__PURE__ */ jsxRuntime.jsx(StyledAction, {
                onClick: (e) => {
                  e.stopPropagation();
                  onRemove();
                },
                children: /* @__PURE__ */ jsxRuntime.jsx(icons.DeleteOutlined, {
                  style: {
                    fontSize: 20
                  }
                })
              })
            })
          ]
        }),
        activeOption === index && /* @__PURE__ */ jsxRuntime.jsxs(StyledTagContent, {
          children: [
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputConfig, {
              formItemProps: {
                label: i18n.t("Nh\xE3n")
              },
              propKey: propKey.PROP_KEY.joinKey(
                propKey.PROP_KEY.Inputs,
                inputIndex.toString(),
                propKey.PROP_KEY.Options,
                index.toString(),
                propKey.PROP_KEY.Label
              )
            }),
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputConfig, {
              formItemProps: {
                label: i18n.t("Gi\xE1 tr\u1ECB")
              },
              propKey: propKey.PROP_KEY.joinKey(
                propKey.PROP_KEY.Inputs,
                inputIndex.toString(),
                propKey.PROP_KEY.Options,
                index.toString(),
                propKey.PROP_KEY.Value
              )
            }),
            /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ImageConfig, {
              propKey: propKey.PROP_KEY.joinKey(
                propKey.PROP_KEY.Inputs,
                inputIndex.toString(),
                propKey.PROP_KEY.Options,
                index.toString(),
                propKey.PROP_KEY.Image,
                propKey.PROP_KEY.Src
              ),
              widthPropKey: propKey.PROP_KEY.joinKey(
                propKey.PROP_KEY.Inputs,
                inputIndex.toString(),
                propKey.PROP_KEY.Options,
                index.toString(),
                propKey.PROP_KEY.Image,
                propKey.PROP_KEY.GoogleImageParams,
                propKey.PROP_KEY.Width
              )
            })
          ]
        })
      ]
    }, index)
  });
};
const StyledTagField = styled__default["default"].div`
  justify-content: space-between;
  padding: 0.5rem;
  border-radius: 4px;
  border: 1px solid var(--border-color);
  margin-bottom: 8px;
`;
const StyledTagHeader = styled__default["default"].div`
  display: flex;
  flex-grow: 1;
  align-items: center;
  justify-content: space-between;
  cursor: pointer;
`;
const StyledTagContent = styled__default["default"].div`
  padding: 0.5rem;
`;
const StyledAction = styled__default["default"].div`
  background-color: transparent;
  cursor: pointer;
`;
const StyledTextLink = styled__default["default"].div(
  ({ theme }) => `
  cursor: pointer;
  color: ${theme.colorPrimary} !important;
`
);

exports.OptionImageSetting = OptionImageSetting;
