// @tempi/core-editor/dist/configs/EffectAttributes/EffectAttributes.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var antd = require('antd');
var AnimationConfig = require('../AnimationConfig/AnimationConfig.js');
var HoverConfig = require('../HoverConfig/HoverConfig.js');
var propKey = require('../../constants/propKey.js');

const { Panel } = antd.Collapse;
const EffectAttributes = ({
  propKey: propKey$1 = propKey.PROP_KEY.CommonStyle,
  separateDevice = false
}) => {
  return /* @__PURE__ */ jsxRuntime.jsxs(antd.Collapse, {
    defaultActiveKey: "AnimationEffect",
    expandIconPosition: "end",
    bordered: false,
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(Panel, {
        header: i18n.t("Hi\u1EC7u \u1EE9ng khi cu\u1ED9n t\u1EDBi"),
        children: /* @__PURE__ */ jsxRuntime.jsx(AnimationConfig.AnimationConfig, {
          propKey: propKey$1,
          separateDevice
        })
      }, "AnimationEffect"),
      /* @__PURE__ */ jsxRuntime.jsx(Panel, {
        header: i18n.t("Hi\u1EC7u \u1EE9ng khi tr\u1ECF chu\u1ED9t"),
        children: /* @__PURE__ */ jsxRuntime.jsx(HoverConfig.HoverConfig, {
          propKey: propKey$1,
          separateDevice
        })
      }, "HoverEffect")
    ]
  });
};

exports.EffectAttributes = EffectAttributes;
