// @tempi/core-editor/dist/configs/EffectAttributes/EffectAttributes.es.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { Collapse } from 'antd';
import { AnimationConfig } from '../AnimationConfig/AnimationConfig.es.js';
import { HoverConfig } from '../HoverConfig/HoverConfig.es.js';
import { PROP_KEY } from '../../constants/propKey.es.js';

const { Panel } = Collapse;
const EffectAttributes = ({
  propKey = PROP_KEY.CommonStyle,
  separateDevice = false
}) => {
  return /* @__PURE__ */ jsxs(Collapse, {
    defaultActiveKey: "AnimationEffect",
    expandIconPosition: "end",
    bordered: false,
    children: [
      /* @__PURE__ */ jsx(Panel, {
        header: t("Hi\u1EC7u \u1EE9ng khi cu\u1ED9n t\u1EDBi"),
        children: /* @__PURE__ */ jsx(AnimationConfig, {
          propKey,
          separateDevice
        })
      }, "AnimationEffect"),
      /* @__PURE__ */ jsx(Panel, {
        header: t("Hi\u1EC7u \u1EE9ng khi tr\u1ECF chu\u1ED9t"),
        children: /* @__PURE__ */ jsx(HoverConfig, {
          propKey,
          separateDevice
        })
      }, "HoverEffect")
    ]
  });
};

export { EffectAttributes };
