// @tempi/core-elements/dist/components/number-counter/constants.es.js
import { t } from '../../i18n.es.js';
import { NumberPosition, NumberCounterLayout, NumberType } from './type.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const DEFAULT_STYLE = {
  duration: 0,
  delay: 0,
  backgroundColor: "#efefef",
  foregroundColor: "#4caf50ff",
  textStyle: {
    fontSize: 16
  },
  numberStyle: {
    fontSize: 32
  }
};
const DEFAULT_BAR_STYLE = {
  barStyle: {
    numberPosition: NumberPosition.inside,
    height: 42
  },
  textStyle: __spreadProps(__spreadValues({}, DEFAULT_STYLE.textStyle), {
    textAlign: "left"
  }),
  numberStyle: __spreadProps(__spreadValues({}, DEFAULT_STYLE.numberStyle), {
    textAlign: "right",
    lineHeight: 42,
    color: "#fff"
  })
};
const DEFAULT_CIRCLE_STYLE = {
  circleStyle: {
    size: 250,
    strokeSize: 10
  },
  textStyle: __spreadProps(__spreadValues({}, DEFAULT_STYLE.textStyle), {
    textAlign: "center"
  }),
  numberStyle: __spreadProps(__spreadValues({}, DEFAULT_STYLE.numberStyle), {
    textAlign: "center",
    fontSize: 40
  })
};
const DEFAULT_NUMBER_STYLE = {
  textStyle: __spreadProps(__spreadValues({}, DEFAULT_STYLE.textStyle), {
    textAlign: "center"
  }),
  numberStyle: __spreadProps(__spreadValues({}, DEFAULT_STYLE.numberStyle), {
    textAlign: "center"
  })
};
const DEFAULT_NUMBER_COUNTER = __spreadValues(__spreadValues({
  textBefore: t("S\u1ED1 l\u01B0\u1EE3ng \u0111\u01A1n th\xE0nh c\xF4ng"),
  layout: NumberCounterLayout.bar,
  number: 89,
  numberType: NumberType.percent
}, DEFAULT_STYLE), DEFAULT_BAR_STYLE);

export { DEFAULT_BAR_STYLE, DEFAULT_CIRCLE_STYLE, DEFAULT_NUMBER_COUNTER, DEFAULT_NUMBER_STYLE, DEFAULT_STYLE };
