// @tempi/core-elements/dist/components/number-counter/selector.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var coreEditor = require('@tempi/core-editor');
var constants = require('./constants.js');
var quickSetting = require('./quick-setting.js');
var setting = require('./setting.js');
var ui = require('./ui.js');
var selector = require('../section/selector.js');
var selector$1 = require('../root/selector.js');

const NumberCounterEditor = coreEditor.withSelector(ui.NumberCounter, {
  displayName: "Progress",
  tag: ui.NumberCounter.displayName.toLowerCase(),
  advanceAttributes: coreEditor.AdvanceAttributes,
  quickSetting: quickSetting.NumberCounterQuickSetting,
  customAttributes: setting.NumberCounterSetting,
  props: constants.DEFAULT_NUMBER_COUNTER,
  rules: {
    canDrop: (targetNode) => {
      const ignoreTargets = [
        coreEditor.getEditorName({ RowEditor: selector.RowEditor }),
        coreEditor.getEditorName({ RootEditor: selector$1.RootEditor })
      ];
      return !ignoreTargets.includes(targetNode.data.name);
    }
  }
});

exports.NumberCounterEditor = NumberCounterEditor;
