// @tempi/core-elements/dist/components/list/v2/selector/ListItemContentV2/ListItemContentEditorV2.es.js
import { jsx } from 'react/jsx-runtime';
import { withSelector, useBuilder, AdvanceAttributes } from '@tempi/core-editor';
import { ParagraphQuickSetting } from '../../../../paragraph/quick-setting.es.js';
import { EditorQuickSetting } from '../../../../paragraph/editor-quick-setting.es.js';
import { DEFAULT_LIST } from '../../constants.es.js';
import { ListItemContentV2 } from '../../ui/ListItemContentV2.es.js';
import { TipTapEditorV2 } from '../../../../../configs/TipTapEditor/TipTapEditorV2.es.js';
import { PROP_KEY } from '../../../../../constants/propKey.es.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
const ListItemContentEditorV2 = withSelector(
  (props) => {
    const { track } = useBuilder();
    return /* @__PURE__ */ jsx(TipTapEditorV2, {
      propKey: PROP_KEY.Text,
      quickSetting: ParagraphQuickSetting,
      editorQuickSetting: EditorQuickSetting,
      track: () => track == null ? void 0 : track({
        regionName: "ListItemContentSetting",
        contentName: "quickSettingBtn"
      }),
      children: /* @__PURE__ */ jsx(ListItemContentV2, __spreadValues({}, props))
    });
  },
  {
    displayName: ListItemContentV2.displayName,
    tag: ListItemContentV2.displayName.toLowerCase(),
    advanceAttributes: AdvanceAttributes,
    props: DEFAULT_LIST.listItemContent,
    rules: {
      canDrag: () => false
    }
  }
);

export { ListItemContentEditorV2 };
