// @tempi/core-elements/dist/components/lucky-draw/v2/configs/LimitationConfig.es.js
import { jsx, Fragment, jsxs } from 'react/jsx-runtime';
import { t } from '../../../../i18n.es.js';
import styled from '@emotion/styled';
import { FormItem, SelectConfig, InputNumber } from '@tempi/core-editor';
import { Form, Radio, Button, DatePicker } from 'antd';
import dayjs from 'dayjs';
import { useState } from 'react';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const LimitationConfig = ({ limit, onSubmit }) => {
  const [form] = Form.useForm();
  const [loading, setLoading] = useState(false);
  const [limitType, setLimitType] = useState((limit == null ? void 0 : limit.type) || 1);
  const RESET_DURATIONS = [
    { value: "day", label: t("Ng\xE0y") },
    { value: "week", label: t("Tu\u1EA7n") },
    { value: "month", label: t("Th\xE1ng") }
  ];
  const onChangeLimitType = (e) => {
    setLimitType(e.target.value);
    form.resetFields();
  };
  const handleSubmitForm = async () => {
    setLoading(true);
    const dataBody = __spreadProps(__spreadValues({}, form.getFieldsValue()), {
      type: limitType
    });
    if (!!dataBody) {
      if (limitType === 1)
        delete dataBody.resetUnit;
      if (limitType === 2)
        dataBody.startAt = dayjs(form.getFieldValue("startAt")).toISOString();
      else
        delete dataBody.startAt;
      try {
        await (onSubmit == null ? void 0 : onSubmit(dataBody));
      } catch (error) {
      }
    }
    setTimeout(() => setLoading(false), 300);
  };
  const UserLimitConfig = () => /* @__PURE__ */ jsx(Fragment, {
    children: /* @__PURE__ */ jsx(StyledFormItem, {
      name: "spinTurn",
      label: t("L\u01B0\u1EE3t tham gia \u0111\u1ED1i \u0111a"),
      rules: [
        { required: true, message: t("H\xE3y nh\u1EADp l\u01B0\u1EE3t tham gia t\u1ED1i \u0111a") }
      ],
      children: /* @__PURE__ */ jsx(InputNumber, __spreadProps(__spreadValues({}, { min: 1, placeholder: t("Nh\u1EADp s\u1ED1") }), {
        style: { width: "100%" }
      }))
    })
  });
  const TimeLimitConfig = () => /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(StyledFormItem, {
        name: "spinTurn",
        label: t("L\u01B0\u1EE3t tham gia"),
        rules: [{ required: true, message: t("H\xE3y nh\u1EADp l\u01B0\u1EE3t tham gia") }],
        children: /* @__PURE__ */ jsx(InputNumber, __spreadProps(__spreadValues({}, { min: 1, placeholder: t("Nh\u1EADp s\u1ED1") }), {
          style: { width: "100%" }
        }))
      }),
      /* @__PURE__ */ jsx(StyledFormItem, {
        name: "startAt",
        label: t("Th\u1EDDi gian b\u1EAFt \u0111\u1EA7u"),
        rules: [{ required: true, message: t("H\xE3y ch\u1ECDn th\u1EDDi gian b\u1EAFt \u0111\u1EA7u") }],
        children: /* @__PURE__ */ jsx(DatePicker, {
          showTime: true,
          format: "DD/MM/YYYY HH:mm:ss",
          placeholder: "dd/mm/yyyy hh:MM:ss",
          style: { width: "100%" }
        })
      }),
      /* @__PURE__ */ jsxs(StyledResetWrapper, {
        label: t("luckydraw_config.condition.reset_after"),
        children: [
          /* @__PURE__ */ jsx(StyledFormItem, {
            noStyle: true,
            name: "resetAfter",
            rules: [{ required: true, message: t("H\xE3y nh\u1EADp gi\xE1 tr\u1ECB") }],
            children: /* @__PURE__ */ jsx(InputNumber, __spreadProps(__spreadValues({}, { min: 1, placeholder: t("Nh\u1EADp s\u1ED1") }), {
              style: { width: "100%", marginBottom: 8 }
            }))
          }),
          /* @__PURE__ */ jsx(StyledFormItem, {
            noStyle: true,
            name: "resetUnit",
            rules: [{ required: true, message: t("H\xE3y ch\u1ECDn \u0111\u01A1n v\u1ECB") }],
            children: /* @__PURE__ */ jsx(StyledSelectConfig, {
              options: RESET_DURATIONS,
              selectProps: { defaultValue: "day" }
            })
          })
        ]
      })
    ]
  });
  const StartLimitConfig = () => /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(StyledFormItem, {
        name: "spinTurn",
        label: t("L\u01B0\u1EE3t tham gia"),
        rules: [{ required: true, message: t("H\xE3y nh\u1EADp l\u01B0\u1EE3t tham gia") }],
        children: /* @__PURE__ */ jsx(InputNumber, __spreadProps(__spreadValues({}, { min: 1, placeholder: t("Nh\u1EADp s\u1ED1") }), {
          style: { width: "100%" }
        }))
      }),
      /* @__PURE__ */ jsxs(StyledResetWrapper, {
        label: t("luckydraw_config.condition.reset_after"),
        children: [
          /* @__PURE__ */ jsx(StyledFormItem, {
            noStyle: true,
            name: "resetAfter",
            rules: [{ required: true, message: t("H\xE3y nh\u1EADp gi\xE1 tr\u1ECB") }],
            children: /* @__PURE__ */ jsx(InputNumber, __spreadProps(__spreadValues({}, { min: 1, placeholder: t("Nh\u1EADp s\u1ED1") }), {
              style: { width: "100%", marginBottom: 8 }
            }))
          }),
          /* @__PURE__ */ jsx(StyledFormItem, {
            noStyle: true,
            name: "resetUnit",
            rules: [{ required: true, message: t("H\xE3y ch\u1ECDn \u0111\u01A1n v\u1ECB") }],
            children: /* @__PURE__ */ jsx(StyledSelectConfig, {
              options: RESET_DURATIONS
            })
          })
        ]
      })
    ]
  });
  return /* @__PURE__ */ jsx(Fragment, {
    children: /* @__PURE__ */ jsxs(Form, {
      form,
      onFinish: handleSubmitForm,
      initialValues: __spreadProps(__spreadValues({}, limit), {
        startAt: dayjs((limit == null ? void 0 : limit.startAt) || dayjs()),
        resetUnit: limit.resetUnit || "day"
      }),
      layout: "vertical",
      children: [
        /* @__PURE__ */ jsx(FormItem, {
          label: t("Gi\u1EDBi h\u1EA1n tham gia"),
          layout: "vertical",
          children: /* @__PURE__ */ jsx(StyledLimitationConfig, {
            children: /* @__PURE__ */ jsxs(Radio.Group, {
              onChange: onChangeLimitType,
              value: limitType,
              children: [
                /* @__PURE__ */ jsx(StyledRadio, {
                  value: 1,
                  children: t("Gi\u1EDBi h\u1EA1n theo m\u1ED7i ng\u01B0\u1EDDi ch\u01A1i")
                }),
                limitType === 1 && /* @__PURE__ */ jsx(UserLimitConfig, {}),
                /* @__PURE__ */ jsx(StyledRadio, {
                  value: 2,
                  children: t("T\xEDnh theo m\u1ED1c th\u1EDDi gian c\u1ED1 \u0111\u1ECBnh")
                }),
                limitType === 2 && /* @__PURE__ */ jsx(TimeLimitConfig, {}),
                /* @__PURE__ */ jsx(StyledRadio, {
                  value: 3,
                  children: t("T\xEDnh theo th\u1EDDi \u0111i\u1EC3m tham gia")
                }),
                limitType === 3 && /* @__PURE__ */ jsx(StartLimitConfig, {})
              ]
            })
          })
        }),
        /* @__PURE__ */ jsx(Button, {
          loading,
          type: "primary",
          htmlType: "submit",
          style: { width: 120 },
          block: true,
          children: t("L\u01B0u")
        })
      ]
    })
  });
};
const StyledLimitationConfig = styled.div`
  margin-top: 16px;
`;
const StyledRadio = styled(Radio)`
  margin-bottom: 12px;
  font-size: 13px;
  .ant-radio-wrapper {
    display: flex;
    align-items: center;
  }
  .ant-radio-inner {
    background-color: transparent;
  }
`;
const StyledFormItem = styled(Form.Item)`
  margin-bottom: 4px;
`;
const StyledResetWrapper = styled(StyledFormItem)`
  .ant-form-item-control-input-content {
    display: grid;
    grid-template-columns: 1fr 1fr;
    align-items: center;
    gap: 8px;
  }

  .custom-form-item {
    margin-bottom: unset;
  }
`;
const StyledSelectConfig = styled(SelectConfig)`
  width: 100% !important;
  margin-bottom: 0;
`;

export { LimitationConfig };
