// @tempi/core-elements/dist/components/lucky-draw/v2/configs/PrizeListConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../../i18n.js');
var icons = require('@ant-design/icons');
var styled = require('@emotion/styled');
var antd = require('antd');
var React = require('react');
var coreEditor = require('@tempi/core-editor');
var LuckyDrawContext = require('../context/LuckyDrawContext.js');
var PrizeConfig = require('./PrizeConfig.js');
var propKey = require('../../../../constants/propKey.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

const PrizeListConfig = () => {
  const [activeIndex, setActiveIndex] = React.useState(null);
  const [showCreateForm, setShowCreateForm] = React.useState(false);
  const { items, loading } = React.useContext(LuckyDrawContext.LuckyDrawContext);
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
        propKey: propKey.PROP_KEY.PrizeAngle,
        formItemProps: {
          label: i18n.t("G\xF3c quay tr\xFAng th\u01B0\u1EDFng")
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
        propKey: propKey.PROP_KEY.joinKey(
          propKey.PROP_KEY.CanvasConfig,
          propKey.PROP_KEY.PerpendicularPrize
        ),
        options: [
          { label: i18n.t("common.horizontal"), value: false },
          { label: i18n.t("common.vertical"), value: true }
        ],
        formItemProps: {
          label: i18n.t("H\u01B0\u1EDBng gi\u1EA3i th\u01B0\u1EDFng")
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
        propKey: propKey.PROP_KEY.ShowTurn,
        formItemProps: {
          label: i18n.t("Hi\u1EC3n th\u1ECB s\u1ED1 l\u01B0\u1EE3t quay")
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsxs(StyledLabel, {
        children: [
          i18n.t("Danh s\xE1ch gi\u1EA3i th\u01B0\u1EDFng"),
          " ",
          /* @__PURE__ */ jsxRuntime.jsx(antd.Popover, {
            content: i18n.t("T\u1ED5ng x\xE1c su\u1EA5t tr\xFAng th\u01B0\u1EDFng ph\u1EA3i b\u1EB1ng 100%"),
            zIndex: 1100,
            children: /* @__PURE__ */ jsxRuntime.jsx(StyledIcon, {
              name: "AntdQuestionCircleOutlined"
            })
          })
        ]
      }),
      /* @__PURE__ */ jsxRuntime.jsxs(antd.Spin, {
        spinning: loading,
        children: [
          items == null ? void 0 : items.map((item, index) => /* @__PURE__ */ jsxRuntime.jsx(PrizeConfig.PrizeConfig, {
            item,
            isActive: index === activeIndex,
            setActive: () => setActiveIndex(index)
          }, item == null ? void 0 : item.id)),
          /* @__PURE__ */ jsxRuntime.jsx(PrizeConfig.PrizeConfig, {
            isActive: showCreateForm,
            setActive: () => {
              setActiveIndex(null);
            },
            onClose: () => {
              setShowCreateForm(false);
            },
            isCreate: true
          }),
          /* @__PURE__ */ jsxRuntime.jsx(StyledBtn, {
            block: true,
            type: "primary",
            icon: /* @__PURE__ */ jsxRuntime.jsx(icons.PlusOutlined, {}),
            onClick: () => setShowCreateForm(true),
            children: i18n.t("Th\xEAm ph\u1EA7n qu\xE0")
          })
        ]
      })
    ]
  });
};
const StyledLabel = styled__default["default"].div`
  font-weight: 500;
  margin: 20px 0 10px 0;
`;
const StyledBtn = styled__default["default"](antd.Button)`
  margin-bottom: 8px;
  display: flex;
  justify-content: center;
  align-items: center;
  span {
    margin-right: 4px;
  }
`;
const StyledIcon = styled__default["default"](icons.QuestionCircleOutlined)`
  font-size: 1.2em;
  color: #82869e;

  &:hover {
    color: var(--text-color);
  }
`;

exports.PrizeListConfig = PrizeListConfig;
