// @tempi/core-elements/dist/components/lucky-draw/v2/configs/PrizeConfig.es.js
import { jsx, jsxs } from 'react/jsx-runtime';
import { t } from '../../../../i18n.es.js';
import { DeleteOutlined } from '@ant-design/icons';
import styled from '@emotion/styled';
import { useBuilder, Input, InputNumber, ImageFormItem } from '@tempi/core-editor';
import { Form, Spin, Divider, Button } from 'antd';
import { useState, useContext, useEffect } from 'react';
import { LuckyDrawContext } from '../context/LuckyDrawContext.es.js';
import { ApiConfig } from '../../../../configs/ApiConfig/ApiConfig.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const PrizeConfig = ({ item, setActive, isActive, isCreate = false, onClose }) => {
  var _a, _b;
  const [isOpen, setOpen] = useState(isActive);
  const [form] = Form.useForm();
  const [loading, setLoading] = useState(false);
  const { id, condition, updateItem, addItem, removeItem } = useContext(LuckyDrawContext);
  const { upsertWheelItem, deleteWheelItem } = useBuilder();
  const handleSubmit = async (values) => {
    var _a2, _b2;
    setLoading(true);
    if (item == null ? void 0 : item.id) {
      const updatedItem = __spreadProps(__spreadValues({
        id: item == null ? void 0 : item.id
      }, values), {
        wheelId: id
      });
      if (!((_a2 = updatedItem == null ? void 0 : updatedItem.image) == null ? void 0 : _a2.src)) {
        updatedItem.image = null;
      }
      await upsertWheelItem(updatedItem).then(() => {
        updateItem(updatedItem);
      });
    } else {
      const newItem = __spreadProps(__spreadValues({}, values), { wheelId: id });
      if (!((_b2 = newItem == null ? void 0 : newItem.image) == null ? void 0 : _b2.src)) {
        delete newItem.image;
      }
      await upsertWheelItem(newItem).then((id2) => {
        addItem(__spreadProps(__spreadValues({}, newItem), { id: id2 }));
        form.resetFields();
      });
    }
    setLoading(false);
    setOpen(false);
    onClose == null ? void 0 : onClose();
  };
  const handleClose = () => {
    form.resetFields();
    setOpen(false);
    onClose == null ? void 0 : onClose();
  };
  const handleDelete = async (id2) => {
    if (confirm(t("B\u1EA1n ch\u1EAFc ch\u1EAFn mu\u1ED1n x\xF3a ph\u1EA7n qu\xE0 n\xE0y?"))) {
      setLoading(true);
      try {
        await deleteWheelItem(id2).then(() => {
          removeItem(id2);
        });
      } catch (err) {
      }
      setLoading(false);
    }
  };
  useEffect(() => {
    if (!isActive) {
      setOpen(false);
    } else {
      setOpen(true);
    }
  }, [isActive]);
  return /* @__PURE__ */ jsx("div", {
    style: { borderRadius: 5, overflow: "hidden" },
    children: /* @__PURE__ */ jsx(Spin, {
      spinning: loading,
      children: /* @__PURE__ */ jsxs(Wrapper, {
        className: isOpen ? "open" : "close",
        isCreate,
        isOpen,
        children: [
          !isCreate && /* @__PURE__ */ jsxs(PrizeHeader, {
            onClick: () => {
              if (!isOpen) {
                setActive();
              }
              setOpen(!isOpen);
            },
            children: [
              /* @__PURE__ */ jsx(StyledLabel, {
                title: item == null ? void 0 : item.name,
                children: item == null ? void 0 : item.name
              }),
              /* @__PURE__ */ jsxs("div", {
                style: { color: "#82869e", marginRight: 8 },
                children: [
                  (item == null ? void 0 : item.percentage) || 0,
                  "%"
                ]
              }),
              /* @__PURE__ */ jsx(StyledIcon, {
                onClick: (e) => {
                  e.stopPropagation();
                  handleDelete(item.id);
                }
              })
            ]
          }),
          /* @__PURE__ */ jsxs("div", {
            style: { padding: "8px 0" },
            children: [
              !isCreate && /* @__PURE__ */ jsx(Divider, {
                style: {
                  borderColor: "var(--border-color)",
                  margin: "0 0 12px 0"
                }
              }),
              /* @__PURE__ */ jsxs(Form, {
                form,
                name: "prize",
                onFinish: handleSubmit,
                initialValues: item,
                children: [
                  /* @__PURE__ */ jsx(StyledFormItem, {
                    name: "name",
                    label: t("T\xEAn qu\xE0"),
                    rules: [
                      {
                        required: true,
                        message: t("T\xEAn qu\xE0 kh\xF4ng \u0111\u01B0\u1EE3c b\u1ECF tr\u1ED1ng")
                      }
                    ],
                    children: /* @__PURE__ */ jsx(Input, {
                      defaultValue: item == null ? void 0 : item.name
                    })
                  }),
                  /* @__PURE__ */ jsx(StyledFormItem, {
                    name: "percentage",
                    label: t("T\u1EC9 l\u1EC7 tr\xFAng th\u01B0\u1EDFng (%)"),
                    rules: [
                      {
                        required: true,
                        message: t("T\u1EC9 l\u1EC7 tr\xFAng th\u01B0\u1EDFng kh\xF4ng \u0111\u01B0\u1EE3c b\u1ECF tr\u1ED1ng")
                      }
                    ],
                    children: /* @__PURE__ */ jsx(InputNumber, {
                      min: 0,
                      max: 100
                    })
                  }),
                  /* @__PURE__ */ jsx(StyledFormItem, {
                    name: "request",
                    noStyle: true,
                    children: /* @__PURE__ */ jsx(ApiConfig, {
                      data: item == null ? void 0 : item.request,
                      onSubmit: async (request) => {
                        form.setFieldValue("request", request);
                      },
                      attributes: [
                        "prizeName",
                        "spinAt",
                        ...((_a = condition == null ? void 0 : condition.form) == null ? void 0 : _a.fields.map((field) => field.name)) || []
                      ]
                    })
                  }),
                  /* @__PURE__ */ jsx(StyledFormItem, {
                    name: ["image", "src"],
                    noStyle: true,
                    children: /* @__PURE__ */ jsx(ImageFormItem, {
                      defaultValue: (_b = item == null ? void 0 : item.image) == null ? void 0 : _b.src,
                      onChange: (src) => {
                        form.setFieldValue(["image", "src"], src);
                      }
                    })
                  }),
                  /* @__PURE__ */ jsx(StyledFormItem, {
                    name: ["image", "width"],
                    label: t("Chi\u1EC1u r\u1ED9ng \u1EA3nh (px)"),
                    children: /* @__PURE__ */ jsx(InputNumber, {})
                  }),
                  /* @__PURE__ */ jsx(StyledFormItem, {
                    name: "quantity",
                    label: t("S\u1ED1 l\u01B0\u1EE3ng gi\u1EA3i th\u01B0\u1EDFng"),
                    children: /* @__PURE__ */ jsx(InputNumber, {})
                  }),
                  /* @__PURE__ */ jsx(Divider, {
                    style: {
                      borderColor: "var(--border-color)",
                      margin: "0 0 12px 0"
                    }
                  }),
                  /* @__PURE__ */ jsxs("div", {
                    style: { display: "flex", gap: 25 },
                    children: [
                      /* @__PURE__ */ jsx(Button, {
                        type: "text",
                        onClick: handleClose,
                        style: {
                          flex: 1,
                          borderRadius: 4
                        },
                        children: t("common.cancel")
                      }),
                      /* @__PURE__ */ jsx(Button, {
                        type: "primary",
                        onClick: () => form.submit(),
                        loading,
                        style: { flex: 1, borderRadius: 4 },
                        children: t("L\u01B0u")
                      })
                    ]
                  })
                ]
              })
            ]
          })
        ]
      })
    })
  });
};
const PrizeHeader = styled.div`
  font-weight: 500;
  user-select: none;
  cursor: pointer;
  display: flex;
  justify-content: space-between;
  width: 100%;
`;
const StyledLabel = styled.div`
  flex-grow: 1;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  padding-right: 0.5em;
`;
const StyledIcon = styled(DeleteOutlined)`
  font-size: 1.2em;
  color: #82869e;

  &:hover {
    color: var(--text-color);
  }
`;
const Wrapper = styled.div(
  ({ isCreate, isOpen }) => `

${(!isCreate || isOpen) && `
  padding: 10px 12px;
  border: 1px solid var(--border-color);
  margin-bottom: 8px;
  `}

padding-right: 12px;
padding-left: 12px;

border-radius: 5px;
overflow: hidden;
transition: 0.5s all;

&.close {
  max-height: ${isCreate ? "0px" : "42px"};
}

&.open {
  max-height: 600px;
}
`
);
const StyledFormItem = styled(Form.Item)`
  margin: 0 0 8px 0;

  .ant-form-item-label {
    flex-grow: 1;
    text-align: left;
  }

  .ant-form-item-control,
  .ant-input-number {
    width: 110px !important;
  }

  .ant-form-horizontal .ant-form-item-control,
  .ant-form-item-control {
    flex: unset;
  }
`;

export { PrizeConfig };
