// @tempi/core-elements/dist/components/lucky-draw/v2/configs/ResponseConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../../i18n.js');
var icons = require('@ant-design/icons');
var styled = require('@emotion/styled');
var coreEditor = require('@tempi/core-editor');
var antd = require('antd');
var React = require('react');
var uuid = require('uuid');
var constants = require('../constants.js');
var ResponseCustomConfig = require('./ResponseCustomConfig.js');
var propKey = require('../../../../constants/propKey.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

const ResponseConfig = () => {
  const { actions, addBlock, nodes } = coreEditor.useEditor((state) => ({
    nodes: state.nodes
  }));
  const { id: currentNodeId, childNodes } = coreEditor.useNode((node) => {
    return {
      childNodes: node.data.nodes
    };
  });
  const [prizePopupId, setPrizePopupId] = coreEditor.useProp(
    propKey.PROP_KEY.joinKey(propKey.PROP_KEY.Response, propKey.PROP_KEY.Prize)
  );
  const [outOfTurnPopupId, setOutOfTurnPopupId] = coreEditor.useProp(
    propKey.PROP_KEY.joinKey(propKey.PROP_KEY.Response, propKey.PROP_KEY.OutOfTurn)
  );
  const [outOfPrizePopupId, setOutOfPrizePopupId] = coreEditor.useProp(
    propKey.PROP_KEY.joinKey(propKey.PROP_KEY.Response, propKey.PROP_KEY.OutOfPrize)
  );
  const handleSwitchPrizePopup = async (flag) => {
    if (flag) {
      const node = coreEditor.deserializePB(
        constants.DEFAULT_POPUP_PRIZE_V2(
          i18n.t("Popup tr\xFAng gi\u1EA3i ({{uuid()}})", { "uuid()": uuid.v4() })
        )
      );
      const newNode = await addBlock(
        node,
        childNodes.length,
        currentNodeId
      );
      setPrizePopupId(newNode.rootNodeId);
    } else {
      if (nodes[prizePopupId]) {
        actions.delete(prizePopupId);
        setPrizePopupId(null);
      }
    }
  };
  const handleSwitchOutOfTurnPopup = async (flag) => {
    if (flag) {
      const node = coreEditor.deserializePB(
        constants.DEFAULT_POPUP_OUT_OF_MOVE_V2(
          i18n.t("Popup h\u1EBFt l\u01B0\u1EE3t ({{uuid()}})", { "uuid()": uuid.v4() })
        )
      );
      const newNode = await addBlock(
        node,
        childNodes.length,
        currentNodeId
      );
      setOutOfTurnPopupId(newNode.rootNodeId);
    } else {
      if (nodes[outOfTurnPopupId]) {
        actions.delete(outOfTurnPopupId);
        setOutOfTurnPopupId(null);
      }
    }
  };
  const handleSwitchOutOfPrizePopup = async (flag) => {
    if (flag) {
      const node = coreEditor.deserializePB(
        constants.DEFAULT_POPUP_OUT_OF_PRIZE_V2(
          i18n.t("Popup h\u1EBFt qu\xE0 ({{uuid()}})", { "uuid()": uuid.v4() })
        )
      );
      const newNode = await addBlock(
        node,
        childNodes.length,
        currentNodeId
      );
      setOutOfPrizePopupId(newNode.rootNodeId);
    } else {
      if (nodes[outOfPrizePopupId]) {
        actions.delete(outOfPrizePopupId);
        setOutOfPrizePopupId(null);
      }
    }
  };
  React.useEffect(() => {
    if (prizePopupId === void 0) {
      handleSwitchPrizePopup(true);
    }
    if (outOfTurnPopupId === void 0) {
      handleSwitchOutOfTurnPopup(true);
    }
    if (outOfPrizePopupId === void 0) {
      handleSwitchOutOfPrizePopup(true);
    }
  }, []);
  const showPrizeDetail = React.useMemo(() => {
    return prizePopupId && (nodes == null ? void 0 : nodes[prizePopupId]);
  }, [prizePopupId, nodes]);
  const showOutOfTurnDetail = React.useMemo(() => {
    return outOfTurnPopupId && (nodes == null ? void 0 : nodes[outOfTurnPopupId]);
  }, [outOfTurnPopupId, nodes]);
  const showOutOfPrizeDetail = React.useMemo(() => {
    return outOfPrizePopupId && (nodes == null ? void 0 : nodes[outOfPrizePopupId]);
  }, [outOfPrizePopupId, nodes]);
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
        label: /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
          children: [
            /* @__PURE__ */ jsxRuntime.jsx("span", {
              children: i18n.t("Popup tr\xFAng th\u01B0\u1EDFng")
            }),
            !!showPrizeDetail && /* @__PURE__ */ jsxRuntime.jsx(StyledIcon, {
              onClick: () => actions.selectNode(prizePopupId)
            })
          ]
        }),
        children: /* @__PURE__ */ jsxRuntime.jsx(antd.Switch, {
          checked: !!showPrizeDetail,
          onChange: handleSwitchPrizePopup
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
        label: /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
          children: [
            /* @__PURE__ */ jsxRuntime.jsx("span", {
              children: i18n.t("Popup h\u1EBFt l\u01B0\u1EE3t")
            }),
            !!showOutOfTurnDetail && /* @__PURE__ */ jsxRuntime.jsx(StyledIcon, {
              onClick: () => actions.selectNode(outOfTurnPopupId)
            })
          ]
        }),
        children: /* @__PURE__ */ jsxRuntime.jsx(antd.Switch, {
          checked: !!showOutOfTurnDetail,
          onChange: handleSwitchOutOfTurnPopup
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
        label: /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
          children: [
            /* @__PURE__ */ jsxRuntime.jsx("span", {
              children: i18n.t("Popup h\u1EBFt qu\xE0")
            }),
            !!showOutOfPrizeDetail && /* @__PURE__ */ jsxRuntime.jsx(StyledIcon, {
              onClick: () => actions.selectNode(outOfPrizePopupId)
            })
          ]
        }),
        children: /* @__PURE__ */ jsxRuntime.jsx(antd.Switch, {
          checked: !!showOutOfPrizeDetail,
          onChange: handleSwitchOutOfPrizePopup
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx(antd.Divider, {
        style: {
          margin: "12px 0"
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
        layout: "horizontal",
        label: i18n.t("N\xE2ng cao: T\u1EF1 c\u1EA5u h\xECnh ph\u1EA3n h\u1ED3i d\u1EF1a theo response code"),
        children: /* @__PURE__ */ jsxRuntime.jsx("div", {})
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
        children: /* @__PURE__ */ jsxRuntime.jsx(ResponseCustomConfig.ResponseCustomConfig, {})
      })
    ]
  });
};
const StyledIcon = styled__default["default"](icons.EditOutlined)(
  ({ theme }) => `
  cursor: pointer;
  color: ${theme.colorPrimary} !important;
  font-size: 20px;
  margin-left: 0.25em;
`
);

exports.ResponseConfig = ResponseConfig;
