// @tempi/core-elements/dist/components/slider/v1/constants.es.js
import { SliderTypeEnum } from './type.es.js';

const DEFAULT_GALLERY_SLIDESHOW = {
  type: SliderTypeEnum.image,
  items: [
    {
      src: "https://lh3.googleusercontent.com/Jj-xgwH9NZCn-WFxpJlkBnfAhTJ5smgyxBUjX9QyUR2b-gSHqHjG3XOIUO7yVAcOmV4dQsNA1KvLDaqZEP9hI7Cq-n6ty_4BHg=w1232-rw",
      googleImageParams: {
        width: 1200
      }
    },
    {
      src: "https://lh3.googleusercontent.com/_2Arqz2Hu80YPgo7TeHDH0smg0jMPimHIuLRZdMHq7Jb4fg7gfnxJofMmIBD8K-b5bPmhEzuLa2hj_lzafkxiSg0zeh8-74VXQ=w1232-rw",
      googleImageParams: {
        width: 1200
      }
    },
    {
      src: "https://lh3.googleusercontent.com/Eci1c3zs_AgF0kahUvV-mjfFGvODH5HaT4bLmeui9f2vF9cZTYIscS7spiQBTXWUPLmH86ysCLtxgCmTYs40gRhsnriBVaf_OA=w1232-rw",
      googleImageParams: {
        width: 1200
      }
    }
  ],
  slidesPerView: 1,
  slidesPerColumn: 1,
  navigation: true,
  navigationStyle: void 0,
  pagination: true,
  paginationStyle: void 0,
  paginationColor: "rgba(37, 141, 255, 1)",
  paginationDeactiveColor: "rgba(221, 221, 221, 1)",
  autoplay: false,
  loop: false,
  spaceBetween: 12,
  speed: 3e3,
  pauseOnMouseEnter: true,
  virtual: false,
  crop: true,
  showThumbnail: false,
  allowSetupThumbnail: true
};

export { DEFAULT_GALLERY_SLIDESHOW };
