// @tempi/core-elements/dist/components/slider/v1/selector.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import styled from '@emotion/styled';
import { withSelector, AdvanceAttributes, getEditorName } from '@tempi/core-editor';
import { DEFAULT_GALLERY_SLIDESHOW } from './constants.es.js';
import { SliderQuickSetting } from './quick-setting.es.js';
import { SliderSetting } from './setting.es.js';
import { SliderStyleSetting } from './style-setting.es.js';
import { Slider } from './ui.es.js';
import { RowEditor } from '../../section/selector.es.js';
import { RootEditor } from '../../root/selector.es.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
const EditorOv = styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 1;
  cursor: pointer;
`;
const Inner = (props) => {
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(EditorOv, {}),
      /* @__PURE__ */ jsx(Slider, __spreadValues(__spreadValues({
        isEditingMode: true
      }, props), { allowTouchMove: false }))
    ]
  });
};
const SliderEditor = withSelector(
  Inner,
  {
    displayName: "Thanh tr\u01B0\u1EE3t",
    tag: Slider.displayName.toLowerCase(),
    advanceAttributes: AdvanceAttributes,
    quickSetting: SliderQuickSetting,
    commonAttributes: SliderStyleSetting,
    customAttributes: SliderSetting,
    props: DEFAULT_GALLERY_SLIDESHOW,
    rules: {
      canDrop: (targetNode) => {
        const ignoreTargets = [
          getEditorName({ RowEditor }),
          getEditorName({ RootEditor })
        ];
        return !ignoreTargets.includes(targetNode.data.name);
      }
    }
  },
  {
    style: {
      position: "relative"
    }
  }
);

export { SliderEditor };
