// @tempi/core-editor/dist/configs/AdvanceAttributes/AdvanceAttributes.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { SpacingAttribute } from './SpacingAttribute.es.js';
import { ResponsiveAttribute } from './ResponsiveAttribute.es.js';
import { AttributeId } from './AttributeId.es.js';
import { Divider, Collapse, Space, Typography } from 'antd';
import { TimingAttributes } from './TimingAttibute.es.js';
import { HoverConfig } from '../HoverConfig/HoverConfig.es.js';
import { useBuilder } from '../../contexts/BuilderContext.es.js';
import { PROP_KEY } from '../../constants/propKey.es.js';

const { Panel } = Collapse;
const AdvanceAttributes = ({
  propKey = PROP_KEY.CommonStyle,
  separateDevice = false,
  readOnlyElementId
}) => {
  const { allowConfigResponsive } = useBuilder();
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(AttributeId, {
        readOnlyElementId
      }),
      /* @__PURE__ */ jsx(Divider, {
        style: {
          margin: "12px 0"
        }
      }),
      /* @__PURE__ */ jsxs(Collapse, {
        defaultActiveKey: [
          "Spacing",
          "ResponsiveVisibility",
          "TimingVisibility"
        ],
        expandIconPosition: "end",
        bordered: false,
        children: [
          /* @__PURE__ */ jsx(Panel, {
            header: t("Kho\u1EA3ng c\xE1ch"),
            children: /* @__PURE__ */ jsxs(Space, {
              direction: "vertical",
              children: [
                /* @__PURE__ */ jsx(Typography.Text, {
                  style: { marginBottom: 8 },
                  type: "warning",
                  children: t(
                    "Ch\xFA \xFD: Margin v\xE0 Padding qu\xE1 cao c\xF3 th\u1EC3 l\xE0m v\u1EE1 thi\u1EBFt k\u1EBF tr\xEAn \u0111i\u1EC7n tho\u1EA1i"
                  )
                }),
                /* @__PURE__ */ jsx(SpacingAttribute, {
                  title: "Margin",
                  separateDevice,
                  propKey: PROP_KEY.joinKey(propKey, PROP_KEY.Margin)
                }),
                /* @__PURE__ */ jsx(SpacingAttribute, {
                  title: "Padding",
                  separateDevice,
                  propKey: PROP_KEY.joinKey(propKey, PROP_KEY.Padding)
                })
              ]
            })
          }, "Spacing"),
          propKey === PROP_KEY.CommonStyle && /* @__PURE__ */ jsxs(Fragment, {
            children: [
              allowConfigResponsive && /* @__PURE__ */ jsx(Panel, {
                header: t("\u1EA8n/Hi\u1EC3n th\u1ECB tr\xEAn thi\u1EBFt b\u1ECB"),
                children: /* @__PURE__ */ jsx(ResponsiveAttribute, {})
              }, "ResponsiveVisibility"),
              /* @__PURE__ */ jsx(Panel, {
                header: t("\u0110\u1EB7t gi\u1EDD hi\u1EC3n th\u1ECB"),
                children: /* @__PURE__ */ jsx(TimingAttributes, {})
              }, "TimingVisibility")
            ]
          }),
          /* @__PURE__ */ jsx(Panel, {
            header: t("Hi\u1EC7u \u1EE9ng khi di chu\u1ED9t v\xE0o"),
            children: /* @__PURE__ */ jsx(HoverConfig, {
              propKey,
              separateDevice
            })
          }, "HoverEffect")
        ]
      })
    ]
  });
};

export { AdvanceAttributes };
