// @tempi/core-editor/dist/configs/AdvanceAttributes/AttributeId.es.js
import { jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { useNode } from '@tempi/dnd-core';
import { useState } from 'react';
import { useEditor } from '../../hooks/useEditor.es.js';
import { useProp } from '../../hooks/useProp.es.js';
import { InputConfig } from '../InputConfig/InputConfig.es.js';

const AttributeId = ({ readOnlyElementId = false }) => {
  const ELEMENT_ID = {
    label: t("M\xE3 th\xE0nh ph\u1EA7n"),
    existed: t("M\xE3 \u0111\xE3 t\u1ED3n t\u1EA1i")
  };
  const { id: nodeId } = useNode();
  const { query } = useEditor();
  const [elementId, setElementId] = useProp("id");
  const [warning, setWarning] = useState("");
  const validateElementId = (value) => {
    let warning2 = "";
    if (value) {
      const nodes = query.getSerializedNodes();
      const existedIds = Object.keys(nodes).filter(
        (key) => {
          var _a;
          return value === ((_a = nodes[key].props) == null ? void 0 : _a.id) && key !== nodeId;
        }
      );
      if (existedIds.length)
        warning2 = ELEMENT_ID.existed;
    }
    setWarning(warning2);
  };
  const onChangeId = (e) => {
    const value = e.target.value.trim() || void 0;
    validateElementId(value);
    setElementId(value);
  };
  return /* @__PURE__ */ jsx(InputConfig, {
    formItemProps: {
      label: t("M\xE3 th\xE0nh ph\u1EA7n"),
      warning,
      style: { margin: "0.5rem" }
    },
    inputProps: {
      value: elementId,
      onChange: onChangeId,
      placeholder: t("Nh\u1EADp m\xE3"),
      readOnly: readOnlyElementId
    }
  });
};

export { AttributeId };
