// @tempi/core-editor/dist/components/RenderQuickSetting/RenderQuickSetting.es.js
import { jsx } from 'react/jsx-runtime';
import { Popover } from 'antd';
import styled from '@emotion/styled';
import { useNode } from '@tempi/dnd-core';
import { useBuilder } from '../../contexts/BuilderContext.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const RenderQuickSetting = (_a) => {
  var _b = _a, {
    children,
    content,
    open
  } = _b, rest = __objRest(_b, [
    "children",
    "content",
    "open"
  ]);
  const { dragging } = useNode((node) => ({
    dragging: node.events.dragged
  }));
  const { frame } = useBuilder();
  return /* @__PURE__ */ jsx(Popover, __spreadProps(__spreadValues({
    placement: "top",
    open: open && !dragging,
    overlayInnerStyle: {
      width: "max-content",
      borderRadius: 8,
      padding: 0
    },
    zIndex: 1050,
    content: /* @__PURE__ */ jsx(ContentWrapper, {
      children: content
    }),
    autoAdjustOverflow: true,
    destroyTooltipOnHide: true,
    getPopupContainer: () => frame == null ? void 0 : frame.body,
    getTooltipContainer: () => frame == null ? void 0 : frame.body
  }, rest), {
    children
  }));
};
const ContentWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  padding-left: 12px;
  padding-right: 12px;
`;

export { RenderQuickSetting };
