// @tempi/core-editor/dist/configs/RangeSliderConfig/RangeSliderConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var antd = require('antd');
var React = require('react');
var InputNumberConfig = require('../InputNumberConfig/InputNumberConfig.js');
var usePropSeparateDevice = require('../../hooks/usePropSeparateDevice.js');
var useProp = require('../../hooks/useProp.js');
var FormItem = require('../../components/FormItem/FormItem.js');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const RangeSliderConfig = (props) => {
  const {
    propKey,
    separateDevice,
    usePropOptions,
    formItemProps,
    inputNumberProps,
    showInput = true,
    unitProps,
    className,
    sliderProps
  } = props;
  const _a = sliderProps || {}, { defaultValue } = _a, rest = __objRest(_a, ["defaultValue"]);
  const [prop, setProp] = usePropSeparateDevice.usePropSeparateDevice(
    propKey,
    separateDevice,
    usePropOptions
  );
  const [unit, setUnit] = useProp.useProp(unitProps == null ? void 0 : unitProps.propKey);
  React.useEffect(() => {
    if (typeof defaultValue !== "undefined" && typeof prop === "undefined") {
      setProp(defaultValue);
    }
    if ((unitProps == null ? void 0 : unitProps.propKey) && (unitProps == null ? void 0 : unitProps.defaultValue) && !unit && setUnit) {
      setUnit(unitProps == null ? void 0 : unitProps.defaultValue);
    }
  }, []);
  const onChange = (value) => {
    if (value === null)
      setProp(void 0);
    else
      setProp(value);
  };
  const getInputNumberWidth = () => {
    if ((unitProps == null ? void 0 : unitProps.options) && (unitProps == null ? void 0 : unitProps.options.length) > 0)
      return 60;
    if (unitProps)
      return 95;
    return 60;
  };
  return /* @__PURE__ */ jsxRuntime.jsx(FormItem.FormItem, __spreadProps(__spreadValues({}, formItemProps), {
    layout: "vertical",
    children: /* @__PURE__ */ jsxRuntime.jsxs(antd.Row, {
      gutter: 16,
      style: { alignItems: "center" },
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(antd.Col, {
          style: { flexGrow: 1 },
          children: /* @__PURE__ */ jsxRuntime.jsx(antd.Slider, __spreadValues({
            className,
            value: prop,
            onChange
          }, rest))
        }),
        /* @__PURE__ */ jsxRuntime.jsx(antd.Col, {
          children: showInput ? /* @__PURE__ */ jsxRuntime.jsx(InputNumberConfig.InputNumber, __spreadProps(__spreadValues(__spreadValues({
            value: prop,
            unitProps,
            onChange
          }, inputNumberProps), rest), {
            style: {
              width: getInputNumberWidth()
            },
            controls: false
          })) : /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
            children: [
              prop != null ? prop : rest == null ? void 0 : rest.value,
              unitProps == null ? void 0 : unitProps.defaultValue
            ]
          })
        })
      ]
    })
  }));
};

exports.RangeSliderConfig = RangeSliderConfig;
