// @tempi/core-elements/dist/quick-configs/BackgroundQuickSetting/BackgroundQuickSetting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var antd = require('antd');
var propKey = require('../../constants/propKey.js');

const BackgroundQuickSetting = ({
  track,
  propKey: propKey$1 = "commonStyle",
  allowConfigBackgroundVideo = false
}) => {
  const tabItems = [
    {
      label: i18n.t("M\xE0u n\u1EC1n"),
      key: "background-color",
      children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
        propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.BackgroundColor),
        formItemProps: {
          label: i18n.t("M\xE0u n\u1EC1n")
        },
        gradientPicker: true
      })
    },
    {
      label: i18n.t("\u1EA2nh n\u1EC1n"),
      key: "background-image",
      children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.BackgroundImageConfig, {
        propKey: propKey$1
      })
    },
    allowConfigBackgroundVideo && {
      label: i18n.t("Video n\u1EC1n"),
      key: "background-video",
      children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.BackgroundVideoConfig, {
        propKey: propKey$1
      })
    }
  ];
  return /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SubQuickSetting, {
    content: /* @__PURE__ */ jsxRuntime.jsx(antd.Tabs, {
      items: tabItems
    }),
    children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.QuickSettingButton, {
      onClick: () => track == null ? void 0 : track(),
      children: i18n.t("background_config.change_background")
    })
  });
};

exports.BackgroundQuickSetting = BackgroundQuickSetting;
