// @tempi/core-elements/dist/quick-configs/BackgroundQuickSetting/BackgroundQuickSetting.es.js
import { jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { ColorPickerConfig, BackgroundImageConfig, BackgroundVideoConfig, SubQuickSetting, QuickSettingButton } from '@tempi/core-editor';
import { Tabs } from 'antd';
import { PROP_KEY } from '../../constants/propKey.es.js';

const BackgroundQuickSetting = ({
  track,
  propKey = "commonStyle",
  allowConfigBackgroundVideo = false
}) => {
  const tabItems = [
    {
      label: t("M\xE0u n\u1EC1n"),
      key: "background-color",
      children: /* @__PURE__ */ jsx(ColorPickerConfig, {
        propKey: PROP_KEY.joinKey(propKey, PROP_KEY.BackgroundColor),
        formItemProps: {
          label: t("M\xE0u n\u1EC1n")
        },
        gradientPicker: true
      })
    },
    {
      label: t("\u1EA2nh n\u1EC1n"),
      key: "background-image",
      children: /* @__PURE__ */ jsx(BackgroundImageConfig, {
        propKey
      })
    },
    allowConfigBackgroundVideo && {
      label: t("Video n\u1EC1n"),
      key: "background-video",
      children: /* @__PURE__ */ jsx(BackgroundVideoConfig, {
        propKey
      })
    }
  ];
  return /* @__PURE__ */ jsx(SubQuickSetting, {
    content: /* @__PURE__ */ jsx(Tabs, {
      items: tabItems
    }),
    children: /* @__PURE__ */ jsx(QuickSettingButton, {
      onClick: () => track == null ? void 0 : track(),
      children: t("background_config.change_background")
    })
  });
};

export { BackgroundQuickSetting };
