// @tempi/core-editor/dist/configs/VideoConfig/VideoConfig.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { UploadOutlined } from '@ant-design/icons';
import styled from '@emotion/styled';
import { Form, Input, Button, Divider, Upload, Typography } from 'antd';
import { useState, useEffect } from 'react';
import { getFileName, beforeUploadVideo } from './helper.es.js';
import { usePropSeparateDevice } from '../../hooks/usePropSeparateDevice.es.js';
import { useBuilder } from '../../contexts/BuilderContext.es.js';
import { InputConfig } from '../InputConfig/InputConfig.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const { Text } = Typography;
const VideoConfig = ({
  youtubeVideoPropKey,
  allowExternalUrl = true,
  uploadVideoPropKey,
  allowUploadVideo = true,
  separateDevice
}) => {
  const [fileList, setFileList] = useState([]);
  const [form] = Form.useForm();
  const [uploading, setUploading] = useState(false);
  const [externalUploading, setExternalUploading] = useState(false);
  const [uploadUrl, setUploadUrl] = usePropSeparateDevice(
    uploadVideoPropKey,
    separateDevice
  );
  useEffect(() => {
    if (!!uploadUrl)
      setFileList([
        { url: uploadUrl, uid: uploadUrl, name: getFileName(uploadUrl) }
      ]);
    else
      setFileList([]);
  }, [uploadUrl]);
  const demoVideo = (file, _type) => {
    return URL.createObjectURL(file);
  };
  const { uploadFileHandler = demoVideo } = useBuilder();
  const onUploadVideo = async (options) => {
    const { file, onSuccess, onError } = options;
    try {
      setUploading(true);
      const url = await uploadFileHandler(file, "video");
      setUploadUrl(url);
      onSuccess(file);
    } catch (e) {
      console.error(e);
      onError(e, file);
    } finally {
      setUploading(false);
    }
  };
  const onRemoveVideo = () => {
    setUploadUrl("");
  };
  const uploadProps = {
    beforeUpload: beforeUploadVideo,
    customRequest: onUploadVideo,
    listType: "picture",
    maxCount: 1,
    multiple: false,
    fileList,
    onRemove: onRemoveVideo
  };
  const handleUploadVideoByURL = async (values) => {
    const { url } = values;
    try {
      setExternalUploading(true);
      const videoUrl = await uploadFileHandler(url, "videoUrl");
      setUploadUrl(videoUrl);
    } finally {
      form.resetFields();
      setExternalUploading(false);
    }
  };
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(InputConfig, {
        propKey: youtubeVideoPropKey,
        separateDevice,
        formItemProps: { label: t("\u0110\u01B0\u1EDDng d\u1EABn Youtube"), layout: "vertical" },
        inputProps: { placeholder: t("Nh\u1EADp \u0111\u01B0\u1EDDng d\u1EABn Youtube") }
      }),
      allowExternalUrl && /* @__PURE__ */ jsxs(Fragment, {
        children: [
          /* @__PURE__ */ jsx(StyledDivider, {
            plain: true,
            children: t("ho\u1EB7c")
          }),
          /* @__PURE__ */ jsx(Form, {
            form,
            onFinish: handleUploadVideoByURL,
            children: /* @__PURE__ */ jsx(Form.Item, {
              name: "url",
              rules: [
                { required: true, message: t("URL kh\xF4ng \u0111\u01B0\u1EE3c \u0111\u1EC3 tr\u1ED1ng") },
                { type: "url", message: t("URL sai \u0111\u1ECBnh d\u1EA1ng") }
              ],
              children: /* @__PURE__ */ jsx(Input.Search, {
                placeholder: t("D\xF9ng \u0111\u01B0\u1EDDng d\u1EABn video"),
                enterButton: t("D\xF9ng video"),
                loading: externalUploading,
                onSearch: () => form.submit()
              })
            })
          }),
          /* @__PURE__ */ jsx(Text, {
            type: "secondary",
            children: t("video_config.note")
          })
        ]
      }),
      allowUploadVideo && /* @__PURE__ */ jsxs(Fragment, {
        children: [
          /* @__PURE__ */ jsx(StyledDivider, {
            plain: true,
            children: t("ho\u1EB7c")
          }),
          /* @__PURE__ */ jsx(StyledUploader, __spreadProps(__spreadValues({}, uploadProps), {
            children: /* @__PURE__ */ jsx(Button, {
              block: true,
              icon: /* @__PURE__ */ jsx(UploadOutlined, {}),
              loading: uploading,
              children: t("T\u1EA3i l\xEAn video (MP4, WebM, OGV)")
            })
          }))
        ]
      })
    ]
  });
};
const StyledDivider = styled(Divider)`
  justify-content: center;
  &::before,
  &::after {
    width: 25% !important;
  }
`;
const StyledUploader = styled(Upload)`
  div.ant-upload {
    width: 100%;
  }
`;

export { VideoConfig };
