// @tempi/core-elements/dist/components/testimonial-carousel/configs/TestimonialItem.es.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { t } from '../../../i18n.es.js';
import { CopyOutlined, CloseOutlined } from '@ant-design/icons';
import styled from '@emotion/styled';
import { useProp, FormItem, ImageConfig, InputConfig } from '@tempi/core-editor';
import { Input } from 'antd';
import { useState } from 'react';
import { PROP_KEY } from '../../../constants/propKey.es.js';

const { TextArea } = Input;
const TestimonialItem = ({
  index,
  handleRemoveItem,
  handleCloneItem
}) => {
  const [expand, setExpand] = useState(false);
  const [content, setContent] = useProp(
    PROP_KEY.joinKey(PROP_KEY.Items, `[${index}]`, PROP_KEY.Content)
  );
  return /* @__PURE__ */ jsxs(Wrapper, {
    children: [
      /* @__PURE__ */ jsxs(Header, {
        onClick: () => setExpand(!expand),
        children: [
          t("Ph\u1EA7n t\u1EED #{{var0}}", { var0: index + 1 }),
          /* @__PURE__ */ jsxs("div", {
            children: [
              /* @__PURE__ */ jsx(CopyOutlined, {
                style: { fontSize: 18, padding: "0 8px" },
                onClick: (e) => {
                  e.stopPropagation();
                  handleCloneItem(index);
                }
              }),
              /* @__PURE__ */ jsx(CloseOutlined, {
                style: { fontSize: 18 },
                onClick: (e) => {
                  e.stopPropagation();
                  handleRemoveItem(index);
                }
              })
            ]
          })
        ]
      }),
      expand && /* @__PURE__ */ jsxs(Body, {
        children: [
          /* @__PURE__ */ jsx(FormItem, {
            label: t("N\u1ED9i dung"),
            layout: "vertical",
            children: /* @__PURE__ */ jsx(TextArea, {
              rows: 5,
              value: content,
              onChange: (e) => setContent(e.target.value)
            })
          }),
          /* @__PURE__ */ jsx(FormItem, {
            label: t("Ch\u1ECDn \u1EA3nh"),
            layout: "vertical",
            children: /* @__PURE__ */ jsx(ImageConfig, {
              propKey: PROP_KEY.joinKey(
                PROP_KEY.Items,
                `[${index}]`,
                PROP_KEY.Image,
                PROP_KEY.Src
              ),
              widthPropKey: PROP_KEY.joinKey(
                PROP_KEY.Items,
                `[${index}]`,
                PROP_KEY.Image,
                PROP_KEY.GoogleImageParams,
                PROP_KEY.Width
              )
            })
          }),
          /* @__PURE__ */ jsx(InputConfig, {
            propKey: PROP_KEY.joinKey(
              PROP_KEY.Items,
              `[${index}]`,
              PROP_KEY.Name
            ),
            formItemProps: {
              label: t("T\xEAn")
            },
            inputProps: {
              placeholder: t("Nh\u1EADp t\xEAn")
            }
          }),
          /* @__PURE__ */ jsx(InputConfig, {
            propKey: PROP_KEY.joinKey(
              PROP_KEY.Items,
              `[${index}]`,
              PROP_KEY.Title
            ),
            formItemProps: {
              label: t("Ti\xEAu \u0111\u1EC1")
            },
            inputProps: {
              placeholder: t("Nh\u1EADp ti\xEAu \u0111\u1EC1")
            }
          })
        ]
      })
    ]
  });
};
const Wrapper = styled.div`
  border: 1px solid rgb(217, 217, 217);
  border-radius: 4px;
  padding: 0 8px;
  margin-bottom: 8px;
`;
const Header = styled.div`
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 40px;
`;
const Body = styled.div``;

export { TestimonialItem };
