// @tempi/core-elements/dist/components/testimonial-carousel/configs/TestimonialList.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../../i18n.es.js';
import { PlusCircleOutlined } from '@ant-design/icons';
import styled from '@emotion/styled';
import { useNode } from '@tempi/core-editor';
import { mappingAssets } from '@tempi/core-renderer';
import { Button } from 'antd';
import get from 'lodash.get';
import { TestimonialItem } from './TestimonialItem.es.js';

const { defaultImage } = mappingAssets;
const TestimonialList = ({ propKey }) => {
  const {
    items,
    actions: { setProp }
  } = useNode((node) => {
    return {
      items: get(node.data.props, propKey)
    };
  });
  const handleAddItem = () => {
    const newItem = {
      content: `item content ${items.length + 1}`,
      image: {
        src: defaultImage
      },
      name: `item name ${items.length + 1}`,
      title: `item title ${items.length + 1}`
    };
    setProp((props) => {
      props["items"] = [...items, newItem];
    }, 500);
  };
  const handleRemoveItem = (itemIndex) => {
    setProp((props) => {
      props["items"] = items.filter((_, index) => index !== itemIndex);
    }, 500);
  };
  const handleCloneItem = (itemIndex) => {
    const newItem = items[itemIndex];
    setProp((props) => {
      props["items"] = [...items, newItem];
    }, 500);
  };
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      items.map((_, index) => /* @__PURE__ */ jsx(TestimonialItem, {
        index,
        handleCloneItem,
        handleRemoveItem
      }, index)),
      /* @__PURE__ */ jsx(BtnWrapper, {
        children: /* @__PURE__ */ jsx(Button, {
          onClick: handleAddItem,
          type: "primary",
          icon: /* @__PURE__ */ jsx(PlusCircleOutlined, {}),
          children: t("Th\xEAm ph\u1EA7n t\u1EED")
        })
      })
    ]
  });
};
const BtnWrapper = styled.div`
  display: flex;
  justify-content: center;
  margin-bottom: 16px;
`;

export { TestimonialList };
