// @tempi/core-elements/dist/components/testimonial-carousel/configs/TestimonialItem.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../i18n.js');
var icons = require('@ant-design/icons');
var styled = require('@emotion/styled');
var coreEditor = require('@tempi/core-editor');
var antd = require('antd');
var React = require('react');
var propKey = require('../../../constants/propKey.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

const { TextArea } = antd.Input;
const TestimonialItem = ({
  index,
  handleRemoveItem,
  handleCloneItem
}) => {
  const [expand, setExpand] = React.useState(false);
  const [content, setContent] = coreEditor.useProp(
    propKey.PROP_KEY.joinKey(propKey.PROP_KEY.Items, `[${index}]`, propKey.PROP_KEY.Content)
  );
  return /* @__PURE__ */ jsxRuntime.jsxs(Wrapper, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsxs(Header, {
        onClick: () => setExpand(!expand),
        children: [
          i18n.t("Ph\u1EA7n t\u1EED #{{var0}}", { var0: index + 1 }),
          /* @__PURE__ */ jsxRuntime.jsxs("div", {
            children: [
              /* @__PURE__ */ jsxRuntime.jsx(icons.CopyOutlined, {
                style: { fontSize: 18, padding: "0 8px" },
                onClick: (e) => {
                  e.stopPropagation();
                  handleCloneItem(index);
                }
              }),
              /* @__PURE__ */ jsxRuntime.jsx(icons.CloseOutlined, {
                style: { fontSize: 18 },
                onClick: (e) => {
                  e.stopPropagation();
                  handleRemoveItem(index);
                }
              })
            ]
          })
        ]
      }),
      expand && /* @__PURE__ */ jsxRuntime.jsxs(Body, {
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
            label: i18n.t("N\u1ED9i dung"),
            layout: "vertical",
            children: /* @__PURE__ */ jsxRuntime.jsx(TextArea, {
              rows: 5,
              value: content,
              onChange: (e) => setContent(e.target.value)
            })
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
            label: i18n.t("Ch\u1ECDn \u1EA3nh"),
            layout: "vertical",
            children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ImageConfig, {
              propKey: propKey.PROP_KEY.joinKey(
                propKey.PROP_KEY.Items,
                `[${index}]`,
                propKey.PROP_KEY.Image,
                propKey.PROP_KEY.Src
              ),
              widthPropKey: propKey.PROP_KEY.joinKey(
                propKey.PROP_KEY.Items,
                `[${index}]`,
                propKey.PROP_KEY.Image,
                propKey.PROP_KEY.GoogleImageParams,
                propKey.PROP_KEY.Width
              )
            })
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputConfig, {
            propKey: propKey.PROP_KEY.joinKey(
              propKey.PROP_KEY.Items,
              `[${index}]`,
              propKey.PROP_KEY.Name
            ),
            formItemProps: {
              label: i18n.t("T\xEAn")
            },
            inputProps: {
              placeholder: i18n.t("Nh\u1EADp t\xEAn")
            }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputConfig, {
            propKey: propKey.PROP_KEY.joinKey(
              propKey.PROP_KEY.Items,
              `[${index}]`,
              propKey.PROP_KEY.Title
            ),
            formItemProps: {
              label: i18n.t("Ti\xEAu \u0111\u1EC1")
            },
            inputProps: {
              placeholder: i18n.t("Nh\u1EADp ti\xEAu \u0111\u1EC1")
            }
          })
        ]
      })
    ]
  });
};
const Wrapper = styled__default["default"].div`
  border: 1px solid rgb(217, 217, 217);
  border-radius: 4px;
  padding: 0 8px;
  margin-bottom: 8px;
`;
const Header = styled__default["default"].div`
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 40px;
`;
const Body = styled__default["default"].div``;

exports.TestimonialItem = TestimonialItem;
