// @tempi/core-editor/dist/components/ModalConfigWrapper/ModalConfigWrapper.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { useState } from 'react';
import styled from '@emotion/styled';
import { Typography, Button } from 'antd';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const ModalConfigWrapper = ({
  title = t("C\u1EA5u h\xECnh s\u1EF1 ki\u1EC7n"),
  actionText = t("C\u1EA5u h\xECnh"),
  customModal: CustomModal,
  subTitle,
  customModalProps
}) => {
  const [visible, setVisible] = useState(false);
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsxs(Wrapper, {
        children: [
          /* @__PURE__ */ jsxs("div", {
            children: [
              /* @__PURE__ */ jsx(Typography.Title, {
                level: 5,
                style: { fontSize: 14, fontWeight: 500 },
                children: title
              }),
              !!subTitle && /* @__PURE__ */ jsx(Typography.Paragraph, {
                type: "secondary",
                className: "mb-0",
                children: subTitle
              })
            ]
          }),
          /* @__PURE__ */ jsx(ActionButton, {
            type: "link",
            onClick: () => setVisible((prevState) => !prevState),
            children: actionText
          })
        ]
      }),
      /* @__PURE__ */ jsx(CustomModal, __spreadValues({}, __spreadProps(__spreadValues({}, customModalProps), { visible, setVisible })))
    ]
  });
};
const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  margin-bottom: 8px;
`;
const ActionButton = styled(Button)`
  padding: 0 !important;
  height: unset !important;
`;

export { ModalConfigWrapper };
