// @tempi/core-elements/dist/configs/TipTapEditor/TiptapEditorSimple.es.js
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
import { RenderQuickSetting } from '@tempi/core-editor';
import React__default from 'react';
import { useCustomTiptap } from './hooks.es.js';
import { CustomKit } from './utils/custom-kit.es.js';
import { EditorContent } from './components/EditorContent.es.js';
import { TypographyQuickSetting } from '../../quick-configs/typography/TypographyQuickSetting.es.js';
import { TextColorQuickSetting } from '../../quick-configs/typography/TextColorQuickSetting.es.js';
import { BoldQuickSetting } from '../../quick-configs/typography/BoldQuickSetting.es.js';
import { ItalicQuickSetting } from '../../quick-configs/typography/ItalicQuickSetting.es.js';
import { UnderlineQuickSetting } from '../../quick-configs/typography/UnderlineQuickSetting.es.js';
import { StrikeThroughQuickSetting } from '../../quick-configs/typography/StrikeThroughQuickSetting.es.js';
import { TextTransformQuickSetting } from '../../quick-configs/typography/TextTransformQuickSetting.es.js';

const TiptapEditorSimple = ({
  propKey
}) => {
  const { editor, isEditable, setIsEditable } = useCustomTiptap(
    {
      editorOptions: {
        extensions: [
          CustomKit.configure({
            list: false,
            table: false
          })
        ]
      }
    },
    propKey
  );
  return /* @__PURE__ */ jsx(RenderQuickSetting, {
    content: React__default.createElement(QuickSetting, { editor }),
    children: /* @__PURE__ */ jsx(EditorContent, {
      editor,
      isEditable,
      setIsEditable
    })
  });
};
const QuickSetting = ({ editor }) => {
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(TypographyQuickSetting, {
        editor
      }),
      /* @__PURE__ */ jsx(TextColorQuickSetting, {
        editor
      }),
      /* @__PURE__ */ jsx(BoldQuickSetting, {
        editor
      }),
      /* @__PURE__ */ jsx(ItalicQuickSetting, {
        editor
      }),
      /* @__PURE__ */ jsx(UnderlineQuickSetting, {
        editor
      }),
      /* @__PURE__ */ jsx(StrikeThroughQuickSetting, {
        editor
      }),
      /* @__PURE__ */ jsx(TextTransformQuickSetting, {
        editor
      })
    ]
  });
};

export { TiptapEditorSimple };
