// @tempi/core-elements/dist/configs/TipTapEditor/TiptapEditorSimple.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var coreEditor = require('@tempi/core-editor');
var React = require('react');
var hooks = require('./hooks.js');
var customKit = require('./utils/custom-kit.js');
var EditorContent = require('./components/EditorContent.js');
var TypographyQuickSetting = require('../../quick-configs/typography/TypographyQuickSetting.js');
var TextColorQuickSetting = require('../../quick-configs/typography/TextColorQuickSetting.js');
var BoldQuickSetting = require('../../quick-configs/typography/BoldQuickSetting.js');
var ItalicQuickSetting = require('../../quick-configs/typography/ItalicQuickSetting.js');
var UnderlineQuickSetting = require('../../quick-configs/typography/UnderlineQuickSetting.js');
var StrikeThroughQuickSetting = require('../../quick-configs/typography/StrikeThroughQuickSetting.js');
var TextTransformQuickSetting = require('../../quick-configs/typography/TextTransformQuickSetting.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

const TiptapEditorSimple = ({
  propKey
}) => {
  const { editor, isEditable, setIsEditable } = hooks.useCustomTiptap(
    {
      editorOptions: {
        extensions: [
          customKit.CustomKit.configure({
            list: false,
            table: false
          })
        ]
      }
    },
    propKey
  );
  return /* @__PURE__ */ jsxRuntime.jsx(coreEditor.RenderQuickSetting, {
    content: React__default["default"].createElement(QuickSetting, { editor }),
    children: /* @__PURE__ */ jsxRuntime.jsx(EditorContent.EditorContent, {
      editor,
      isEditable,
      setIsEditable
    })
  });
};
const QuickSetting = ({ editor }) => {
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(TypographyQuickSetting.TypographyQuickSetting, {
        editor
      }),
      /* @__PURE__ */ jsxRuntime.jsx(TextColorQuickSetting.TextColorQuickSetting, {
        editor
      }),
      /* @__PURE__ */ jsxRuntime.jsx(BoldQuickSetting.BoldQuickSetting, {
        editor
      }),
      /* @__PURE__ */ jsxRuntime.jsx(ItalicQuickSetting.ItalicQuickSetting, {
        editor
      }),
      /* @__PURE__ */ jsxRuntime.jsx(UnderlineQuickSetting.UnderlineQuickSetting, {
        editor
      }),
      /* @__PURE__ */ jsxRuntime.jsx(StrikeThroughQuickSetting.StrikeThroughQuickSetting, {
        editor
      }),
      /* @__PURE__ */ jsxRuntime.jsx(TextTransformQuickSetting.TextTransformQuickSetting, {
        editor
      })
    ]
  });
};

exports.TiptapEditorSimple = TiptapEditorSimple;
