// @tempi/core-elements/dist/components/button/v2/constants.es.js
import { t } from '../../../i18n.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const DEFAULT_STYLE = {
  background: null,
  backgroundColor: null,
  borderStyle: null,
  borderWidth: null,
  borderRadius: null,
  borderColor: null
};
const DEFAULT_BUTTON_STYLE = {
  primary: __spreadProps(__spreadValues({}, DEFAULT_STYLE), {
    background: "rgba(37, 141, 255, 1)"
  }),
  primaryRound: __spreadProps(__spreadValues({}, DEFAULT_STYLE), {
    background: "rgba(37, 141, 255, 1)",
    borderRadius: 999
  }),
  outline: __spreadProps(__spreadValues({}, DEFAULT_STYLE), {
    borderStyle: "solid",
    borderWidth: 1,
    borderColor: "rgba(37, 141, 255, 1)"
  }),
  outlineRound: __spreadProps(__spreadValues({}, DEFAULT_STYLE), {
    borderStyle: "solid",
    borderWidth: 1,
    borderColor: "rgba(37, 141, 255, 1)",
    borderRadius: 999
  }),
  link: DEFAULT_STYLE
};
const DEFAULT_BUTTON = {
  primary: {
    text: t("B\u1EA5m v\xE0o \u0111\xE2y"),
    buttonStyle: __spreadProps(__spreadValues({}, DEFAULT_BUTTON_STYLE.primary), {
      color: "rgba(255, 255, 255, 1)",
      height: 32,
      width: 120
    }),
    iconConfig: {
      position: "left",
      spacing: 8,
      show: true,
      icon: {
        size: 24
      }
    }
  },
  primaryRound: {
    buttonStyle: __spreadProps(__spreadValues({}, DEFAULT_BUTTON_STYLE.primaryRound), {
      color: "rgba(255, 255, 255, 1)",
      height: 32,
      width: 120
    }),
    iconConfig: {
      position: "left",
      spacing: 8,
      show: true,
      icon: {
        size: 24
      }
    }
  },
  outlineRound: {
    buttonStyle: __spreadProps(__spreadValues({}, DEFAULT_BUTTON_STYLE.outlineRound), {
      color: "rgba(37, 141, 255, 1)",
      height: 32,
      width: 120
    }),
    iconConfig: {
      position: "left",
      spacing: 8,
      show: true,
      icon: {
        size: 24
      }
    }
  },
  link: {
    buttonStyle: __spreadProps(__spreadValues({}, DEFAULT_BUTTON_STYLE.link), {
      color: "rgba(37, 141, 255, 1)",
      height: 32,
      width: 120
    }),
    iconConfig: {
      position: "left",
      spacing: 8,
      show: true,
      icon: {
        size: 24
      }
    }
  },
  outline: {
    buttonStyle: __spreadProps(__spreadValues({}, DEFAULT_BUTTON_STYLE.outline), {
      color: "rgba(37, 141, 255, 1)",
      height: 32,
      width: 120
    }),
    iconConfig: {
      position: "left",
      spacing: 8,
      show: true,
      icon: {
        size: 24
      }
    }
  }
};

export { DEFAULT_BUTTON, DEFAULT_BUTTON_STYLE };
