// @tempi/core-elements/dist/components/button/v2/quick-setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../i18n.js');
var reactIcons = require('@fluentui/react-icons');
var iconsDev = require('@tempi/icons-dev');
var coreEditor = require('@tempi/core-editor');
var TypographyQuickSetting = require('../../../quick-configs/typography/TypographyQuickSetting.js');
var ButtonConfig = require('./configs/ButtonConfig.js');
var IconButtonConfig = require('./configs/IconButtonConfig.js');
var AlignmentQuickSetting = require('../../../quick-configs/AlignmentQuickSetting/AlignmentQuickSetting.js');
var ButtonColorConfig = require('./configs/ButtonColorConfig.js');
var propKey = require('../../../constants/propKey.js');

const ButtonQuickSetting = ({
  editor,
  propKey: propKey$1,
  configs = {},
  separateDevice
}) => {
  const alignPropKey = propKey.PROP_KEY.joinKey(
    propKey$1,
    propKey.PROP_KEY.CommonStyle,
    propKey.PROP_KEY.JustifyContent
  );
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(TypographyQuickSetting.TypographyQuickSetting, {
        config: { responsiveFontSize: separateDevice },
        editor
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SubQuickSetting, {
        content: /* @__PURE__ */ jsxRuntime.jsx(ButtonConfig.ButtonConfig, {
          separateDevice,
          propKey: propKey$1
        }),
        children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.QuickSettingButton, {
          tooltip: i18n.t("T\xF9y ch\u1EC9nh n\xFAt b\u1EA5m"),
          children: /* @__PURE__ */ jsxRuntime.jsx(iconsDev.AntdCustomIcon, {
            name: "ButtonPointerOutlined",
            style: {
              fontSize: 16
            }
          })
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SubQuickSetting, {
        content: /* @__PURE__ */ jsxRuntime.jsx(IconButtonConfig.IconButtonConfig, {
          separateDevice,
          propKey: propKey$1
        }),
        children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.QuickSettingButton, {
          tooltip: i18n.t("T\xF9y ch\u1EC9nh bi\u1EC3u t\u01B0\u1EE3ng"),
          children: /* @__PURE__ */ jsxRuntime.jsx(reactIcons.Star16Regular, {})
        })
      }),
      configs.alignment && /* @__PURE__ */ jsxRuntime.jsx(AlignmentQuickSetting.AlignmentQuickSetting, {
        propKey: alignPropKey,
        listAlignment: ["flex-start", "center", "flex-end"],
        separateDevice
      }),
      /* @__PURE__ */ jsxRuntime.jsx(ButtonColorConfig.ButtonColorConfig, {
        separateDevice,
        propKey: propKey$1
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.EventQuickSetting, {
        propKey: propKey$1
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.OpenSettingDialog, {}),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.DuplicateNode, {}),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.DeleteNode, {})
    ]
  });
};

exports.ButtonQuickSetting = ButtonQuickSetting;
