// @tempi/core-elements/dist/configs/storage/GoogleSheetConfig/GoogleSheetConfig.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../../i18n.es.js';
import { LoadingOutlined, ExclamationCircleOutlined, CheckCircleOutlined } from '@ant-design/icons';
import { useBuilder } from '@tempi/core-editor';
import { theme, Form, Input } from 'antd';
import debounce from 'lodash.debounce';
import { GGSheetAccountConfig } from './GGSheetAccountConfig.es.js';
import { isGoogleSheetUrl } from '../../../utils/isGoogleSheetUrl.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const GoogleSheetConfig = ({
  index,
  ggSheetsAccounts = [],
  handleFetchFormAccount,
  displayUrl,
  defaultField,
  form
}) => {
  const { authorizeGGSheet } = useBuilder();
  const { token } = theme.useToken();
  const setFieldValueForm = (fieldName, fieldVal) => {
    form.setFieldValue(["configs", index, fieldName], fieldVal);
  };
  const getFieldValueForm = (fieldName) => {
    return form.getFieldValue(["configs", index, fieldName]);
  };
  const checkAuthorization = async (url, ggSheetAccId) => {
    if (!url || !ggSheetAccId || !isGoogleSheetUrl(url)) {
      setFieldValueForm("statusGoogleSheetUrl", "");
      return;
    }
    if (ggSheetAccId > 0) {
      const body = {
        spreadSheetUrl: url,
        formAccountId: ggSheetAccId
      };
      try {
        setFieldValueForm("statusGoogleSheetUrl", "validating");
        const data = await authorizeGGSheet(body);
        if (data.granted) {
          setFieldValueForm("sheetId", data.sheetId);
          setFieldValueForm("statusGoogleSheetUrl", "valid");
        } else {
          setFieldValueForm("statusGoogleSheetUrl", "invalid");
        }
      } catch (err) {
        setFieldValueForm("statusGoogleSheetUrl", "invalid");
      } finally {
        form.validateFields([["configs", index, "spreadSheetUrl"]]);
      }
    }
  };
  const renderSuffix = () => {
    let icon = /* @__PURE__ */ jsx(Fragment, {});
    switch (getFieldValueForm("statusGoogleSheetUrl")) {
      case "valid":
        icon = /* @__PURE__ */ jsx(CheckCircleOutlined, {
          style: { color: "#59C36A", fontSize: 20 }
        });
        break;
      case "validating":
        icon = /* @__PURE__ */ jsx(LoadingOutlined, {
          style: {
            animation: "loadingCircle 1s infinite linear",
            color: token.colorPrimary,
            fontSize: 20
          }
        });
        break;
      case "invalid":
        icon = /* @__PURE__ */ jsx(ExclamationCircleOutlined, {
          style: { color: "#EE112E", fontSize: 20 }
        });
        break;
    }
    return icon;
  };
  const debounceCheckAuthorization = debounce(
    (spreadSheetUrl, formAccountId) => checkAuthorization(spreadSheetUrl, formAccountId),
    300
  );
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(Form.Item, __spreadProps(__spreadValues({
        validateFirst: true,
        labelCol: {
          span: 8
        },
        wrapperCol: {
          span: 16
        },
        rules: [
          {
            required: true,
            message: t("Vui l\xF2ng ch\u1ECDn t\xE0i kho\u1EA3n li\xEAn k\u1EBFt")
          }
        ],
        name: [index, "formAccountId"],
        label: getFieldValueForm("formAccountId") ? t("T\xE0i kho\u1EA3n Google") : null
      }, defaultField), {
        children: /* @__PURE__ */ jsx(GGSheetAccountConfig, {
          ggSheetsAccounts,
          handleFetchFormAccount,
          onChange: (ggSheetAccId) => checkAuthorization(
            getFieldValueForm("spreadSheetUrl"),
            ggSheetAccId
          )
        })
      })),
      displayUrl && /* @__PURE__ */ jsx(Fragment, {
        children: /* @__PURE__ */ jsx(Form.Item, {
          noStyle: true,
          shouldUpdate: (prev, curr) => {
            var _a, _b, _c, _d;
            return ((_b = (_a = prev == null ? void 0 : prev.configs) == null ? void 0 : _a[index]) == null ? void 0 : _b.statusGoogleSheetUrl) !== ((_d = (_c = curr == null ? void 0 : curr.configs) == null ? void 0 : _c[index]) == null ? void 0 : _d.statusGoogleSheetUrl);
          },
          children: () => {
            return /* @__PURE__ */ jsx(Form.Item, __spreadProps(__spreadValues({
              labelCol: {
                span: 8
              },
              wrapperCol: {
                span: 16
              }
            }, defaultField), {
              validateFirst: true,
              rules: [
                {
                  required: true,
                  message: t("Vui l\xF2ng nh\u1EADp Google Sheet URL")
                },
                {
                  message: t("Vui l\xF2ng nh\u1EADp \u0111\xFAng Google Sheet URL"),
                  validator: (_, value) => {
                    if (isGoogleSheetUrl(value)) {
                      return Promise.resolve();
                    } else {
                      return Promise.reject(
                        t("Vui l\xF2ng nh\u1EADp \u0111\xFAng Google Sheet URL")
                      );
                    }
                  }
                },
                {
                  message: t("Kh\xF4ng th\u1EC3 truy c\u1EADp trang t\xEDnh"),
                  validator: (_, value) => {
                    if (value && getFieldValueForm("statusGoogleSheetUrl") === "invalid") {
                      return Promise.reject(
                        t("Vui l\xF2ng nh\u1EADp \u0111\xFAng Google Sheet URL")
                      );
                    } else {
                      return Promise.resolve();
                    }
                  }
                }
              ],
              name: [index, "spreadSheetUrl"],
              label: t("\u0110\u01B0\u1EDDng d\u1EABn"),
              children: /* @__PURE__ */ jsx(Input, {
                autoComplete: "off",
                placeholder: t("Nh\u1EADp \u0111\u01B0\u1EDDng d\u1EABn Google Sheet"),
                readOnly: getFieldValueForm("statusGoogleSheetUrl") === "validating",
                suffix: renderSuffix(),
                onChange: () => {
                  setFieldValueForm("statusGoogleSheetUrl", "");
                  debounceCheckAuthorization(
                    getFieldValueForm("spreadSheetUrl"),
                    getFieldValueForm("formAccountId")
                  );
                }
              })
            }));
          }
        })
      })
    ]
  });
};

export { GoogleSheetConfig };
