// @tempi/core-editor/dist/configs/TekoTrackingConfig/TekoTrackingConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var propKey = require('../../constants/propKey.js');
var InputConfig = require('../InputConfig/InputConfig.js');

const TekoTrackingConfig = ({
  propKey: propKey$1 = propKey.PROP_KEY.Event,
  separateDevice
}) => {
  const regionName = propKey.PROP_KEY.joinKey(
    propKey$1,
    propKey.PROP_KEY.TekoTrackingEvent,
    propKey.PROP_KEY.RegionName
  );
  const contentName = propKey.PROP_KEY.joinKey(
    propKey$1,
    propKey.PROP_KEY.TekoTrackingEvent,
    propKey.PROP_KEY.ContentName
  );
  const target = propKey.PROP_KEY.joinKey(
    propKey$1,
    propKey.PROP_KEY.TekoTrackingEvent,
    propKey.PROP_KEY.Target
  );
  const payload = propKey.PROP_KEY.joinKey(
    propKey$1,
    propKey.PROP_KEY.TekoTrackingEvent,
    propKey.PROP_KEY.Payload
  );
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(InputConfig.InputConfig, {
        separateDevice,
        propKey: regionName,
        formItemProps: {
          label: "Region name",
          style: {
            display: "flex",
            justifyContent: "space-between",
            alignItems: "center",
            marginBottom: "0.5rem"
          }
        },
        inputProps: {
          placeholder: "Region name",
          style: { width: "180px" }
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(InputConfig.InputConfig, {
        separateDevice,
        propKey: contentName,
        formItemProps: {
          label: "Content name",
          style: {
            display: "flex",
            justifyContent: "space-between",
            alignItems: "center",
            marginBottom: "0.5rem"
          }
        },
        inputProps: {
          placeholder: "Content name",
          style: { width: "180px" }
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(InputConfig.InputConfig, {
        separateDevice,
        propKey: target,
        formItemProps: {
          label: "Target",
          style: {
            display: "flex",
            justifyContent: "space-between",
            alignItems: "center",
            marginBottom: "0.5rem"
          }
        },
        inputProps: {
          placeholder: "Target",
          style: { width: "180px" }
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(InputConfig.InputConfig, {
        separateDevice,
        propKey: payload,
        formItemProps: {
          label: "Payload",
          style: {
            display: "flex",
            justifyContent: "space-between",
            alignItems: "center",
            marginBottom: "0.5rem"
          }
        },
        inputProps: {
          placeholder: "Payload",
          style: { width: "180px" }
        }
      })
    ]
  });
};

exports.TekoTrackingConfig = TekoTrackingConfig;
