// @tempi/core-editor/dist/configs/HoverConfig/HoverConfig.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { UnitEnum } from '@tempi/core-renderer';
import { ColorPickerConfig } from '../ColorPickerConfig/ColorPickerConfig.es.js';
import { PROP_KEY } from '../../constants/propKey.es.js';
import { InputNumberConfig } from '../InputNumberConfig/InputNumberConfig.es.js';

const HoverConfig = ({
  propKey = PROP_KEY.CommonStyle,
  separateDevice
}) => {
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(ColorPickerConfig, {
        separateDevice,
        propKey: PROP_KEY.joinKey(propKey, PROP_KEY.HoverBorderColor),
        formItemProps: { label: t("M\xE0u vi\u1EC1n") }
      }),
      /* @__PURE__ */ jsx(ColorPickerConfig, {
        separateDevice,
        propKey: PROP_KEY.joinKey(propKey, PROP_KEY.HoverBackgroundColor),
        formItemProps: { label: t("M\xE0u n\u1EC1n") },
        gradientPicker: true
      }),
      /* @__PURE__ */ jsx(ColorPickerConfig, {
        separateDevice,
        propKey: PROP_KEY.joinKey(propKey, PROP_KEY.HoverColor),
        formItemProps: { label: t("M\xE0u ch\u1EEF") }
      }),
      /* @__PURE__ */ jsx(InputNumberConfig, {
        separateDevice,
        propKey: PROP_KEY.joinKey(propKey, PROP_KEY.HoverTransparent),
        formItemProps: {
          label: t("Trong su\u1ED1t")
        },
        inputNumberProps: {
          min: 0,
          max: 100,
          placeholder: "100"
        },
        unitProps: { defaultValue: UnitEnum.percent }
      }),
      /* @__PURE__ */ jsx(InputNumberConfig, {
        separateDevice,
        propKey: PROP_KEY.joinKey(propKey, PROP_KEY.HoverZoom),
        formItemProps: {
          label: t("Ph\xF3ng to")
        },
        inputNumberProps: {
          min: 0,
          placeholder: "100"
        },
        unitProps: { defaultValue: UnitEnum.percent }
      })
    ]
  });
};

export { HoverConfig };
