// @tempi/core-elements/dist/components/store-list/constants.es.js
import { t } from '../../i18n.es.js';

const DEFAULT_STORES = [
  {
    name: t("Teko H\xE0 N\u1ED9i"),
    address: t("Peakview Tower, 36 Ho\xE0ng C\u1EA7u, P. \xD4 Ch\u1EE3 D\u1EEBa, \u0110\u1ED1ng \u0110a, H\xE0 N\u1ED9i"),
    phone: "024 7308 6336",
    workHours: "8h30 - 17h30",
    map: '<iframe src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d59597.414738390646!2d105.79102373768603!3d20.999113110094758!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x3135ab10f1fc1aab%3A0x66f26185c594c82a!2sTechnology%20Joint%20Stock%20Company%20Vietnam%20TEKO!5e0!3m2!1sen!2s!4v1647959420865!5m2!1sen!2s" width="400" height="300" style="border:0;" allowfullscreen="" loading="lazy"></iframe>',
    cityId: "01",
    districtId: "006"
  },
  {
    name: t("Teko H\u1ED3 Ch\xED Minh"),
    address: t("15 Tr\u01B0\u01A1ng \u0110\u1ECBnh, Ph\u01B0\u1EDDng 06, Qu\u1EADn 3, TP H\u1ED3 Ch\xED Minh"),
    phone: "024 7308 6336",
    workHours: "8h30 - 17h30",
    map: '<iframe src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d3919.4581561482833!2d106.68890011520631!3d10.77617896214396!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x31752f3a5570ff1d%3A0xb1c763a459280a36!2zMTUgVHLGsMahbmcgxJDhu4tuaCwgUGjGsOG7nW5nIDYsIFF14bqtbiAzLCBUaMOgbmggcGjhu5EgSOG7kyBDaMOtIE1pbmgsIFZpZXRuYW0!5e0!3m2!1sen!2s!4v1647959488552!5m2!1sen!2s" width="400" height="300" style="border:0;" allowfullscreen="" loading="lazy"></iframe>',
    cityId: "79",
    districtId: "770"
  }
];
const DEFAULT_NEW_STORE = {
  name: "",
  address: "",
  phone: "",
  workHours: "",
  map: "",
  cityId: "",
  districtId: ""
};
const DEFAULT_STORE_LIST = {
  title: t("Ch\u1ECDn c\u1EEDa h\xE0ng theo t\u1EC9nh/th\xE0nh ph\u1ED1"),
  stores: DEFAULT_STORES,
  stickyTop: 20,
  mainColor: "#2e7ac1"
};

export { DEFAULT_NEW_STORE, DEFAULT_STORES, DEFAULT_STORE_LIST };
