// @tempi/core-elements/dist/components/store-list/selector.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var coreEditor = require('@tempi/core-editor');
var constants = require('./constants.js');
var quickSetting = require('./quick-setting.js');
var setting = require('./setting.js');
var ui = require('./ui.js');
var selector = require('../section/selector.js');
var selector$1 = require('../root/selector.js');

const StoreListEditor = coreEditor.withSelector(ui.StoreList, {
  displayName: "Danh s\xE1ch c\u1EEDa h\xE0ng",
  tag: ui.StoreList.displayName.toLowerCase(),
  advanceAttributes: coreEditor.AdvanceAttributes,
  quickSetting: quickSetting.StoreListQuickSetting,
  customAttributes: setting.StoreListSetting,
  props: constants.DEFAULT_STORE_LIST,
  rules: {
    canDrop: (targetNode) => {
      const ignoreTargets = [
        coreEditor.getEditorName({ RowEditor: selector.RowEditor }),
        coreEditor.getEditorName({ RootEditor: selector$1.RootEditor })
      ];
      return !ignoreTargets.includes(targetNode.data.name);
    }
  }
});

exports.StoreListEditor = StoreListEditor;
