// @tempi/core-elements/dist/components/store-list/setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var coreRenderer = require('@tempi/core-renderer');
var antd = require('antd');
var StoreSetting = require('./configs/StoreSetting.js');
var ContentWrapper = require('../../core/ContentWrapper/ContentWrapper.js');
var propKey = require('../../constants/propKey.js');

const StoreListSetting = () => {
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsxs(ContentWrapper.ContentWrapper, {
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputConfig, {
            propKey: "title",
            formItemProps: {
              label: i18n.t("Ti\xEAu \u0111\u1EC1")
            }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
            propKey: propKey.PROP_KEY.MainColor,
            placeholder: i18n.t("Ch\u1ECDn m\xE0u"),
            formItemProps: {
              label: i18n.t("M\xE0u s\u1EAFc ch\xEDnh")
            }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
            propKey: propKey.PROP_KEY.StickyTop,
            formItemProps: {
              label: i18n.t("Kho\u1EA3ng c\xE1ch tr\xEAn khi scroll")
            },
            unitProps: { defaultValue: coreRenderer.UnitEnum.pixel }
          })
        ]
      }),
      /* @__PURE__ */ jsxRuntime.jsx(antd.Typography.Title, {
        level: 5,
        children: i18n.t("Danh s\xE1ch c\u1EEDa h\xE0ng")
      }),
      /* @__PURE__ */ jsxRuntime.jsx(StoreSetting.StoreSetting, {})
    ]
  });
};

exports.StoreListSetting = StoreListSetting;
