// @tempi/core-elements/dist/components/store-list/setting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { InputConfig, ColorPickerConfig, InputNumberConfig } from '@tempi/core-editor';
import { UnitEnum } from '@tempi/core-renderer';
import { Typography } from 'antd';
import { StoreSetting } from './configs/StoreSetting.es.js';
import { ContentWrapper } from '../../core/ContentWrapper/ContentWrapper.es.js';
import { PROP_KEY } from '../../constants/propKey.es.js';

const StoreListSetting = () => {
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsxs(ContentWrapper, {
        children: [
          /* @__PURE__ */ jsx(InputConfig, {
            propKey: "title",
            formItemProps: {
              label: t("Ti\xEAu \u0111\u1EC1")
            }
          }),
          /* @__PURE__ */ jsx(ColorPickerConfig, {
            propKey: PROP_KEY.MainColor,
            placeholder: t("Ch\u1ECDn m\xE0u"),
            formItemProps: {
              label: t("M\xE0u s\u1EAFc ch\xEDnh")
            }
          }),
          /* @__PURE__ */ jsx(InputNumberConfig, {
            propKey: PROP_KEY.StickyTop,
            formItemProps: {
              label: t("Kho\u1EA3ng c\xE1ch tr\xEAn khi scroll")
            },
            unitProps: { defaultValue: UnitEnum.pixel }
          })
        ]
      }),
      /* @__PURE__ */ jsx(Typography.Title, {
        level: 5,
        children: t("Danh s\xE1ch c\u1EEDa h\xE0ng")
      }),
      /* @__PURE__ */ jsx(StoreSetting, {})
    ]
  });
};

export { StoreListSetting };
