// @tempi/core-elements/dist/components/store-list/ui.es.js
import { jsx, jsxs } from 'react/jsx-runtime';
import styled from '@emotion/styled';
import { getStyle } from '@tempi/core-renderer';
import { useState, useEffect } from 'react';
import { useGetCitiesAndDistricts } from './hooks.es.js';
import { Store } from './components/Store.es.js';
import { Row } from '../../core/Grid/Row.es.js';
import { Col } from '../../core/Grid/Col.es.js';

const StoreList = ({
  stores,
  mainColor = "#2e7ac1",
  commonStyle,
  stickyTop,
  title
}) => {
  const [cityId, setCityId] = useState();
  const [districtId, setDistrictId] = useState();
  const [filteredStores, setFilteredStores] = useState(stores);
  const { cities: _cities, districts: _districts } = useGetCitiesAndDistricts();
  const cities = _cities.filter(
    (city) => !!stores.find((store) => store.cityId === city.idProvince)
  );
  const districts = _districts.filter(
    (district) => district.idProvince === cityId && !!stores.find((store) => store.districtId === district.idDistrict)
  );
  const handleChangeLocation = (_cityId, _districtId) => {
    if (_cityId) {
      setCityId(_cityId);
      setDistrictId(null);
    }
    if (_districtId) {
      setDistrictId(_districtId);
    }
  };
  useEffect(() => {
    if (!cityId || cityId === "0") {
      setFilteredStores(stores);
    } else if (cityId && (!districtId || districtId === "0")) {
      setFilteredStores(stores.filter((store) => store.cityId === cityId));
    } else
      setFilteredStores(
        stores.filter(
          (store) => store.cityId === cityId && store.districtId === districtId
        )
      );
  }, [cityId, districtId, stores]);
  return /* @__PURE__ */ jsx(Wrapper, {
    stickyTop,
    mainColor,
    commonStyle,
    children: /* @__PURE__ */ jsxs(Row, {
      gutter: [24, 16],
      children: [
        /* @__PURE__ */ jsxs(Col, {
          md: 5,
          xs: 12,
          children: [
            /* @__PURE__ */ jsxs("div", {
              className: "label",
              children: [
                "H\u1EC7 th\u1ED1ng ",
                stores.length,
                " c\u1EEDa h\xE0ng"
              ]
            }),
            /* @__PURE__ */ jsxs("div", {
              className: "select-location",
              children: [
                /* @__PURE__ */ jsx("h5", {
                  children: /* @__PURE__ */ jsx("b", {
                    children: title || "Ch\u1ECDn c\u1EEDa h\xE0ng theo t\u1EC9nh/th\xE0nh ph\u1ED1"
                  })
                }),
                /* @__PURE__ */ jsxs("select", {
                  onChange: (e) => handleChangeLocation(e.target.value),
                  children: [
                    /* @__PURE__ */ jsx("option", {
                      value: 0,
                      children: "Ch\u1ECDn t\u1EC9nh/ th\xE0nh ph\u1ED1"
                    }),
                    cities.map((city) => /* @__PURE__ */ jsx("option", {
                      value: city.idProvince,
                      children: city.name
                    }, city.idProvince))
                  ]
                }),
                /* @__PURE__ */ jsxs("select", {
                  onChange: (e) => handleChangeLocation(null, e.target.value),
                  children: [
                    /* @__PURE__ */ jsx("option", {
                      value: 0,
                      children: "Ch\u1ECDn qu\u1EADn/ huy\u1EC7n"
                    }),
                    districts.sort((a, b) => a.name < b.name ? -1 : 1).map((district) => /* @__PURE__ */ jsx("option", {
                      value: district.idDistrict,
                      children: district.name
                    }, district.idDistrict))
                  ]
                })
              ]
            })
          ]
        }),
        /* @__PURE__ */ jsxs(Col, {
          md: 7,
          xs: 12,
          children: [
            /* @__PURE__ */ jsxs("div", {
              className: "label",
              children: [
                filteredStores.length,
                " k\u1EBFt qu\u1EA3"
              ]
            }),
            /* @__PURE__ */ jsx("div", {
              className: "store-list",
              children: filteredStores.length > 0 ? filteredStores.map((store, i) => /* @__PURE__ */ jsx(Store, {
                store,
                mainColor
              }, i)) : /* @__PURE__ */ jsx(Empty, {
                mainColor,
                children: "Kh\xF4ng c\xF3 c\u1EEDa h\xE0ng"
              })
            })
          ]
        })
      ]
    })
  });
};
StoreList.displayName = "StoreList";
const Empty = styled.div(
  ({ mainColor }) => `
  font-size: 1.8em;
  text-align: center;
  padding: 2em;
  color: ${mainColor};
  opacity: 0.85;

`
);
const Wrapper = styled.div(
  ({ mainColor, commonStyle, stickyTop }) => `   
  ${getStyle(commonStyle)}

  .label{
    font-size: 1em;
    margin-bottom: 0.5em;
  }

  .select-location{    
    padding: 0.8em 1.5em;
    background-color: ${mainColor};
    border-radius: 3px;
    font-size: 1.1em;
    position: sticky;
    top: ${stickyTop}px;
    
    h5 {
      color: white;
      margin-top: 0.5em;
      margin-bottom: 1em;
    }
  
    select{
      width: 100%;
      margin-bottom: 1em;
      border-radius: 4px;
      -webkit-appearance: none;
      padding: 0.5em 3.5em 0.5em 1em;

      background-color: white;
      background-image: linear-gradient(45deg, transparent 50%, gray 50%), linear-gradient(135deg, gray 50%, transparent 50%), linear-gradient(to right, #ccc, #ccc);
      background-position: calc(100% - 20px) calc(1em + 2px), calc(100% - 15px) calc(1em + 2px), calc(100% - 2.5em) 0.5em;
      background-size: 5px 5px, 5px 5px, 1px 1.5em;
      background-repeat: no-repeat;      
    }
  }
  
  .store-list{
    border: 1px solid ${mainColor};
    border-radius: 5px;
    overflow: hidden;
    background-color: white;
  }
`
);

export { StoreList };
