// @tempi/core-elements/dist/components/header/selector/Header/HeaderEditor.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var coreEditor = require('@tempi/core-editor');
var constants = require('../../constants.js');
var HeaderQuickSetting = require('./HeaderQuickSetting.js');
var Header = require('../../ui/Header/Header.js');
var selector = require('../../../root/selector.js');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const HeaderEditor = coreEditor.withSelector(
  (props) => {
    const { realDevice: device } = coreEditor.useBuilder();
    return /* @__PURE__ */ jsxRuntime.jsx(Header.Header, __spreadProps(__spreadValues({}, props), {
      isMobile: device === "mobile"
    }));
  },
  {
    displayName: "\u0110\u1EA7u trang",
    tag: Header.Header.displayName.toLowerCase(),
    advanceAttributes: coreEditor.AdvanceAttributes,
    quickSetting: HeaderQuickSetting.HeaderQuickSetting,
    props: constants.DEFAULT_HEADER.header,
    rules: {
      canDrop: (targetNode) => {
        const allowTargets = [coreEditor.getEditorName({ RootEditor: selector.RootEditor })];
        return allowTargets.includes(targetNode.data.name);
      }
    }
  }
);

exports.HeaderEditor = HeaderEditor;
