// @tempi/core-elements/dist/components/header/selector/Header/HeaderQuickSetting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../../i18n.js');
var iconsDev = require('@tempi/icons-dev');
var coreEditor = require('@tempi/core-editor');
var HeaderSetting = require('./HeaderSetting.js');
var ColorPickerQuickSetting = require('../../../../quick-configs/ColorPickerQuickSetting/ColorPickerQuickSetting.js');
var BackgroundImageQuickSetting = require('../../../../quick-configs/BackgroundImageQuickSetting/BackgroundImageQuickSetting.js');

const HeaderQuickSetting = () => {
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SubQuickSetting, {
        content: HeaderSetting.HeaderSetting,
        children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.QuickSettingButton, {
          tooltip: i18n.t("T\xF9y ch\u1EC9nh Header"),
          children: /* @__PURE__ */ jsxRuntime.jsx(iconsDev.AntdCustomIcon, {
            name: "HeaderOutlined",
            style: {
              fontSize: 16
            }
          })
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx(ColorPickerQuickSetting.ColorPickerQuickSetting, {
        gradientPicker: true
      }),
      /* @__PURE__ */ jsxRuntime.jsx(BackgroundImageQuickSetting.BackgroundImageQuickSetting, {}),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.DuplicateNode, {}),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.OpenSettingDialog, {}),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.DeleteNode, {})
    ]
  });
};

exports.HeaderQuickSetting = HeaderQuickSetting;
