// @tempi/core-elements/dist/components/header/selector/Header/HeaderQuickSetting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../../../i18n.es.js';
import { AntdCustomIcon } from '@tempi/icons-dev';
import { SubQuickSetting, QuickSettingButton, DuplicateNode, OpenSettingDialog, DeleteNode } from '@tempi/core-editor';
import { HeaderSetting } from './HeaderSetting.es.js';
import { ColorPickerQuickSetting } from '../../../../quick-configs/ColorPickerQuickSetting/ColorPickerQuickSetting.es.js';
import { BackgroundImageQuickSetting } from '../../../../quick-configs/BackgroundImageQuickSetting/BackgroundImageQuickSetting.es.js';

const HeaderQuickSetting = () => {
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(SubQuickSetting, {
        content: HeaderSetting,
        children: /* @__PURE__ */ jsx(QuickSettingButton, {
          tooltip: t("T\xF9y ch\u1EC9nh Header"),
          children: /* @__PURE__ */ jsx(AntdCustomIcon, {
            name: "HeaderOutlined",
            style: {
              fontSize: 16
            }
          })
        })
      }),
      /* @__PURE__ */ jsx(ColorPickerQuickSetting, {
        gradientPicker: true
      }),
      /* @__PURE__ */ jsx(BackgroundImageQuickSetting, {}),
      /* @__PURE__ */ jsx(DuplicateNode, {}),
      /* @__PURE__ */ jsx(OpenSettingDialog, {}),
      /* @__PURE__ */ jsx(DeleteNode, {})
    ]
  });
};

export { HeaderQuickSetting };
