// @tempi/core-elements/dist/components/video/selector.es.js
import { jsx } from 'react/jsx-runtime';
import { useBuilder, useEditor, useNode, crafting, AdvanceAttributes, getEditorName } from '@tempi/core-editor';
import { DEFAULT_VIDEO } from './constants.es.js';
import { VideoQuickSetting } from './quick-setting.es.js';
import { VideoSetting } from './setting.es.js';
import { Video } from './ui.es.js';
import { RowEditor } from '../section/selector.es.js';
import { RootEditor } from '../root/selector.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const VideoEditor = (props) => {
  const { track } = useBuilder();
  const { enabled } = useEditor((state) => ({
    enabled: state.options.enabled
  }));
  const {
    connectors: { connect, drag }
  } = useNode();
  if (!enabled)
    return /* @__PURE__ */ jsx(Video, __spreadProps(__spreadValues({}, props), {
      isEditorMode: enabled
    }));
  return /* @__PURE__ */ jsx("div", {
    ref: (ref) => connect(drag(ref)),
    style: { cursor: "pointer" },
    children: /* @__PURE__ */ jsx(Video, __spreadProps(__spreadValues({
      onClick: (e) => {
        e.preventDefault();
        track == null ? void 0 : track({
          regionName: "videoSetting",
          contentName: "quickSettingBtn"
        });
      }
    }, props), {
      isEditorMode: enabled
    }))
  });
};
VideoEditor.craft = crafting({
  displayName: "Video",
  tag: Video.displayName.toLowerCase(),
  advanceAttributes: AdvanceAttributes,
  quickSetting: VideoQuickSetting,
  customAttributes: VideoSetting,
  props: DEFAULT_VIDEO,
  rules: {
    canDrop: (targetNode) => {
      const ignoreTargets = [
        getEditorName({ RowEditor }),
        getEditorName({ RootEditor })
      ];
      return !ignoreTargets.includes(targetNode.data.name);
    }
  }
});

export { VideoEditor };
