// @tempi/core-elements/dist/components/video/setting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { VideoConfig, SwitchConfig } from '@tempi/core-editor';
import { PROP_KEY } from '../../constants/propKey.es.js';

const VideoSetting = ({
  youtubeSrc = PROP_KEY.Src,
  uploadSrc = PROP_KEY.UploadSrc,
  autoplay = PROP_KEY.Autoplay,
  loop = PROP_KEY.Loop,
  showControllerPropKey = PROP_KEY.ShowController
}) => {
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(VideoConfig, {
        youtubeVideoPropKey: youtubeSrc,
        uploadVideoPropKey: uploadSrc
      }),
      /* @__PURE__ */ jsxs("div", {
        style: { marginTop: "86px" },
        children: [
          /* @__PURE__ */ jsx(SwitchConfig, {
            propKey: autoplay,
            formItemProps: {
              label: t("T\u1EF1 \u0111\u1ED9ng ph\xE1t")
            }
          }),
          /* @__PURE__ */ jsx(SwitchConfig, {
            propKey: loop,
            formItemProps: {
              label: t("L\u1EB7p l\u1EA1i")
            }
          }),
          /* @__PURE__ */ jsx(SwitchConfig, {
            propKey: showControllerPropKey,
            formItemProps: {
              label: t("Hi\u1EC3n th\u1ECB thanh \u0111i\u1EC1u khi\u1EC3n Video")
            }
          })
        ]
      })
    ]
  });
};

export { VideoSetting };
