// @tempi/core-elements/dist/components/video/setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var propKey = require('../../constants/propKey.js');

const VideoSetting = ({
  youtubeSrc = propKey.PROP_KEY.Src,
  uploadSrc = propKey.PROP_KEY.UploadSrc,
  autoplay = propKey.PROP_KEY.Autoplay,
  loop = propKey.PROP_KEY.Loop,
  showControllerPropKey = propKey.PROP_KEY.ShowController
}) => {
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.VideoConfig, {
        youtubeVideoPropKey: youtubeSrc,
        uploadVideoPropKey: uploadSrc
      }),
      /* @__PURE__ */ jsxRuntime.jsxs("div", {
        style: { marginTop: "86px" },
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
            propKey: autoplay,
            formItemProps: {
              label: i18n.t("T\u1EF1 \u0111\u1ED9ng ph\xE1t")
            }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
            propKey: loop,
            formItemProps: {
              label: i18n.t("L\u1EB7p l\u1EA1i")
            }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
            propKey: showControllerPropKey,
            formItemProps: {
              label: i18n.t("Hi\u1EC3n th\u1ECB thanh \u0111i\u1EC1u khi\u1EC3n Video")
            }
          })
        ]
      })
    ]
  });
};

exports.VideoSetting = VideoSetting;
