// @tempi/core-editor/dist/configs/TimingConfig/TimeVisibilityConfig.es.js
import { jsx, Fragment } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { Modal, Button } from 'antd';
import { useFrame } from 'react-frame-component';
import { TimingConfig } from './TimingConfig.es.js';
import { usePropSeparateDevice } from '../../hooks/usePropSeparateDevice.es.js';
import { PROP_KEY } from '../../constants/propKey.es.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
const TimeVisibilityConfig = ({
  propKey,
  separateDevice = false,
  visible,
  setVisible
}) => {
  const [startTime, setStartTime] = usePropSeparateDevice(
    PROP_KEY.joinKey(propKey, PROP_KEY.TimeVisibility, PROP_KEY.StartTime),
    separateDevice
  );
  const [endTime, setEndTime] = usePropSeparateDevice(
    PROP_KEY.joinKey(propKey, PROP_KEY.TimeVisibility, PROP_KEY.EndTime),
    separateDevice
  );
  const [ruleRepeat, setRuleRepeat] = usePropSeparateDevice(
    PROP_KEY.joinKey(propKey, PROP_KEY.TimeVisibility, PROP_KEY.RuleRepeat),
    separateDevice
  );
  const [date, setDate] = usePropSeparateDevice(
    PROP_KEY.joinKey(propKey, PROP_KEY.TimeVisibility, PROP_KEY.Date),
    separateDevice
  );
  const [duration, setDuration] = usePropSeparateDevice(
    PROP_KEY.joinKey(propKey, PROP_KEY.TimeVisibility, PROP_KEY.Duration),
    separateDevice
  );
  const { document: frameDocument } = useFrame();
  return /* @__PURE__ */ jsx(Fragment, {
    children: /* @__PURE__ */ jsx(Modal, {
      title: t("\u0110\u1EB7t gi\u1EDD hi\u1EC3n th\u1ECB"),
      onCancel: () => setVisible(false),
      open: visible,
      destroyOnClose: true,
      bodyStyle: { paddingTop: 0 },
      getContainer: () => (frameDocument == null ? void 0 : frameDocument.body) || (document == null ? void 0 : document.body),
      closable: false,
      footer: [
        /* @__PURE__ */ jsx(Button, {
          type: "primary",
          shape: "round",
          onClick: () => setVisible(false),
          children: t("\u0110\xF3ng")
        })
      ],
      zIndex: 1080,
      children: /* @__PURE__ */ jsx(TimingConfig, __spreadValues({}, {
        startTime,
        endTime,
        ruleRepeat,
        date,
        duration,
        setDate,
        setStartTime,
        setEndTime,
        setRuleRepeat,
        setDuration
      }))
    })
  });
};

export { TimeVisibilityConfig };
