// @tempi/core-editor/dist/configs/TimingConfig/TimingConfig.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import styled from '@emotion/styled';
import dayjs from 'dayjs';
import { useState, useEffect, useRef } from 'react';
import { RepeatTypeEnum, parseRuleRepeat, RuleRepeatEnum, UnitRepeatEnum } from '@tempi/core-renderer';
import { DatePicker } from 'antd';
import { SelectSegment } from './SelectSegment.es.js';
import { t } from '../../i18n.es.js';
import { SelectConfig } from '../SelectConfig/SelectConfig.es.js';
import { FormItem } from '../../components/FormItem/FormItem.es.js';
import { InputNumberConfig } from '../InputNumberConfig/InputNumberConfig.es.js';

const formatDate = (date) => {
  if (!dayjs.isDayjs(date))
    return void 0;
  return dayjs(date).toISOString();
};
const TimingConfig = ({
  startTime,
  endTime,
  ruleRepeat,
  date,
  duration,
  setStartTime,
  setEndTime,
  setRuleRepeat,
  setDate,
  setDuration
}) => {
  var _a;
  const [unitRepeat, setUnitRepeat] = useState();
  const [endTimeRepeat, setEndTimeRepeat] = useState("");
  const [segmentSelected, setSegmentSelected] = useState();
  const [numberRepeat, setNumberRepeat] = useState();
  const [typeSelectTime, setTypeSelectTime] = useState("");
  const [startDuration, setStartDuration] = useState();
  const [endDuration, setEndDuration] = useState();
  const timeFormat = "DD/MM/YYYY HH:mm:ss";
  useEffect(() => {
    if (duration) {
      const [start, end] = duration;
      setStartDuration(start);
      setEndDuration(end);
    }
    initRuleRepeat();
  }, []);
  const initRuleRepeat = () => {
    var _a2;
    if (!ruleRepeat) {
      (startTime || endTime) && setTypeSelectTime(RepeatTypeEnum.NO_REPEAT);
      return;
    }
    const freq = parseRuleRepeat(RuleRepeatEnum.FREQ, ruleRepeat);
    const interval = parseRuleRepeat(RuleRepeatEnum.INTERVAL, ruleRepeat) || "1";
    const until = parseRuleRepeat(RuleRepeatEnum.UNTIL, ruleRepeat);
    const byday = parseRuleRepeat(RuleRepeatEnum.BYDAY, ruleRepeat);
    setNumberRepeat(parseInt(interval));
    if (freq)
      setUnitRepeat(freq);
    if (until) {
      setEndTimeRepeat(until);
    }
    if (byday)
      setSegmentSelected((_a2 = byday.split(",")) == null ? void 0 : _a2.map((item) => parseInt(item)));
    if (freq === UnitRepeatEnum.DURATION || startTime || endTime) {
      setTypeSelectTime(RepeatTypeEnum.NO_REPEAT);
    }
    if (+interval === 1 && freq === UnitRepeatEnum.DAY) {
      setTypeSelectTime(RepeatTypeEnum.DAILY);
    }
    if (+interval > 1 || freq !== UnitRepeatEnum.DAY && freq !== UnitRepeatEnum.DURATION) {
      setTypeSelectTime(RepeatTypeEnum.CUSTOM);
    }
  };
  const convertToDateTime = (time) => {
    if (time === void 0)
      return void 0;
    const day = date ? dayjs(date) : dayjs();
    const hour = Math.floor(time / 60);
    const minute = time - hour * 60;
    return day.hour(hour).minute(minute);
  };
  const disabledStartDate = (current) => {
    if (!endTime)
      return false;
    return current == null ? void 0 : current.isAfter(endTime, "day");
  };
  const disabledEndDate = (current) => {
    if (!startTime)
      return false;
    return current == null ? void 0 : current.isBefore(startTime, "day");
  };
  const setTime = (startTime2, endTime2) => {
    if (startTime2 && endTime2 && dayjs(startTime2).isAfter(endTime2)) {
      setStartTime(endTime2);
      setEndTime(startTime2);
    } else {
      setStartTime(startTime2);
      setEndTime(endTime2);
    }
  };
  const onChangeSegment = (item) => {
    setSegmentSelected((prev) => {
      const indexOfItem = prev == null ? void 0 : prev.indexOf(item);
      if (indexOfItem !== void 0) {
        if (indexOfItem < 0) {
          return [...prev || [], item];
        } else {
          const res = [...prev || []];
          res.splice(indexOfItem, 1);
          return res;
        }
      }
    });
  };
  useEffect(() => {
    handlerRuleRepeat({
      freq: unitRepeat,
      byDay: segmentSelected == null ? void 0 : segmentSelected.join(","),
      until: endTimeRepeat,
      interval: numberRepeat
    });
  }, [
    unitRepeat,
    segmentSelected,
    endTimeRepeat,
    typeSelectTime,
    numberRepeat
  ]);
  const handlerRuleRepeat = ({
    freq,
    byDay,
    until,
    interval
  }) => {
    const newRuleRepeat = `${freq ? `${RuleRepeatEnum.FREQ}:${freq};` : ""}${interval ? `${RuleRepeatEnum.INTERVAL}:${interval};` : ""}${byDay ? `${RuleRepeatEnum.BYDAY}:${byDay};` : ""}${until ? `${RuleRepeatEnum.UNTIL}:${until};` : ""}`;
    if (!ruleRepeat && !newRuleRepeat)
      return;
    setRuleRepeat(newRuleRepeat);
  };
  const handlerDuration = (value) => {
    const [start, end] = value || [void 0, void 0];
    const startToMinute = start ? start.hour() * 60 + start.minute() : void 0;
    const endToMinute = end ? end.hour() * 60 + end.minute() : void 0;
    setStartDuration(startToMinute);
    setEndDuration(endToMinute);
    setDuration(
      startToMinute !== void 0 && endToMinute !== void 0 ? [startToMinute, endToMinute] : void 0
    );
  };
  const onChangeTypeSelectTime = (value) => {
    resetRepeat();
    setTypeSelectTime(value);
    switch (value) {
      case RepeatTypeEnum.CUSTOM:
        setUnitRepeat(UnitRepeatEnum.WEEK);
        setSegmentSelected([1]);
        setNumberRepeat(1);
        break;
      case RepeatTypeEnum.DAILY:
        setUnitRepeat(UnitRepeatEnum.DAY);
        setNumberRepeat(1);
        break;
      case RepeatTypeEnum.NO_REPEAT:
        setUnitRepeat(UnitRepeatEnum.DURATION);
        break;
    }
  };
  const textDescForRuleRepeat = (interval, unitRepeat2, dateTime) => {
    if (!dateTime || !(unitRepeat2 === UnitRepeatEnum.YEAR || unitRepeat2 === UnitRepeatEnum.MONTH))
      return "";
    const unit = {
      month: t("th\xE1ng"),
      year: t("n\u0103m")
    };
    const format = unitRepeat2 == UnitRepeatEnum.YEAR ? "DD/MM" : "DD";
    return `${interval > 1 ? `${interval} ${unit[unitRepeat2.toLowerCase()]} ${t("m\u1ED9t l\u1EA7n")}` : `${t("H\xE0ng")} ${unit[unitRepeat2.toLowerCase()]}`} ${t("v\xE0o ng\xE0y")} ${dayjs(dateTime).format(format)}`;
  };
  const resetRepeat = () => {
    setStartTime(void 0);
    setEndTime(void 0);
    setDate(void 0);
    handlerDuration(void 0);
    setNumberRepeat(void 0);
    setUnitRepeat(void 0);
    setSegmentSelected(void 0);
    setEndTimeRepeat(void 0);
  };
  const isOldConfig = () => {
    if (numberRepeat && numberRepeat > 1)
      return true;
    if (unitRepeat !== UnitRepeatEnum.WEEK && unitRepeat !== UnitRepeatEnum.DAY)
      return true;
    return false;
  };
  const ref = useRef(null);
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx("div", {
        ref,
        style: { display: "none" }
      }),
      /* @__PURE__ */ jsx(SelectConfig, {
        options: [
          { label: t("Lu\xF4n hi\u1EC3n th\u1ECB"), value: "" },
          { label: t("Kho\u1EA3ng th\u1EDDi gian"), value: RepeatTypeEnum.NO_REPEAT },
          { label: t("L\u1EB7p l\u1EA1i h\xE0ng ng\xE0y"), value: RepeatTypeEnum.DAILY },
          {
            label: t("T\xF9y ch\u1EC9nh"),
            value: RepeatTypeEnum.CUSTOM
          }
        ],
        formItemProps: {
          label: t("Ki\u1EC3u hi\u1EC3n th\u1ECB"),
          layout: "vertical"
        },
        selectProps: {
          style: {
            width: "100%"
          },
          value: typeSelectTime,
          onChange: onChangeTypeSelectTime,
          allowClear: false
        }
      }),
      typeSelectTime === RepeatTypeEnum.NO_REPEAT && /* @__PURE__ */ jsxs(Fragment, {
        children: [
          /* @__PURE__ */ jsx(FormItem, {
            label: t("Th\u1EDDi gian b\u1EAFt \u0111\u1EA7u"),
            layout: "vertical",
            children: /* @__PURE__ */ jsx(StyledDatePicker, {
              showTime: true,
              placeholder: t("Ch\u1ECDn th\u1EDDi gian b\u1EAFt \u0111\u1EA7u"),
              disabledDate: disabledStartDate,
              format: timeFormat,
              value: startTime ? dayjs(startTime) : void 0,
              onChange: (value) => setTime(formatDate(value), endTime)
            })
          }),
          /* @__PURE__ */ jsx(FormItem, {
            label: t("Th\u1EDDi gian k\u1EBFt th\xFAc"),
            layout: "vertical",
            children: /* @__PURE__ */ jsx(StyledDatePicker, {
              showTime: true,
              placeholder: t("Ch\u1ECDn th\u1EDDi gian k\u1EBFt th\xFAc"),
              disabledDate: disabledEndDate,
              format: timeFormat,
              value: endTime ? dayjs(endTime) : void 0,
              onChange: (value) => setTime(startTime, formatDate(value))
            })
          })
        ]
      }),
      (typeSelectTime === RepeatTypeEnum.CUSTOM || typeSelectTime === RepeatTypeEnum.DAILY) && /* @__PURE__ */ jsxs(Fragment, {
        children: [
          isOldConfig() && /* @__PURE__ */ jsxs(Fragment, {
            children: [
              /* @__PURE__ */ jsx(FormItem, {
                label: t("L\u1EB7p l\u1EA1i m\u1ED7i"),
                style: { marginBottom: 0 },
                children: /* @__PURE__ */ jsxs(StyledDiv, {
                  children: [
                    /* @__PURE__ */ jsx(InputNumberConfig, {
                      inputNumberProps: {
                        min: 1,
                        value: numberRepeat,
                        onChange: (value) => setNumberRepeat(value)
                      }
                    }),
                    /* @__PURE__ */ jsx(SelectConfig, {
                      options: [
                        { label: t("Ng\xE0y"), value: UnitRepeatEnum.DAY },
                        { label: t("Tu\u1EA7n"), value: UnitRepeatEnum.WEEK },
                        {
                          label: t("Th\xE1ng"),
                          value: UnitRepeatEnum.MONTH
                        },
                        { label: t("N\u0103m"), value: UnitRepeatEnum.YEAR }
                      ],
                      selectProps: {
                        value: unitRepeat,
                        onChange: (value) => {
                          setUnitRepeat(value);
                        }
                      }
                    })
                  ]
                })
              }),
              !!numberRepeat && !!unitRepeat && !!textDescForRuleRepeat(numberRepeat, unitRepeat, date) && /* @__PURE__ */ jsx("div", {
                style: { marginBottom: 16, color: "#82869E" },
                children: textDescForRuleRepeat(numberRepeat, unitRepeat, date)
              })
            ]
          }),
          unitRepeat === UnitRepeatEnum.WEEK && /* @__PURE__ */ jsx(FormItem, {
            label: t("L\u1EB7p l\u1EA1i v\xE0o"),
            layout: "vertical",
            children: /* @__PURE__ */ jsx(SelectSegment, {
              arrSelected: segmentSelected,
              onchange: onChangeSegment
            })
          }),
          /* @__PURE__ */ jsx(FormItem, {
            label: t("Hi\u1EC3n th\u1ECB trong khung gi\u1EDD"),
            layout: "vertical",
            children: /* @__PURE__ */ jsx(StyledTimePicker, {
              placeholder: [t("B\u1EAFt \u0111\u1EA7u"), t("K\u1EBFt th\xFAc")],
              format: "HH:mm",
              picker: "time",
              value: [
                convertToDateTime(startDuration),
                convertToDateTime(endDuration)
              ],
              changeOnBlur: ((_a = ref == null ? void 0 : ref.current) == null ? void 0 : _a.ownerDocument) !== document,
              allowClear: true,
              onChange: handlerDuration
            })
          }),
          /* @__PURE__ */ jsx(FormItem, {
            label: t("Ng\xE0y b\u1EAFt \u0111\u1EA7u"),
            layout: "vertical",
            children: /* @__PURE__ */ jsx(StyledDatePicker, {
              placeholder: t("Ch\u1ECDn th\u1EDDi gian b\u1EAFt \u0111\u1EA7u"),
              value: date ? dayjs(date) : void 0,
              format: "DD/MM/YYYY",
              onChange: (value) => setDate(formatDate(value))
            })
          }),
          /* @__PURE__ */ jsx(FormItem, {
            label: t("Ng\xE0y k\u1EBFt th\xFAc"),
            layout: "vertical",
            children: /* @__PURE__ */ jsx(StyledDatePicker, {
              placeholder: t("Ch\u1ECDn th\u1EDDi gian k\u1EBFt th\xFAc"),
              format: "DD/MM/YYYY",
              value: endTimeRepeat ? dayjs(endTimeRepeat) : void 0,
              onChange: (value) => setEndTimeRepeat(formatDate(value))
            })
          })
        ]
      })
    ]
  });
};
const StyledDatePicker = styled(DatePicker)`
  width: 100%;
`;
const StyledDiv = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
`;
const StyledTimePicker = styled(DatePicker.RangePicker)`
  width: 100%;
`;

export { TimingConfig };
