// @tempi/core-elements/dist/components/global-block/constants.es.js
import { Alignment } from '@tempi/core-renderer';
import { BLOCK_TYPES } from './type.es.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
const DEFAULT_GLOBAL_HEADER_DESKTOP = {
  stickyTop: true,
  menuConfig: {
    commonStyle: {
      columnGap: 32
    },
    align: Alignment.right
  },
  logoConfig: {
    show: true,
    src: "https://lh3.googleusercontent.com/bhg6FoxZPA5S6q50_Gmu6YgMt2GcKxpjsbnaRvvHUwSV9JSnAj83TpbJEHIJjgKnbNoZmM7Mnx0pBlUau5CG2yIOG5v7rGc",
    align: Alignment.left,
    commonStyle: {
      width: 100
    },
    spacingWithMenu: 32
  },
  menuItemConfig: {
    commonStyle: {
      color: "#000000",
      hover: {
        color: "#008cff"
      }
    }
  },
  commonStyle: {
    backgroundColor: "#ffffff",
    padding: {
      top: 30,
      right: 16,
      bottom: 30,
      left: 16
    }
  }
};
const DEFAULT_GLOBAL_HEADER_MOBILE = {
  commonStyle: {
    padding: {
      top: 24,
      right: 16,
      bottom: 24,
      left: 16
    }
  }
};
const DEFAULT_GLOBAL_BLOCK_HEADER = {
  desktop: __spreadValues({
    blockType: BLOCK_TYPES.HEADER
  }, DEFAULT_GLOBAL_HEADER_DESKTOP),
  mobile: __spreadValues({
    blockType: BLOCK_TYPES.HEADER
  }, DEFAULT_GLOBAL_HEADER_MOBILE),
  blockType: BLOCK_TYPES.HEADER,
  isActive: true
};
const DEFAULT_SECTION_PADDING_STYLE = {
  top: 8,
  bottom: 8,
  left: 16,
  right: 16
};
const DEFAULT_GLOBAL_BLOCK = {
  gutter: [0, 0],
  commonStyle: {
    padding: DEFAULT_SECTION_PADDING_STYLE
  }
};

export { DEFAULT_GLOBAL_BLOCK, DEFAULT_GLOBAL_BLOCK_HEADER, DEFAULT_GLOBAL_HEADER_DESKTOP, DEFAULT_GLOBAL_HEADER_MOBILE };
