// @tempi/core-elements/dist/components/global-block/selector.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var coreEditor = require('@tempi/core-editor');
var quickSetting$1 = require('../section/quick-setting.js');
var advancedSetting = require('./advanced-setting.js');
var setting = require('./setting.js');
var styleSetting = require('./style-setting.js');
var type = require('./type.js');
var ui = require('./ui.js');
var RenderFloatingQuickSetting = require('../../core/RenderFloatingQuickSetting/RenderFloatingQuickSetting.js');
var quickSetting = require('./header-v2/quick-setting.js');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const GlobalBlockEditor = (_a) => {
  var _b = _a, {
    addBlockButton
  } = _b, restProps = __objRest(_b, [
    "addBlockButton"
  ]);
  const { blockType } = restProps;
  const { menu } = coreEditor.useBuilder();
  const {
    connectors: { connect }
  } = coreEditor.useNode();
  return blockType === type.BLOCK_TYPES.HEADER ? /* @__PURE__ */ jsxRuntime.jsx("div", {
    ref: connect,
    children: /* @__PURE__ */ jsxRuntime.jsxs(RenderFloatingQuickSetting.RenderFloatingQuickSetting, {
      quickSetting: /* @__PURE__ */ jsxRuntime.jsx(quickSetting.HeaderV2QuickSetting, {}),
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(ui.GlobalBlock, __spreadProps(__spreadValues({}, restProps), {
          isEditingMode: true,
          menu
        })),
        addBlockButton
      ]
    })
  }) : /* @__PURE__ */ jsxRuntime.jsx("div", {
    ref: connect,
    style: {
      width: "100%",
      position: "relative"
    },
    children: /* @__PURE__ */ jsxRuntime.jsx(RenderFloatingQuickSetting.RenderFloatingQuickSetting, {
      quickSetting: /* @__PURE__ */ jsxRuntime.jsx(quickSetting$1.SectionQuickSetting, {}),
      children: /* @__PURE__ */ jsxRuntime.jsx(ui.GlobalBlock, __spreadProps(__spreadValues({}, restProps), {
        isEditingMode: true
      }))
    })
  });
};
GlobalBlockEditor.craft = coreEditor.crafting({
  displayName: ui.GlobalBlock.displayName,
  tag: ui.GlobalBlock.displayName.toLowerCase(),
  advanceAttributes: advancedSetting.GlobalBlockAdvancedSetting,
  customAttributes: setting.GlobalBlockSetting,
  commonAttributes: styleSetting.GlobalBlockStyleSetting,
  rules: {
    canDrag: () => false,
    canMoveIn: (_incomingNodes, currentNode) => {
      return currentNode.data.props.type !== type.BLOCK_TYPES.HEADER;
    }
  }
});

exports.GlobalBlockEditor = GlobalBlockEditor;
