// @tempi/core-editor/dist/configs/SwitchConfig/SwitchConfig.es.js
import { jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { Switch } from 'antd';
import { usePropSeparateDevice } from '../../hooks/usePropSeparateDevice.es.js';
import { FormItem } from '../../components/FormItem/FormItem.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const SwitchConfig = (props) => {
  const { propKey, switchProps, formItemProps, separateDevice = false } = props;
  const [prop, setProp] = usePropSeparateDevice(
    propKey,
    separateDevice
  );
  return /* @__PURE__ */ jsx(FormItem, __spreadProps(__spreadValues({}, formItemProps), {
    children: /* @__PURE__ */ jsx(Switch, __spreadValues({
      checkedChildren: t("B\u1EADt"),
      unCheckedChildren: t("T\u1EAFt"),
      checked: prop,
      onChange: (checked) => setProp(checked)
    }, switchProps))
  }));
};

export { SwitchConfig };
