// @tempi/core-elements/dist/components/global-block/header-v2/ui/components/Item.es.js
import styled from '@emotion/styled';
import { Alignment } from '@tempi/core-renderer';
import { MOBILE_WIDTH } from '../../constants.es.js';

const Item = styled.li`
  user-select: none;
`;
const ItemLabel = styled.label`
  display: block;
`;
const ItemContentWrapper = styled.div(
  ({ alignMenu, hoverColor }) => `
  display: flex;
  align-items: center;
  text-align: left;
  @media screen and (max-width: ${MOBILE_WIDTH}px) {
    grid-template-columns: 
      ${alignMenu === Alignment.left ? "auto 1fr auto" : ""}
      ${alignMenu === Alignment.center ? "1fr auto 1fr" : ""}
      ${alignMenu === Alignment.right ? "1fr auto" : ""}
    ;
    display: grid;
  }

  ${hoverColor ? `a:hover ~ svg{
    color: ${hoverColor} !important;
  }` : ""}
`
);
const ItemContent = styled.a(({ grid, alignMenu }) => {
  const {
    menuItemSpacing,
    menuColor,
    menuHoverColor,
    mobileNavigationTextAlign
  } = grid || {};
  return `
  position: relative;
  align-self: center;
  margin: 0 0 0 ${menuItemSpacing}px;
  text-decoration: none;
  white-space: nowrap;
  cursor: pointer;
  color: ${menuColor};

  &:hover {
    text-decoration: underline;
    text-underline-offset: 7px;
    color: ${menuHoverColor};
  }

  @media screen and (max-width: ${MOBILE_WIDTH}px){
    margin: 0;
    text-align: ${mobileNavigationTextAlign};

    grid-column: 
    ${alignMenu === Alignment.left ? "1/2" : ""}
    ${alignMenu === Alignment.center ? "2/3" : ""}
    ${alignMenu === Alignment.right ? "3/4" : ""}
  }
`;
});
const ItemIcon = styled.img`
  width: 20px;
  height: 20px;
  margin-right: 0.4em;
`;

export { Item, ItemContent, ItemContentWrapper, ItemIcon, ItemLabel };
