// @tempi/core-elements/dist/components/contact-bubble/constants.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var type = require('./type.js');

const DEFAULT_CONTACT_BUBBLE_ITEMS = [
  {
    title: "Facebook",
    icon: {
      src: "https://lh3.googleusercontent.com/qDVYa2xp4agHphLw8UW983epO55Fbr2UeoQICoIX6OsS7-9OJaC80LF0RxWixBnusNBhd0E4NWYCIHK9d6UdJ6Ut9UzMO0I4rA=rw-w200"
    },
    href: "",
    size: 50
  },
  {
    title: "Messenger",
    icon: {
      src: "https://lh3.googleusercontent.com/Y4thC3FPnYVBwkSW-II5lOsI09U3dICX789slwIIrItvfQ24TW__Whe1d8d1XI8de0D4U6v6y9nUQ5R1GGnop7X-Zdhnk36m=rw-w200"
    },
    href: "",
    size: 50
  }
];
const DEFAULT_CONTACT_BUBBLE = {
  enable: true,
  desktop: {
    items: DEFAULT_CONTACT_BUBBLE_ITEMS,
    layout: type.ContactBubbleLayout.VERTICAL,
    position: type.ContactBubblePosition.BOTTOM_RIGHT,
    gap: {
      horizontal: 20,
      vertical: 20
    }
  },
  mobile: {
    items: DEFAULT_CONTACT_BUBBLE_ITEMS,
    layout: type.ContactBubbleLayout.FLASH,
    position: type.ContactBubblePosition.BOTTOM_RIGHT,
    gap: {
      horizontal: 20,
      vertical: 20
    }
  }
};

exports.DEFAULT_CONTACT_BUBBLE = DEFAULT_CONTACT_BUBBLE;
exports.DEFAULT_CONTACT_BUBBLE_ITEMS = DEFAULT_CONTACT_BUBBLE_ITEMS;
