// @tempi/core-elements/dist/components/contact-bubble/selector.es.js
import { jsx } from 'react/jsx-runtime';
import { useBuilder, crafting } from '@tempi/core-editor';
import { ContactBubbleSetting } from './setting.es.js';
import { ContactBubble } from './ui.es.js';
import { DEFAULT_CONTACT_BUBBLE } from './constants.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const ContactBubbleEditor = (props) => {
  const { hideBubbleOnEditor } = useBuilder();
  if (hideBubbleOnEditor)
    return null;
  return /* @__PURE__ */ jsx(ContactBubble, __spreadProps(__spreadValues({}, props), {
    isEditorMode: true
  }));
};
ContactBubbleEditor.craft = crafting({
  displayName: ContactBubble.displayName,
  tag: ContactBubble.displayName.toLowerCase(),
  customAttributes: ContactBubbleSetting,
  props: DEFAULT_CONTACT_BUBBLE
});

export { ContactBubbleEditor };
