// @tempi/core-elements/dist/components/contact-bubble/setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var antd = require('antd');
var coreEditor = require('@tempi/core-editor');
var reactIcons = require('@fluentui/react-icons');
var styled = require('@emotion/styled');
var BubbleSettingByDevice = require('./settings/BubbleSettingByDevice.js');
var propKey = require('../../constants/propKey.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

const ContactBubbleSetting = () => {
  const { relatedNodeId } = coreEditor.useBuilder();
  const [enable] = coreEditor.useProp(propKey.PROP_KEY.Enable);
  if (!enable)
    return null;
  return /* @__PURE__ */ jsxRuntime.jsx(jsxRuntime.Fragment, {
    children: /* @__PURE__ */ jsxRuntime.jsx(antd.Tabs, {
      centered: true,
      type: "card",
      defaultActiveKey: (relatedNodeId == null ? void 0 : relatedNodeId.device) || "desktop",
      items: [
        {
          label: /* @__PURE__ */ jsxRuntime.jsxs(Label, {
            children: [
              /* @__PURE__ */ jsxRuntime.jsx(reactIcons.Desktop16Regular, {
                style: { marginRight: "0.25em" }
              }),
              i18n.t("M\xE1y t\xEDnh")
            ]
          }),
          key: "desktop",
          children: /* @__PURE__ */ jsxRuntime.jsx(BubbleSettingByDevice.BubbleSettingByDevice, {
            device: "desktop"
          })
        },
        {
          label: /* @__PURE__ */ jsxRuntime.jsxs(Label, {
            children: [
              /* @__PURE__ */ jsxRuntime.jsx(reactIcons.Phone16Regular, {
                style: { marginRight: "0.25em" }
              }),
              i18n.t("\u0110i\u1EC7n tho\u1EA1i")
            ]
          }),
          key: "mobile",
          children: /* @__PURE__ */ jsxRuntime.jsx(BubbleSettingByDevice.BubbleSettingByDevice, {
            device: "mobile"
          })
        }
      ]
    })
  });
};
const Label = styled__default["default"].div`
  display: flex;
  align-items: center;
`;

exports.ContactBubbleSetting = ContactBubbleSetting;
