// @tempi/core-elements/dist/components/contact-bubble/setting.es.js
import { jsx, Fragment, jsxs } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { Tabs } from 'antd';
import { useBuilder, useProp } from '@tempi/core-editor';
import { Desktop16Regular, Phone16Regular } from '@fluentui/react-icons';
import styled from '@emotion/styled';
import { BubbleSettingByDevice } from './settings/BubbleSettingByDevice.es.js';
import { PROP_KEY } from '../../constants/propKey.es.js';

const ContactBubbleSetting = () => {
  const { relatedNodeId } = useBuilder();
  const [enable] = useProp(PROP_KEY.Enable);
  if (!enable)
    return null;
  return /* @__PURE__ */ jsx(Fragment, {
    children: /* @__PURE__ */ jsx(Tabs, {
      centered: true,
      type: "card",
      defaultActiveKey: (relatedNodeId == null ? void 0 : relatedNodeId.device) || "desktop",
      items: [
        {
          label: /* @__PURE__ */ jsxs(Label, {
            children: [
              /* @__PURE__ */ jsx(Desktop16Regular, {
                style: { marginRight: "0.25em" }
              }),
              t("M\xE1y t\xEDnh")
            ]
          }),
          key: "desktop",
          children: /* @__PURE__ */ jsx(BubbleSettingByDevice, {
            device: "desktop"
          })
        },
        {
          label: /* @__PURE__ */ jsxs(Label, {
            children: [
              /* @__PURE__ */ jsx(Phone16Regular, {
                style: { marginRight: "0.25em" }
              }),
              t("\u0110i\u1EC7n tho\u1EA1i")
            ]
          }),
          key: "mobile",
          children: /* @__PURE__ */ jsx(BubbleSettingByDevice, {
            device: "mobile"
          })
        }
      ]
    })
  });
};
const Label = styled.div`
  display: flex;
  align-items: center;
`;

export { ContactBubbleSetting };
