// @tempi/core-elements/dist/constants/grid.es.js
import { VerticalAlignment, Alignment } from '@tempi/core-renderer';
import { WidthTypeEnum } from '../types/index.es.js';

const BREAK_POINTS = {
  xs: "480px",
  sm: "576px",
  md: "768px",
  lg: "1024px",
  xl: "1440px",
  xxl: "1600px"
};
const GRID_COLUMNS = 12;
const MAPPING_VERTICAL_ALIGNMENT_FLEX = {
  [VerticalAlignment.top]: "flex-start",
  [VerticalAlignment.center]: "center",
  [VerticalAlignment.bottom]: "flex-end"
};
const MAPPING_ALIGNMENT_FLEX = {
  [Alignment.center]: "center",
  [Alignment.left]: "flex-start",
  [Alignment.right]: "flex-end",
  [Alignment.justify]: "space-between"
};
const DEFAULT_WIDTH = 1232;
const DEFAULT_HEIGHT = 400;
const DEFAULT_GUTTER = [0, 0];
const WIDTH_TYPE_OPTIONS = [
  { label: "common.full_width", value: WidthTypeEnum.full },
  { label: "common.custom", value: WidthTypeEnum.fixed }
];

export { BREAK_POINTS, DEFAULT_GUTTER, DEFAULT_HEIGHT, DEFAULT_WIDTH, GRID_COLUMNS, MAPPING_ALIGNMENT_FLEX, MAPPING_VERTICAL_ALIGNMENT_FLEX, WIDTH_TYPE_OPTIONS };
