// @tempi/core-elements/dist/constants/propKey.es.js
import { PROP_KEY as PROP_KEY$1 } from '@tempi/core-editor';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const PROP_KEY = __spreadProps(__spreadValues({}, PROP_KEY$1), {
  Name: "name",
  Color: "color",
  Size: "size",
  TextAlign: "textAlign",
  WidthType: "widthType",
  ContentWidthType: "contentWidthType",
  ContentMaxWidthKey: "row.commonStyle.maxWidth",
  HiddenXs: "responsive.hiddenXs",
  HiddenMd: "responsive.hiddenMd",
  Md: "md.flex",
  Xs: "xs.flex",
  MinHeight: "minHeight",
  Gutter: "gutter",
  Spacing: "spacing",
  Icon: "icon",
  IconConfig: "iconConfig",
  SearchConfig: "searchConfig",
  Position: "position",
  ButtonStyle: "buttonStyle",
  Shadow: "shadow",
  JustifyContent: "justifyContent",
  Text: "text",
  Align: "align",
  Weight: "weight",
  Gap: "gap",
  Style: "style",
  TextConfig: "textConfig",
  Type: "type",
  CountdownStyle: "countdownStyle",
  ShowLabel: "showLabel",
  Layout: "layout",
  ElementGap: "elementGap",
  LabelGap: "labelGap",
  LabelPosition: "labelPosition",
  DigitStyle: "digitStyle",
  LabelStyle: "labelStyle",
  LetterSpacing: "letterSpacing",
  TextTransform: "textTransform",
  TextDecoration: "textDecoration",
  FontStyle: "fontStyle",
  FontVariant: "fontVariant",
  TextShadow: "textShadow",
  Address: "address",
  Zoom: "zoom",
  Alignment: "alignment",
  ZoomPreview: "zoomPreview",
  Images: "images",
  ColumnCount: "columnCount",
  HtmlCode: "htmlCode",
  BarStyle: "barStyle",
  CircleStyle: "circleStyle",
  TextStyle: "textStyle",
  NumberStyle: "numberStyle",
  NumberType: "numberType",
  Total: "total",
  PrefixNumber: "prefixNumber",
  SuffixNumber: "suffixNumber",
  NumberPosition: "numberPosition",
  Number: "number",
  TextBefore: "textBefore",
  TextAfter: "textAfter",
  Delay: "delay",
  ForegroundColor: "foregroundColor",
  StrokeSize: "strokeSize",
  MainColor: "mainColor",
  StickyTop: "stickyTop",
  Stores: "stores",
  CityId: "cityId",
  DistrictId: "districtId",
  WorkHours: "workHours",
  Map: "map",
  ShowThumbnail: "showThumbnail",
  PaginationColor: "paginationColor",
  PaginationDeactiveColor: "paginationDeactiveColor",
  NavigationColor: "navigationColor",
  Autoplay: "autoplay",
  ShowController: "showController",
  Loop: "loop",
  Crop: "crop",
  AllowSetupThumbnail: "allowSetupThumbnail",
  Title: "title",
  ItemKey: "itemKey",
  TitleStyle: "titleStyle",
  ContentStyle: "contentStyle",
  ContentAlign: "contentAlign",
  HeaderStyle: "headerStyle",
  ColumnGap: "columnGap",
  RowGap: "rowGap",
  AlignItems: "alignItems",
  IconStyle: "iconStyle",
  FormStyle: "formStyle",
  FlexDirection: "flexDirection",
  InputStyle: "inputStyle",
  OptionStyle: "optionStyle",
  ErrorStyle: "errorStyle",
  HiddenFields: "hiddenFields",
  Submits: "submits",
  Version: "version",
  Captcha: "captcha",
  SiteKey: "siteKey",
  RequestKey: "requestKey",
  Action: "action",
  ConfigStorage: "configStorage",
  FormConfigId: "formConfigId",
  Inputs: "inputs",
  Multiple: "multiple",
  Options: "options",
  ImageConfig: "imageConfig",
  Placeholder: "placeholder",
  Label: "label",
  Required: "required",
  Radius: "radius",
  Value: "value",
  Image: "image",
  Percent: "percent",
  Response: "response",
  IsMainPopup: "isMainPopup",
  ShowCloseIcon: "showCloseIcon",
  CloseIcon: "closeIcon",
  ZIndex: "zIndex",
  MaxWidth: "maxWidth",
  PositionType: "positionType",
  Toggle: "toggle",
  ItemData: "itemData",
  Id: "id",
  CanvasConfig: "canvasConfig",
  BackgroundColors: "backgroundColors",
  Rotate: "rotate",
  SpinTrigger: "spinTrigger",
  Roulette: "roulette",
  PrizeAngle: "prizeAngle",
  PerpendicularPrize: "perpendicularPrize",
  ShowTurn: "showTurn",
  Prize: "prize",
  OutOfTurn: "outOfTurn",
  OutOfPrize: "outOfPrize",
  Custom: "custom",
  Key: "key",
  Popup: "popup",
  TriggerUpdate: "triggerUpdate",
  Tabs: "tabs",
  ActionButtons: "actionButtons",
  ActionButton: "actionButton",
  OrderButton: "orderButton",
  ButtonLink: "buttonLink",
  MenuItemId: "menuItemId",
  Slug: "slug",
  Limit: "limit",
  ItemSettings: "itemSettings",
  PriceColor: "priceColor",
  ResponsiveSettings: "responsiveSettings",
  ItemsPerRow: "itemsPerRow",
  CarouselSettings: "carouselSettings",
  GridRows: "gridRows",
  ArrowStyle: "arrowStyle",
  PaginationStyle: "paginationStyle",
  Interval: "interval",
  GridCols: "gridCols",
  PaginationPosition: "paginationPosition",
  Rewind: "rewind",
  PauseOnHover: "pauseOnHover",
  Button: "button",
  Items: "items",
  Skin: "skin",
  Content: "content",
  ImageStyle: "imageStyle",
  NameStyle: "nameStyle",
  BubbleStyle: "bubbleStyle",
  PageWidthType: "pageWidthType",
  PageMaxWidth: "pageMaxWidth",
  BlockType: "blockType",
  MenuConfig: "menuConfig",
  LogoConfig: "logoConfig",
  LanguageConfig: "languageConfig",
  SpacingWithMenu: "spacingWithMenu",
  MenuItemConfig: "menuItemConfig",
  Top: "top",
  Bottom: "bottom",
  Show: "show",
  ShowFlag: "showFlag",
  FlagStyle: "flagStyle",
  FlagSize: "flagSize",
  Enable: "enable",
  Horizontal: "horizontal",
  Vertical: "vertical",
  CarouselOptions: "carouselOptions",
  GridOptions: "gridOptions",
  Speed: "speed",
  Columns: "columns",
  SlidesPerView: "slidesPerView",
  PriceStyle: "priceStyle",
  VirtualRating: "virtualRating",
  VirtualSold: "virtualSold",
  QRCodeStyle: "qrCodeStyle",
  Bordered: "bordered",
  CurrencyUnit: "currencyUnit",
  CurrencyAlignment: "currencyAlignment",
  HeaderBackgroundColor: "headerBackgroundColor",
  CategoryIds: "categoryIds",
  GridConfig: "gridConfig",
  ListConfig: "listConfig",
  ShowLoadMore: "showLoadMore",
  LoadMoreType: "loadMoreType",
  Row: "row",
  Column: "column",
  IsOpenNewTab: "isOpenNewTab",
  TitleFontSize: "titleFontSize",
  ShowImage: "showImage",
  ShowTitle: "showTitle",
  ShowDesc: "showDesc",
  ShowPublishDate: "showPublishDate",
  ShowCategories: "showCategories",
  ShowAuthor: "showAuthor",
  ShowButton: "showButton",
  DisplayConfig: "displayConfig",
  CategoryType: "categoryType",
  CategoryColor: "categoryColor",
  TitleLevel: "titleLevel",
  CssSelector: "cssSelector",
  CustomCss: "customCss"
});

export { PROP_KEY };
