// @tempi/core-elements/dist/components/tab/selector/TabEditor/TabStyleSetting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../../i18n.js');
var styled = require('@emotion/styled');
var coreEditor = require('@tempi/core-editor');
var antd = require('antd');
var TabItemHead = require('../../ui/TabItemHead/TabItemHead.js');
var propKey = require('../../../../constants/propKey.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

const TabStyleSetting = () => {
  const [type = "button", setType] = coreEditor.useProp("type");
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
        label: i18n.t("Ki\u1EC3u hi\u1EC3n th\u1ECB"),
        layout: "vertical",
        children: /* @__PURE__ */ jsxRuntime.jsxs(antd.Radio.Group, {
          defaultValue: type,
          onChange: (e) => setType(e.target.value),
          style: { width: "100%" },
          children: [
            /* @__PURE__ */ jsxRuntime.jsx(RadioButtonWrapper, {
              children: /* @__PURE__ */ jsxRuntime.jsx(StyledRadioButton, {
                value: "button",
                children: /* @__PURE__ */ jsxRuntime.jsx(TabHeadPreview, {
                  Head: TabItemHead.TabHeadButton
                })
              })
            }),
            /* @__PURE__ */ jsxRuntime.jsx(RadioButtonWrapper, {
              children: /* @__PURE__ */ jsxRuntime.jsx(StyledRadioButton, {
                value: "rectangle",
                children: /* @__PURE__ */ jsxRuntime.jsx(TabHeadPreview, {
                  Head: TabItemHead.TabHeadRectangle
                })
              })
            }),
            /* @__PURE__ */ jsxRuntime.jsx(RadioButtonWrapper, {
              children: /* @__PURE__ */ jsxRuntime.jsx(StyledRadioButton, {
                value: "line",
                children: /* @__PURE__ */ jsxRuntime.jsx(TabHeadPreview, {
                  Head: TabItemHead.TabHeadLine
                })
              })
            }),
            /* @__PURE__ */ jsxRuntime.jsx(RadioButtonWrapper, {
              children: /* @__PURE__ */ jsxRuntime.jsx(StyledRadioButton, {
                value: "card",
                children: /* @__PURE__ */ jsxRuntime.jsx(TabHeadPreview, {
                  Head: TabItemHead.TabHeadCard
                })
              })
            })
          ]
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.AlignmentConfig, {
        formItemProps: {
          label: i18n.t("C\u0103n ch\u1EC9nh")
        },
        buttonGroupProps: {
          listAlignment: ["left", "center", "right", "justify"]
        },
        propKey: "alignment"
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
        formItemProps: {
          label: i18n.t("M\xE0u ch\u1EEF")
        },
        propKey: "textColor"
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
        formItemProps: {
          label: i18n.t("M\xE0u tab")
        },
        propKey: "activeColor",
        gradientPicker: true
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.BackgroundConfig, {
        propKey: propKey.PROP_KEY.CommonStyle
      })
    ]
  });
};
const TabHeadPreview = ({ Head }) => {
  const [activeColor = "#258DFF"] = coreEditor.useProp("activeColor");
  const [textColor] = coreEditor.useProp("textColor");
  return /* @__PURE__ */ jsxRuntime.jsxs("div", {
    style: { display: "flex", width: "100%", justifyContent: "space-around" },
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(Head, {
        className: "active",
        textColor,
        activeColor,
        children: "Tab 1"
      }),
      /* @__PURE__ */ jsxRuntime.jsx(Head, {
        textColor: textColor || "black",
        activeColor,
        children: "Tab 2"
      }),
      /* @__PURE__ */ jsxRuntime.jsx(Head, {
        textColor: textColor || "black",
        activeColor,
        children: "Tab 3"
      })
    ]
  });
};
const StyledRadioButton = styled__default["default"](antd.Radio.Button)`
  width: 100%;
  padding: 6px 0;
  height: unset;
`;
const RadioButtonWrapper = styled__default["default"].div`
  margin-bottom: 8px;
`;

exports.TabStyleSetting = TabStyleSetting;
