// @tempi/core-elements/dist/components/tab/selector/TabEditor/TabStyleSetting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../../../i18n.es.js';
import styled from '@emotion/styled';
import { useProp, FormItem, AlignmentConfig, ColorPickerConfig, BackgroundConfig } from '@tempi/core-editor';
import { Radio } from 'antd';
import { TabHeadButton, TabHeadRectangle, TabHeadLine, TabHeadCard } from '../../ui/TabItemHead/TabItemHead.es.js';
import { PROP_KEY } from '../../../../constants/propKey.es.js';

const TabStyleSetting = () => {
  const [type = "button", setType] = useProp("type");
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(FormItem, {
        label: t("Ki\u1EC3u hi\u1EC3n th\u1ECB"),
        layout: "vertical",
        children: /* @__PURE__ */ jsxs(Radio.Group, {
          defaultValue: type,
          onChange: (e) => setType(e.target.value),
          style: { width: "100%" },
          children: [
            /* @__PURE__ */ jsx(RadioButtonWrapper, {
              children: /* @__PURE__ */ jsx(StyledRadioButton, {
                value: "button",
                children: /* @__PURE__ */ jsx(TabHeadPreview, {
                  Head: TabHeadButton
                })
              })
            }),
            /* @__PURE__ */ jsx(RadioButtonWrapper, {
              children: /* @__PURE__ */ jsx(StyledRadioButton, {
                value: "rectangle",
                children: /* @__PURE__ */ jsx(TabHeadPreview, {
                  Head: TabHeadRectangle
                })
              })
            }),
            /* @__PURE__ */ jsx(RadioButtonWrapper, {
              children: /* @__PURE__ */ jsx(StyledRadioButton, {
                value: "line",
                children: /* @__PURE__ */ jsx(TabHeadPreview, {
                  Head: TabHeadLine
                })
              })
            }),
            /* @__PURE__ */ jsx(RadioButtonWrapper, {
              children: /* @__PURE__ */ jsx(StyledRadioButton, {
                value: "card",
                children: /* @__PURE__ */ jsx(TabHeadPreview, {
                  Head: TabHeadCard
                })
              })
            })
          ]
        })
      }),
      /* @__PURE__ */ jsx(AlignmentConfig, {
        formItemProps: {
          label: t("C\u0103n ch\u1EC9nh")
        },
        buttonGroupProps: {
          listAlignment: ["left", "center", "right", "justify"]
        },
        propKey: "alignment"
      }),
      /* @__PURE__ */ jsx(ColorPickerConfig, {
        formItemProps: {
          label: t("M\xE0u ch\u1EEF")
        },
        propKey: "textColor"
      }),
      /* @__PURE__ */ jsx(ColorPickerConfig, {
        formItemProps: {
          label: t("M\xE0u tab")
        },
        propKey: "activeColor",
        gradientPicker: true
      }),
      /* @__PURE__ */ jsx(BackgroundConfig, {
        propKey: PROP_KEY.CommonStyle
      })
    ]
  });
};
const TabHeadPreview = ({ Head }) => {
  const [activeColor = "#258DFF"] = useProp("activeColor");
  const [textColor] = useProp("textColor");
  return /* @__PURE__ */ jsxs("div", {
    style: { display: "flex", width: "100%", justifyContent: "space-around" },
    children: [
      /* @__PURE__ */ jsx(Head, {
        className: "active",
        textColor,
        activeColor,
        children: "Tab 1"
      }),
      /* @__PURE__ */ jsx(Head, {
        textColor: textColor || "black",
        activeColor,
        children: "Tab 2"
      }),
      /* @__PURE__ */ jsx(Head, {
        textColor: textColor || "black",
        activeColor,
        children: "Tab 3"
      })
    ]
  });
};
const StyledRadioButton = styled(Radio.Button)`
  width: 100%;
  padding: 6px 0;
  height: unset;
`;
const RadioButtonWrapper = styled.div`
  margin-bottom: 8px;
`;

export { TabStyleSetting };
