// @tempi/core-elements/dist/components/tab/selector/TabEditor/TabSetting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../../i18n.js');
var styled = require('@emotion/styled');
var coreEditor = require('@tempi/core-editor');
var TabItemEditor = require('../TabItemEditor/TabItemEditor.js');
var propKey = require('../../../../constants/propKey.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

const TabSetting = () => {
  const {
    actions,
    query: { parseReactElement, node },
    duplicateNode
  } = coreEditor.useEditor();
  const { id, childNodes } = coreEditor.useNode((currentNode) => ({
    childNodes: currentNode.data.nodes.map(
      (childNodeId) => node(childNodeId).get()
    )
  }));
  const tabs = childNodes.map((node2) => {
    const nodeProps = node2.data.props;
    return {
      id: node2.id,
      title: nodeProps.title
    };
  });
  const deleteTab = (index) => {
    actions.delete(tabs.length === 1 ? id : tabs[index].id);
  };
  const duplicateTab = (index) => {
    duplicateNode(node(tabs[index].id).get(), id);
  };
  const addNewTab = () => {
    const newNodeTree = parseReactElement(
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.Element, {
        canvas: true,
        is: TabItemEditor.TabItemEditor,
        title: "New Tab"
      })
    ).toNodeTree();
    actions.history.throttle().addNodeTree(newNodeTree, id);
  };
  return /* @__PURE__ */ jsxRuntime.jsx(jsxRuntime.Fragment, {
    children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
      label: i18n.t("Danh s\xE1ch Tab"),
      layout: "vertical",
      children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ItemListConfig, {
        activeItemConfig: TabTitleConfig,
        propKey: propKey.PROP_KEY.Tabs,
        customRemove: deleteTab,
        customDuplicate: duplicateTab,
        customInsert: addNewTab,
        allowDrag: node(id).isCanvas(),
        addItemText: i18n.t("Th\xEAm tab m\u1EDBi")
      })
    })
  });
};
const TabTitleConfig = ({ index }) => {
  const {
    actions: { setProp }
  } = coreEditor.useEditor();
  const [tab] = coreEditor.useProp(propKey.PROP_KEY.joinKey(propKey.PROP_KEY.Tabs, index.toString()));
  const onChangeTitle = (title) => {
    setProp(tab.id, (props) => props.title = title);
  };
  return /* @__PURE__ */ jsxRuntime.jsx(StyledTagContent, {
    children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputConfig, {
      formItemProps: {
        label: i18n.t("Ti\xEAu \u0111\u1EC1"),
        layout: "vertical"
      },
      inputProps: {
        placeholder: i18n.t("Nh\u1EADp ti\xEAu \u0111\u1EC1"),
        style: { width: "100%" },
        value: tab.title,
        onChange: (e) => onChangeTitle(e.target.value)
      }
    })
  });
};
const StyledTagContent = styled__default["default"].div`
  padding: 0.5rem;
`;

exports.TabSetting = TabSetting;
