// @tempi/core-elements/dist/components/article-header/selector.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var coreEditor = require('@tempi/core-editor');
var constants = require('./constants.js');
var quickSetting = require('./quick-setting.js');
var setting = require('./setting.js');
var styleSetting = require('./style-setting.js');
var ui = require('./ui.js');
var RenderFloatingQuickSetting = require('../../core/RenderFloatingQuickSetting/RenderFloatingQuickSetting.js');
var AddBlockButton = require('../../core/Indicator/AddBlockButton.js');

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
const ArticleHeaderEditor = (props) => {
  const {
    connectors: { connect }
  } = coreEditor.useNode();
  return /* @__PURE__ */ jsxRuntime.jsx("div", {
    ref: connect,
    children: /* @__PURE__ */ jsxRuntime.jsxs(RenderFloatingQuickSetting.RenderFloatingQuickSetting, {
      quickSetting: /* @__PURE__ */ jsxRuntime.jsx(quickSetting.ArticleHeaderQuickSetting, {}),
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(ui.ArticleHeader, __spreadValues({}, props)),
        " ",
        /* @__PURE__ */ jsxRuntime.jsx(AddBlockButton.AddBlockButton, {})
      ]
    })
  });
};
ArticleHeaderEditor.craft = coreEditor.crafting({
  displayName: ui.ArticleHeader.displayName,
  tag: ui.ArticleHeader.displayName.toLowerCase(),
  rules: {
    canDrag: () => false
  },
  props: constants.DEFAULT_ARTICLE_HEADER,
  customAttributes: setting.ArticleHeaderSetting,
  commonAttributes: styleSetting.ArticleHeaderStyleSetting
});

exports.ArticleHeaderEditor = ArticleHeaderEditor;
