// @tempi/core-elements/dist/components/article-header/style-setting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { ColorPickerConfig, RangeSliderConfig } from '@tempi/core-editor';
import { PROP_KEY } from '../../constants/propKey.es.js';

const ArticleHeaderStyleSetting = () => {
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(ColorPickerConfig, {
        formItemProps: {
          label: t("M\xE0u n\u1EC1n")
        },
        propKey: PROP_KEY.joinKey(
          PROP_KEY.CommonStyle,
          PROP_KEY.BackgroundColor
        )
      }),
      /* @__PURE__ */ jsx(ColorPickerConfig, {
        formItemProps: {
          label: t("M\xE0u ti\xEAu \u0111\u1EC1")
        },
        propKey: PROP_KEY.joinKey(PROP_KEY.CommonStyle, PROP_KEY.Color)
      }),
      /* @__PURE__ */ jsx(RangeSliderConfig, {
        propKey: PROP_KEY.joinKey(PROP_KEY.CommonStyle, PROP_KEY.FontSize),
        formItemProps: {
          label: t("C\u1EE1 ch\u1EEF ti\xEAu \u0111\u1EC1"),
          layout: "vertical"
        },
        unitProps: {
          defaultValue: "px"
        },
        sliderProps: {
          min: 14,
          max: 60
        }
      })
    ]
  });
};

export { ArticleHeaderStyleSetting };
