// @tempi/core-elements/dist/components/article-header/style-setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var propKey = require('../../constants/propKey.js');

const ArticleHeaderStyleSetting = () => {
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
        formItemProps: {
          label: i18n.t("M\xE0u n\u1EC1n")
        },
        propKey: propKey.PROP_KEY.joinKey(
          propKey.PROP_KEY.CommonStyle,
          propKey.PROP_KEY.BackgroundColor
        )
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
        formItemProps: {
          label: i18n.t("M\xE0u ti\xEAu \u0111\u1EC1")
        },
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.CommonStyle, propKey.PROP_KEY.Color)
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.RangeSliderConfig, {
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.CommonStyle, propKey.PROP_KEY.FontSize),
        formItemProps: {
          label: i18n.t("C\u1EE1 ch\u1EEF ti\xEAu \u0111\u1EC1"),
          layout: "vertical"
        },
        unitProps: {
          defaultValue: "px"
        },
        sliderProps: {
          min: 14,
          max: 60
        }
      })
    ]
  });
};

exports.ArticleHeaderStyleSetting = ArticleHeaderStyleSetting;
