// @tempi/core-editor/dist/configs/ButtonGroup/ButtonGroupItem.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var styled = require('@emotion/styled');
var antd = require('antd');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

const StyledButton = styled__default["default"](antd.Button)`
  border: none;
  height: 32px;
  min-width: 30px;
  display: flex;
  align-items: center;
  justify-content: center;
  path,
  rect {
    fill: ${(props) => props.theme.colorText};
  }
  border-radius: 0;
  cursor: pointer;
  padding: 6px 8px;
  &.is-active {
    color: ${(props) => props.theme.colorPrimary} !important;
    path,
    rect {
      fill: ${(props) => props.theme.colorPrimary} !important;
    }
  }
`;
const ButtonGroupItem = ({
  isActive = false,
  children,
  onClick,
  className
}) => {
  return /* @__PURE__ */ jsxRuntime.jsx(StyledButton, {
    type: "text",
    onClick,
    className: `${isActive ? "is-active" : ""} ${className}`,
    children
  });
};

exports.ButtonGroupItem = ButtonGroupItem;
