// @tempi/core-elements/dist/components/accordion/selector/configs/AccordionSetting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../../i18n.js');
var antd = require('antd');
var CommonSetting = require('./CommonSetting.js');
var ListAccordionItemSetting = require('./ListAccordionItemSetting.js');

const { Panel } = antd.Collapse;
const AccordionSetting = () => {
  return /* @__PURE__ */ jsxRuntime.jsx(jsxRuntime.Fragment, {
    children: /* @__PURE__ */ jsxRuntime.jsxs(antd.Collapse, {
      defaultActiveKey: ["AccordionItems", "Display"],
      expandIconPosition: "end",
      bordered: false,
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(Panel, {
          header: i18n.t("Danh s\xE1ch ph\u1EA7n t\u1EED"),
          children: /* @__PURE__ */ jsxRuntime.jsx(ListAccordionItemSetting.ListAccordionSetting, {})
        }, "AccordionItems"),
        /* @__PURE__ */ jsxRuntime.jsx(Panel, {
          header: i18n.t("C\u1EA5u h\xECnh chung"),
          children: /* @__PURE__ */ jsxRuntime.jsx(CommonSetting.CommonSetting, {})
        }, "Display")
      ]
    })
  });
};

exports.AccordionSetting = AccordionSetting;
