// @tempi/core-elements/dist/components/accordion/selector/configs/ColorSetting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../../i18n.js');
var icons = require('@ant-design/icons');
var coreEditor = require('@tempi/core-editor');
var antd = require('antd');
var QuickSettingTabWrapper = require('../../../../core/QuickSettingTabWrapper/QuickSettingTabWrapper.js');
var propKey = require('../../../../constants/propKey.js');
var ShadowConfig = require('../../../../configs/ShadowConfig/ShadowConfig.js');

const ColorSetting = () => {
  const content = () => /* @__PURE__ */ jsxRuntime.jsxs(QuickSettingTabWrapper.QuickSettingTabWrapper, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(antd.Tabs.TabPane, {
        tab: i18n.t("Ch\u1EEF"),
        children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
          formItemProps: {
            label: i18n.t("M\xE0u ch\u1EEF")
          },
          propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.TitleStyle, propKey.PROP_KEY.Color)
        })
      }, "1"),
      /* @__PURE__ */ jsxRuntime.jsx(antd.Tabs.TabPane, {
        tab: "Icon",
        children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
          formItemProps: {
            label: i18n.t("M\xE0u icon")
          },
          propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.Icon, propKey.PROP_KEY.Color)
        })
      }, "2"),
      /* @__PURE__ */ jsxRuntime.jsx(antd.Tabs.TabPane, {
        tab: i18n.t("N\u1EC1n"),
        children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
          propKey: propKey.PROP_KEY.joinKey(
            propKey.PROP_KEY.CommonStyle,
            propKey.PROP_KEY.BackgroundColor
          ),
          formItemProps: {
            label: i18n.t("M\xE0u n\u1EC1n")
          },
          gradientPicker: true
        })
      }, "3"),
      /* @__PURE__ */ jsxRuntime.jsx(antd.Tabs.TabPane, {
        tab: i18n.t("Vi\u1EC1n"),
        children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.BorderConfig, {
          propKey: propKey.PROP_KEY.CommonStyle
        })
      }, "4"),
      /* @__PURE__ */ jsxRuntime.jsx(antd.Tabs.TabPane, {
        tab: i18n.t("B\xF3ng"),
        children: /* @__PURE__ */ jsxRuntime.jsx(ShadowConfig.ShadowConfig, {
          propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.CommonStyle, propKey.PROP_KEY.Shadow)
        })
      }, "5")
    ]
  });
  return /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SubQuickSetting, {
    content,
    children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.QuickSettingButton, {
      tooltip: i18n.t("M\xE0u kh\u1ED1i"),
      children: /* @__PURE__ */ jsxRuntime.jsx(icons.BorderOutlined, {
        style: {
          fontSize: "16px"
        }
      })
    })
  });
};

exports.ColorSetting = ColorSetting;
