// @tempi/core-elements/dist/components/accordion/selector/configs/ColorSetting.es.js
import { jsx, jsxs } from 'react/jsx-runtime';
import { t } from '../../../../i18n.es.js';
import { BorderOutlined } from '@ant-design/icons';
import { SubQuickSetting, QuickSettingButton, ColorPickerConfig, BorderConfig } from '@tempi/core-editor';
import { Tabs } from 'antd';
import { QuickSettingTabWrapper } from '../../../../core/QuickSettingTabWrapper/QuickSettingTabWrapper.es.js';
import { PROP_KEY } from '../../../../constants/propKey.es.js';
import { ShadowConfig } from '../../../../configs/ShadowConfig/ShadowConfig.es.js';

const ColorSetting = () => {
  const content = () => /* @__PURE__ */ jsxs(QuickSettingTabWrapper, {
    children: [
      /* @__PURE__ */ jsx(Tabs.TabPane, {
        tab: t("Ch\u1EEF"),
        children: /* @__PURE__ */ jsx(ColorPickerConfig, {
          formItemProps: {
            label: t("M\xE0u ch\u1EEF")
          },
          propKey: PROP_KEY.joinKey(PROP_KEY.TitleStyle, PROP_KEY.Color)
        })
      }, "1"),
      /* @__PURE__ */ jsx(Tabs.TabPane, {
        tab: "Icon",
        children: /* @__PURE__ */ jsx(ColorPickerConfig, {
          formItemProps: {
            label: t("M\xE0u icon")
          },
          propKey: PROP_KEY.joinKey(PROP_KEY.Icon, PROP_KEY.Color)
        })
      }, "2"),
      /* @__PURE__ */ jsx(Tabs.TabPane, {
        tab: t("N\u1EC1n"),
        children: /* @__PURE__ */ jsx(ColorPickerConfig, {
          propKey: PROP_KEY.joinKey(
            PROP_KEY.CommonStyle,
            PROP_KEY.BackgroundColor
          ),
          formItemProps: {
            label: t("M\xE0u n\u1EC1n")
          },
          gradientPicker: true
        })
      }, "3"),
      /* @__PURE__ */ jsx(Tabs.TabPane, {
        tab: t("Vi\u1EC1n"),
        children: /* @__PURE__ */ jsx(BorderConfig, {
          propKey: PROP_KEY.CommonStyle
        })
      }, "4"),
      /* @__PURE__ */ jsx(Tabs.TabPane, {
        tab: t("B\xF3ng"),
        children: /* @__PURE__ */ jsx(ShadowConfig, {
          propKey: PROP_KEY.joinKey(PROP_KEY.CommonStyle, PROP_KEY.Shadow)
        })
      }, "5")
    ]
  });
  return /* @__PURE__ */ jsx(SubQuickSetting, {
    content,
    children: /* @__PURE__ */ jsx(QuickSettingButton, {
      tooltip: t("M\xE0u kh\u1ED1i"),
      children: /* @__PURE__ */ jsx(BorderOutlined, {
        style: {
          fontSize: "16px"
        }
      })
    })
  });
};

export { ColorSetting };
