// @tempi/core-elements/dist/components/accordion/selector/configs/CommonSetting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../../../i18n.es.js';
import { useProp, SizeAttribute, InputNumberConfig, AlignmentConfig } from '@tempi/core-editor';
import { UnitEnum, Alignment } from '@tempi/core-renderer';
import { PROP_KEY } from '../../../../constants/propKey.es.js';

const CommonSetting = () => {
  const iconPostionKey = PROP_KEY.joinKey(PROP_KEY.Icon, PROP_KEY.Align);
  const [iconPosition] = useProp(iconPostionKey);
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(SizeAttribute, {
        configs: { allowConfigHeight: false }
      }),
      /* @__PURE__ */ jsx(InputNumberConfig, {
        propKey: PROP_KEY.joinKey(PROP_KEY.Duration),
        formItemProps: {
          label: t("Kho\u1EA3ng th\u1EDDi gian (s)")
        },
        inputNumberProps: {
          step: 0.1,
          placeholder: "0"
        }
      }),
      /* @__PURE__ */ jsx(InputNumberConfig, {
        propKey: PROP_KEY.joinKey(PROP_KEY.CommonStyle, PROP_KEY.RowGap),
        formItemProps: {
          label: t("Kho\u1EA3ng c\xE1ch gi\u1EEFa c\xE1c m\u1EE5c")
        },
        unitProps: { defaultValue: UnitEnum.pixel },
        inputNumberProps: {
          placeholder: "0"
        }
      }),
      /* @__PURE__ */ jsx(AlignmentConfig, {
        formItemProps: {
          label: t("V\u1ECB tr\xED bi\u1EC3u t\u01B0\u1EE3ng")
        },
        propKey: iconPostionKey,
        buttonGroupProps: {
          listAlignment: ["left", "right"]
        }
      }),
      iconPosition === Alignment.left && /* @__PURE__ */ jsx(InputNumberConfig, {
        propKey: PROP_KEY.joinKey(PROP_KEY.HeaderStyle, PROP_KEY.ColumnGap),
        formItemProps: {
          label: t("Kho\u1EA3ng c\xE1ch gi\u1EEFa ti\xEAu \u0111\u1EC1 v\xE0 icon")
        },
        inputNumberProps: {
          placeholder: "0"
        },
        unitProps: { defaultValue: UnitEnum.pixel }
      })
    ]
  });
};

export { CommonSetting };
