// @tempi/core-elements/dist/components/accordion/selector/configs/ListAccordionItemSetting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../../i18n.js');
var core = require('@dnd-kit/core');
var modifiers = require('@dnd-kit/modifiers');
var sortable = require('@dnd-kit/sortable');
var utilities = require('@dnd-kit/utilities');
var styled = require('@emotion/styled');
var antd = require('antd');
var React = require('react');
var icons = require('@ant-design/icons');
var reactIcons = require('@fluentui/react-icons');
var coreEditor = require('@tempi/core-editor');
var constants = require('../../constants.js');
var coreRenderer = require('@tempi/core-renderer');
var utils = require('../../utils.js');
var AccordionContentEditor = require('../AccordionContent/AccordionContentEditor.js');
var AccordionItemEditor = require('../AccordionItem/AccordionItemEditor.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const truncate = (input, length) => {
  if ((input == null ? void 0 : input.length) > length) {
    return input.substring(0, length) + "...";
  }
  return input;
};
const ItemConfig = ({
  index,
  item,
  onDuplicate,
  onRemove,
  activeItem: activeField,
  setActiveItem: setActiveField,
  changeItem
}) => {
  const {
    actions: { setProp }
  } = coreEditor.useEditor();
  const { setNodeRef, attributes, listeners, transform } = sortable.useSortable({
    id: item == null ? void 0 : item.dndId
  });
  const onChangeTitle = (title) => {
    changeItem(__spreadProps(__spreadValues({}, item), { title }), index);
    setProp(item.nodeId, (props) => props.title = coreRenderer.escapeString(title));
  };
  return /* @__PURE__ */ jsxRuntime.jsx(jsxRuntime.Fragment, {
    children: /* @__PURE__ */ jsxRuntime.jsxs(StyledTagField, {
      ref: setNodeRef,
      style: { transform: utilities.CSS.Transform.toString(transform) },
      className: activeField === index ? "active" : null,
      children: [
        /* @__PURE__ */ jsxRuntime.jsxs(StyledTagHeader, {
          onClick: () => {
            if (activeField === index)
              setActiveField(void 0);
            else
              setActiveField(index);
          },
          children: [
            /* @__PURE__ */ jsxRuntime.jsxs("div", {
              className: "flex",
              children: [
                /* @__PURE__ */ jsxRuntime.jsx(StyledAction, __spreadProps(__spreadValues(__spreadValues({}, listeners), attributes), {
                  style: { cursor: "move", marginRight: 8 },
                  children: /* @__PURE__ */ jsxRuntime.jsx(icons.HolderOutlined, {
                    style: {
                      fontSize: 20
                    }
                  })
                })),
                /* @__PURE__ */ jsxRuntime.jsx("div", {
                  children: truncate(item.title, 30) || index + 1
                })
              ]
            }),
            /* @__PURE__ */ jsxRuntime.jsxs("div", {
              className: "flex",
              children: [
                /* @__PURE__ */ jsxRuntime.jsx(StyledAction, {
                  style: { marginRight: 8 },
                  onClick: (e) => {
                    e.stopPropagation();
                    onDuplicate();
                  },
                  children: /* @__PURE__ */ jsxRuntime.jsx(reactIcons.CopyRegular, {
                    fontSize: 20
                  })
                }),
                /* @__PURE__ */ jsxRuntime.jsx(StyledAction, {
                  onClick: (e) => {
                    e.stopPropagation();
                    onRemove();
                  },
                  children: /* @__PURE__ */ jsxRuntime.jsx(reactIcons.DeleteRegular, {
                    fontSize: 20
                  })
                })
              ]
            })
          ]
        }),
        activeField === index && /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
          children: [
            /* @__PURE__ */ jsxRuntime.jsx(antd.Divider, {
              className: "my-half mb-0"
            }),
            /* @__PURE__ */ jsxRuntime.jsx(StyledTagContent, {
              children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputConfig, {
                formItemProps: {
                  label: i18n.t("Ti\xEAu \u0111\u1EC1"),
                  layout: "vertical"
                },
                inputProps: {
                  placeholder: i18n.t("Nh\u1EADp ti\xEAu \u0111\u1EC1"),
                  style: { width: "100%" },
                  value: item.title,
                  onChange: (e) => onChangeTitle(e.target.value),
                  maxLength: 255
                }
              })
            })
          ]
        })
      ]
    }, index)
  });
};
const ListAccordionSetting = ({ maxItem }) => {
  var _a, _b;
  const {
    actions,
    query: { parseReactElement, node },
    duplicateNode
  } = coreEditor.useEditor();
  const { id, childNodes } = coreEditor.useNode((currentNode) => ({
    childNodes: currentNode.data.nodes.map(
      (childNodeId) => node(childNodeId).get()
    )
  }));
  const getAccordionFromChildNode = () => {
    return childNodes.map((node2) => {
      const nodeProps = node2.data.props;
      return {
        title: coreRenderer.unescapeString(utils.convertTitleToText(nodeProps.title)),
        dndId: coreRenderer.generateRandomId(),
        nodeId: node2 == null ? void 0 : node2.id
      };
    });
  };
  const accordionItems = getAccordionFromChildNode();
  const [items, setItems] = React.useState(accordionItems);
  const [activeItem, setActiveItem] = React.useState(0);
  const deleteTab = (index) => {
    actions.delete(childNodes.length === 1 ? id : childNodes[index].id);
  };
  const duplicateTab = (index) => {
    duplicateNode(childNodes[index], id);
  };
  const addNewTab = () => {
    const newNodeTree = parseReactElement(
      /* @__PURE__ */ jsxRuntime.jsx(AccordionItemEditor.AccordionItemEditor, __spreadProps(__spreadValues({}, constants.DEFAULT_ACCORDION.accordionItem), {
        children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.Element, {
          is: AccordionContentEditor.AccordionContentEditor,
          canvas: true
        })
      }))
    ).toNodeTree();
    actions.history.throttle().addNodeTree(newNodeTree, id);
  };
  const handleDragEnd = ({ active, over }) => {
    if ((active == null ? void 0 : active.id) !== (over == null ? void 0 : over.id)) {
      const _inputs = [...items];
      const oldIndex = _inputs.findIndex((el) => el.dndId === (active == null ? void 0 : active.id));
      const newIndex = _inputs.findIndex((el) => el.dndId === (over == null ? void 0 : over.id));
      setActiveItem(void 0);
      const newPos = newIndex > oldIndex ? newIndex + 1 : newIndex;
      actions.history.ignore().move(items[oldIndex].nodeId, id, newPos);
    }
  };
  React.useEffect(() => {
    if ((accordionItems == null ? void 0 : accordionItems.length) !== (items == null ? void 0 : items.length)) {
      setItems(getAccordionFromChildNode());
    } else {
      const check = items.some((item, index) => {
        return item.title !== accordionItems[index].title;
      });
      if (check) {
        setItems(getAccordionFromChildNode());
      }
    }
  }, [accordionItems]);
  const updateItem = (item, index) => {
    const cloneInputs = [...items];
    cloneInputs[index] = item;
    setItems(cloneInputs);
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(core.DndContext, {
        modifiers: [
          modifiers.restrictToVerticalAxis,
          modifiers.restrictToWindowEdges,
          modifiers.restrictToFirstScrollableAncestor
        ],
        onDragEnd: handleDragEnd,
        children: items.length === 0 ? /* @__PURE__ */ jsxRuntime.jsx(antd.Empty, {
          description: i18n.t("Danh s\xE1ch r\u1ED7ng")
        }) : /* @__PURE__ */ jsxRuntime.jsx(sortable.SortableContext, {
          items: (_a = items.map) == null ? void 0 : _a.call(items, (el) => el.dndId),
          strategy: sortable.verticalListSortingStrategy,
          disabled: false,
          children: (_b = items.map) == null ? void 0 : _b.call(items, (el, index) => /* @__PURE__ */ jsxRuntime.jsx(ItemConfig, {
            activeItem,
            setActiveItem,
            item: el,
            index,
            onDuplicate: () => duplicateTab(index),
            onRemove: () => deleteTab(index),
            changeItem: updateItem
          }, index))
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsx(antd.Button, {
        style: {
          marginTop: 8,
          marginBottom: 8
        },
        disabled: maxItem && items.length >= maxItem,
        type: "primary",
        block: true,
        icon: /* @__PURE__ */ jsxRuntime.jsx(icons.PlusOutlined, {}),
        onClick: () => addNewTab(),
        children: i18n.t("Th\xEAm ph\u1EA7n t\u1EED") + (maxItem ? ` (${items.length}/${maxItem})` : "")
      })
    ]
  });
};
const StyledTagField = styled__default["default"].div`
  justify-content: space-between;
  padding: 0.5rem;
  border-radius: 4px;
  border: 1px solid var(--border-color);
  margin-bottom: 8px;
  &.active {
    border-color: ${(props) => props.theme.colorPrimary};
  }
`;
const StyledTagHeader = styled__default["default"].div`
  display: flex;
  flex-grow: 1;
  align-items: center;
  justify-content: space-between;
  cursor: pointer;
`;
const StyledAction = styled__default["default"].div`
  background-color: transparent;
  cursor: pointer;
`;
const StyledTagContent = styled__default["default"].div`
  padding: 0.5rem;
`;

exports.ListAccordionSetting = ListAccordionSetting;
