// @tempi/core-editor/dist/configs/ImageConfig/ImageFormItem.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import styled from '@emotion/styled';
import { useState, useEffect } from 'react';
import defaultImage from '../../assets/images/default_image.es.js';
import { beforeUploadImage } from './helpers.es.js';
import { ImageUpload } from './ImageUpload.es.js';
import { Spin, Button } from 'antd';
import { CloseCircleFilled, LoadingOutlined, UploadOutlined, CloudUploadOutlined } from '@ant-design/icons';
import { FormItem } from '../../components/FormItem/FormItem.es.js';
import { InputConfig } from '../InputConfig/InputConfig.es.js';
import { isPrefixUrl } from '../../utils/text.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const ImageFormItem = ({
  onChange,
  defaultValue
}) => {
  const [imageUrl, setImageUrl] = useState(defaultValue);
  useEffect(() => {
    const getBaseImageUrl = (imageUrl2) => {
      if (isPrefixUrl(imageUrl2)) {
        return imageUrl2.slice(4, imageUrl2.lastIndexOf("="));
      }
      return imageUrl2;
    };
    const baseImageUrl = getBaseImageUrl(imageUrl);
    if (imageUrl !== baseImageUrl)
      setImageUrl(baseImageUrl);
  }, []);
  const onChangeImageUrlInput = (e) => {
    const newUrl = e.target.value;
    setImageUrl(newUrl);
  };
  const arr = [void 0, "", defaultImage];
  useEffect(() => {
    onChange(imageUrl);
  }, [imageUrl]);
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(FormItem, {
        label: t("common.image"),
        children: /* @__PURE__ */ jsxs("div", {
          style: { position: "relative" },
          children: [
            /* @__PURE__ */ jsx(StyledImageUploadButton, {
              accept: "image/jpg, image/jpeg, image/png, image/gif, image/webp",
              beforeUpload: beforeUploadImage,
              image: imageUrl,
              setImage: setImageUrl
            }),
            !arr.includes(imageUrl) && /* @__PURE__ */ jsx(CloseCircleFilled, {
              onClick: () => {
                setImageUrl(void 0);
              },
              style: {
                position: "absolute",
                top: -6,
                right: -6,
                fontSize: 16,
                borderRadius: 8
              }
            })
          ]
        })
      }),
      /* @__PURE__ */ jsx(InputConfig, {
        formItemProps: {
          label: t("\u0110\u01B0\u1EDDng d\u1EABn \u1EA3nh")
        },
        inputProps: {
          style: {
            width: 140
          },
          placeholder: t("Nh\u1EADp URL"),
          value: imageUrl === defaultImage ? null : imageUrl,
          onChange: onChangeImageUrlInput
        }
      })
    ]
  });
};
const ImageUploadButton = (_a) => {
  var _b = _a, {
    multiple = false,
    image,
    setImage,
    listType = "picture-card"
  } = _b, rest = __objRest(_b, [
    "multiple",
    "image",
    "setImage",
    "listType"
  ]);
  const [uploading, setUploading] = useState(false);
  const uploadAction = (data) => {
    if (!data)
      return;
    const imageUrl = data.imageUrl;
    if (imageUrl) {
      setImage(imageUrl);
    }
    setUploading(data.uploading);
  };
  const uploadImageListAction = (data) => {
    if (!data)
      return;
    const imageUrl = data.imageUrl;
    if (imageUrl) {
      setImage(imageUrl);
    }
    setUploading(data.uploading);
  };
  const renderContent = () => {
    if (multiple) {
      return /* @__PURE__ */ jsx(Button, {
        block: true,
        type: "primary",
        icon: /* @__PURE__ */ jsx(UploadOutlined, {}),
        loading: uploading,
        children: t("T\u1EA3i \u1EA3nh l\xEAn (T\u1ED1i \u0111a 20MB)")
      });
    }
    if (image) {
      return /* @__PURE__ */ jsx(PreviewImage, {
        src: image,
        alt: "uploaded img"
      });
    }
    return /* @__PURE__ */ jsxs("div", {
      children: [
        /* @__PURE__ */ jsx(CloudUploadOutlined, {
          style: { fontSize: 20 }
        }),
        /* @__PURE__ */ jsx("div", {
          style: { marginTop: 8 },
          children: t("T\u1EA3i \u1EA3nh l\xEAn (T\u1ED1i \u0111a 20MB)")
        })
      ]
    });
  };
  return /* @__PURE__ */ jsx(Spin, {
    indicator: /* @__PURE__ */ jsx(LoadingOutlined, {
      style: { fontSize: 20 }
    }),
    spinning: uploading,
    children: /* @__PURE__ */ jsx(ImageUpload, __spreadProps(__spreadValues({
      multiple,
      showUploadList: false,
      disabled: uploading,
      listType,
      onUpdateImageUrl: multiple ? uploadImageListAction : uploadAction
    }, rest), {
      children: renderContent()
    }))
  });
};
const StyledImageUploadButton = styled(ImageUploadButton)`
  width: 140px;
  height: 140px;
`;
const PreviewImage = styled.img`
  width: 100%;
  height: 100%;
  object-fit: contain;
`;

export { ImageFormItem };
