// @tempi/core-editor/dist/configs/TrackingConfig/TrackingConfig.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import React, { useState, useEffect } from 'react';
import { unescapeString, escapeString } from '@tempi/core-renderer';
import { PROP_KEY } from '../../constants/propKey.es.js';
import { usePropSeparateDevice } from '../../hooks/usePropSeparateDevice.es.js';
import { InputConfig } from '../InputConfig/InputConfig.es.js';
import { SelectConfig } from '../SelectConfig/SelectConfig.es.js';
import { InputNumberConfig } from '../InputNumberConfig/InputNumberConfig.es.js';
import { FormItem } from '../../components/FormItem/FormItem.es.js';

const AceEditor = React.lazy(() => import('react-ace'));
const optionTrackingName = [
  {
    label: "CompleteRegistration",
    value: "CompleteRegistration"
  },
  {
    label: "Purchase",
    value: "Purchase"
  },
  {
    label: "Lead",
    value: "Lead"
  }
];
const TrackingConfig = ({
  propKey = PROP_KEY.Event,
  config = "event",
  separateDevice
}) => {
  const trackingEventNamePropKey = PROP_KEY.joinKey(
    propKey,
    PROP_KEY.TrackingEventName
  );
  const trackingConversionValuePropKey = PROP_KEY.joinKey(
    propKey,
    PROP_KEY.ConversionValue
  );
  const idGoogleAdsPropKey = PROP_KEY.joinKey(
    propKey,
    PROP_KEY.ConversionGGAdsId
  );
  const labelGoogleAdsPropKey = PROP_KEY.joinKey(
    propKey,
    PROP_KEY.ConversionGGAdsLabel
  );
  const scriptPropKey = PROP_KEY.joinKey(propKey, PROP_KEY.ScriptCustom);
  const [script, setScript] = usePropSeparateDevice(
    scriptPropKey,
    separateDevice
  );
  const [aceEditor, setAceEditor] = useState(null);
  useEffect(() => {
    var _a, _b;
    (_b = (_a = aceEditor == null ? void 0 : aceEditor.editor) == null ? void 0 : _a.renderer) == null ? void 0 : _b.attachToShadowRoot();
    return () => {
      var _a2, _b2;
      (_b2 = (_a2 = aceEditor == null ? void 0 : aceEditor.editor) == null ? void 0 : _a2.renderer) == null ? void 0 : _b2.attachToShadowRoot();
    };
  });
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      config === "event" && /* @__PURE__ */ jsx(InputConfig, {
        separateDevice,
        propKey: trackingEventNamePropKey,
        formItemProps: {
          label: t("T\xEAn s\u1EF1 ki\u1EC7n FB/GA"),
          style: {
            display: "flex",
            justifyContent: "space-between",
            alignItems: "center",
            marginBottom: "0.5rem"
          }
        },
        inputProps: {
          placeholder: t("Nh\u1EADp t\xEAn s\u1EF1 ki\u1EC7n"),
          style: { width: "180px" }
        }
      }),
      config === "form" && /* @__PURE__ */ jsxs(Fragment, {
        children: [
          /* @__PURE__ */ jsx(SelectConfig, {
            separateDevice,
            propKey: trackingEventNamePropKey,
            options: optionTrackingName,
            formItemProps: {
              label: t("M\xE3 chuy\u1EC3n \u0111\u1ED5i FB/GA")
            },
            selectProps: {
              placeholder: t("Ch\u1ECDn m\xE3 chuy\u1EC3n \u0111\u1ED5i"),
              allowClear: true
            }
          }),
          /* @__PURE__ */ jsx(InputNumberConfig, {
            separateDevice,
            propKey: trackingConversionValuePropKey,
            formItemProps: {
              label: t("Gi\xE1 tr\u1ECB chuy\u1EC3n \u0111\u1ED5i")
            },
            inputNumberProps: {
              placeholder: "0",
              min: 0,
              style: { width: "180px" }
            }
          })
        ]
      }),
      /* @__PURE__ */ jsx(InputConfig, {
        separateDevice,
        propKey: idGoogleAdsPropKey,
        formItemProps: {
          label: t("ID chuy\u1EC3n \u0111\u1ED5i Google Ads"),
          style: {
            display: "flex",
            justifyContent: "space-between",
            alignItems: "center",
            marginBottom: "0.5rem"
          }
        },
        inputProps: {
          placeholder: t("ID chuy\u1EC3n \u0111\u1ED5i"),
          style: { width: "180px" }
        }
      }),
      /* @__PURE__ */ jsx(InputConfig, {
        separateDevice,
        propKey: labelGoogleAdsPropKey,
        formItemProps: {
          label: t("Nh\xE3n chuy\u1EC3n \u0111\u1ED5i Google Ads"),
          style: {
            display: "flex",
            justifyContent: "space-between",
            alignItems: "center",
            marginBottom: "0.5rem"
          }
        },
        inputProps: {
          placeholder: t("Nh\xE3n chuy\u1EC3n \u0111\u1ED5i"),
          style: { width: "180px" }
        }
      }),
      /* @__PURE__ */ jsx(React.Suspense, {
        fallback: /* @__PURE__ */ jsx("div", {}),
        children: /* @__PURE__ */ jsx(FormItem, {
          label: t("M\xE3 t\xF9y ch\u1EC9nh n\xE2ng cao (kh\xF4ng b\u1EAFt bu\u1ED9c)"),
          children: /* @__PURE__ */ jsx(AceEditor, {
            ref: (ref) => setAceEditor(ref),
            placeholder: "VD: fbq('track', 'Purchase', {currency: 'VND', value: 300000})",
            mode: "javascript",
            fontSize: 14,
            height: "200px",
            width: "100%",
            setOptions: {
              showLineNumbers: true,
              tabSize: 2
            },
            highlightActiveLine: true,
            value: unescapeString(script),
            onChange: (value) => setScript(escapeString(value)),
            showPrintMargin: true,
            showGutter: true
          })
        })
      })
    ]
  });
};

export { TrackingConfig };
