// @tempi/core-elements/dist/components/divider/configs/TextTabSetting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var coreRenderer = require('@tempi/core-renderer');
var antd = require('antd');
var propKey = require('../../../constants/propKey.js');

const TextTabSetting = ({ propKey: propKey$1 }) => {
  const [textSpacing, setTextSpacing] = coreEditor.useProp(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.TextConfig, propKey.PROP_KEY.Spacing)
  );
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.AlignmentConfig, {
        propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.TextConfig, propKey.PROP_KEY.Align),
        formItemProps: {
          label: i18n.t("V\u1ECB tr\xED ch\u1EEF")
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
        label: i18n.t("Kho\u1EA3ng c\xE1ch"),
        children: /* @__PURE__ */ jsxRuntime.jsx(antd.InputNumber, {
          value: textSpacing == null ? void 0 : textSpacing.size,
          onChange: (val) => {
            setTextSpacing({
              unit: coreRenderer.UnitEnum.pixel,
              size: val
            });
          },
          addonAfter: "px",
          min: 0,
          style: { width: 120 }
        })
      })
    ]
  });
};

exports.TextTabSetting = TextTabSetting;
