// @tempi/core-elements/dist/components/divider/configs/TextTabSetting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../../i18n.es.js';
import { useProp, AlignmentConfig, FormItem } from '@tempi/core-editor';
import { UnitEnum } from '@tempi/core-renderer';
import { InputNumber } from 'antd';
import { PROP_KEY } from '../../../constants/propKey.es.js';

const TextTabSetting = ({ propKey }) => {
  const [textSpacing, setTextSpacing] = useProp(
    PROP_KEY.joinKey(propKey, PROP_KEY.TextConfig, PROP_KEY.Spacing)
  );
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(AlignmentConfig, {
        propKey: PROP_KEY.joinKey(propKey, PROP_KEY.TextConfig, PROP_KEY.Align),
        formItemProps: {
          label: t("V\u1ECB tr\xED ch\u1EEF")
        }
      }),
      /* @__PURE__ */ jsx(FormItem, {
        label: t("Kho\u1EA3ng c\xE1ch"),
        children: /* @__PURE__ */ jsx(InputNumber, {
          value: textSpacing == null ? void 0 : textSpacing.size,
          onChange: (val) => {
            setTextSpacing({
              unit: UnitEnum.pixel,
              size: val
            });
          },
          addonAfter: "px",
          min: 0,
          style: { width: 120 }
        })
      })
    ]
  });
};

export { TextTabSetting };
