// @tempi/core-renderer/dist/renderPage/RenderPage.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var react = require('@emotion/react');
var React = require('react');
var index = require('../constants/index.js');
var NodeElements = require('../components/NodeElement/NodeElements.js');
var getComponentProps = require('../utils/getComponentProps.js');
var StoreContext = require('../contexts/StoreContext.js');
var reset = require('../constants/reset.js');
var rcCascader = require('../constants/rc-cascader.js');
var animation = require('../constants/animation.js');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const RenderPage = ({
  content = {},
  configs,
  ssrData = {},
  mappingDynamicSlots,
  rootId = "ROOT",
  addGlobalStyle = true,
  numOfHydratedBlocks,
  addLandingLinks = true,
  componentDict
}) => {
  var _a, _b, _c;
  const [client, setIsClient] = React.useState(false);
  React.useEffect(() => {
    setIsClient(true);
  }, []);
  const RootComp = componentDict[(_c = (_b = (_a = content[rootId]) == null ? void 0 : _a.tag) == null ? void 0 : _b.toString()) == null ? void 0 : _c.toLowerCase()];
  const rootProps = React.useMemo(() => {
    return getComponentProps.getComponentProps(content[rootId], configs, ssrData[rootId]);
  }, [content[rootId]]);
  const childNode = React.useMemo(() => {
    var _a2;
    const childLength = ((_a2 = rootProps.nodes) == null ? void 0 : _a2.length) || 0;
    return !client && numOfHydratedBlocks && childLength > numOfHydratedBlocks ? rootProps.nodes.slice(0, numOfHydratedBlocks) : rootProps.nodes;
  }, [rootProps.nodes, client]);
  return /* @__PURE__ */ jsxRuntime.jsxs(StoreContext.StoreProvider, __spreadProps(__spreadValues({
    ssrData,
    content,
    mappingDynamicSlots,
    componentDict
  }, configs), {
    children: [
      addGlobalStyle && /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(react.Global, {
            styles: reset.reset
          }),
          /* @__PURE__ */ jsxRuntime.jsx(react.Global, {
            styles: rcCascader.rcCascaderCss
          }),
          /* @__PURE__ */ jsxRuntime.jsx(react.Global, {
            styles: animation.animationCss
          })
        ]
      }),
      addLandingLinks && /* @__PURE__ */ jsxRuntime.jsx(index.LandingLinks, {}),
      RootComp && rootProps.isVisible && /* @__PURE__ */ jsxRuntime.jsx(RootComp, __spreadProps(__spreadValues({
        className: rootProps.className,
        ssrData: ssrData[rootId]
      }, rootProps.props), {
        children: childNode.map((nodeId) => /* @__PURE__ */ jsxRuntime.jsx(NodeElements.NodeElement, {
          id: nodeId
        }, nodeId))
      }), rootProps.key)
    ]
  }));
};

exports.RenderPage = RenderPage;
