// @tempi/core-elements/dist/components/testimonial-carousel/selector.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import styled from '@emotion/styled';
import { withSelector, AdvanceAttributes, getEditorName } from '@tempi/core-editor';
import { TestimonialCarouselQuickSetting } from './quick-setting.es.js';
import { TestimonialCarouselSetting } from './setting.es.js';
import { TestimonialCarouselStyleSetting } from './style-setting.es.js';
import { TestimonialCarousel } from './ui.es.js';
import { RowEditor } from '../section/selector.es.js';
import { RootEditor } from '../root/selector.es.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
const TestimonialCarouselEditor = withSelector(
  (props) => {
    return /* @__PURE__ */ jsxs(Fragment, {
      children: [
        /* @__PURE__ */ jsx(EditorOv, {}),
        /* @__PURE__ */ jsx(TestimonialCarousel, __spreadValues({}, props))
      ]
    });
  },
  {
    displayName: "TestimonialCarousel",
    tag: TestimonialCarousel.displayName.toLowerCase(),
    quickSetting: TestimonialCarouselQuickSetting,
    customAttributes: TestimonialCarouselSetting,
    commonAttributes: TestimonialCarouselStyleSetting,
    advanceAttributes: AdvanceAttributes,
    rules: {
      canDrop: (targetNode) => {
        const ignoreTargets = [
          getEditorName({ RowEditor }),
          getEditorName({ RootEditor })
        ];
        return !ignoreTargets.includes(targetNode.data.name);
      }
    }
  }
);
const EditorOv = styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 1;
  cursor: pointer;
`;

export { TestimonialCarouselEditor };
