// @tempi/core-elements/dist/components/testimonial-carousel/setting.es.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { InputNumberConfig, AlignmentConfig, SelectConfig } from '@tempi/core-editor';
import { UnitEnum } from '@tempi/core-renderer';
import { Collapse } from 'antd';
import { TestimonialList } from './configs/TestimonialList.es.js';
import { PROP_KEY } from '../../constants/propKey.es.js';
import { SwiperSetting } from '../../configs/Swiper/SwiperSetting.es.js';

const { Panel } = Collapse;
const TestimonialCarouselSetting = () => {
  const layoutOptions = [
    { label: t("\u1EA2nh b\xEAn tr\xEAn"), value: "imageAbove" },
    { label: t("\u1EA2nh x\u1EBFp ch\u1ED3ng"), value: "imageStacked" },
    { label: t("\u1EA2nh n\u1ED9i d\xF2ng"), value: "imageInline" },
    { label: t("\u1EA2nh b\xEAn tr\xE1i"), value: "imageLeft" },
    { label: t("\u1EA2nh b\xEAn ph\u1EA3i"), value: "imageRight" }
  ];
  const skinOptions = [
    { label: t("M\u1EB7c \u0111\u1ECBnh"), value: "default" },
    { label: "Bubble", value: "bubble" }
  ];
  return /* @__PURE__ */ jsxs(Collapse, {
    defaultActiveKey: ["Items"],
    expandIconPosition: "end",
    bordered: false,
    children: [
      /* @__PURE__ */ jsxs(Panel, {
        header: t("Danh s\xE1ch ch\u1EE9ng th\u1EF1c"),
        children: [
          /* @__PURE__ */ jsx(TestimonialList, {
            propKey: PROP_KEY.Items
          }),
          /* @__PURE__ */ jsx(InputNumberConfig, {
            propKey: PROP_KEY.Width,
            formItemProps: {
              label: t("Chi\u1EC1u r\u1ED9ng")
            },
            inputNumberProps: { min: 1, max: 100 },
            unitProps: { defaultValue: UnitEnum.percent }
          }),
          /* @__PURE__ */ jsx(AlignmentConfig, {
            propKey: PROP_KEY.Alignment,
            buttonGroupProps: {
              listAlignment: ["left", "center", "right"]
            },
            formItemProps: {
              label: t("C\u0103n ch\u1EC9nh")
            }
          }),
          /* @__PURE__ */ jsx(SelectConfig, {
            propKey: PROP_KEY.Layout,
            options: layoutOptions,
            formItemProps: {
              label: t("B\u1ED1 c\u1EE5c")
            }
          }),
          /* @__PURE__ */ jsx(SelectConfig, {
            propKey: PROP_KEY.Skin,
            options: skinOptions,
            formItemProps: {
              label: "Skin"
            }
          })
        ]
      }, "Items"),
      /* @__PURE__ */ jsx(Panel, {
        header: t("C\xE0i \u0111\u1EB7t thanh tr\u01B0\u1EE3t"),
        children: /* @__PURE__ */ jsx(SwiperSetting, {})
      }, "Swiper")
    ]
  });
};

export { TestimonialCarouselSetting };
