// @tempi/core-elements/dist/components/testimonial-carousel/setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var coreRenderer = require('@tempi/core-renderer');
var antd = require('antd');
var TestimonialList = require('./configs/TestimonialList.js');
var propKey = require('../../constants/propKey.js');
var SwiperSetting = require('../../configs/Swiper/SwiperSetting.js');

const { Panel } = antd.Collapse;
const TestimonialCarouselSetting = () => {
  const layoutOptions = [
    { label: i18n.t("\u1EA2nh b\xEAn tr\xEAn"), value: "imageAbove" },
    { label: i18n.t("\u1EA2nh x\u1EBFp ch\u1ED3ng"), value: "imageStacked" },
    { label: i18n.t("\u1EA2nh n\u1ED9i d\xF2ng"), value: "imageInline" },
    { label: i18n.t("\u1EA2nh b\xEAn tr\xE1i"), value: "imageLeft" },
    { label: i18n.t("\u1EA2nh b\xEAn ph\u1EA3i"), value: "imageRight" }
  ];
  const skinOptions = [
    { label: i18n.t("M\u1EB7c \u0111\u1ECBnh"), value: "default" },
    { label: "Bubble", value: "bubble" }
  ];
  return /* @__PURE__ */ jsxRuntime.jsxs(antd.Collapse, {
    defaultActiveKey: ["Items"],
    expandIconPosition: "end",
    bordered: false,
    children: [
      /* @__PURE__ */ jsxRuntime.jsxs(Panel, {
        header: i18n.t("Danh s\xE1ch ch\u1EE9ng th\u1EF1c"),
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(TestimonialList.TestimonialList, {
            propKey: propKey.PROP_KEY.Items
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
            propKey: propKey.PROP_KEY.Width,
            formItemProps: {
              label: i18n.t("Chi\u1EC1u r\u1ED9ng")
            },
            inputNumberProps: { min: 1, max: 100 },
            unitProps: { defaultValue: coreRenderer.UnitEnum.percent }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.AlignmentConfig, {
            propKey: propKey.PROP_KEY.Alignment,
            buttonGroupProps: {
              listAlignment: ["left", "center", "right"]
            },
            formItemProps: {
              label: i18n.t("C\u0103n ch\u1EC9nh")
            }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
            propKey: propKey.PROP_KEY.Layout,
            options: layoutOptions,
            formItemProps: {
              label: i18n.t("B\u1ED1 c\u1EE5c")
            }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SelectConfig, {
            propKey: propKey.PROP_KEY.Skin,
            options: skinOptions,
            formItemProps: {
              label: "Skin"
            }
          })
        ]
      }, "Items"),
      /* @__PURE__ */ jsxRuntime.jsx(Panel, {
        header: i18n.t("C\xE0i \u0111\u1EB7t thanh tr\u01B0\u1EE3t"),
        children: /* @__PURE__ */ jsxRuntime.jsx(SwiperSetting.SwiperSetting, {})
      }, "Swiper")
    ]
  });
};

exports.TestimonialCarouselSetting = TestimonialCarouselSetting;
