// @tempi/core-elements/dist/components/testimonial-carousel/style-setting.es.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { useProp, InputNumberConfig, AlignmentConfig, BorderConfig, ColorPickerConfig, SpacingAttribute, FormItem } from '@tempi/core-editor';
import { Collapse, Switch } from 'antd';
import { useState } from 'react';
import { PROP_KEY } from '../../constants/propKey.es.js';
import { ContentWrapper } from '../../core/ContentWrapper/ContentWrapper.es.js';
import { TextConfig } from '../../configs/TextConfig/TextConfig.es.js';

const { Panel } = Collapse;
const TestimonialCarouselStyleSetting = () => {
  const [skin] = useProp(PROP_KEY.Skin);
  const [selectBorder, setSelectBorder] = useState(false);
  const typographyStyle = (propKey) => {
    return /* @__PURE__ */ jsxs(ContentWrapper, {
      children: [
        /* @__PURE__ */ jsx(ColorPickerConfig, {
          propKey: PROP_KEY.joinKey(propKey, PROP_KEY.Color),
          formItemProps: {
            label: t("M\xE0u")
          }
        }),
        /* @__PURE__ */ jsx(TextConfig, {
          propKey,
          config: { textAlign: true, textShadow: true, fontSize: true }
        })
      ]
    });
  };
  return /* @__PURE__ */ jsxs(Collapse, {
    defaultActiveKey: [],
    expandIconPosition: "end",
    bordered: false,
    children: [
      /* @__PURE__ */ jsxs(Panel, {
        header: t("common.image"),
        children: [
          /* @__PURE__ */ jsx(InputNumberConfig, {
            propKey: PROP_KEY.joinKey(PROP_KEY.ImageStyle, PROP_KEY.Width),
            formItemProps: {
              label: t("Chi\u1EC1u r\u1ED9ng")
            }
          }),
          /* @__PURE__ */ jsx(AlignmentConfig, {
            propKey: PROP_KEY.joinKey(PROP_KEY.ImageStyle, PROP_KEY.Alignment),
            buttonGroupProps: {
              listAlignment: ["left", "center", "right"]
            },
            formItemProps: {
              label: t("C\u0103n ch\u1EC9nh")
            }
          }),
          /* @__PURE__ */ jsx(BorderConfig, {
            propKey: PROP_KEY.ImageStyle
          })
        ]
      }, "Image"),
      /* @__PURE__ */ jsx(Panel, {
        header: t("N\u1ED9i dung"),
        children: typographyStyle(PROP_KEY.ContentStyle)
      }, "Content"),
      /* @__PURE__ */ jsx(Panel, {
        header: t("T\xEAn"),
        children: typographyStyle(PROP_KEY.NameStyle)
      }, "Name"),
      /* @__PURE__ */ jsx(Panel, {
        header: t("Ti\xEAu \u0111\u1EC1"),
        children: typographyStyle(PROP_KEY.TitleStyle)
      }, "Title"),
      skin === "bubble" && /* @__PURE__ */ jsxs(Panel, {
        header: "Bubble",
        children: [
          /* @__PURE__ */ jsx(ColorPickerConfig, {
            propKey: PROP_KEY.joinKey(PROP_KEY.BubbleStyle, PROP_KEY.Color),
            placeholder: t("Ch\u1ECDn m\xE0u"),
            formItemProps: {
              label: t("N\u1EC1n")
            }
          }),
          /* @__PURE__ */ jsx(SpacingAttribute, {
            title: "Padidng",
            propKey: PROP_KEY.joinKey(PROP_KEY.BubbleStyle, PROP_KEY.Padding)
          }),
          /* @__PURE__ */ jsx(FormItem, {
            label: t("Vi\u1EC1n"),
            children: /* @__PURE__ */ jsx(Switch, {
              checked: selectBorder,
              onChange: setSelectBorder
            })
          }),
          selectBorder && /* @__PURE__ */ jsx(BorderConfig, {
            propKey: PROP_KEY.BubbleStyle
          })
        ]
      }, "Bubble")
    ]
  });
};

export { TestimonialCarouselStyleSetting };
