// @tempi/core-elements/dist/components/testimonial-carousel/style-setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var antd = require('antd');
var React = require('react');
var propKey = require('../../constants/propKey.js');
var ContentWrapper = require('../../core/ContentWrapper/ContentWrapper.js');
var TextConfig = require('../../configs/TextConfig/TextConfig.js');

const { Panel } = antd.Collapse;
const TestimonialCarouselStyleSetting = () => {
  const [skin] = coreEditor.useProp(propKey.PROP_KEY.Skin);
  const [selectBorder, setSelectBorder] = React.useState(false);
  const typographyStyle = (propKey$1) => {
    return /* @__PURE__ */ jsxRuntime.jsxs(ContentWrapper.ContentWrapper, {
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
          propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.Color),
          formItemProps: {
            label: i18n.t("M\xE0u")
          }
        }),
        /* @__PURE__ */ jsxRuntime.jsx(TextConfig.TextConfig, {
          propKey: propKey$1,
          config: { textAlign: true, textShadow: true, fontSize: true }
        })
      ]
    });
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(antd.Collapse, {
    defaultActiveKey: [],
    expandIconPosition: "end",
    bordered: false,
    children: [
      /* @__PURE__ */ jsxRuntime.jsxs(Panel, {
        header: i18n.t("common.image"),
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputNumberConfig, {
            propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.ImageStyle, propKey.PROP_KEY.Width),
            formItemProps: {
              label: i18n.t("Chi\u1EC1u r\u1ED9ng")
            }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.AlignmentConfig, {
            propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.ImageStyle, propKey.PROP_KEY.Alignment),
            buttonGroupProps: {
              listAlignment: ["left", "center", "right"]
            },
            formItemProps: {
              label: i18n.t("C\u0103n ch\u1EC9nh")
            }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.BorderConfig, {
            propKey: propKey.PROP_KEY.ImageStyle
          })
        ]
      }, "Image"),
      /* @__PURE__ */ jsxRuntime.jsx(Panel, {
        header: i18n.t("N\u1ED9i dung"),
        children: typographyStyle(propKey.PROP_KEY.ContentStyle)
      }, "Content"),
      /* @__PURE__ */ jsxRuntime.jsx(Panel, {
        header: i18n.t("T\xEAn"),
        children: typographyStyle(propKey.PROP_KEY.NameStyle)
      }, "Name"),
      /* @__PURE__ */ jsxRuntime.jsx(Panel, {
        header: i18n.t("Ti\xEAu \u0111\u1EC1"),
        children: typographyStyle(propKey.PROP_KEY.TitleStyle)
      }, "Title"),
      skin === "bubble" && /* @__PURE__ */ jsxRuntime.jsxs(Panel, {
        header: "Bubble",
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.ColorPickerConfig, {
            propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.BubbleStyle, propKey.PROP_KEY.Color),
            placeholder: i18n.t("Ch\u1ECDn m\xE0u"),
            formItemProps: {
              label: i18n.t("N\u1EC1n")
            }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SpacingAttribute, {
            title: "Padidng",
            propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.BubbleStyle, propKey.PROP_KEY.Padding)
          }),
          /* @__PURE__ */ jsxRuntime.jsx(coreEditor.FormItem, {
            label: i18n.t("Vi\u1EC1n"),
            children: /* @__PURE__ */ jsxRuntime.jsx(antd.Switch, {
              checked: selectBorder,
              onChange: setSelectBorder
            })
          }),
          selectBorder && /* @__PURE__ */ jsxRuntime.jsx(coreEditor.BorderConfig, {
            propKey: propKey.PROP_KEY.BubbleStyle
          })
        ]
      }, "Bubble")
    ]
  });
};

exports.TestimonialCarouselStyleSetting = TestimonialCarouselStyleSetting;
