// @tempi/core-editor/dist/configs/ShadowConfig/ShadowConfig.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { UnitEnum } from '@tempi/core-renderer';
import { useProp } from '../../hooks/useProp.es.js';
import { SwitchConfig } from '../SwitchConfig/SwitchConfig.es.js';
import { RangeSliderConfig } from '../RangeSliderConfig/RangeSliderConfig.es.js';

const ShadowConfig = () => {
  const propKey = "commonStyle.shadow";
  const [onShadow, _] = useProp(`${propKey}.on`);
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(SwitchConfig, {
        formItemProps: {
          label: t("\u0110\u1ED5 b\xF3ng")
        },
        propKey: `${propKey}.on`
      }),
      onShadow && /* @__PURE__ */ jsxs(Fragment, {
        children: [
          /* @__PURE__ */ jsx(RangeSliderConfig, {
            formItemProps: {
              label: t("Tr\u1EE5c X")
            },
            propKey: `${propKey}.x`,
            sliderProps: {
              min: -125,
              max: 125,
              defaultValue: 0
            },
            unitProps: { defaultValue: UnitEnum.pixel }
          }),
          /* @__PURE__ */ jsx(RangeSliderConfig, {
            formItemProps: {
              label: t("Tr\u1EE5c Y")
            },
            propKey: `${propKey}.y`,
            sliderProps: {
              min: -125,
              max: 125,
              defaultValue: 0
            },
            unitProps: { defaultValue: UnitEnum.pixel }
          }),
          /* @__PURE__ */ jsx(RangeSliderConfig, {
            formItemProps: {
              label: t("\u0110\u1ED9 lan")
            },
            propKey: `${propKey}.spread`,
            sliderProps: {
              min: -125,
              max: 125,
              defaultValue: 0
            },
            unitProps: { defaultValue: UnitEnum.pixel }
          }),
          /* @__PURE__ */ jsx(RangeSliderConfig, {
            propKey: `${propKey}.blur`,
            formItemProps: {
              label: t("\u0110\u1ED9 m\u1EDD")
            },
            sliderProps: {
              min: 0,
              max: 120,
              defaultValue: 20
            },
            unitProps: { defaultValue: UnitEnum.pixel }
          }),
          /* @__PURE__ */ jsx(RangeSliderConfig, {
            formItemProps: {
              label: t("\u0110\u1ED9 \u0111\u1EADm")
            },
            propKey: `${propKey}.opacity`,
            sliderProps: {
              min: 0,
              max: 1,
              defaultValue: 0.25,
              step: 0.25
            }
          })
        ]
      })
    ]
  });
};

export { ShadowConfig };
