// @tempi/core-editor/dist/configs/ShadowConfig/ShadowConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var coreRenderer = require('@tempi/core-renderer');
var useProp = require('../../hooks/useProp.js');
var SwitchConfig = require('../SwitchConfig/SwitchConfig.js');
var RangeSliderConfig = require('../RangeSliderConfig/RangeSliderConfig.js');

const ShadowConfig = () => {
  const propKey = "commonStyle.shadow";
  const [onShadow, _] = useProp.useProp(`${propKey}.on`);
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(SwitchConfig.SwitchConfig, {
        formItemProps: {
          label: i18n.t("\u0110\u1ED5 b\xF3ng")
        },
        propKey: `${propKey}.on`
      }),
      onShadow && /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(RangeSliderConfig.RangeSliderConfig, {
            formItemProps: {
              label: i18n.t("Tr\u1EE5c X")
            },
            propKey: `${propKey}.x`,
            sliderProps: {
              min: -125,
              max: 125,
              defaultValue: 0
            },
            unitProps: { defaultValue: coreRenderer.UnitEnum.pixel }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(RangeSliderConfig.RangeSliderConfig, {
            formItemProps: {
              label: i18n.t("Tr\u1EE5c Y")
            },
            propKey: `${propKey}.y`,
            sliderProps: {
              min: -125,
              max: 125,
              defaultValue: 0
            },
            unitProps: { defaultValue: coreRenderer.UnitEnum.pixel }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(RangeSliderConfig.RangeSliderConfig, {
            formItemProps: {
              label: i18n.t("\u0110\u1ED9 lan")
            },
            propKey: `${propKey}.spread`,
            sliderProps: {
              min: -125,
              max: 125,
              defaultValue: 0
            },
            unitProps: { defaultValue: coreRenderer.UnitEnum.pixel }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(RangeSliderConfig.RangeSliderConfig, {
            propKey: `${propKey}.blur`,
            formItemProps: {
              label: i18n.t("\u0110\u1ED9 m\u1EDD")
            },
            sliderProps: {
              min: 0,
              max: 120,
              defaultValue: 20
            },
            unitProps: { defaultValue: coreRenderer.UnitEnum.pixel }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(RangeSliderConfig.RangeSliderConfig, {
            formItemProps: {
              label: i18n.t("\u0110\u1ED9 \u0111\u1EADm")
            },
            propKey: `${propKey}.opacity`,
            sliderProps: {
              min: 0,
              max: 1,
              defaultValue: 0.25,
              step: 0.25
            }
          })
        ]
      })
    ]
  });
};

exports.ShadowConfig = ShadowConfig;
