// @tempi/core-elements/dist/components/section/quick-setting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { Settings16Regular } from '@fluentui/react-icons';
import { useEditor, useNode, useBuilder, QuickSettingButton, DuplicateNode, DeleteNode } from '@tempi/core-editor';
import { DragNode } from '../../quick-configs/DragNode/DragNode.es.js';
import { QuickLayout } from '../../quick-configs/QuickLayout/QuickLayout.es.js';
import { MoveDownBlock } from '../../quick-configs/MoveDownBlock/MoveDownBlock.es.js';
import { MoveUpBlock } from '../../quick-configs/MoveUpBlock/MoveUpBlock.es.js';
import { VisibleOnDevice } from '../../quick-configs/VisibleOnDevice/VisibleOnDevice.es.js';

const SectionQuickSetting = () => {
  const { actions, query } = useEditor();
  const { id } = useNode((node) => ({
    descendants: query.node(node.id).descendants(true),
    node
  }));
  const { setShowSettingDialog, track, isEditingBlock } = useBuilder();
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(DragNode, {}),
      !isEditingBlock && /* @__PURE__ */ jsxs(Fragment, {
        children: [
          /* @__PURE__ */ jsx(QuickLayout, {}),
          /* @__PURE__ */ jsx(MoveDownBlock, {}),
          /* @__PURE__ */ jsx(MoveUpBlock, {})
        ]
      }),
      /* @__PURE__ */ jsx(VisibleOnDevice, {}),
      /* @__PURE__ */ jsx(QuickSettingButton, {
        onClick: () => {
          actions.selectNode(id);
          setShowSettingDialog(true);
          track == null ? void 0 : track({
            regionName: "commonSetting",
            contentName: "advancedSettingBtn"
          });
        },
        children: /* @__PURE__ */ jsx(Settings16Regular, {})
      }),
      !isEditingBlock && /* @__PURE__ */ jsx(DuplicateNode, {
        track: () => {
          track == null ? void 0 : track({
            regionName: "commonSetting",
            contentName: "duplicateBtn"
          });
        }
      }),
      /* @__PURE__ */ jsx(DeleteNode, {
        customDelete: (id2) => {
          actions.delete(id2);
          track == null ? void 0 : track({
            regionName: "commonSetting",
            contentName: "deleteBtn"
          });
        }
      })
    ]
  });
};

export { SectionQuickSetting };
