// @tempi/core-elements/dist/components/section/selector.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var coreEditor = require('@tempi/core-editor');
var coreRenderer = require('@tempi/core-renderer');
var set = require('lodash.set');
var React = require('react');
var index = require('../../types/index.js');
var ui$1 = require('../row/ui.js');
var constant = require('./constant.js');
var quickSetting = require('./quick-setting.js');
var ui = require('./ui.js');
var AddBlockButton = require('../../core/Indicator/AddBlockButton.js');
var quickSetting$1 = require('../row/quick-setting.js');
var propKey = require('../../constants/propKey.js');
var grid = require('../../constants/grid.js');
var RenderFloatingQuickSetting = require('../../core/RenderFloatingQuickSetting/RenderFloatingQuickSetting.js');
var setting = require('../row/setting.js');
var styleSetting = require('../row/style-setting.js');
var selector = require('../col/selector.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var set__default = /*#__PURE__*/_interopDefaultLegacy(set);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const RowEditor = (_a) => {
  var _b = _a, {
    commonStyle,
    children,
    addBlockButton: passingAddBlockButton = /* @__PURE__ */ jsxRuntime.jsx(AddBlockButton.AddBlockButton, {})
  } = _b, rest = __objRest(_b, [
    "commonStyle",
    "children",
    "addBlockButton"
  ]);
  const { isEditingBlock } = coreEditor.useBuilder();
  const addBlockButton = isEditingBlock ? null : passingAddBlockButton;
  const {
    actions,
    query: { node },
    hoveredNodes,
    resizeColumns
  } = coreEditor.useEditor((state) => ({
    hoveredNodes: state.events.hovered
  }));
  const {
    id,
    widthType,
    parent,
    isHover,
    connectors: { connect },
    childNodes,
    allDescendants,
    isActive,
    tag
  } = coreEditor.useNode(({ data, events, id: id2 }) => {
    var _a2, _b2;
    return {
      parent: data == null ? void 0 : data.parent,
      widthType: (_a2 = data == null ? void 0 : data.props) == null ? void 0 : _a2.widthType,
      isHover: events.hovered,
      allDescendants: node(id2).descendants(true),
      childNodes: data == null ? void 0 : data.nodes,
      isActive: events.selected,
      tag: (_b2 = data == null ? void 0 : data.custom) == null ? void 0 : _b2.tag
    };
  });
  const isSection = [coreRenderer.ROOT, coreRenderer.FOOTER].includes(parent);
  const [gutter, setGutter] = coreEditor.useProp(propKey.PROP_KEY.Gutter);
  const handleAutoResizeColumns = () => {
    const totalColWidth = childNodes.reduce((total, chilId) => {
      var _a2, _b2;
      const colWidth = parseFloat(((_b2 = (_a2 = node(chilId).get().data.props) == null ? void 0 : _a2.md) == null ? void 0 : _b2.flex) || 0);
      return total + colWidth;
    }, 0);
    if (totalColWidth > 100 || totalColWidth < 99)
      resizeColumns(id, childNodes.length);
  };
  React.useEffect(() => {
    handleAutoResizeColumns();
  }, [childNodes.length]);
  React.useEffect(() => {
    if (!gutter) {
      setGutter(grid.DEFAULT_GUTTER);
    }
  }, [gutter, setGutter]);
  React.useEffect(() => {
    if (!widthType && [coreRenderer.ROOT, coreRenderer.HEADER, coreRenderer.FOOTER].includes(parent)) {
      actions.history.ignore().setProp(id, (props) => {
        set__default["default"](props, propKey.PROP_KEY.WidthType, index.WidthTypeEnum.full);
        set__default["default"](props, propKey.PROP_KEY.ContentWidthType, index.WidthTypeEnum.fixed);
        set__default["default"](props, propKey.PROP_KEY.ContentMaxWidthKey, grid.DEFAULT_WIDTH);
      });
    }
  }, []);
  React.useEffect(() => {
    const handleUpdateTag = (tag2) => {
      actions.history.ignore().setCustom(id, (custom) => {
        custom.tag = tag2;
      });
    };
    if (isSection && tag !== coreRenderer.COMPONENT_TAG.SECTION) {
      handleUpdateTag(coreRenderer.COMPONENT_TAG.SECTION);
    } else if (!isSection && tag !== coreRenderer.COMPONENT_TAG.ROW) {
      handleUpdateTag(coreRenderer.COMPONENT_TAG.ROW);
    }
  }, [parent]);
  const hoveredChilds = allDescendants.some((el) => hoveredNodes.has(el));
  const showRowQuickSetting = hoveredChilds || isHover || isActive;
  if (isSection)
    return /* @__PURE__ */ jsxRuntime.jsx("div", {
      ref: connect,
      style: {
        width: "100%",
        position: "relative"
      },
      children: /* @__PURE__ */ jsxRuntime.jsxs(RenderFloatingQuickSetting.RenderFloatingQuickSetting, {
        quickSetting: /* @__PURE__ */ jsxRuntime.jsx(quickSetting.SectionQuickSetting, {}),
        children: [
          addBlockButton,
          /* @__PURE__ */ jsxRuntime.jsx(ui.PbSection, __spreadProps(__spreadValues({
            commonStyle,
            gutter
          }, rest), {
            children
          }))
        ]
      })
    });
  return /* @__PURE__ */ jsxRuntime.jsxs(ui$1.PbRow, __spreadProps(__spreadValues({
    ref: connect,
    commonStyle: __spreadProps(__spreadValues({}, commonStyle), { position: "relative" }),
    gutter
  }, rest), {
    children: [
      showRowQuickSetting && /* @__PURE__ */ jsxRuntime.jsx(quickSetting$1.RowQuickSetting, {}),
      children
    ]
  }));
};
RowEditor.craft = coreEditor.crafting({
  displayName: "H\xE0ng",
  tag: "row",
  props: constant.DEFAULT_SECTION,
  advanceAttributes: coreEditor.AdvanceAttributes,
  customAttributes: setting.RowSetting,
  commonAttributes: styleSetting.RowStyle,
  effectAttributes: coreEditor.EffectAttributes,
  rules: {
    canMoveIn: (incomingNodes) => {
      const allowSources = [coreEditor.getEditorName({ ColEditor: selector.ColEditor })];
      return incomingNodes.every(
        (incomingNode) => allowSources.includes(incomingNode.data.name)
      );
    }
  }
});

exports.RowEditor = RowEditor;
