// @tempi/core-elements/dist/components/section/ui.es.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { BackgroundVideo } from '@tempi/core-renderer';
import React__default from 'react';
import { WidthTypeEnum } from '../../types/index.es.js';
import { PbContainer } from '../container/ui.es.js';
import { PbRow } from '../row/ui.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const PbSection = React__default.forwardRef(
  (props, ref) => {
    var _b;
    const _a = props, {
      children,
      commonStyle,
      verticalAlignment,
      widthType,
      row,
      gutter
    } = _a, other = __objRest(_a, [
      "children",
      "commonStyle",
      "verticalAlignment",
      "widthType",
      "row",
      "gutter"
    ]);
    const { maxWidth } = commonStyle;
    let rowStyle = row == null ? void 0 : row.commonStyle;
    rowStyle = __spreadValues({ minHeight: "inherit", height: "100%" }, rowStyle);
    if (rowStyle == null ? void 0 : rowStyle.maxWidth) {
      rowStyle = __spreadValues({
        marginLeft: "auto !important",
        marginRight: "auto !important"
      }, rowStyle);
    }
    return /* @__PURE__ */ jsxs(PbContainer, __spreadProps(__spreadValues({
      maxWidth,
      fluid: widthType === WidthTypeEnum.full,
      ref,
      commonStyle
    }, other), {
      children: [
        (commonStyle == null ? void 0 : commonStyle.backgroundVideo) && /* @__PURE__ */ jsx(BackgroundVideo, __spreadProps(__spreadValues({}, commonStyle.backgroundVideo), {
          background: (_b = commonStyle.background) == null ? void 0 : _b.toString()
        })),
        /* @__PURE__ */ jsx(PbRow, {
          commonStyle: rowStyle,
          verticalAlignment,
          gutter,
          children
        })
      ]
    }));
  }
);
PbSection.displayName = "Section";

export { PbSection };
