// @tempi/core-elements/dist/components/header/selector/MenuItem/MenuItemEditor.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var coreEditor = require('@tempi/core-editor');
var type = require('../../type.js');
var ButtonMenuItem = require('./ButtonMenuItem.js');
var LinkEditorQuickSetting = require('./LinkEditorQuickSetting.js');
var MenuItemQuickSetting = require('./MenuItemQuickSetting.js');
var MenuItem = require('../../ui/MenuItem/MenuItem.js');
var TipTapEditorV2 = require('../../../../configs/TipTapEditor/TipTapEditorV2.js');
var propKey = require('../../../../constants/propKey.js');
var ui = require('../../../heading/ui.js');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const MenuItemEditor = coreEditor.withSelector(
  (props) => {
    const { enabled } = coreEditor.useEditor((state) => ({
      enabled: state.options.enabled
    }));
    return /* @__PURE__ */ jsxRuntime.jsx(MenuItem.MenuItem, __spreadProps(__spreadValues({}, props), {
      enabled,
      children: props.type === type.MenuItemType.link ? /* @__PURE__ */ jsxRuntime.jsx(TipTapEditorV2.TipTapEditorV2, {
        propKey: propKey.PROP_KEY.joinKey(propKey.PROP_KEY.ItemData, propKey.PROP_KEY.Text),
        quickSetting: MenuItemQuickSetting.QuickSetting,
        editorQuickSetting: LinkEditorQuickSetting.LinkEditorQuickSetting,
        children: /* @__PURE__ */ jsxRuntime.jsx(ui.Heading, {
          commonStyle: props == null ? void 0 : props.commonStyle
        })
      }) : /* @__PURE__ */ jsxRuntime.jsx(ButtonMenuItem.ButtonMenuItem, __spreadValues({}, props))
    }));
  },
  {
    displayName: MenuItem.MenuItem.displayName,
    tag: MenuItem.MenuItem.displayName.toLowerCase(),
    advanceAttributes: coreEditor.AdvanceAttributes
  }
);

exports.MenuItemEditor = MenuItemEditor;
