// @tempi/core-elements/dist/components/header/selector/MenuItem/MenuItemQuickSetting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../../../i18n.es.js';
import { AntdCustomIcon } from '@tempi/icons-dev';
import { useProp, SubQuickSetting, QuickSettingButton } from '@tempi/core-editor';
import { ButtonQuickSetting } from '../../../button/v2/quick-setting.es.js';
import { HeadingQuickSetting } from '../../../heading/quick-setting.es.js';
import { MenuItemType } from '../../type.es.js';
import { MenuItemSetting } from './MenuItemSetting.es.js';
import { PROP_KEY } from '../../../../constants/propKey.es.js';

const QuickSetting = ({ editor, setIsEditable } = {}) => {
  const [menuItemType] = useProp(PROP_KEY.Type);
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(SubQuickSetting, {
        content: MenuItemSetting,
        children: /* @__PURE__ */ jsx(QuickSettingButton, {
          tooltip: t("T\xF9y ch\u1EC9nh Menu Item"),
          children: /* @__PURE__ */ jsx(AntdCustomIcon, {
            name: "HeaderOutlined",
            style: {
              fontSize: 16
            }
          })
        })
      }),
      menuItemType === MenuItemType.link ? /* @__PURE__ */ jsx(HeadingQuickSetting, {
        setIsEditable
      }) : /* @__PURE__ */ jsx(ButtonQuickSetting, {
        editor,
        propKey: PROP_KEY.ItemData,
        configs: { alignment: false }
      })
    ]
  });
};

export { QuickSetting };
