// @tempi/core-elements/dist/components/header/selector/MenuItem/MenuItemQuickSetting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../../../i18n.js');
var iconsDev = require('@tempi/icons-dev');
var coreEditor = require('@tempi/core-editor');
var quickSetting$1 = require('../../../button/v2/quick-setting.js');
var quickSetting = require('../../../heading/quick-setting.js');
var type = require('../../type.js');
var MenuItemSetting = require('./MenuItemSetting.js');
var propKey = require('../../../../constants/propKey.js');

const QuickSetting = ({ editor, setIsEditable } = {}) => {
  const [menuItemType] = coreEditor.useProp(propKey.PROP_KEY.Type);
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SubQuickSetting, {
        content: MenuItemSetting.MenuItemSetting,
        children: /* @__PURE__ */ jsxRuntime.jsx(coreEditor.QuickSettingButton, {
          tooltip: i18n.t("T\xF9y ch\u1EC9nh Menu Item"),
          children: /* @__PURE__ */ jsxRuntime.jsx(iconsDev.AntdCustomIcon, {
            name: "HeaderOutlined",
            style: {
              fontSize: 16
            }
          })
        })
      }),
      menuItemType === type.MenuItemType.link ? /* @__PURE__ */ jsxRuntime.jsx(quickSetting.HeadingQuickSetting, {
        setIsEditable
      }) : /* @__PURE__ */ jsxRuntime.jsx(quickSetting$1.ButtonQuickSetting, {
        editor,
        propKey: propKey.PROP_KEY.ItemData,
        configs: { alignment: false }
      })
    ]
  });
};

exports.QuickSetting = QuickSetting;
