// @tempi/core-editor/dist/configs/AdvanceAttributes/AdvanceAttributes.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var SpacingAttribute = require('./SpacingAttribute.js');
var ResponsiveAttribute = require('./ResponsiveAttribute.js');
var AttributeId = require('./AttributeId.js');
var antd = require('antd');
var TimingAttibute = require('./TimingAttibute.js');
var HoverConfig = require('../HoverConfig/HoverConfig.js');
var BuilderContext = require('../../contexts/BuilderContext.js');
var propKey = require('../../constants/propKey.js');

const { Panel } = antd.Collapse;
const AdvanceAttributes = ({
  propKey: propKey$1 = propKey.PROP_KEY.CommonStyle,
  separateDevice = false,
  readOnlyElementId
}) => {
  const { allowConfigResponsive } = BuilderContext.useBuilder();
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(AttributeId.AttributeId, {
        readOnlyElementId
      }),
      /* @__PURE__ */ jsxRuntime.jsx(antd.Divider, {
        style: {
          margin: "12px 0"
        }
      }),
      /* @__PURE__ */ jsxRuntime.jsxs(antd.Collapse, {
        defaultActiveKey: [
          "Spacing",
          "ResponsiveVisibility",
          "TimingVisibility"
        ],
        expandIconPosition: "end",
        bordered: false,
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(Panel, {
            header: i18n.t("Kho\u1EA3ng c\xE1ch"),
            children: /* @__PURE__ */ jsxRuntime.jsxs(antd.Space, {
              direction: "vertical",
              children: [
                /* @__PURE__ */ jsxRuntime.jsx(antd.Typography.Text, {
                  style: { marginBottom: 8 },
                  type: "warning",
                  children: i18n.t(
                    "Ch\xFA \xFD: Margin v\xE0 Padding qu\xE1 cao c\xF3 th\u1EC3 l\xE0m v\u1EE1 thi\u1EBFt k\u1EBF tr\xEAn \u0111i\u1EC7n tho\u1EA1i"
                  )
                }),
                /* @__PURE__ */ jsxRuntime.jsx(SpacingAttribute.SpacingAttribute, {
                  title: "Margin",
                  separateDevice,
                  propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.Margin)
                }),
                /* @__PURE__ */ jsxRuntime.jsx(SpacingAttribute.SpacingAttribute, {
                  title: "Padding",
                  separateDevice,
                  propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.Padding)
                })
              ]
            })
          }, "Spacing"),
          propKey$1 === propKey.PROP_KEY.CommonStyle && /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
            children: [
              allowConfigResponsive && /* @__PURE__ */ jsxRuntime.jsx(Panel, {
                header: i18n.t("\u1EA8n/Hi\u1EC3n th\u1ECB tr\xEAn thi\u1EBFt b\u1ECB"),
                children: /* @__PURE__ */ jsxRuntime.jsx(ResponsiveAttribute.ResponsiveAttribute, {})
              }, "ResponsiveVisibility"),
              /* @__PURE__ */ jsxRuntime.jsx(Panel, {
                header: i18n.t("\u0110\u1EB7t gi\u1EDD hi\u1EC3n th\u1ECB"),
                children: /* @__PURE__ */ jsxRuntime.jsx(TimingAttibute.TimingAttributes, {})
              }, "TimingVisibility")
            ]
          }),
          /* @__PURE__ */ jsxRuntime.jsx(Panel, {
            header: i18n.t("Hi\u1EC7u \u1EE9ng khi di chu\u1ED9t v\xE0o"),
            children: /* @__PURE__ */ jsxRuntime.jsx(HoverConfig.HoverConfig, {
              propKey: propKey$1,
              separateDevice
            })
          }, "HoverEffect")
        ]
      })
    ]
  });
};

exports.AdvanceAttributes = AdvanceAttributes;
