// @tempi/core-elements/dist/configs/storage-v2/StorageDataConfig.es.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { ChevronDown16Regular } from '@fluentui/react-icons';
import { Spin, Empty, List, Row, Dropdown, Button } from 'antd';
import { useState } from 'react';
import { useFrame } from 'react-frame-component';
import { FormAccountType, RequestMethod } from '../storage/type.es.js';
import { DataConfigRow } from './components/DataConfigRow.es.js';
import { EmailConfigModal } from './components/EmailConfigModal.es.js';
import { GGSheetConfigModal } from './components/GGSheetConfigModal.es.js';
import { ModalApiConfig } from '../ApiConfig/ModalApiConfig.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const StorageDataConfig = ({
  handleFetchFormAccount,
  loading,
  configStorage,
  handleSaveDataConfig,
  ggSheetsAccounts
}) => {
  var _a;
  const [isOpenModal, setOpenModal] = useState(null);
  const [activeIndex, setActiveIndex] = useState(-1);
  const configs = (configStorage == null ? void 0 : configStorage.configs) || [];
  const { document: frameDocument } = useFrame();
  const formAccountType = [
    { key: FormAccountType.EMAIL, label: t("L\u01B0u d\u1EEF li\u1EC7u v\u1EC1 Gmail") },
    {
      key: FormAccountType.GOOGLE_SHEET,
      label: t("L\u01B0u d\u1EEF li\u1EC7u v\u1EC1 Google Sheet")
    },
    { key: FormAccountType.API, label: t("L\u01B0u d\u1EEF li\u1EC7u qua API") }
  ];
  const onAddDataConfig = (type) => {
    const insertIndex = configs.length || 0;
    setOpenModal(type);
    setActiveIndex(insertIndex);
  };
  const onEditConfig = (index) => {
    var _a2;
    setActiveIndex(index);
    setOpenModal((_a2 = configs == null ? void 0 : configs[index]) == null ? void 0 : _a2.type);
  };
  const onDeleteConfig = async (index) => {
    const newConfigs = [...configs];
    newConfigs.splice(index, 1);
    await handleSaveDataConfig(newConfigs);
  };
  const onCloseModal = () => setOpenModal(null);
  const handleSaveModal = (value) => {
    if (activeIndex === configs.length) {
      handleSaveDataConfig([...configs, value]);
      return;
    }
    const newConfigs = [...configs];
    newConfigs[activeIndex] = value;
    handleSaveDataConfig(newConfigs);
  };
  return /* @__PURE__ */ jsxs(Spin, {
    spinning: loading,
    children: [
      !(configs == null ? void 0 : configs.length) ? /* @__PURE__ */ jsx(Empty, {
        description: t(
          "Th\xEAm k\u1EBFt n\u1ED1i \u0111\u1EC3 l\u01B0u d\u1EEF li\u1EC7u v\u1EC1 Gmail, Google Sheet ho\u1EB7c API"
        ),
        image: Empty.PRESENTED_IMAGE_DEFAULT
      }) : /* @__PURE__ */ jsx(List, {
        dataSource: configs,
        renderItem: (item, index) => /* @__PURE__ */ jsx(DataConfigRow, {
          config: item,
          onEdit: () => onEditConfig(index),
          onDelete: () => onDeleteConfig(index),
          editable: item.editable,
          deletable: item.deletable
        })
      }),
      /* @__PURE__ */ jsxs(Row, {
        justify: "center",
        className: "my-half",
        children: [
          /* @__PURE__ */ jsx(Dropdown, {
            menu: {
              items: formAccountType,
              onClick: ({ key }) => onAddDataConfig(key)
            },
            trigger: ["hover"],
            overlayStyle: { zIndex: 1081 },
            placement: "bottomLeft",
            disabled: (configs == null ? void 0 : configs.length) >= 3,
            children: /* @__PURE__ */ jsxs(Button, {
              type: "primary",
              className: "d-flex align-items-center",
              children: [
                t("Th\xEAm k\u1EBFt n\u1ED1i ("),
                (configs == null ? void 0 : configs.length) || 0,
                "/3)",
                /* @__PURE__ */ jsx(ChevronDown16Regular, {
                  className: "ml-half"
                })
              ]
            })
          }),
          /* @__PURE__ */ jsx(EmailConfigModal, {
            open: isOpenModal === FormAccountType.EMAIL,
            onClose: onCloseModal,
            loading,
            data: configs == null ? void 0 : configs[activeIndex],
            handleSaveModal
          }),
          /* @__PURE__ */ jsx(GGSheetConfigModal, {
            open: isOpenModal === FormAccountType.GOOGLE_SHEET,
            onClose: onCloseModal,
            loading,
            data: configs == null ? void 0 : configs[activeIndex],
            handleSaveModal,
            ggSheetsAccounts,
            handleFetchFormAccount,
            displayUrl: (ggSheetsAccounts == null ? void 0 : ggSheetsAccounts.length) > 1 || !!((_a = configs == null ? void 0 : configs[activeIndex]) == null ? void 0 : _a.formAccountId)
          }),
          /* @__PURE__ */ jsx(ModalApiConfig, {
            getContainer: () => (frameDocument == null ? void 0 : frameDocument.body) || (document == null ? void 0 : document.body),
            title: t("L\u01B0u d\u1EEF li\u1EC7u qua API"),
            data: !!(configs == null ? void 0 : configs[activeIndex]) && {
              endpoint: configs == null ? void 0 : configs[activeIndex].endpoint,
              method: (configs == null ? void 0 : configs[activeIndex].method) || RequestMethod.POST,
              body: configs == null ? void 0 : configs[activeIndex].body,
              headers: configs == null ? void 0 : configs[activeIndex].headers
            },
            onSubmit: async (value) => handleSaveModal(__spreadProps(__spreadValues({}, value), { type: FormAccountType.API })),
            isModalOpen: isOpenModal === FormAccountType.API,
            closeModal: onCloseModal,
            optionMethods: [
              {
                label: "GET",
                value: RequestMethod.GET
              },
              {
                label: "POST",
                value: RequestMethod.POST
              },
              {
                label: "PATCH",
                value: RequestMethod.PATCH
              }
            ]
          })
        ]
      })
    ]
  });
};

export { StorageDataConfig };
