// @tempi/core-elements/dist/configs/storage-v2/StorageDataConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var reactIcons = require('@fluentui/react-icons');
var antd = require('antd');
var React = require('react');
var reactFrameComponent = require('react-frame-component');
var type = require('../storage/type.js');
var DataConfigRow = require('./components/DataConfigRow.js');
var EmailConfigModal = require('./components/EmailConfigModal.js');
var GGSheetConfigModal = require('./components/GGSheetConfigModal.js');
var ModalApiConfig = require('../ApiConfig/ModalApiConfig.js');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const StorageDataConfig = ({
  handleFetchFormAccount,
  loading,
  configStorage,
  handleSaveDataConfig,
  ggSheetsAccounts
}) => {
  var _a;
  const [isOpenModal, setOpenModal] = React.useState(null);
  const [activeIndex, setActiveIndex] = React.useState(-1);
  const configs = (configStorage == null ? void 0 : configStorage.configs) || [];
  const { document: frameDocument } = reactFrameComponent.useFrame();
  const formAccountType = [
    { key: type.FormAccountType.EMAIL, label: i18n.t("L\u01B0u d\u1EEF li\u1EC7u v\u1EC1 Gmail") },
    {
      key: type.FormAccountType.GOOGLE_SHEET,
      label: i18n.t("L\u01B0u d\u1EEF li\u1EC7u v\u1EC1 Google Sheet")
    },
    { key: type.FormAccountType.API, label: i18n.t("L\u01B0u d\u1EEF li\u1EC7u qua API") }
  ];
  const onAddDataConfig = (type) => {
    const insertIndex = configs.length || 0;
    setOpenModal(type);
    setActiveIndex(insertIndex);
  };
  const onEditConfig = (index) => {
    var _a2;
    setActiveIndex(index);
    setOpenModal((_a2 = configs == null ? void 0 : configs[index]) == null ? void 0 : _a2.type);
  };
  const onDeleteConfig = async (index) => {
    const newConfigs = [...configs];
    newConfigs.splice(index, 1);
    await handleSaveDataConfig(newConfigs);
  };
  const onCloseModal = () => setOpenModal(null);
  const handleSaveModal = (value) => {
    if (activeIndex === configs.length) {
      handleSaveDataConfig([...configs, value]);
      return;
    }
    const newConfigs = [...configs];
    newConfigs[activeIndex] = value;
    handleSaveDataConfig(newConfigs);
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(antd.Spin, {
    spinning: loading,
    children: [
      !(configs == null ? void 0 : configs.length) ? /* @__PURE__ */ jsxRuntime.jsx(antd.Empty, {
        description: i18n.t(
          "Th\xEAm k\u1EBFt n\u1ED1i \u0111\u1EC3 l\u01B0u d\u1EEF li\u1EC7u v\u1EC1 Gmail, Google Sheet ho\u1EB7c API"
        ),
        image: antd.Empty.PRESENTED_IMAGE_DEFAULT
      }) : /* @__PURE__ */ jsxRuntime.jsx(antd.List, {
        dataSource: configs,
        renderItem: (item, index) => /* @__PURE__ */ jsxRuntime.jsx(DataConfigRow.DataConfigRow, {
          config: item,
          onEdit: () => onEditConfig(index),
          onDelete: () => onDeleteConfig(index),
          editable: item.editable,
          deletable: item.deletable
        })
      }),
      /* @__PURE__ */ jsxRuntime.jsxs(antd.Row, {
        justify: "center",
        className: "my-half",
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(antd.Dropdown, {
            menu: {
              items: formAccountType,
              onClick: ({ key }) => onAddDataConfig(key)
            },
            trigger: ["hover"],
            overlayStyle: { zIndex: 1081 },
            placement: "bottomLeft",
            disabled: (configs == null ? void 0 : configs.length) >= 3,
            children: /* @__PURE__ */ jsxRuntime.jsxs(antd.Button, {
              type: "primary",
              className: "d-flex align-items-center",
              children: [
                i18n.t("Th\xEAm k\u1EBFt n\u1ED1i ("),
                (configs == null ? void 0 : configs.length) || 0,
                "/3)",
                /* @__PURE__ */ jsxRuntime.jsx(reactIcons.ChevronDown16Regular, {
                  className: "ml-half"
                })
              ]
            })
          }),
          /* @__PURE__ */ jsxRuntime.jsx(EmailConfigModal.EmailConfigModal, {
            open: isOpenModal === type.FormAccountType.EMAIL,
            onClose: onCloseModal,
            loading,
            data: configs == null ? void 0 : configs[activeIndex],
            handleSaveModal
          }),
          /* @__PURE__ */ jsxRuntime.jsx(GGSheetConfigModal.GGSheetConfigModal, {
            open: isOpenModal === type.FormAccountType.GOOGLE_SHEET,
            onClose: onCloseModal,
            loading,
            data: configs == null ? void 0 : configs[activeIndex],
            handleSaveModal,
            ggSheetsAccounts,
            handleFetchFormAccount,
            displayUrl: (ggSheetsAccounts == null ? void 0 : ggSheetsAccounts.length) > 1 || !!((_a = configs == null ? void 0 : configs[activeIndex]) == null ? void 0 : _a.formAccountId)
          }),
          /* @__PURE__ */ jsxRuntime.jsx(ModalApiConfig.ModalApiConfig, {
            getContainer: () => (frameDocument == null ? void 0 : frameDocument.body) || (document == null ? void 0 : document.body),
            title: i18n.t("L\u01B0u d\u1EEF li\u1EC7u qua API"),
            data: !!(configs == null ? void 0 : configs[activeIndex]) && {
              endpoint: configs == null ? void 0 : configs[activeIndex].endpoint,
              method: (configs == null ? void 0 : configs[activeIndex].method) || type.RequestMethod.POST,
              body: configs == null ? void 0 : configs[activeIndex].body,
              headers: configs == null ? void 0 : configs[activeIndex].headers
            },
            onSubmit: async (value) => handleSaveModal(__spreadProps(__spreadValues({}, value), { type: type.FormAccountType.API })),
            isModalOpen: isOpenModal === type.FormAccountType.API,
            closeModal: onCloseModal,
            optionMethods: [
              {
                label: "GET",
                value: type.RequestMethod.GET
              },
              {
                label: "POST",
                value: type.RequestMethod.POST
              },
              {
                label: "PATCH",
                value: type.RequestMethod.PATCH
              }
            ]
          })
        ]
      })
    ]
  });
};

exports.StorageDataConfig = StorageDataConfig;
