// @tempi/core-editor/dist/components/ModalConfigWrapper/ModalConfigWrapper.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var React = require('react');
var styled = require('@emotion/styled');
var antd = require('antd');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const ModalConfigWrapper = ({
  title = i18n.t("C\u1EA5u h\xECnh s\u1EF1 ki\u1EC7n"),
  actionText = i18n.t("C\u1EA5u h\xECnh"),
  customModal: CustomModal,
  subTitle,
  customModalProps
}) => {
  const [visible, setVisible] = React.useState(false);
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsxs(Wrapper, {
        children: [
          /* @__PURE__ */ jsxRuntime.jsxs("div", {
            children: [
              /* @__PURE__ */ jsxRuntime.jsx(antd.Typography.Title, {
                level: 5,
                style: { fontSize: 14, fontWeight: 500 },
                children: title
              }),
              !!subTitle && /* @__PURE__ */ jsxRuntime.jsx(antd.Typography.Paragraph, {
                type: "secondary",
                className: "mb-0",
                children: subTitle
              })
            ]
          }),
          /* @__PURE__ */ jsxRuntime.jsx(ActionButton, {
            type: "link",
            onClick: () => setVisible((prevState) => !prevState),
            children: actionText
          })
        ]
      }),
      /* @__PURE__ */ jsxRuntime.jsx(CustomModal, __spreadValues({}, __spreadProps(__spreadValues({}, customModalProps), { visible, setVisible })))
    ]
  });
};
const Wrapper = styled__default["default"].div`
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  margin-bottom: 8px;
`;
const ActionButton = styled__default["default"](antd.Button)`
  padding: 0 !important;
  height: unset !important;
`;

exports.ModalConfigWrapper = ModalConfigWrapper;
