// @tempi/dnd-utils/dist/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var constants = require('./constants.js');
var useMethods = require('./useMethods.js');
var getDOMInfo = require('./getDOMInfo.js');
var useCollector = require('./useCollector.js');
var RenderIndicator = require('./RenderIndicator.js');
var useEffectOnce = require('./useEffectOnce.js');
var deprecate = require('./deprecate.js');
var History = require('./History.js');
var getRandomId = require('./getRandomId.js');
var platform = require('./platform.js');
var EventHandlers = require('./EventHandlers/EventHandlers.js');
var DerivedEventHandlers = require('./EventHandlers/DerivedEventHandlers.js');
var wrapConnectorHooks = require('./EventHandlers/wrapConnectorHooks.js');
var interfaces = require('./EventHandlers/interfaces.js');



exports.DEPRECATED_ROOT_NODE = constants.DEPRECATED_ROOT_NODE;
exports.ERROR_CANNOT_DRAG = constants.ERROR_CANNOT_DRAG;
exports.ERROR_DELETE_TOP_LEVEL_NODE = constants.ERROR_DELETE_TOP_LEVEL_NODE;
exports.ERROR_DESERIALIZE_COMPONENT_NOT_IN_RESOLVER = constants.ERROR_DESERIALIZE_COMPONENT_NOT_IN_RESOLVER;
exports.ERROR_DUPLICATE_NODEID = constants.ERROR_DUPLICATE_NODEID;
exports.ERROR_INFINITE_CANVAS = constants.ERROR_INFINITE_CANVAS;
exports.ERROR_INVALID_NODEID = constants.ERROR_INVALID_NODEID;
exports.ERROR_INVALID_NODE_ID = constants.ERROR_INVALID_NODE_ID;
exports.ERROR_MISSING_PLACEHOLDER_PLACEMENT = constants.ERROR_MISSING_PLACEHOLDER_PLACEMENT;
exports.ERROR_MOVE_CANNOT_DROP = constants.ERROR_MOVE_CANNOT_DROP;
exports.ERROR_MOVE_INCOMING_PARENT = constants.ERROR_MOVE_INCOMING_PARENT;
exports.ERROR_MOVE_NONCANVAS_CHILD = constants.ERROR_MOVE_NONCANVAS_CHILD;
exports.ERROR_MOVE_OUTGOING_PARENT = constants.ERROR_MOVE_OUTGOING_PARENT;
exports.ERROR_MOVE_ROOT_NODE = constants.ERROR_MOVE_ROOT_NODE;
exports.ERROR_MOVE_TOP_LEVEL_NODE = constants.ERROR_MOVE_TOP_LEVEL_NODE;
exports.ERROR_MOVE_TO_DESCENDANT = constants.ERROR_MOVE_TO_DESCENDANT;
exports.ERROR_MOVE_TO_NONCANVAS_PARENT = constants.ERROR_MOVE_TO_NONCANVAS_PARENT;
exports.ERROR_NOPARENT = constants.ERROR_NOPARENT;
exports.ERROR_NOT_IN_RESOLVER = constants.ERROR_NOT_IN_RESOLVER;
exports.ERROR_RESOLVER_NOT_AN_OBJECT = constants.ERROR_RESOLVER_NOT_AN_OBJECT;
exports.ERROR_TOP_LEVEL_ELEMENT_NO_ID = constants.ERROR_TOP_LEVEL_ELEMENT_NO_ID;
exports.ERROR_USE_EDITOR_OUTSIDE_OF_EDITOR_CONTEXT = constants.ERROR_USE_EDITOR_OUTSIDE_OF_EDITOR_CONTEXT;
exports.ERROR_USE_NODE_OUTSIDE_OF_EDITOR_CONTEXT = constants.ERROR_USE_NODE_OUTSIDE_OF_EDITOR_CONTEXT;
exports.ROOT_NODE = constants.ROOT_NODE;
exports.createQuery = useMethods.createQuery;
exports.useMethods = useMethods.useMethods;
exports.getDOMInfo = getDOMInfo.getDOMInfo;
exports.useCollector = useCollector.useCollector;
exports.RenderIndicator = RenderIndicator.RenderIndicator;
exports.useEffectOnce = useEffectOnce.useEffectOnce;
exports.deprecationWarning = deprecate.deprecationWarning;
exports.HISTORY_ACTIONS = History.HISTORY_ACTIONS;
exports.History = History.History;
exports.getRandomId = getRandomId.getRandomId;
exports.isChromium = platform.isChromium;
exports.isClientSide = platform.isClientSide;
exports.isLinux = platform.isLinux;
exports.EventHandlers = EventHandlers.EventHandlers;
exports.DerivedEventHandlers = DerivedEventHandlers.DerivedEventHandlers;
exports.cloneWithRef = wrapConnectorHooks.cloneWithRef;
exports.wrapConnectorHooks = wrapConnectorHooks.wrapConnectorHooks;
exports.wrapHookToRecognizeElement = wrapConnectorHooks.wrapHookToRecognizeElement;
exports.EventHandlerUpdates = interfaces.EventHandlerUpdates;
