// @tempi/core-editor/dist/configs/ImageListConfig/ImageListConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var core = require('@dnd-kit/core');
var sortable = require('@dnd-kit/sortable');
var styled = require('@emotion/styled');
var antd = require('antd');
var React = require('react');
var reactFrameComponent = require('react-frame-component');
var SortableImage = require('./SortableImage.js');
var useSortable = require('../../hooks/useSortable.js');
var ImageUploadButton = require('../ImageConfig/ImageUploadButton.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const ImageListConfig = ({
  propKey = "items",
  separateDevice,
  visible,
  setVisible,
  imageHeight,
  actions,
  defaultNewImage
}) => {
  var _a;
  const [items, setItems] = useSortable.useSortable(propKey, separateDevice);
  const [activeId, setActiveId] = React.useState(null);
  const activeIndex = activeId ? items.findIndex((e) => e.id === activeId) : -1;
  const { document: frameDocument } = reactFrameComponent.useFrame();
  const onClose = () => setVisible(false);
  const handleRemove = (index) => {
    setItems((items2) => {
      items2 == null ? void 0 : items2.splice(index, 1);
      return items2;
    });
  };
  const onDragStart = ({ active }) => {
    if (!active) {
      return;
    }
    setActiveId(active.id);
  };
  const onDragEnd = ({ over }) => {
    setActiveId(null);
    if (over) {
      const overIndex = items.findIndex((e) => e.id === over.id);
      if (activeIndex !== overIndex) {
        setItems((items2) => sortable.arrayMove(items2, activeIndex, overIndex));
      }
    }
  };
  const renderImageList = /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsxs(StyledDiv, {
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(antd.Typography.Text, {
            type: "secondary",
            children: i18n.t("Click 2 l\u1EA7n v\xE0o \u1EA3nh \u0111\u1EC3 thay \u1EA3nh m\u1EDBi")
          }),
          /* @__PURE__ */ jsxRuntime.jsx(ImageUploadButton.ImageUploadButton, {
            separateDevice,
            imagePropKey: propKey,
            multiple: true,
            listType: "picture",
            defaultNewImage
          })
        ]
      }),
      /* @__PURE__ */ jsxRuntime.jsx(core.DndContext, {
        onDragStart,
        onDragEnd,
        onDragCancel: () => setActiveId(null),
        children: /* @__PURE__ */ jsxRuntime.jsx(sortable.SortableContext, {
          items,
          children: /* @__PURE__ */ jsxRuntime.jsx(List, {
            children: (_a = items == null ? void 0 : items.map) == null ? void 0 : _a.call(items, (item, index) => /* @__PURE__ */ jsxRuntime.jsx(SortableImage.SortableImage, __spreadProps(__spreadValues({
              actions,
              propKey,
              onRemove: () => handleRemove(index),
              index,
              id: item.id
            }, item.item), {
              imageHeight,
              separateDevice
            }), index))
          })
        })
      })
    ]
  });
  return /* @__PURE__ */ jsxRuntime.jsxs(antd.Modal, {
    width: "70%",
    bodyStyle: {
      minHeight: 500
    },
    zIndex: 1080,
    title: i18n.t("Qu\u1EA3n l\xFD \u1EA3nh"),
    okText: i18n.t("\u0110\xF3ng"),
    destroyOnClose: true,
    onCancel: onClose,
    open: visible,
    footer: /* @__PURE__ */ jsxRuntime.jsx(jsxRuntime.Fragment, {
      children: /* @__PURE__ */ jsxRuntime.jsx(antd.Button, {
        type: "primary",
        onClick: onClose,
        children: i18n.t("\u0110\xF3ng")
      })
    }),
    getContainer: (frameDocument == null ? void 0 : frameDocument.body) || (document == null ? void 0 : document.body),
    children: [
      (items == null ? void 0 : items.length) ? renderImageList : null,
      /* @__PURE__ */ jsxRuntime.jsx(antd.Empty, {
        description: i18n.t("Ch\u01B0a c\xF3 h\xECnh \u1EA3nh n\xE0o"),
        style: {
          paddingTop: "10%",
          display: (items == null ? void 0 : items.length) ? "none" : "block"
        },
        children: /* @__PURE__ */ jsxRuntime.jsx(ImageUploadButton.ImageUploadButton, {
          separateDevice,
          imagePropKey: propKey,
          multiple: true,
          listType: "picture",
          defaultNewImage
        })
      })
    ]
  });
};
const List = styled__default["default"].ul`
  display: grid;
  gap: 0.75rem;
  padding: 0;

  transition: all 2s;

  /* Desktop */
  grid-template-columns: repeat(4, 1fr);
  /* Tablet */
  @media only screen and (max-width: 800px) {
    grid-template-columns: repeat(2, 1fr);
  }
  /* Mobile */
  @media only screen and (max-width: 460px) {
    grid-template-columns: repeat(1, 1fr);
  }
`;
const StyledDiv = styled__default["default"].div`
  display: flex;
  justify-content: space-between;
  margin-top: 1rem;
  margin-bottom: 1rem;
`;

exports.ImageListConfig = ImageListConfig;
