// @tempi/core-editor/dist/configs/AnimationConfig/AnimationConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var antd = require('antd');
var RangeSliderConfig = require('../RangeSliderConfig/RangeSliderConfig.js');
var usePropSeparateDevice = require('../../hooks/usePropSeparateDevice.js');
var propKey = require('../../constants/propKey.js');
var useProp = require('../../hooks/useProp.js');
var SelectConfig = require('../SelectConfig/SelectConfig.js');
var SwitchConfig = require('../SwitchConfig/SwitchConfig.js');

const animationNameOptions = [
  {
    label: "Fading entrances",
    options: [
      "fadeIn",
      "fadeInUp",
      "fadeInUpBig",
      "fadeInDown",
      "fadeInLeft",
      "fadeInRight",
      "fadeInTopLeft",
      "fadeInTopRight",
      "fadeInBottomLeft",
      "fadeInBottomRight"
    ]
  },
  {
    label: "Sliding entrances",
    options: ["slideInUp", "slideInDown", "slideInLeft", "slideInRight"]
  },
  {
    label: "Rotating entrances",
    options: [
      "rotateIn",
      "rotateInUpLeft",
      "rotateInUpRight",
      "rotateInDownLeft",
      "rotateInDownRight"
    ]
  },
  {
    label: "Bouncing entrances",
    options: [
      "bounceIn",
      "bounceInUp",
      "bounceInDown",
      "bounceInLeft",
      "bounceInRight"
    ]
  },
  {
    label: "Zooming entrances",
    options: ["zoomIn", "zoomInUp", "zoomInDown", "zoomInLeft", "zoomInRight"]
  },
  {
    label: "Lightspeed",
    options: [
      "lightSpeedInRight",
      "lightSpeedInLeft",
      "lightSpeedOutRight",
      "lightSpeedOutLeft"
    ]
  },
  {
    label: "Attention seekers",
    options: [
      "bounce",
      "flash",
      "pulse",
      "rubberBand",
      "shakeX",
      "shakeY",
      "headShake",
      "swing",
      "tada",
      "wobble",
      "jello",
      "heartBeat"
    ]
  },
  {
    label: "Specials",
    options: ["hinge", "jackInTheBox", "rollIn"]
  }
];
const AnimationConfig = ({
  propKey: propKey$1 = propKey.PROP_KEY.CommonStyle,
  separateDevice
}) => {
  const { Option, OptGroup } = antd.Select;
  const [repeatType, setRepeatType] = usePropSeparateDevice.usePropSeparateDevice(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.AnimationRepeatType),
    separateDevice
  );
  const [animation, _] = useProp.useProp(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.AnimationName)
  );
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(SelectConfig.SelectConfig, {
        separateDevice,
        propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.AnimationName),
        formItemProps: {
          label: i18n.t("Lo\u1EA1i hi\u1EC7u \u1EE9ng"),
          placeholder: i18n.t("Ch\u1ECDn hi\u1EC7u \u1EE9ng"),
          layout: "vertical"
        },
        selectProps: {
          style: { width: "100%" },
          children: /* @__PURE__ */ jsxRuntime.jsx(jsxRuntime.Fragment, {
            children: animationNameOptions.map((opt) => /* @__PURE__ */ jsxRuntime.jsx(OptGroup, {
              label: opt.label,
              children: opt.options.map((name) => /* @__PURE__ */ jsxRuntime.jsx(Option, {
                value: name,
                children: name[0].toUpperCase() + name.slice(1).split(/(?=[A-Z])/).join(" ")
              }, name))
            }, opt.label))
          })
        }
      }),
      animation && /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(RangeSliderConfig.RangeSliderConfig, {
            separateDevice,
            propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.AnimationDuration),
            formItemProps: {
              label: i18n.t("Kho\u1EA3ng th\u1EDDi gian di\u1EC5n ra hi\u1EC7u \u1EE9ng (s)")
            },
            sliderProps: {
              defaultValue: 1,
              min: 0,
              max: 5
            },
            unitProps: {
              defaultValue: "s"
            }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(RangeSliderConfig.RangeSliderConfig, {
            separateDevice,
            propKey: propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.AnimationDelay),
            formItemProps: {
              label: i18n.t("\u0110\u1ED9 tr\u1EC5 xu\u1EA5t hi\u1EC7n hi\u1EC7u \u1EE9ng (s)")
            },
            sliderProps: {
              defaultValue: 1,
              min: 0,
              max: 5
            },
            unitProps: {
              defaultValue: "s"
            }
          }),
          /* @__PURE__ */ jsxRuntime.jsx(SwitchConfig.SwitchConfig, {
            formItemProps: { label: i18n.t("L\u1EB7p l\u1EA1i hi\u1EC7u \u1EE9ng") },
            switchProps: {
              checked: repeatType === "infinite",
              onChange: (checked) => {
                if (checked)
                  setRepeatType("infinite");
                else
                  setRepeatType(void 0);
              }
            }
          })
        ]
      })
    ]
  });
};

exports.AnimationConfig = AnimationConfig;
