// @tempi/core-editor/dist/configs/TextConfig/TextConfig.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var reactIcons = require('@fluentui/react-icons');
var antd = require('antd');
var usePropSeparateDevice = require('../../hooks/usePropSeparateDevice.js');
var propKey = require('../../constants/propKey.js');
var FormItem = require('../../components/FormItem/FormItem.js');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const TextConfig = (props) => {
  const { propKey: propKey$1, separateDevice, formItemProps } = props;
  const {
    token: { colorPrimary }
  } = antd.theme.useToken();
  const [fontWeight, setFontWeight] = usePropSeparateDevice.usePropSeparateDevice(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.FontWeight),
    separateDevice
  );
  const [fontStyle, setFontStyle] = usePropSeparateDevice.usePropSeparateDevice(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.FontStyle),
    separateDevice
  );
  const [textDecoration, setTextDecoration] = usePropSeparateDevice.usePropSeparateDevice(
    propKey.PROP_KEY.joinKey(propKey$1, propKey.PROP_KEY.TextDecoration),
    separateDevice
  );
  const onChangeFontWeight = () => {
    if (!fontWeight) {
      setFontWeight("bold");
      return;
    }
    setFontWeight(void 0);
  };
  const onChangeFontStyle = () => {
    if (!fontStyle) {
      setFontStyle("italic");
      return;
    }
    setFontStyle(void 0);
  };
  const onChangeTextDecoration = () => {
    if (!textDecoration) {
      setTextDecoration("underline");
      return;
    }
    setTextDecoration(void 0);
  };
  return /* @__PURE__ */ jsxRuntime.jsx(FormItem.FormItem, __spreadProps(__spreadValues({}, formItemProps), {
    children: /* @__PURE__ */ jsxRuntime.jsxs(antd.Space, {
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(antd.Button, {
          style: { color: fontWeight ? colorPrimary : void 0 },
          type: "text",
          icon: /* @__PURE__ */ jsxRuntime.jsx(reactIcons.TextBold16Regular, {}),
          onClick: onChangeFontWeight
        }),
        /* @__PURE__ */ jsxRuntime.jsx(antd.Button, {
          style: { color: fontStyle ? colorPrimary : void 0 },
          type: "text",
          icon: /* @__PURE__ */ jsxRuntime.jsx(reactIcons.TextItalic16Regular, {}),
          onClick: onChangeFontStyle
        }),
        /* @__PURE__ */ jsxRuntime.jsx(antd.Button, {
          style: { color: textDecoration ? colorPrimary : void 0 },
          type: "text",
          icon: /* @__PURE__ */ jsxRuntime.jsx(reactIcons.TextUnderline16Regular, {}),
          onClick: onChangeTextDecoration
        })
      ]
    })
  }));
};

exports.TextConfig = TextConfig;
