// @tempi/core-editor/dist/configs/InputSelectConfig/InputSelectConfig.es.js
import { jsx } from 'react/jsx-runtime';
import styled from '@emotion/styled';
import { ChevronDown16Regular } from '@fluentui/react-icons';
import { Input, Dropdown } from 'antd';
import { useState, useEffect } from 'react';
import { usePropSeparateDevice } from '../../hooks/usePropSeparateDevice.es.js';
import { FormItem } from '../../components/FormItem/FormItem.es.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const InputSelectConfig = ({
  propKey,
  formItemProps,
  options,
  value: defaultValue,
  onChange,
  width = 100,
  inputProps
}) => {
  const [prop, setProp] = usePropSeparateDevice(propKey);
  const [value, setValue] = useState(defaultValue);
  const onChangeValue = (value2) => {
    if (value2 === (defaultValue || prop))
      return;
    if (typeof onChange === "function") {
      onChange(value2);
      return;
    }
    setProp(value2);
  };
  useEffect(() => {
    if (typeof defaultValue !== "undefined")
      setValue(defaultValue);
    else
      setValue(prop);
  }, [defaultValue, prop]);
  return /* @__PURE__ */ jsx(FormItem, __spreadProps(__spreadValues({}, formItemProps), {
    children: /* @__PURE__ */ jsx(StyledDropdown, {
      menu: {
        items: options,
        selectedKeys: [value == null ? void 0 : value.toString()],
        onClick: ({ key }) => onChangeValue(key)
      },
      trigger: ["click"],
      children: /* @__PURE__ */ jsx(StyledWrapper, {
        width,
        children: /* @__PURE__ */ jsx(Input, __spreadValues({
          value,
          bordered: false,
          suffix: /* @__PURE__ */ jsx(ChevronDown16Regular, {}),
          onChange: (e) => setValue(e.target.value),
          onPressEnter: () => onChangeValue(value),
          onBlur: () => onChangeValue(value)
        }, inputProps))
      })
    })
  }));
};
const StyledWrapper = styled.div(
  ({ width, theme }) => `
  width: ${typeof width === "number" ? width + "px" : width};
  position: relative;

  &:hover {
    cursor: pointer;

    &::before {
      content: '';
      background: ${theme == null ? void 0 : theme.colorPrimaryBgHover};
      position: absolute;
      width: 100%;
      height: 100%;
      top: 0;
      left: 0;
      border-radius: 8px;
    }

    * {
      color: ${theme == null ? void 0 : theme.colorPrimaryHover};
    }
  }

  input::-webkit-outer-spin-button,
  input::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }
  
  input[type=number] {
    -moz-appearance: textfield;
  }
`
);
const StyledDropdown = styled(Dropdown)`
  min-width: unset !important;
`;

export { InputSelectConfig };
