// @tempi/core-elements/dist/core/RenderFloatingQuickSetting/RenderFloatingQuickSetting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import styled from '@emotion/styled';
import { useEditor, useNode } from '@tempi/core-editor';

const RenderFloatingQuickSetting = ({ quickSetting, children }) => {
  const {
    query: { node },
    hoveredNodes,
    selectedNodes
  } = useEditor((state) => ({
    hoveredNodes: state.events.hovered,
    selectedNodes: state.events.selected
  }));
  const { allDescendants, isActive, isHover } = useNode(({ events, id }) => ({
    allDescendants: node(id).descendants(true),
    isActive: events.selected,
    isHover: events.hovered
  }));
  const showQuickSetting = () => {
    if (allDescendants.some((el) => selectedNodes.has(el)))
      return false;
    const hoveredChilds = allDescendants.some((el) => hoveredNodes.has(el));
    return hoveredChilds || isHover || isActive;
  };
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      showQuickSetting() && /* @__PURE__ */ jsx("div", {
        style: { position: "relative" },
        children: /* @__PURE__ */ jsx(Wrapper, {
          children: /* @__PURE__ */ jsx(QuickSettingWrapper, {
            children: quickSetting
          })
        })
      }),
      children
    ]
  });
};
const QuickSettingWrapper = styled.div`
  height: 40px;
  background: ${(props) => props.theme.colorBgBase};
  box-shadow: ${(props) => props.theme.boxShadowSecondary};
  border-radius: 999px;
  display: flex;
  align-items: center;
`;
const Wrapper = styled.div`
  position: absolute;
  top: 4px;
  right: 24px;
  z-index: 1000;
`;

export { RenderFloatingQuickSetting };
