// @tempi/core-elements/dist/components/qr-code/selector.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var coreEditor = require('@tempi/core-editor');
var constants = require('./constants.js');
var quickSetting = require('./quick-setting.js');
var setting = require('./setting.js');
var styleSetting = require('./style-setting.js');
var ui = require('./ui.js');
var selector = require('../section/selector.js');
var selector$1 = require('../root/selector.js');

const QRCodeEditor = coreEditor.withSelector(ui.QRCode, {
  displayName: "QR code",
  tag: ui.QRCode.displayName.toLowerCase(),
  quickSetting: quickSetting.QuickSetting,
  advanceAttributes: coreEditor.AdvanceAttributes,
  commonAttributes: styleSetting.StyleSetting,
  customAttributes: setting.Setting,
  props: constants.DEFAULT_QR_CODE,
  rules: {
    canDrop: (targetNode) => {
      const ignoreTargets = [
        coreEditor.getEditorName({ RowEditor: selector.RowEditor }),
        coreEditor.getEditorName({ RootEditor: selector$1.RootEditor })
      ];
      return !ignoreTargets.includes(targetNode.data.name);
    }
  }
});

exports.QRCodeEditor = QRCodeEditor;
