// @tempi/core-elements/dist/components/qr-code/setting.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var i18n = require('../../i18n.js');
var coreEditor = require('@tempi/core-editor');
var propKey = require('../../constants/propKey.js');

const Setting = () => {
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, {
    children: [
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.InputConfig, {
        formItemProps: { label: i18n.t("\u0110\u01B0\u1EDDng d\u1EABn"), layout: "vertical" },
        propKey: propKey.PROP_KEY.Value
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.RangeSliderConfig, {
        sliderProps: { min: 100, max: 500, step: 1 },
        propKey: propKey.PROP_KEY.Size,
        formItemProps: { label: i18n.t("K\xEDch th\u01B0\u1EDBc") }
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.SwitchConfig, {
        formItemProps: { label: i18n.t("Vi\u1EC1n") },
        propKey: propKey.PROP_KEY.Bordered
      }),
      /* @__PURE__ */ jsxRuntime.jsx(coreEditor.AlignmentConfig, {
        formItemProps: {
          label: i18n.t("C\u0103n ch\u1EC9nh")
        },
        buttonGroupProps: {
          listAlignment: ["left", "center", "right"]
        },
        propKey: propKey.PROP_KEY.Alignment
      })
    ]
  });
};

exports.Setting = Setting;
