// @tempi/core-elements/dist/components/qr-code/setting.es.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { t } from '../../i18n.es.js';
import { InputConfig, RangeSliderConfig, SwitchConfig, AlignmentConfig } from '@tempi/core-editor';
import { PROP_KEY } from '../../constants/propKey.es.js';

const Setting = () => {
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      /* @__PURE__ */ jsx(InputConfig, {
        formItemProps: { label: t("\u0110\u01B0\u1EDDng d\u1EABn"), layout: "vertical" },
        propKey: PROP_KEY.Value
      }),
      /* @__PURE__ */ jsx(RangeSliderConfig, {
        sliderProps: { min: 100, max: 500, step: 1 },
        propKey: PROP_KEY.Size,
        formItemProps: { label: t("K\xEDch th\u01B0\u1EDBc") }
      }),
      /* @__PURE__ */ jsx(SwitchConfig, {
        formItemProps: { label: t("Vi\u1EC1n") },
        propKey: PROP_KEY.Bordered
      }),
      /* @__PURE__ */ jsx(AlignmentConfig, {
        formItemProps: {
          label: t("C\u0103n ch\u1EC9nh")
        },
        buttonGroupProps: {
          listAlignment: ["left", "center", "right"]
        },
        propKey: PROP_KEY.Alignment
      })
    ]
  });
};

export { Setting };
